(function(){const t=document.createElement("link").relList;if(t&&t.supports&&t.supports("modulepreload"))return;for(const l of document.querySelectorAll('link[rel="modulepreload"]'))s(l);new MutationObserver(l=>{for(const u of l)if(u.type==="childList")for(const p of u.addedNodes)p.tagName==="LINK"&&p.rel==="modulepreload"&&s(p)}).observe(document,{childList:!0,subtree:!0});function r(l){const u={};return l.integrity&&(u.integrity=l.integrity),l.referrerPolicy&&(u.referrerPolicy=l.referrerPolicy),l.crossOrigin==="use-credentials"?u.credentials="include":l.crossOrigin==="anonymous"?u.credentials="omit":u.credentials="same-origin",u}function s(l){if(l.ep)return;l.ep=!0;const u=r(l);fetch(l.href,u)}})();function KR(i){return i&&i.__esModule&&Object.prototype.hasOwnProperty.call(i,"default")?i.default:i}var Au={exports:{}},ls={};/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var Bg;function ZR(){if(Bg)return ls;Bg=1;var i=Symbol.for("react.transitional.element"),t=Symbol.for("react.fragment");function r(s,l,u){var p=null;if(u!==void 0&&(p=""+u),l.key!==void 0&&(p=""+l.key),"key"in l){u={};for(var h in l)h!=="key"&&(u[h]=l[h])}else u=l;return l=u.ref,{$$typeof:i,type:s,key:p,ref:l!==void 0?l:null,props:u}}return ls.Fragment=t,ls.jsx=r,ls.jsxs=r,ls}var Mg;function JR(){return Mg||(Mg=1,Au.exports=ZR()),Au.exports}var I=JR(),Lu={exports:{}},ue={};/**
 * @license React
 * react.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var qg;function $R(){if(qg)return ue;qg=1;var i=Symbol.for("react.transitional.element"),t=Symbol.for("react.portal"),r=Symbol.for("react.fragment"),s=Symbol.for("react.strict_mode"),l=Symbol.for("react.profiler"),u=Symbol.for("react.consumer"),p=Symbol.for("react.context"),h=Symbol.for("react.forward_ref"),f=Symbol.for("react.suspense"),g=Symbol.for("react.memo"),E=Symbol.for("react.lazy"),R=Symbol.iterator;function y(_){return _===null||typeof _!="object"?null:(_=R&&_[R]||_["@@iterator"],typeof _=="function"?_:null)}var w={isMounted:function(){return!1},enqueueForceUpdate:function(){},enqueueReplaceState:function(){},enqueueSetState:function(){}},b=Object.assign,X={};function N(_,B,z){this.props=_,this.context=B,this.refs=X,this.updater=z||w}N.prototype.isReactComponent={},N.prototype.setState=function(_,B){if(typeof _!="object"&&typeof _!="function"&&_!=null)throw Error("takes an object of state variables to update or a function which returns an object of state variables.");this.updater.enqueueSetState(this,_,B,"setState")},N.prototype.forceUpdate=function(_){this.updater.enqueueForceUpdate(this,_,"forceUpdate")};function v(){}v.prototype=N.prototype;function A(_,B,z){this.props=_,this.context=B,this.refs=X,this.updater=z||w}var k=A.prototype=new v;k.constructor=A,b(k,N.prototype),k.isPureReactComponent=!0;var U=Array.isArray,j={H:null,A:null,T:null,S:null,V:null},K=Object.prototype.hasOwnProperty;function Z(_,B,z,G,$,de){return z=de.ref,{$$typeof:i,type:_,key:B,ref:z!==void 0?z:null,props:de}}function ce(_,B){return Z(_.type,B,void 0,void 0,void 0,_.props)}function re(_){return typeof _=="object"&&_!==null&&_.$$typeof===i}function te(_){var B={"=":"=0",":":"=2"};return"$"+_.replace(/[=:]/g,function(z){return B[z]})}var Xe=/\/+/g;function _e(_,B){return typeof _=="object"&&_!==null&&_.key!=null?te(""+_.key):B.toString(36)}function Ee(){}function qe(_){switch(_.status){case"fulfilled":return _.value;case"rejected":throw _.reason;default:switch(typeof _.status=="string"?_.then(Ee,Ee):(_.status="pending",_.then(function(B){_.status==="pending"&&(_.status="fulfilled",_.value=B)},function(B){_.status==="pending"&&(_.status="rejected",_.reason=B)})),_.status){case"fulfilled":return _.value;case"rejected":throw _.reason}}throw _}function J(_,B,z,G,$){var de=typeof _;(de==="undefined"||de==="boolean")&&(_=null);var le=!1;if(_===null)le=!0;else switch(de){case"bigint":case"string":case"number":le=!0;break;case"object":switch(_.$$typeof){case i:case t:le=!0;break;case E:return le=_._init,J(le(_._payload),B,z,G,$)}}if(le)return $=$(_),le=G===""?"."+_e(_,0):G,U($)?(z="",le!=null&&(z=le.replace(Xe,"$&/")+"/"),J($,B,z,"",function(kn){return kn})):$!=null&&(re($)&&($=ce($,z+($.key==null||_&&_.key===$.key?"":(""+$.key).replace(Xe,"$&/")+"/")+le)),B.push($)),1;le=0;var Tt=G===""?".":G+":";if(U(_))for(var Fe=0;Fe<_.length;Fe++)G=_[Fe],de=Tt+_e(G,Fe),le+=J(G,B,z,de,$);else if(Fe=y(_),typeof Fe=="function")for(_=Fe.call(_),Fe=0;!(G=_.next()).done;)G=G.value,de=Tt+_e(G,Fe++),le+=J(G,B,z,de,$);else if(de==="object"){if(typeof _.then=="function")return J(qe(_),B,z,G,$);throw B=String(_),Error("Objects are not valid as a React child (found: "+(B==="[object Object]"?"object with keys {"+Object.keys(_).join(", ")+"}":B)+"). If you meant to render a collection of children, use an array instead.")}return le}function L(_,B,z){if(_==null)return _;var G=[],$=0;return J(_,G,"","",function(de){return B.call(z,de,$++)}),G}function V(_){if(_._status===-1){var B=_._result;B=B(),B.then(function(z){(_._status===0||_._status===-1)&&(_._status=1,_._result=z)},function(z){(_._status===0||_._status===-1)&&(_._status=2,_._result=z)}),_._status===-1&&(_._status=0,_._result=B)}if(_._status===1)return _._result.default;throw _._result}var ne=typeof reportError=="function"?reportError:function(_){if(typeof window=="object"&&typeof window.ErrorEvent=="function"){var B=new window.ErrorEvent("error",{bubbles:!0,cancelable:!0,message:typeof _=="object"&&_!==null&&typeof _.message=="string"?String(_.message):String(_),error:_});if(!window.dispatchEvent(B))return}else if(typeof process=="object"&&typeof process.emit=="function"){process.emit("uncaughtException",_);return}console.error(_)};function Re(){}return ue.Children={map:L,forEach:function(_,B,z){L(_,function(){B.apply(this,arguments)},z)},count:function(_){var B=0;return L(_,function(){B++}),B},toArray:function(_){return L(_,function(B){return B})||[]},only:function(_){if(!re(_))throw Error("React.Children.only expected to receive a single React element child.");return _}},ue.Component=N,ue.Fragment=r,ue.Profiler=l,ue.PureComponent=A,ue.StrictMode=s,ue.Suspense=f,ue.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE=j,ue.__COMPILER_RUNTIME={__proto__:null,c:function(_){return j.H.useMemoCache(_)}},ue.cache=function(_){return function(){return _.apply(null,arguments)}},ue.cloneElement=function(_,B,z){if(_==null)throw Error("The argument must be a React element, but you passed "+_+".");var G=b({},_.props),$=_.key,de=void 0;if(B!=null)for(le in B.ref!==void 0&&(de=void 0),B.key!==void 0&&($=""+B.key),B)!K.call(B,le)||le==="key"||le==="__self"||le==="__source"||le==="ref"&&B.ref===void 0||(G[le]=B[le]);var le=arguments.length-2;if(le===1)G.children=z;else if(1<le){for(var Tt=Array(le),Fe=0;Fe<le;Fe++)Tt[Fe]=arguments[Fe+2];G.children=Tt}return Z(_.type,$,void 0,void 0,de,G)},ue.createContext=function(_){return _={$$typeof:p,_currentValue:_,_currentValue2:_,_threadCount:0,Provider:null,Consumer:null},_.Provider=_,_.Consumer={$$typeof:u,_context:_},_},ue.createElement=function(_,B,z){var G,$={},de=null;if(B!=null)for(G in B.key!==void 0&&(de=""+B.key),B)K.call(B,G)&&G!=="key"&&G!=="__self"&&G!=="__source"&&($[G]=B[G]);var le=arguments.length-2;if(le===1)$.children=z;else if(1<le){for(var Tt=Array(le),Fe=0;Fe<le;Fe++)Tt[Fe]=arguments[Fe+2];$.children=Tt}if(_&&_.defaultProps)for(G in le=_.defaultProps,le)$[G]===void 0&&($[G]=le[G]);return Z(_,de,void 0,void 0,null,$)},ue.createRef=function(){return{current:null}},ue.forwardRef=function(_){return{$$typeof:h,render:_}},ue.isValidElement=re,ue.lazy=function(_){return{$$typeof:E,_payload:{_status:-1,_result:_},_init:V}},ue.memo=function(_,B){return{$$typeof:g,type:_,compare:B===void 0?null:B}},ue.startTransition=function(_){var B=j.T,z={};j.T=z;try{var G=_(),$=j.S;$!==null&&$(z,G),typeof G=="object"&&G!==null&&typeof G.then=="function"&&G.then(Re,ne)}catch(de){ne(de)}finally{j.T=B}},ue.unstable_useCacheRefresh=function(){return j.H.useCacheRefresh()},ue.use=function(_){return j.H.use(_)},ue.useActionState=function(_,B,z){return j.H.useActionState(_,B,z)},ue.useCallback=function(_,B){return j.H.useCallback(_,B)},ue.useContext=function(_){return j.H.useContext(_)},ue.useDebugValue=function(){},ue.useDeferredValue=function(_,B){return j.H.useDeferredValue(_,B)},ue.useEffect=function(_,B,z){var G=j.H;if(typeof z=="function")throw Error("useEffect CRUD overload is not enabled in this build of React.");return G.useEffect(_,B)},ue.useId=function(){return j.H.useId()},ue.useImperativeHandle=function(_,B,z){return j.H.useImperativeHandle(_,B,z)},ue.useInsertionEffect=function(_,B){return j.H.useInsertionEffect(_,B)},ue.useLayoutEffect=function(_,B){return j.H.useLayoutEffect(_,B)},ue.useMemo=function(_,B){return j.H.useMemo(_,B)},ue.useOptimistic=function(_,B){return j.H.useOptimistic(_,B)},ue.useReducer=function(_,B,z){return j.H.useReducer(_,B,z)},ue.useRef=function(_){return j.H.useRef(_)},ue.useState=function(_){return j.H.useState(_)},ue.useSyncExternalStore=function(_,B,z){return j.H.useSyncExternalStore(_,B,z)},ue.useTransition=function(){return j.H.useTransition()},ue.version="19.1.0",ue}var Ug;function Td(){return Ug||(Ug=1,Lu.exports=$R()),Lu.exports}var P=Td();const wt=KR(P);var Bu={exports:{}},cs={},Mu={exports:{}},qu={};/**
 * @license React
 * scheduler.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var Wg;function eI(){return Wg||(Wg=1,function(i){function t(L,V){var ne=L.length;L.push(V);e:for(;0<ne;){var Re=ne-1>>>1,_=L[Re];if(0<l(_,V))L[Re]=V,L[ne]=_,ne=Re;else break e}}function r(L){return L.length===0?null:L[0]}function s(L){if(L.length===0)return null;var V=L[0],ne=L.pop();if(ne!==V){L[0]=ne;e:for(var Re=0,_=L.length,B=_>>>1;Re<B;){var z=2*(Re+1)-1,G=L[z],$=z+1,de=L[$];if(0>l(G,ne))$<_&&0>l(de,G)?(L[Re]=de,L[$]=ne,Re=$):(L[Re]=G,L[z]=ne,Re=z);else if($<_&&0>l(de,ne))L[Re]=de,L[$]=ne,Re=$;else break e}}return V}function l(L,V){var ne=L.sortIndex-V.sortIndex;return ne!==0?ne:L.id-V.id}if(i.unstable_now=void 0,typeof performance=="object"&&typeof performance.now=="function"){var u=performance;i.unstable_now=function(){return u.now()}}else{var p=Date,h=p.now();i.unstable_now=function(){return p.now()-h}}var f=[],g=[],E=1,R=null,y=3,w=!1,b=!1,X=!1,N=!1,v=typeof setTimeout=="function"?setTimeout:null,A=typeof clearTimeout=="function"?clearTimeout:null,k=typeof setImmediate<"u"?setImmediate:null;function U(L){for(var V=r(g);V!==null;){if(V.callback===null)s(g);else if(V.startTime<=L)s(g),V.sortIndex=V.expirationTime,t(f,V);else break;V=r(g)}}function j(L){if(X=!1,U(L),!b)if(r(f)!==null)b=!0,K||(K=!0,_e());else{var V=r(g);V!==null&&J(j,V.startTime-L)}}var K=!1,Z=-1,ce=5,re=-1;function te(){return N?!0:!(i.unstable_now()-re<ce)}function Xe(){if(N=!1,K){var L=i.unstable_now();re=L;var V=!0;try{e:{b=!1,X&&(X=!1,A(Z),Z=-1),w=!0;var ne=y;try{t:{for(U(L),R=r(f);R!==null&&!(R.expirationTime>L&&te());){var Re=R.callback;if(typeof Re=="function"){R.callback=null,y=R.priorityLevel;var _=Re(R.expirationTime<=L);if(L=i.unstable_now(),typeof _=="function"){R.callback=_,U(L),V=!0;break t}R===r(f)&&s(f),U(L)}else s(f);R=r(f)}if(R!==null)V=!0;else{var B=r(g);B!==null&&J(j,B.startTime-L),V=!1}}break e}finally{R=null,y=ne,w=!1}V=void 0}}finally{V?_e():K=!1}}}var _e;if(typeof k=="function")_e=function(){k(Xe)};else if(typeof MessageChannel<"u"){var Ee=new MessageChannel,qe=Ee.port2;Ee.port1.onmessage=Xe,_e=function(){qe.postMessage(null)}}else _e=function(){v(Xe,0)};function J(L,V){Z=v(function(){L(i.unstable_now())},V)}i.unstable_IdlePriority=5,i.unstable_ImmediatePriority=1,i.unstable_LowPriority=4,i.unstable_NormalPriority=3,i.unstable_Profiling=null,i.unstable_UserBlockingPriority=2,i.unstable_cancelCallback=function(L){L.callback=null},i.unstable_forceFrameRate=function(L){0>L||125<L?console.error("forceFrameRate takes a positive int between 0 and 125, forcing frame rates higher than 125 fps is not supported"):ce=0<L?Math.floor(1e3/L):5},i.unstable_getCurrentPriorityLevel=function(){return y},i.unstable_next=function(L){switch(y){case 1:case 2:case 3:var V=3;break;default:V=y}var ne=y;y=V;try{return L()}finally{y=ne}},i.unstable_requestPaint=function(){N=!0},i.unstable_runWithPriority=function(L,V){switch(L){case 1:case 2:case 3:case 4:case 5:break;default:L=3}var ne=y;y=L;try{return V()}finally{y=ne}},i.unstable_scheduleCallback=function(L,V,ne){var Re=i.unstable_now();switch(typeof ne=="object"&&ne!==null?(ne=ne.delay,ne=typeof ne=="number"&&0<ne?Re+ne:Re):ne=Re,L){case 1:var _=-1;break;case 2:_=250;break;case 5:_=1073741823;break;case 4:_=1e4;break;default:_=5e3}return _=ne+_,L={id:E++,callback:V,priorityLevel:L,startTime:ne,expirationTime:_,sortIndex:-1},ne>Re?(L.sortIndex=ne,t(g,L),r(f)===null&&L===r(g)&&(X?(A(Z),Z=-1):X=!0,J(j,ne-Re))):(L.sortIndex=_,t(f,L),b||w||(b=!0,K||(K=!0,_e()))),L},i.unstable_shouldYield=te,i.unstable_wrapCallback=function(L){var V=y;return function(){var ne=y;y=V;try{return L.apply(this,arguments)}finally{y=ne}}}}(qu)),qu}var jg;function tI(){return jg||(jg=1,Mu.exports=eI()),Mu.exports}var Uu={exports:{}},lt={};/**
 * @license React
 * react-dom.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var kg;function nI(){if(kg)return lt;kg=1;var i=Td();function t(f){var g="https://react.dev/errors/"+f;if(1<arguments.length){g+="?args[]="+encodeURIComponent(arguments[1]);for(var E=2;E<arguments.length;E++)g+="&args[]="+encodeURIComponent(arguments[E])}return"Minified React error #"+f+"; visit "+g+" for the full message or use the non-minified dev environment for full errors and additional helpful warnings."}function r(){}var s={d:{f:r,r:function(){throw Error(t(522))},D:r,C:r,L:r,m:r,X:r,S:r,M:r},p:0,findDOMNode:null},l=Symbol.for("react.portal");function u(f,g,E){var R=3<arguments.length&&arguments[3]!==void 0?arguments[3]:null;return{$$typeof:l,key:R==null?null:""+R,children:f,containerInfo:g,implementation:E}}var p=i.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE;function h(f,g){if(f==="font")return"";if(typeof g=="string")return g==="use-credentials"?g:""}return lt.__DOM_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE=s,lt.createPortal=function(f,g){var E=2<arguments.length&&arguments[2]!==void 0?arguments[2]:null;if(!g||g.nodeType!==1&&g.nodeType!==9&&g.nodeType!==11)throw Error(t(299));return u(f,g,null,E)},lt.flushSync=function(f){var g=p.T,E=s.p;try{if(p.T=null,s.p=2,f)return f()}finally{p.T=g,s.p=E,s.d.f()}},lt.preconnect=function(f,g){typeof f=="string"&&(g?(g=g.crossOrigin,g=typeof g=="string"?g==="use-credentials"?g:"":void 0):g=null,s.d.C(f,g))},lt.prefetchDNS=function(f){typeof f=="string"&&s.d.D(f)},lt.preinit=function(f,g){if(typeof f=="string"&&g&&typeof g.as=="string"){var E=g.as,R=h(E,g.crossOrigin),y=typeof g.integrity=="string"?g.integrity:void 0,w=typeof g.fetchPriority=="string"?g.fetchPriority:void 0;E==="style"?s.d.S(f,typeof g.precedence=="string"?g.precedence:void 0,{crossOrigin:R,integrity:y,fetchPriority:w}):E==="script"&&s.d.X(f,{crossOrigin:R,integrity:y,fetchPriority:w,nonce:typeof g.nonce=="string"?g.nonce:void 0})}},lt.preinitModule=function(f,g){if(typeof f=="string")if(typeof g=="object"&&g!==null){if(g.as==null||g.as==="script"){var E=h(g.as,g.crossOrigin);s.d.M(f,{crossOrigin:E,integrity:typeof g.integrity=="string"?g.integrity:void 0,nonce:typeof g.nonce=="string"?g.nonce:void 0})}}else g==null&&s.d.M(f)},lt.preload=function(f,g){if(typeof f=="string"&&typeof g=="object"&&g!==null&&typeof g.as=="string"){var E=g.as,R=h(E,g.crossOrigin);s.d.L(f,E,{crossOrigin:R,integrity:typeof g.integrity=="string"?g.integrity:void 0,nonce:typeof g.nonce=="string"?g.nonce:void 0,type:typeof g.type=="string"?g.type:void 0,fetchPriority:typeof g.fetchPriority=="string"?g.fetchPriority:void 0,referrerPolicy:typeof g.referrerPolicy=="string"?g.referrerPolicy:void 0,imageSrcSet:typeof g.imageSrcSet=="string"?g.imageSrcSet:void 0,imageSizes:typeof g.imageSizes=="string"?g.imageSizes:void 0,media:typeof g.media=="string"?g.media:void 0})}},lt.preloadModule=function(f,g){if(typeof f=="string")if(g){var E=h(g.as,g.crossOrigin);s.d.m(f,{as:typeof g.as=="string"&&g.as!=="script"?g.as:void 0,crossOrigin:E,integrity:typeof g.integrity=="string"?g.integrity:void 0})}else s.d.m(f)},lt.requestFormReset=function(f){s.d.r(f)},lt.unstable_batchedUpdates=function(f,g){return f(g)},lt.useFormState=function(f,g,E){return p.H.useFormState(f,g,E)},lt.useFormStatus=function(){return p.H.useHostTransitionStatus()},lt.version="19.1.0",lt}var zg;function iI(){if(zg)return Uu.exports;zg=1;function i(){if(!(typeof __REACT_DEVTOOLS_GLOBAL_HOOK__>"u"||typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE!="function"))try{__REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE(i)}catch(t){console.error(t)}}return i(),Uu.exports=nI(),Uu.exports}/**
 * @license React
 * react-dom-client.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var Gg;function oI(){if(Gg)return cs;Gg=1;var i=tI(),t=Td(),r=iI();function s(e){var n="https://react.dev/errors/"+e;if(1<arguments.length){n+="?args[]="+encodeURIComponent(arguments[1]);for(var o=2;o<arguments.length;o++)n+="&args[]="+encodeURIComponent(arguments[o])}return"Minified React error #"+e+"; visit "+n+" for the full message or use the non-minified dev environment for full errors and additional helpful warnings."}function l(e){return!(!e||e.nodeType!==1&&e.nodeType!==9&&e.nodeType!==11)}function u(e){var n=e,o=e;if(e.alternate)for(;n.return;)n=n.return;else{e=n;do n=e,(n.flags&4098)!==0&&(o=n.return),e=n.return;while(e)}return n.tag===3?o:null}function p(e){if(e.tag===13){var n=e.memoizedState;if(n===null&&(e=e.alternate,e!==null&&(n=e.memoizedState)),n!==null)return n.dehydrated}return null}function h(e){if(u(e)!==e)throw Error(s(188))}function f(e){var n=e.alternate;if(!n){if(n=u(e),n===null)throw Error(s(188));return n!==e?null:e}for(var o=e,a=n;;){var c=o.return;if(c===null)break;var d=c.alternate;if(d===null){if(a=c.return,a!==null){o=a;continue}break}if(c.child===d.child){for(d=c.child;d;){if(d===o)return h(c),e;if(d===a)return h(c),n;d=d.sibling}throw Error(s(188))}if(o.return!==a.return)o=c,a=d;else{for(var m=!1,S=c.child;S;){if(S===o){m=!0,o=c,a=d;break}if(S===a){m=!0,a=c,o=d;break}S=S.sibling}if(!m){for(S=d.child;S;){if(S===o){m=!0,o=d,a=c;break}if(S===a){m=!0,a=d,o=c;break}S=S.sibling}if(!m)throw Error(s(189))}}if(o.alternate!==a)throw Error(s(190))}if(o.tag!==3)throw Error(s(188));return o.stateNode.current===o?e:n}function g(e){var n=e.tag;if(n===5||n===26||n===27||n===6)return e;for(e=e.child;e!==null;){if(n=g(e),n!==null)return n;e=e.sibling}return null}var E=Object.assign,R=Symbol.for("react.element"),y=Symbol.for("react.transitional.element"),w=Symbol.for("react.portal"),b=Symbol.for("react.fragment"),X=Symbol.for("react.strict_mode"),N=Symbol.for("react.profiler"),v=Symbol.for("react.provider"),A=Symbol.for("react.consumer"),k=Symbol.for("react.context"),U=Symbol.for("react.forward_ref"),j=Symbol.for("react.suspense"),K=Symbol.for("react.suspense_list"),Z=Symbol.for("react.memo"),ce=Symbol.for("react.lazy"),re=Symbol.for("react.activity"),te=Symbol.for("react.memo_cache_sentinel"),Xe=Symbol.iterator;function _e(e){return e===null||typeof e!="object"?null:(e=Xe&&e[Xe]||e["@@iterator"],typeof e=="function"?e:null)}var Ee=Symbol.for("react.client.reference");function qe(e){if(e==null)return null;if(typeof e=="function")return e.$$typeof===Ee?null:e.displayName||e.name||null;if(typeof e=="string")return e;switch(e){case b:return"Fragment";case N:return"Profiler";case X:return"StrictMode";case j:return"Suspense";case K:return"SuspenseList";case re:return"Activity"}if(typeof e=="object")switch(e.$$typeof){case w:return"Portal";case k:return(e.displayName||"Context")+".Provider";case A:return(e._context.displayName||"Context")+".Consumer";case U:var n=e.render;return e=e.displayName,e||(e=n.displayName||n.name||"",e=e!==""?"ForwardRef("+e+")":"ForwardRef"),e;case Z:return n=e.displayName||null,n!==null?n:qe(e.type)||"Memo";case ce:n=e._payload,e=e._init;try{return qe(e(n))}catch{}}return null}var J=Array.isArray,L=t.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE,V=r.__DOM_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE,ne={pending:!1,data:null,method:null,action:null},Re=[],_=-1;function B(e){return{current:e}}function z(e){0>_||(e.current=Re[_],Re[_]=null,_--)}function G(e,n){_++,Re[_]=e.current,e.current=n}var $=B(null),de=B(null),le=B(null),Tt=B(null);function Fe(e,n){switch(G(le,n),G(de,e),G($,null),n.nodeType){case 9:case 11:e=(e=n.documentElement)&&(e=e.namespaceURI)?pg(e):0;break;default:if(e=n.tagName,n=n.namespaceURI)n=pg(n),e=hg(n,e);else switch(e){case"svg":e=1;break;case"math":e=2;break;default:e=0}}z($),G($,e)}function kn(){z($),z(de),z(le)}function yl(e){e.memoizedState!==null&&G(Tt,e);var n=$.current,o=hg(n,e.type);n!==o&&(G(de,e),G($,o))}function vs(e){de.current===e&&(z($),z(de)),Tt.current===e&&(z(Tt),is._currentValue=ne)}var bl=Object.prototype.hasOwnProperty,Cl=i.unstable_scheduleCallback,wl=i.unstable_cancelCallback,xS=i.unstable_shouldYield,PS=i.unstable_requestPaint,cn=i.unstable_now,XS=i.unstable_getCurrentPriorityLevel,zd=i.unstable_ImmediatePriority,Gd=i.unstable_UserBlockingPriority,xs=i.unstable_NormalPriority,NS=i.unstable_LowPriority,Qd=i.unstable_IdlePriority,OS=i.log,FS=i.unstable_setDisableYieldValue,dr=null,vt=null;function zn(e){if(typeof OS=="function"&&FS(e),vt&&typeof vt.setStrictMode=="function")try{vt.setStrictMode(dr,e)}catch{}}var xt=Math.clz32?Math.clz32:AS,DS=Math.log,HS=Math.LN2;function AS(e){return e>>>=0,e===0?32:31-(DS(e)/HS|0)|0}var Ps=256,Xs=4194304;function wi(e){var n=e&42;if(n!==0)return n;switch(e&-e){case 1:return 1;case 2:return 2;case 4:return 4;case 8:return 8;case 16:return 16;case 32:return 32;case 64:return 64;case 128:return 128;case 256:case 512:case 1024:case 2048:case 4096:case 8192:case 16384:case 32768:case 65536:case 131072:case 262144:case 524288:case 1048576:case 2097152:return e&4194048;case 4194304:case 8388608:case 16777216:case 33554432:return e&62914560;case 67108864:return 67108864;case 134217728:return 134217728;case 268435456:return 268435456;case 536870912:return 536870912;case 1073741824:return 0;default:return e}}function Ns(e,n,o){var a=e.pendingLanes;if(a===0)return 0;var c=0,d=e.suspendedLanes,m=e.pingedLanes;e=e.warmLanes;var S=a&134217727;return S!==0?(a=S&~d,a!==0?c=wi(a):(m&=S,m!==0?c=wi(m):o||(o=S&~e,o!==0&&(c=wi(o))))):(S=a&~d,S!==0?c=wi(S):m!==0?c=wi(m):o||(o=a&~e,o!==0&&(c=wi(o)))),c===0?0:n!==0&&n!==c&&(n&d)===0&&(d=c&-c,o=n&-n,d>=o||d===32&&(o&4194048)!==0)?n:c}function pr(e,n){return(e.pendingLanes&~(e.suspendedLanes&~e.pingedLanes)&n)===0}function LS(e,n){switch(e){case 1:case 2:case 4:case 8:case 64:return n+250;case 16:case 32:case 128:case 256:case 512:case 1024:case 2048:case 4096:case 8192:case 16384:case 32768:case 65536:case 131072:case 262144:case 524288:case 1048576:case 2097152:return n+5e3;case 4194304:case 8388608:case 16777216:case 33554432:return-1;case 67108864:case 134217728:case 268435456:case 536870912:case 1073741824:return-1;default:return-1}}function Vd(){var e=Ps;return Ps<<=1,(Ps&4194048)===0&&(Ps=256),e}function Yd(){var e=Xs;return Xs<<=1,(Xs&62914560)===0&&(Xs=4194304),e}function _l(e){for(var n=[],o=0;31>o;o++)n.push(e);return n}function hr(e,n){e.pendingLanes|=n,n!==268435456&&(e.suspendedLanes=0,e.pingedLanes=0,e.warmLanes=0)}function BS(e,n,o,a,c,d){var m=e.pendingLanes;e.pendingLanes=o,e.suspendedLanes=0,e.pingedLanes=0,e.warmLanes=0,e.expiredLanes&=o,e.entangledLanes&=o,e.errorRecoveryDisabledLanes&=o,e.shellSuspendCounter=0;var S=e.entanglements,C=e.expirationTimes,F=e.hiddenUpdates;for(o=m&~o;0<o;){var M=31-xt(o),W=1<<M;S[M]=0,C[M]=-1;var D=F[M];if(D!==null)for(F[M]=null,M=0;M<D.length;M++){var H=D[M];H!==null&&(H.lane&=-536870913)}o&=~W}a!==0&&Kd(e,a,0),d!==0&&c===0&&e.tag!==0&&(e.suspendedLanes|=d&~(m&~n))}function Kd(e,n,o){e.pendingLanes|=n,e.suspendedLanes&=~n;var a=31-xt(n);e.entangledLanes|=n,e.entanglements[a]=e.entanglements[a]|1073741824|o&4194090}function Zd(e,n){var o=e.entangledLanes|=n;for(e=e.entanglements;o;){var a=31-xt(o),c=1<<a;c&n|e[a]&n&&(e[a]|=n),o&=~c}}function Tl(e){switch(e){case 2:e=1;break;case 8:e=4;break;case 32:e=16;break;case 256:case 512:case 1024:case 2048:case 4096:case 8192:case 16384:case 32768:case 65536:case 131072:case 262144:case 524288:case 1048576:case 2097152:case 4194304:case 8388608:case 16777216:case 33554432:e=128;break;case 268435456:e=134217728;break;default:e=0}return e}function vl(e){return e&=-e,2<e?8<e?(e&134217727)!==0?32:268435456:8:2}function Jd(){var e=V.p;return e!==0?e:(e=window.event,e===void 0?32:Og(e.type))}function MS(e,n){var o=V.p;try{return V.p=e,n()}finally{V.p=o}}var Gn=Math.random().toString(36).slice(2),st="__reactFiber$"+Gn,St="__reactProps$"+Gn,ro="__reactContainer$"+Gn,xl="__reactEvents$"+Gn,qS="__reactListeners$"+Gn,US="__reactHandles$"+Gn,$d="__reactResources$"+Gn,fr="__reactMarker$"+Gn;function Pl(e){delete e[st],delete e[St],delete e[xl],delete e[qS],delete e[US]}function so(e){var n=e[st];if(n)return n;for(var o=e.parentNode;o;){if(n=o[ro]||o[st]){if(o=n.alternate,n.child!==null||o!==null&&o.child!==null)for(e=Eg(e);e!==null;){if(o=e[st])return o;e=Eg(e)}return n}e=o,o=e.parentNode}return null}function ao(e){if(e=e[st]||e[ro]){var n=e.tag;if(n===5||n===6||n===13||n===26||n===27||n===3)return e}return null}function gr(e){var n=e.tag;if(n===5||n===26||n===27||n===6)return e.stateNode;throw Error(s(33))}function lo(e){var n=e[$d];return n||(n=e[$d]={hoistableStyles:new Map,hoistableScripts:new Map}),n}function Ze(e){e[fr]=!0}var ep=new Set,tp={};function _i(e,n){co(e,n),co(e+"Capture",n)}function co(e,n){for(tp[e]=n,e=0;e<n.length;e++)ep.add(n[e])}var WS=RegExp("^[:A-Z_a-z\\u00C0-\\u00D6\\u00D8-\\u00F6\\u00F8-\\u02FF\\u0370-\\u037D\\u037F-\\u1FFF\\u200C-\\u200D\\u2070-\\u218F\\u2C00-\\u2FEF\\u3001-\\uD7FF\\uF900-\\uFDCF\\uFDF0-\\uFFFD][:A-Z_a-z\\u00C0-\\u00D6\\u00D8-\\u00F6\\u00F8-\\u02FF\\u0370-\\u037D\\u037F-\\u1FFF\\u200C-\\u200D\\u2070-\\u218F\\u2C00-\\u2FEF\\u3001-\\uD7FF\\uF900-\\uFDCF\\uFDF0-\\uFFFD\\-.0-9\\u00B7\\u0300-\\u036F\\u203F-\\u2040]*$"),np={},ip={};function jS(e){return bl.call(ip,e)?!0:bl.call(np,e)?!1:WS.test(e)?ip[e]=!0:(np[e]=!0,!1)}function Os(e,n,o){if(jS(n))if(o===null)e.removeAttribute(n);else{switch(typeof o){case"undefined":case"function":case"symbol":e.removeAttribute(n);return;case"boolean":var a=n.toLowerCase().slice(0,5);if(a!=="data-"&&a!=="aria-"){e.removeAttribute(n);return}}e.setAttribute(n,""+o)}}function Fs(e,n,o){if(o===null)e.removeAttribute(n);else{switch(typeof o){case"undefined":case"function":case"symbol":case"boolean":e.removeAttribute(n);return}e.setAttribute(n,""+o)}}function Cn(e,n,o,a){if(a===null)e.removeAttribute(o);else{switch(typeof a){case"undefined":case"function":case"symbol":case"boolean":e.removeAttribute(o);return}e.setAttributeNS(n,o,""+a)}}var Xl,op;function uo(e){if(Xl===void 0)try{throw Error()}catch(o){var n=o.stack.trim().match(/\n( *(at )?)/);Xl=n&&n[1]||"",op=-1<o.stack.indexOf(`
    at`)?" (<anonymous>)":-1<o.stack.indexOf("@")?"@unknown:0:0":""}return`
`+Xl+e+op}var Nl=!1;function Ol(e,n){if(!e||Nl)return"";Nl=!0;var o=Error.prepareStackTrace;Error.prepareStackTrace=void 0;try{var a={DetermineComponentFrameRoot:function(){try{if(n){var W=function(){throw Error()};if(Object.defineProperty(W.prototype,"props",{set:function(){throw Error()}}),typeof Reflect=="object"&&Reflect.construct){try{Reflect.construct(W,[])}catch(H){var D=H}Reflect.construct(e,[],W)}else{try{W.call()}catch(H){D=H}e.call(W.prototype)}}else{try{throw Error()}catch(H){D=H}(W=e())&&typeof W.catch=="function"&&W.catch(function(){})}}catch(H){if(H&&D&&typeof H.stack=="string")return[H.stack,D.stack]}return[null,null]}};a.DetermineComponentFrameRoot.displayName="DetermineComponentFrameRoot";var c=Object.getOwnPropertyDescriptor(a.DetermineComponentFrameRoot,"name");c&&c.configurable&&Object.defineProperty(a.DetermineComponentFrameRoot,"name",{value:"DetermineComponentFrameRoot"});var d=a.DetermineComponentFrameRoot(),m=d[0],S=d[1];if(m&&S){var C=m.split(`
`),F=S.split(`
`);for(c=a=0;a<C.length&&!C[a].includes("DetermineComponentFrameRoot");)a++;for(;c<F.length&&!F[c].includes("DetermineComponentFrameRoot");)c++;if(a===C.length||c===F.length)for(a=C.length-1,c=F.length-1;1<=a&&0<=c&&C[a]!==F[c];)c--;for(;1<=a&&0<=c;a--,c--)if(C[a]!==F[c]){if(a!==1||c!==1)do if(a--,c--,0>c||C[a]!==F[c]){var M=`
`+C[a].replace(" at new "," at ");return e.displayName&&M.includes("<anonymous>")&&(M=M.replace("<anonymous>",e.displayName)),M}while(1<=a&&0<=c);break}}}finally{Nl=!1,Error.prepareStackTrace=o}return(o=e?e.displayName||e.name:"")?uo(o):""}function kS(e){switch(e.tag){case 26:case 27:case 5:return uo(e.type);case 16:return uo("Lazy");case 13:return uo("Suspense");case 19:return uo("SuspenseList");case 0:case 15:return Ol(e.type,!1);case 11:return Ol(e.type.render,!1);case 1:return Ol(e.type,!0);case 31:return uo("Activity");default:return""}}function rp(e){try{var n="";do n+=kS(e),e=e.return;while(e);return n}catch(o){return`
Error generating stack: `+o.message+`
`+o.stack}}function qt(e){switch(typeof e){case"bigint":case"boolean":case"number":case"string":case"undefined":return e;case"object":return e;default:return""}}function sp(e){var n=e.type;return(e=e.nodeName)&&e.toLowerCase()==="input"&&(n==="checkbox"||n==="radio")}function zS(e){var n=sp(e)?"checked":"value",o=Object.getOwnPropertyDescriptor(e.constructor.prototype,n),a=""+e[n];if(!e.hasOwnProperty(n)&&typeof o<"u"&&typeof o.get=="function"&&typeof o.set=="function"){var c=o.get,d=o.set;return Object.defineProperty(e,n,{configurable:!0,get:function(){return c.call(this)},set:function(m){a=""+m,d.call(this,m)}}),Object.defineProperty(e,n,{enumerable:o.enumerable}),{getValue:function(){return a},setValue:function(m){a=""+m},stopTracking:function(){e._valueTracker=null,delete e[n]}}}}function Ds(e){e._valueTracker||(e._valueTracker=zS(e))}function ap(e){if(!e)return!1;var n=e._valueTracker;if(!n)return!0;var o=n.getValue(),a="";return e&&(a=sp(e)?e.checked?"true":"false":e.value),e=a,e!==o?(n.setValue(e),!0):!1}function Hs(e){if(e=e||(typeof document<"u"?document:void 0),typeof e>"u")return null;try{return e.activeElement||e.body}catch{return e.body}}var GS=/[\n"\\]/g;function Ut(e){return e.replace(GS,function(n){return"\\"+n.charCodeAt(0).toString(16)+" "})}function Fl(e,n,o,a,c,d,m,S){e.name="",m!=null&&typeof m!="function"&&typeof m!="symbol"&&typeof m!="boolean"?e.type=m:e.removeAttribute("type"),n!=null?m==="number"?(n===0&&e.value===""||e.value!=n)&&(e.value=""+qt(n)):e.value!==""+qt(n)&&(e.value=""+qt(n)):m!=="submit"&&m!=="reset"||e.removeAttribute("value"),n!=null?Dl(e,m,qt(n)):o!=null?Dl(e,m,qt(o)):a!=null&&e.removeAttribute("value"),c==null&&d!=null&&(e.defaultChecked=!!d),c!=null&&(e.checked=c&&typeof c!="function"&&typeof c!="symbol"),S!=null&&typeof S!="function"&&typeof S!="symbol"&&typeof S!="boolean"?e.name=""+qt(S):e.removeAttribute("name")}function lp(e,n,o,a,c,d,m,S){if(d!=null&&typeof d!="function"&&typeof d!="symbol"&&typeof d!="boolean"&&(e.type=d),n!=null||o!=null){if(!(d!=="submit"&&d!=="reset"||n!=null))return;o=o!=null?""+qt(o):"",n=n!=null?""+qt(n):o,S||n===e.value||(e.value=n),e.defaultValue=n}a=a??c,a=typeof a!="function"&&typeof a!="symbol"&&!!a,e.checked=S?e.checked:!!a,e.defaultChecked=!!a,m!=null&&typeof m!="function"&&typeof m!="symbol"&&typeof m!="boolean"&&(e.name=m)}function Dl(e,n,o){n==="number"&&Hs(e.ownerDocument)===e||e.defaultValue===""+o||(e.defaultValue=""+o)}function po(e,n,o,a){if(e=e.options,n){n={};for(var c=0;c<o.length;c++)n["$"+o[c]]=!0;for(o=0;o<e.length;o++)c=n.hasOwnProperty("$"+e[o].value),e[o].selected!==c&&(e[o].selected=c),c&&a&&(e[o].defaultSelected=!0)}else{for(o=""+qt(o),n=null,c=0;c<e.length;c++){if(e[c].value===o){e[c].selected=!0,a&&(e[c].defaultSelected=!0);return}n!==null||e[c].disabled||(n=e[c])}n!==null&&(n.selected=!0)}}function cp(e,n,o){if(n!=null&&(n=""+qt(n),n!==e.value&&(e.value=n),o==null)){e.defaultValue!==n&&(e.defaultValue=n);return}e.defaultValue=o!=null?""+qt(o):""}function up(e,n,o,a){if(n==null){if(a!=null){if(o!=null)throw Error(s(92));if(J(a)){if(1<a.length)throw Error(s(93));a=a[0]}o=a}o==null&&(o=""),n=o}o=qt(n),e.defaultValue=o,a=e.textContent,a===o&&a!==""&&a!==null&&(e.value=a)}function ho(e,n){if(n){var o=e.firstChild;if(o&&o===e.lastChild&&o.nodeType===3){o.nodeValue=n;return}}e.textContent=n}var QS=new Set("animationIterationCount aspectRatio borderImageOutset borderImageSlice borderImageWidth boxFlex boxFlexGroup boxOrdinalGroup columnCount columns flex flexGrow flexPositive flexShrink flexNegative flexOrder gridArea gridRow gridRowEnd gridRowSpan gridRowStart gridColumn gridColumnEnd gridColumnSpan gridColumnStart fontWeight lineClamp lineHeight opacity order orphans scale tabSize widows zIndex zoom fillOpacity floodOpacity stopOpacity strokeDasharray strokeDashoffset strokeMiterlimit strokeOpacity strokeWidth MozAnimationIterationCount MozBoxFlex MozBoxFlexGroup MozLineClamp msAnimationIterationCount msFlex msZoom msFlexGrow msFlexNegative msFlexOrder msFlexPositive msFlexShrink msGridColumn msGridColumnSpan msGridRow msGridRowSpan WebkitAnimationIterationCount WebkitBoxFlex WebKitBoxFlexGroup WebkitBoxOrdinalGroup WebkitColumnCount WebkitColumns WebkitFlex WebkitFlexGrow WebkitFlexPositive WebkitFlexShrink WebkitLineClamp".split(" "));function dp(e,n,o){var a=n.indexOf("--")===0;o==null||typeof o=="boolean"||o===""?a?e.setProperty(n,""):n==="float"?e.cssFloat="":e[n]="":a?e.setProperty(n,o):typeof o!="number"||o===0||QS.has(n)?n==="float"?e.cssFloat=o:e[n]=(""+o).trim():e[n]=o+"px"}function pp(e,n,o){if(n!=null&&typeof n!="object")throw Error(s(62));if(e=e.style,o!=null){for(var a in o)!o.hasOwnProperty(a)||n!=null&&n.hasOwnProperty(a)||(a.indexOf("--")===0?e.setProperty(a,""):a==="float"?e.cssFloat="":e[a]="");for(var c in n)a=n[c],n.hasOwnProperty(c)&&o[c]!==a&&dp(e,c,a)}else for(var d in n)n.hasOwnProperty(d)&&dp(e,d,n[d])}function Hl(e){if(e.indexOf("-")===-1)return!1;switch(e){case"annotation-xml":case"color-profile":case"font-face":case"font-face-src":case"font-face-uri":case"font-face-format":case"font-face-name":case"missing-glyph":return!1;default:return!0}}var VS=new Map([["acceptCharset","accept-charset"],["htmlFor","for"],["httpEquiv","http-equiv"],["crossOrigin","crossorigin"],["accentHeight","accent-height"],["alignmentBaseline","alignment-baseline"],["arabicForm","arabic-form"],["baselineShift","baseline-shift"],["capHeight","cap-height"],["clipPath","clip-path"],["clipRule","clip-rule"],["colorInterpolation","color-interpolation"],["colorInterpolationFilters","color-interpolation-filters"],["colorProfile","color-profile"],["colorRendering","color-rendering"],["dominantBaseline","dominant-baseline"],["enableBackground","enable-background"],["fillOpacity","fill-opacity"],["fillRule","fill-rule"],["floodColor","flood-color"],["floodOpacity","flood-opacity"],["fontFamily","font-family"],["fontSize","font-size"],["fontSizeAdjust","font-size-adjust"],["fontStretch","font-stretch"],["fontStyle","font-style"],["fontVariant","font-variant"],["fontWeight","font-weight"],["glyphName","glyph-name"],["glyphOrientationHorizontal","glyph-orientation-horizontal"],["glyphOrientationVertical","glyph-orientation-vertical"],["horizAdvX","horiz-adv-x"],["horizOriginX","horiz-origin-x"],["imageRendering","image-rendering"],["letterSpacing","letter-spacing"],["lightingColor","lighting-color"],["markerEnd","marker-end"],["markerMid","marker-mid"],["markerStart","marker-start"],["overlinePosition","overline-position"],["overlineThickness","overline-thickness"],["paintOrder","paint-order"],["panose-1","panose-1"],["pointerEvents","pointer-events"],["renderingIntent","rendering-intent"],["shapeRendering","shape-rendering"],["stopColor","stop-color"],["stopOpacity","stop-opacity"],["strikethroughPosition","strikethrough-position"],["strikethroughThickness","strikethrough-thickness"],["strokeDasharray","stroke-dasharray"],["strokeDashoffset","stroke-dashoffset"],["strokeLinecap","stroke-linecap"],["strokeLinejoin","stroke-linejoin"],["strokeMiterlimit","stroke-miterlimit"],["strokeOpacity","stroke-opacity"],["strokeWidth","stroke-width"],["textAnchor","text-anchor"],["textDecoration","text-decoration"],["textRendering","text-rendering"],["transformOrigin","transform-origin"],["underlinePosition","underline-position"],["underlineThickness","underline-thickness"],["unicodeBidi","unicode-bidi"],["unicodeRange","unicode-range"],["unitsPerEm","units-per-em"],["vAlphabetic","v-alphabetic"],["vHanging","v-hanging"],["vIdeographic","v-ideographic"],["vMathematical","v-mathematical"],["vectorEffect","vector-effect"],["vertAdvY","vert-adv-y"],["vertOriginX","vert-origin-x"],["vertOriginY","vert-origin-y"],["wordSpacing","word-spacing"],["writingMode","writing-mode"],["xmlnsXlink","xmlns:xlink"],["xHeight","x-height"]]),YS=/^[\u0000-\u001F ]*j[\r\n\t]*a[\r\n\t]*v[\r\n\t]*a[\r\n\t]*s[\r\n\t]*c[\r\n\t]*r[\r\n\t]*i[\r\n\t]*p[\r\n\t]*t[\r\n\t]*:/i;function As(e){return YS.test(""+e)?"javascript:throw new Error('React has blocked a javascript: URL as a security precaution.')":e}var Al=null;function Ll(e){return e=e.target||e.srcElement||window,e.correspondingUseElement&&(e=e.correspondingUseElement),e.nodeType===3?e.parentNode:e}var fo=null,go=null;function hp(e){var n=ao(e);if(n&&(e=n.stateNode)){var o=e[St]||null;e:switch(e=n.stateNode,n.type){case"input":if(Fl(e,o.value,o.defaultValue,o.defaultValue,o.checked,o.defaultChecked,o.type,o.name),n=o.name,o.type==="radio"&&n!=null){for(o=e;o.parentNode;)o=o.parentNode;for(o=o.querySelectorAll('input[name="'+Ut(""+n)+'"][type="radio"]'),n=0;n<o.length;n++){var a=o[n];if(a!==e&&a.form===e.form){var c=a[St]||null;if(!c)throw Error(s(90));Fl(a,c.value,c.defaultValue,c.defaultValue,c.checked,c.defaultChecked,c.type,c.name)}}for(n=0;n<o.length;n++)a=o[n],a.form===e.form&&ap(a)}break e;case"textarea":cp(e,o.value,o.defaultValue);break e;case"select":n=o.value,n!=null&&po(e,!!o.multiple,n,!1)}}}var Bl=!1;function fp(e,n,o){if(Bl)return e(n,o);Bl=!0;try{var a=e(n);return a}finally{if(Bl=!1,(fo!==null||go!==null)&&(ya(),fo&&(n=fo,e=go,go=fo=null,hp(n),e)))for(n=0;n<e.length;n++)hp(e[n])}}function mr(e,n){var o=e.stateNode;if(o===null)return null;var a=o[St]||null;if(a===null)return null;o=a[n];e:switch(n){case"onClick":case"onClickCapture":case"onDoubleClick":case"onDoubleClickCapture":case"onMouseDown":case"onMouseDownCapture":case"onMouseMove":case"onMouseMoveCapture":case"onMouseUp":case"onMouseUpCapture":case"onMouseEnter":(a=!a.disabled)||(e=e.type,a=!(e==="button"||e==="input"||e==="select"||e==="textarea")),e=!a;break e;default:e=!1}if(e)return null;if(o&&typeof o!="function")throw Error(s(231,n,typeof o));return o}var wn=!(typeof window>"u"||typeof window.document>"u"||typeof window.document.createElement>"u"),Ml=!1;if(wn)try{var Er={};Object.defineProperty(Er,"passive",{get:function(){Ml=!0}}),window.addEventListener("test",Er,Er),window.removeEventListener("test",Er,Er)}catch{Ml=!1}var Qn=null,ql=null,Ls=null;function gp(){if(Ls)return Ls;var e,n=ql,o=n.length,a,c="value"in Qn?Qn.value:Qn.textContent,d=c.length;for(e=0;e<o&&n[e]===c[e];e++);var m=o-e;for(a=1;a<=m&&n[o-a]===c[d-a];a++);return Ls=c.slice(e,1<a?1-a:void 0)}function Bs(e){var n=e.keyCode;return"charCode"in e?(e=e.charCode,e===0&&n===13&&(e=13)):e=n,e===10&&(e=13),32<=e||e===13?e:0}function Ms(){return!0}function mp(){return!1}function Rt(e){function n(o,a,c,d,m){this._reactName=o,this._targetInst=c,this.type=a,this.nativeEvent=d,this.target=m,this.currentTarget=null;for(var S in e)e.hasOwnProperty(S)&&(o=e[S],this[S]=o?o(d):d[S]);return this.isDefaultPrevented=(d.defaultPrevented!=null?d.defaultPrevented:d.returnValue===!1)?Ms:mp,this.isPropagationStopped=mp,this}return E(n.prototype,{preventDefault:function(){this.defaultPrevented=!0;var o=this.nativeEvent;o&&(o.preventDefault?o.preventDefault():typeof o.returnValue!="unknown"&&(o.returnValue=!1),this.isDefaultPrevented=Ms)},stopPropagation:function(){var o=this.nativeEvent;o&&(o.stopPropagation?o.stopPropagation():typeof o.cancelBubble!="unknown"&&(o.cancelBubble=!0),this.isPropagationStopped=Ms)},persist:function(){},isPersistent:Ms}),n}var Ti={eventPhase:0,bubbles:0,cancelable:0,timeStamp:function(e){return e.timeStamp||Date.now()},defaultPrevented:0,isTrusted:0},qs=Rt(Ti),Sr=E({},Ti,{view:0,detail:0}),KS=Rt(Sr),Ul,Wl,Rr,Us=E({},Sr,{screenX:0,screenY:0,clientX:0,clientY:0,pageX:0,pageY:0,ctrlKey:0,shiftKey:0,altKey:0,metaKey:0,getModifierState:kl,button:0,buttons:0,relatedTarget:function(e){return e.relatedTarget===void 0?e.fromElement===e.srcElement?e.toElement:e.fromElement:e.relatedTarget},movementX:function(e){return"movementX"in e?e.movementX:(e!==Rr&&(Rr&&e.type==="mousemove"?(Ul=e.screenX-Rr.screenX,Wl=e.screenY-Rr.screenY):Wl=Ul=0,Rr=e),Ul)},movementY:function(e){return"movementY"in e?e.movementY:Wl}}),Ep=Rt(Us),ZS=E({},Us,{dataTransfer:0}),JS=Rt(ZS),$S=E({},Sr,{relatedTarget:0}),jl=Rt($S),e0=E({},Ti,{animationName:0,elapsedTime:0,pseudoElement:0}),t0=Rt(e0),n0=E({},Ti,{clipboardData:function(e){return"clipboardData"in e?e.clipboardData:window.clipboardData}}),i0=Rt(n0),o0=E({},Ti,{data:0}),Sp=Rt(o0),r0={Esc:"Escape",Spacebar:" ",Left:"ArrowLeft",Up:"ArrowUp",Right:"ArrowRight",Down:"ArrowDown",Del:"Delete",Win:"OS",Menu:"ContextMenu",Apps:"ContextMenu",Scroll:"ScrollLock",MozPrintableKey:"Unidentified"},s0={8:"Backspace",9:"Tab",12:"Clear",13:"Enter",16:"Shift",17:"Control",18:"Alt",19:"Pause",20:"CapsLock",27:"Escape",32:" ",33:"PageUp",34:"PageDown",35:"End",36:"Home",37:"ArrowLeft",38:"ArrowUp",39:"ArrowRight",40:"ArrowDown",45:"Insert",46:"Delete",112:"F1",113:"F2",114:"F3",115:"F4",116:"F5",117:"F6",118:"F7",119:"F8",120:"F9",121:"F10",122:"F11",123:"F12",144:"NumLock",145:"ScrollLock",224:"Meta"},a0={Alt:"altKey",Control:"ctrlKey",Meta:"metaKey",Shift:"shiftKey"};function l0(e){var n=this.nativeEvent;return n.getModifierState?n.getModifierState(e):(e=a0[e])?!!n[e]:!1}function kl(){return l0}var c0=E({},Sr,{key:function(e){if(e.key){var n=r0[e.key]||e.key;if(n!=="Unidentified")return n}return e.type==="keypress"?(e=Bs(e),e===13?"Enter":String.fromCharCode(e)):e.type==="keydown"||e.type==="keyup"?s0[e.keyCode]||"Unidentified":""},code:0,location:0,ctrlKey:0,shiftKey:0,altKey:0,metaKey:0,repeat:0,locale:0,getModifierState:kl,charCode:function(e){return e.type==="keypress"?Bs(e):0},keyCode:function(e){return e.type==="keydown"||e.type==="keyup"?e.keyCode:0},which:function(e){return e.type==="keypress"?Bs(e):e.type==="keydown"||e.type==="keyup"?e.keyCode:0}}),u0=Rt(c0),d0=E({},Us,{pointerId:0,width:0,height:0,pressure:0,tangentialPressure:0,tiltX:0,tiltY:0,twist:0,pointerType:0,isPrimary:0}),Rp=Rt(d0),p0=E({},Sr,{touches:0,targetTouches:0,changedTouches:0,altKey:0,metaKey:0,ctrlKey:0,shiftKey:0,getModifierState:kl}),h0=Rt(p0),f0=E({},Ti,{propertyName:0,elapsedTime:0,pseudoElement:0}),g0=Rt(f0),m0=E({},Us,{deltaX:function(e){return"deltaX"in e?e.deltaX:"wheelDeltaX"in e?-e.wheelDeltaX:0},deltaY:function(e){return"deltaY"in e?e.deltaY:"wheelDeltaY"in e?-e.wheelDeltaY:"wheelDelta"in e?-e.wheelDelta:0},deltaZ:0,deltaMode:0}),E0=Rt(m0),S0=E({},Ti,{newState:0,oldState:0}),R0=Rt(S0),I0=[9,13,27,32],zl=wn&&"CompositionEvent"in window,Ir=null;wn&&"documentMode"in document&&(Ir=document.documentMode);var y0=wn&&"TextEvent"in window&&!Ir,Ip=wn&&(!zl||Ir&&8<Ir&&11>=Ir),yp=" ",bp=!1;function Cp(e,n){switch(e){case"keyup":return I0.indexOf(n.keyCode)!==-1;case"keydown":return n.keyCode!==229;case"keypress":case"mousedown":case"focusout":return!0;default:return!1}}function wp(e){return e=e.detail,typeof e=="object"&&"data"in e?e.data:null}var mo=!1;function b0(e,n){switch(e){case"compositionend":return wp(n);case"keypress":return n.which!==32?null:(bp=!0,yp);case"textInput":return e=n.data,e===yp&&bp?null:e;default:return null}}function C0(e,n){if(mo)return e==="compositionend"||!zl&&Cp(e,n)?(e=gp(),Ls=ql=Qn=null,mo=!1,e):null;switch(e){case"paste":return null;case"keypress":if(!(n.ctrlKey||n.altKey||n.metaKey)||n.ctrlKey&&n.altKey){if(n.char&&1<n.char.length)return n.char;if(n.which)return String.fromCharCode(n.which)}return null;case"compositionend":return Ip&&n.locale!=="ko"?null:n.data;default:return null}}var w0={color:!0,date:!0,datetime:!0,"datetime-local":!0,email:!0,month:!0,number:!0,password:!0,range:!0,search:!0,tel:!0,text:!0,time:!0,url:!0,week:!0};function _p(e){var n=e&&e.nodeName&&e.nodeName.toLowerCase();return n==="input"?!!w0[e.type]:n==="textarea"}function Tp(e,n,o,a){fo?go?go.push(a):go=[a]:fo=a,n=va(n,"onChange"),0<n.length&&(o=new qs("onChange","change",null,o,a),e.push({event:o,listeners:n}))}var yr=null,br=null;function _0(e){ag(e,0)}function Ws(e){var n=gr(e);if(ap(n))return e}function vp(e,n){if(e==="change")return n}var xp=!1;if(wn){var Gl;if(wn){var Ql="oninput"in document;if(!Ql){var Pp=document.createElement("div");Pp.setAttribute("oninput","return;"),Ql=typeof Pp.oninput=="function"}Gl=Ql}else Gl=!1;xp=Gl&&(!document.documentMode||9<document.documentMode)}function Xp(){yr&&(yr.detachEvent("onpropertychange",Np),br=yr=null)}function Np(e){if(e.propertyName==="value"&&Ws(br)){var n=[];Tp(n,br,e,Ll(e)),fp(_0,n)}}function T0(e,n,o){e==="focusin"?(Xp(),yr=n,br=o,yr.attachEvent("onpropertychange",Np)):e==="focusout"&&Xp()}function v0(e){if(e==="selectionchange"||e==="keyup"||e==="keydown")return Ws(br)}function x0(e,n){if(e==="click")return Ws(n)}function P0(e,n){if(e==="input"||e==="change")return Ws(n)}function X0(e,n){return e===n&&(e!==0||1/e===1/n)||e!==e&&n!==n}var Pt=typeof Object.is=="function"?Object.is:X0;function Cr(e,n){if(Pt(e,n))return!0;if(typeof e!="object"||e===null||typeof n!="object"||n===null)return!1;var o=Object.keys(e),a=Object.keys(n);if(o.length!==a.length)return!1;for(a=0;a<o.length;a++){var c=o[a];if(!bl.call(n,c)||!Pt(e[c],n[c]))return!1}return!0}function Op(e){for(;e&&e.firstChild;)e=e.firstChild;return e}function Fp(e,n){var o=Op(e);e=0;for(var a;o;){if(o.nodeType===3){if(a=e+o.textContent.length,e<=n&&a>=n)return{node:o,offset:n-e};e=a}e:{for(;o;){if(o.nextSibling){o=o.nextSibling;break e}o=o.parentNode}o=void 0}o=Op(o)}}function Dp(e,n){return e&&n?e===n?!0:e&&e.nodeType===3?!1:n&&n.nodeType===3?Dp(e,n.parentNode):"contains"in e?e.contains(n):e.compareDocumentPosition?!!(e.compareDocumentPosition(n)&16):!1:!1}function Hp(e){e=e!=null&&e.ownerDocument!=null&&e.ownerDocument.defaultView!=null?e.ownerDocument.defaultView:window;for(var n=Hs(e.document);n instanceof e.HTMLIFrameElement;){try{var o=typeof n.contentWindow.location.href=="string"}catch{o=!1}if(o)e=n.contentWindow;else break;n=Hs(e.document)}return n}function Vl(e){var n=e&&e.nodeName&&e.nodeName.toLowerCase();return n&&(n==="input"&&(e.type==="text"||e.type==="search"||e.type==="tel"||e.type==="url"||e.type==="password")||n==="textarea"||e.contentEditable==="true")}var N0=wn&&"documentMode"in document&&11>=document.documentMode,Eo=null,Yl=null,wr=null,Kl=!1;function Ap(e,n,o){var a=o.window===o?o.document:o.nodeType===9?o:o.ownerDocument;Kl||Eo==null||Eo!==Hs(a)||(a=Eo,"selectionStart"in a&&Vl(a)?a={start:a.selectionStart,end:a.selectionEnd}:(a=(a.ownerDocument&&a.ownerDocument.defaultView||window).getSelection(),a={anchorNode:a.anchorNode,anchorOffset:a.anchorOffset,focusNode:a.focusNode,focusOffset:a.focusOffset}),wr&&Cr(wr,a)||(wr=a,a=va(Yl,"onSelect"),0<a.length&&(n=new qs("onSelect","select",null,n,o),e.push({event:n,listeners:a}),n.target=Eo)))}function vi(e,n){var o={};return o[e.toLowerCase()]=n.toLowerCase(),o["Webkit"+e]="webkit"+n,o["Moz"+e]="moz"+n,o}var So={animationend:vi("Animation","AnimationEnd"),animationiteration:vi("Animation","AnimationIteration"),animationstart:vi("Animation","AnimationStart"),transitionrun:vi("Transition","TransitionRun"),transitionstart:vi("Transition","TransitionStart"),transitioncancel:vi("Transition","TransitionCancel"),transitionend:vi("Transition","TransitionEnd")},Zl={},Lp={};wn&&(Lp=document.createElement("div").style,"AnimationEvent"in window||(delete So.animationend.animation,delete So.animationiteration.animation,delete So.animationstart.animation),"TransitionEvent"in window||delete So.transitionend.transition);function xi(e){if(Zl[e])return Zl[e];if(!So[e])return e;var n=So[e],o;for(o in n)if(n.hasOwnProperty(o)&&o in Lp)return Zl[e]=n[o];return e}var Bp=xi("animationend"),Mp=xi("animationiteration"),qp=xi("animationstart"),O0=xi("transitionrun"),F0=xi("transitionstart"),D0=xi("transitioncancel"),Up=xi("transitionend"),Wp=new Map,Jl="abort auxClick beforeToggle cancel canPlay canPlayThrough click close contextMenu copy cut drag dragEnd dragEnter dragExit dragLeave dragOver dragStart drop durationChange emptied encrypted ended error gotPointerCapture input invalid keyDown keyPress keyUp load loadedData loadedMetadata loadStart lostPointerCapture mouseDown mouseMove mouseOut mouseOver mouseUp paste pause play playing pointerCancel pointerDown pointerMove pointerOut pointerOver pointerUp progress rateChange reset resize seeked seeking stalled submit suspend timeUpdate touchCancel touchEnd touchStart volumeChange scroll toggle touchMove waiting wheel".split(" ");Jl.push("scrollEnd");function $t(e,n){Wp.set(e,n),_i(n,[e])}var jp=new WeakMap;function Wt(e,n){if(typeof e=="object"&&e!==null){var o=jp.get(e);return o!==void 0?o:(n={value:e,source:n,stack:rp(n)},jp.set(e,n),n)}return{value:e,source:n,stack:rp(n)}}var jt=[],Ro=0,$l=0;function js(){for(var e=Ro,n=$l=Ro=0;n<e;){var o=jt[n];jt[n++]=null;var a=jt[n];jt[n++]=null;var c=jt[n];jt[n++]=null;var d=jt[n];if(jt[n++]=null,a!==null&&c!==null){var m=a.pending;m===null?c.next=c:(c.next=m.next,m.next=c),a.pending=c}d!==0&&kp(o,c,d)}}function ks(e,n,o,a){jt[Ro++]=e,jt[Ro++]=n,jt[Ro++]=o,jt[Ro++]=a,$l|=a,e.lanes|=a,e=e.alternate,e!==null&&(e.lanes|=a)}function ec(e,n,o,a){return ks(e,n,o,a),zs(e)}function Io(e,n){return ks(e,null,null,n),zs(e)}function kp(e,n,o){e.lanes|=o;var a=e.alternate;a!==null&&(a.lanes|=o);for(var c=!1,d=e.return;d!==null;)d.childLanes|=o,a=d.alternate,a!==null&&(a.childLanes|=o),d.tag===22&&(e=d.stateNode,e===null||e._visibility&1||(c=!0)),e=d,d=d.return;return e.tag===3?(d=e.stateNode,c&&n!==null&&(c=31-xt(o),e=d.hiddenUpdates,a=e[c],a===null?e[c]=[n]:a.push(n),n.lane=o|536870912),d):null}function zs(e){if(50<Yr)throw Yr=0,su=null,Error(s(185));for(var n=e.return;n!==null;)e=n,n=e.return;return e.tag===3?e.stateNode:null}var yo={};function H0(e,n,o,a){this.tag=e,this.key=o,this.sibling=this.child=this.return=this.stateNode=this.type=this.elementType=null,this.index=0,this.refCleanup=this.ref=null,this.pendingProps=n,this.dependencies=this.memoizedState=this.updateQueue=this.memoizedProps=null,this.mode=a,this.subtreeFlags=this.flags=0,this.deletions=null,this.childLanes=this.lanes=0,this.alternate=null}function Xt(e,n,o,a){return new H0(e,n,o,a)}function tc(e){return e=e.prototype,!(!e||!e.isReactComponent)}function _n(e,n){var o=e.alternate;return o===null?(o=Xt(e.tag,n,e.key,e.mode),o.elementType=e.elementType,o.type=e.type,o.stateNode=e.stateNode,o.alternate=e,e.alternate=o):(o.pendingProps=n,o.type=e.type,o.flags=0,o.subtreeFlags=0,o.deletions=null),o.flags=e.flags&65011712,o.childLanes=e.childLanes,o.lanes=e.lanes,o.child=e.child,o.memoizedProps=e.memoizedProps,o.memoizedState=e.memoizedState,o.updateQueue=e.updateQueue,n=e.dependencies,o.dependencies=n===null?null:{lanes:n.lanes,firstContext:n.firstContext},o.sibling=e.sibling,o.index=e.index,o.ref=e.ref,o.refCleanup=e.refCleanup,o}function zp(e,n){e.flags&=65011714;var o=e.alternate;return o===null?(e.childLanes=0,e.lanes=n,e.child=null,e.subtreeFlags=0,e.memoizedProps=null,e.memoizedState=null,e.updateQueue=null,e.dependencies=null,e.stateNode=null):(e.childLanes=o.childLanes,e.lanes=o.lanes,e.child=o.child,e.subtreeFlags=0,e.deletions=null,e.memoizedProps=o.memoizedProps,e.memoizedState=o.memoizedState,e.updateQueue=o.updateQueue,e.type=o.type,n=o.dependencies,e.dependencies=n===null?null:{lanes:n.lanes,firstContext:n.firstContext}),e}function Gs(e,n,o,a,c,d){var m=0;if(a=e,typeof e=="function")tc(e)&&(m=1);else if(typeof e=="string")m=LR(e,o,$.current)?26:e==="html"||e==="head"||e==="body"?27:5;else e:switch(e){case re:return e=Xt(31,o,n,c),e.elementType=re,e.lanes=d,e;case b:return Pi(o.children,c,d,n);case X:m=8,c|=24;break;case N:return e=Xt(12,o,n,c|2),e.elementType=N,e.lanes=d,e;case j:return e=Xt(13,o,n,c),e.elementType=j,e.lanes=d,e;case K:return e=Xt(19,o,n,c),e.elementType=K,e.lanes=d,e;default:if(typeof e=="object"&&e!==null)switch(e.$$typeof){case v:case k:m=10;break e;case A:m=9;break e;case U:m=11;break e;case Z:m=14;break e;case ce:m=16,a=null;break e}m=29,o=Error(s(130,e===null?"null":typeof e,"")),a=null}return n=Xt(m,o,n,c),n.elementType=e,n.type=a,n.lanes=d,n}function Pi(e,n,o,a){return e=Xt(7,e,a,n),e.lanes=o,e}function nc(e,n,o){return e=Xt(6,e,null,n),e.lanes=o,e}function ic(e,n,o){return n=Xt(4,e.children!==null?e.children:[],e.key,n),n.lanes=o,n.stateNode={containerInfo:e.containerInfo,pendingChildren:null,implementation:e.implementation},n}var bo=[],Co=0,Qs=null,Vs=0,kt=[],zt=0,Xi=null,Tn=1,vn="";function Ni(e,n){bo[Co++]=Vs,bo[Co++]=Qs,Qs=e,Vs=n}function Gp(e,n,o){kt[zt++]=Tn,kt[zt++]=vn,kt[zt++]=Xi,Xi=e;var a=Tn;e=vn;var c=32-xt(a)-1;a&=~(1<<c),o+=1;var d=32-xt(n)+c;if(30<d){var m=c-c%5;d=(a&(1<<m)-1).toString(32),a>>=m,c-=m,Tn=1<<32-xt(n)+c|o<<c|a,vn=d+e}else Tn=1<<d|o<<c|a,vn=e}function oc(e){e.return!==null&&(Ni(e,1),Gp(e,1,0))}function rc(e){for(;e===Qs;)Qs=bo[--Co],bo[Co]=null,Vs=bo[--Co],bo[Co]=null;for(;e===Xi;)Xi=kt[--zt],kt[zt]=null,vn=kt[--zt],kt[zt]=null,Tn=kt[--zt],kt[zt]=null}var gt=null,Le=null,be=!1,Oi=null,un=!1,sc=Error(s(519));function Fi(e){var n=Error(s(418,""));throw vr(Wt(n,e)),sc}function Qp(e){var n=e.stateNode,o=e.type,a=e.memoizedProps;switch(n[st]=e,n[St]=a,o){case"dialog":ge("cancel",n),ge("close",n);break;case"iframe":case"object":case"embed":ge("load",n);break;case"video":case"audio":for(o=0;o<Zr.length;o++)ge(Zr[o],n);break;case"source":ge("error",n);break;case"img":case"image":case"link":ge("error",n),ge("load",n);break;case"details":ge("toggle",n);break;case"input":ge("invalid",n),lp(n,a.value,a.defaultValue,a.checked,a.defaultChecked,a.type,a.name,!0),Ds(n);break;case"select":ge("invalid",n);break;case"textarea":ge("invalid",n),up(n,a.value,a.defaultValue,a.children),Ds(n)}o=a.children,typeof o!="string"&&typeof o!="number"&&typeof o!="bigint"||n.textContent===""+o||a.suppressHydrationWarning===!0||dg(n.textContent,o)?(a.popover!=null&&(ge("beforetoggle",n),ge("toggle",n)),a.onScroll!=null&&ge("scroll",n),a.onScrollEnd!=null&&ge("scrollend",n),a.onClick!=null&&(n.onclick=xa),n=!0):n=!1,n||Fi(e)}function Vp(e){for(gt=e.return;gt;)switch(gt.tag){case 5:case 13:un=!1;return;case 27:case 3:un=!0;return;default:gt=gt.return}}function _r(e){if(e!==gt)return!1;if(!be)return Vp(e),be=!0,!1;var n=e.tag,o;if((o=n!==3&&n!==27)&&((o=n===5)&&(o=e.type,o=!(o!=="form"&&o!=="button")||bu(e.type,e.memoizedProps)),o=!o),o&&Le&&Fi(e),Vp(e),n===13){if(e=e.memoizedState,e=e!==null?e.dehydrated:null,!e)throw Error(s(317));e:{for(e=e.nextSibling,n=0;e;){if(e.nodeType===8)if(o=e.data,o==="/$"){if(n===0){Le=tn(e.nextSibling);break e}n--}else o!=="$"&&o!=="$!"&&o!=="$?"||n++;e=e.nextSibling}Le=null}}else n===27?(n=Le,ci(e.type)?(e=Tu,Tu=null,Le=e):Le=n):Le=gt?tn(e.stateNode.nextSibling):null;return!0}function Tr(){Le=gt=null,be=!1}function Yp(){var e=Oi;return e!==null&&(bt===null?bt=e:bt.push.apply(bt,e),Oi=null),e}function vr(e){Oi===null?Oi=[e]:Oi.push(e)}var ac=B(null),Di=null,xn=null;function Vn(e,n,o){G(ac,n._currentValue),n._currentValue=o}function Pn(e){e._currentValue=ac.current,z(ac)}function lc(e,n,o){for(;e!==null;){var a=e.alternate;if((e.childLanes&n)!==n?(e.childLanes|=n,a!==null&&(a.childLanes|=n)):a!==null&&(a.childLanes&n)!==n&&(a.childLanes|=n),e===o)break;e=e.return}}function cc(e,n,o,a){var c=e.child;for(c!==null&&(c.return=e);c!==null;){var d=c.dependencies;if(d!==null){var m=c.child;d=d.firstContext;e:for(;d!==null;){var S=d;d=c;for(var C=0;C<n.length;C++)if(S.context===n[C]){d.lanes|=o,S=d.alternate,S!==null&&(S.lanes|=o),lc(d.return,o,e),a||(m=null);break e}d=S.next}}else if(c.tag===18){if(m=c.return,m===null)throw Error(s(341));m.lanes|=o,d=m.alternate,d!==null&&(d.lanes|=o),lc(m,o,e),m=null}else m=c.child;if(m!==null)m.return=c;else for(m=c;m!==null;){if(m===e){m=null;break}if(c=m.sibling,c!==null){c.return=m.return,m=c;break}m=m.return}c=m}}function xr(e,n,o,a){e=null;for(var c=n,d=!1;c!==null;){if(!d){if((c.flags&524288)!==0)d=!0;else if((c.flags&262144)!==0)break}if(c.tag===10){var m=c.alternate;if(m===null)throw Error(s(387));if(m=m.memoizedProps,m!==null){var S=c.type;Pt(c.pendingProps.value,m.value)||(e!==null?e.push(S):e=[S])}}else if(c===Tt.current){if(m=c.alternate,m===null)throw Error(s(387));m.memoizedState.memoizedState!==c.memoizedState.memoizedState&&(e!==null?e.push(is):e=[is])}c=c.return}e!==null&&cc(n,e,o,a),n.flags|=262144}function Ys(e){for(e=e.firstContext;e!==null;){if(!Pt(e.context._currentValue,e.memoizedValue))return!0;e=e.next}return!1}function Hi(e){Di=e,xn=null,e=e.dependencies,e!==null&&(e.firstContext=null)}function at(e){return Kp(Di,e)}function Ks(e,n){return Di===null&&Hi(e),Kp(e,n)}function Kp(e,n){var o=n._currentValue;if(n={context:n,memoizedValue:o,next:null},xn===null){if(e===null)throw Error(s(308));xn=n,e.dependencies={lanes:0,firstContext:n},e.flags|=524288}else xn=xn.next=n;return o}var A0=typeof AbortController<"u"?AbortController:function(){var e=[],n=this.signal={aborted:!1,addEventListener:function(o,a){e.push(a)}};this.abort=function(){n.aborted=!0,e.forEach(function(o){return o()})}},L0=i.unstable_scheduleCallback,B0=i.unstable_NormalPriority,Ve={$$typeof:k,Consumer:null,Provider:null,_currentValue:null,_currentValue2:null,_threadCount:0};function uc(){return{controller:new A0,data:new Map,refCount:0}}function Pr(e){e.refCount--,e.refCount===0&&L0(B0,function(){e.controller.abort()})}var Xr=null,dc=0,wo=0,_o=null;function M0(e,n){if(Xr===null){var o=Xr=[];dc=0,wo=hu(),_o={status:"pending",value:void 0,then:function(a){o.push(a)}}}return dc++,n.then(Zp,Zp),n}function Zp(){if(--dc===0&&Xr!==null){_o!==null&&(_o.status="fulfilled");var e=Xr;Xr=null,wo=0,_o=null;for(var n=0;n<e.length;n++)(0,e[n])()}}function q0(e,n){var o=[],a={status:"pending",value:null,reason:null,then:function(c){o.push(c)}};return e.then(function(){a.status="fulfilled",a.value=n;for(var c=0;c<o.length;c++)(0,o[c])(n)},function(c){for(a.status="rejected",a.reason=c,c=0;c<o.length;c++)(0,o[c])(void 0)}),a}var Jp=L.S;L.S=function(e,n){typeof n=="object"&&n!==null&&typeof n.then=="function"&&M0(e,n),Jp!==null&&Jp(e,n)};var Ai=B(null);function pc(){var e=Ai.current;return e!==null?e:Ne.pooledCache}function Zs(e,n){n===null?G(Ai,Ai.current):G(Ai,n.pool)}function $p(){var e=pc();return e===null?null:{parent:Ve._currentValue,pool:e}}var Nr=Error(s(460)),eh=Error(s(474)),Js=Error(s(542)),hc={then:function(){}};function th(e){return e=e.status,e==="fulfilled"||e==="rejected"}function $s(){}function nh(e,n,o){switch(o=e[o],o===void 0?e.push(n):o!==n&&(n.then($s,$s),n=o),n.status){case"fulfilled":return n.value;case"rejected":throw e=n.reason,oh(e),e;default:if(typeof n.status=="string")n.then($s,$s);else{if(e=Ne,e!==null&&100<e.shellSuspendCounter)throw Error(s(482));e=n,e.status="pending",e.then(function(a){if(n.status==="pending"){var c=n;c.status="fulfilled",c.value=a}},function(a){if(n.status==="pending"){var c=n;c.status="rejected",c.reason=a}})}switch(n.status){case"fulfilled":return n.value;case"rejected":throw e=n.reason,oh(e),e}throw Or=n,Nr}}var Or=null;function ih(){if(Or===null)throw Error(s(459));var e=Or;return Or=null,e}function oh(e){if(e===Nr||e===Js)throw Error(s(483))}var Yn=!1;function fc(e){e.updateQueue={baseState:e.memoizedState,firstBaseUpdate:null,lastBaseUpdate:null,shared:{pending:null,lanes:0,hiddenCallbacks:null},callbacks:null}}function gc(e,n){e=e.updateQueue,n.updateQueue===e&&(n.updateQueue={baseState:e.baseState,firstBaseUpdate:e.firstBaseUpdate,lastBaseUpdate:e.lastBaseUpdate,shared:e.shared,callbacks:null})}function Kn(e){return{lane:e,tag:0,payload:null,callback:null,next:null}}function Zn(e,n,o){var a=e.updateQueue;if(a===null)return null;if(a=a.shared,(Ce&2)!==0){var c=a.pending;return c===null?n.next=n:(n.next=c.next,c.next=n),a.pending=n,n=zs(e),kp(e,null,o),n}return ks(e,a,n,o),zs(e)}function Fr(e,n,o){if(n=n.updateQueue,n!==null&&(n=n.shared,(o&4194048)!==0)){var a=n.lanes;a&=e.pendingLanes,o|=a,n.lanes=o,Zd(e,o)}}function mc(e,n){var o=e.updateQueue,a=e.alternate;if(a!==null&&(a=a.updateQueue,o===a)){var c=null,d=null;if(o=o.firstBaseUpdate,o!==null){do{var m={lane:o.lane,tag:o.tag,payload:o.payload,callback:null,next:null};d===null?c=d=m:d=d.next=m,o=o.next}while(o!==null);d===null?c=d=n:d=d.next=n}else c=d=n;o={baseState:a.baseState,firstBaseUpdate:c,lastBaseUpdate:d,shared:a.shared,callbacks:a.callbacks},e.updateQueue=o;return}e=o.lastBaseUpdate,e===null?o.firstBaseUpdate=n:e.next=n,o.lastBaseUpdate=n}var Ec=!1;function Dr(){if(Ec){var e=_o;if(e!==null)throw e}}function Hr(e,n,o,a){Ec=!1;var c=e.updateQueue;Yn=!1;var d=c.firstBaseUpdate,m=c.lastBaseUpdate,S=c.shared.pending;if(S!==null){c.shared.pending=null;var C=S,F=C.next;C.next=null,m===null?d=F:m.next=F,m=C;var M=e.alternate;M!==null&&(M=M.updateQueue,S=M.lastBaseUpdate,S!==m&&(S===null?M.firstBaseUpdate=F:S.next=F,M.lastBaseUpdate=C))}if(d!==null){var W=c.baseState;m=0,M=F=C=null,S=d;do{var D=S.lane&-536870913,H=D!==S.lane;if(H?(Se&D)===D:(a&D)===D){D!==0&&D===wo&&(Ec=!0),M!==null&&(M=M.next={lane:0,tag:S.tag,payload:S.payload,callback:null,next:null});e:{var ae=e,oe=S;D=n;var xe=o;switch(oe.tag){case 1:if(ae=oe.payload,typeof ae=="function"){W=ae.call(xe,W,D);break e}W=ae;break e;case 3:ae.flags=ae.flags&-65537|128;case 0:if(ae=oe.payload,D=typeof ae=="function"?ae.call(xe,W,D):ae,D==null)break e;W=E({},W,D);break e;case 2:Yn=!0}}D=S.callback,D!==null&&(e.flags|=64,H&&(e.flags|=8192),H=c.callbacks,H===null?c.callbacks=[D]:H.push(D))}else H={lane:D,tag:S.tag,payload:S.payload,callback:S.callback,next:null},M===null?(F=M=H,C=W):M=M.next=H,m|=D;if(S=S.next,S===null){if(S=c.shared.pending,S===null)break;H=S,S=H.next,H.next=null,c.lastBaseUpdate=H,c.shared.pending=null}}while(!0);M===null&&(C=W),c.baseState=C,c.firstBaseUpdate=F,c.lastBaseUpdate=M,d===null&&(c.shared.lanes=0),ri|=m,e.lanes=m,e.memoizedState=W}}function rh(e,n){if(typeof e!="function")throw Error(s(191,e));e.call(n)}function sh(e,n){var o=e.callbacks;if(o!==null)for(e.callbacks=null,e=0;e<o.length;e++)rh(o[e],n)}var To=B(null),ea=B(0);function ah(e,n){e=An,G(ea,e),G(To,n),An=e|n.baseLanes}function Sc(){G(ea,An),G(To,To.current)}function Rc(){An=ea.current,z(To),z(ea)}var Jn=0,pe=null,Te=null,ke=null,ta=!1,vo=!1,Li=!1,na=0,Ar=0,xo=null,U0=0;function Ue(){throw Error(s(321))}function Ic(e,n){if(n===null)return!1;for(var o=0;o<n.length&&o<e.length;o++)if(!Pt(e[o],n[o]))return!1;return!0}function yc(e,n,o,a,c,d){return Jn=d,pe=n,n.memoizedState=null,n.updateQueue=null,n.lanes=0,L.H=e===null||e.memoizedState===null?kh:zh,Li=!1,d=o(a,c),Li=!1,vo&&(d=ch(n,o,a,c)),lh(e),d}function lh(e){L.H=la;var n=Te!==null&&Te.next!==null;if(Jn=0,ke=Te=pe=null,ta=!1,Ar=0,xo=null,n)throw Error(s(300));e===null||Je||(e=e.dependencies,e!==null&&Ys(e)&&(Je=!0))}function ch(e,n,o,a){pe=e;var c=0;do{if(vo&&(xo=null),Ar=0,vo=!1,25<=c)throw Error(s(301));if(c+=1,ke=Te=null,e.updateQueue!=null){var d=e.updateQueue;d.lastEffect=null,d.events=null,d.stores=null,d.memoCache!=null&&(d.memoCache.index=0)}L.H=V0,d=n(o,a)}while(vo);return d}function W0(){var e=L.H,n=e.useState()[0];return n=typeof n.then=="function"?Lr(n):n,e=e.useState()[0],(Te!==null?Te.memoizedState:null)!==e&&(pe.flags|=1024),n}function bc(){var e=na!==0;return na=0,e}function Cc(e,n,o){n.updateQueue=e.updateQueue,n.flags&=-2053,e.lanes&=~o}function wc(e){if(ta){for(e=e.memoizedState;e!==null;){var n=e.queue;n!==null&&(n.pending=null),e=e.next}ta=!1}Jn=0,ke=Te=pe=null,vo=!1,Ar=na=0,xo=null}function It(){var e={memoizedState:null,baseState:null,baseQueue:null,queue:null,next:null};return ke===null?pe.memoizedState=ke=e:ke=ke.next=e,ke}function ze(){if(Te===null){var e=pe.alternate;e=e!==null?e.memoizedState:null}else e=Te.next;var n=ke===null?pe.memoizedState:ke.next;if(n!==null)ke=n,Te=e;else{if(e===null)throw pe.alternate===null?Error(s(467)):Error(s(310));Te=e,e={memoizedState:Te.memoizedState,baseState:Te.baseState,baseQueue:Te.baseQueue,queue:Te.queue,next:null},ke===null?pe.memoizedState=ke=e:ke=ke.next=e}return ke}function _c(){return{lastEffect:null,events:null,stores:null,memoCache:null}}function Lr(e){var n=Ar;return Ar+=1,xo===null&&(xo=[]),e=nh(xo,e,n),n=pe,(ke===null?n.memoizedState:ke.next)===null&&(n=n.alternate,L.H=n===null||n.memoizedState===null?kh:zh),e}function ia(e){if(e!==null&&typeof e=="object"){if(typeof e.then=="function")return Lr(e);if(e.$$typeof===k)return at(e)}throw Error(s(438,String(e)))}function Tc(e){var n=null,o=pe.updateQueue;if(o!==null&&(n=o.memoCache),n==null){var a=pe.alternate;a!==null&&(a=a.updateQueue,a!==null&&(a=a.memoCache,a!=null&&(n={data:a.data.map(function(c){return c.slice()}),index:0})))}if(n==null&&(n={data:[],index:0}),o===null&&(o=_c(),pe.updateQueue=o),o.memoCache=n,o=n.data[n.index],o===void 0)for(o=n.data[n.index]=Array(e),a=0;a<e;a++)o[a]=te;return n.index++,o}function Xn(e,n){return typeof n=="function"?n(e):n}function oa(e){var n=ze();return vc(n,Te,e)}function vc(e,n,o){var a=e.queue;if(a===null)throw Error(s(311));a.lastRenderedReducer=o;var c=e.baseQueue,d=a.pending;if(d!==null){if(c!==null){var m=c.next;c.next=d.next,d.next=m}n.baseQueue=c=d,a.pending=null}if(d=e.baseState,c===null)e.memoizedState=d;else{n=c.next;var S=m=null,C=null,F=n,M=!1;do{var W=F.lane&-536870913;if(W!==F.lane?(Se&W)===W:(Jn&W)===W){var D=F.revertLane;if(D===0)C!==null&&(C=C.next={lane:0,revertLane:0,action:F.action,hasEagerState:F.hasEagerState,eagerState:F.eagerState,next:null}),W===wo&&(M=!0);else if((Jn&D)===D){F=F.next,D===wo&&(M=!0);continue}else W={lane:0,revertLane:F.revertLane,action:F.action,hasEagerState:F.hasEagerState,eagerState:F.eagerState,next:null},C===null?(S=C=W,m=d):C=C.next=W,pe.lanes|=D,ri|=D;W=F.action,Li&&o(d,W),d=F.hasEagerState?F.eagerState:o(d,W)}else D={lane:W,revertLane:F.revertLane,action:F.action,hasEagerState:F.hasEagerState,eagerState:F.eagerState,next:null},C===null?(S=C=D,m=d):C=C.next=D,pe.lanes|=W,ri|=W;F=F.next}while(F!==null&&F!==n);if(C===null?m=d:C.next=S,!Pt(d,e.memoizedState)&&(Je=!0,M&&(o=_o,o!==null)))throw o;e.memoizedState=d,e.baseState=m,e.baseQueue=C,a.lastRenderedState=d}return c===null&&(a.lanes=0),[e.memoizedState,a.dispatch]}function xc(e){var n=ze(),o=n.queue;if(o===null)throw Error(s(311));o.lastRenderedReducer=e;var a=o.dispatch,c=o.pending,d=n.memoizedState;if(c!==null){o.pending=null;var m=c=c.next;do d=e(d,m.action),m=m.next;while(m!==c);Pt(d,n.memoizedState)||(Je=!0),n.memoizedState=d,n.baseQueue===null&&(n.baseState=d),o.lastRenderedState=d}return[d,a]}function uh(e,n,o){var a=pe,c=ze(),d=be;if(d){if(o===void 0)throw Error(s(407));o=o()}else o=n();var m=!Pt((Te||c).memoizedState,o);m&&(c.memoizedState=o,Je=!0),c=c.queue;var S=hh.bind(null,a,c,e);if(Br(2048,8,S,[e]),c.getSnapshot!==n||m||ke!==null&&ke.memoizedState.tag&1){if(a.flags|=2048,Po(9,ra(),ph.bind(null,a,c,o,n),null),Ne===null)throw Error(s(349));d||(Jn&124)!==0||dh(a,n,o)}return o}function dh(e,n,o){e.flags|=16384,e={getSnapshot:n,value:o},n=pe.updateQueue,n===null?(n=_c(),pe.updateQueue=n,n.stores=[e]):(o=n.stores,o===null?n.stores=[e]:o.push(e))}function ph(e,n,o,a){n.value=o,n.getSnapshot=a,fh(n)&&gh(e)}function hh(e,n,o){return o(function(){fh(n)&&gh(e)})}function fh(e){var n=e.getSnapshot;e=e.value;try{var o=n();return!Pt(e,o)}catch{return!0}}function gh(e){var n=Io(e,2);n!==null&&Ht(n,e,2)}function Pc(e){var n=It();if(typeof e=="function"){var o=e;if(e=o(),Li){zn(!0);try{o()}finally{zn(!1)}}}return n.memoizedState=n.baseState=e,n.queue={pending:null,lanes:0,dispatch:null,lastRenderedReducer:Xn,lastRenderedState:e},n}function mh(e,n,o,a){return e.baseState=o,vc(e,Te,typeof a=="function"?a:Xn)}function j0(e,n,o,a,c){if(aa(e))throw Error(s(485));if(e=n.action,e!==null){var d={payload:c,action:e,next:null,isTransition:!0,status:"pending",value:null,reason:null,listeners:[],then:function(m){d.listeners.push(m)}};L.T!==null?o(!0):d.isTransition=!1,a(d),o=n.pending,o===null?(d.next=n.pending=d,Eh(n,d)):(d.next=o.next,n.pending=o.next=d)}}function Eh(e,n){var o=n.action,a=n.payload,c=e.state;if(n.isTransition){var d=L.T,m={};L.T=m;try{var S=o(c,a),C=L.S;C!==null&&C(m,S),Sh(e,n,S)}catch(F){Xc(e,n,F)}finally{L.T=d}}else try{d=o(c,a),Sh(e,n,d)}catch(F){Xc(e,n,F)}}function Sh(e,n,o){o!==null&&typeof o=="object"&&typeof o.then=="function"?o.then(function(a){Rh(e,n,a)},function(a){return Xc(e,n,a)}):Rh(e,n,o)}function Rh(e,n,o){n.status="fulfilled",n.value=o,Ih(n),e.state=o,n=e.pending,n!==null&&(o=n.next,o===n?e.pending=null:(o=o.next,n.next=o,Eh(e,o)))}function Xc(e,n,o){var a=e.pending;if(e.pending=null,a!==null){a=a.next;do n.status="rejected",n.reason=o,Ih(n),n=n.next;while(n!==a)}e.action=null}function Ih(e){e=e.listeners;for(var n=0;n<e.length;n++)(0,e[n])()}function yh(e,n){return n}function bh(e,n){if(be){var o=Ne.formState;if(o!==null){e:{var a=pe;if(be){if(Le){t:{for(var c=Le,d=un;c.nodeType!==8;){if(!d){c=null;break t}if(c=tn(c.nextSibling),c===null){c=null;break t}}d=c.data,c=d==="F!"||d==="F"?c:null}if(c){Le=tn(c.nextSibling),a=c.data==="F!";break e}}Fi(a)}a=!1}a&&(n=o[0])}}return o=It(),o.memoizedState=o.baseState=n,a={pending:null,lanes:0,dispatch:null,lastRenderedReducer:yh,lastRenderedState:n},o.queue=a,o=Uh.bind(null,pe,a),a.dispatch=o,a=Pc(!1),d=Hc.bind(null,pe,!1,a.queue),a=It(),c={state:n,dispatch:null,action:e,pending:null},a.queue=c,o=j0.bind(null,pe,c,d,o),c.dispatch=o,a.memoizedState=e,[n,o,!1]}function Ch(e){var n=ze();return wh(n,Te,e)}function wh(e,n,o){if(n=vc(e,n,yh)[0],e=oa(Xn)[0],typeof n=="object"&&n!==null&&typeof n.then=="function")try{var a=Lr(n)}catch(m){throw m===Nr?Js:m}else a=n;n=ze();var c=n.queue,d=c.dispatch;return o!==n.memoizedState&&(pe.flags|=2048,Po(9,ra(),k0.bind(null,c,o),null)),[a,d,e]}function k0(e,n){e.action=n}function _h(e){var n=ze(),o=Te;if(o!==null)return wh(n,o,e);ze(),n=n.memoizedState,o=ze();var a=o.queue.dispatch;return o.memoizedState=e,[n,a,!1]}function Po(e,n,o,a){return e={tag:e,create:o,deps:a,inst:n,next:null},n=pe.updateQueue,n===null&&(n=_c(),pe.updateQueue=n),o=n.lastEffect,o===null?n.lastEffect=e.next=e:(a=o.next,o.next=e,e.next=a,n.lastEffect=e),e}function ra(){return{destroy:void 0,resource:void 0}}function Th(){return ze().memoizedState}function sa(e,n,o,a){var c=It();a=a===void 0?null:a,pe.flags|=e,c.memoizedState=Po(1|n,ra(),o,a)}function Br(e,n,o,a){var c=ze();a=a===void 0?null:a;var d=c.memoizedState.inst;Te!==null&&a!==null&&Ic(a,Te.memoizedState.deps)?c.memoizedState=Po(n,d,o,a):(pe.flags|=e,c.memoizedState=Po(1|n,d,o,a))}function vh(e,n){sa(8390656,8,e,n)}function xh(e,n){Br(2048,8,e,n)}function Ph(e,n){return Br(4,2,e,n)}function Xh(e,n){return Br(4,4,e,n)}function Nh(e,n){if(typeof n=="function"){e=e();var o=n(e);return function(){typeof o=="function"?o():n(null)}}if(n!=null)return e=e(),n.current=e,function(){n.current=null}}function Oh(e,n,o){o=o!=null?o.concat([e]):null,Br(4,4,Nh.bind(null,n,e),o)}function Nc(){}function Fh(e,n){var o=ze();n=n===void 0?null:n;var a=o.memoizedState;return n!==null&&Ic(n,a[1])?a[0]:(o.memoizedState=[e,n],e)}function Dh(e,n){var o=ze();n=n===void 0?null:n;var a=o.memoizedState;if(n!==null&&Ic(n,a[1]))return a[0];if(a=e(),Li){zn(!0);try{e()}finally{zn(!1)}}return o.memoizedState=[a,n],a}function Oc(e,n,o){return o===void 0||(Jn&1073741824)!==0?e.memoizedState=n:(e.memoizedState=o,e=Bf(),pe.lanes|=e,ri|=e,o)}function Hh(e,n,o,a){return Pt(o,n)?o:To.current!==null?(e=Oc(e,o,a),Pt(e,n)||(Je=!0),e):(Jn&42)===0?(Je=!0,e.memoizedState=o):(e=Bf(),pe.lanes|=e,ri|=e,n)}function Ah(e,n,o,a,c){var d=V.p;V.p=d!==0&&8>d?d:8;var m=L.T,S={};L.T=S,Hc(e,!1,n,o);try{var C=c(),F=L.S;if(F!==null&&F(S,C),C!==null&&typeof C=="object"&&typeof C.then=="function"){var M=q0(C,a);Mr(e,n,M,Dt(e))}else Mr(e,n,a,Dt(e))}catch(W){Mr(e,n,{then:function(){},status:"rejected",reason:W},Dt())}finally{V.p=d,L.T=m}}function z0(){}function Fc(e,n,o,a){if(e.tag!==5)throw Error(s(476));var c=Lh(e).queue;Ah(e,c,n,ne,o===null?z0:function(){return Bh(e),o(a)})}function Lh(e){var n=e.memoizedState;if(n!==null)return n;n={memoizedState:ne,baseState:ne,baseQueue:null,queue:{pending:null,lanes:0,dispatch:null,lastRenderedReducer:Xn,lastRenderedState:ne},next:null};var o={};return n.next={memoizedState:o,baseState:o,baseQueue:null,queue:{pending:null,lanes:0,dispatch:null,lastRenderedReducer:Xn,lastRenderedState:o},next:null},e.memoizedState=n,e=e.alternate,e!==null&&(e.memoizedState=n),n}function Bh(e){var n=Lh(e).next.queue;Mr(e,n,{},Dt())}function Dc(){return at(is)}function Mh(){return ze().memoizedState}function qh(){return ze().memoizedState}function G0(e){for(var n=e.return;n!==null;){switch(n.tag){case 24:case 3:var o=Dt();e=Kn(o);var a=Zn(n,e,o);a!==null&&(Ht(a,n,o),Fr(a,n,o)),n={cache:uc()},e.payload=n;return}n=n.return}}function Q0(e,n,o){var a=Dt();o={lane:a,revertLane:0,action:o,hasEagerState:!1,eagerState:null,next:null},aa(e)?Wh(n,o):(o=ec(e,n,o,a),o!==null&&(Ht(o,e,a),jh(o,n,a)))}function Uh(e,n,o){var a=Dt();Mr(e,n,o,a)}function Mr(e,n,o,a){var c={lane:a,revertLane:0,action:o,hasEagerState:!1,eagerState:null,next:null};if(aa(e))Wh(n,c);else{var d=e.alternate;if(e.lanes===0&&(d===null||d.lanes===0)&&(d=n.lastRenderedReducer,d!==null))try{var m=n.lastRenderedState,S=d(m,o);if(c.hasEagerState=!0,c.eagerState=S,Pt(S,m))return ks(e,n,c,0),Ne===null&&js(),!1}catch{}finally{}if(o=ec(e,n,c,a),o!==null)return Ht(o,e,a),jh(o,n,a),!0}return!1}function Hc(e,n,o,a){if(a={lane:2,revertLane:hu(),action:a,hasEagerState:!1,eagerState:null,next:null},aa(e)){if(n)throw Error(s(479))}else n=ec(e,o,a,2),n!==null&&Ht(n,e,2)}function aa(e){var n=e.alternate;return e===pe||n!==null&&n===pe}function Wh(e,n){vo=ta=!0;var o=e.pending;o===null?n.next=n:(n.next=o.next,o.next=n),e.pending=n}function jh(e,n,o){if((o&4194048)!==0){var a=n.lanes;a&=e.pendingLanes,o|=a,n.lanes=o,Zd(e,o)}}var la={readContext:at,use:ia,useCallback:Ue,useContext:Ue,useEffect:Ue,useImperativeHandle:Ue,useLayoutEffect:Ue,useInsertionEffect:Ue,useMemo:Ue,useReducer:Ue,useRef:Ue,useState:Ue,useDebugValue:Ue,useDeferredValue:Ue,useTransition:Ue,useSyncExternalStore:Ue,useId:Ue,useHostTransitionStatus:Ue,useFormState:Ue,useActionState:Ue,useOptimistic:Ue,useMemoCache:Ue,useCacheRefresh:Ue},kh={readContext:at,use:ia,useCallback:function(e,n){return It().memoizedState=[e,n===void 0?null:n],e},useContext:at,useEffect:vh,useImperativeHandle:function(e,n,o){o=o!=null?o.concat([e]):null,sa(4194308,4,Nh.bind(null,n,e),o)},useLayoutEffect:function(e,n){return sa(4194308,4,e,n)},useInsertionEffect:function(e,n){sa(4,2,e,n)},useMemo:function(e,n){var o=It();n=n===void 0?null:n;var a=e();if(Li){zn(!0);try{e()}finally{zn(!1)}}return o.memoizedState=[a,n],a},useReducer:function(e,n,o){var a=It();if(o!==void 0){var c=o(n);if(Li){zn(!0);try{o(n)}finally{zn(!1)}}}else c=n;return a.memoizedState=a.baseState=c,e={pending:null,lanes:0,dispatch:null,lastRenderedReducer:e,lastRenderedState:c},a.queue=e,e=e.dispatch=Q0.bind(null,pe,e),[a.memoizedState,e]},useRef:function(e){var n=It();return e={current:e},n.memoizedState=e},useState:function(e){e=Pc(e);var n=e.queue,o=Uh.bind(null,pe,n);return n.dispatch=o,[e.memoizedState,o]},useDebugValue:Nc,useDeferredValue:function(e,n){var o=It();return Oc(o,e,n)},useTransition:function(){var e=Pc(!1);return e=Ah.bind(null,pe,e.queue,!0,!1),It().memoizedState=e,[!1,e]},useSyncExternalStore:function(e,n,o){var a=pe,c=It();if(be){if(o===void 0)throw Error(s(407));o=o()}else{if(o=n(),Ne===null)throw Error(s(349));(Se&124)!==0||dh(a,n,o)}c.memoizedState=o;var d={value:o,getSnapshot:n};return c.queue=d,vh(hh.bind(null,a,d,e),[e]),a.flags|=2048,Po(9,ra(),ph.bind(null,a,d,o,n),null),o},useId:function(){var e=It(),n=Ne.identifierPrefix;if(be){var o=vn,a=Tn;o=(a&~(1<<32-xt(a)-1)).toString(32)+o,n="«"+n+"R"+o,o=na++,0<o&&(n+="H"+o.toString(32)),n+="»"}else o=U0++,n="«"+n+"r"+o.toString(32)+"»";return e.memoizedState=n},useHostTransitionStatus:Dc,useFormState:bh,useActionState:bh,useOptimistic:function(e){var n=It();n.memoizedState=n.baseState=e;var o={pending:null,lanes:0,dispatch:null,lastRenderedReducer:null,lastRenderedState:null};return n.queue=o,n=Hc.bind(null,pe,!0,o),o.dispatch=n,[e,n]},useMemoCache:Tc,useCacheRefresh:function(){return It().memoizedState=G0.bind(null,pe)}},zh={readContext:at,use:ia,useCallback:Fh,useContext:at,useEffect:xh,useImperativeHandle:Oh,useInsertionEffect:Ph,useLayoutEffect:Xh,useMemo:Dh,useReducer:oa,useRef:Th,useState:function(){return oa(Xn)},useDebugValue:Nc,useDeferredValue:function(e,n){var o=ze();return Hh(o,Te.memoizedState,e,n)},useTransition:function(){var e=oa(Xn)[0],n=ze().memoizedState;return[typeof e=="boolean"?e:Lr(e),n]},useSyncExternalStore:uh,useId:Mh,useHostTransitionStatus:Dc,useFormState:Ch,useActionState:Ch,useOptimistic:function(e,n){var o=ze();return mh(o,Te,e,n)},useMemoCache:Tc,useCacheRefresh:qh},V0={readContext:at,use:ia,useCallback:Fh,useContext:at,useEffect:xh,useImperativeHandle:Oh,useInsertionEffect:Ph,useLayoutEffect:Xh,useMemo:Dh,useReducer:xc,useRef:Th,useState:function(){return xc(Xn)},useDebugValue:Nc,useDeferredValue:function(e,n){var o=ze();return Te===null?Oc(o,e,n):Hh(o,Te.memoizedState,e,n)},useTransition:function(){var e=xc(Xn)[0],n=ze().memoizedState;return[typeof e=="boolean"?e:Lr(e),n]},useSyncExternalStore:uh,useId:Mh,useHostTransitionStatus:Dc,useFormState:_h,useActionState:_h,useOptimistic:function(e,n){var o=ze();return Te!==null?mh(o,Te,e,n):(o.baseState=e,[e,o.queue.dispatch])},useMemoCache:Tc,useCacheRefresh:qh},Xo=null,qr=0;function ca(e){var n=qr;return qr+=1,Xo===null&&(Xo=[]),nh(Xo,e,n)}function Ur(e,n){n=n.props.ref,e.ref=n!==void 0?n:null}function ua(e,n){throw n.$$typeof===R?Error(s(525)):(e=Object.prototype.toString.call(n),Error(s(31,e==="[object Object]"?"object with keys {"+Object.keys(n).join(", ")+"}":e)))}function Gh(e){var n=e._init;return n(e._payload)}function Qh(e){function n(x,T){if(e){var O=x.deletions;O===null?(x.deletions=[T],x.flags|=16):O.push(T)}}function o(x,T){if(!e)return null;for(;T!==null;)n(x,T),T=T.sibling;return null}function a(x){for(var T=new Map;x!==null;)x.key!==null?T.set(x.key,x):T.set(x.index,x),x=x.sibling;return T}function c(x,T){return x=_n(x,T),x.index=0,x.sibling=null,x}function d(x,T,O){return x.index=O,e?(O=x.alternate,O!==null?(O=O.index,O<T?(x.flags|=67108866,T):O):(x.flags|=67108866,T)):(x.flags|=1048576,T)}function m(x){return e&&x.alternate===null&&(x.flags|=67108866),x}function S(x,T,O,q){return T===null||T.tag!==6?(T=nc(O,x.mode,q),T.return=x,T):(T=c(T,O),T.return=x,T)}function C(x,T,O,q){var Y=O.type;return Y===b?M(x,T,O.props.children,q,O.key):T!==null&&(T.elementType===Y||typeof Y=="object"&&Y!==null&&Y.$$typeof===ce&&Gh(Y)===T.type)?(T=c(T,O.props),Ur(T,O),T.return=x,T):(T=Gs(O.type,O.key,O.props,null,x.mode,q),Ur(T,O),T.return=x,T)}function F(x,T,O,q){return T===null||T.tag!==4||T.stateNode.containerInfo!==O.containerInfo||T.stateNode.implementation!==O.implementation?(T=ic(O,x.mode,q),T.return=x,T):(T=c(T,O.children||[]),T.return=x,T)}function M(x,T,O,q,Y){return T===null||T.tag!==7?(T=Pi(O,x.mode,q,Y),T.return=x,T):(T=c(T,O),T.return=x,T)}function W(x,T,O){if(typeof T=="string"&&T!==""||typeof T=="number"||typeof T=="bigint")return T=nc(""+T,x.mode,O),T.return=x,T;if(typeof T=="object"&&T!==null){switch(T.$$typeof){case y:return O=Gs(T.type,T.key,T.props,null,x.mode,O),Ur(O,T),O.return=x,O;case w:return T=ic(T,x.mode,O),T.return=x,T;case ce:var q=T._init;return T=q(T._payload),W(x,T,O)}if(J(T)||_e(T))return T=Pi(T,x.mode,O,null),T.return=x,T;if(typeof T.then=="function")return W(x,ca(T),O);if(T.$$typeof===k)return W(x,Ks(x,T),O);ua(x,T)}return null}function D(x,T,O,q){var Y=T!==null?T.key:null;if(typeof O=="string"&&O!==""||typeof O=="number"||typeof O=="bigint")return Y!==null?null:S(x,T,""+O,q);if(typeof O=="object"&&O!==null){switch(O.$$typeof){case y:return O.key===Y?C(x,T,O,q):null;case w:return O.key===Y?F(x,T,O,q):null;case ce:return Y=O._init,O=Y(O._payload),D(x,T,O,q)}if(J(O)||_e(O))return Y!==null?null:M(x,T,O,q,null);if(typeof O.then=="function")return D(x,T,ca(O),q);if(O.$$typeof===k)return D(x,T,Ks(x,O),q);ua(x,O)}return null}function H(x,T,O,q,Y){if(typeof q=="string"&&q!==""||typeof q=="number"||typeof q=="bigint")return x=x.get(O)||null,S(T,x,""+q,Y);if(typeof q=="object"&&q!==null){switch(q.$$typeof){case y:return x=x.get(q.key===null?O:q.key)||null,C(T,x,q,Y);case w:return x=x.get(q.key===null?O:q.key)||null,F(T,x,q,Y);case ce:var he=q._init;return q=he(q._payload),H(x,T,O,q,Y)}if(J(q)||_e(q))return x=x.get(O)||null,M(T,x,q,Y,null);if(typeof q.then=="function")return H(x,T,O,ca(q),Y);if(q.$$typeof===k)return H(x,T,O,Ks(T,q),Y);ua(T,q)}return null}function ae(x,T,O,q){for(var Y=null,he=null,ee=T,se=T=0,et=null;ee!==null&&se<O.length;se++){ee.index>se?(et=ee,ee=null):et=ee.sibling;var Ie=D(x,ee,O[se],q);if(Ie===null){ee===null&&(ee=et);break}e&&ee&&Ie.alternate===null&&n(x,ee),T=d(Ie,T,se),he===null?Y=Ie:he.sibling=Ie,he=Ie,ee=et}if(se===O.length)return o(x,ee),be&&Ni(x,se),Y;if(ee===null){for(;se<O.length;se++)ee=W(x,O[se],q),ee!==null&&(T=d(ee,T,se),he===null?Y=ee:he.sibling=ee,he=ee);return be&&Ni(x,se),Y}for(ee=a(ee);se<O.length;se++)et=H(ee,x,se,O[se],q),et!==null&&(e&&et.alternate!==null&&ee.delete(et.key===null?se:et.key),T=d(et,T,se),he===null?Y=et:he.sibling=et,he=et);return e&&ee.forEach(function(fi){return n(x,fi)}),be&&Ni(x,se),Y}function oe(x,T,O,q){if(O==null)throw Error(s(151));for(var Y=null,he=null,ee=T,se=T=0,et=null,Ie=O.next();ee!==null&&!Ie.done;se++,Ie=O.next()){ee.index>se?(et=ee,ee=null):et=ee.sibling;var fi=D(x,ee,Ie.value,q);if(fi===null){ee===null&&(ee=et);break}e&&ee&&fi.alternate===null&&n(x,ee),T=d(fi,T,se),he===null?Y=fi:he.sibling=fi,he=fi,ee=et}if(Ie.done)return o(x,ee),be&&Ni(x,se),Y;if(ee===null){for(;!Ie.done;se++,Ie=O.next())Ie=W(x,Ie.value,q),Ie!==null&&(T=d(Ie,T,se),he===null?Y=Ie:he.sibling=Ie,he=Ie);return be&&Ni(x,se),Y}for(ee=a(ee);!Ie.done;se++,Ie=O.next())Ie=H(ee,x,se,Ie.value,q),Ie!==null&&(e&&Ie.alternate!==null&&ee.delete(Ie.key===null?se:Ie.key),T=d(Ie,T,se),he===null?Y=Ie:he.sibling=Ie,he=Ie);return e&&ee.forEach(function(YR){return n(x,YR)}),be&&Ni(x,se),Y}function xe(x,T,O,q){if(typeof O=="object"&&O!==null&&O.type===b&&O.key===null&&(O=O.props.children),typeof O=="object"&&O!==null){switch(O.$$typeof){case y:e:{for(var Y=O.key;T!==null;){if(T.key===Y){if(Y=O.type,Y===b){if(T.tag===7){o(x,T.sibling),q=c(T,O.props.children),q.return=x,x=q;break e}}else if(T.elementType===Y||typeof Y=="object"&&Y!==null&&Y.$$typeof===ce&&Gh(Y)===T.type){o(x,T.sibling),q=c(T,O.props),Ur(q,O),q.return=x,x=q;break e}o(x,T);break}else n(x,T);T=T.sibling}O.type===b?(q=Pi(O.props.children,x.mode,q,O.key),q.return=x,x=q):(q=Gs(O.type,O.key,O.props,null,x.mode,q),Ur(q,O),q.return=x,x=q)}return m(x);case w:e:{for(Y=O.key;T!==null;){if(T.key===Y)if(T.tag===4&&T.stateNode.containerInfo===O.containerInfo&&T.stateNode.implementation===O.implementation){o(x,T.sibling),q=c(T,O.children||[]),q.return=x,x=q;break e}else{o(x,T);break}else n(x,T);T=T.sibling}q=ic(O,x.mode,q),q.return=x,x=q}return m(x);case ce:return Y=O._init,O=Y(O._payload),xe(x,T,O,q)}if(J(O))return ae(x,T,O,q);if(_e(O)){if(Y=_e(O),typeof Y!="function")throw Error(s(150));return O=Y.call(O),oe(x,T,O,q)}if(typeof O.then=="function")return xe(x,T,ca(O),q);if(O.$$typeof===k)return xe(x,T,Ks(x,O),q);ua(x,O)}return typeof O=="string"&&O!==""||typeof O=="number"||typeof O=="bigint"?(O=""+O,T!==null&&T.tag===6?(o(x,T.sibling),q=c(T,O),q.return=x,x=q):(o(x,T),q=nc(O,x.mode,q),q.return=x,x=q),m(x)):o(x,T)}return function(x,T,O,q){try{qr=0;var Y=xe(x,T,O,q);return Xo=null,Y}catch(ee){if(ee===Nr||ee===Js)throw ee;var he=Xt(29,ee,null,x.mode);return he.lanes=q,he.return=x,he}finally{}}}var No=Qh(!0),Vh=Qh(!1),Gt=B(null),dn=null;function $n(e){var n=e.alternate;G(Ye,Ye.current&1),G(Gt,e),dn===null&&(n===null||To.current!==null||n.memoizedState!==null)&&(dn=e)}function Yh(e){if(e.tag===22){if(G(Ye,Ye.current),G(Gt,e),dn===null){var n=e.alternate;n!==null&&n.memoizedState!==null&&(dn=e)}}else ei()}function ei(){G(Ye,Ye.current),G(Gt,Gt.current)}function Nn(e){z(Gt),dn===e&&(dn=null),z(Ye)}var Ye=B(0);function da(e){for(var n=e;n!==null;){if(n.tag===13){var o=n.memoizedState;if(o!==null&&(o=o.dehydrated,o===null||o.data==="$?"||_u(o)))return n}else if(n.tag===19&&n.memoizedProps.revealOrder!==void 0){if((n.flags&128)!==0)return n}else if(n.child!==null){n.child.return=n,n=n.child;continue}if(n===e)break;for(;n.sibling===null;){if(n.return===null||n.return===e)return null;n=n.return}n.sibling.return=n.return,n=n.sibling}return null}function Ac(e,n,o,a){n=e.memoizedState,o=o(a,n),o=o==null?n:E({},n,o),e.memoizedState=o,e.lanes===0&&(e.updateQueue.baseState=o)}var Lc={enqueueSetState:function(e,n,o){e=e._reactInternals;var a=Dt(),c=Kn(a);c.payload=n,o!=null&&(c.callback=o),n=Zn(e,c,a),n!==null&&(Ht(n,e,a),Fr(n,e,a))},enqueueReplaceState:function(e,n,o){e=e._reactInternals;var a=Dt(),c=Kn(a);c.tag=1,c.payload=n,o!=null&&(c.callback=o),n=Zn(e,c,a),n!==null&&(Ht(n,e,a),Fr(n,e,a))},enqueueForceUpdate:function(e,n){e=e._reactInternals;var o=Dt(),a=Kn(o);a.tag=2,n!=null&&(a.callback=n),n=Zn(e,a,o),n!==null&&(Ht(n,e,o),Fr(n,e,o))}};function Kh(e,n,o,a,c,d,m){return e=e.stateNode,typeof e.shouldComponentUpdate=="function"?e.shouldComponentUpdate(a,d,m):n.prototype&&n.prototype.isPureReactComponent?!Cr(o,a)||!Cr(c,d):!0}function Zh(e,n,o,a){e=n.state,typeof n.componentWillReceiveProps=="function"&&n.componentWillReceiveProps(o,a),typeof n.UNSAFE_componentWillReceiveProps=="function"&&n.UNSAFE_componentWillReceiveProps(o,a),n.state!==e&&Lc.enqueueReplaceState(n,n.state,null)}function Bi(e,n){var o=n;if("ref"in n){o={};for(var a in n)a!=="ref"&&(o[a]=n[a])}if(e=e.defaultProps){o===n&&(o=E({},o));for(var c in e)o[c]===void 0&&(o[c]=e[c])}return o}var pa=typeof reportError=="function"?reportError:function(e){if(typeof window=="object"&&typeof window.ErrorEvent=="function"){var n=new window.ErrorEvent("error",{bubbles:!0,cancelable:!0,message:typeof e=="object"&&e!==null&&typeof e.message=="string"?String(e.message):String(e),error:e});if(!window.dispatchEvent(n))return}else if(typeof process=="object"&&typeof process.emit=="function"){process.emit("uncaughtException",e);return}console.error(e)};function Jh(e){pa(e)}function $h(e){console.error(e)}function ef(e){pa(e)}function ha(e,n){try{var o=e.onUncaughtError;o(n.value,{componentStack:n.stack})}catch(a){setTimeout(function(){throw a})}}function tf(e,n,o){try{var a=e.onCaughtError;a(o.value,{componentStack:o.stack,errorBoundary:n.tag===1?n.stateNode:null})}catch(c){setTimeout(function(){throw c})}}function Bc(e,n,o){return o=Kn(o),o.tag=3,o.payload={element:null},o.callback=function(){ha(e,n)},o}function nf(e){return e=Kn(e),e.tag=3,e}function of(e,n,o,a){var c=o.type.getDerivedStateFromError;if(typeof c=="function"){var d=a.value;e.payload=function(){return c(d)},e.callback=function(){tf(n,o,a)}}var m=o.stateNode;m!==null&&typeof m.componentDidCatch=="function"&&(e.callback=function(){tf(n,o,a),typeof c!="function"&&(si===null?si=new Set([this]):si.add(this));var S=a.stack;this.componentDidCatch(a.value,{componentStack:S!==null?S:""})})}function Y0(e,n,o,a,c){if(o.flags|=32768,a!==null&&typeof a=="object"&&typeof a.then=="function"){if(n=o.alternate,n!==null&&xr(n,o,c,!0),o=Gt.current,o!==null){switch(o.tag){case 13:return dn===null?lu():o.alternate===null&&Be===0&&(Be=3),o.flags&=-257,o.flags|=65536,o.lanes=c,a===hc?o.flags|=16384:(n=o.updateQueue,n===null?o.updateQueue=new Set([a]):n.add(a),uu(e,a,c)),!1;case 22:return o.flags|=65536,a===hc?o.flags|=16384:(n=o.updateQueue,n===null?(n={transitions:null,markerInstances:null,retryQueue:new Set([a])},o.updateQueue=n):(o=n.retryQueue,o===null?n.retryQueue=new Set([a]):o.add(a)),uu(e,a,c)),!1}throw Error(s(435,o.tag))}return uu(e,a,c),lu(),!1}if(be)return n=Gt.current,n!==null?((n.flags&65536)===0&&(n.flags|=256),n.flags|=65536,n.lanes=c,a!==sc&&(e=Error(s(422),{cause:a}),vr(Wt(e,o)))):(a!==sc&&(n=Error(s(423),{cause:a}),vr(Wt(n,o))),e=e.current.alternate,e.flags|=65536,c&=-c,e.lanes|=c,a=Wt(a,o),c=Bc(e.stateNode,a,c),mc(e,c),Be!==4&&(Be=2)),!1;var d=Error(s(520),{cause:a});if(d=Wt(d,o),Vr===null?Vr=[d]:Vr.push(d),Be!==4&&(Be=2),n===null)return!0;a=Wt(a,o),o=n;do{switch(o.tag){case 3:return o.flags|=65536,e=c&-c,o.lanes|=e,e=Bc(o.stateNode,a,e),mc(o,e),!1;case 1:if(n=o.type,d=o.stateNode,(o.flags&128)===0&&(typeof n.getDerivedStateFromError=="function"||d!==null&&typeof d.componentDidCatch=="function"&&(si===null||!si.has(d))))return o.flags|=65536,c&=-c,o.lanes|=c,c=nf(c),of(c,e,o,a),mc(o,c),!1}o=o.return}while(o!==null);return!1}var rf=Error(s(461)),Je=!1;function nt(e,n,o,a){n.child=e===null?Vh(n,null,o,a):No(n,e.child,o,a)}function sf(e,n,o,a,c){o=o.render;var d=n.ref;if("ref"in a){var m={};for(var S in a)S!=="ref"&&(m[S]=a[S])}else m=a;return Hi(n),a=yc(e,n,o,m,d,c),S=bc(),e!==null&&!Je?(Cc(e,n,c),On(e,n,c)):(be&&S&&oc(n),n.flags|=1,nt(e,n,a,c),n.child)}function af(e,n,o,a,c){if(e===null){var d=o.type;return typeof d=="function"&&!tc(d)&&d.defaultProps===void 0&&o.compare===null?(n.tag=15,n.type=d,lf(e,n,d,a,c)):(e=Gs(o.type,null,a,n,n.mode,c),e.ref=n.ref,e.return=n,n.child=e)}if(d=e.child,!Gc(e,c)){var m=d.memoizedProps;if(o=o.compare,o=o!==null?o:Cr,o(m,a)&&e.ref===n.ref)return On(e,n,c)}return n.flags|=1,e=_n(d,a),e.ref=n.ref,e.return=n,n.child=e}function lf(e,n,o,a,c){if(e!==null){var d=e.memoizedProps;if(Cr(d,a)&&e.ref===n.ref)if(Je=!1,n.pendingProps=a=d,Gc(e,c))(e.flags&131072)!==0&&(Je=!0);else return n.lanes=e.lanes,On(e,n,c)}return Mc(e,n,o,a,c)}function cf(e,n,o){var a=n.pendingProps,c=a.children,d=e!==null?e.memoizedState:null;if(a.mode==="hidden"){if((n.flags&128)!==0){if(a=d!==null?d.baseLanes|o:o,e!==null){for(c=n.child=e.child,d=0;c!==null;)d=d|c.lanes|c.childLanes,c=c.sibling;n.childLanes=d&~a}else n.childLanes=0,n.child=null;return uf(e,n,a,o)}if((o&536870912)!==0)n.memoizedState={baseLanes:0,cachePool:null},e!==null&&Zs(n,d!==null?d.cachePool:null),d!==null?ah(n,d):Sc(),Yh(n);else return n.lanes=n.childLanes=536870912,uf(e,n,d!==null?d.baseLanes|o:o,o)}else d!==null?(Zs(n,d.cachePool),ah(n,d),ei(),n.memoizedState=null):(e!==null&&Zs(n,null),Sc(),ei());return nt(e,n,c,o),n.child}function uf(e,n,o,a){var c=pc();return c=c===null?null:{parent:Ve._currentValue,pool:c},n.memoizedState={baseLanes:o,cachePool:c},e!==null&&Zs(n,null),Sc(),Yh(n),e!==null&&xr(e,n,a,!0),null}function fa(e,n){var o=n.ref;if(o===null)e!==null&&e.ref!==null&&(n.flags|=4194816);else{if(typeof o!="function"&&typeof o!="object")throw Error(s(284));(e===null||e.ref!==o)&&(n.flags|=4194816)}}function Mc(e,n,o,a,c){return Hi(n),o=yc(e,n,o,a,void 0,c),a=bc(),e!==null&&!Je?(Cc(e,n,c),On(e,n,c)):(be&&a&&oc(n),n.flags|=1,nt(e,n,o,c),n.child)}function df(e,n,o,a,c,d){return Hi(n),n.updateQueue=null,o=ch(n,a,o,c),lh(e),a=bc(),e!==null&&!Je?(Cc(e,n,d),On(e,n,d)):(be&&a&&oc(n),n.flags|=1,nt(e,n,o,d),n.child)}function pf(e,n,o,a,c){if(Hi(n),n.stateNode===null){var d=yo,m=o.contextType;typeof m=="object"&&m!==null&&(d=at(m)),d=new o(a,d),n.memoizedState=d.state!==null&&d.state!==void 0?d.state:null,d.updater=Lc,n.stateNode=d,d._reactInternals=n,d=n.stateNode,d.props=a,d.state=n.memoizedState,d.refs={},fc(n),m=o.contextType,d.context=typeof m=="object"&&m!==null?at(m):yo,d.state=n.memoizedState,m=o.getDerivedStateFromProps,typeof m=="function"&&(Ac(n,o,m,a),d.state=n.memoizedState),typeof o.getDerivedStateFromProps=="function"||typeof d.getSnapshotBeforeUpdate=="function"||typeof d.UNSAFE_componentWillMount!="function"&&typeof d.componentWillMount!="function"||(m=d.state,typeof d.componentWillMount=="function"&&d.componentWillMount(),typeof d.UNSAFE_componentWillMount=="function"&&d.UNSAFE_componentWillMount(),m!==d.state&&Lc.enqueueReplaceState(d,d.state,null),Hr(n,a,d,c),Dr(),d.state=n.memoizedState),typeof d.componentDidMount=="function"&&(n.flags|=4194308),a=!0}else if(e===null){d=n.stateNode;var S=n.memoizedProps,C=Bi(o,S);d.props=C;var F=d.context,M=o.contextType;m=yo,typeof M=="object"&&M!==null&&(m=at(M));var W=o.getDerivedStateFromProps;M=typeof W=="function"||typeof d.getSnapshotBeforeUpdate=="function",S=n.pendingProps!==S,M||typeof d.UNSAFE_componentWillReceiveProps!="function"&&typeof d.componentWillReceiveProps!="function"||(S||F!==m)&&Zh(n,d,a,m),Yn=!1;var D=n.memoizedState;d.state=D,Hr(n,a,d,c),Dr(),F=n.memoizedState,S||D!==F||Yn?(typeof W=="function"&&(Ac(n,o,W,a),F=n.memoizedState),(C=Yn||Kh(n,o,C,a,D,F,m))?(M||typeof d.UNSAFE_componentWillMount!="function"&&typeof d.componentWillMount!="function"||(typeof d.componentWillMount=="function"&&d.componentWillMount(),typeof d.UNSAFE_componentWillMount=="function"&&d.UNSAFE_componentWillMount()),typeof d.componentDidMount=="function"&&(n.flags|=4194308)):(typeof d.componentDidMount=="function"&&(n.flags|=4194308),n.memoizedProps=a,n.memoizedState=F),d.props=a,d.state=F,d.context=m,a=C):(typeof d.componentDidMount=="function"&&(n.flags|=4194308),a=!1)}else{d=n.stateNode,gc(e,n),m=n.memoizedProps,M=Bi(o,m),d.props=M,W=n.pendingProps,D=d.context,F=o.contextType,C=yo,typeof F=="object"&&F!==null&&(C=at(F)),S=o.getDerivedStateFromProps,(F=typeof S=="function"||typeof d.getSnapshotBeforeUpdate=="function")||typeof d.UNSAFE_componentWillReceiveProps!="function"&&typeof d.componentWillReceiveProps!="function"||(m!==W||D!==C)&&Zh(n,d,a,C),Yn=!1,D=n.memoizedState,d.state=D,Hr(n,a,d,c),Dr();var H=n.memoizedState;m!==W||D!==H||Yn||e!==null&&e.dependencies!==null&&Ys(e.dependencies)?(typeof S=="function"&&(Ac(n,o,S,a),H=n.memoizedState),(M=Yn||Kh(n,o,M,a,D,H,C)||e!==null&&e.dependencies!==null&&Ys(e.dependencies))?(F||typeof d.UNSAFE_componentWillUpdate!="function"&&typeof d.componentWillUpdate!="function"||(typeof d.componentWillUpdate=="function"&&d.componentWillUpdate(a,H,C),typeof d.UNSAFE_componentWillUpdate=="function"&&d.UNSAFE_componentWillUpdate(a,H,C)),typeof d.componentDidUpdate=="function"&&(n.flags|=4),typeof d.getSnapshotBeforeUpdate=="function"&&(n.flags|=1024)):(typeof d.componentDidUpdate!="function"||m===e.memoizedProps&&D===e.memoizedState||(n.flags|=4),typeof d.getSnapshotBeforeUpdate!="function"||m===e.memoizedProps&&D===e.memoizedState||(n.flags|=1024),n.memoizedProps=a,n.memoizedState=H),d.props=a,d.state=H,d.context=C,a=M):(typeof d.componentDidUpdate!="function"||m===e.memoizedProps&&D===e.memoizedState||(n.flags|=4),typeof d.getSnapshotBeforeUpdate!="function"||m===e.memoizedProps&&D===e.memoizedState||(n.flags|=1024),a=!1)}return d=a,fa(e,n),a=(n.flags&128)!==0,d||a?(d=n.stateNode,o=a&&typeof o.getDerivedStateFromError!="function"?null:d.render(),n.flags|=1,e!==null&&a?(n.child=No(n,e.child,null,c),n.child=No(n,null,o,c)):nt(e,n,o,c),n.memoizedState=d.state,e=n.child):e=On(e,n,c),e}function hf(e,n,o,a){return Tr(),n.flags|=256,nt(e,n,o,a),n.child}var qc={dehydrated:null,treeContext:null,retryLane:0,hydrationErrors:null};function Uc(e){return{baseLanes:e,cachePool:$p()}}function Wc(e,n,o){return e=e!==null?e.childLanes&~o:0,n&&(e|=Qt),e}function ff(e,n,o){var a=n.pendingProps,c=!1,d=(n.flags&128)!==0,m;if((m=d)||(m=e!==null&&e.memoizedState===null?!1:(Ye.current&2)!==0),m&&(c=!0,n.flags&=-129),m=(n.flags&32)!==0,n.flags&=-33,e===null){if(be){if(c?$n(n):ei(),be){var S=Le,C;if(C=S){e:{for(C=S,S=un;C.nodeType!==8;){if(!S){S=null;break e}if(C=tn(C.nextSibling),C===null){S=null;break e}}S=C}S!==null?(n.memoizedState={dehydrated:S,treeContext:Xi!==null?{id:Tn,overflow:vn}:null,retryLane:536870912,hydrationErrors:null},C=Xt(18,null,null,0),C.stateNode=S,C.return=n,n.child=C,gt=n,Le=null,C=!0):C=!1}C||Fi(n)}if(S=n.memoizedState,S!==null&&(S=S.dehydrated,S!==null))return _u(S)?n.lanes=32:n.lanes=536870912,null;Nn(n)}return S=a.children,a=a.fallback,c?(ei(),c=n.mode,S=ga({mode:"hidden",children:S},c),a=Pi(a,c,o,null),S.return=n,a.return=n,S.sibling=a,n.child=S,c=n.child,c.memoizedState=Uc(o),c.childLanes=Wc(e,m,o),n.memoizedState=qc,a):($n(n),jc(n,S))}if(C=e.memoizedState,C!==null&&(S=C.dehydrated,S!==null)){if(d)n.flags&256?($n(n),n.flags&=-257,n=kc(e,n,o)):n.memoizedState!==null?(ei(),n.child=e.child,n.flags|=128,n=null):(ei(),c=a.fallback,S=n.mode,a=ga({mode:"visible",children:a.children},S),c=Pi(c,S,o,null),c.flags|=2,a.return=n,c.return=n,a.sibling=c,n.child=a,No(n,e.child,null,o),a=n.child,a.memoizedState=Uc(o),a.childLanes=Wc(e,m,o),n.memoizedState=qc,n=c);else if($n(n),_u(S)){if(m=S.nextSibling&&S.nextSibling.dataset,m)var F=m.dgst;m=F,a=Error(s(419)),a.stack="",a.digest=m,vr({value:a,source:null,stack:null}),n=kc(e,n,o)}else if(Je||xr(e,n,o,!1),m=(o&e.childLanes)!==0,Je||m){if(m=Ne,m!==null&&(a=o&-o,a=(a&42)!==0?1:Tl(a),a=(a&(m.suspendedLanes|o))!==0?0:a,a!==0&&a!==C.retryLane))throw C.retryLane=a,Io(e,a),Ht(m,e,a),rf;S.data==="$?"||lu(),n=kc(e,n,o)}else S.data==="$?"?(n.flags|=192,n.child=e.child,n=null):(e=C.treeContext,Le=tn(S.nextSibling),gt=n,be=!0,Oi=null,un=!1,e!==null&&(kt[zt++]=Tn,kt[zt++]=vn,kt[zt++]=Xi,Tn=e.id,vn=e.overflow,Xi=n),n=jc(n,a.children),n.flags|=4096);return n}return c?(ei(),c=a.fallback,S=n.mode,C=e.child,F=C.sibling,a=_n(C,{mode:"hidden",children:a.children}),a.subtreeFlags=C.subtreeFlags&65011712,F!==null?c=_n(F,c):(c=Pi(c,S,o,null),c.flags|=2),c.return=n,a.return=n,a.sibling=c,n.child=a,a=c,c=n.child,S=e.child.memoizedState,S===null?S=Uc(o):(C=S.cachePool,C!==null?(F=Ve._currentValue,C=C.parent!==F?{parent:F,pool:F}:C):C=$p(),S={baseLanes:S.baseLanes|o,cachePool:C}),c.memoizedState=S,c.childLanes=Wc(e,m,o),n.memoizedState=qc,a):($n(n),o=e.child,e=o.sibling,o=_n(o,{mode:"visible",children:a.children}),o.return=n,o.sibling=null,e!==null&&(m=n.deletions,m===null?(n.deletions=[e],n.flags|=16):m.push(e)),n.child=o,n.memoizedState=null,o)}function jc(e,n){return n=ga({mode:"visible",children:n},e.mode),n.return=e,e.child=n}function ga(e,n){return e=Xt(22,e,null,n),e.lanes=0,e.stateNode={_visibility:1,_pendingMarkers:null,_retryCache:null,_transitions:null},e}function kc(e,n,o){return No(n,e.child,null,o),e=jc(n,n.pendingProps.children),e.flags|=2,n.memoizedState=null,e}function gf(e,n,o){e.lanes|=n;var a=e.alternate;a!==null&&(a.lanes|=n),lc(e.return,n,o)}function zc(e,n,o,a,c){var d=e.memoizedState;d===null?e.memoizedState={isBackwards:n,rendering:null,renderingStartTime:0,last:a,tail:o,tailMode:c}:(d.isBackwards=n,d.rendering=null,d.renderingStartTime=0,d.last=a,d.tail=o,d.tailMode=c)}function mf(e,n,o){var a=n.pendingProps,c=a.revealOrder,d=a.tail;if(nt(e,n,a.children,o),a=Ye.current,(a&2)!==0)a=a&1|2,n.flags|=128;else{if(e!==null&&(e.flags&128)!==0)e:for(e=n.child;e!==null;){if(e.tag===13)e.memoizedState!==null&&gf(e,o,n);else if(e.tag===19)gf(e,o,n);else if(e.child!==null){e.child.return=e,e=e.child;continue}if(e===n)break e;for(;e.sibling===null;){if(e.return===null||e.return===n)break e;e=e.return}e.sibling.return=e.return,e=e.sibling}a&=1}switch(G(Ye,a),c){case"forwards":for(o=n.child,c=null;o!==null;)e=o.alternate,e!==null&&da(e)===null&&(c=o),o=o.sibling;o=c,o===null?(c=n.child,n.child=null):(c=o.sibling,o.sibling=null),zc(n,!1,c,o,d);break;case"backwards":for(o=null,c=n.child,n.child=null;c!==null;){if(e=c.alternate,e!==null&&da(e)===null){n.child=c;break}e=c.sibling,c.sibling=o,o=c,c=e}zc(n,!0,o,null,d);break;case"together":zc(n,!1,null,null,void 0);break;default:n.memoizedState=null}return n.child}function On(e,n,o){if(e!==null&&(n.dependencies=e.dependencies),ri|=n.lanes,(o&n.childLanes)===0)if(e!==null){if(xr(e,n,o,!1),(o&n.childLanes)===0)return null}else return null;if(e!==null&&n.child!==e.child)throw Error(s(153));if(n.child!==null){for(e=n.child,o=_n(e,e.pendingProps),n.child=o,o.return=n;e.sibling!==null;)e=e.sibling,o=o.sibling=_n(e,e.pendingProps),o.return=n;o.sibling=null}return n.child}function Gc(e,n){return(e.lanes&n)!==0?!0:(e=e.dependencies,!!(e!==null&&Ys(e)))}function K0(e,n,o){switch(n.tag){case 3:Fe(n,n.stateNode.containerInfo),Vn(n,Ve,e.memoizedState.cache),Tr();break;case 27:case 5:yl(n);break;case 4:Fe(n,n.stateNode.containerInfo);break;case 10:Vn(n,n.type,n.memoizedProps.value);break;case 13:var a=n.memoizedState;if(a!==null)return a.dehydrated!==null?($n(n),n.flags|=128,null):(o&n.child.childLanes)!==0?ff(e,n,o):($n(n),e=On(e,n,o),e!==null?e.sibling:null);$n(n);break;case 19:var c=(e.flags&128)!==0;if(a=(o&n.childLanes)!==0,a||(xr(e,n,o,!1),a=(o&n.childLanes)!==0),c){if(a)return mf(e,n,o);n.flags|=128}if(c=n.memoizedState,c!==null&&(c.rendering=null,c.tail=null,c.lastEffect=null),G(Ye,Ye.current),a)break;return null;case 22:case 23:return n.lanes=0,cf(e,n,o);case 24:Vn(n,Ve,e.memoizedState.cache)}return On(e,n,o)}function Ef(e,n,o){if(e!==null)if(e.memoizedProps!==n.pendingProps)Je=!0;else{if(!Gc(e,o)&&(n.flags&128)===0)return Je=!1,K0(e,n,o);Je=(e.flags&131072)!==0}else Je=!1,be&&(n.flags&1048576)!==0&&Gp(n,Vs,n.index);switch(n.lanes=0,n.tag){case 16:e:{e=n.pendingProps;var a=n.elementType,c=a._init;if(a=c(a._payload),n.type=a,typeof a=="function")tc(a)?(e=Bi(a,e),n.tag=1,n=pf(null,n,a,e,o)):(n.tag=0,n=Mc(null,n,a,e,o));else{if(a!=null){if(c=a.$$typeof,c===U){n.tag=11,n=sf(null,n,a,e,o);break e}else if(c===Z){n.tag=14,n=af(null,n,a,e,o);break e}}throw n=qe(a)||a,Error(s(306,n,""))}}return n;case 0:return Mc(e,n,n.type,n.pendingProps,o);case 1:return a=n.type,c=Bi(a,n.pendingProps),pf(e,n,a,c,o);case 3:e:{if(Fe(n,n.stateNode.containerInfo),e===null)throw Error(s(387));a=n.pendingProps;var d=n.memoizedState;c=d.element,gc(e,n),Hr(n,a,null,o);var m=n.memoizedState;if(a=m.cache,Vn(n,Ve,a),a!==d.cache&&cc(n,[Ve],o,!0),Dr(),a=m.element,d.isDehydrated)if(d={element:a,isDehydrated:!1,cache:m.cache},n.updateQueue.baseState=d,n.memoizedState=d,n.flags&256){n=hf(e,n,a,o);break e}else if(a!==c){c=Wt(Error(s(424)),n),vr(c),n=hf(e,n,a,o);break e}else{switch(e=n.stateNode.containerInfo,e.nodeType){case 9:e=e.body;break;default:e=e.nodeName==="HTML"?e.ownerDocument.body:e}for(Le=tn(e.firstChild),gt=n,be=!0,Oi=null,un=!0,o=Vh(n,null,a,o),n.child=o;o;)o.flags=o.flags&-3|4096,o=o.sibling}else{if(Tr(),a===c){n=On(e,n,o);break e}nt(e,n,a,o)}n=n.child}return n;case 26:return fa(e,n),e===null?(o=yg(n.type,null,n.pendingProps,null))?n.memoizedState=o:be||(o=n.type,e=n.pendingProps,a=Pa(le.current).createElement(o),a[st]=n,a[St]=e,ot(a,o,e),Ze(a),n.stateNode=a):n.memoizedState=yg(n.type,e.memoizedProps,n.pendingProps,e.memoizedState),null;case 27:return yl(n),e===null&&be&&(a=n.stateNode=Sg(n.type,n.pendingProps,le.current),gt=n,un=!0,c=Le,ci(n.type)?(Tu=c,Le=tn(a.firstChild)):Le=c),nt(e,n,n.pendingProps.children,o),fa(e,n),e===null&&(n.flags|=4194304),n.child;case 5:return e===null&&be&&((c=a=Le)&&(a=wR(a,n.type,n.pendingProps,un),a!==null?(n.stateNode=a,gt=n,Le=tn(a.firstChild),un=!1,c=!0):c=!1),c||Fi(n)),yl(n),c=n.type,d=n.pendingProps,m=e!==null?e.memoizedProps:null,a=d.children,bu(c,d)?a=null:m!==null&&bu(c,m)&&(n.flags|=32),n.memoizedState!==null&&(c=yc(e,n,W0,null,null,o),is._currentValue=c),fa(e,n),nt(e,n,a,o),n.child;case 6:return e===null&&be&&((e=o=Le)&&(o=_R(o,n.pendingProps,un),o!==null?(n.stateNode=o,gt=n,Le=null,e=!0):e=!1),e||Fi(n)),null;case 13:return ff(e,n,o);case 4:return Fe(n,n.stateNode.containerInfo),a=n.pendingProps,e===null?n.child=No(n,null,a,o):nt(e,n,a,o),n.child;case 11:return sf(e,n,n.type,n.pendingProps,o);case 7:return nt(e,n,n.pendingProps,o),n.child;case 8:return nt(e,n,n.pendingProps.children,o),n.child;case 12:return nt(e,n,n.pendingProps.children,o),n.child;case 10:return a=n.pendingProps,Vn(n,n.type,a.value),nt(e,n,a.children,o),n.child;case 9:return c=n.type._context,a=n.pendingProps.children,Hi(n),c=at(c),a=a(c),n.flags|=1,nt(e,n,a,o),n.child;case 14:return af(e,n,n.type,n.pendingProps,o);case 15:return lf(e,n,n.type,n.pendingProps,o);case 19:return mf(e,n,o);case 31:return a=n.pendingProps,o=n.mode,a={mode:a.mode,children:a.children},e===null?(o=ga(a,o),o.ref=n.ref,n.child=o,o.return=n,n=o):(o=_n(e.child,a),o.ref=n.ref,n.child=o,o.return=n,n=o),n;case 22:return cf(e,n,o);case 24:return Hi(n),a=at(Ve),e===null?(c=pc(),c===null&&(c=Ne,d=uc(),c.pooledCache=d,d.refCount++,d!==null&&(c.pooledCacheLanes|=o),c=d),n.memoizedState={parent:a,cache:c},fc(n),Vn(n,Ve,c)):((e.lanes&o)!==0&&(gc(e,n),Hr(n,null,null,o),Dr()),c=e.memoizedState,d=n.memoizedState,c.parent!==a?(c={parent:a,cache:a},n.memoizedState=c,n.lanes===0&&(n.memoizedState=n.updateQueue.baseState=c),Vn(n,Ve,a)):(a=d.cache,Vn(n,Ve,a),a!==c.cache&&cc(n,[Ve],o,!0))),nt(e,n,n.pendingProps.children,o),n.child;case 29:throw n.pendingProps}throw Error(s(156,n.tag))}function Fn(e){e.flags|=4}function Sf(e,n){if(n.type!=="stylesheet"||(n.state.loading&4)!==0)e.flags&=-16777217;else if(e.flags|=16777216,!Tg(n)){if(n=Gt.current,n!==null&&((Se&4194048)===Se?dn!==null:(Se&62914560)!==Se&&(Se&536870912)===0||n!==dn))throw Or=hc,eh;e.flags|=8192}}function ma(e,n){n!==null&&(e.flags|=4),e.flags&16384&&(n=e.tag!==22?Yd():536870912,e.lanes|=n,Ho|=n)}function Wr(e,n){if(!be)switch(e.tailMode){case"hidden":n=e.tail;for(var o=null;n!==null;)n.alternate!==null&&(o=n),n=n.sibling;o===null?e.tail=null:o.sibling=null;break;case"collapsed":o=e.tail;for(var a=null;o!==null;)o.alternate!==null&&(a=o),o=o.sibling;a===null?n||e.tail===null?e.tail=null:e.tail.sibling=null:a.sibling=null}}function He(e){var n=e.alternate!==null&&e.alternate.child===e.child,o=0,a=0;if(n)for(var c=e.child;c!==null;)o|=c.lanes|c.childLanes,a|=c.subtreeFlags&65011712,a|=c.flags&65011712,c.return=e,c=c.sibling;else for(c=e.child;c!==null;)o|=c.lanes|c.childLanes,a|=c.subtreeFlags,a|=c.flags,c.return=e,c=c.sibling;return e.subtreeFlags|=a,e.childLanes=o,n}function Z0(e,n,o){var a=n.pendingProps;switch(rc(n),n.tag){case 31:case 16:case 15:case 0:case 11:case 7:case 8:case 12:case 9:case 14:return He(n),null;case 1:return He(n),null;case 3:return o=n.stateNode,a=null,e!==null&&(a=e.memoizedState.cache),n.memoizedState.cache!==a&&(n.flags|=2048),Pn(Ve),kn(),o.pendingContext&&(o.context=o.pendingContext,o.pendingContext=null),(e===null||e.child===null)&&(_r(n)?Fn(n):e===null||e.memoizedState.isDehydrated&&(n.flags&256)===0||(n.flags|=1024,Yp())),He(n),null;case 26:return o=n.memoizedState,e===null?(Fn(n),o!==null?(He(n),Sf(n,o)):(He(n),n.flags&=-16777217)):o?o!==e.memoizedState?(Fn(n),He(n),Sf(n,o)):(He(n),n.flags&=-16777217):(e.memoizedProps!==a&&Fn(n),He(n),n.flags&=-16777217),null;case 27:vs(n),o=le.current;var c=n.type;if(e!==null&&n.stateNode!=null)e.memoizedProps!==a&&Fn(n);else{if(!a){if(n.stateNode===null)throw Error(s(166));return He(n),null}e=$.current,_r(n)?Qp(n):(e=Sg(c,a,o),n.stateNode=e,Fn(n))}return He(n),null;case 5:if(vs(n),o=n.type,e!==null&&n.stateNode!=null)e.memoizedProps!==a&&Fn(n);else{if(!a){if(n.stateNode===null)throw Error(s(166));return He(n),null}if(e=$.current,_r(n))Qp(n);else{switch(c=Pa(le.current),e){case 1:e=c.createElementNS("http://www.w3.org/2000/svg",o);break;case 2:e=c.createElementNS("http://www.w3.org/1998/Math/MathML",o);break;default:switch(o){case"svg":e=c.createElementNS("http://www.w3.org/2000/svg",o);break;case"math":e=c.createElementNS("http://www.w3.org/1998/Math/MathML",o);break;case"script":e=c.createElement("div"),e.innerHTML="<script><\/script>",e=e.removeChild(e.firstChild);break;case"select":e=typeof a.is=="string"?c.createElement("select",{is:a.is}):c.createElement("select"),a.multiple?e.multiple=!0:a.size&&(e.size=a.size);break;default:e=typeof a.is=="string"?c.createElement(o,{is:a.is}):c.createElement(o)}}e[st]=n,e[St]=a;e:for(c=n.child;c!==null;){if(c.tag===5||c.tag===6)e.appendChild(c.stateNode);else if(c.tag!==4&&c.tag!==27&&c.child!==null){c.child.return=c,c=c.child;continue}if(c===n)break e;for(;c.sibling===null;){if(c.return===null||c.return===n)break e;c=c.return}c.sibling.return=c.return,c=c.sibling}n.stateNode=e;e:switch(ot(e,o,a),o){case"button":case"input":case"select":case"textarea":e=!!a.autoFocus;break e;case"img":e=!0;break e;default:e=!1}e&&Fn(n)}}return He(n),n.flags&=-16777217,null;case 6:if(e&&n.stateNode!=null)e.memoizedProps!==a&&Fn(n);else{if(typeof a!="string"&&n.stateNode===null)throw Error(s(166));if(e=le.current,_r(n)){if(e=n.stateNode,o=n.memoizedProps,a=null,c=gt,c!==null)switch(c.tag){case 27:case 5:a=c.memoizedProps}e[st]=n,e=!!(e.nodeValue===o||a!==null&&a.suppressHydrationWarning===!0||dg(e.nodeValue,o)),e||Fi(n)}else e=Pa(e).createTextNode(a),e[st]=n,n.stateNode=e}return He(n),null;case 13:if(a=n.memoizedState,e===null||e.memoizedState!==null&&e.memoizedState.dehydrated!==null){if(c=_r(n),a!==null&&a.dehydrated!==null){if(e===null){if(!c)throw Error(s(318));if(c=n.memoizedState,c=c!==null?c.dehydrated:null,!c)throw Error(s(317));c[st]=n}else Tr(),(n.flags&128)===0&&(n.memoizedState=null),n.flags|=4;He(n),c=!1}else c=Yp(),e!==null&&e.memoizedState!==null&&(e.memoizedState.hydrationErrors=c),c=!0;if(!c)return n.flags&256?(Nn(n),n):(Nn(n),null)}if(Nn(n),(n.flags&128)!==0)return n.lanes=o,n;if(o=a!==null,e=e!==null&&e.memoizedState!==null,o){a=n.child,c=null,a.alternate!==null&&a.alternate.memoizedState!==null&&a.alternate.memoizedState.cachePool!==null&&(c=a.alternate.memoizedState.cachePool.pool);var d=null;a.memoizedState!==null&&a.memoizedState.cachePool!==null&&(d=a.memoizedState.cachePool.pool),d!==c&&(a.flags|=2048)}return o!==e&&o&&(n.child.flags|=8192),ma(n,n.updateQueue),He(n),null;case 4:return kn(),e===null&&Eu(n.stateNode.containerInfo),He(n),null;case 10:return Pn(n.type),He(n),null;case 19:if(z(Ye),c=n.memoizedState,c===null)return He(n),null;if(a=(n.flags&128)!==0,d=c.rendering,d===null)if(a)Wr(c,!1);else{if(Be!==0||e!==null&&(e.flags&128)!==0)for(e=n.child;e!==null;){if(d=da(e),d!==null){for(n.flags|=128,Wr(c,!1),e=d.updateQueue,n.updateQueue=e,ma(n,e),n.subtreeFlags=0,e=o,o=n.child;o!==null;)zp(o,e),o=o.sibling;return G(Ye,Ye.current&1|2),n.child}e=e.sibling}c.tail!==null&&cn()>Ra&&(n.flags|=128,a=!0,Wr(c,!1),n.lanes=4194304)}else{if(!a)if(e=da(d),e!==null){if(n.flags|=128,a=!0,e=e.updateQueue,n.updateQueue=e,ma(n,e),Wr(c,!0),c.tail===null&&c.tailMode==="hidden"&&!d.alternate&&!be)return He(n),null}else 2*cn()-c.renderingStartTime>Ra&&o!==536870912&&(n.flags|=128,a=!0,Wr(c,!1),n.lanes=4194304);c.isBackwards?(d.sibling=n.child,n.child=d):(e=c.last,e!==null?e.sibling=d:n.child=d,c.last=d)}return c.tail!==null?(n=c.tail,c.rendering=n,c.tail=n.sibling,c.renderingStartTime=cn(),n.sibling=null,e=Ye.current,G(Ye,a?e&1|2:e&1),n):(He(n),null);case 22:case 23:return Nn(n),Rc(),a=n.memoizedState!==null,e!==null?e.memoizedState!==null!==a&&(n.flags|=8192):a&&(n.flags|=8192),a?(o&536870912)!==0&&(n.flags&128)===0&&(He(n),n.subtreeFlags&6&&(n.flags|=8192)):He(n),o=n.updateQueue,o!==null&&ma(n,o.retryQueue),o=null,e!==null&&e.memoizedState!==null&&e.memoizedState.cachePool!==null&&(o=e.memoizedState.cachePool.pool),a=null,n.memoizedState!==null&&n.memoizedState.cachePool!==null&&(a=n.memoizedState.cachePool.pool),a!==o&&(n.flags|=2048),e!==null&&z(Ai),null;case 24:return o=null,e!==null&&(o=e.memoizedState.cache),n.memoizedState.cache!==o&&(n.flags|=2048),Pn(Ve),He(n),null;case 25:return null;case 30:return null}throw Error(s(156,n.tag))}function J0(e,n){switch(rc(n),n.tag){case 1:return e=n.flags,e&65536?(n.flags=e&-65537|128,n):null;case 3:return Pn(Ve),kn(),e=n.flags,(e&65536)!==0&&(e&128)===0?(n.flags=e&-65537|128,n):null;case 26:case 27:case 5:return vs(n),null;case 13:if(Nn(n),e=n.memoizedState,e!==null&&e.dehydrated!==null){if(n.alternate===null)throw Error(s(340));Tr()}return e=n.flags,e&65536?(n.flags=e&-65537|128,n):null;case 19:return z(Ye),null;case 4:return kn(),null;case 10:return Pn(n.type),null;case 22:case 23:return Nn(n),Rc(),e!==null&&z(Ai),e=n.flags,e&65536?(n.flags=e&-65537|128,n):null;case 24:return Pn(Ve),null;case 25:return null;default:return null}}function Rf(e,n){switch(rc(n),n.tag){case 3:Pn(Ve),kn();break;case 26:case 27:case 5:vs(n);break;case 4:kn();break;case 13:Nn(n);break;case 19:z(Ye);break;case 10:Pn(n.type);break;case 22:case 23:Nn(n),Rc(),e!==null&&z(Ai);break;case 24:Pn(Ve)}}function jr(e,n){try{var o=n.updateQueue,a=o!==null?o.lastEffect:null;if(a!==null){var c=a.next;o=c;do{if((o.tag&e)===e){a=void 0;var d=o.create,m=o.inst;a=d(),m.destroy=a}o=o.next}while(o!==c)}}catch(S){Pe(n,n.return,S)}}function ti(e,n,o){try{var a=n.updateQueue,c=a!==null?a.lastEffect:null;if(c!==null){var d=c.next;a=d;do{if((a.tag&e)===e){var m=a.inst,S=m.destroy;if(S!==void 0){m.destroy=void 0,c=n;var C=o,F=S;try{F()}catch(M){Pe(c,C,M)}}}a=a.next}while(a!==d)}}catch(M){Pe(n,n.return,M)}}function If(e){var n=e.updateQueue;if(n!==null){var o=e.stateNode;try{sh(n,o)}catch(a){Pe(e,e.return,a)}}}function yf(e,n,o){o.props=Bi(e.type,e.memoizedProps),o.state=e.memoizedState;try{o.componentWillUnmount()}catch(a){Pe(e,n,a)}}function kr(e,n){try{var o=e.ref;if(o!==null){switch(e.tag){case 26:case 27:case 5:var a=e.stateNode;break;case 30:a=e.stateNode;break;default:a=e.stateNode}typeof o=="function"?e.refCleanup=o(a):o.current=a}}catch(c){Pe(e,n,c)}}function pn(e,n){var o=e.ref,a=e.refCleanup;if(o!==null)if(typeof a=="function")try{a()}catch(c){Pe(e,n,c)}finally{e.refCleanup=null,e=e.alternate,e!=null&&(e.refCleanup=null)}else if(typeof o=="function")try{o(null)}catch(c){Pe(e,n,c)}else o.current=null}function bf(e){var n=e.type,o=e.memoizedProps,a=e.stateNode;try{e:switch(n){case"button":case"input":case"select":case"textarea":o.autoFocus&&a.focus();break e;case"img":o.src?a.src=o.src:o.srcSet&&(a.srcset=o.srcSet)}}catch(c){Pe(e,e.return,c)}}function Qc(e,n,o){try{var a=e.stateNode;RR(a,e.type,o,n),a[St]=n}catch(c){Pe(e,e.return,c)}}function Cf(e){return e.tag===5||e.tag===3||e.tag===26||e.tag===27&&ci(e.type)||e.tag===4}function Vc(e){e:for(;;){for(;e.sibling===null;){if(e.return===null||Cf(e.return))return null;e=e.return}for(e.sibling.return=e.return,e=e.sibling;e.tag!==5&&e.tag!==6&&e.tag!==18;){if(e.tag===27&&ci(e.type)||e.flags&2||e.child===null||e.tag===4)continue e;e.child.return=e,e=e.child}if(!(e.flags&2))return e.stateNode}}function Yc(e,n,o){var a=e.tag;if(a===5||a===6)e=e.stateNode,n?(o.nodeType===9?o.body:o.nodeName==="HTML"?o.ownerDocument.body:o).insertBefore(e,n):(n=o.nodeType===9?o.body:o.nodeName==="HTML"?o.ownerDocument.body:o,n.appendChild(e),o=o._reactRootContainer,o!=null||n.onclick!==null||(n.onclick=xa));else if(a!==4&&(a===27&&ci(e.type)&&(o=e.stateNode,n=null),e=e.child,e!==null))for(Yc(e,n,o),e=e.sibling;e!==null;)Yc(e,n,o),e=e.sibling}function Ea(e,n,o){var a=e.tag;if(a===5||a===6)e=e.stateNode,n?o.insertBefore(e,n):o.appendChild(e);else if(a!==4&&(a===27&&ci(e.type)&&(o=e.stateNode),e=e.child,e!==null))for(Ea(e,n,o),e=e.sibling;e!==null;)Ea(e,n,o),e=e.sibling}function wf(e){var n=e.stateNode,o=e.memoizedProps;try{for(var a=e.type,c=n.attributes;c.length;)n.removeAttributeNode(c[0]);ot(n,a,o),n[st]=e,n[St]=o}catch(d){Pe(e,e.return,d)}}var Dn=!1,We=!1,Kc=!1,_f=typeof WeakSet=="function"?WeakSet:Set,$e=null;function $0(e,n){if(e=e.containerInfo,Iu=Ha,e=Hp(e),Vl(e)){if("selectionStart"in e)var o={start:e.selectionStart,end:e.selectionEnd};else e:{o=(o=e.ownerDocument)&&o.defaultView||window;var a=o.getSelection&&o.getSelection();if(a&&a.rangeCount!==0){o=a.anchorNode;var c=a.anchorOffset,d=a.focusNode;a=a.focusOffset;try{o.nodeType,d.nodeType}catch{o=null;break e}var m=0,S=-1,C=-1,F=0,M=0,W=e,D=null;t:for(;;){for(var H;W!==o||c!==0&&W.nodeType!==3||(S=m+c),W!==d||a!==0&&W.nodeType!==3||(C=m+a),W.nodeType===3&&(m+=W.nodeValue.length),(H=W.firstChild)!==null;)D=W,W=H;for(;;){if(W===e)break t;if(D===o&&++F===c&&(S=m),D===d&&++M===a&&(C=m),(H=W.nextSibling)!==null)break;W=D,D=W.parentNode}W=H}o=S===-1||C===-1?null:{start:S,end:C}}else o=null}o=o||{start:0,end:0}}else o=null;for(yu={focusedElem:e,selectionRange:o},Ha=!1,$e=n;$e!==null;)if(n=$e,e=n.child,(n.subtreeFlags&1024)!==0&&e!==null)e.return=n,$e=e;else for(;$e!==null;){switch(n=$e,d=n.alternate,e=n.flags,n.tag){case 0:break;case 11:case 15:break;case 1:if((e&1024)!==0&&d!==null){e=void 0,o=n,c=d.memoizedProps,d=d.memoizedState,a=o.stateNode;try{var ae=Bi(o.type,c,o.elementType===o.type);e=a.getSnapshotBeforeUpdate(ae,d),a.__reactInternalSnapshotBeforeUpdate=e}catch(oe){Pe(o,o.return,oe)}}break;case 3:if((e&1024)!==0){if(e=n.stateNode.containerInfo,o=e.nodeType,o===9)wu(e);else if(o===1)switch(e.nodeName){case"HEAD":case"HTML":case"BODY":wu(e);break;default:e.textContent=""}}break;case 5:case 26:case 27:case 6:case 4:case 17:break;default:if((e&1024)!==0)throw Error(s(163))}if(e=n.sibling,e!==null){e.return=n.return,$e=e;break}$e=n.return}}function Tf(e,n,o){var a=o.flags;switch(o.tag){case 0:case 11:case 15:ni(e,o),a&4&&jr(5,o);break;case 1:if(ni(e,o),a&4)if(e=o.stateNode,n===null)try{e.componentDidMount()}catch(m){Pe(o,o.return,m)}else{var c=Bi(o.type,n.memoizedProps);n=n.memoizedState;try{e.componentDidUpdate(c,n,e.__reactInternalSnapshotBeforeUpdate)}catch(m){Pe(o,o.return,m)}}a&64&&If(o),a&512&&kr(o,o.return);break;case 3:if(ni(e,o),a&64&&(e=o.updateQueue,e!==null)){if(n=null,o.child!==null)switch(o.child.tag){case 27:case 5:n=o.child.stateNode;break;case 1:n=o.child.stateNode}try{sh(e,n)}catch(m){Pe(o,o.return,m)}}break;case 27:n===null&&a&4&&wf(o);case 26:case 5:ni(e,o),n===null&&a&4&&bf(o),a&512&&kr(o,o.return);break;case 12:ni(e,o);break;case 13:ni(e,o),a&4&&Pf(e,o),a&64&&(e=o.memoizedState,e!==null&&(e=e.dehydrated,e!==null&&(o=lR.bind(null,o),TR(e,o))));break;case 22:if(a=o.memoizedState!==null||Dn,!a){n=n!==null&&n.memoizedState!==null||We,c=Dn;var d=We;Dn=a,(We=n)&&!d?ii(e,o,(o.subtreeFlags&8772)!==0):ni(e,o),Dn=c,We=d}break;case 30:break;default:ni(e,o)}}function vf(e){var n=e.alternate;n!==null&&(e.alternate=null,vf(n)),e.child=null,e.deletions=null,e.sibling=null,e.tag===5&&(n=e.stateNode,n!==null&&Pl(n)),e.stateNode=null,e.return=null,e.dependencies=null,e.memoizedProps=null,e.memoizedState=null,e.pendingProps=null,e.stateNode=null,e.updateQueue=null}var De=null,yt=!1;function Hn(e,n,o){for(o=o.child;o!==null;)xf(e,n,o),o=o.sibling}function xf(e,n,o){if(vt&&typeof vt.onCommitFiberUnmount=="function")try{vt.onCommitFiberUnmount(dr,o)}catch{}switch(o.tag){case 26:We||pn(o,n),Hn(e,n,o),o.memoizedState?o.memoizedState.count--:o.stateNode&&(o=o.stateNode,o.parentNode.removeChild(o));break;case 27:We||pn(o,n);var a=De,c=yt;ci(o.type)&&(De=o.stateNode,yt=!1),Hn(e,n,o),$r(o.stateNode),De=a,yt=c;break;case 5:We||pn(o,n);case 6:if(a=De,c=yt,De=null,Hn(e,n,o),De=a,yt=c,De!==null)if(yt)try{(De.nodeType===9?De.body:De.nodeName==="HTML"?De.ownerDocument.body:De).removeChild(o.stateNode)}catch(d){Pe(o,n,d)}else try{De.removeChild(o.stateNode)}catch(d){Pe(o,n,d)}break;case 18:De!==null&&(yt?(e=De,mg(e.nodeType===9?e.body:e.nodeName==="HTML"?e.ownerDocument.body:e,o.stateNode),as(e)):mg(De,o.stateNode));break;case 4:a=De,c=yt,De=o.stateNode.containerInfo,yt=!0,Hn(e,n,o),De=a,yt=c;break;case 0:case 11:case 14:case 15:We||ti(2,o,n),We||ti(4,o,n),Hn(e,n,o);break;case 1:We||(pn(o,n),a=o.stateNode,typeof a.componentWillUnmount=="function"&&yf(o,n,a)),Hn(e,n,o);break;case 21:Hn(e,n,o);break;case 22:We=(a=We)||o.memoizedState!==null,Hn(e,n,o),We=a;break;default:Hn(e,n,o)}}function Pf(e,n){if(n.memoizedState===null&&(e=n.alternate,e!==null&&(e=e.memoizedState,e!==null&&(e=e.dehydrated,e!==null))))try{as(e)}catch(o){Pe(n,n.return,o)}}function eR(e){switch(e.tag){case 13:case 19:var n=e.stateNode;return n===null&&(n=e.stateNode=new _f),n;case 22:return e=e.stateNode,n=e._retryCache,n===null&&(n=e._retryCache=new _f),n;default:throw Error(s(435,e.tag))}}function Zc(e,n){var o=eR(e);n.forEach(function(a){var c=cR.bind(null,e,a);o.has(a)||(o.add(a),a.then(c,c))})}function Nt(e,n){var o=n.deletions;if(o!==null)for(var a=0;a<o.length;a++){var c=o[a],d=e,m=n,S=m;e:for(;S!==null;){switch(S.tag){case 27:if(ci(S.type)){De=S.stateNode,yt=!1;break e}break;case 5:De=S.stateNode,yt=!1;break e;case 3:case 4:De=S.stateNode.containerInfo,yt=!0;break e}S=S.return}if(De===null)throw Error(s(160));xf(d,m,c),De=null,yt=!1,d=c.alternate,d!==null&&(d.return=null),c.return=null}if(n.subtreeFlags&13878)for(n=n.child;n!==null;)Xf(n,e),n=n.sibling}var en=null;function Xf(e,n){var o=e.alternate,a=e.flags;switch(e.tag){case 0:case 11:case 14:case 15:Nt(n,e),Ot(e),a&4&&(ti(3,e,e.return),jr(3,e),ti(5,e,e.return));break;case 1:Nt(n,e),Ot(e),a&512&&(We||o===null||pn(o,o.return)),a&64&&Dn&&(e=e.updateQueue,e!==null&&(a=e.callbacks,a!==null&&(o=e.shared.hiddenCallbacks,e.shared.hiddenCallbacks=o===null?a:o.concat(a))));break;case 26:var c=en;if(Nt(n,e),Ot(e),a&512&&(We||o===null||pn(o,o.return)),a&4){var d=o!==null?o.memoizedState:null;if(a=e.memoizedState,o===null)if(a===null)if(e.stateNode===null){e:{a=e.type,o=e.memoizedProps,c=c.ownerDocument||c;t:switch(a){case"title":d=c.getElementsByTagName("title")[0],(!d||d[fr]||d[st]||d.namespaceURI==="http://www.w3.org/2000/svg"||d.hasAttribute("itemprop"))&&(d=c.createElement(a),c.head.insertBefore(d,c.querySelector("head > title"))),ot(d,a,o),d[st]=e,Ze(d),a=d;break e;case"link":var m=wg("link","href",c).get(a+(o.href||""));if(m){for(var S=0;S<m.length;S++)if(d=m[S],d.getAttribute("href")===(o.href==null||o.href===""?null:o.href)&&d.getAttribute("rel")===(o.rel==null?null:o.rel)&&d.getAttribute("title")===(o.title==null?null:o.title)&&d.getAttribute("crossorigin")===(o.crossOrigin==null?null:o.crossOrigin)){m.splice(S,1);break t}}d=c.createElement(a),ot(d,a,o),c.head.appendChild(d);break;case"meta":if(m=wg("meta","content",c).get(a+(o.content||""))){for(S=0;S<m.length;S++)if(d=m[S],d.getAttribute("content")===(o.content==null?null:""+o.content)&&d.getAttribute("name")===(o.name==null?null:o.name)&&d.getAttribute("property")===(o.property==null?null:o.property)&&d.getAttribute("http-equiv")===(o.httpEquiv==null?null:o.httpEquiv)&&d.getAttribute("charset")===(o.charSet==null?null:o.charSet)){m.splice(S,1);break t}}d=c.createElement(a),ot(d,a,o),c.head.appendChild(d);break;default:throw Error(s(468,a))}d[st]=e,Ze(d),a=d}e.stateNode=a}else _g(c,e.type,e.stateNode);else e.stateNode=Cg(c,a,e.memoizedProps);else d!==a?(d===null?o.stateNode!==null&&(o=o.stateNode,o.parentNode.removeChild(o)):d.count--,a===null?_g(c,e.type,e.stateNode):Cg(c,a,e.memoizedProps)):a===null&&e.stateNode!==null&&Qc(e,e.memoizedProps,o.memoizedProps)}break;case 27:Nt(n,e),Ot(e),a&512&&(We||o===null||pn(o,o.return)),o!==null&&a&4&&Qc(e,e.memoizedProps,o.memoizedProps);break;case 5:if(Nt(n,e),Ot(e),a&512&&(We||o===null||pn(o,o.return)),e.flags&32){c=e.stateNode;try{ho(c,"")}catch(H){Pe(e,e.return,H)}}a&4&&e.stateNode!=null&&(c=e.memoizedProps,Qc(e,c,o!==null?o.memoizedProps:c)),a&1024&&(Kc=!0);break;case 6:if(Nt(n,e),Ot(e),a&4){if(e.stateNode===null)throw Error(s(162));a=e.memoizedProps,o=e.stateNode;try{o.nodeValue=a}catch(H){Pe(e,e.return,H)}}break;case 3:if(Oa=null,c=en,en=Xa(n.containerInfo),Nt(n,e),en=c,Ot(e),a&4&&o!==null&&o.memoizedState.isDehydrated)try{as(n.containerInfo)}catch(H){Pe(e,e.return,H)}Kc&&(Kc=!1,Nf(e));break;case 4:a=en,en=Xa(e.stateNode.containerInfo),Nt(n,e),Ot(e),en=a;break;case 12:Nt(n,e),Ot(e);break;case 13:Nt(n,e),Ot(e),e.child.flags&8192&&e.memoizedState!==null!=(o!==null&&o.memoizedState!==null)&&(iu=cn()),a&4&&(a=e.updateQueue,a!==null&&(e.updateQueue=null,Zc(e,a)));break;case 22:c=e.memoizedState!==null;var C=o!==null&&o.memoizedState!==null,F=Dn,M=We;if(Dn=F||c,We=M||C,Nt(n,e),We=M,Dn=F,Ot(e),a&8192)e:for(n=e.stateNode,n._visibility=c?n._visibility&-2:n._visibility|1,c&&(o===null||C||Dn||We||Mi(e)),o=null,n=e;;){if(n.tag===5||n.tag===26){if(o===null){C=o=n;try{if(d=C.stateNode,c)m=d.style,typeof m.setProperty=="function"?m.setProperty("display","none","important"):m.display="none";else{S=C.stateNode;var W=C.memoizedProps.style,D=W!=null&&W.hasOwnProperty("display")?W.display:null;S.style.display=D==null||typeof D=="boolean"?"":(""+D).trim()}}catch(H){Pe(C,C.return,H)}}}else if(n.tag===6){if(o===null){C=n;try{C.stateNode.nodeValue=c?"":C.memoizedProps}catch(H){Pe(C,C.return,H)}}}else if((n.tag!==22&&n.tag!==23||n.memoizedState===null||n===e)&&n.child!==null){n.child.return=n,n=n.child;continue}if(n===e)break e;for(;n.sibling===null;){if(n.return===null||n.return===e)break e;o===n&&(o=null),n=n.return}o===n&&(o=null),n.sibling.return=n.return,n=n.sibling}a&4&&(a=e.updateQueue,a!==null&&(o=a.retryQueue,o!==null&&(a.retryQueue=null,Zc(e,o))));break;case 19:Nt(n,e),Ot(e),a&4&&(a=e.updateQueue,a!==null&&(e.updateQueue=null,Zc(e,a)));break;case 30:break;case 21:break;default:Nt(n,e),Ot(e)}}function Ot(e){var n=e.flags;if(n&2){try{for(var o,a=e.return;a!==null;){if(Cf(a)){o=a;break}a=a.return}if(o==null)throw Error(s(160));switch(o.tag){case 27:var c=o.stateNode,d=Vc(e);Ea(e,d,c);break;case 5:var m=o.stateNode;o.flags&32&&(ho(m,""),o.flags&=-33);var S=Vc(e);Ea(e,S,m);break;case 3:case 4:var C=o.stateNode.containerInfo,F=Vc(e);Yc(e,F,C);break;default:throw Error(s(161))}}catch(M){Pe(e,e.return,M)}e.flags&=-3}n&4096&&(e.flags&=-4097)}function Nf(e){if(e.subtreeFlags&1024)for(e=e.child;e!==null;){var n=e;Nf(n),n.tag===5&&n.flags&1024&&n.stateNode.reset(),e=e.sibling}}function ni(e,n){if(n.subtreeFlags&8772)for(n=n.child;n!==null;)Tf(e,n.alternate,n),n=n.sibling}function Mi(e){for(e=e.child;e!==null;){var n=e;switch(n.tag){case 0:case 11:case 14:case 15:ti(4,n,n.return),Mi(n);break;case 1:pn(n,n.return);var o=n.stateNode;typeof o.componentWillUnmount=="function"&&yf(n,n.return,o),Mi(n);break;case 27:$r(n.stateNode);case 26:case 5:pn(n,n.return),Mi(n);break;case 22:n.memoizedState===null&&Mi(n);break;case 30:Mi(n);break;default:Mi(n)}e=e.sibling}}function ii(e,n,o){for(o=o&&(n.subtreeFlags&8772)!==0,n=n.child;n!==null;){var a=n.alternate,c=e,d=n,m=d.flags;switch(d.tag){case 0:case 11:case 15:ii(c,d,o),jr(4,d);break;case 1:if(ii(c,d,o),a=d,c=a.stateNode,typeof c.componentDidMount=="function")try{c.componentDidMount()}catch(F){Pe(a,a.return,F)}if(a=d,c=a.updateQueue,c!==null){var S=a.stateNode;try{var C=c.shared.hiddenCallbacks;if(C!==null)for(c.shared.hiddenCallbacks=null,c=0;c<C.length;c++)rh(C[c],S)}catch(F){Pe(a,a.return,F)}}o&&m&64&&If(d),kr(d,d.return);break;case 27:wf(d);case 26:case 5:ii(c,d,o),o&&a===null&&m&4&&bf(d),kr(d,d.return);break;case 12:ii(c,d,o);break;case 13:ii(c,d,o),o&&m&4&&Pf(c,d);break;case 22:d.memoizedState===null&&ii(c,d,o),kr(d,d.return);break;case 30:break;default:ii(c,d,o)}n=n.sibling}}function Jc(e,n){var o=null;e!==null&&e.memoizedState!==null&&e.memoizedState.cachePool!==null&&(o=e.memoizedState.cachePool.pool),e=null,n.memoizedState!==null&&n.memoizedState.cachePool!==null&&(e=n.memoizedState.cachePool.pool),e!==o&&(e!=null&&e.refCount++,o!=null&&Pr(o))}function $c(e,n){e=null,n.alternate!==null&&(e=n.alternate.memoizedState.cache),n=n.memoizedState.cache,n!==e&&(n.refCount++,e!=null&&Pr(e))}function hn(e,n,o,a){if(n.subtreeFlags&10256)for(n=n.child;n!==null;)Of(e,n,o,a),n=n.sibling}function Of(e,n,o,a){var c=n.flags;switch(n.tag){case 0:case 11:case 15:hn(e,n,o,a),c&2048&&jr(9,n);break;case 1:hn(e,n,o,a);break;case 3:hn(e,n,o,a),c&2048&&(e=null,n.alternate!==null&&(e=n.alternate.memoizedState.cache),n=n.memoizedState.cache,n!==e&&(n.refCount++,e!=null&&Pr(e)));break;case 12:if(c&2048){hn(e,n,o,a),e=n.stateNode;try{var d=n.memoizedProps,m=d.id,S=d.onPostCommit;typeof S=="function"&&S(m,n.alternate===null?"mount":"update",e.passiveEffectDuration,-0)}catch(C){Pe(n,n.return,C)}}else hn(e,n,o,a);break;case 13:hn(e,n,o,a);break;case 23:break;case 22:d=n.stateNode,m=n.alternate,n.memoizedState!==null?d._visibility&2?hn(e,n,o,a):zr(e,n):d._visibility&2?hn(e,n,o,a):(d._visibility|=2,Oo(e,n,o,a,(n.subtreeFlags&10256)!==0)),c&2048&&Jc(m,n);break;case 24:hn(e,n,o,a),c&2048&&$c(n.alternate,n);break;default:hn(e,n,o,a)}}function Oo(e,n,o,a,c){for(c=c&&(n.subtreeFlags&10256)!==0,n=n.child;n!==null;){var d=e,m=n,S=o,C=a,F=m.flags;switch(m.tag){case 0:case 11:case 15:Oo(d,m,S,C,c),jr(8,m);break;case 23:break;case 22:var M=m.stateNode;m.memoizedState!==null?M._visibility&2?Oo(d,m,S,C,c):zr(d,m):(M._visibility|=2,Oo(d,m,S,C,c)),c&&F&2048&&Jc(m.alternate,m);break;case 24:Oo(d,m,S,C,c),c&&F&2048&&$c(m.alternate,m);break;default:Oo(d,m,S,C,c)}n=n.sibling}}function zr(e,n){if(n.subtreeFlags&10256)for(n=n.child;n!==null;){var o=e,a=n,c=a.flags;switch(a.tag){case 22:zr(o,a),c&2048&&Jc(a.alternate,a);break;case 24:zr(o,a),c&2048&&$c(a.alternate,a);break;default:zr(o,a)}n=n.sibling}}var Gr=8192;function Fo(e){if(e.subtreeFlags&Gr)for(e=e.child;e!==null;)Ff(e),e=e.sibling}function Ff(e){switch(e.tag){case 26:Fo(e),e.flags&Gr&&e.memoizedState!==null&&MR(en,e.memoizedState,e.memoizedProps);break;case 5:Fo(e);break;case 3:case 4:var n=en;en=Xa(e.stateNode.containerInfo),Fo(e),en=n;break;case 22:e.memoizedState===null&&(n=e.alternate,n!==null&&n.memoizedState!==null?(n=Gr,Gr=16777216,Fo(e),Gr=n):Fo(e));break;default:Fo(e)}}function Df(e){var n=e.alternate;if(n!==null&&(e=n.child,e!==null)){n.child=null;do n=e.sibling,e.sibling=null,e=n;while(e!==null)}}function Qr(e){var n=e.deletions;if((e.flags&16)!==0){if(n!==null)for(var o=0;o<n.length;o++){var a=n[o];$e=a,Af(a,e)}Df(e)}if(e.subtreeFlags&10256)for(e=e.child;e!==null;)Hf(e),e=e.sibling}function Hf(e){switch(e.tag){case 0:case 11:case 15:Qr(e),e.flags&2048&&ti(9,e,e.return);break;case 3:Qr(e);break;case 12:Qr(e);break;case 22:var n=e.stateNode;e.memoizedState!==null&&n._visibility&2&&(e.return===null||e.return.tag!==13)?(n._visibility&=-3,Sa(e)):Qr(e);break;default:Qr(e)}}function Sa(e){var n=e.deletions;if((e.flags&16)!==0){if(n!==null)for(var o=0;o<n.length;o++){var a=n[o];$e=a,Af(a,e)}Df(e)}for(e=e.child;e!==null;){switch(n=e,n.tag){case 0:case 11:case 15:ti(8,n,n.return),Sa(n);break;case 22:o=n.stateNode,o._visibility&2&&(o._visibility&=-3,Sa(n));break;default:Sa(n)}e=e.sibling}}function Af(e,n){for(;$e!==null;){var o=$e;switch(o.tag){case 0:case 11:case 15:ti(8,o,n);break;case 23:case 22:if(o.memoizedState!==null&&o.memoizedState.cachePool!==null){var a=o.memoizedState.cachePool.pool;a!=null&&a.refCount++}break;case 24:Pr(o.memoizedState.cache)}if(a=o.child,a!==null)a.return=o,$e=a;else e:for(o=e;$e!==null;){a=$e;var c=a.sibling,d=a.return;if(vf(a),a===o){$e=null;break e}if(c!==null){c.return=d,$e=c;break e}$e=d}}}var tR={getCacheForType:function(e){var n=at(Ve),o=n.data.get(e);return o===void 0&&(o=e(),n.data.set(e,o)),o}},nR=typeof WeakMap=="function"?WeakMap:Map,Ce=0,Ne=null,fe=null,Se=0,we=0,Ft=null,oi=!1,Do=!1,eu=!1,An=0,Be=0,ri=0,qi=0,tu=0,Qt=0,Ho=0,Vr=null,bt=null,nu=!1,iu=0,Ra=1/0,Ia=null,si=null,it=0,ai=null,Ao=null,Lo=0,ou=0,ru=null,Lf=null,Yr=0,su=null;function Dt(){if((Ce&2)!==0&&Se!==0)return Se&-Se;if(L.T!==null){var e=wo;return e!==0?e:hu()}return Jd()}function Bf(){Qt===0&&(Qt=(Se&536870912)===0||be?Vd():536870912);var e=Gt.current;return e!==null&&(e.flags|=32),Qt}function Ht(e,n,o){(e===Ne&&(we===2||we===9)||e.cancelPendingCommit!==null)&&(Bo(e,0),li(e,Se,Qt,!1)),hr(e,o),((Ce&2)===0||e!==Ne)&&(e===Ne&&((Ce&2)===0&&(qi|=o),Be===4&&li(e,Se,Qt,!1)),fn(e))}function Mf(e,n,o){if((Ce&6)!==0)throw Error(s(327));var a=!o&&(n&124)===0&&(n&e.expiredLanes)===0||pr(e,n),c=a?rR(e,n):cu(e,n,!0),d=a;do{if(c===0){Do&&!a&&li(e,n,0,!1);break}else{if(o=e.current.alternate,d&&!iR(o)){c=cu(e,n,!1),d=!1;continue}if(c===2){if(d=n,e.errorRecoveryDisabledLanes&d)var m=0;else m=e.pendingLanes&-536870913,m=m!==0?m:m&536870912?536870912:0;if(m!==0){n=m;e:{var S=e;c=Vr;var C=S.current.memoizedState.isDehydrated;if(C&&(Bo(S,m).flags|=256),m=cu(S,m,!1),m!==2){if(eu&&!C){S.errorRecoveryDisabledLanes|=d,qi|=d,c=4;break e}d=bt,bt=c,d!==null&&(bt===null?bt=d:bt.push.apply(bt,d))}c=m}if(d=!1,c!==2)continue}}if(c===1){Bo(e,0),li(e,n,0,!0);break}e:{switch(a=e,d=c,d){case 0:case 1:throw Error(s(345));case 4:if((n&4194048)!==n)break;case 6:li(a,n,Qt,!oi);break e;case 2:bt=null;break;case 3:case 5:break;default:throw Error(s(329))}if((n&62914560)===n&&(c=iu+300-cn(),10<c)){if(li(a,n,Qt,!oi),Ns(a,0,!0)!==0)break e;a.timeoutHandle=fg(qf.bind(null,a,o,bt,Ia,nu,n,Qt,qi,Ho,oi,d,2,-0,0),c);break e}qf(a,o,bt,Ia,nu,n,Qt,qi,Ho,oi,d,0,-0,0)}}break}while(!0);fn(e)}function qf(e,n,o,a,c,d,m,S,C,F,M,W,D,H){if(e.timeoutHandle=-1,W=n.subtreeFlags,(W&8192||(W&16785408)===16785408)&&(ns={stylesheets:null,count:0,unsuspend:BR},Ff(n),W=qR(),W!==null)){e.cancelPendingCommit=W(Qf.bind(null,e,n,d,o,a,c,m,S,C,M,1,D,H)),li(e,d,m,!F);return}Qf(e,n,d,o,a,c,m,S,C)}function iR(e){for(var n=e;;){var o=n.tag;if((o===0||o===11||o===15)&&n.flags&16384&&(o=n.updateQueue,o!==null&&(o=o.stores,o!==null)))for(var a=0;a<o.length;a++){var c=o[a],d=c.getSnapshot;c=c.value;try{if(!Pt(d(),c))return!1}catch{return!1}}if(o=n.child,n.subtreeFlags&16384&&o!==null)o.return=n,n=o;else{if(n===e)break;for(;n.sibling===null;){if(n.return===null||n.return===e)return!0;n=n.return}n.sibling.return=n.return,n=n.sibling}}return!0}function li(e,n,o,a){n&=~tu,n&=~qi,e.suspendedLanes|=n,e.pingedLanes&=~n,a&&(e.warmLanes|=n),a=e.expirationTimes;for(var c=n;0<c;){var d=31-xt(c),m=1<<d;a[d]=-1,c&=~m}o!==0&&Kd(e,o,n)}function ya(){return(Ce&6)===0?(Kr(0),!1):!0}function au(){if(fe!==null){if(we===0)var e=fe.return;else e=fe,xn=Di=null,wc(e),Xo=null,qr=0,e=fe;for(;e!==null;)Rf(e.alternate,e),e=e.return;fe=null}}function Bo(e,n){var o=e.timeoutHandle;o!==-1&&(e.timeoutHandle=-1,yR(o)),o=e.cancelPendingCommit,o!==null&&(e.cancelPendingCommit=null,o()),au(),Ne=e,fe=o=_n(e.current,null),Se=n,we=0,Ft=null,oi=!1,Do=pr(e,n),eu=!1,Ho=Qt=tu=qi=ri=Be=0,bt=Vr=null,nu=!1,(n&8)!==0&&(n|=n&32);var a=e.entangledLanes;if(a!==0)for(e=e.entanglements,a&=n;0<a;){var c=31-xt(a),d=1<<c;n|=e[c],a&=~d}return An=n,js(),o}function Uf(e,n){pe=null,L.H=la,n===Nr||n===Js?(n=ih(),we=3):n===eh?(n=ih(),we=4):we=n===rf?8:n!==null&&typeof n=="object"&&typeof n.then=="function"?6:1,Ft=n,fe===null&&(Be=1,ha(e,Wt(n,e.current)))}function Wf(){var e=L.H;return L.H=la,e===null?la:e}function jf(){var e=L.A;return L.A=tR,e}function lu(){Be=4,oi||(Se&4194048)!==Se&&Gt.current!==null||(Do=!0),(ri&134217727)===0&&(qi&134217727)===0||Ne===null||li(Ne,Se,Qt,!1)}function cu(e,n,o){var a=Ce;Ce|=2;var c=Wf(),d=jf();(Ne!==e||Se!==n)&&(Ia=null,Bo(e,n)),n=!1;var m=Be;e:do try{if(we!==0&&fe!==null){var S=fe,C=Ft;switch(we){case 8:au(),m=6;break e;case 3:case 2:case 9:case 6:Gt.current===null&&(n=!0);var F=we;if(we=0,Ft=null,Mo(e,S,C,F),o&&Do){m=0;break e}break;default:F=we,we=0,Ft=null,Mo(e,S,C,F)}}oR(),m=Be;break}catch(M){Uf(e,M)}while(!0);return n&&e.shellSuspendCounter++,xn=Di=null,Ce=a,L.H=c,L.A=d,fe===null&&(Ne=null,Se=0,js()),m}function oR(){for(;fe!==null;)kf(fe)}function rR(e,n){var o=Ce;Ce|=2;var a=Wf(),c=jf();Ne!==e||Se!==n?(Ia=null,Ra=cn()+500,Bo(e,n)):Do=pr(e,n);e:do try{if(we!==0&&fe!==null){n=fe;var d=Ft;t:switch(we){case 1:we=0,Ft=null,Mo(e,n,d,1);break;case 2:case 9:if(th(d)){we=0,Ft=null,zf(n);break}n=function(){we!==2&&we!==9||Ne!==e||(we=7),fn(e)},d.then(n,n);break e;case 3:we=7;break e;case 4:we=5;break e;case 7:th(d)?(we=0,Ft=null,zf(n)):(we=0,Ft=null,Mo(e,n,d,7));break;case 5:var m=null;switch(fe.tag){case 26:m=fe.memoizedState;case 5:case 27:var S=fe;if(!m||Tg(m)){we=0,Ft=null;var C=S.sibling;if(C!==null)fe=C;else{var F=S.return;F!==null?(fe=F,ba(F)):fe=null}break t}}we=0,Ft=null,Mo(e,n,d,5);break;case 6:we=0,Ft=null,Mo(e,n,d,6);break;case 8:au(),Be=6;break e;default:throw Error(s(462))}}sR();break}catch(M){Uf(e,M)}while(!0);return xn=Di=null,L.H=a,L.A=c,Ce=o,fe!==null?0:(Ne=null,Se=0,js(),Be)}function sR(){for(;fe!==null&&!xS();)kf(fe)}function kf(e){var n=Ef(e.alternate,e,An);e.memoizedProps=e.pendingProps,n===null?ba(e):fe=n}function zf(e){var n=e,o=n.alternate;switch(n.tag){case 15:case 0:n=df(o,n,n.pendingProps,n.type,void 0,Se);break;case 11:n=df(o,n,n.pendingProps,n.type.render,n.ref,Se);break;case 5:wc(n);default:Rf(o,n),n=fe=zp(n,An),n=Ef(o,n,An)}e.memoizedProps=e.pendingProps,n===null?ba(e):fe=n}function Mo(e,n,o,a){xn=Di=null,wc(n),Xo=null,qr=0;var c=n.return;try{if(Y0(e,c,n,o,Se)){Be=1,ha(e,Wt(o,e.current)),fe=null;return}}catch(d){if(c!==null)throw fe=c,d;Be=1,ha(e,Wt(o,e.current)),fe=null;return}n.flags&32768?(be||a===1?e=!0:Do||(Se&536870912)!==0?e=!1:(oi=e=!0,(a===2||a===9||a===3||a===6)&&(a=Gt.current,a!==null&&a.tag===13&&(a.flags|=16384))),Gf(n,e)):ba(n)}function ba(e){var n=e;do{if((n.flags&32768)!==0){Gf(n,oi);return}e=n.return;var o=Z0(n.alternate,n,An);if(o!==null){fe=o;return}if(n=n.sibling,n!==null){fe=n;return}fe=n=e}while(n!==null);Be===0&&(Be=5)}function Gf(e,n){do{var o=J0(e.alternate,e);if(o!==null){o.flags&=32767,fe=o;return}if(o=e.return,o!==null&&(o.flags|=32768,o.subtreeFlags=0,o.deletions=null),!n&&(e=e.sibling,e!==null)){fe=e;return}fe=e=o}while(e!==null);Be=6,fe=null}function Qf(e,n,o,a,c,d,m,S,C){e.cancelPendingCommit=null;do Ca();while(it!==0);if((Ce&6)!==0)throw Error(s(327));if(n!==null){if(n===e.current)throw Error(s(177));if(d=n.lanes|n.childLanes,d|=$l,BS(e,o,d,m,S,C),e===Ne&&(fe=Ne=null,Se=0),Ao=n,ai=e,Lo=o,ou=d,ru=c,Lf=a,(n.subtreeFlags&10256)!==0||(n.flags&10256)!==0?(e.callbackNode=null,e.callbackPriority=0,uR(xs,function(){return Jf(),null})):(e.callbackNode=null,e.callbackPriority=0),a=(n.flags&13878)!==0,(n.subtreeFlags&13878)!==0||a){a=L.T,L.T=null,c=V.p,V.p=2,m=Ce,Ce|=4;try{$0(e,n,o)}finally{Ce=m,V.p=c,L.T=a}}it=1,Vf(),Yf(),Kf()}}function Vf(){if(it===1){it=0;var e=ai,n=Ao,o=(n.flags&13878)!==0;if((n.subtreeFlags&13878)!==0||o){o=L.T,L.T=null;var a=V.p;V.p=2;var c=Ce;Ce|=4;try{Xf(n,e);var d=yu,m=Hp(e.containerInfo),S=d.focusedElem,C=d.selectionRange;if(m!==S&&S&&S.ownerDocument&&Dp(S.ownerDocument.documentElement,S)){if(C!==null&&Vl(S)){var F=C.start,M=C.end;if(M===void 0&&(M=F),"selectionStart"in S)S.selectionStart=F,S.selectionEnd=Math.min(M,S.value.length);else{var W=S.ownerDocument||document,D=W&&W.defaultView||window;if(D.getSelection){var H=D.getSelection(),ae=S.textContent.length,oe=Math.min(C.start,ae),xe=C.end===void 0?oe:Math.min(C.end,ae);!H.extend&&oe>xe&&(m=xe,xe=oe,oe=m);var x=Fp(S,oe),T=Fp(S,xe);if(x&&T&&(H.rangeCount!==1||H.anchorNode!==x.node||H.anchorOffset!==x.offset||H.focusNode!==T.node||H.focusOffset!==T.offset)){var O=W.createRange();O.setStart(x.node,x.offset),H.removeAllRanges(),oe>xe?(H.addRange(O),H.extend(T.node,T.offset)):(O.setEnd(T.node,T.offset),H.addRange(O))}}}}for(W=[],H=S;H=H.parentNode;)H.nodeType===1&&W.push({element:H,left:H.scrollLeft,top:H.scrollTop});for(typeof S.focus=="function"&&S.focus(),S=0;S<W.length;S++){var q=W[S];q.element.scrollLeft=q.left,q.element.scrollTop=q.top}}Ha=!!Iu,yu=Iu=null}finally{Ce=c,V.p=a,L.T=o}}e.current=n,it=2}}function Yf(){if(it===2){it=0;var e=ai,n=Ao,o=(n.flags&8772)!==0;if((n.subtreeFlags&8772)!==0||o){o=L.T,L.T=null;var a=V.p;V.p=2;var c=Ce;Ce|=4;try{Tf(e,n.alternate,n)}finally{Ce=c,V.p=a,L.T=o}}it=3}}function Kf(){if(it===4||it===3){it=0,PS();var e=ai,n=Ao,o=Lo,a=Lf;(n.subtreeFlags&10256)!==0||(n.flags&10256)!==0?it=5:(it=0,Ao=ai=null,Zf(e,e.pendingLanes));var c=e.pendingLanes;if(c===0&&(si=null),vl(o),n=n.stateNode,vt&&typeof vt.onCommitFiberRoot=="function")try{vt.onCommitFiberRoot(dr,n,void 0,(n.current.flags&128)===128)}catch{}if(a!==null){n=L.T,c=V.p,V.p=2,L.T=null;try{for(var d=e.onRecoverableError,m=0;m<a.length;m++){var S=a[m];d(S.value,{componentStack:S.stack})}}finally{L.T=n,V.p=c}}(Lo&3)!==0&&Ca(),fn(e),c=e.pendingLanes,(o&4194090)!==0&&(c&42)!==0?e===su?Yr++:(Yr=0,su=e):Yr=0,Kr(0)}}function Zf(e,n){(e.pooledCacheLanes&=n)===0&&(n=e.pooledCache,n!=null&&(e.pooledCache=null,Pr(n)))}function Ca(e){return Vf(),Yf(),Kf(),Jf()}function Jf(){if(it!==5)return!1;var e=ai,n=ou;ou=0;var o=vl(Lo),a=L.T,c=V.p;try{V.p=32>o?32:o,L.T=null,o=ru,ru=null;var d=ai,m=Lo;if(it=0,Ao=ai=null,Lo=0,(Ce&6)!==0)throw Error(s(331));var S=Ce;if(Ce|=4,Hf(d.current),Of(d,d.current,m,o),Ce=S,Kr(0,!1),vt&&typeof vt.onPostCommitFiberRoot=="function")try{vt.onPostCommitFiberRoot(dr,d)}catch{}return!0}finally{V.p=c,L.T=a,Zf(e,n)}}function $f(e,n,o){n=Wt(o,n),n=Bc(e.stateNode,n,2),e=Zn(e,n,2),e!==null&&(hr(e,2),fn(e))}function Pe(e,n,o){if(e.tag===3)$f(e,e,o);else for(;n!==null;){if(n.tag===3){$f(n,e,o);break}else if(n.tag===1){var a=n.stateNode;if(typeof n.type.getDerivedStateFromError=="function"||typeof a.componentDidCatch=="function"&&(si===null||!si.has(a))){e=Wt(o,e),o=nf(2),a=Zn(n,o,2),a!==null&&(of(o,a,n,e),hr(a,2),fn(a));break}}n=n.return}}function uu(e,n,o){var a=e.pingCache;if(a===null){a=e.pingCache=new nR;var c=new Set;a.set(n,c)}else c=a.get(n),c===void 0&&(c=new Set,a.set(n,c));c.has(o)||(eu=!0,c.add(o),e=aR.bind(null,e,n,o),n.then(e,e))}function aR(e,n,o){var a=e.pingCache;a!==null&&a.delete(n),e.pingedLanes|=e.suspendedLanes&o,e.warmLanes&=~o,Ne===e&&(Se&o)===o&&(Be===4||Be===3&&(Se&62914560)===Se&&300>cn()-iu?(Ce&2)===0&&Bo(e,0):tu|=o,Ho===Se&&(Ho=0)),fn(e)}function eg(e,n){n===0&&(n=Yd()),e=Io(e,n),e!==null&&(hr(e,n),fn(e))}function lR(e){var n=e.memoizedState,o=0;n!==null&&(o=n.retryLane),eg(e,o)}function cR(e,n){var o=0;switch(e.tag){case 13:var a=e.stateNode,c=e.memoizedState;c!==null&&(o=c.retryLane);break;case 19:a=e.stateNode;break;case 22:a=e.stateNode._retryCache;break;default:throw Error(s(314))}a!==null&&a.delete(n),eg(e,o)}function uR(e,n){return Cl(e,n)}var wa=null,qo=null,du=!1,_a=!1,pu=!1,Ui=0;function fn(e){e!==qo&&e.next===null&&(qo===null?wa=qo=e:qo=qo.next=e),_a=!0,du||(du=!0,pR())}function Kr(e,n){if(!pu&&_a){pu=!0;do for(var o=!1,a=wa;a!==null;){if(e!==0){var c=a.pendingLanes;if(c===0)var d=0;else{var m=a.suspendedLanes,S=a.pingedLanes;d=(1<<31-xt(42|e)+1)-1,d&=c&~(m&~S),d=d&201326741?d&201326741|1:d?d|2:0}d!==0&&(o=!0,og(a,d))}else d=Se,d=Ns(a,a===Ne?d:0,a.cancelPendingCommit!==null||a.timeoutHandle!==-1),(d&3)===0||pr(a,d)||(o=!0,og(a,d));a=a.next}while(o);pu=!1}}function dR(){tg()}function tg(){_a=du=!1;var e=0;Ui!==0&&(IR()&&(e=Ui),Ui=0);for(var n=cn(),o=null,a=wa;a!==null;){var c=a.next,d=ng(a,n);d===0?(a.next=null,o===null?wa=c:o.next=c,c===null&&(qo=o)):(o=a,(e!==0||(d&3)!==0)&&(_a=!0)),a=c}Kr(e)}function ng(e,n){for(var o=e.suspendedLanes,a=e.pingedLanes,c=e.expirationTimes,d=e.pendingLanes&-62914561;0<d;){var m=31-xt(d),S=1<<m,C=c[m];C===-1?((S&o)===0||(S&a)!==0)&&(c[m]=LS(S,n)):C<=n&&(e.expiredLanes|=S),d&=~S}if(n=Ne,o=Se,o=Ns(e,e===n?o:0,e.cancelPendingCommit!==null||e.timeoutHandle!==-1),a=e.callbackNode,o===0||e===n&&(we===2||we===9)||e.cancelPendingCommit!==null)return a!==null&&a!==null&&wl(a),e.callbackNode=null,e.callbackPriority=0;if((o&3)===0||pr(e,o)){if(n=o&-o,n===e.callbackPriority)return n;switch(a!==null&&wl(a),vl(o)){case 2:case 8:o=Gd;break;case 32:o=xs;break;case 268435456:o=Qd;break;default:o=xs}return a=ig.bind(null,e),o=Cl(o,a),e.callbackPriority=n,e.callbackNode=o,n}return a!==null&&a!==null&&wl(a),e.callbackPriority=2,e.callbackNode=null,2}function ig(e,n){if(it!==0&&it!==5)return e.callbackNode=null,e.callbackPriority=0,null;var o=e.callbackNode;if(Ca()&&e.callbackNode!==o)return null;var a=Se;return a=Ns(e,e===Ne?a:0,e.cancelPendingCommit!==null||e.timeoutHandle!==-1),a===0?null:(Mf(e,a,n),ng(e,cn()),e.callbackNode!=null&&e.callbackNode===o?ig.bind(null,e):null)}function og(e,n){if(Ca())return null;Mf(e,n,!0)}function pR(){bR(function(){(Ce&6)!==0?Cl(zd,dR):tg()})}function hu(){return Ui===0&&(Ui=Vd()),Ui}function rg(e){return e==null||typeof e=="symbol"||typeof e=="boolean"?null:typeof e=="function"?e:As(""+e)}function sg(e,n){var o=n.ownerDocument.createElement("input");return o.name=n.name,o.value=n.value,e.id&&o.setAttribute("form",e.id),n.parentNode.insertBefore(o,n),e=new FormData(e),o.parentNode.removeChild(o),e}function hR(e,n,o,a,c){if(n==="submit"&&o&&o.stateNode===c){var d=rg((c[St]||null).action),m=a.submitter;m&&(n=(n=m[St]||null)?rg(n.formAction):m.getAttribute("formAction"),n!==null&&(d=n,m=null));var S=new qs("action","action",null,a,c);e.push({event:S,listeners:[{instance:null,listener:function(){if(a.defaultPrevented){if(Ui!==0){var C=m?sg(c,m):new FormData(c);Fc(o,{pending:!0,data:C,method:c.method,action:d},null,C)}}else typeof d=="function"&&(S.preventDefault(),C=m?sg(c,m):new FormData(c),Fc(o,{pending:!0,data:C,method:c.method,action:d},d,C))},currentTarget:c}]})}}for(var fu=0;fu<Jl.length;fu++){var gu=Jl[fu],fR=gu.toLowerCase(),gR=gu[0].toUpperCase()+gu.slice(1);$t(fR,"on"+gR)}$t(Bp,"onAnimationEnd"),$t(Mp,"onAnimationIteration"),$t(qp,"onAnimationStart"),$t("dblclick","onDoubleClick"),$t("focusin","onFocus"),$t("focusout","onBlur"),$t(O0,"onTransitionRun"),$t(F0,"onTransitionStart"),$t(D0,"onTransitionCancel"),$t(Up,"onTransitionEnd"),co("onMouseEnter",["mouseout","mouseover"]),co("onMouseLeave",["mouseout","mouseover"]),co("onPointerEnter",["pointerout","pointerover"]),co("onPointerLeave",["pointerout","pointerover"]),_i("onChange","change click focusin focusout input keydown keyup selectionchange".split(" ")),_i("onSelect","focusout contextmenu dragend focusin keydown keyup mousedown mouseup selectionchange".split(" ")),_i("onBeforeInput",["compositionend","keypress","textInput","paste"]),_i("onCompositionEnd","compositionend focusout keydown keypress keyup mousedown".split(" ")),_i("onCompositionStart","compositionstart focusout keydown keypress keyup mousedown".split(" ")),_i("onCompositionUpdate","compositionupdate focusout keydown keypress keyup mousedown".split(" "));var Zr="abort canplay canplaythrough durationchange emptied encrypted ended error loadeddata loadedmetadata loadstart pause play playing progress ratechange resize seeked seeking stalled suspend timeupdate volumechange waiting".split(" "),mR=new Set("beforetoggle cancel close invalid load scroll scrollend toggle".split(" ").concat(Zr));function ag(e,n){n=(n&4)!==0;for(var o=0;o<e.length;o++){var a=e[o],c=a.event;a=a.listeners;e:{var d=void 0;if(n)for(var m=a.length-1;0<=m;m--){var S=a[m],C=S.instance,F=S.currentTarget;if(S=S.listener,C!==d&&c.isPropagationStopped())break e;d=S,c.currentTarget=F;try{d(c)}catch(M){pa(M)}c.currentTarget=null,d=C}else for(m=0;m<a.length;m++){if(S=a[m],C=S.instance,F=S.currentTarget,S=S.listener,C!==d&&c.isPropagationStopped())break e;d=S,c.currentTarget=F;try{d(c)}catch(M){pa(M)}c.currentTarget=null,d=C}}}}function ge(e,n){var o=n[xl];o===void 0&&(o=n[xl]=new Set);var a=e+"__bubble";o.has(a)||(lg(n,e,2,!1),o.add(a))}function mu(e,n,o){var a=0;n&&(a|=4),lg(o,e,a,n)}var Ta="_reactListening"+Math.random().toString(36).slice(2);function Eu(e){if(!e[Ta]){e[Ta]=!0,ep.forEach(function(o){o!=="selectionchange"&&(mR.has(o)||mu(o,!1,e),mu(o,!0,e))});var n=e.nodeType===9?e:e.ownerDocument;n===null||n[Ta]||(n[Ta]=!0,mu("selectionchange",!1,n))}}function lg(e,n,o,a){switch(Og(n)){case 2:var c=jR;break;case 8:c=kR;break;default:c=Nu}o=c.bind(null,n,o,e),c=void 0,!Ml||n!=="touchstart"&&n!=="touchmove"&&n!=="wheel"||(c=!0),a?c!==void 0?e.addEventListener(n,o,{capture:!0,passive:c}):e.addEventListener(n,o,!0):c!==void 0?e.addEventListener(n,o,{passive:c}):e.addEventListener(n,o,!1)}function Su(e,n,o,a,c){var d=a;if((n&1)===0&&(n&2)===0&&a!==null)e:for(;;){if(a===null)return;var m=a.tag;if(m===3||m===4){var S=a.stateNode.containerInfo;if(S===c)break;if(m===4)for(m=a.return;m!==null;){var C=m.tag;if((C===3||C===4)&&m.stateNode.containerInfo===c)return;m=m.return}for(;S!==null;){if(m=so(S),m===null)return;if(C=m.tag,C===5||C===6||C===26||C===27){a=d=m;continue e}S=S.parentNode}}a=a.return}fp(function(){var F=d,M=Ll(o),W=[];e:{var D=Wp.get(e);if(D!==void 0){var H=qs,ae=e;switch(e){case"keypress":if(Bs(o)===0)break e;case"keydown":case"keyup":H=u0;break;case"focusin":ae="focus",H=jl;break;case"focusout":ae="blur",H=jl;break;case"beforeblur":case"afterblur":H=jl;break;case"click":if(o.button===2)break e;case"auxclick":case"dblclick":case"mousedown":case"mousemove":case"mouseup":case"mouseout":case"mouseover":case"contextmenu":H=Ep;break;case"drag":case"dragend":case"dragenter":case"dragexit":case"dragleave":case"dragover":case"dragstart":case"drop":H=JS;break;case"touchcancel":case"touchend":case"touchmove":case"touchstart":H=h0;break;case Bp:case Mp:case qp:H=t0;break;case Up:H=g0;break;case"scroll":case"scrollend":H=KS;break;case"wheel":H=E0;break;case"copy":case"cut":case"paste":H=i0;break;case"gotpointercapture":case"lostpointercapture":case"pointercancel":case"pointerdown":case"pointermove":case"pointerout":case"pointerover":case"pointerup":H=Rp;break;case"toggle":case"beforetoggle":H=R0}var oe=(n&4)!==0,xe=!oe&&(e==="scroll"||e==="scrollend"),x=oe?D!==null?D+"Capture":null:D;oe=[];for(var T=F,O;T!==null;){var q=T;if(O=q.stateNode,q=q.tag,q!==5&&q!==26&&q!==27||O===null||x===null||(q=mr(T,x),q!=null&&oe.push(Jr(T,q,O))),xe)break;T=T.return}0<oe.length&&(D=new H(D,ae,null,o,M),W.push({event:D,listeners:oe}))}}if((n&7)===0){e:{if(D=e==="mouseover"||e==="pointerover",H=e==="mouseout"||e==="pointerout",D&&o!==Al&&(ae=o.relatedTarget||o.fromElement)&&(so(ae)||ae[ro]))break e;if((H||D)&&(D=M.window===M?M:(D=M.ownerDocument)?D.defaultView||D.parentWindow:window,H?(ae=o.relatedTarget||o.toElement,H=F,ae=ae?so(ae):null,ae!==null&&(xe=u(ae),oe=ae.tag,ae!==xe||oe!==5&&oe!==27&&oe!==6)&&(ae=null)):(H=null,ae=F),H!==ae)){if(oe=Ep,q="onMouseLeave",x="onMouseEnter",T="mouse",(e==="pointerout"||e==="pointerover")&&(oe=Rp,q="onPointerLeave",x="onPointerEnter",T="pointer"),xe=H==null?D:gr(H),O=ae==null?D:gr(ae),D=new oe(q,T+"leave",H,o,M),D.target=xe,D.relatedTarget=O,q=null,so(M)===F&&(oe=new oe(x,T+"enter",ae,o,M),oe.target=O,oe.relatedTarget=xe,q=oe),xe=q,H&&ae)t:{for(oe=H,x=ae,T=0,O=oe;O;O=Uo(O))T++;for(O=0,q=x;q;q=Uo(q))O++;for(;0<T-O;)oe=Uo(oe),T--;for(;0<O-T;)x=Uo(x),O--;for(;T--;){if(oe===x||x!==null&&oe===x.alternate)break t;oe=Uo(oe),x=Uo(x)}oe=null}else oe=null;H!==null&&cg(W,D,H,oe,!1),ae!==null&&xe!==null&&cg(W,xe,ae,oe,!0)}}e:{if(D=F?gr(F):window,H=D.nodeName&&D.nodeName.toLowerCase(),H==="select"||H==="input"&&D.type==="file")var Y=vp;else if(_p(D))if(xp)Y=P0;else{Y=v0;var he=T0}else H=D.nodeName,!H||H.toLowerCase()!=="input"||D.type!=="checkbox"&&D.type!=="radio"?F&&Hl(F.elementType)&&(Y=vp):Y=x0;if(Y&&(Y=Y(e,F))){Tp(W,Y,o,M);break e}he&&he(e,D,F),e==="focusout"&&F&&D.type==="number"&&F.memoizedProps.value!=null&&Dl(D,"number",D.value)}switch(he=F?gr(F):window,e){case"focusin":(_p(he)||he.contentEditable==="true")&&(Eo=he,Yl=F,wr=null);break;case"focusout":wr=Yl=Eo=null;break;case"mousedown":Kl=!0;break;case"contextmenu":case"mouseup":case"dragend":Kl=!1,Ap(W,o,M);break;case"selectionchange":if(N0)break;case"keydown":case"keyup":Ap(W,o,M)}var ee;if(zl)e:{switch(e){case"compositionstart":var se="onCompositionStart";break e;case"compositionend":se="onCompositionEnd";break e;case"compositionupdate":se="onCompositionUpdate";break e}se=void 0}else mo?Cp(e,o)&&(se="onCompositionEnd"):e==="keydown"&&o.keyCode===229&&(se="onCompositionStart");se&&(Ip&&o.locale!=="ko"&&(mo||se!=="onCompositionStart"?se==="onCompositionEnd"&&mo&&(ee=gp()):(Qn=M,ql="value"in Qn?Qn.value:Qn.textContent,mo=!0)),he=va(F,se),0<he.length&&(se=new Sp(se,e,null,o,M),W.push({event:se,listeners:he}),ee?se.data=ee:(ee=wp(o),ee!==null&&(se.data=ee)))),(ee=y0?b0(e,o):C0(e,o))&&(se=va(F,"onBeforeInput"),0<se.length&&(he=new Sp("onBeforeInput","beforeinput",null,o,M),W.push({event:he,listeners:se}),he.data=ee)),hR(W,e,F,o,M)}ag(W,n)})}function Jr(e,n,o){return{instance:e,listener:n,currentTarget:o}}function va(e,n){for(var o=n+"Capture",a=[];e!==null;){var c=e,d=c.stateNode;if(c=c.tag,c!==5&&c!==26&&c!==27||d===null||(c=mr(e,o),c!=null&&a.unshift(Jr(e,c,d)),c=mr(e,n),c!=null&&a.push(Jr(e,c,d))),e.tag===3)return a;e=e.return}return[]}function Uo(e){if(e===null)return null;do e=e.return;while(e&&e.tag!==5&&e.tag!==27);return e||null}function cg(e,n,o,a,c){for(var d=n._reactName,m=[];o!==null&&o!==a;){var S=o,C=S.alternate,F=S.stateNode;if(S=S.tag,C!==null&&C===a)break;S!==5&&S!==26&&S!==27||F===null||(C=F,c?(F=mr(o,d),F!=null&&m.unshift(Jr(o,F,C))):c||(F=mr(o,d),F!=null&&m.push(Jr(o,F,C)))),o=o.return}m.length!==0&&e.push({event:n,listeners:m})}var ER=/\r\n?/g,SR=/\u0000|\uFFFD/g;function ug(e){return(typeof e=="string"?e:""+e).replace(ER,`
`).replace(SR,"")}function dg(e,n){return n=ug(n),ug(e)===n}function xa(){}function ve(e,n,o,a,c,d){switch(o){case"children":typeof a=="string"?n==="body"||n==="textarea"&&a===""||ho(e,a):(typeof a=="number"||typeof a=="bigint")&&n!=="body"&&ho(e,""+a);break;case"className":Fs(e,"class",a);break;case"tabIndex":Fs(e,"tabindex",a);break;case"dir":case"role":case"viewBox":case"width":case"height":Fs(e,o,a);break;case"style":pp(e,a,d);break;case"data":if(n!=="object"){Fs(e,"data",a);break}case"src":case"href":if(a===""&&(n!=="a"||o!=="href")){e.removeAttribute(o);break}if(a==null||typeof a=="function"||typeof a=="symbol"||typeof a=="boolean"){e.removeAttribute(o);break}a=As(""+a),e.setAttribute(o,a);break;case"action":case"formAction":if(typeof a=="function"){e.setAttribute(o,"javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')");break}else typeof d=="function"&&(o==="formAction"?(n!=="input"&&ve(e,n,"name",c.name,c,null),ve(e,n,"formEncType",c.formEncType,c,null),ve(e,n,"formMethod",c.formMethod,c,null),ve(e,n,"formTarget",c.formTarget,c,null)):(ve(e,n,"encType",c.encType,c,null),ve(e,n,"method",c.method,c,null),ve(e,n,"target",c.target,c,null)));if(a==null||typeof a=="symbol"||typeof a=="boolean"){e.removeAttribute(o);break}a=As(""+a),e.setAttribute(o,a);break;case"onClick":a!=null&&(e.onclick=xa);break;case"onScroll":a!=null&&ge("scroll",e);break;case"onScrollEnd":a!=null&&ge("scrollend",e);break;case"dangerouslySetInnerHTML":if(a!=null){if(typeof a!="object"||!("__html"in a))throw Error(s(61));if(o=a.__html,o!=null){if(c.children!=null)throw Error(s(60));e.innerHTML=o}}break;case"multiple":e.multiple=a&&typeof a!="function"&&typeof a!="symbol";break;case"muted":e.muted=a&&typeof a!="function"&&typeof a!="symbol";break;case"suppressContentEditableWarning":case"suppressHydrationWarning":case"defaultValue":case"defaultChecked":case"innerHTML":case"ref":break;case"autoFocus":break;case"xlinkHref":if(a==null||typeof a=="function"||typeof a=="boolean"||typeof a=="symbol"){e.removeAttribute("xlink:href");break}o=As(""+a),e.setAttributeNS("http://www.w3.org/1999/xlink","xlink:href",o);break;case"contentEditable":case"spellCheck":case"draggable":case"value":case"autoReverse":case"externalResourcesRequired":case"focusable":case"preserveAlpha":a!=null&&typeof a!="function"&&typeof a!="symbol"?e.setAttribute(o,""+a):e.removeAttribute(o);break;case"inert":case"allowFullScreen":case"async":case"autoPlay":case"controls":case"default":case"defer":case"disabled":case"disablePictureInPicture":case"disableRemotePlayback":case"formNoValidate":case"hidden":case"loop":case"noModule":case"noValidate":case"open":case"playsInline":case"readOnly":case"required":case"reversed":case"scoped":case"seamless":case"itemScope":a&&typeof a!="function"&&typeof a!="symbol"?e.setAttribute(o,""):e.removeAttribute(o);break;case"capture":case"download":a===!0?e.setAttribute(o,""):a!==!1&&a!=null&&typeof a!="function"&&typeof a!="symbol"?e.setAttribute(o,a):e.removeAttribute(o);break;case"cols":case"rows":case"size":case"span":a!=null&&typeof a!="function"&&typeof a!="symbol"&&!isNaN(a)&&1<=a?e.setAttribute(o,a):e.removeAttribute(o);break;case"rowSpan":case"start":a==null||typeof a=="function"||typeof a=="symbol"||isNaN(a)?e.removeAttribute(o):e.setAttribute(o,a);break;case"popover":ge("beforetoggle",e),ge("toggle",e),Os(e,"popover",a);break;case"xlinkActuate":Cn(e,"http://www.w3.org/1999/xlink","xlink:actuate",a);break;case"xlinkArcrole":Cn(e,"http://www.w3.org/1999/xlink","xlink:arcrole",a);break;case"xlinkRole":Cn(e,"http://www.w3.org/1999/xlink","xlink:role",a);break;case"xlinkShow":Cn(e,"http://www.w3.org/1999/xlink","xlink:show",a);break;case"xlinkTitle":Cn(e,"http://www.w3.org/1999/xlink","xlink:title",a);break;case"xlinkType":Cn(e,"http://www.w3.org/1999/xlink","xlink:type",a);break;case"xmlBase":Cn(e,"http://www.w3.org/XML/1998/namespace","xml:base",a);break;case"xmlLang":Cn(e,"http://www.w3.org/XML/1998/namespace","xml:lang",a);break;case"xmlSpace":Cn(e,"http://www.w3.org/XML/1998/namespace","xml:space",a);break;case"is":Os(e,"is",a);break;case"innerText":case"textContent":break;default:(!(2<o.length)||o[0]!=="o"&&o[0]!=="O"||o[1]!=="n"&&o[1]!=="N")&&(o=VS.get(o)||o,Os(e,o,a))}}function Ru(e,n,o,a,c,d){switch(o){case"style":pp(e,a,d);break;case"dangerouslySetInnerHTML":if(a!=null){if(typeof a!="object"||!("__html"in a))throw Error(s(61));if(o=a.__html,o!=null){if(c.children!=null)throw Error(s(60));e.innerHTML=o}}break;case"children":typeof a=="string"?ho(e,a):(typeof a=="number"||typeof a=="bigint")&&ho(e,""+a);break;case"onScroll":a!=null&&ge("scroll",e);break;case"onScrollEnd":a!=null&&ge("scrollend",e);break;case"onClick":a!=null&&(e.onclick=xa);break;case"suppressContentEditableWarning":case"suppressHydrationWarning":case"innerHTML":case"ref":break;case"innerText":case"textContent":break;default:if(!tp.hasOwnProperty(o))e:{if(o[0]==="o"&&o[1]==="n"&&(c=o.endsWith("Capture"),n=o.slice(2,c?o.length-7:void 0),d=e[St]||null,d=d!=null?d[o]:null,typeof d=="function"&&e.removeEventListener(n,d,c),typeof a=="function")){typeof d!="function"&&d!==null&&(o in e?e[o]=null:e.hasAttribute(o)&&e.removeAttribute(o)),e.addEventListener(n,a,c);break e}o in e?e[o]=a:a===!0?e.setAttribute(o,""):Os(e,o,a)}}}function ot(e,n,o){switch(n){case"div":case"span":case"svg":case"path":case"a":case"g":case"p":case"li":break;case"img":ge("error",e),ge("load",e);var a=!1,c=!1,d;for(d in o)if(o.hasOwnProperty(d)){var m=o[d];if(m!=null)switch(d){case"src":a=!0;break;case"srcSet":c=!0;break;case"children":case"dangerouslySetInnerHTML":throw Error(s(137,n));default:ve(e,n,d,m,o,null)}}c&&ve(e,n,"srcSet",o.srcSet,o,null),a&&ve(e,n,"src",o.src,o,null);return;case"input":ge("invalid",e);var S=d=m=c=null,C=null,F=null;for(a in o)if(o.hasOwnProperty(a)){var M=o[a];if(M!=null)switch(a){case"name":c=M;break;case"type":m=M;break;case"checked":C=M;break;case"defaultChecked":F=M;break;case"value":d=M;break;case"defaultValue":S=M;break;case"children":case"dangerouslySetInnerHTML":if(M!=null)throw Error(s(137,n));break;default:ve(e,n,a,M,o,null)}}lp(e,d,S,C,F,m,c,!1),Ds(e);return;case"select":ge("invalid",e),a=m=d=null;for(c in o)if(o.hasOwnProperty(c)&&(S=o[c],S!=null))switch(c){case"value":d=S;break;case"defaultValue":m=S;break;case"multiple":a=S;default:ve(e,n,c,S,o,null)}n=d,o=m,e.multiple=!!a,n!=null?po(e,!!a,n,!1):o!=null&&po(e,!!a,o,!0);return;case"textarea":ge("invalid",e),d=c=a=null;for(m in o)if(o.hasOwnProperty(m)&&(S=o[m],S!=null))switch(m){case"value":a=S;break;case"defaultValue":c=S;break;case"children":d=S;break;case"dangerouslySetInnerHTML":if(S!=null)throw Error(s(91));break;default:ve(e,n,m,S,o,null)}up(e,a,c,d),Ds(e);return;case"option":for(C in o)if(o.hasOwnProperty(C)&&(a=o[C],a!=null))switch(C){case"selected":e.selected=a&&typeof a!="function"&&typeof a!="symbol";break;default:ve(e,n,C,a,o,null)}return;case"dialog":ge("beforetoggle",e),ge("toggle",e),ge("cancel",e),ge("close",e);break;case"iframe":case"object":ge("load",e);break;case"video":case"audio":for(a=0;a<Zr.length;a++)ge(Zr[a],e);break;case"image":ge("error",e),ge("load",e);break;case"details":ge("toggle",e);break;case"embed":case"source":case"link":ge("error",e),ge("load",e);case"area":case"base":case"br":case"col":case"hr":case"keygen":case"meta":case"param":case"track":case"wbr":case"menuitem":for(F in o)if(o.hasOwnProperty(F)&&(a=o[F],a!=null))switch(F){case"children":case"dangerouslySetInnerHTML":throw Error(s(137,n));default:ve(e,n,F,a,o,null)}return;default:if(Hl(n)){for(M in o)o.hasOwnProperty(M)&&(a=o[M],a!==void 0&&Ru(e,n,M,a,o,void 0));return}}for(S in o)o.hasOwnProperty(S)&&(a=o[S],a!=null&&ve(e,n,S,a,o,null))}function RR(e,n,o,a){switch(n){case"div":case"span":case"svg":case"path":case"a":case"g":case"p":case"li":break;case"input":var c=null,d=null,m=null,S=null,C=null,F=null,M=null;for(H in o){var W=o[H];if(o.hasOwnProperty(H)&&W!=null)switch(H){case"checked":break;case"value":break;case"defaultValue":C=W;default:a.hasOwnProperty(H)||ve(e,n,H,null,a,W)}}for(var D in a){var H=a[D];if(W=o[D],a.hasOwnProperty(D)&&(H!=null||W!=null))switch(D){case"type":d=H;break;case"name":c=H;break;case"checked":F=H;break;case"defaultChecked":M=H;break;case"value":m=H;break;case"defaultValue":S=H;break;case"children":case"dangerouslySetInnerHTML":if(H!=null)throw Error(s(137,n));break;default:H!==W&&ve(e,n,D,H,a,W)}}Fl(e,m,S,C,F,M,d,c);return;case"select":H=m=S=D=null;for(d in o)if(C=o[d],o.hasOwnProperty(d)&&C!=null)switch(d){case"value":break;case"multiple":H=C;default:a.hasOwnProperty(d)||ve(e,n,d,null,a,C)}for(c in a)if(d=a[c],C=o[c],a.hasOwnProperty(c)&&(d!=null||C!=null))switch(c){case"value":D=d;break;case"defaultValue":S=d;break;case"multiple":m=d;default:d!==C&&ve(e,n,c,d,a,C)}n=S,o=m,a=H,D!=null?po(e,!!o,D,!1):!!a!=!!o&&(n!=null?po(e,!!o,n,!0):po(e,!!o,o?[]:"",!1));return;case"textarea":H=D=null;for(S in o)if(c=o[S],o.hasOwnProperty(S)&&c!=null&&!a.hasOwnProperty(S))switch(S){case"value":break;case"children":break;default:ve(e,n,S,null,a,c)}for(m in a)if(c=a[m],d=o[m],a.hasOwnProperty(m)&&(c!=null||d!=null))switch(m){case"value":D=c;break;case"defaultValue":H=c;break;case"children":break;case"dangerouslySetInnerHTML":if(c!=null)throw Error(s(91));break;default:c!==d&&ve(e,n,m,c,a,d)}cp(e,D,H);return;case"option":for(var ae in o)if(D=o[ae],o.hasOwnProperty(ae)&&D!=null&&!a.hasOwnProperty(ae))switch(ae){case"selected":e.selected=!1;break;default:ve(e,n,ae,null,a,D)}for(C in a)if(D=a[C],H=o[C],a.hasOwnProperty(C)&&D!==H&&(D!=null||H!=null))switch(C){case"selected":e.selected=D&&typeof D!="function"&&typeof D!="symbol";break;default:ve(e,n,C,D,a,H)}return;case"img":case"link":case"area":case"base":case"br":case"col":case"embed":case"hr":case"keygen":case"meta":case"param":case"source":case"track":case"wbr":case"menuitem":for(var oe in o)D=o[oe],o.hasOwnProperty(oe)&&D!=null&&!a.hasOwnProperty(oe)&&ve(e,n,oe,null,a,D);for(F in a)if(D=a[F],H=o[F],a.hasOwnProperty(F)&&D!==H&&(D!=null||H!=null))switch(F){case"children":case"dangerouslySetInnerHTML":if(D!=null)throw Error(s(137,n));break;default:ve(e,n,F,D,a,H)}return;default:if(Hl(n)){for(var xe in o)D=o[xe],o.hasOwnProperty(xe)&&D!==void 0&&!a.hasOwnProperty(xe)&&Ru(e,n,xe,void 0,a,D);for(M in a)D=a[M],H=o[M],!a.hasOwnProperty(M)||D===H||D===void 0&&H===void 0||Ru(e,n,M,D,a,H);return}}for(var x in o)D=o[x],o.hasOwnProperty(x)&&D!=null&&!a.hasOwnProperty(x)&&ve(e,n,x,null,a,D);for(W in a)D=a[W],H=o[W],!a.hasOwnProperty(W)||D===H||D==null&&H==null||ve(e,n,W,D,a,H)}var Iu=null,yu=null;function Pa(e){return e.nodeType===9?e:e.ownerDocument}function pg(e){switch(e){case"http://www.w3.org/2000/svg":return 1;case"http://www.w3.org/1998/Math/MathML":return 2;default:return 0}}function hg(e,n){if(e===0)switch(n){case"svg":return 1;case"math":return 2;default:return 0}return e===1&&n==="foreignObject"?0:e}function bu(e,n){return e==="textarea"||e==="noscript"||typeof n.children=="string"||typeof n.children=="number"||typeof n.children=="bigint"||typeof n.dangerouslySetInnerHTML=="object"&&n.dangerouslySetInnerHTML!==null&&n.dangerouslySetInnerHTML.__html!=null}var Cu=null;function IR(){var e=window.event;return e&&e.type==="popstate"?e===Cu?!1:(Cu=e,!0):(Cu=null,!1)}var fg=typeof setTimeout=="function"?setTimeout:void 0,yR=typeof clearTimeout=="function"?clearTimeout:void 0,gg=typeof Promise=="function"?Promise:void 0,bR=typeof queueMicrotask=="function"?queueMicrotask:typeof gg<"u"?function(e){return gg.resolve(null).then(e).catch(CR)}:fg;function CR(e){setTimeout(function(){throw e})}function ci(e){return e==="head"}function mg(e,n){var o=n,a=0,c=0;do{var d=o.nextSibling;if(e.removeChild(o),d&&d.nodeType===8)if(o=d.data,o==="/$"){if(0<a&&8>a){o=a;var m=e.ownerDocument;if(o&1&&$r(m.documentElement),o&2&&$r(m.body),o&4)for(o=m.head,$r(o),m=o.firstChild;m;){var S=m.nextSibling,C=m.nodeName;m[fr]||C==="SCRIPT"||C==="STYLE"||C==="LINK"&&m.rel.toLowerCase()==="stylesheet"||o.removeChild(m),m=S}}if(c===0){e.removeChild(d),as(n);return}c--}else o==="$"||o==="$?"||o==="$!"?c++:a=o.charCodeAt(0)-48;else a=0;o=d}while(o);as(n)}function wu(e){var n=e.firstChild;for(n&&n.nodeType===10&&(n=n.nextSibling);n;){var o=n;switch(n=n.nextSibling,o.nodeName){case"HTML":case"HEAD":case"BODY":wu(o),Pl(o);continue;case"SCRIPT":case"STYLE":continue;case"LINK":if(o.rel.toLowerCase()==="stylesheet")continue}e.removeChild(o)}}function wR(e,n,o,a){for(;e.nodeType===1;){var c=o;if(e.nodeName.toLowerCase()!==n.toLowerCase()){if(!a&&(e.nodeName!=="INPUT"||e.type!=="hidden"))break}else if(a){if(!e[fr])switch(n){case"meta":if(!e.hasAttribute("itemprop"))break;return e;case"link":if(d=e.getAttribute("rel"),d==="stylesheet"&&e.hasAttribute("data-precedence"))break;if(d!==c.rel||e.getAttribute("href")!==(c.href==null||c.href===""?null:c.href)||e.getAttribute("crossorigin")!==(c.crossOrigin==null?null:c.crossOrigin)||e.getAttribute("title")!==(c.title==null?null:c.title))break;return e;case"style":if(e.hasAttribute("data-precedence"))break;return e;case"script":if(d=e.getAttribute("src"),(d!==(c.src==null?null:c.src)||e.getAttribute("type")!==(c.type==null?null:c.type)||e.getAttribute("crossorigin")!==(c.crossOrigin==null?null:c.crossOrigin))&&d&&e.hasAttribute("async")&&!e.hasAttribute("itemprop"))break;return e;default:return e}}else if(n==="input"&&e.type==="hidden"){var d=c.name==null?null:""+c.name;if(c.type==="hidden"&&e.getAttribute("name")===d)return e}else return e;if(e=tn(e.nextSibling),e===null)break}return null}function _R(e,n,o){if(n==="")return null;for(;e.nodeType!==3;)if((e.nodeType!==1||e.nodeName!=="INPUT"||e.type!=="hidden")&&!o||(e=tn(e.nextSibling),e===null))return null;return e}function _u(e){return e.data==="$!"||e.data==="$?"&&e.ownerDocument.readyState==="complete"}function TR(e,n){var o=e.ownerDocument;if(e.data!=="$?"||o.readyState==="complete")n();else{var a=function(){n(),o.removeEventListener("DOMContentLoaded",a)};o.addEventListener("DOMContentLoaded",a),e._reactRetry=a}}function tn(e){for(;e!=null;e=e.nextSibling){var n=e.nodeType;if(n===1||n===3)break;if(n===8){if(n=e.data,n==="$"||n==="$!"||n==="$?"||n==="F!"||n==="F")break;if(n==="/$")return null}}return e}var Tu=null;function Eg(e){e=e.previousSibling;for(var n=0;e;){if(e.nodeType===8){var o=e.data;if(o==="$"||o==="$!"||o==="$?"){if(n===0)return e;n--}else o==="/$"&&n++}e=e.previousSibling}return null}function Sg(e,n,o){switch(n=Pa(o),e){case"html":if(e=n.documentElement,!e)throw Error(s(452));return e;case"head":if(e=n.head,!e)throw Error(s(453));return e;case"body":if(e=n.body,!e)throw Error(s(454));return e;default:throw Error(s(451))}}function $r(e){for(var n=e.attributes;n.length;)e.removeAttributeNode(n[0]);Pl(e)}var Vt=new Map,Rg=new Set;function Xa(e){return typeof e.getRootNode=="function"?e.getRootNode():e.nodeType===9?e:e.ownerDocument}var Ln=V.d;V.d={f:vR,r:xR,D:PR,C:XR,L:NR,m:OR,X:DR,S:FR,M:HR};function vR(){var e=Ln.f(),n=ya();return e||n}function xR(e){var n=ao(e);n!==null&&n.tag===5&&n.type==="form"?Bh(n):Ln.r(e)}var Wo=typeof document>"u"?null:document;function Ig(e,n,o){var a=Wo;if(a&&typeof n=="string"&&n){var c=Ut(n);c='link[rel="'+e+'"][href="'+c+'"]',typeof o=="string"&&(c+='[crossorigin="'+o+'"]'),Rg.has(c)||(Rg.add(c),e={rel:e,crossOrigin:o,href:n},a.querySelector(c)===null&&(n=a.createElement("link"),ot(n,"link",e),Ze(n),a.head.appendChild(n)))}}function PR(e){Ln.D(e),Ig("dns-prefetch",e,null)}function XR(e,n){Ln.C(e,n),Ig("preconnect",e,n)}function NR(e,n,o){Ln.L(e,n,o);var a=Wo;if(a&&e&&n){var c='link[rel="preload"][as="'+Ut(n)+'"]';n==="image"&&o&&o.imageSrcSet?(c+='[imagesrcset="'+Ut(o.imageSrcSet)+'"]',typeof o.imageSizes=="string"&&(c+='[imagesizes="'+Ut(o.imageSizes)+'"]')):c+='[href="'+Ut(e)+'"]';var d=c;switch(n){case"style":d=jo(e);break;case"script":d=ko(e)}Vt.has(d)||(e=E({rel:"preload",href:n==="image"&&o&&o.imageSrcSet?void 0:e,as:n},o),Vt.set(d,e),a.querySelector(c)!==null||n==="style"&&a.querySelector(es(d))||n==="script"&&a.querySelector(ts(d))||(n=a.createElement("link"),ot(n,"link",e),Ze(n),a.head.appendChild(n)))}}function OR(e,n){Ln.m(e,n);var o=Wo;if(o&&e){var a=n&&typeof n.as=="string"?n.as:"script",c='link[rel="modulepreload"][as="'+Ut(a)+'"][href="'+Ut(e)+'"]',d=c;switch(a){case"audioworklet":case"paintworklet":case"serviceworker":case"sharedworker":case"worker":case"script":d=ko(e)}if(!Vt.has(d)&&(e=E({rel:"modulepreload",href:e},n),Vt.set(d,e),o.querySelector(c)===null)){switch(a){case"audioworklet":case"paintworklet":case"serviceworker":case"sharedworker":case"worker":case"script":if(o.querySelector(ts(d)))return}a=o.createElement("link"),ot(a,"link",e),Ze(a),o.head.appendChild(a)}}}function FR(e,n,o){Ln.S(e,n,o);var a=Wo;if(a&&e){var c=lo(a).hoistableStyles,d=jo(e);n=n||"default";var m=c.get(d);if(!m){var S={loading:0,preload:null};if(m=a.querySelector(es(d)))S.loading=5;else{e=E({rel:"stylesheet",href:e,"data-precedence":n},o),(o=Vt.get(d))&&vu(e,o);var C=m=a.createElement("link");Ze(C),ot(C,"link",e),C._p=new Promise(function(F,M){C.onload=F,C.onerror=M}),C.addEventListener("load",function(){S.loading|=1}),C.addEventListener("error",function(){S.loading|=2}),S.loading|=4,Na(m,n,a)}m={type:"stylesheet",instance:m,count:1,state:S},c.set(d,m)}}}function DR(e,n){Ln.X(e,n);var o=Wo;if(o&&e){var a=lo(o).hoistableScripts,c=ko(e),d=a.get(c);d||(d=o.querySelector(ts(c)),d||(e=E({src:e,async:!0},n),(n=Vt.get(c))&&xu(e,n),d=o.createElement("script"),Ze(d),ot(d,"link",e),o.head.appendChild(d)),d={type:"script",instance:d,count:1,state:null},a.set(c,d))}}function HR(e,n){Ln.M(e,n);var o=Wo;if(o&&e){var a=lo(o).hoistableScripts,c=ko(e),d=a.get(c);d||(d=o.querySelector(ts(c)),d||(e=E({src:e,async:!0,type:"module"},n),(n=Vt.get(c))&&xu(e,n),d=o.createElement("script"),Ze(d),ot(d,"link",e),o.head.appendChild(d)),d={type:"script",instance:d,count:1,state:null},a.set(c,d))}}function yg(e,n,o,a){var c=(c=le.current)?Xa(c):null;if(!c)throw Error(s(446));switch(e){case"meta":case"title":return null;case"style":return typeof o.precedence=="string"&&typeof o.href=="string"?(n=jo(o.href),o=lo(c).hoistableStyles,a=o.get(n),a||(a={type:"style",instance:null,count:0,state:null},o.set(n,a)),a):{type:"void",instance:null,count:0,state:null};case"link":if(o.rel==="stylesheet"&&typeof o.href=="string"&&typeof o.precedence=="string"){e=jo(o.href);var d=lo(c).hoistableStyles,m=d.get(e);if(m||(c=c.ownerDocument||c,m={type:"stylesheet",instance:null,count:0,state:{loading:0,preload:null}},d.set(e,m),(d=c.querySelector(es(e)))&&!d._p&&(m.instance=d,m.state.loading=5),Vt.has(e)||(o={rel:"preload",as:"style",href:o.href,crossOrigin:o.crossOrigin,integrity:o.integrity,media:o.media,hrefLang:o.hrefLang,referrerPolicy:o.referrerPolicy},Vt.set(e,o),d||AR(c,e,o,m.state))),n&&a===null)throw Error(s(528,""));return m}if(n&&a!==null)throw Error(s(529,""));return null;case"script":return n=o.async,o=o.src,typeof o=="string"&&n&&typeof n!="function"&&typeof n!="symbol"?(n=ko(o),o=lo(c).hoistableScripts,a=o.get(n),a||(a={type:"script",instance:null,count:0,state:null},o.set(n,a)),a):{type:"void",instance:null,count:0,state:null};default:throw Error(s(444,e))}}function jo(e){return'href="'+Ut(e)+'"'}function es(e){return'link[rel="stylesheet"]['+e+"]"}function bg(e){return E({},e,{"data-precedence":e.precedence,precedence:null})}function AR(e,n,o,a){e.querySelector('link[rel="preload"][as="style"]['+n+"]")?a.loading=1:(n=e.createElement("link"),a.preload=n,n.addEventListener("load",function(){return a.loading|=1}),n.addEventListener("error",function(){return a.loading|=2}),ot(n,"link",o),Ze(n),e.head.appendChild(n))}function ko(e){return'[src="'+Ut(e)+'"]'}function ts(e){return"script[async]"+e}function Cg(e,n,o){if(n.count++,n.instance===null)switch(n.type){case"style":var a=e.querySelector('style[data-href~="'+Ut(o.href)+'"]');if(a)return n.instance=a,Ze(a),a;var c=E({},o,{"data-href":o.href,"data-precedence":o.precedence,href:null,precedence:null});return a=(e.ownerDocument||e).createElement("style"),Ze(a),ot(a,"style",c),Na(a,o.precedence,e),n.instance=a;case"stylesheet":c=jo(o.href);var d=e.querySelector(es(c));if(d)return n.state.loading|=4,n.instance=d,Ze(d),d;a=bg(o),(c=Vt.get(c))&&vu(a,c),d=(e.ownerDocument||e).createElement("link"),Ze(d);var m=d;return m._p=new Promise(function(S,C){m.onload=S,m.onerror=C}),ot(d,"link",a),n.state.loading|=4,Na(d,o.precedence,e),n.instance=d;case"script":return d=ko(o.src),(c=e.querySelector(ts(d)))?(n.instance=c,Ze(c),c):(a=o,(c=Vt.get(d))&&(a=E({},o),xu(a,c)),e=e.ownerDocument||e,c=e.createElement("script"),Ze(c),ot(c,"link",a),e.head.appendChild(c),n.instance=c);case"void":return null;default:throw Error(s(443,n.type))}else n.type==="stylesheet"&&(n.state.loading&4)===0&&(a=n.instance,n.state.loading|=4,Na(a,o.precedence,e));return n.instance}function Na(e,n,o){for(var a=o.querySelectorAll('link[rel="stylesheet"][data-precedence],style[data-precedence]'),c=a.length?a[a.length-1]:null,d=c,m=0;m<a.length;m++){var S=a[m];if(S.dataset.precedence===n)d=S;else if(d!==c)break}d?d.parentNode.insertBefore(e,d.nextSibling):(n=o.nodeType===9?o.head:o,n.insertBefore(e,n.firstChild))}function vu(e,n){e.crossOrigin==null&&(e.crossOrigin=n.crossOrigin),e.referrerPolicy==null&&(e.referrerPolicy=n.referrerPolicy),e.title==null&&(e.title=n.title)}function xu(e,n){e.crossOrigin==null&&(e.crossOrigin=n.crossOrigin),e.referrerPolicy==null&&(e.referrerPolicy=n.referrerPolicy),e.integrity==null&&(e.integrity=n.integrity)}var Oa=null;function wg(e,n,o){if(Oa===null){var a=new Map,c=Oa=new Map;c.set(o,a)}else c=Oa,a=c.get(o),a||(a=new Map,c.set(o,a));if(a.has(e))return a;for(a.set(e,null),o=o.getElementsByTagName(e),c=0;c<o.length;c++){var d=o[c];if(!(d[fr]||d[st]||e==="link"&&d.getAttribute("rel")==="stylesheet")&&d.namespaceURI!=="http://www.w3.org/2000/svg"){var m=d.getAttribute(n)||"";m=e+m;var S=a.get(m);S?S.push(d):a.set(m,[d])}}return a}function _g(e,n,o){e=e.ownerDocument||e,e.head.insertBefore(o,n==="title"?e.querySelector("head > title"):null)}function LR(e,n,o){if(o===1||n.itemProp!=null)return!1;switch(e){case"meta":case"title":return!0;case"style":if(typeof n.precedence!="string"||typeof n.href!="string"||n.href==="")break;return!0;case"link":if(typeof n.rel!="string"||typeof n.href!="string"||n.href===""||n.onLoad||n.onError)break;switch(n.rel){case"stylesheet":return e=n.disabled,typeof n.precedence=="string"&&e==null;default:return!0}case"script":if(n.async&&typeof n.async!="function"&&typeof n.async!="symbol"&&!n.onLoad&&!n.onError&&n.src&&typeof n.src=="string")return!0}return!1}function Tg(e){return!(e.type==="stylesheet"&&(e.state.loading&3)===0)}var ns=null;function BR(){}function MR(e,n,o){if(ns===null)throw Error(s(475));var a=ns;if(n.type==="stylesheet"&&(typeof o.media!="string"||matchMedia(o.media).matches!==!1)&&(n.state.loading&4)===0){if(n.instance===null){var c=jo(o.href),d=e.querySelector(es(c));if(d){e=d._p,e!==null&&typeof e=="object"&&typeof e.then=="function"&&(a.count++,a=Fa.bind(a),e.then(a,a)),n.state.loading|=4,n.instance=d,Ze(d);return}d=e.ownerDocument||e,o=bg(o),(c=Vt.get(c))&&vu(o,c),d=d.createElement("link"),Ze(d);var m=d;m._p=new Promise(function(S,C){m.onload=S,m.onerror=C}),ot(d,"link",o),n.instance=d}a.stylesheets===null&&(a.stylesheets=new Map),a.stylesheets.set(n,e),(e=n.state.preload)&&(n.state.loading&3)===0&&(a.count++,n=Fa.bind(a),e.addEventListener("load",n),e.addEventListener("error",n))}}function qR(){if(ns===null)throw Error(s(475));var e=ns;return e.stylesheets&&e.count===0&&Pu(e,e.stylesheets),0<e.count?function(n){var o=setTimeout(function(){if(e.stylesheets&&Pu(e,e.stylesheets),e.unsuspend){var a=e.unsuspend;e.unsuspend=null,a()}},6e4);return e.unsuspend=n,function(){e.unsuspend=null,clearTimeout(o)}}:null}function Fa(){if(this.count--,this.count===0){if(this.stylesheets)Pu(this,this.stylesheets);else if(this.unsuspend){var e=this.unsuspend;this.unsuspend=null,e()}}}var Da=null;function Pu(e,n){e.stylesheets=null,e.unsuspend!==null&&(e.count++,Da=new Map,n.forEach(UR,e),Da=null,Fa.call(e))}function UR(e,n){if(!(n.state.loading&4)){var o=Da.get(e);if(o)var a=o.get(null);else{o=new Map,Da.set(e,o);for(var c=e.querySelectorAll("link[data-precedence],style[data-precedence]"),d=0;d<c.length;d++){var m=c[d];(m.nodeName==="LINK"||m.getAttribute("media")!=="not all")&&(o.set(m.dataset.precedence,m),a=m)}a&&o.set(null,a)}c=n.instance,m=c.getAttribute("data-precedence"),d=o.get(m)||a,d===a&&o.set(null,c),o.set(m,c),this.count++,a=Fa.bind(this),c.addEventListener("load",a),c.addEventListener("error",a),d?d.parentNode.insertBefore(c,d.nextSibling):(e=e.nodeType===9?e.head:e,e.insertBefore(c,e.firstChild)),n.state.loading|=4}}var is={$$typeof:k,Provider:null,Consumer:null,_currentValue:ne,_currentValue2:ne,_threadCount:0};function WR(e,n,o,a,c,d,m,S){this.tag=1,this.containerInfo=e,this.pingCache=this.current=this.pendingChildren=null,this.timeoutHandle=-1,this.callbackNode=this.next=this.pendingContext=this.context=this.cancelPendingCommit=null,this.callbackPriority=0,this.expirationTimes=_l(-1),this.entangledLanes=this.shellSuspendCounter=this.errorRecoveryDisabledLanes=this.expiredLanes=this.warmLanes=this.pingedLanes=this.suspendedLanes=this.pendingLanes=0,this.entanglements=_l(0),this.hiddenUpdates=_l(null),this.identifierPrefix=a,this.onUncaughtError=c,this.onCaughtError=d,this.onRecoverableError=m,this.pooledCache=null,this.pooledCacheLanes=0,this.formState=S,this.incompleteTransitions=new Map}function vg(e,n,o,a,c,d,m,S,C,F,M,W){return e=new WR(e,n,o,m,S,C,F,W),n=1,d===!0&&(n|=24),d=Xt(3,null,null,n),e.current=d,d.stateNode=e,n=uc(),n.refCount++,e.pooledCache=n,n.refCount++,d.memoizedState={element:a,isDehydrated:o,cache:n},fc(d),e}function xg(e){return e?(e=yo,e):yo}function Pg(e,n,o,a,c,d){c=xg(c),a.context===null?a.context=c:a.pendingContext=c,a=Kn(n),a.payload={element:o},d=d===void 0?null:d,d!==null&&(a.callback=d),o=Zn(e,a,n),o!==null&&(Ht(o,e,n),Fr(o,e,n))}function Xg(e,n){if(e=e.memoizedState,e!==null&&e.dehydrated!==null){var o=e.retryLane;e.retryLane=o!==0&&o<n?o:n}}function Xu(e,n){Xg(e,n),(e=e.alternate)&&Xg(e,n)}function Ng(e){if(e.tag===13){var n=Io(e,67108864);n!==null&&Ht(n,e,67108864),Xu(e,67108864)}}var Ha=!0;function jR(e,n,o,a){var c=L.T;L.T=null;var d=V.p;try{V.p=2,Nu(e,n,o,a)}finally{V.p=d,L.T=c}}function kR(e,n,o,a){var c=L.T;L.T=null;var d=V.p;try{V.p=8,Nu(e,n,o,a)}finally{V.p=d,L.T=c}}function Nu(e,n,o,a){if(Ha){var c=Ou(a);if(c===null)Su(e,n,a,Aa,o),Fg(e,a);else if(GR(c,e,n,o,a))a.stopPropagation();else if(Fg(e,a),n&4&&-1<zR.indexOf(e)){for(;c!==null;){var d=ao(c);if(d!==null)switch(d.tag){case 3:if(d=d.stateNode,d.current.memoizedState.isDehydrated){var m=wi(d.pendingLanes);if(m!==0){var S=d;for(S.pendingLanes|=2,S.entangledLanes|=2;m;){var C=1<<31-xt(m);S.entanglements[1]|=C,m&=~C}fn(d),(Ce&6)===0&&(Ra=cn()+500,Kr(0))}}break;case 13:S=Io(d,2),S!==null&&Ht(S,d,2),ya(),Xu(d,2)}if(d=Ou(a),d===null&&Su(e,n,a,Aa,o),d===c)break;c=d}c!==null&&a.stopPropagation()}else Su(e,n,a,null,o)}}function Ou(e){return e=Ll(e),Fu(e)}var Aa=null;function Fu(e){if(Aa=null,e=so(e),e!==null){var n=u(e);if(n===null)e=null;else{var o=n.tag;if(o===13){if(e=p(n),e!==null)return e;e=null}else if(o===3){if(n.stateNode.current.memoizedState.isDehydrated)return n.tag===3?n.stateNode.containerInfo:null;e=null}else n!==e&&(e=null)}}return Aa=e,null}function Og(e){switch(e){case"beforetoggle":case"cancel":case"click":case"close":case"contextmenu":case"copy":case"cut":case"auxclick":case"dblclick":case"dragend":case"dragstart":case"drop":case"focusin":case"focusout":case"input":case"invalid":case"keydown":case"keypress":case"keyup":case"mousedown":case"mouseup":case"paste":case"pause":case"play":case"pointercancel":case"pointerdown":case"pointerup":case"ratechange":case"reset":case"resize":case"seeked":case"submit":case"toggle":case"touchcancel":case"touchend":case"touchstart":case"volumechange":case"change":case"selectionchange":case"textInput":case"compositionstart":case"compositionend":case"compositionupdate":case"beforeblur":case"afterblur":case"beforeinput":case"blur":case"fullscreenchange":case"focus":case"hashchange":case"popstate":case"select":case"selectstart":return 2;case"drag":case"dragenter":case"dragexit":case"dragleave":case"dragover":case"mousemove":case"mouseout":case"mouseover":case"pointermove":case"pointerout":case"pointerover":case"scroll":case"touchmove":case"wheel":case"mouseenter":case"mouseleave":case"pointerenter":case"pointerleave":return 8;case"message":switch(XS()){case zd:return 2;case Gd:return 8;case xs:case NS:return 32;case Qd:return 268435456;default:return 32}default:return 32}}var Du=!1,ui=null,di=null,pi=null,os=new Map,rs=new Map,hi=[],zR="mousedown mouseup touchcancel touchend touchstart auxclick dblclick pointercancel pointerdown pointerup dragend dragstart drop compositionend compositionstart keydown keypress keyup input textInput copy cut paste click change contextmenu reset".split(" ");function Fg(e,n){switch(e){case"focusin":case"focusout":ui=null;break;case"dragenter":case"dragleave":di=null;break;case"mouseover":case"mouseout":pi=null;break;case"pointerover":case"pointerout":os.delete(n.pointerId);break;case"gotpointercapture":case"lostpointercapture":rs.delete(n.pointerId)}}function ss(e,n,o,a,c,d){return e===null||e.nativeEvent!==d?(e={blockedOn:n,domEventName:o,eventSystemFlags:a,nativeEvent:d,targetContainers:[c]},n!==null&&(n=ao(n),n!==null&&Ng(n)),e):(e.eventSystemFlags|=a,n=e.targetContainers,c!==null&&n.indexOf(c)===-1&&n.push(c),e)}function GR(e,n,o,a,c){switch(n){case"focusin":return ui=ss(ui,e,n,o,a,c),!0;case"dragenter":return di=ss(di,e,n,o,a,c),!0;case"mouseover":return pi=ss(pi,e,n,o,a,c),!0;case"pointerover":var d=c.pointerId;return os.set(d,ss(os.get(d)||null,e,n,o,a,c)),!0;case"gotpointercapture":return d=c.pointerId,rs.set(d,ss(rs.get(d)||null,e,n,o,a,c)),!0}return!1}function Dg(e){var n=so(e.target);if(n!==null){var o=u(n);if(o!==null){if(n=o.tag,n===13){if(n=p(o),n!==null){e.blockedOn=n,MS(e.priority,function(){if(o.tag===13){var a=Dt();a=Tl(a);var c=Io(o,a);c!==null&&Ht(c,o,a),Xu(o,a)}});return}}else if(n===3&&o.stateNode.current.memoizedState.isDehydrated){e.blockedOn=o.tag===3?o.stateNode.containerInfo:null;return}}}e.blockedOn=null}function La(e){if(e.blockedOn!==null)return!1;for(var n=e.targetContainers;0<n.length;){var o=Ou(e.nativeEvent);if(o===null){o=e.nativeEvent;var a=new o.constructor(o.type,o);Al=a,o.target.dispatchEvent(a),Al=null}else return n=ao(o),n!==null&&Ng(n),e.blockedOn=o,!1;n.shift()}return!0}function Hg(e,n,o){La(e)&&o.delete(n)}function QR(){Du=!1,ui!==null&&La(ui)&&(ui=null),di!==null&&La(di)&&(di=null),pi!==null&&La(pi)&&(pi=null),os.forEach(Hg),rs.forEach(Hg)}function Ba(e,n){e.blockedOn===n&&(e.blockedOn=null,Du||(Du=!0,i.unstable_scheduleCallback(i.unstable_NormalPriority,QR)))}var Ma=null;function Ag(e){Ma!==e&&(Ma=e,i.unstable_scheduleCallback(i.unstable_NormalPriority,function(){Ma===e&&(Ma=null);for(var n=0;n<e.length;n+=3){var o=e[n],a=e[n+1],c=e[n+2];if(typeof a!="function"){if(Fu(a||o)===null)continue;break}var d=ao(o);d!==null&&(e.splice(n,3),n-=3,Fc(d,{pending:!0,data:c,method:o.method,action:a},a,c))}}))}function as(e){function n(C){return Ba(C,e)}ui!==null&&Ba(ui,e),di!==null&&Ba(di,e),pi!==null&&Ba(pi,e),os.forEach(n),rs.forEach(n);for(var o=0;o<hi.length;o++){var a=hi[o];a.blockedOn===e&&(a.blockedOn=null)}for(;0<hi.length&&(o=hi[0],o.blockedOn===null);)Dg(o),o.blockedOn===null&&hi.shift();if(o=(e.ownerDocument||e).$$reactFormReplay,o!=null)for(a=0;a<o.length;a+=3){var c=o[a],d=o[a+1],m=c[St]||null;if(typeof d=="function")m||Ag(o);else if(m){var S=null;if(d&&d.hasAttribute("formAction")){if(c=d,m=d[St]||null)S=m.formAction;else if(Fu(c)!==null)continue}else S=m.action;typeof S=="function"?o[a+1]=S:(o.splice(a,3),a-=3),Ag(o)}}}function Hu(e){this._internalRoot=e}qa.prototype.render=Hu.prototype.render=function(e){var n=this._internalRoot;if(n===null)throw Error(s(409));var o=n.current,a=Dt();Pg(o,a,e,n,null,null)},qa.prototype.unmount=Hu.prototype.unmount=function(){var e=this._internalRoot;if(e!==null){this._internalRoot=null;var n=e.containerInfo;Pg(e.current,2,null,e,null,null),ya(),n[ro]=null}};function qa(e){this._internalRoot=e}qa.prototype.unstable_scheduleHydration=function(e){if(e){var n=Jd();e={blockedOn:null,target:e,priority:n};for(var o=0;o<hi.length&&n!==0&&n<hi[o].priority;o++);hi.splice(o,0,e),o===0&&Dg(e)}};var Lg=t.version;if(Lg!=="19.1.0")throw Error(s(527,Lg,"19.1.0"));V.findDOMNode=function(e){var n=e._reactInternals;if(n===void 0)throw typeof e.render=="function"?Error(s(188)):(e=Object.keys(e).join(","),Error(s(268,e)));return e=f(n),e=e!==null?g(e):null,e=e===null?null:e.stateNode,e};var VR={bundleType:0,version:"19.1.0",rendererPackageName:"react-dom",currentDispatcherRef:L,reconcilerVersion:"19.1.0"};if(typeof __REACT_DEVTOOLS_GLOBAL_HOOK__<"u"){var Ua=__REACT_DEVTOOLS_GLOBAL_HOOK__;if(!Ua.isDisabled&&Ua.supportsFiber)try{dr=Ua.inject(VR),vt=Ua}catch{}}return cs.createRoot=function(e,n){if(!l(e))throw Error(s(299));var o=!1,a="",c=Jh,d=$h,m=ef,S=null;return n!=null&&(n.unstable_strictMode===!0&&(o=!0),n.identifierPrefix!==void 0&&(a=n.identifierPrefix),n.onUncaughtError!==void 0&&(c=n.onUncaughtError),n.onCaughtError!==void 0&&(d=n.onCaughtError),n.onRecoverableError!==void 0&&(m=n.onRecoverableError),n.unstable_transitionCallbacks!==void 0&&(S=n.unstable_transitionCallbacks)),n=vg(e,1,!1,null,null,o,a,c,d,m,S,null),e[ro]=n.current,Eu(e),new Hu(n)},cs.hydrateRoot=function(e,n,o){if(!l(e))throw Error(s(299));var a=!1,c="",d=Jh,m=$h,S=ef,C=null,F=null;return o!=null&&(o.unstable_strictMode===!0&&(a=!0),o.identifierPrefix!==void 0&&(c=o.identifierPrefix),o.onUncaughtError!==void 0&&(d=o.onUncaughtError),o.onCaughtError!==void 0&&(m=o.onCaughtError),o.onRecoverableError!==void 0&&(S=o.onRecoverableError),o.unstable_transitionCallbacks!==void 0&&(C=o.unstable_transitionCallbacks),o.formState!==void 0&&(F=o.formState)),n=vg(e,1,!0,n,o??null,a,c,d,m,S,C,F),n.context=xg(null),o=n.current,a=Dt(),a=Tl(a),c=Kn(a),c.callback=null,Zn(o,c,a),o=a,n.current.lanes=o,hr(n,o),fn(n),e[ro]=n.current,Eu(e),new qa(n)},cs.version="19.1.0",cs}var Qg;function rI(){if(Qg)return Bu.exports;Qg=1;function i(){if(!(typeof __REACT_DEVTOOLS_GLOBAL_HOOK__>"u"||typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE!="function"))try{__REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE(i)}catch(t){console.error(t)}}return i(),Bu.exports=oI(),Bu.exports}var sI=rI();const pE="/Sparql-fest/assets/logo-PAti6GV0.png";var hE={color:void 0,size:void 0,className:void 0,style:void 0,attr:void 0},Vg=wt.createContext&&wt.createContext(hE),aI=["attr","size","title"];function lI(i,t){if(i==null)return{};var r=cI(i,t),s,l;if(Object.getOwnPropertySymbols){var u=Object.getOwnPropertySymbols(i);for(l=0;l<u.length;l++)s=u[l],!(t.indexOf(s)>=0)&&Object.prototype.propertyIsEnumerable.call(i,s)&&(r[s]=i[s])}return r}function cI(i,t){if(i==null)return{};var r={};for(var s in i)if(Object.prototype.hasOwnProperty.call(i,s)){if(t.indexOf(s)>=0)continue;r[s]=i[s]}return r}function sl(){return sl=Object.assign?Object.assign.bind():function(i){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var s in r)Object.prototype.hasOwnProperty.call(r,s)&&(i[s]=r[s])}return i},sl.apply(this,arguments)}function Yg(i,t){var r=Object.keys(i);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(i);t&&(s=s.filter(function(l){return Object.getOwnPropertyDescriptor(i,l).enumerable})),r.push.apply(r,s)}return r}function al(i){for(var t=1;t<arguments.length;t++){var r=arguments[t]!=null?arguments[t]:{};t%2?Yg(Object(r),!0).forEach(function(s){uI(i,s,r[s])}):Object.getOwnPropertyDescriptors?Object.defineProperties(i,Object.getOwnPropertyDescriptors(r)):Yg(Object(r)).forEach(function(s){Object.defineProperty(i,s,Object.getOwnPropertyDescriptor(r,s))})}return i}function uI(i,t,r){return t=dI(t),t in i?Object.defineProperty(i,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):i[t]=r,i}function dI(i){var t=pI(i,"string");return typeof t=="symbol"?t:t+""}function pI(i,t){if(typeof i!="object"||!i)return i;var r=i[Symbol.toPrimitive];if(r!==void 0){var s=r.call(i,t);if(typeof s!="object")return s;throw new TypeError("@@toPrimitive must return a primitive value.")}return(t==="string"?String:Number)(i)}function fE(i){return i&&i.map((t,r)=>wt.createElement(t.tag,al({key:r},t.attr),fE(t.child)))}function bi(i){return t=>wt.createElement(hI,sl({attr:al({},i.attr)},t),fE(i.child))}function hI(i){var t=r=>{var{attr:s,size:l,title:u}=i,p=lI(i,aI),h=l||r.size||"1em",f;return r.className&&(f=r.className),i.className&&(f=(f?f+" ":"")+i.className),wt.createElement("svg",sl({stroke:"currentColor",fill:"currentColor",strokeWidth:"0"},r.attr,s,p,{className:f,style:al(al({color:i.color||r.color},r.style),i.style),height:h,width:h,xmlns:"http://www.w3.org/2000/svg"}),u&&wt.createElement("title",null,u),i.children)};return Vg!==void 0?wt.createElement(Vg.Consumer,null,r=>t(r)):t(hE)}function fI(i){return bi({attr:{viewBox:"0 0 512 512"},child:[{tag:"path",attr:{d:"M352 96c0-53.02-42.98-96-96-96s-96 42.98-96 96 42.98 96 96 96 96-42.98 96-96zM233.59 241.1c-59.33-36.32-155.43-46.3-203.79-49.05C13.55 191.13 0 203.51 0 219.14v222.8c0 14.33 11.59 26.28 26.49 27.05 43.66 2.29 131.99 10.68 193.04 41.43 9.37 4.72 20.48-1.71 20.48-11.87V252.56c-.01-4.67-2.32-8.95-6.42-11.46zm248.61-49.05c-48.35 2.74-144.46 12.73-203.78 49.05-4.1 2.51-6.41 6.96-6.41 11.63v245.79c0 10.19 11.14 16.63 20.54 11.9 61.04-30.72 149.32-39.11 192.97-41.4 14.9-.78 26.49-12.73 26.49-27.06V219.14c-.01-15.63-13.56-28.01-29.81-27.09z"},child:[]}]})(i)}function gI(i){return bi({attr:{viewBox:"0 0 512 512"},child:[{tag:"path",attr:{d:"M501.1 395.7L384 278.6c-23.1-23.1-57.6-27.6-85.4-13.9L192 158.1V96L64 0 0 64l96 128h62.1l106.6 106.6c-13.6 27.8-9.2 62.3 13.9 85.4l117.1 117.1c14.6 14.6 38.2 14.6 52.7 0l52.7-52.7c14.5-14.6 14.5-38.2 0-52.7zM331.7 225c28.3 0 54.9 11 74.9 31l19.4 19.4c15.8-6.9 30.8-16.5 43.8-29.5 37.1-37.1 49.7-89.3 37.9-136.7-2.2-9-13.5-12.1-20.1-5.5l-74.4 74.4-67.9-11.3L334 98.9l74.4-74.4c6.6-6.6 3.4-17.9-5.7-20.2-47.4-11.7-99.6.9-136.6 37.9-28.5 28.5-41.9 66.1-41.2 103.6l82.1 82.1c8.1-1.9 16.5-2.9 24.7-2.9zm-103.9 82l-56.7-56.7L18.7 402.8c-25 25-25 65.5 0 90.5s65.5 25 90.5 0l123.6-123.6c-7.6-19.9-9.9-41.6-5-62.7zM64 472c-13.2 0-24-10.8-24-24 0-13.3 10.7-24 24-24s24 10.7 24 24c0 13.2-10.7 24-24 24z"},child:[]}]})(i)}function mI(i){return bi({attr:{viewBox:"0 0 24 24"},child:[{tag:"path",attr:{fill:"none",d:"M0 0h24v24H0V0z"},child:[]},{tag:"path",attr:{d:"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5 7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"},child:[]}]})(i)}function EI(i){return bi({attr:{viewBox:"0 0 24 24"},child:[{tag:"path",attr:{fill:"none",d:"M0 0h24v24H0z"},child:[]},{tag:"path",attr:{d:"M19.3 16.9c.4-.7.7-1.5.7-2.4 0-2.5-2-4.5-4.5-4.5S11 12 11 14.5s2 4.5 4.5 4.5c.9 0 1.7-.3 2.4-.7l3.2 3.2 1.4-1.4-3.2-3.2zm-3.8.1c-1.4 0-2.5-1.1-2.5-2.5s1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5-1.1 2.5-2.5 2.5zM12 20v2C6.48 22 2 17.52 2 12S6.48 2 12 2c4.84 0 8.87 3.44 9.8 8h-2.07A8 8 0 0 0 15 4.59V5c0 1.1-.9 2-2 2h-2v2c0 .55-.45 1-1 1H8v2h2v3H9l-4.79-4.79C4.08 10.79 4 11.38 4 12c0 4.41 3.59 8 8 8z"},child:[]}]})(i)}function SI(i){return bi({attr:{viewBox:"0 0 24 24",fill:"currentColor","aria-hidden":"true"},child:[{tag:"path",attr:{d:"M11.584 2.376a.75.75 0 0 1 .832 0l9 6a.75.75 0 1 1-.832 1.248L12 3.901 3.416 9.624a.75.75 0 0 1-.832-1.248l9-6Z"},child:[]},{tag:"path",attr:{fillRule:"evenodd",d:"M20.25 10.332v9.918H21a.75.75 0 0 1 0 1.5H3a.75.75 0 0 1 0-1.5h.75v-9.918a.75.75 0 0 1 .634-.74A49.109 49.109 0 0 1 12 9c2.59 0 5.134.202 7.616.592a.75.75 0 0 1 .634.74Zm-7.5 2.418a.75.75 0 0 0-1.5 0v6.75a.75.75 0 0 0 1.5 0v-6.75Zm3-.75a.75.75 0 0 1 .75.75v6.75a.75.75 0 0 1-1.5 0v-6.75a.75.75 0 0 1 .75-.75ZM9 12.75a.75.75 0 0 0-1.5 0v6.75a.75.75 0 0 0 1.5 0v-6.75Z",clipRule:"evenodd"},child:[]},{tag:"path",attr:{d:"M12 7.875a1.125 1.125 0 1 0 0-2.25 1.125 1.125 0 0 0 0 2.25Z"},child:[]}]})(i)}function RI(i){return bi({attr:{fill:"none",viewBox:"0 0 24 24",strokeWidth:"2",stroke:"currentColor","aria-hidden":"true"},child:[{tag:"path",attr:{strokeLinecap:"round",strokeLinejoin:"round",d:"M9.663 17h4.673M12 3v1m6.364 1.636l-.707.707M21 12h-1M4 12H3m3.343-5.657l-.707-.707m2.828 9.9a5 5 0 117.072 0l-.548.547A3.374 3.374 0 0014 18.469V19a2 2 0 11-4 0v-.531c0-.895-.356-1.754-.988-2.386l-.548-.547z"},child:[]}]})(i)}const Kg=[{label:"SPARQL Queries",href:"/sparql-queries"},{label:"SPARQL Prefixes",href:"/sparql-prefixes"},{label:"Tutorials",href:"/tutorials"},{label:"Other Resources",href:"/other-resources"}],II=[{icon:I.jsx(EI,{}),text:"Explore the Semantic Web",description:"SPARQL is the gateway to the Semantic Web. It lets you interact with structured knowledge published by institutions, researchers, and communities across the globe."},{icon:I.jsx(mI,{}),text:"Navigate Ontologies and Linked Data",description:"SPARQL gives you the power to query complex ontologies like Wikidata, Orphanet, and more. You can extract relationships, hierarchies, and rich metadata effortlessly."},{icon:I.jsx(SI,{}),text:"Used by Major Organizations",description:"Governments, universities, libraries, biotech companies, and open data initiatives all use SPARQL. It’s a trusted standard backed by the W3C."},{icon:I.jsx(RI,{}),text:"Make Sense of Complex Data",description:"With SPARQL, you can ask meaningful questions across distributed datasets. It helps reveal connections hidden in traditional databases."},{icon:I.jsx(gI,{}),text:"Build Interoperable Applications",description:"SPARQL supports data integration across systems. It’s ideal for apps that consume, transform, and connect semantic data from diverse sources."},{icon:I.jsx(fI,{}),text:"Learn Concepts That Last",description:"SPARQL teaches you how to think semantically. You’ll learn about triples, classes, relationships, and how meaning is represented in data."}],yI=[{title:"Learn SPARQL through real examples",description:"Practice SPARQL with authentic queries to grasp its practical use quickly"},{title:"Explore queries from various domains",description:"Discover diverse SPARQL queries covering healthcare, culture, and biodiversity."},{title:"Connect with people, data, and ideas",description:"Engage with a global community to exchange knowledge and build collaborations."},{title:"Discover what’s possible with the Semantic Web",description:"Unlock new potentials by integrating data and innovating with SPARQL."}];/**
 * @license lucide-react v0.513.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */const bI=i=>i.replace(/([a-z0-9])([A-Z])/g,"$1-$2").toLowerCase(),CI=i=>i.replace(/^([A-Z])|[\s-_]+(\w)/g,(t,r,s)=>s?s.toUpperCase():r.toLowerCase()),Zg=i=>{const t=CI(i);return t.charAt(0).toUpperCase()+t.slice(1)},gE=(...i)=>i.filter((t,r,s)=>!!t&&t.trim()!==""&&s.indexOf(t)===r).join(" ").trim(),wI=i=>{for(const t in i)if(t.startsWith("aria-")||t==="role"||t==="title")return!0};/**
 * @license lucide-react v0.513.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */var _I={xmlns:"http://www.w3.org/2000/svg",width:24,height:24,viewBox:"0 0 24 24",fill:"none",stroke:"currentColor",strokeWidth:2,strokeLinecap:"round",strokeLinejoin:"round"};/**
 * @license lucide-react v0.513.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */const TI=P.forwardRef(({color:i="currentColor",size:t=24,strokeWidth:r=2,absoluteStrokeWidth:s,className:l="",children:u,iconNode:p,...h},f)=>P.createElement("svg",{ref:f,..._I,width:t,height:t,stroke:i,strokeWidth:s?Number(r)*24/Number(t):r,className:gE("lucide",l),...!u&&!wI(h)&&{"aria-hidden":"true"},...h},[...p.map(([g,E])=>P.createElement(g,E)),...Array.isArray(u)?u:[u]]));/**
 * @license lucide-react v0.513.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */const vd=(i,t)=>{const r=P.forwardRef(({className:s,...l},u)=>P.createElement(TI,{ref:u,iconNode:t,className:gE(`lucide-${bI(Zg(i))}`,`lucide-${i}`,s),...l}));return r.displayName=Zg(i),r};/**
 * @license lucide-react v0.513.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */const vI=[["circle",{cx:"12",cy:"12",r:"10",key:"1mglay"}],["path",{d:"m9 12 2 2 4-4",key:"dzmm74"}]],xI=vd("circle-check",vI);/**
 * @license lucide-react v0.513.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */const PI=[["path",{d:"M4 12h16",key:"1lakjw"}],["path",{d:"M4 18h16",key:"19g7jn"}],["path",{d:"M4 6h16",key:"1o0s65"}]],XI=vd("menu",PI);/**
 * @license lucide-react v0.513.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */const NI=[["path",{d:"M18 6 6 18",key:"1bl5f8"}],["path",{d:"m6 6 12 12",key:"d8bk6v"}]],OI=vd("x",NI);var us={},Jg;function FI(){if(Jg)return us;Jg=1,Object.defineProperty(us,"__esModule",{value:!0}),us.parse=p,us.serialize=g;const i=/^[\u0021-\u003A\u003C\u003E-\u007E]+$/,t=/^[\u0021-\u003A\u003C-\u007E]*$/,r=/^([.]?[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)([.][a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)*$/i,s=/^[\u0020-\u003A\u003D-\u007E]*$/,l=Object.prototype.toString,u=(()=>{const y=function(){};return y.prototype=Object.create(null),y})();function p(y,w){const b=new u,X=y.length;if(X<2)return b;const N=(w==null?void 0:w.decode)||E;let v=0;do{const A=y.indexOf("=",v);if(A===-1)break;const k=y.indexOf(";",v),U=k===-1?X:k;if(A>U){v=y.lastIndexOf(";",A-1)+1;continue}const j=h(y,v,A),K=f(y,A,j),Z=y.slice(j,K);if(b[Z]===void 0){let ce=h(y,A+1,U),re=f(y,U,ce);const te=N(y.slice(ce,re));b[Z]=te}v=U+1}while(v<X);return b}function h(y,w,b){do{const X=y.charCodeAt(w);if(X!==32&&X!==9)return w}while(++w<b);return b}function f(y,w,b){for(;w>b;){const X=y.charCodeAt(--w);if(X!==32&&X!==9)return w+1}return b}function g(y,w,b){const X=(b==null?void 0:b.encode)||encodeURIComponent;if(!i.test(y))throw new TypeError(`argument name is invalid: ${y}`);const N=X(w);if(!t.test(N))throw new TypeError(`argument val is invalid: ${w}`);let v=y+"="+N;if(!b)return v;if(b.maxAge!==void 0){if(!Number.isInteger(b.maxAge))throw new TypeError(`option maxAge is invalid: ${b.maxAge}`);v+="; Max-Age="+b.maxAge}if(b.domain){if(!r.test(b.domain))throw new TypeError(`option domain is invalid: ${b.domain}`);v+="; Domain="+b.domain}if(b.path){if(!s.test(b.path))throw new TypeError(`option path is invalid: ${b.path}`);v+="; Path="+b.path}if(b.expires){if(!R(b.expires)||!Number.isFinite(b.expires.valueOf()))throw new TypeError(`option expires is invalid: ${b.expires}`);v+="; Expires="+b.expires.toUTCString()}if(b.httpOnly&&(v+="; HttpOnly"),b.secure&&(v+="; Secure"),b.partitioned&&(v+="; Partitioned"),b.priority)switch(typeof b.priority=="string"?b.priority.toLowerCase():void 0){case"low":v+="; Priority=Low";break;case"medium":v+="; Priority=Medium";break;case"high":v+="; Priority=High";break;default:throw new TypeError(`option priority is invalid: ${b.priority}`)}if(b.sameSite)switch(typeof b.sameSite=="string"?b.sameSite.toLowerCase():b.sameSite){case!0:case"strict":v+="; SameSite=Strict";break;case"lax":v+="; SameSite=Lax";break;case"none":v+="; SameSite=None";break;default:throw new TypeError(`option sameSite is invalid: ${b.sameSite}`)}return v}function E(y){if(y.indexOf("%")===-1)return y;try{return decodeURIComponent(y)}catch{return y}}function R(y){return l.call(y)==="[object Date]"}return us}FI();var $g="popstate";function DI(i={}){function t(s,l){let{pathname:u,search:p,hash:h}=s.location;return dd("",{pathname:u,search:p,hash:h},l.state&&l.state.usr||null,l.state&&l.state.key||"default")}function r(s,l){return typeof l=="string"?l:ms(l)}return AI(t,r,null,i)}function Ae(i,t){if(i===!1||i===null||typeof i>"u")throw new Error(t)}function En(i,t){if(!i){typeof console<"u"&&console.warn(t);try{throw new Error(t)}catch{}}}function HI(){return Math.random().toString(36).substring(2,10)}function em(i,t){return{usr:i.state,key:i.key,idx:t}}function dd(i,t,r=null,s){return{pathname:typeof i=="string"?i:i.pathname,search:"",hash:"",...typeof t=="string"?cr(t):t,state:r,key:t&&t.key||s||HI()}}function ms({pathname:i="/",search:t="",hash:r=""}){return t&&t!=="?"&&(i+=t.charAt(0)==="?"?t:"?"+t),r&&r!=="#"&&(i+=r.charAt(0)==="#"?r:"#"+r),i}function cr(i){let t={};if(i){let r=i.indexOf("#");r>=0&&(t.hash=i.substring(r),i=i.substring(0,r));let s=i.indexOf("?");s>=0&&(t.search=i.substring(s),i=i.substring(0,s)),i&&(t.pathname=i)}return t}function AI(i,t,r,s={}){let{window:l=document.defaultView,v5Compat:u=!1}=s,p=l.history,h="POP",f=null,g=E();g==null&&(g=0,p.replaceState({...p.state,idx:g},""));function E(){return(p.state||{idx:null}).idx}function R(){h="POP";let N=E(),v=N==null?null:N-g;g=N,f&&f({action:h,location:X.location,delta:v})}function y(N,v){h="PUSH";let A=dd(X.location,N,v);g=E()+1;let k=em(A,g),U=X.createHref(A);try{p.pushState(k,"",U)}catch(j){if(j instanceof DOMException&&j.name==="DataCloneError")throw j;l.location.assign(U)}u&&f&&f({action:h,location:X.location,delta:1})}function w(N,v){h="REPLACE";let A=dd(X.location,N,v);g=E();let k=em(A,g),U=X.createHref(A);p.replaceState(k,"",U),u&&f&&f({action:h,location:X.location,delta:0})}function b(N){return LI(N)}let X={get action(){return h},get location(){return i(l,p)},listen(N){if(f)throw new Error("A history only accepts one active listener");return l.addEventListener($g,R),f=N,()=>{l.removeEventListener($g,R),f=null}},createHref(N){return t(l,N)},createURL:b,encodeLocation(N){let v=b(N);return{pathname:v.pathname,search:v.search,hash:v.hash}},push:y,replace:w,go(N){return p.go(N)}};return X}function LI(i,t=!1){let r="http://localhost";typeof window<"u"&&(r=window.location.origin!=="null"?window.location.origin:window.location.href),Ae(r,"No window.location.(origin|href) available to create URL");let s=typeof i=="string"?i:ms(i);return s=s.replace(/ $/,"%20"),!t&&s.startsWith("//")&&(s=r+s),new URL(s,r)}function mE(i,t,r="/"){return BI(i,t,r,!1)}function BI(i,t,r,s){let l=typeof t=="string"?cr(t):t,u=jn(l.pathname||"/",r);if(u==null)return null;let p=EE(i);MI(p);let h=null;for(let f=0;h==null&&f<p.length;++f){let g=KI(u);h=VI(p[f],g,s)}return h}function EE(i,t=[],r=[],s=""){let l=(u,p,h)=>{let f={relativePath:h===void 0?u.path||"":h,caseSensitive:u.caseSensitive===!0,childrenIndex:p,route:u};f.relativePath.startsWith("/")&&(Ae(f.relativePath.startsWith(s),`Absolute route path "${f.relativePath}" nested under path "${s}" is not valid. An absolute child route path must start with the combined path of all its parent routes.`),f.relativePath=f.relativePath.slice(s.length));let g=Un([s,f.relativePath]),E=r.concat(f);u.children&&u.children.length>0&&(Ae(u.index!==!0,`Index routes must not have child routes. Please remove all child routes from route path "${g}".`),EE(u.children,t,E,g)),!(u.path==null&&!u.index)&&t.push({path:g,score:GI(g,u.index),routesMeta:E})};return i.forEach((u,p)=>{var h;if(u.path===""||!((h=u.path)!=null&&h.includes("?")))l(u,p);else for(let f of SE(u.path))l(u,p,f)}),t}function SE(i){let t=i.split("/");if(t.length===0)return[];let[r,...s]=t,l=r.endsWith("?"),u=r.replace(/\?$/,"");if(s.length===0)return l?[u,""]:[u];let p=SE(s.join("/")),h=[];return h.push(...p.map(f=>f===""?u:[u,f].join("/"))),l&&h.push(...p),h.map(f=>i.startsWith("/")&&f===""?"/":f)}function MI(i){i.sort((t,r)=>t.score!==r.score?r.score-t.score:QI(t.routesMeta.map(s=>s.childrenIndex),r.routesMeta.map(s=>s.childrenIndex)))}var qI=/^:[\w-]+$/,UI=3,WI=2,jI=1,kI=10,zI=-2,tm=i=>i==="*";function GI(i,t){let r=i.split("/"),s=r.length;return r.some(tm)&&(s+=zI),t&&(s+=WI),r.filter(l=>!tm(l)).reduce((l,u)=>l+(qI.test(u)?UI:u===""?jI:kI),s)}function QI(i,t){return i.length===t.length&&i.slice(0,-1).every((s,l)=>s===t[l])?i[i.length-1]-t[t.length-1]:0}function VI(i,t,r=!1){let{routesMeta:s}=i,l={},u="/",p=[];for(let h=0;h<s.length;++h){let f=s[h],g=h===s.length-1,E=u==="/"?t:t.slice(u.length)||"/",R=ll({path:f.relativePath,caseSensitive:f.caseSensitive,end:g},E),y=f.route;if(!R&&g&&r&&!s[s.length-1].route.index&&(R=ll({path:f.relativePath,caseSensitive:f.caseSensitive,end:!1},E)),!R)return null;Object.assign(l,R.params),p.push({params:l,pathname:Un([u,R.pathname]),pathnameBase:ey(Un([u,R.pathnameBase])),route:y}),R.pathnameBase!=="/"&&(u=Un([u,R.pathnameBase]))}return p}function ll(i,t){typeof i=="string"&&(i={path:i,caseSensitive:!1,end:!0});let[r,s]=YI(i.path,i.caseSensitive,i.end),l=t.match(r);if(!l)return null;let u=l[0],p=u.replace(/(.)\/+$/,"$1"),h=l.slice(1);return{params:s.reduce((g,{paramName:E,isOptional:R},y)=>{if(E==="*"){let b=h[y]||"";p=u.slice(0,u.length-b.length).replace(/(.)\/+$/,"$1")}const w=h[y];return R&&!w?g[E]=void 0:g[E]=(w||"").replace(/%2F/g,"/"),g},{}),pathname:u,pathnameBase:p,pattern:i}}function YI(i,t=!1,r=!0){En(i==="*"||!i.endsWith("*")||i.endsWith("/*"),`Route path "${i}" will be treated as if it were "${i.replace(/\*$/,"/*")}" because the \`*\` character must always follow a \`/\` in the pattern. To get rid of this warning, please change the route path to "${i.replace(/\*$/,"/*")}".`);let s=[],l="^"+i.replace(/\/*\*?$/,"").replace(/^\/*/,"/").replace(/[\\.*+^${}|()[\]]/g,"\\$&").replace(/\/:([\w-]+)(\?)?/g,(p,h,f)=>(s.push({paramName:h,isOptional:f!=null}),f?"/?([^\\/]+)?":"/([^\\/]+)"));return i.endsWith("*")?(s.push({paramName:"*"}),l+=i==="*"||i==="/*"?"(.*)$":"(?:\\/(.+)|\\/*)$"):r?l+="\\/*$":i!==""&&i!=="/"&&(l+="(?:(?=\\/|$))"),[new RegExp(l,t?void 0:"i"),s]}function KI(i){try{return i.split("/").map(t=>decodeURIComponent(t).replace(/\//g,"%2F")).join("/")}catch(t){return En(!1,`The URL path "${i}" could not be decoded because it is a malformed URL segment. This is probably due to a bad percent encoding (${t}).`),i}}function jn(i,t){if(t==="/")return i;if(!i.toLowerCase().startsWith(t.toLowerCase()))return null;let r=t.endsWith("/")?t.length-1:t.length,s=i.charAt(r);return s&&s!=="/"?null:i.slice(r)||"/"}function ZI(i,t="/"){let{pathname:r,search:s="",hash:l=""}=typeof i=="string"?cr(i):i;return{pathname:r?r.startsWith("/")?r:JI(r,t):t,search:ty(s),hash:ny(l)}}function JI(i,t){let r=t.replace(/\/+$/,"").split("/");return i.split("/").forEach(l=>{l===".."?r.length>1&&r.pop():l!=="."&&r.push(l)}),r.length>1?r.join("/"):"/"}function Wu(i,t,r,s){return`Cannot include a '${i}' character in a manually specified \`to.${t}\` field [${JSON.stringify(s)}].  Please separate it out to the \`to.${r}\` field. Alternatively you may provide the full path as a string in <Link to="..."> and the router will parse it for you.`}function $I(i){return i.filter((t,r)=>r===0||t.route.path&&t.route.path.length>0)}function RE(i){let t=$I(i);return t.map((r,s)=>s===t.length-1?r.pathname:r.pathnameBase)}function IE(i,t,r,s=!1){let l;typeof i=="string"?l=cr(i):(l={...i},Ae(!l.pathname||!l.pathname.includes("?"),Wu("?","pathname","search",l)),Ae(!l.pathname||!l.pathname.includes("#"),Wu("#","pathname","hash",l)),Ae(!l.search||!l.search.includes("#"),Wu("#","search","hash",l)));let u=i===""||l.pathname==="",p=u?"/":l.pathname,h;if(p==null)h=r;else{let R=t.length-1;if(!s&&p.startsWith("..")){let y=p.split("/");for(;y[0]==="..";)y.shift(),R-=1;l.pathname=y.join("/")}h=R>=0?t[R]:"/"}let f=ZI(l,h),g=p&&p!=="/"&&p.endsWith("/"),E=(u||p===".")&&r.endsWith("/");return!f.pathname.endsWith("/")&&(g||E)&&(f.pathname+="/"),f}var Un=i=>i.join("/").replace(/\/\/+/g,"/"),ey=i=>i.replace(/\/+$/,"").replace(/^\/*/,"/"),ty=i=>!i||i==="?"?"":i.startsWith("?")?i:"?"+i,ny=i=>!i||i==="#"?"":i.startsWith("#")?i:"#"+i;function iy(i){return i!=null&&typeof i.status=="number"&&typeof i.statusText=="string"&&typeof i.internal=="boolean"&&"data"in i}var yE=["POST","PUT","PATCH","DELETE"];new Set(yE);var oy=["GET",...yE];new Set(oy);var ur=P.createContext(null);ur.displayName="DataRouter";var gl=P.createContext(null);gl.displayName="DataRouterState";var bE=P.createContext({isTransitioning:!1});bE.displayName="ViewTransition";var ry=P.createContext(new Map);ry.displayName="Fetchers";var sy=P.createContext(null);sy.displayName="Await";var Sn=P.createContext(null);Sn.displayName="Navigation";var bs=P.createContext(null);bs.displayName="Location";var Rn=P.createContext({outlet:null,matches:[],isDataRoute:!1});Rn.displayName="Route";var xd=P.createContext(null);xd.displayName="RouteError";function ay(i,{relative:t}={}){Ae(Cs(),"useHref() may be used only in the context of a <Router> component.");let{basename:r,navigator:s}=P.useContext(Sn),{hash:l,pathname:u,search:p}=ws(i,{relative:t}),h=u;return r!=="/"&&(h=u==="/"?r:Un([r,u])),s.createHref({pathname:h,search:p,hash:l})}function Cs(){return P.useContext(bs)!=null}function Ci(){return Ae(Cs(),"useLocation() may be used only in the context of a <Router> component."),P.useContext(bs).location}var CE="You should call navigate() in a React.useEffect(), not when your component is first rendered.";function wE(i){P.useContext(Sn).static||P.useLayoutEffect(i)}function ly(){let{isDataRoute:i}=P.useContext(Rn);return i?yy():cy()}function cy(){Ae(Cs(),"useNavigate() may be used only in the context of a <Router> component.");let i=P.useContext(ur),{basename:t,navigator:r}=P.useContext(Sn),{matches:s}=P.useContext(Rn),{pathname:l}=Ci(),u=JSON.stringify(RE(s)),p=P.useRef(!1);return wE(()=>{p.current=!0}),P.useCallback((f,g={})=>{if(En(p.current,CE),!p.current)return;if(typeof f=="number"){r.go(f);return}let E=IE(f,JSON.parse(u),l,g.relative==="path");i==null&&t!=="/"&&(E.pathname=E.pathname==="/"?t:Un([t,E.pathname])),(g.replace?r.replace:r.push)(E,g.state,g)},[t,r,u,l,i])}P.createContext(null);function Pd(){let{matches:i}=P.useContext(Rn),t=i[i.length-1];return t?t.params:{}}function ws(i,{relative:t}={}){let{matches:r}=P.useContext(Rn),{pathname:s}=Ci(),l=JSON.stringify(RE(r));return P.useMemo(()=>IE(i,JSON.parse(l),s,t==="path"),[i,l,s,t])}function uy(i,t){return _E(i,t)}function _E(i,t,r,s){var v;Ae(Cs(),"useRoutes() may be used only in the context of a <Router> component.");let{navigator:l}=P.useContext(Sn),{matches:u}=P.useContext(Rn),p=u[u.length-1],h=p?p.params:{},f=p?p.pathname:"/",g=p?p.pathnameBase:"/",E=p&&p.route;{let A=E&&E.path||"";TE(f,!E||A.endsWith("*")||A.endsWith("*?"),`You rendered descendant <Routes> (or called \`useRoutes()\`) at "${f}" (under <Route path="${A}">) but the parent route path has no trailing "*". This means if you navigate deeper, the parent won't match anymore and therefore the child routes will never render.

Please change the parent <Route path="${A}"> to <Route path="${A==="/"?"*":`${A}/*`}">.`)}let R=Ci(),y;if(t){let A=typeof t=="string"?cr(t):t;Ae(g==="/"||((v=A.pathname)==null?void 0:v.startsWith(g)),`When overriding the location using \`<Routes location>\` or \`useRoutes(routes, location)\`, the location pathname must begin with the portion of the URL pathname that was matched by all parent routes. The current pathname base is "${g}" but pathname "${A.pathname}" was given in the \`location\` prop.`),y=A}else y=R;let w=y.pathname||"/",b=w;if(g!=="/"){let A=g.replace(/^\//,"").split("/");b="/"+w.replace(/^\//,"").split("/").slice(A.length).join("/")}let X=mE(i,{pathname:b});En(E||X!=null,`No routes matched location "${y.pathname}${y.search}${y.hash}" `),En(X==null||X[X.length-1].route.element!==void 0||X[X.length-1].route.Component!==void 0||X[X.length-1].route.lazy!==void 0,`Matched leaf route at location "${y.pathname}${y.search}${y.hash}" does not have an element or Component. This means it will render an <Outlet /> with a null value by default resulting in an "empty" page.`);let N=gy(X&&X.map(A=>Object.assign({},A,{params:Object.assign({},h,A.params),pathname:Un([g,l.encodeLocation?l.encodeLocation(A.pathname).pathname:A.pathname]),pathnameBase:A.pathnameBase==="/"?g:Un([g,l.encodeLocation?l.encodeLocation(A.pathnameBase).pathname:A.pathnameBase])})),u,r,s);return t&&N?P.createElement(bs.Provider,{value:{location:{pathname:"/",search:"",hash:"",state:null,key:"default",...y},navigationType:"POP"}},N):N}function dy(){let i=Iy(),t=iy(i)?`${i.status} ${i.statusText}`:i instanceof Error?i.message:JSON.stringify(i),r=i instanceof Error?i.stack:null,s="rgba(200,200,200, 0.5)",l={padding:"0.5rem",backgroundColor:s},u={padding:"2px 4px",backgroundColor:s},p=null;return console.error("Error handled by React Router default ErrorBoundary:",i),p=P.createElement(P.Fragment,null,P.createElement("p",null,"💿 Hey developer 👋"),P.createElement("p",null,"You can provide a way better UX than this when your app throws errors by providing your own ",P.createElement("code",{style:u},"ErrorBoundary")," or"," ",P.createElement("code",{style:u},"errorElement")," prop on your route.")),P.createElement(P.Fragment,null,P.createElement("h2",null,"Unexpected Application Error!"),P.createElement("h3",{style:{fontStyle:"italic"}},t),r?P.createElement("pre",{style:l},r):null,p)}var py=P.createElement(dy,null),hy=class extends P.Component{constructor(i){super(i),this.state={location:i.location,revalidation:i.revalidation,error:i.error}}static getDerivedStateFromError(i){return{error:i}}static getDerivedStateFromProps(i,t){return t.location!==i.location||t.revalidation!=="idle"&&i.revalidation==="idle"?{error:i.error,location:i.location,revalidation:i.revalidation}:{error:i.error!==void 0?i.error:t.error,location:t.location,revalidation:i.revalidation||t.revalidation}}componentDidCatch(i,t){console.error("React Router caught the following error during render",i,t)}render(){return this.state.error!==void 0?P.createElement(Rn.Provider,{value:this.props.routeContext},P.createElement(xd.Provider,{value:this.state.error,children:this.props.component})):this.props.children}};function fy({routeContext:i,match:t,children:r}){let s=P.useContext(ur);return s&&s.static&&s.staticContext&&(t.route.errorElement||t.route.ErrorBoundary)&&(s.staticContext._deepestRenderedBoundaryId=t.route.id),P.createElement(Rn.Provider,{value:i},r)}function gy(i,t=[],r=null,s=null){if(i==null){if(!r)return null;if(r.errors)i=r.matches;else if(t.length===0&&!r.initialized&&r.matches.length>0)i=r.matches;else return null}let l=i,u=r==null?void 0:r.errors;if(u!=null){let f=l.findIndex(g=>g.route.id&&(u==null?void 0:u[g.route.id])!==void 0);Ae(f>=0,`Could not find a matching route for errors on route IDs: ${Object.keys(u).join(",")}`),l=l.slice(0,Math.min(l.length,f+1))}let p=!1,h=-1;if(r)for(let f=0;f<l.length;f++){let g=l[f];if((g.route.HydrateFallback||g.route.hydrateFallbackElement)&&(h=f),g.route.id){let{loaderData:E,errors:R}=r,y=g.route.loader&&!E.hasOwnProperty(g.route.id)&&(!R||R[g.route.id]===void 0);if(g.route.lazy||y){p=!0,h>=0?l=l.slice(0,h+1):l=[l[0]];break}}}return l.reduceRight((f,g,E)=>{let R,y=!1,w=null,b=null;r&&(R=u&&g.route.id?u[g.route.id]:void 0,w=g.route.errorElement||py,p&&(h<0&&E===0?(TE("route-fallback",!1,"No `HydrateFallback` element provided to render during initial hydration"),y=!0,b=null):h===E&&(y=!0,b=g.route.hydrateFallbackElement||null)));let X=t.concat(l.slice(0,E+1)),N=()=>{let v;return R?v=w:y?v=b:g.route.Component?v=P.createElement(g.route.Component,null):g.route.element?v=g.route.element:v=f,P.createElement(fy,{match:g,routeContext:{outlet:f,matches:X,isDataRoute:r!=null},children:v})};return r&&(g.route.ErrorBoundary||g.route.errorElement||E===0)?P.createElement(hy,{location:r.location,revalidation:r.revalidation,component:w,error:R,children:N(),routeContext:{outlet:null,matches:X,isDataRoute:!0}}):N()},null)}function Xd(i){return`${i} must be used within a data router.  See https://reactrouter.com/en/main/routers/picking-a-router.`}function my(i){let t=P.useContext(ur);return Ae(t,Xd(i)),t}function Ey(i){let t=P.useContext(gl);return Ae(t,Xd(i)),t}function Sy(i){let t=P.useContext(Rn);return Ae(t,Xd(i)),t}function Nd(i){let t=Sy(i),r=t.matches[t.matches.length-1];return Ae(r.route.id,`${i} can only be used on routes that contain a unique "id"`),r.route.id}function Ry(){return Nd("useRouteId")}function Iy(){var s;let i=P.useContext(xd),t=Ey("useRouteError"),r=Nd("useRouteError");return i!==void 0?i:(s=t.errors)==null?void 0:s[r]}function yy(){let{router:i}=my("useNavigate"),t=Nd("useNavigate"),r=P.useRef(!1);return wE(()=>{r.current=!0}),P.useCallback(async(l,u={})=>{En(r.current,CE),r.current&&(typeof l=="number"?i.navigate(l):await i.navigate(l,{fromRouteId:t,...u}))},[i,t])}var nm={};function TE(i,t,r){!t&&!nm[i]&&(nm[i]=!0,En(!1,r))}P.memo(by);function by({routes:i,future:t,state:r}){return _E(i,void 0,r,t)}function Bn(i){Ae(!1,"A <Route> is only ever to be used as the child of <Routes> element, never rendered directly. Please wrap your <Route> in a <Routes>.")}function Cy({basename:i="/",children:t=null,location:r,navigationType:s="POP",navigator:l,static:u=!1}){Ae(!Cs(),"You cannot render a <Router> inside another <Router>. You should never have more than one in your app.");let p=i.replace(/^\/*/,"/"),h=P.useMemo(()=>({basename:p,navigator:l,static:u,future:{}}),[p,l,u]);typeof r=="string"&&(r=cr(r));let{pathname:f="/",search:g="",hash:E="",state:R=null,key:y="default"}=r,w=P.useMemo(()=>{let b=jn(f,p);return b==null?null:{location:{pathname:b,search:g,hash:E,state:R,key:y},navigationType:s}},[p,f,g,E,R,y,s]);return En(w!=null,`<Router basename="${p}"> is not able to match the URL "${f}${g}${E}" because it does not start with the basename, so the <Router> won't render anything.`),w==null?null:P.createElement(Sn.Provider,{value:h},P.createElement(bs.Provider,{children:t,value:w}))}function wy({children:i,location:t}){return uy(pd(i),t)}function pd(i,t=[]){let r=[];return P.Children.forEach(i,(s,l)=>{if(!P.isValidElement(s))return;let u=[...t,l];if(s.type===P.Fragment){r.push.apply(r,pd(s.props.children,u));return}Ae(s.type===Bn,`[${typeof s.type=="string"?s.type:s.type.name}] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>`),Ae(!s.props.index||!s.props.children,"An index route cannot have child routes.");let p={id:s.props.id||u.join("-"),caseSensitive:s.props.caseSensitive,element:s.props.element,Component:s.props.Component,index:s.props.index,path:s.props.path,loader:s.props.loader,action:s.props.action,hydrateFallbackElement:s.props.hydrateFallbackElement,HydrateFallback:s.props.HydrateFallback,errorElement:s.props.errorElement,ErrorBoundary:s.props.ErrorBoundary,hasErrorBoundary:s.props.hasErrorBoundary===!0||s.props.ErrorBoundary!=null||s.props.errorElement!=null,shouldRevalidate:s.props.shouldRevalidate,handle:s.props.handle,lazy:s.props.lazy};s.props.children&&(p.children=pd(s.props.children,u)),r.push(p)}),r}var tl="get",nl="application/x-www-form-urlencoded";function ml(i){return i!=null&&typeof i.tagName=="string"}function _y(i){return ml(i)&&i.tagName.toLowerCase()==="button"}function Ty(i){return ml(i)&&i.tagName.toLowerCase()==="form"}function vy(i){return ml(i)&&i.tagName.toLowerCase()==="input"}function xy(i){return!!(i.metaKey||i.altKey||i.ctrlKey||i.shiftKey)}function Py(i,t){return i.button===0&&(!t||t==="_self")&&!xy(i)}var Wa=null;function Xy(){if(Wa===null)try{new FormData(document.createElement("form"),0),Wa=!1}catch{Wa=!0}return Wa}var Ny=new Set(["application/x-www-form-urlencoded","multipart/form-data","text/plain"]);function ju(i){return i!=null&&!Ny.has(i)?(En(!1,`"${i}" is not a valid \`encType\` for \`<Form>\`/\`<fetcher.Form>\` and will default to "${nl}"`),null):i}function Oy(i,t){let r,s,l,u,p;if(Ty(i)){let h=i.getAttribute("action");s=h?jn(h,t):null,r=i.getAttribute("method")||tl,l=ju(i.getAttribute("enctype"))||nl,u=new FormData(i)}else if(_y(i)||vy(i)&&(i.type==="submit"||i.type==="image")){let h=i.form;if(h==null)throw new Error('Cannot submit a <button> or <input type="submit"> without a <form>');let f=i.getAttribute("formaction")||h.getAttribute("action");if(s=f?jn(f,t):null,r=i.getAttribute("formmethod")||h.getAttribute("method")||tl,l=ju(i.getAttribute("formenctype"))||ju(h.getAttribute("enctype"))||nl,u=new FormData(h,i),!Xy()){let{name:g,type:E,value:R}=i;if(E==="image"){let y=g?`${g}.`:"";u.append(`${y}x`,"0"),u.append(`${y}y`,"0")}else g&&u.append(g,R)}}else{if(ml(i))throw new Error('Cannot submit element that is not <form>, <button>, or <input type="submit|image">');r=tl,s=null,l=nl,p=i}return u&&l==="text/plain"&&(p=u,u=void 0),{action:s,method:r.toLowerCase(),encType:l,formData:u,body:p}}function Od(i,t){if(i===!1||i===null||typeof i>"u")throw new Error(t)}async function Fy(i,t){if(i.id in t)return t[i.id];try{let r=await import(i.module);return t[i.id]=r,r}catch(r){return console.error(`Error loading route module \`${i.module}\`, reloading page...`),console.error(r),window.__reactRouterContext&&window.__reactRouterContext.isSpaMode,window.location.reload(),new Promise(()=>{})}}function Dy(i){return i==null?!1:i.href==null?i.rel==="preload"&&typeof i.imageSrcSet=="string"&&typeof i.imageSizes=="string":typeof i.rel=="string"&&typeof i.href=="string"}async function Hy(i,t,r){let s=await Promise.all(i.map(async l=>{let u=t.routes[l.route.id];if(u){let p=await Fy(u,r);return p.links?p.links():[]}return[]}));return My(s.flat(1).filter(Dy).filter(l=>l.rel==="stylesheet"||l.rel==="preload").map(l=>l.rel==="stylesheet"?{...l,rel:"prefetch",as:"style"}:{...l,rel:"prefetch"}))}function im(i,t,r,s,l,u){let p=(f,g)=>r[g]?f.route.id!==r[g].route.id:!0,h=(f,g)=>{var E;return r[g].pathname!==f.pathname||((E=r[g].route.path)==null?void 0:E.endsWith("*"))&&r[g].params["*"]!==f.params["*"]};return u==="assets"?t.filter((f,g)=>p(f,g)||h(f,g)):u==="data"?t.filter((f,g)=>{var R;let E=s.routes[f.route.id];if(!E||!E.hasLoader)return!1;if(p(f,g)||h(f,g))return!0;if(f.route.shouldRevalidate){let y=f.route.shouldRevalidate({currentUrl:new URL(l.pathname+l.search+l.hash,window.origin),currentParams:((R=r[0])==null?void 0:R.params)||{},nextUrl:new URL(i,window.origin),nextParams:f.params,defaultShouldRevalidate:!0});if(typeof y=="boolean")return y}return!0}):[]}function Ay(i,t,{includeHydrateFallback:r}={}){return Ly(i.map(s=>{let l=t.routes[s.route.id];if(!l)return[];let u=[l.module];return l.clientActionModule&&(u=u.concat(l.clientActionModule)),l.clientLoaderModule&&(u=u.concat(l.clientLoaderModule)),r&&l.hydrateFallbackModule&&(u=u.concat(l.hydrateFallbackModule)),l.imports&&(u=u.concat(l.imports)),u}).flat(1))}function Ly(i){return[...new Set(i)]}function By(i){let t={},r=Object.keys(i).sort();for(let s of r)t[s]=i[s];return t}function My(i,t){let r=new Set;return new Set(t),i.reduce((s,l)=>{let u=JSON.stringify(By(l));return r.has(u)||(r.add(u),s.push({key:u,link:l})),s},[])}Object.getOwnPropertyNames(Object.prototype).sort().join("\0");var qy=new Set([100,101,204,205]);function Uy(i,t){let r=typeof i=="string"?new URL(i,typeof window>"u"?"server://singlefetch/":window.location.origin):i;return r.pathname==="/"?r.pathname="_root.data":t&&jn(r.pathname,t)==="/"?r.pathname=`${t.replace(/\/$/,"")}/_root.data`:r.pathname=`${r.pathname.replace(/\/$/,"")}.data`,r}function vE(){let i=P.useContext(ur);return Od(i,"You must render this element inside a <DataRouterContext.Provider> element"),i}function Wy(){let i=P.useContext(gl);return Od(i,"You must render this element inside a <DataRouterStateContext.Provider> element"),i}var Fd=P.createContext(void 0);Fd.displayName="FrameworkContext";function xE(){let i=P.useContext(Fd);return Od(i,"You must render this element inside a <HydratedRouter> element"),i}function jy(i,t){let r=P.useContext(Fd),[s,l]=P.useState(!1),[u,p]=P.useState(!1),{onFocus:h,onBlur:f,onMouseEnter:g,onMouseLeave:E,onTouchStart:R}=t,y=P.useRef(null);P.useEffect(()=>{if(i==="render"&&p(!0),i==="viewport"){let X=v=>{v.forEach(A=>{p(A.isIntersecting)})},N=new IntersectionObserver(X,{threshold:.5});return y.current&&N.observe(y.current),()=>{N.disconnect()}}},[i]),P.useEffect(()=>{if(s){let X=setTimeout(()=>{p(!0)},100);return()=>{clearTimeout(X)}}},[s]);let w=()=>{l(!0)},b=()=>{l(!1),p(!1)};return r?i!=="intent"?[u,y,{}]:[u,y,{onFocus:ds(h,w),onBlur:ds(f,b),onMouseEnter:ds(g,w),onMouseLeave:ds(E,b),onTouchStart:ds(R,w)}]:[!1,y,{}]}function ds(i,t){return r=>{i&&i(r),r.defaultPrevented||t(r)}}function ky({page:i,...t}){let{router:r}=vE(),s=P.useMemo(()=>mE(r.routes,i,r.basename),[r.routes,i,r.basename]);return s?P.createElement(Gy,{page:i,matches:s,...t}):null}function zy(i){let{manifest:t,routeModules:r}=xE(),[s,l]=P.useState([]);return P.useEffect(()=>{let u=!1;return Hy(i,t,r).then(p=>{u||l(p)}),()=>{u=!0}},[i,t,r]),s}function Gy({page:i,matches:t,...r}){let s=Ci(),{manifest:l,routeModules:u}=xE(),{basename:p}=vE(),{loaderData:h,matches:f}=Wy(),g=P.useMemo(()=>im(i,t,f,l,s,"data"),[i,t,f,l,s]),E=P.useMemo(()=>im(i,t,f,l,s,"assets"),[i,t,f,l,s]),R=P.useMemo(()=>{if(i===s.pathname+s.search+s.hash)return[];let b=new Set,X=!1;if(t.forEach(v=>{var k;let A=l.routes[v.route.id];!A||!A.hasLoader||(!g.some(U=>U.route.id===v.route.id)&&v.route.id in h&&((k=u[v.route.id])!=null&&k.shouldRevalidate)||A.hasClientLoader?X=!0:b.add(v.route.id))}),b.size===0)return[];let N=Uy(i,p);return X&&b.size>0&&N.searchParams.set("_routes",t.filter(v=>b.has(v.route.id)).map(v=>v.route.id).join(",")),[N.pathname+N.search]},[p,h,s,l,g,t,i,u]),y=P.useMemo(()=>Ay(E,l),[E,l]),w=zy(E);return P.createElement(P.Fragment,null,R.map(b=>P.createElement("link",{key:b,rel:"prefetch",as:"fetch",href:b,...r})),y.map(b=>P.createElement("link",{key:b,rel:"modulepreload",href:b,...r})),w.map(({key:b,link:X})=>P.createElement("link",{key:b,...X})))}function Qy(...i){return t=>{i.forEach(r=>{typeof r=="function"?r(t):r!=null&&(r.current=t)})}}var PE=typeof window<"u"&&typeof window.document<"u"&&typeof window.document.createElement<"u";try{PE&&(window.__reactRouterVersion="7.6.2")}catch{}function Vy({basename:i,children:t,window:r}){let s=P.useRef();s.current==null&&(s.current=DI({window:r,v5Compat:!0}));let l=s.current,[u,p]=P.useState({action:l.action,location:l.location}),h=P.useCallback(f=>{P.startTransition(()=>p(f))},[p]);return P.useLayoutEffect(()=>l.listen(h),[l,h]),P.createElement(Cy,{basename:i,children:t,location:u.location,navigationType:u.action,navigator:l})}var XE=/^(?:[a-z][a-z0-9+.-]*:|\/\/)/i,ut=P.forwardRef(function({onClick:t,discover:r="render",prefetch:s="none",relative:l,reloadDocument:u,replace:p,state:h,target:f,to:g,preventScrollReset:E,viewTransition:R,...y},w){let{basename:b}=P.useContext(Sn),X=typeof g=="string"&&XE.test(g),N,v=!1;if(typeof g=="string"&&X&&(N=g,PE))try{let re=new URL(window.location.href),te=g.startsWith("//")?new URL(re.protocol+g):new URL(g),Xe=jn(te.pathname,b);te.origin===re.origin&&Xe!=null?g=Xe+te.search+te.hash:v=!0}catch{En(!1,`<Link to="${g}"> contains an invalid URL which will probably break when clicked - please update to a valid URL path.`)}let A=ay(g,{relative:l}),[k,U,j]=jy(s,y),K=Jy(g,{replace:p,state:h,target:f,preventScrollReset:E,relative:l,viewTransition:R});function Z(re){t&&t(re),re.defaultPrevented||K(re)}let ce=P.createElement("a",{...y,...j,href:N||A,onClick:v||u?t:Z,ref:Qy(w,U),target:f,"data-discover":!X&&r==="render"?"true":void 0});return k&&!X?P.createElement(P.Fragment,null,ce,P.createElement(ky,{page:A})):ce});ut.displayName="Link";var Yy=P.forwardRef(function({"aria-current":t="page",caseSensitive:r=!1,className:s="",end:l=!1,style:u,to:p,viewTransition:h,children:f,...g},E){let R=ws(p,{relative:g.relative}),y=Ci(),w=P.useContext(gl),{navigator:b,basename:X}=P.useContext(Sn),N=w!=null&&ib(R)&&h===!0,v=b.encodeLocation?b.encodeLocation(R).pathname:R.pathname,A=y.pathname,k=w&&w.navigation&&w.navigation.location?w.navigation.location.pathname:null;r||(A=A.toLowerCase(),k=k?k.toLowerCase():null,v=v.toLowerCase()),k&&X&&(k=jn(k,X)||k);const U=v!=="/"&&v.endsWith("/")?v.length-1:v.length;let j=A===v||!l&&A.startsWith(v)&&A.charAt(U)==="/",K=k!=null&&(k===v||!l&&k.startsWith(v)&&k.charAt(v.length)==="/"),Z={isActive:j,isPending:K,isTransitioning:N},ce=j?t:void 0,re;typeof s=="function"?re=s(Z):re=[s,j?"active":null,K?"pending":null,N?"transitioning":null].filter(Boolean).join(" ");let te=typeof u=="function"?u(Z):u;return P.createElement(ut,{...g,"aria-current":ce,className:re,ref:E,style:te,to:p,viewTransition:h},typeof f=="function"?f(Z):f)});Yy.displayName="NavLink";var Ky=P.forwardRef(({discover:i="render",fetcherKey:t,navigate:r,reloadDocument:s,replace:l,state:u,method:p=tl,action:h,onSubmit:f,relative:g,preventScrollReset:E,viewTransition:R,...y},w)=>{let b=tb(),X=nb(h,{relative:g}),N=p.toLowerCase()==="get"?"get":"post",v=typeof h=="string"&&XE.test(h),A=k=>{if(f&&f(k),k.defaultPrevented)return;k.preventDefault();let U=k.nativeEvent.submitter,j=(U==null?void 0:U.getAttribute("formmethod"))||p;b(U||k.currentTarget,{fetcherKey:t,method:j,navigate:r,replace:l,state:u,relative:g,preventScrollReset:E,viewTransition:R})};return P.createElement("form",{ref:w,method:N,action:X,onSubmit:s?f:A,...y,"data-discover":!v&&i==="render"?"true":void 0})});Ky.displayName="Form";function Zy(i){return`${i} must be used within a data router.  See https://reactrouter.com/en/main/routers/picking-a-router.`}function NE(i){let t=P.useContext(ur);return Ae(t,Zy(i)),t}function Jy(i,{target:t,replace:r,state:s,preventScrollReset:l,relative:u,viewTransition:p}={}){let h=ly(),f=Ci(),g=ws(i,{relative:u});return P.useCallback(E=>{if(Py(E,t)){E.preventDefault();let R=r!==void 0?r:ms(f)===ms(g);h(i,{replace:R,state:s,preventScrollReset:l,relative:u,viewTransition:p})}},[f,h,g,r,s,t,i,l,u,p])}var $y=0,eb=()=>`__${String(++$y)}__`;function tb(){let{router:i}=NE("useSubmit"),{basename:t}=P.useContext(Sn),r=Ry();return P.useCallback(async(s,l={})=>{let{action:u,method:p,encType:h,formData:f,body:g}=Oy(s,t);if(l.navigate===!1){let E=l.fetcherKey||eb();await i.fetch(E,r,l.action||u,{preventScrollReset:l.preventScrollReset,formData:f,body:g,formMethod:l.method||p,formEncType:l.encType||h,flushSync:l.flushSync})}else await i.navigate(l.action||u,{preventScrollReset:l.preventScrollReset,formData:f,body:g,formMethod:l.method||p,formEncType:l.encType||h,replace:l.replace,state:l.state,fromRouteId:r,flushSync:l.flushSync,viewTransition:l.viewTransition})},[i,t,r])}function nb(i,{relative:t}={}){let{basename:r}=P.useContext(Sn),s=P.useContext(Rn);Ae(s,"useFormAction must be used inside a RouteContext");let[l]=s.matches.slice(-1),u={...ws(i||".",{relative:t})},p=Ci();if(i==null){u.search=p.search;let h=new URLSearchParams(u.search),f=h.getAll("index");if(f.some(E=>E==="")){h.delete("index"),f.filter(R=>R).forEach(R=>h.append("index",R));let E=h.toString();u.search=E?`?${E}`:""}}return(!i||i===".")&&l.route.index&&(u.search=u.search?u.search.replace(/^\?/,"?index&"):"?index"),r!=="/"&&(u.pathname=u.pathname==="/"?r:Un([r,u.pathname])),ms(u)}function ib(i,t={}){let r=P.useContext(bE);Ae(r!=null,"`useViewTransitionState` must be used within `react-router-dom`'s `RouterProvider`.  Did you accidentally import `RouterProvider` from `react-router`?");let{basename:s}=NE("useViewTransitionState"),l=ws(i,{relative:t.relative});if(!r.isTransitioning)return!1;let u=jn(r.currentLocation.pathname,s)||r.currentLocation.pathname,p=jn(r.nextLocation.pathname,s)||r.nextLocation.pathname;return ll(l.pathname,p)!=null||ll(l.pathname,u)!=null}[...qy];const ob=()=>{const[i,t]=P.useState(!1),r=()=>{t(!i)};return I.jsx("nav",{className:`sticky top-0 z-50 py-3 backdrop-blur-lg 
    border-b border-neutral-700/80`,children:I.jsxs("div",{className:"container px-4 mx-auto relative text-sm",children:[I.jsxs("div",{className:"flex justify-between items-center",children:[I.jsxs(ut,{to:"/",className:"flex items-center flex-shrink-0",children:[I.jsx("img",{className:"h-10 w-10 mr-2",src:pE,alt:"logo"}),I.jsx("span",{className:"text-xl tracking-tight bg-gradient-to-r bg-clip-text text-transparent from-orange-400 to-orange-600",children:"SPARQL Fest"})]}),I.jsx("ul",{className:"hidden lg:flex ml-14 space-x-12",children:Kg.map((s,l)=>I.jsx("li",{children:I.jsx(ut,{to:s.href,children:s.label})},l))}),I.jsx("div",{className:"hidden lg:flex justify-center items-center",children:I.jsx(ut,{to:"/#contact",className:"bg-gradient-to-r from-orange-500 to-orange-800 py-2 px-3 rounded-md",children:"Contact us"})}),I.jsx("div",{className:"lg:hidden md:flex flex-col justify-end",children:I.jsx("button",{onClick:r,children:i?I.jsx(OI,{}):I.jsx(XI,{})})})]}),i&&I.jsxs("div",{className:`fixed right-0 z-20 bg-stone-900/95 w-full p-12
                flex flex-col justify-center items-center lg:hidden`,children:[I.jsx("ul",{children:Kg.map((s,l)=>I.jsx("li",{className:"py-4",children:I.jsx(ut,{to:s.href,children:s.label})},l))}),I.jsx("div",{className:"flex mt-4",children:I.jsx(ut,{to:"/#contact",className:`py-2 px-3 rounded-md bg-gradient-to-r
                        from-orange-500 to-orange-800 `,children:"Contact us"})})]})]})})},rb=()=>I.jsx("div",{className:"w-full py-16 bg-stone-900/50",children:I.jsx("footer",{className:"text-white py-8 px-[12vw] md:px-[7vw] lg:px-[20vw]",children:I.jsxs("div",{className:"container mx-auto text-center",children:[I.jsx("div",{className:"flex justify-center",children:I.jsxs(ut,{to:"/",className:"flex items-center flex-shrink-0",children:[I.jsx("img",{className:"h-10 w-10 mr-2",src:pE,alt:"logo"}),I.jsx("span",{className:"text-xl tracking-tight bg-gradient-to-r bg-clip-text text-transparent from-orange-400 to-orange-600",children:"SPARQL Fest"})]})}),I.jsx("nav",{className:"flex flex-wrap justify-center space-x-4 sm:space-x-6 mt-4",children:[{name:"About",id:"about"},{name:"SPARQL queries",id:"sparql-queries"},{name:"Contact",id:"contact"},{name:"Prefixes",id:"prefixes"}].map((i,t)=>I.jsx("button",{onClick:()=>handleScroll(i.id),className:"hover:text-orange-500 text-sm sm:text-base my-1",children:i.name},t))}),I.jsx("p",{className:"text-sm text-gray-400 mt-6",children:"© 2025 SPARQL Fest. All rights reserved."})]})})}),Yi="generated",sb="pointerdown",ab="pointerup",hd="pointerleave",lb="pointerout",eo="pointermove",cb="touchstart",om="touchend",ub="touchmove",db="touchcancel",pb="resize",hb="visibilitychange",Kt="tsParticles - Error",mi=100,Ke=.5,dt=1e3,ct={x:0,y:0,z:0},ja={a:1,b:0,c:0,d:1},ir="random",il="mid",pt=2,fb=Math.PI*pt,ku=60,OE="true",rm="false",zu="canvas",sm=0,Mn=2,am=4,gb=1,lm=1,cm=1,mb=4,FE=1,ol=255,Ki=360,fd=100,gd=100,um=0,md=0,Eb=60,Sb=0,Ed=.25,dm=Ke+Ed,pm=0,Rb=1,Ib=0,yb=0,bb=1,Dd=1,Cb=1,hm=1,cl=0,DE=1,wb=0,_b=120,Tb=0,vb=0,xb=1e4,Pb=0,Xb=1,ul=0,HE=1,Nb=1,Ob=0,fm=1,Fb=0,Db=0,gm=-.25,mm=1.5,Em=0,Hb=1,Ab=0,Sd=0,Lb=0,AE=1,Bb=1,dl=1,Mb=500,Sm=50,qb=0,Rd=1,LE=0,Ub=1,Wb=0,Wi=255,Gu=3,Qu=6,jb=1,kb=1,zb=0,Gb=0,Qb=0,Vb=0;var rt;(function(i){i.bottom="bottom",i.bottomLeft="bottom-left",i.bottomRight="bottom-right",i.left="left",i.none="none",i.right="right",i.top="top",i.topLeft="top-left",i.topRight="top-right",i.outside="outside",i.inside="inside"})(rt||(rt={}));function BE(i){return typeof i=="boolean"}function Ii(i){return typeof i=="string"}function yi(i){return typeof i=="number"}function Vo(i){return typeof i=="object"&&i!==null}function In(i){return Array.isArray(i)}function Q(i){return i==null}class At{constructor(t,r,s){if(this._updateFromAngle=(l,u)=>{this.x=Math.cos(l)*u,this.y=Math.sin(l)*u},!yi(t)&&t){this.x=t.x,this.y=t.y;const l=t;this.z=l.z?l.z:ct.z}else if(t!==void 0&&r!==void 0)this.x=t,this.y=r,this.z=s??ct.z;else throw new Error(`${Kt} Vector3d not initialized correctly`)}static get origin(){return At.create(ct.x,ct.y,ct.z)}get angle(){return Math.atan2(this.y,this.x)}set angle(t){this._updateFromAngle(t,this.length)}get length(){return Math.sqrt(this.getLengthSq())}set length(t){this._updateFromAngle(this.angle,t)}static clone(t){return At.create(t.x,t.y,t.z)}static create(t,r,s){return new At(t,r,s)}add(t){return At.create(this.x+t.x,this.y+t.y,this.z+t.z)}addTo(t){this.x+=t.x,this.y+=t.y,this.z+=t.z}copy(){return At.clone(this)}distanceTo(t){return this.sub(t).length}distanceToSq(t){return this.sub(t).getLengthSq()}div(t){return At.create(this.x/t,this.y/t,this.z/t)}divTo(t){this.x/=t,this.y/=t,this.z/=t}getLengthSq(){return this.x**Mn+this.y**Mn}mult(t){return At.create(this.x*t,this.y*t,this.z*t)}multTo(t){this.x*=t,this.y*=t,this.z*=t}normalize(){const t=this.length;t!=ul&&this.multTo(FE/t)}rotate(t){return At.create(this.x*Math.cos(t)-this.y*Math.sin(t),this.x*Math.sin(t)+this.y*Math.cos(t),ct.z)}setTo(t){this.x=t.x,this.y=t.y;const r=t;this.z=r.z?r.z:ct.z}sub(t){return At.create(this.x-t.x,this.y-t.y,this.z-t.z)}subFrom(t){this.x-=t.x,this.y-=t.y,this.z-=t.z}}class tt extends At{constructor(t,r){super(t,r,ct.z)}static get origin(){return tt.create(ct.x,ct.y)}static clone(t){return tt.create(t.x,t.y)}static create(t,r){return new tt(t,r)}}let Yb=Math.random;const ME={nextFrame:i=>requestAnimationFrame(i),cancel:i=>cancelAnimationFrame(i)};function Oe(){return Zt(Yb(),0,1-Number.EPSILON)}function Kb(i){return ME.nextFrame(i)}function Zb(i){ME.cancel(i)}function Zt(i,t,r){return Math.min(Math.max(i,t),r)}function Vu(i,t,r,s){return Math.floor((i*r+t*s)/(r+s))}function Bt(i){const t=mn(i),r=0;let s=El(i);return t===s&&(s=r),Oe()*(t-s)+s}function ie(i){return yi(i)?i:Bt(i)}function El(i){return yi(i)?i:i.min}function mn(i){return yi(i)?i:i.max}function ye(i,t){if(i===t||t===void 0&&yi(i))return i;const r=El(i),s=mn(i);return t!==void 0?{min:Math.min(r,t),max:Math.max(s,t)}:ye(r,s)}function Et(i,t){const r=i.x-t.x,s=i.y-t.y,l=2;return{dx:r,dy:s,distance:Math.sqrt(r**l+s**l)}}function Mt(i,t){return Et(i,t).distance}function Si(i){return i*Math.PI/180}function Jb(i,t,r){if(yi(i))return Si(i);switch(i){case rt.top:return-Math.PI*Ke;case rt.topRight:return-Math.PI*Ed;case rt.right:return Sb;case rt.bottomRight:return Math.PI*Ed;case rt.bottom:return Math.PI*Ke;case rt.bottomLeft:return Math.PI*dm;case rt.left:return Math.PI;case rt.topLeft:return-Math.PI*dm;case rt.inside:return Math.atan2(r.y-t.y,r.x-t.x);case rt.outside:return Math.atan2(t.y-r.y,t.x-r.x);default:return Oe()*fb}}function $b(i){const t=tt.origin;return t.length=1,t.angle=i,t}function Rm(i,t,r,s){return tt.create(i.x*(r-s)/(r+s)+t.x*pt*s/(r+s),i.y)}function eC(i){var t,r;return{x:((t=i.position)==null?void 0:t.x)??Oe()*i.size.width,y:((r=i.position)==null?void 0:r.y)??Oe()*i.size.height}}function qE(i){return i?i.endsWith("%")?parseFloat(i)/mi:parseFloat(i):1}var Zi;(function(i){i.auto="auto",i.increase="increase",i.decrease="decrease",i.random="random"})(Zi||(Zi={}));var Ge;(function(i){i.increasing="increasing",i.decreasing="decreasing"})(Ge||(Ge={}));var to;(function(i){i.none="none",i.max="max",i.min="min"})(to||(to={}));var me;(function(i){i.bottom="bottom",i.left="left",i.right="right",i.top="top"})(me||(me={}));var or;(function(i){i.precise="precise",i.percent="percent"})(or||(or={}));var Jo;(function(i){i.max="max",i.min="min",i.random="random"})(Jo||(Jo={}));const tC={debug:console.debug,error:console.error,info:console.info,log:console.log,verbose:console.log,warning:console.warn};function no(){return tC}function nC(i){const t=new Map;return(...r)=>{const s=JSON.stringify(r);if(t.has(s))return t.get(s);const l=i(...r);return t.set(s,l),l}}function Im(i){const t={bounced:!1},{pSide:r,pOtherSide:s,rectSide:l,rectOtherSide:u,velocity:p,factor:h}=i;return s.min<u.min||s.min>u.max||s.max<u.min||s.max>u.max||(r.max>=l.min&&r.max<=(l.max+l.min)*Ke&&p>pm||r.min<=l.max&&r.min>(l.max+l.min)*Ke&&p<pm)&&(t.velocity=p*-h,t.bounced=!0),t}function iC(i,t){const r=Jt(t,s=>i.matches(s));return In(r)?r.some(s=>s):r}function io(){return typeof window>"u"||!window||typeof window.document>"u"||!window.document}function oC(){return!io()&&typeof matchMedia<"u"}function UE(i){if(oC())return matchMedia(i)}function rC(i){if(!(io()||typeof IntersectionObserver>"u"))return new IntersectionObserver(i)}function sC(i){if(!(io()||typeof MutationObserver>"u"))return new MutationObserver(i)}function je(i,t){return i===t||In(t)&&t.indexOf(i)>-1}async function ym(i,t){try{await document.fonts.load(`${t??"400"} 36px '${i??"Verdana"}'`)}catch{}}function aC(i){return Math.floor(Oe()*i.length)}function Sl(i,t,r=!0){return i[t!==void 0&&r?t%i.length:aC(i)]}function Hd(i,t,r,s,l){return lC(_s(i,s??0),t,r,l)}function lC(i,t,r,s){let l=!0;return(!s||s===me.bottom)&&(l=i.top<t.height+r.x),l&&(!s||s===me.left)&&(l=i.right>r.x),l&&(!s||s===me.right)&&(l=i.left<t.width+r.y),l&&(!s||s===me.top)&&(l=i.bottom>r.y),l}function _s(i,t){return{bottom:i.y+t,left:i.x-t,right:i.x+t,top:i.y-t}}function _t(i,...t){for(const r of t){if(r==null)continue;if(!Vo(r)){i=r;continue}const s=Array.isArray(r);s&&(Vo(i)||!i||!Array.isArray(i))?i=[]:!s&&(Vo(i)||!i||Array.isArray(i))&&(i={});for(const l in r){if(l==="__proto__")continue;const u=r,p=u[l],h=i;h[l]=Vo(p)&&Array.isArray(p)?p.map(f=>_t(h[l],f)):_t(h[l],p)}}return i}function Ad(i,t){return!!kE(t,r=>r.enable&&je(i,r.mode))}function Ld(i,t,r){Jt(t,s=>{const l=s.mode;s.enable&&je(i,l)&&cC(s,r)})}function cC(i,t){const r=i.selectors;Jt(r,s=>{t(s,i)})}function WE(i,t){if(!(!t||!i))return kE(i,r=>iC(t,r.selectors))}function Id(i){return{position:i.getPosition(),radius:i.getRadius(),mass:i.getMass(),velocity:i.velocity,factor:tt.create(ie(i.options.bounce.horizontal.value),ie(i.options.bounce.vertical.value))}}function jE(i,t){const{x:r,y:s}=i.velocity.sub(t.velocity),[l,u]=[i.position,t.position],{dx:p,dy:h}=Et(u,l);if(r*p+s*h<0)return;const g=-Math.atan2(h,p),E=i.mass,R=t.mass,y=i.velocity.rotate(g),w=t.velocity.rotate(g),b=Rm(y,w,E,R),X=Rm(w,y,E,R),N=b.rotate(-g),v=X.rotate(-g);i.velocity.x=N.x*i.factor.x,i.velocity.y=N.y*i.factor.y,t.velocity.x=v.x*t.factor.x,t.velocity.y=v.y*t.factor.y}function uC(i,t){const r=i.getPosition(),s=i.getRadius(),l=_s(r,s),u=i.options.bounce,p=Im({pSide:{min:l.left,max:l.right},pOtherSide:{min:l.top,max:l.bottom},rectSide:{min:t.left,max:t.right},rectOtherSide:{min:t.top,max:t.bottom},velocity:i.velocity.x,factor:ie(u.horizontal.value)});p.bounced&&(p.velocity!==void 0&&(i.velocity.x=p.velocity),p.position!==void 0&&(i.position.x=p.position));const h=Im({pSide:{min:l.top,max:l.bottom},pOtherSide:{min:l.left,max:l.right},rectSide:{min:t.top,max:t.bottom},rectOtherSide:{min:t.left,max:t.right},velocity:i.velocity.y,factor:ie(u.vertical.value)});h.bounced&&(h.velocity!==void 0&&(i.velocity.y=h.velocity),h.position!==void 0&&(i.position.y=h.position))}function Jt(i,t){return In(i)?i.map((s,l)=>t(s,l)):t(i,0)}function rn(i,t,r){return In(i)?Sl(i,t,r):i}function kE(i,t){return In(i)?i.find((s,l)=>t(s,l)):t(i,0)?i:void 0}function zE(i,t){const r=i.value,s=i.animation,l={delayTime:ie(s.delay)*dt,enable:s.enable,value:ie(i.value)*t,max:mn(r)*t,min:El(r)*t,loops:0,maxLoops:ie(s.count),time:0},u=1;if(s.enable){switch(l.decay=u-ie(s.decay),s.mode){case Zi.increase:l.status=Ge.increasing;break;case Zi.decrease:l.status=Ge.decreasing;break;case Zi.random:l.status=Oe()>=Ke?Ge.increasing:Ge.decreasing;break}const p=s.mode===Zi.auto;switch(s.startValue){case Jo.min:l.value=l.min,p&&(l.status=Ge.increasing);break;case Jo.max:l.value=l.max,p&&(l.status=Ge.decreasing);break;case Jo.random:default:l.value=Bt(l),p&&(l.status=Oe()>=Ke?Ge.increasing:Ge.decreasing);break}}return l.initialValue=l.value,l}function dC(i,t){if(!(i.mode===or.percent)){const{mode:l,...u}=i;return u}return"x"in i?{x:i.x/mi*t.width,y:i.y/mi*t.height}:{width:i.width/mi*t.width,height:i.height/mi*t.height}}function GE(i,t){return dC(i,t)}function pC(i,t,r,s,l){switch(t){case to.max:r>=l&&i.destroy();break;case to.min:r<=s&&i.destroy();break}}function Bd(i,t,r,s,l){if(i.destroyed||!t||!t.enable||(t.maxLoops??0)>0&&(t.loops??0)>(t.maxLoops??0))return;const E=(t.velocity??0)*l.factor,R=t.min,y=t.max,w=t.decay??1;if(t.time||(t.time=0),(t.delayTime??0)>0&&t.time<(t.delayTime??0)&&(t.time+=l.value),!((t.delayTime??0)>0&&t.time<(t.delayTime??0))){switch(t.status){case Ge.increasing:t.value>=y?(r?t.status=Ge.decreasing:t.value-=y,t.loops||(t.loops=0),t.loops++):t.value+=E;break;case Ge.decreasing:t.value<=R?(r?t.status=Ge.increasing:t.value+=y,t.loops||(t.loops=0),t.loops++):t.value-=E}t.velocity&&w!==1&&(t.velocity*=w),pC(i,s,t.value,R,y),i.destroyed||(t.value=Zt(t.value,R,y))}}function hC(i){var r,s,l,u;const t=document.createElement("div").style;if(!i)return t;for(const p in i){const h=i[p];if(!Object.prototype.hasOwnProperty.call(i,p)||Q(h))continue;const f=(r=i.getPropertyValue)==null?void 0:r.call(i,h);if(!f)continue;const g=(s=i.getPropertyPriority)==null?void 0:s.call(i,h);g?(u=t.setProperty)==null||u.call(t,h,f,g):(l=t.setProperty)==null||l.call(t,h,f)}return t}function fC(i){const t=document.createElement("div").style,r=10,s={width:"100%",height:"100%",margin:"0",padding:"0",borderWidth:"0",position:"fixed",zIndex:i.toString(r),"z-index":i.toString(r),top:"0",left:"0"};for(const l in s){const u=s[l];t.setProperty(l,u)}return t}const gC=nC(fC);var yd;(function(i){i.darken="darken",i.enlighten="enlighten"})(yd||(yd={}));function mC(i,t){if(t){for(const r of i.colorManagers.values())if(t.startsWith(r.stringPrefix))return r.parseString(t)}}function sn(i,t,r,s=!0){if(!t)return;const l=Ii(t)?{value:t}:t;if(Ii(l.value))return QE(i,l.value,r,s);if(In(l.value))return sn(i,{value:Sl(l.value,r,s)});for(const u of i.colorManagers.values()){const p=u.handleRangeColor(l);if(p)return p}}function QE(i,t,r,s=!0){if(!t)return;const l=Ii(t)?{value:t}:t;if(Ii(l.value))return l.value===ir?YE():EC(i,l.value);if(In(l.value))return QE(i,{value:Sl(l.value,r,s)});for(const u of i.colorManagers.values()){const p=u.handleColor(l);if(p)return p}}function Es(i,t,r,s=!0){const l=sn(i,t,r,s);return l?VE(l):void 0}function VE(i){const t=i.r/ol,r=i.g/ol,s=i.b/ol,l=Math.max(t,r,s),u=Math.min(t,r,s),p={h:um,l:(l+u)*Ke,s:md};return l!==u&&(p.s=p.l<Ke?(l-u)/(l+u):(l-u)/(pt-l-u),p.h=t===l?(r-s)/(l-u):p.h=r===l?pt+(s-t)/(l-u):pt*pt+(t-r)/(l-u)),p.l*=gd,p.s*=fd,p.h*=Eb,p.h<um&&(p.h+=Ki),p.h>=Ki&&(p.h-=Ki),p}function EC(i,t){return mC(i,t)}function rr(i){const t=(i.h%Ki+Ki)%Ki,r=Math.max(md,Math.min(fd,i.s)),s=Math.max(Wb,Math.min(gd,i.l)),l=t/Ki,u=r/fd,p=s/gd;if(r===md){const b=Math.round(p*Wi);return{r:b,g:b,b}}const h=(b,X,N)=>{if(N<0&&N++,N>1&&N--,N*Qu<1)return b+(X-b)*Qu*N;if(N*pt<1)return X;if(N*Gu<1*pt){const k=pt/Gu;return b+(X-b)*(k-N)*Qu}return b},f=p<Ke?p*(jb+u):p+u-p*u,g=pt*p-f,E=kb/Gu,R=Math.min(Wi,Wi*h(g,f,l+E)),y=Math.min(Wi,Wi*h(g,f,l)),w=Math.min(Wi,Wi*h(g,f,l-E));return{r:Math.round(R),g:Math.round(y),b:Math.round(w)}}function SC(i){const t=rr(i);return{a:i.a,b:t.b,g:t.g,r:t.r}}function YE(i){const t=zb,r=ol+Rd;return{b:Math.floor(Bt(ye(t,r))),g:Math.floor(Bt(ye(t,r))),r:Math.floor(Bt(ye(t,r)))}}function Wn(i,t){return`rgba(${i.r}, ${i.g}, ${i.b}, ${t??Dd})`}function Ss(i,t){return`hsla(${i.h}, ${i.s}%, ${i.l}%, ${t??Dd})`}function Md(i,t,r,s){let l=i,u=t;return l.r===void 0&&(l=rr(i)),u.r===void 0&&(u=rr(t)),{b:Vu(l.b,u.b,r,s),g:Vu(l.g,u.g,r,s),r:Vu(l.r,u.r,r,s)}}function bd(i,t,r){if(r===ir)return YE();if(r===il){const s=i.getFillColor()??i.getStrokeColor(),l=(t==null?void 0:t.getFillColor())??(t==null?void 0:t.getStrokeColor());if(s&&l&&t)return Md(s,l,i.getRadius(),t.getRadius());{const u=s??l;if(u)return rr(u)}}else return r}function KE(i,t,r,s){const l=Ii(t)?t:t.value;return l===ir?s?sn(i,{value:l}):r?ir:il:l===il?il:sn(i,{value:l})}function bm(i){return i!==void 0?{h:i.h.value,s:i.s.value,l:i.l.value}:void 0}function ZE(i,t,r){const s={h:{enable:!1,value:i.h},s:{enable:!1,value:i.s},l:{enable:!1,value:i.l}};return t&&(Yu(s.h,t.h,r),Yu(s.s,t.s,r),Yu(s.l,t.l,r)),s}function Yu(i,t,r){i.enable=t.enable,i.enable?(i.velocity=ie(t.speed)/mi*r,i.decay=HE-ie(t.decay),i.status=Ge.increasing,i.loops=Qb,i.maxLoops=ie(t.count),i.time=Vb,i.delayTime=ie(t.delay)*dt,t.sync||(i.velocity*=Oe(),i.value*=Oe()),i.initialValue=i.value,i.offset=ye(t.offset)):i.velocity=Gb}function Ku(i,t,r,s){if(!i||!i.enable||(i.maxLoops??0)>0&&(i.loops??0)>(i.maxLoops??0)||(i.time||(i.time=0),(i.delayTime??0)>0&&i.time<(i.delayTime??0)&&(i.time+=s.value),(i.delayTime??0)>0&&i.time<(i.delayTime??0)))return;const E=i.offset?Bt(i.offset):0,R=(i.velocity??0)*s.factor+E*3.6,y=i.decay??1,w=mn(t),b=El(t);!r||i.status===Ge.increasing?(i.value+=R,i.value>w&&(i.loops||(i.loops=0),i.loops++,r?i.status=Ge.decreasing:i.value-=w)):(i.value-=R,i.value<0&&(i.loops||(i.loops=0),i.loops++,i.status=Ge.increasing)),i.velocity&&y!==1&&(i.velocity*=y),i.value=Zt(i.value,b,w)}function JE(i,t){if(!i)return;const{h:r,s,l}=i,u={h:{min:0,max:360},s:{min:0,max:100},l:{min:0,max:100}};r&&Ku(r,u.h,!1,t),s&&Ku(s,u.s,!0,t),l&&Ku(l,u.l,!0,t)}function fs(i,t,r){i.beginPath(),i.moveTo(t.x,t.y),i.lineTo(r.x,r.y),i.closePath()}function RC(i,t,r){i.fillStyle=r??"rgba(0,0,0,0)",i.fillRect(ct.x,ct.y,t.width,t.height)}function IC(i,t,r,s){r&&(i.globalAlpha=s,i.drawImage(r,ct.x,ct.y,t.width,t.height),i.globalAlpha=1)}function Zu(i,t){i.clearRect(ct.x,ct.y,t.width,t.height)}function yC(i){const{container:t,context:r,particle:s,delta:l,colorStyles:u,backgroundMask:p,composite:h,radius:f,opacity:g,shadow:E,transform:R}=i,y=s.getPosition(),w=s.rotation+(s.pathRotation?s.velocity.angle:qb),b={sin:Math.sin(w),cos:Math.cos(w)},X=!!w,N={a:b.cos*(R.a??ja.a),b:X?b.sin*(R.b??Rd):R.b??ja.b,c:X?-b.sin*(R.c??Rd):R.c??ja.c,d:b.cos*(R.d??ja.d)};r.setTransform(N.a,N.b,N.c,N.d,y.x,y.y),p&&(r.globalCompositeOperation=h);const v=s.shadowColor;E.enable&&v&&(r.shadowBlur=E.blur,r.shadowColor=Wn(v),r.shadowOffsetX=E.offset.x,r.shadowOffsetY=E.offset.y),u.fill&&(r.fillStyle=u.fill);const A=s.strokeWidth??LE;r.lineWidth=A,u.stroke&&(r.strokeStyle=u.stroke);const k={container:t,context:r,particle:s,radius:f,opacity:g,delta:l,transformData:N,strokeWidth:A};CC(k),wC(k),bC(k),r.globalCompositeOperation="source-over",r.resetTransform()}function bC(i){const{container:t,context:r,particle:s,radius:l,opacity:u,delta:p,transformData:h}=i;if(!s.effect)return;const f=t.effectDrawers.get(s.effect);f&&f.draw({context:r,particle:s,radius:l,opacity:u,delta:p,pixelRatio:t.retina.pixelRatio,transformData:{...h}})}function CC(i){const{container:t,context:r,particle:s,radius:l,opacity:u,delta:p,strokeWidth:h,transformData:f}=i;if(!s.shape)return;const g=t.shapeDrawers.get(s.shape);g&&(r.beginPath(),g.draw({context:r,particle:s,radius:l,opacity:u,delta:p,pixelRatio:t.retina.pixelRatio,transformData:{...f}}),s.shapeClose&&r.closePath(),h>LE&&r.stroke(),s.shapeFill&&r.fill())}function wC(i){const{container:t,context:r,particle:s,radius:l,opacity:u,delta:p,transformData:h}=i;if(!s.shape)return;const f=t.shapeDrawers.get(s.shape);f!=null&&f.afterDraw&&f.afterDraw({context:r,particle:s,radius:l,opacity:u,delta:p,pixelRatio:t.retina.pixelRatio,transformData:{...h}})}function _C(i,t,r){t.draw&&t.draw(i,r)}function TC(i,t,r,s){t.drawParticle&&t.drawParticle(i,r,s)}function vC(i,t,r){return{h:i.h,s:i.s,l:i.l+(t===yd.darken?-1:Ub)*r}}function xC(i,t,r){const s=t[r];s!==void 0&&(i[r]=(i[r]??Rb)*s)}function Cm(i,t,r=!1){if(!t)return;const s=i;if(!s)return;const l=s.style;if(!l)return;const u=new Set;for(const p in l)Object.prototype.hasOwnProperty.call(l,p)&&u.add(l[p]);for(const p in t)Object.prototype.hasOwnProperty.call(t,p)&&u.add(t[p]);for(const p of u){const h=t.getPropertyValue(p);h?l.setProperty(p,h,r?"important":""):l.removeProperty(p)}}class PC{constructor(t,r){this.container=t,this._applyPostDrawUpdaters=u=>{var p;for(const h of this._postDrawUpdaters)(p=h.afterDraw)==null||p.call(h,u)},this._applyPreDrawUpdaters=(u,p,h,f,g,E)=>{var R;for(const y of this._preDrawUpdaters){if(y.getColorStyles){const{fill:w,stroke:b}=y.getColorStyles(p,u,h,f);w&&(g.fill=w),b&&(g.stroke=b)}if(y.getTransformValues){const w=y.getTransformValues(p);for(const b in w)xC(E,w,b)}(R=y.beforeDraw)==null||R.call(y,p)}},this._applyResizePlugins=()=>{var u;for(const p of this._resizePlugins)(u=p.resize)==null||u.call(p)},this._getPluginParticleColors=u=>{let p,h;for(const f of this._colorPlugins)if(!p&&f.particleFillColor&&(p=Es(this._engine,f.particleFillColor(u))),!h&&f.particleStrokeColor&&(h=Es(this._engine,f.particleStrokeColor(u))),p&&h)break;return[p,h]},this._initCover=async()=>{const u=this.container.actualOptions,p=u.backgroundMask.cover,h=p.color;if(h){const f=sn(this._engine,h);if(f){const g={...f,a:p.opacity};this._coverColorStyle=Wn(g,g.a)}}else await new Promise((f,g)=>{if(!p.image)return;const E=document.createElement("img");E.addEventListener("load",()=>{this._coverImage={image:E,opacity:p.opacity},f()}),E.addEventListener("error",R=>{g(R.error)}),E.src=p.image})},this._initStyle=()=>{const u=this.element,p=this.container.actualOptions;if(u){this._fullScreen?this._setFullScreenStyle():this._resetOriginalStyle();for(const h in p.style){if(!h||!p.style||!Object.prototype.hasOwnProperty.call(p.style,h))continue;const f=p.style[h];f&&u.style.setProperty(h,f,"important")}}},this._initTrail=async()=>{const u=this.container.actualOptions,p=u.particles.move.trail,h=p.fill;if(!p.enable)return;const f=FE/p.length;if(h.color){const g=sn(this._engine,h.color);if(!g)return;this._trailFill={color:{...g},opacity:f}}else await new Promise((g,E)=>{if(!h.image)return;const R=document.createElement("img");R.addEventListener("load",()=>{this._trailFill={image:R,opacity:f},g()}),R.addEventListener("error",y=>{E(y.error)}),R.src=h.image})},this._paintBase=u=>{this.draw(p=>RC(p,this.size,u))},this._paintImage=(u,p)=>{this.draw(h=>IC(h,this.size,u,p))},this._repairStyle=()=>{const u=this.element;u&&(this._safeMutationObserver(p=>p.disconnect()),this._initStyle(),this.initBackground(),this._safeMutationObserver(p=>{!u||!(u instanceof Node)||p.observe(u,{attributes:!0})}))},this._resetOriginalStyle=()=>{const u=this.element,p=this._originalStyle;!u||!p||Cm(u,p,!0)},this._safeMutationObserver=u=>{this._mutationObserver&&u(this._mutationObserver)},this._setFullScreenStyle=()=>{const u=this.element;u&&Cm(u,gC(this.container.actualOptions.fullScreen.zIndex),!0)},this._engine=r,this._standardSize={height:0,width:0};const s=t.retina.pixelRatio,l=this._standardSize;this.size={height:l.height*s,width:l.width*s},this._context=null,this._generated=!1,this._preDrawUpdaters=[],this._postDrawUpdaters=[],this._resizePlugins=[],this._colorPlugins=[]}get _fullScreen(){return this.container.actualOptions.fullScreen.enable}clear(){const t=this.container.actualOptions,r=t.particles.move.trail,s=this._trailFill;t.backgroundMask.enable?this.paint():r.enable&&r.length>yb&&s?s.color?this._paintBase(Wn(s.color,s.opacity)):s.image&&this._paintImage(s.image,s.opacity):t.clear&&this.draw(l=>{Zu(l,this.size)})}destroy(){if(this.stop(),this._generated){const t=this.element;t==null||t.remove(),this.element=void 0}else this._resetOriginalStyle();this._preDrawUpdaters=[],this._postDrawUpdaters=[],this._resizePlugins=[],this._colorPlugins=[]}draw(t){const r=this._context;if(r)return t(r)}drawAsync(t){const r=this._context;if(r)return t(r)}drawParticle(t,r){if(t.spawning||t.destroyed)return;const s=t.getRadius();if(s<=Ib)return;const l=t.getFillColor(),u=t.getStrokeColor()??l;let[p,h]=this._getPluginParticleColors(t);p||(p=l),h||(h=u),!(!p&&!h)&&this.draw(f=>{var U;const g=this.container,E=g.actualOptions,R=t.options.zIndex,y=bb-t.zIndexFactor,w=y**R.opacityRate,b=t.bubble.opacity??((U=t.opacity)==null?void 0:U.value)??Dd,X=t.strokeOpacity??b,N=b*w,v=X*w,A={},k={fill:p?Ss(p,N):void 0};k.stroke=h?Ss(h,v):k.fill,this._applyPreDrawUpdaters(f,t,s,N,k,A),yC({container:g,context:f,particle:t,delta:r,colorStyles:k,backgroundMask:E.backgroundMask.enable,composite:E.backgroundMask.composite,radius:s*y**R.sizeRate,opacity:N,shadow:t.options.shadow,transform:A}),this._applyPostDrawUpdaters(t)})}drawParticlePlugin(t,r,s){this.draw(l=>TC(l,t,r,s))}drawPlugin(t,r){this.draw(s=>_C(s,t,r))}async init(){this._safeMutationObserver(t=>t.disconnect()),this._mutationObserver=sC(t=>{for(const r of t)r.type==="attributes"&&r.attributeName==="style"&&this._repairStyle()}),this.resize(),this._initStyle(),await this._initCover();try{await this._initTrail()}catch(t){no().error(t)}this.initBackground(),this._safeMutationObserver(t=>{!this.element||!(this.element instanceof Node)||t.observe(this.element,{attributes:!0})}),this.initUpdaters(),this.initPlugins(),this.paint()}initBackground(){const t=this.container.actualOptions,r=t.background,s=this.element;if(!s)return;const l=s.style;if(l){if(r.color){const u=sn(this._engine,r.color);l.backgroundColor=u?Wn(u,r.opacity):""}else l.backgroundColor="";l.backgroundImage=r.image||"",l.backgroundPosition=r.position||"",l.backgroundRepeat=r.repeat||"",l.backgroundSize=r.size||""}}initPlugins(){this._resizePlugins=[];for(const t of this.container.plugins.values())t.resize&&this._resizePlugins.push(t),(t.particleFillColor??t.particleStrokeColor)&&this._colorPlugins.push(t)}initUpdaters(){this._preDrawUpdaters=[],this._postDrawUpdaters=[];for(const t of this.container.particles.updaters)t.afterDraw&&this._postDrawUpdaters.push(t),(t.getColorStyles??t.getTransformValues??t.beforeDraw)&&this._preDrawUpdaters.push(t)}loadCanvas(t){this._generated&&this.element&&this.element.remove(),this._generated=t.dataset&&Yi in t.dataset?t.dataset[Yi]==="true":this._generated,this.element=t,this.element.ariaHidden="true",this._originalStyle=hC(this.element.style);const r=this._standardSize;r.height=t.offsetHeight,r.width=t.offsetWidth;const s=this.container.retina.pixelRatio,l=this.size;t.height=l.height=r.height*s,t.width=l.width=r.width*s,this._context=this.element.getContext("2d"),this._safeMutationObserver(u=>u.disconnect()),this.container.retina.init(),this.initBackground(),this._safeMutationObserver(u=>{!this.element||!(this.element instanceof Node)||u.observe(this.element,{attributes:!0})})}paint(){const t=this.container.actualOptions;this.draw(r=>{t.backgroundMask.enable&&t.backgroundMask.cover?(Zu(r,this.size),this._coverImage?this._paintImage(this._coverImage.image,this._coverImage.opacity):this._coverColorStyle?this._paintBase(this._coverColorStyle):this._paintBase()):this._paintBase()})}resize(){if(!this.element)return!1;const t=this.container,r=t.canvas._standardSize,s={width:this.element.offsetWidth,height:this.element.offsetHeight},l=t.retina.pixelRatio,u={width:s.width*l,height:s.height*l};if(s.height===r.height&&s.width===r.width&&u.height===this.element.height&&u.width===this.element.width)return!1;const p={...r};r.height=s.height,r.width=s.width;const h=this.size;return this.element.width=h.width=u.width,this.element.height=h.height=u.height,this.container.started&&t.particles.setResizeFactor({width:r.width/p.width,height:r.height/p.height}),!0}stop(){this._safeMutationObserver(t=>t.disconnect()),this._mutationObserver=void 0,this.draw(t=>Zu(t,this.size))}async windowResize(){if(!this.element||!this.resize())return;const t=this.container,r=t.updateActualOptions();t.particles.setDensity(),this._applyResizePlugins(),r&&await t.refresh()}}var $o;(function(i){i.canvas="canvas",i.parent="parent",i.window="window"})($o||($o={}));function Yt(i,t,r,s,l){if(s){let u={passive:!0};BE(l)?u.capture=l:l!==void 0&&(u=l),i.addEventListener(t,r,u)}else{const u=l;i.removeEventListener(t,r,u)}}class XC{constructor(t){this.container=t,this._doMouseTouchClick=r=>{const s=this.container,l=s.actualOptions;if(this._canPush){const u=s.interactivity.mouse,p=u.position;if(!p)return;u.clickPosition={...p},u.clickTime=new Date().getTime();const h=l.interactivity.events.onClick;Jt(h.mode,f=>this.container.handleClickMode(f))}r.type==="touchend"&&setTimeout(()=>this._mouseTouchFinish(),Mb)},this._handleThemeChange=r=>{const s=r,l=this.container,u=l.options,p=u.defaultThemes,h=s.matches?p.dark:p.light,f=u.themes.find(g=>g.name===h);f!=null&&f.default.auto&&l.loadTheme(h)},this._handleVisibilityChange=()=>{const r=this.container,s=r.actualOptions;this._mouseTouchFinish(),s.pauseOnBlur&&(document!=null&&document.hidden?(r.pageHidden=!0,r.pause()):(r.pageHidden=!1,r.animationStatus?r.play(!0):r.draw(!0)))},this._handleWindowResize=()=>{this._resizeTimeout&&(clearTimeout(this._resizeTimeout),delete this._resizeTimeout);const r=async()=>{const s=this.container.canvas;await(s==null?void 0:s.windowResize())};this._resizeTimeout=setTimeout(()=>void r(),this.container.actualOptions.interactivity.events.resize.delay*dt)},this._manageInteractivityListeners=(r,s)=>{const l=this._handlers,u=this.container,p=u.actualOptions,h=u.interactivity.element;if(!h)return;const f=h,g=u.canvas.element;g&&(g.style.pointerEvents=f===g?"initial":"none"),(p.interactivity.events.onHover.enable||p.interactivity.events.onClick.enable)&&(Yt(h,eo,l.mouseMove,s),Yt(h,cb,l.touchStart,s),Yt(h,ub,l.touchMove,s),p.interactivity.events.onClick.enable?(Yt(h,om,l.touchEndClick,s),Yt(h,ab,l.mouseUp,s),Yt(h,sb,l.mouseDown,s)):Yt(h,om,l.touchEnd,s),Yt(h,r,l.mouseLeave,s),Yt(h,db,l.touchCancel,s))},this._manageListeners=r=>{const s=this._handlers,l=this.container,u=l.actualOptions,p=u.interactivity.detectsOn,h=l.canvas.element;let f=hd;p===$o.window?(l.interactivity.element=window,f=lb):p===$o.parent&&h?l.interactivity.element=h.parentElement??h.parentNode:l.interactivity.element=h,this._manageMediaMatch(r),this._manageResize(r),this._manageInteractivityListeners(f,r),document&&Yt(document,hb,s.visibilityChange,r,!1)},this._manageMediaMatch=r=>{const s=this._handlers,l=UE("(prefers-color-scheme: dark)");if(l){if(l.addEventListener!==void 0){Yt(l,"change",s.themeChange,r);return}l.addListener!==void 0&&(r?l.addListener(s.oldThemeChange):l.removeListener(s.oldThemeChange))}},this._manageResize=r=>{const s=this._handlers,l=this.container;if(!l.actualOptions.interactivity.events.resize)return;if(typeof ResizeObserver>"u"){Yt(window,pb,s.resize,r);return}const p=l.canvas.element;this._resizeObserver&&!r?(p&&this._resizeObserver.unobserve(p),this._resizeObserver.disconnect(),delete this._resizeObserver):!this._resizeObserver&&r&&p&&(this._resizeObserver=new ResizeObserver(h=>{h.find(g=>g.target===p)&&this._handleWindowResize()}),this._resizeObserver.observe(p))},this._mouseDown=()=>{const{interactivity:r}=this.container;if(!r)return;const{mouse:s}=r;s.clicking=!0,s.downPosition=s.position},this._mouseTouchClick=r=>{const s=this.container,l=s.actualOptions,{mouse:u}=s.interactivity;u.inside=!0;let p=!1;const h=u.position;if(!(!h||!l.interactivity.events.onClick.enable)){for(const f of s.plugins.values())if(f.clickPositionValid&&(p=f.clickPositionValid(h),p))break;p||this._doMouseTouchClick(r),u.clicking=!1}},this._mouseTouchFinish=()=>{const r=this.container.interactivity;if(!r)return;const s=r.mouse;delete s.position,delete s.clickPosition,delete s.downPosition,r.status=hd,s.inside=!1,s.clicking=!1},this._mouseTouchMove=r=>{const s=this.container,l=s.actualOptions,u=s.interactivity,p=s.canvas.element;if(!(u!=null&&u.element))return;u.mouse.inside=!0;let h;if(r.type.startsWith("pointer")){this._canPush=!0;const g=r;if(u.element===window){if(p){const E=p.getBoundingClientRect();h={x:g.clientX-E.left,y:g.clientY-E.top}}}else if(l.interactivity.detectsOn===$o.parent){const E=g.target,R=g.currentTarget;if(E&&R&&p){const y=E.getBoundingClientRect(),w=R.getBoundingClientRect(),b=p.getBoundingClientRect();h={x:g.offsetX+pt*y.left-(w.left+b.left),y:g.offsetY+pt*y.top-(w.top+b.top)}}else h={x:g.offsetX??g.clientX,y:g.offsetY??g.clientY}}else g.target===p&&(h={x:g.offsetX??g.clientX,y:g.offsetY??g.clientY})}else if(this._canPush=r.type!=="touchmove",p){const g=r,E=g.touches[g.touches.length-AE],R=p.getBoundingClientRect();h={x:E.clientX-(R.left??cl),y:E.clientY-(R.top??cl)}}const f=s.retina.pixelRatio;h&&(h.x*=f,h.y*=f),u.mouse.position=h,u.status=eo},this._touchEnd=r=>{const s=r,l=Array.from(s.changedTouches);for(const u of l)this._touches.delete(u.identifier);this._mouseTouchFinish()},this._touchEndClick=r=>{const s=r,l=Array.from(s.changedTouches);for(const u of l)this._touches.delete(u.identifier);this._mouseTouchClick(r)},this._touchStart=r=>{const s=r,l=Array.from(s.changedTouches);for(const u of l)this._touches.set(u.identifier,performance.now());this._mouseTouchMove(r)},this._canPush=!0,this._touches=new Map,this._handlers={mouseDown:()=>this._mouseDown(),mouseLeave:()=>this._mouseTouchFinish(),mouseMove:r=>this._mouseTouchMove(r),mouseUp:r=>this._mouseTouchClick(r),touchStart:r=>this._touchStart(r),touchMove:r=>this._mouseTouchMove(r),touchEnd:r=>this._touchEnd(r),touchCancel:r=>this._touchEnd(r),touchEndClick:r=>this._touchEndClick(r),visibilityChange:()=>this._handleVisibilityChange(),themeChange:r=>this._handleThemeChange(r),oldThemeChange:r=>this._handleThemeChange(r),resize:()=>{this._handleWindowResize()}}}addListeners(){this._manageListeners(!0)}removeListeners(){this._manageListeners(!1)}}var mt;(function(i){i.configAdded="configAdded",i.containerInit="containerInit",i.particlesSetup="particlesSetup",i.containerStarted="containerStarted",i.containerStopped="containerStopped",i.containerDestroyed="containerDestroyed",i.containerPaused="containerPaused",i.containerPlay="containerPlay",i.containerBuilt="containerBuilt",i.particleAdded="particleAdded",i.particleDestroyed="particleDestroyed",i.particleRemoved="particleRemoved"})(mt||(mt={}));class ft{constructor(){this.value=""}static create(t,r){const s=new ft;return s.load(t),r!==void 0&&(Ii(r)||In(r)?s.load({value:r}):s.load(r)),s}load(t){Q(t)||Q(t.value)||(this.value=t.value)}}class NC{constructor(){this.color=new ft,this.color.value="",this.image="",this.position="",this.repeat="",this.size="",this.opacity=1}load(t){Q(t)||(t.color!==void 0&&(this.color=ft.create(this.color,t.color)),t.image!==void 0&&(this.image=t.image),t.position!==void 0&&(this.position=t.position),t.repeat!==void 0&&(this.repeat=t.repeat),t.size!==void 0&&(this.size=t.size),t.opacity!==void 0&&(this.opacity=t.opacity))}}class OC{constructor(){this.opacity=1}load(t){Q(t)||(t.color!==void 0&&(this.color=ft.create(this.color,t.color)),t.image!==void 0&&(this.image=t.image),t.opacity!==void 0&&(this.opacity=t.opacity))}}class FC{constructor(){this.composite="destination-out",this.cover=new OC,this.enable=!1}load(t){if(!Q(t)){if(t.composite!==void 0&&(this.composite=t.composite),t.cover!==void 0){const r=t.cover,s=Ii(t.cover)?{color:t.cover}:t.cover;this.cover.load(r.color!==void 0||r.image!==void 0?r:{color:s})}t.enable!==void 0&&(this.enable=t.enable)}}}class DC{constructor(){this.enable=!0,this.zIndex=0}load(t){Q(t)||(t.enable!==void 0&&(this.enable=t.enable),t.zIndex!==void 0&&(this.zIndex=t.zIndex))}}class HC{constructor(){this.enable=!1,this.mode=[]}load(t){Q(t)||(t.enable!==void 0&&(this.enable=t.enable),t.mode!==void 0&&(this.mode=t.mode))}}var sr;(function(i){i.circle="circle",i.rectangle="rectangle"})(sr||(sr={}));class wm{constructor(){this.selectors=[],this.enable=!1,this.mode=[],this.type=sr.circle}load(t){Q(t)||(t.selectors!==void 0&&(this.selectors=t.selectors),t.enable!==void 0&&(this.enable=t.enable),t.mode!==void 0&&(this.mode=t.mode),t.type!==void 0&&(this.type=t.type))}}class AC{constructor(){this.enable=!1,this.force=2,this.smooth=10}load(t){Q(t)||(t.enable!==void 0&&(this.enable=t.enable),t.force!==void 0&&(this.force=t.force),t.smooth!==void 0&&(this.smooth=t.smooth))}}class LC{constructor(){this.enable=!1,this.mode=[],this.parallax=new AC}load(t){Q(t)||(t.enable!==void 0&&(this.enable=t.enable),t.mode!==void 0&&(this.mode=t.mode),this.parallax.load(t.parallax))}}class BC{constructor(){this.delay=.5,this.enable=!0}load(t){Q(t)||(t.delay!==void 0&&(this.delay=t.delay),t.enable!==void 0&&(this.enable=t.enable))}}class MC{constructor(){this.onClick=new HC,this.onDiv=new wm,this.onHover=new LC,this.resize=new BC}load(t){if(Q(t))return;this.onClick.load(t.onClick);const r=t.onDiv;r!==void 0&&(this.onDiv=Jt(r,s=>{const l=new wm;return l.load(s),l})),this.onHover.load(t.onHover),this.resize.load(t.resize)}}class qC{constructor(t,r){this._engine=t,this._container=r}load(t){if(Q(t)||!this._container)return;const r=this._engine.interactors.get(this._container);if(r)for(const s of r)s.loadModeOptions&&s.loadModeOptions(this,t)}}class $E{constructor(t,r){this.detectsOn=$o.window,this.events=new MC,this.modes=new qC(t,r)}load(t){if(Q(t))return;const r=t.detectsOn;r!==void 0&&(this.detectsOn=r),this.events.load(t.events),this.modes.load(t.modes)}}class UC{load(t){Q(t)||(t.position&&(this.position={x:t.position.x??Sm,y:t.position.y??Sm,mode:t.position.mode??or.percent}),t.options&&(this.options=_t({},t.options)))}}var Ji;(function(i){i.screen="screen",i.canvas="canvas"})(Ji||(Ji={}));class WC{constructor(){this.maxWidth=1/0,this.options={},this.mode=Ji.canvas}load(t){Q(t)||(Q(t.maxWidth)||(this.maxWidth=t.maxWidth),Q(t.mode)||(t.mode===Ji.screen?this.mode=Ji.screen:this.mode=Ji.canvas),Q(t.options)||(this.options=_t({},t.options)))}}var gi;(function(i){i.any="any",i.dark="dark",i.light="light"})(gi||(gi={}));class jC{constructor(){this.auto=!1,this.mode=gi.any,this.value=!1}load(t){Q(t)||(t.auto!==void 0&&(this.auto=t.auto),t.mode!==void 0&&(this.mode=t.mode),t.value!==void 0&&(this.value=t.value))}}class kC{constructor(){this.name="",this.default=new jC}load(t){Q(t)||(t.name!==void 0&&(this.name=t.name),this.default.load(t.default),t.options!==void 0&&(this.options=_t({},t.options)))}}class qd{constructor(){this.count=0,this.enable=!1,this.speed=1,this.decay=0,this.delay=0,this.sync=!1}load(t){Q(t)||(t.count!==void 0&&(this.count=ye(t.count)),t.enable!==void 0&&(this.enable=t.enable),t.speed!==void 0&&(this.speed=ye(t.speed)),t.decay!==void 0&&(this.decay=ye(t.decay)),t.delay!==void 0&&(this.delay=ye(t.delay)),t.sync!==void 0&&(this.sync=t.sync))}}class Ud extends qd{constructor(){super(),this.mode=Zi.auto,this.startValue=Jo.random}load(t){super.load(t),!Q(t)&&(t.mode!==void 0&&(this.mode=t.mode),t.startValue!==void 0&&(this.startValue=t.startValue))}}class Ju extends qd{constructor(){super(),this.offset=0,this.sync=!0}load(t){super.load(t),!Q(t)&&t.offset!==void 0&&(this.offset=ye(t.offset))}}class zC{constructor(){this.h=new Ju,this.s=new Ju,this.l=new Ju}load(t){Q(t)||(this.h.load(t.h),this.s.load(t.s),this.l.load(t.l))}}class Rs extends ft{constructor(){super(),this.animation=new zC}static create(t,r){const s=new Rs;return s.load(t),r!==void 0&&(Ii(r)||In(r)?s.load({value:r}):s.load(r)),s}load(t){if(super.load(t),Q(t))return;const r=t.animation;r!==void 0&&(r.enable!==void 0?this.animation.h.load(r):this.animation.load(t.animation))}}var er;(function(i){i.absorb="absorb",i.bounce="bounce",i.destroy="destroy"})(er||(er={}));class GC{constructor(){this.speed=2}load(t){Q(t)||t.speed!==void 0&&(this.speed=t.speed)}}class QC{constructor(){this.enable=!0,this.retries=0}load(t){Q(t)||(t.enable!==void 0&&(this.enable=t.enable),t.retries!==void 0&&(this.retries=t.retries))}}class oo{constructor(){this.value=0}load(t){Q(t)||Q(t.value)||(this.value=ye(t.value))}}class VC extends oo{constructor(){super(),this.animation=new qd}load(t){if(super.load(t),Q(t))return;const r=t.animation;r!==void 0&&this.animation.load(r)}}class eS extends VC{constructor(){super(),this.animation=new Ud}load(t){super.load(t)}}class _m extends oo{constructor(){super(),this.value=1}}class tS{constructor(){this.horizontal=new _m,this.vertical=new _m}load(t){Q(t)||(this.horizontal.load(t.horizontal),this.vertical.load(t.vertical))}}class YC{constructor(){this.absorb=new GC,this.bounce=new tS,this.enable=!1,this.maxSpeed=50,this.mode=er.bounce,this.overlap=new QC}load(t){Q(t)||(this.absorb.load(t.absorb),this.bounce.load(t.bounce),t.enable!==void 0&&(this.enable=t.enable),t.maxSpeed!==void 0&&(this.maxSpeed=ye(t.maxSpeed)),t.mode!==void 0&&(this.mode=t.mode),this.overlap.load(t.overlap))}}class KC{constructor(){this.close=!0,this.fill=!0,this.options={},this.type=[]}load(t){if(Q(t))return;const r=t.options;if(r!==void 0)for(const s in r){const l=r[s];l&&(this.options[s]=_t(this.options[s]??{},l))}t.close!==void 0&&(this.close=t.close),t.fill!==void 0&&(this.fill=t.fill),t.type!==void 0&&(this.type=t.type)}}class ZC{constructor(){this.offset=0,this.value=90}load(t){Q(t)||(t.offset!==void 0&&(this.offset=ye(t.offset)),t.value!==void 0&&(this.value=ye(t.value)))}}class JC{constructor(){this.distance=200,this.enable=!1,this.rotate={x:3e3,y:3e3}}load(t){if(!Q(t)&&(t.distance!==void 0&&(this.distance=ye(t.distance)),t.enable!==void 0&&(this.enable=t.enable),t.rotate)){const r=t.rotate.x;r!==void 0&&(this.rotate.x=r);const s=t.rotate.y;s!==void 0&&(this.rotate.y=s)}}}class $C{constructor(){this.x=50,this.y=50,this.mode=or.percent,this.radius=0}load(t){Q(t)||(t.x!==void 0&&(this.x=t.x),t.y!==void 0&&(this.y=t.y),t.mode!==void 0&&(this.mode=t.mode),t.radius!==void 0&&(this.radius=t.radius))}}class ew{constructor(){this.acceleration=9.81,this.enable=!1,this.inverse=!1,this.maxSpeed=50}load(t){Q(t)||(t.acceleration!==void 0&&(this.acceleration=ye(t.acceleration)),t.enable!==void 0&&(this.enable=t.enable),t.inverse!==void 0&&(this.inverse=t.inverse),t.maxSpeed!==void 0&&(this.maxSpeed=ye(t.maxSpeed)))}}class tw{constructor(){this.clamp=!0,this.delay=new oo,this.enable=!1,this.options={}}load(t){Q(t)||(t.clamp!==void 0&&(this.clamp=t.clamp),this.delay.load(t.delay),t.enable!==void 0&&(this.enable=t.enable),this.generator=t.generator,t.options&&(this.options=_t(this.options,t.options)))}}class nw{load(t){Q(t)||(t.color!==void 0&&(this.color=ft.create(this.color,t.color)),t.image!==void 0&&(this.image=t.image))}}class iw{constructor(){this.enable=!1,this.length=10,this.fill=new nw}load(t){Q(t)||(t.enable!==void 0&&(this.enable=t.enable),t.fill!==void 0&&this.fill.load(t.fill),t.length!==void 0&&(this.length=t.length))}}var Qe;(function(i){i.bounce="bounce",i.none="none",i.out="out",i.destroy="destroy",i.split="split"})(Qe||(Qe={}));class ow{constructor(){this.default=Qe.out}load(t){Q(t)||(t.default!==void 0&&(this.default=t.default),this.bottom=t.bottom??t.default,this.left=t.left??t.default,this.right=t.right??t.default,this.top=t.top??t.default)}}class rw{constructor(){this.acceleration=0,this.enable=!1}load(t){Q(t)||(t.acceleration!==void 0&&(this.acceleration=ye(t.acceleration)),t.enable!==void 0&&(this.enable=t.enable),t.position&&(this.position=_t({},t.position)))}}class sw{constructor(){this.angle=new ZC,this.attract=new JC,this.center=new $C,this.decay=0,this.distance={},this.direction=rt.none,this.drift=0,this.enable=!1,this.gravity=new ew,this.path=new tw,this.outModes=new ow,this.random=!1,this.size=!1,this.speed=2,this.spin=new rw,this.straight=!1,this.trail=new iw,this.vibrate=!1,this.warp=!1}load(t){if(Q(t))return;this.angle.load(yi(t.angle)?{value:t.angle}:t.angle),this.attract.load(t.attract),this.center.load(t.center),t.decay!==void 0&&(this.decay=ye(t.decay)),t.direction!==void 0&&(this.direction=t.direction),t.distance!==void 0&&(this.distance=yi(t.distance)?{horizontal:t.distance,vertical:t.distance}:{...t.distance}),t.drift!==void 0&&(this.drift=ye(t.drift)),t.enable!==void 0&&(this.enable=t.enable),this.gravity.load(t.gravity);const r=t.outModes;r!==void 0&&(Vo(r)?this.outModes.load(r):this.outModes.load({default:r})),this.path.load(t.path),t.random!==void 0&&(this.random=t.random),t.size!==void 0&&(this.size=t.size),t.speed!==void 0&&(this.speed=ye(t.speed)),this.spin.load(t.spin),t.straight!==void 0&&(this.straight=t.straight),this.trail.load(t.trail),t.vibrate!==void 0&&(this.vibrate=t.vibrate),t.warp!==void 0&&(this.warp=t.warp)}}class aw extends Ud{constructor(){super(),this.destroy=to.none,this.speed=2}load(t){super.load(t),!Q(t)&&t.destroy!==void 0&&(this.destroy=t.destroy)}}class lw extends eS{constructor(){super(),this.animation=new aw,this.value=1}load(t){if(Q(t))return;super.load(t);const r=t.animation;r!==void 0&&this.animation.load(r)}}class cw{constructor(){this.enable=!1,this.width=1920,this.height=1080}load(t){if(Q(t))return;t.enable!==void 0&&(this.enable=t.enable);const r=t.width;r!==void 0&&(this.width=r);const s=t.height;s!==void 0&&(this.height=s)}}var Is;(function(i){i.delete="delete",i.wait="wait"})(Is||(Is={}));class uw{constructor(){this.mode=Is.delete,this.value=0}load(t){Q(t)||(t.mode!==void 0&&(this.mode=t.mode),t.value!==void 0&&(this.value=t.value))}}class dw{constructor(){this.density=new cw,this.limit=new uw,this.value=0}load(t){Q(t)||(this.density.load(t.density),this.limit.load(t.limit),t.value!==void 0&&(this.value=t.value))}}class pw{constructor(){this.blur=0,this.color=new ft,this.enable=!1,this.offset={x:0,y:0},this.color.value="#000"}load(t){Q(t)||(t.blur!==void 0&&(this.blur=t.blur),this.color=ft.create(this.color,t.color),t.enable!==void 0&&(this.enable=t.enable),t.offset!==void 0&&(t.offset.x!==void 0&&(this.offset.x=t.offset.x),t.offset.y!==void 0&&(this.offset.y=t.offset.y)))}}class hw{constructor(){this.close=!0,this.fill=!0,this.options={},this.type="circle"}load(t){if(Q(t))return;const r=t.options;if(r!==void 0)for(const s in r){const l=r[s];l&&(this.options[s]=_t(this.options[s]??{},l))}t.close!==void 0&&(this.close=t.close),t.fill!==void 0&&(this.fill=t.fill),t.type!==void 0&&(this.type=t.type)}}class fw extends Ud{constructor(){super(),this.destroy=to.none,this.speed=5}load(t){super.load(t),!Q(t)&&t.destroy!==void 0&&(this.destroy=t.destroy)}}class gw extends eS{constructor(){super(),this.animation=new fw,this.value=3}load(t){if(super.load(t),Q(t))return;const r=t.animation;r!==void 0&&this.animation.load(r)}}class Tm{constructor(){this.width=0}load(t){Q(t)||(t.color!==void 0&&(this.color=Rs.create(this.color,t.color)),t.width!==void 0&&(this.width=ye(t.width)),t.opacity!==void 0&&(this.opacity=ye(t.opacity)))}}class mw extends oo{constructor(){super(),this.opacityRate=1,this.sizeRate=1,this.velocityRate=1}load(t){super.load(t),!Q(t)&&(t.opacityRate!==void 0&&(this.opacityRate=t.opacityRate),t.sizeRate!==void 0&&(this.sizeRate=t.sizeRate),t.velocityRate!==void 0&&(this.velocityRate=t.velocityRate))}}class Ew{constructor(t,r){this._engine=t,this._container=r,this.bounce=new tS,this.collisions=new YC,this.color=new Rs,this.color.value="#fff",this.effect=new KC,this.groups={},this.move=new sw,this.number=new dw,this.opacity=new lw,this.reduceDuplicates=!1,this.shadow=new pw,this.shape=new hw,this.size=new gw,this.stroke=new Tm,this.zIndex=new mw}load(t){if(Q(t))return;if(t.groups!==void 0)for(const s of Object.keys(t.groups)){if(!Object.hasOwn(t.groups,s))continue;const l=t.groups[s];l!==void 0&&(this.groups[s]=_t(this.groups[s]??{},l))}t.reduceDuplicates!==void 0&&(this.reduceDuplicates=t.reduceDuplicates),this.bounce.load(t.bounce),this.color.load(Rs.create(this.color,t.color)),this.effect.load(t.effect),this.move.load(t.move),this.number.load(t.number),this.opacity.load(t.opacity),this.shape.load(t.shape),this.size.load(t.size),this.shadow.load(t.shadow),this.zIndex.load(t.zIndex),this.collisions.load(t.collisions),t.interactivity!==void 0&&(this.interactivity=_t({},t.interactivity));const r=t.stroke;if(r&&(this.stroke=Jt(r,s=>{const l=new Tm;return l.load(s),l})),this._container){const s=this._engine.updaters.get(this._container);if(s)for(const u of s)u.loadOptions&&u.loadOptions(this,t);const l=this._engine.interactors.get(this._container);if(l)for(const u of l)u.loadParticlesOptions&&u.loadParticlesOptions(this,t)}}}function nS(i,...t){for(const r of t)i.load(r)}function iS(i,t,...r){const s=new Ew(i,t);return nS(s,...r),s}class Sw{constructor(t,r){this._findDefaultTheme=s=>this.themes.find(l=>l.default.value&&l.default.mode===s)??this.themes.find(l=>l.default.value&&l.default.mode===gi.any),this._importPreset=s=>{this.load(this._engine.getPreset(s))},this._engine=t,this._container=r,this.autoPlay=!0,this.background=new NC,this.backgroundMask=new FC,this.clear=!0,this.defaultThemes={},this.delay=0,this.fullScreen=new DC,this.detectRetina=!0,this.duration=0,this.fpsLimit=120,this.interactivity=new $E(t,r),this.manualParticles=[],this.particles=iS(this._engine,this._container),this.pauseOnBlur=!0,this.pauseOnOutsideViewport=!0,this.responsive=[],this.smooth=!1,this.style={},this.themes=[],this.zLayers=100}load(t){var p,h;if(Q(t))return;t.preset!==void 0&&Jt(t.preset,f=>this._importPreset(f)),t.autoPlay!==void 0&&(this.autoPlay=t.autoPlay),t.clear!==void 0&&(this.clear=t.clear),t.key!==void 0&&(this.key=t.key),t.name!==void 0&&(this.name=t.name),t.delay!==void 0&&(this.delay=ye(t.delay));const r=t.detectRetina;r!==void 0&&(this.detectRetina=r),t.duration!==void 0&&(this.duration=ye(t.duration));const s=t.fpsLimit;s!==void 0&&(this.fpsLimit=s),t.pauseOnBlur!==void 0&&(this.pauseOnBlur=t.pauseOnBlur),t.pauseOnOutsideViewport!==void 0&&(this.pauseOnOutsideViewport=t.pauseOnOutsideViewport),t.zLayers!==void 0&&(this.zLayers=t.zLayers),this.background.load(t.background);const l=t.fullScreen;BE(l)?this.fullScreen.enable=l:this.fullScreen.load(l),this.backgroundMask.load(t.backgroundMask),this.interactivity.load(t.interactivity),t.manualParticles&&(this.manualParticles=t.manualParticles.map(f=>{const g=new UC;return g.load(f),g})),this.particles.load(t.particles),this.style=_t(this.style,t.style),this._engine.loadOptions(this,t),t.smooth!==void 0&&(this.smooth=t.smooth);const u=this._engine.interactors.get(this._container);if(u)for(const f of u)f.loadOptions&&f.loadOptions(this,t);if(t.responsive!==void 0)for(const f of t.responsive){const g=new WC;g.load(f),this.responsive.push(g)}if(this.responsive.sort((f,g)=>f.maxWidth-g.maxWidth),t.themes!==void 0)for(const f of t.themes){const g=this.themes.find(E=>E.name===f.name);if(g)g.load(f);else{const E=new kC;E.load(f),this.themes.push(E)}}this.defaultThemes.dark=(p=this._findDefaultTheme(gi.dark))==null?void 0:p.name,this.defaultThemes.light=(h=this._findDefaultTheme(gi.light))==null?void 0:h.name}setResponsive(t,r,s){this.load(s);const l=this.responsive.find(u=>u.mode===Ji.screen&&screen?u.maxWidth>screen.availWidth:u.maxWidth*r>t);return this.load(l==null?void 0:l.options),l==null?void 0:l.maxWidth}setTheme(t){if(t){const r=this.themes.find(s=>s.name===t);r&&this.load(r.options)}else{const r=UE("(prefers-color-scheme: dark)"),s=r==null?void 0:r.matches,l=this._findDefaultTheme(s?gi.dark:gi.light);l&&this.load(l.options)}}}var ar;(function(i){i.external="external",i.particles="particles"})(ar||(ar={}));class Rw{constructor(t,r){this.container=r,this._engine=t,this._interactors=[],this._externalInteractors=[],this._particleInteractors=[]}externalInteract(t){for(const r of this._externalInteractors)r.isEnabled()&&r.interact(t)}handleClickMode(t){var r;for(const s of this._externalInteractors)(r=s.handleClickMode)==null||r.call(s,t)}async init(){this._interactors=await this._engine.getInteractors(this.container,!0),this._externalInteractors=[],this._particleInteractors=[];for(const t of this._interactors){switch(t.type){case ar.external:this._externalInteractors.push(t);break;case ar.particles:this._particleInteractors.push(t);break}t.init()}}particlesInteract(t,r){for(const s of this._externalInteractors)s.clear(t,r);for(const s of this._particleInteractors)s.isEnabled(t)&&s.interact(t,r)}reset(t){for(const r of this._externalInteractors)r.isEnabled()&&r.reset(t);for(const r of this._particleInteractors)r.isEnabled(t)&&r.reset(t)}}var an;(function(i){i.normal="normal",i.inside="inside",i.outside="outside"})(an||(an={}));function Iw(i,t,r,s){const l=t.options[i];if(l)return _t({close:t.close,fill:t.fill},rn(l,r,s))}function yw(i,t,r,s){const l=t.options[i];if(l)return _t({close:t.close,fill:t.fill},rn(l,r,s))}function vm(i){if(!je(i.outMode,i.checkModes))return;const t=i.radius*pt;i.coord>i.maxCoord-t?i.setCb(-i.radius):i.coord<t&&i.setCb(i.radius)}class bw{constructor(t,r){this.container=r,this._calcPosition=(s,l,u,p=sm)=>{for(const b of s.plugins.values()){const X=b.particlePosition!==void 0?b.particlePosition(l,this):void 0;if(X)return At.create(X.x,X.y,u)}const h=s.canvas.size,f=eC({size:h,position:l}),g=At.create(f.x,f.y,u),E=this.getRadius(),R=this.options.move.outModes,y=b=>{vm({outMode:b,checkModes:[Qe.bounce],coord:g.x,maxCoord:s.canvas.size.width,setCb:X=>g.x+=X,radius:E})},w=b=>{vm({outMode:b,checkModes:[Qe.bounce],coord:g.y,maxCoord:s.canvas.size.height,setCb:X=>g.y+=X,radius:E})};return y(R.left??R.default),y(R.right??R.default),w(R.top??R.default),w(R.bottom??R.default),this._checkOverlap(g,p)?this._calcPosition(s,void 0,u,p+Nb):g},this._calculateVelocity=()=>{const s=$b(this.direction),l=s.copy(),u=this.options.move;if(u.direction===rt.inside||u.direction===rt.outside)return l;const p=Si(ie(u.angle.value)),h=Si(ie(u.angle.offset)),f={left:h-p*Ke,right:h+p*Ke};return u.straight||(l.angle+=Bt(ye(f.left,f.right))),u.random&&typeof u.speed=="number"&&(l.length*=Oe()),l},this._checkOverlap=(s,l=sm)=>{const u=this.options.collisions,p=this.getRadius();if(!u.enable)return!1;const h=u.overlap;if(h.enable)return!1;const f=h.retries;if(f>=Ob&&l>f)throw new Error(`${Kt} particle is overlapping and can't be placed`);return!!this.container.particles.find(g=>Mt(s,g.position)<p+g.getRadius())},this._getRollColor=s=>{if(!s||!this.roll||!this.backColor&&!this.roll.alter)return s;const l=this.roll.horizontal&&this.roll.vertical?pt*fm:fm,u=this.roll.horizontal?Math.PI*Ke:ul;return Math.floor(((this.roll.angle??ul)+u)/(Math.PI/l))%pt?this.backColor?this.backColor:this.roll.alter?vC(s,this.roll.alter.type,this.roll.alter.value):s:s},this._initPosition=s=>{const l=this.container,u=ie(this.options.zIndex.value);this.position=this._calcPosition(l,s,Zt(u,Fb,l.zLayers)),this.initialPosition=this.position.copy();const p=l.canvas.size;switch(this.moveCenter={...GE(this.options.move.center,p),radius:this.options.move.center.radius??Db,mode:this.options.move.center.mode??or.percent},this.direction=Jb(this.options.move.direction,this.position,this.moveCenter),this.options.move.direction){case rt.inside:this.outType=an.inside;break;case rt.outside:this.outType=an.outside;break}this.offset=tt.origin},this._engine=t}destroy(t){var u,p,h;if(this.unbreakable||this.destroyed)return;this.destroyed=!0,this.bubble.inRange=!1,this.slow.inRange=!1;const r=this.container,s=this.pathGenerator,l=r.shapeDrawers.get(this.shape);(u=l==null?void 0:l.particleDestroy)==null||u.call(l,this);for(const f of r.plugins.values())(p=f.particleDestroyed)==null||p.call(f,this,t);for(const f of r.particles.updaters)(h=f.particleDestroyed)==null||h.call(f,this,t);s==null||s.reset(this),this._engine.dispatchEvent(mt.particleDestroyed,{container:this.container,data:{particle:this}})}draw(t){const r=this.container,s=r.canvas;for(const l of r.plugins.values())s.drawParticlePlugin(l,this,t);s.drawParticle(this,t)}getFillColor(){return this._getRollColor(this.bubble.color??bm(this.color))}getMass(){return this.getRadius()**Mn*Math.PI*Ke}getPosition(){return{x:this.position.x+this.offset.x,y:this.position.y+this.offset.y,z:this.position.z}}getRadius(){return this.bubble.radius??this.size.value}getStrokeColor(){return this._getRollColor(this.bubble.color??bm(this.strokeColor))}init(t,r,s,l){var Z,ce,re,te,Xe,_e;const u=this.container,p=this._engine;this.id=t,this.group=l,this.effectClose=!0,this.effectFill=!0,this.shapeClose=!0,this.shapeFill=!0,this.pathRotation=!1,this.lastPathTime=0,this.destroyed=!1,this.unbreakable=!1,this.isRotating=!1,this.rotation=0,this.misplaced=!1,this.retina={maxDistance:{}},this.outType=an.normal,this.ignoresResizeRatio=!0;const h=u.retina.pixelRatio,f=u.actualOptions,g=iS(this._engine,u,f.particles),{reduceDuplicates:E}=g,R=g.effect.type,y=g.shape.type;this.effect=rn(R,this.id,E),this.shape=rn(y,this.id,E);const w=g.effect,b=g.shape;if(s){if((Z=s.effect)!=null&&Z.type){const Ee=s.effect.type,qe=rn(Ee,this.id,E);qe&&(this.effect=qe,w.load(s.effect))}if((ce=s.shape)!=null&&ce.type){const Ee=s.shape.type,qe=rn(Ee,this.id,E);qe&&(this.shape=qe,b.load(s.shape))}}if(this.effect===ir){const Ee=[...this.container.effectDrawers.keys()];this.effect=Ee[Math.floor(Math.random()*Ee.length)]}if(this.shape===ir){const Ee=[...this.container.shapeDrawers.keys()];this.shape=Ee[Math.floor(Math.random()*Ee.length)]}this.effectData=Iw(this.effect,w,this.id,E),this.shapeData=yw(this.shape,b,this.id,E),g.load(s);const X=this.effectData;X&&g.load(X.particles);const N=this.shapeData;N&&g.load(N.particles);const v=new $E(p,u);v.load(u.actualOptions.interactivity),v.load(g.interactivity),this.interactivity=v,this.effectFill=(X==null?void 0:X.fill)??g.effect.fill,this.effectClose=(X==null?void 0:X.close)??g.effect.close,this.shapeFill=(N==null?void 0:N.fill)??g.shape.fill,this.shapeClose=(N==null?void 0:N.close)??g.shape.close,this.options=g;const A=this.options.move.path;this.pathDelay=ie(A.delay.value)*dt,A.generator&&(this.pathGenerator=this._engine.getPathGenerator(A.generator),this.pathGenerator&&u.addPath(A.generator,this.pathGenerator)&&this.pathGenerator.init(u)),u.retina.initParticle(this),this.size=zE(this.options.size,h),this.bubble={inRange:!1},this.slow={inRange:!1,factor:1},this._initPosition(r),this.initialVelocity=this._calculateVelocity(),this.velocity=this.initialVelocity.copy(),this.moveDecay=HE-ie(this.options.move.decay);const k=u.particles;k.setLastZIndex(this.position.z),this.zIndexFactor=this.position.z/u.zLayers,this.sides=24;let U=u.effectDrawers.get(this.effect);U||(U=this._engine.getEffectDrawer(this.effect),U&&u.effectDrawers.set(this.effect,U)),U!=null&&U.loadEffect&&U.loadEffect(this);let j=u.shapeDrawers.get(this.shape);j||(j=this._engine.getShapeDrawer(this.shape),j&&u.shapeDrawers.set(this.shape,j)),j!=null&&j.loadShape&&j.loadShape(this);const K=j==null?void 0:j.getSidesCount;K&&(this.sides=K(this)),this.spawning=!1,this.shadowColor=sn(this._engine,this.options.shadow.color);for(const Ee of k.updaters)Ee.init(this);for(const Ee of k.movers)(re=Ee.init)==null||re.call(Ee,this);(te=U==null?void 0:U.particleInit)==null||te.call(U,u,this),(Xe=j==null?void 0:j.particleInit)==null||Xe.call(j,u,this);for(const Ee of u.plugins.values())(_e=Ee.particleCreated)==null||_e.call(Ee,this)}isInsideCanvas(){const t=this.getRadius(),r=this.container.canvas.size,s=this.position;return s.x>=-t&&s.y>=-t&&s.y<=r.height+t&&s.x<=r.width+t}isVisible(){return!this.destroyed&&!this.spawning&&this.isInsideCanvas()}reset(){var t;for(const r of this.container.particles.updaters)(t=r.reset)==null||t.call(r,this)}}class Cw{constructor(t,r){this.position=t,this.particle=r}}var tr;(function(i){i.circle="circle",i.rectangle="rectangle"})(tr||(tr={}));class oS{constructor(t,r,s){this.position={x:t,y:r},this.type=s}}class ht extends oS{constructor(t,r,s){super(t,r,tr.circle),this.radius=s}contains(t){return Mt(t,this.position)<=this.radius}intersects(t){const r=this.position,s=t.position,l={x:Math.abs(s.x-r.x),y:Math.abs(s.y-r.y)},u=this.radius;if(t instanceof ht||t.type===tr.circle){const p=t,h=u+p.radius,f=Math.sqrt(l.x**Mn+l.y**Mn);return h>f}else if(t instanceof ln||t.type===tr.rectangle){const p=t,{width:h,height:f}=p.size;return Math.pow(l.x-h,Mn)+Math.pow(l.y-f,Mn)<=u**Mn||l.x<=u+h&&l.y<=u+f||l.x<=h||l.y<=f}return!1}}class ln extends oS{constructor(t,r,s,l){super(t,r,tr.rectangle),this.size={height:l,width:s}}contains(t){const r=this.size.width,s=this.size.height,l=this.position;return t.x>=l.x&&t.x<=l.x+r&&t.y>=l.y&&t.y<=l.y+s}intersects(t){if(t instanceof ht)return t.intersects(this);const r=this.size.width,s=this.size.height,l=this.position,u=t.position,p=t instanceof ln?t.size:{width:0,height:0},h=p.width,f=p.height;return u.x<l.x+r&&u.x+h>l.x&&u.y<l.y+s&&u.y+f>l.y}}class pl{constructor(t,r){this.rectangle=t,this.capacity=r,this._subdivide=()=>{const{x:s,y:l}=this.rectangle.position,{width:u,height:p}=this.rectangle.size,{capacity:h}=this;for(let f=0;f<mb;f++){const g=f%pt;this._subs.push(new pl(new ln(s+u*Ke*g,l+p*Ke*(Math.round(f*Ke)-g),u*Ke,p*Ke),h))}this._divided=!0},this._points=[],this._divided=!1,this._subs=[]}insert(t){return this.rectangle.contains(t.position)?this._points.length<this.capacity?(this._points.push(t),!0):(this._divided||this._subdivide(),this._subs.some(r=>r.insert(t))):!1}query(t,r){const s=[];if(!t.intersects(this.rectangle))return[];for(const l of this._points)!t.contains(l.position)&&Mt(t.position,l.position)>l.particle.getRadius()&&(!r||r(l.particle))||s.push(l.particle);if(this._divided)for(const l of this._subs)s.push(...l.query(t,r));return s}queryCircle(t,r,s){return this.query(new ht(t.x,t.y,r),s)}queryRectangle(t,r,s){return this.query(new ln(t.x,t.y,r.width,r.height),s)}}const xm=i=>{const{height:t,width:r}=i;return new ln(gm*r,gm*t,mm*r,mm*t)};class ww{constructor(t,r){this._addToPool=(...l)=>{this._pool.push(...l)},this._applyDensity=(l,u,p)=>{var w;const h=l.number;if(!((w=l.number.density)!=null&&w.enable)){p===void 0?this._limit=h.limit.value:h.limit&&this._groupLimits.set(p,h.limit.value);return}const f=this._initDensityFactor(h.density),g=h.value,E=h.limit.value>Em?h.limit.value:g,R=Math.min(g,E)*f+u,y=Math.min(this.count,this.filter(b=>b.group===p).length);p===void 0?this._limit=h.limit.value*f:this._groupLimits.set(p,h.limit.value*f),y<R?this.push(Math.abs(R-y),void 0,l,p):y>R&&this.removeQuantity(y-R,p)},this._initDensityFactor=l=>{const u=this._container;if(!u.canvas.element||!l.enable)return Bb;const p=u.canvas.element,h=u.retina.pixelRatio;return p.width*p.height/(l.height*l.width*h**Mn)},this._pushParticle=(l,u,p,h)=>{try{let f=this._pool.pop();f||(f=new bw(this._engine,this._container)),f.init(this._nextId,l,u,p);let g=!0;return h&&(g=h(f)),g?(this._array.push(f),this._zArray.push(f),this._nextId++,this._engine.dispatchEvent(mt.particleAdded,{container:this._container,data:{particle:f}}),f):void 0}catch(f){no().warning(`${Kt} adding particle: ${f}`)}},this._removeParticle=(l,u,p)=>{const h=this._array[l];if(!h||h.group!==u)return!1;const f=this._zArray.indexOf(h);return this._array.splice(l,dl),this._zArray.splice(f,dl),h.destroy(p),this._engine.dispatchEvent(mt.particleRemoved,{container:this._container,data:{particle:h}}),this._addToPool(h),!0},this._engine=t,this._container=r,this._nextId=0,this._array=[],this._zArray=[],this._pool=[],this._limit=0,this._groupLimits=new Map,this._needsSort=!1,this._lastZIndex=0,this._interactionManager=new Rw(t,r),this._pluginsInitialized=!1;const s=r.canvas.size;this.quadTree=new pl(xm(s),am),this.movers=[],this.updaters=[]}get count(){return this._array.length}addManualParticles(){const t=this._container;t.actualOptions.manualParticles.forEach(s=>this.addParticle(s.position?GE(s.position,t.canvas.size):void 0,s.options))}addParticle(t,r,s,l){const u=this._container.actualOptions.particles.number.limit.mode,p=s===void 0?this._limit:this._groupLimits.get(s)??this._limit,h=this.count;if(p>Em)switch(u){case Is.delete:{const f=h+Hb-p;f>Ab&&this.removeQuantity(f);break}case Is.wait:if(h>=p)return;break}return this._pushParticle(t,r,s,l)}clear(){this._array=[],this._zArray=[],this._pluginsInitialized=!1}destroy(){this._array=[],this._zArray=[],this.movers=[],this.updaters=[]}draw(t){const r=this._container,s=r.canvas;s.clear(),this.update(t);for(const l of r.plugins.values())s.drawPlugin(l,t);for(const l of this._zArray)l.draw(t)}filter(t){return this._array.filter(t)}find(t){return this._array.find(t)}get(t){return this._array[t]}handleClickMode(t){this._interactionManager.handleClickMode(t)}async init(){var l,u;const t=this._container,r=t.actualOptions;this._lastZIndex=0,this._needsSort=!1,await this.initPlugins();let s=!1;for(const p of t.plugins.values())if(s=((l=p.particlesInitialization)==null?void 0:l.call(p))??s,s)break;if(this.addManualParticles(),!s){const p=r.particles,h=p.groups;for(const f in h){const g=h[f];for(let E=this.count,R=0;R<((u=g.number)==null?void 0:u.value)&&E<p.number.value;E++,R++)this.addParticle(void 0,g,f)}for(let f=this.count;f<p.number.value;f++)this.addParticle()}}async initPlugins(){if(this._pluginsInitialized)return;const t=this._container;this.movers=await this._engine.getMovers(t,!0),this.updaters=await this._engine.getUpdaters(t,!0),await this._interactionManager.init();for(const r of t.pathGenerators.values())r.init(t)}push(t,r,s,l){for(let u=0;u<t;u++)this.addParticle(r==null?void 0:r.position,s,l)}async redraw(){this.clear(),await this.init(),this.draw({value:0,factor:0})}remove(t,r,s){this.removeAt(this._array.indexOf(t),void 0,r,s)}removeAt(t,r=gb,s,l){if(t<Sd||t>this.count)return;let u=0;for(let p=t;u<r&&p<this.count;p++)this._removeParticle(p,s,l)&&(p--,u++)}removeQuantity(t,r){this.removeAt(Sd,t,r)}setDensity(){const t=this._container.actualOptions,r=t.particles.groups;for(const s in r)this._applyDensity(r[s],Lb,s);this._applyDensity(t.particles,t.manualParticles.length)}setLastZIndex(t){this._lastZIndex=t,this._needsSort=this._needsSort||this._lastZIndex<t}setResizeFactor(t){this._resizeFactor=t}update(t){var u,p;const r=this._container,s=new Set;this.quadTree=new pl(xm(r.canvas.size),am);for(const h of r.pathGenerators.values())h.update();for(const h of r.plugins.values())(u=h.update)==null||u.call(h,t);const l=this._resizeFactor;for(const h of this._array){l&&!h.ignoresResizeRatio&&(h.position.x*=l.width,h.position.y*=l.height,h.initialPosition.x*=l.width,h.initialPosition.y*=l.height),h.ignoresResizeRatio=!1,this._interactionManager.reset(h);for(const f of this._container.plugins.values()){if(h.destroyed)break;(p=f.particleUpdate)==null||p.call(f,h,t)}for(const f of this.movers)f.isEnabled(h)&&f.move(h,t);if(h.destroyed){s.add(h);continue}this.quadTree.insert(new Cw(h.getPosition(),h))}if(s.size){const h=f=>!s.has(f);this._array=this.filter(h),this._zArray=this._zArray.filter(h);for(const f of s)this._engine.dispatchEvent(mt.particleRemoved,{container:this._container,data:{particle:f}});this._addToPool(...s)}this._interactionManager.externalInteract(t);for(const h of this._array){for(const f of this.updaters)f.update(h,t);!h.destroyed&&!h.spawning&&this._interactionManager.particlesInteract(h,t)}if(delete this._resizeFactor,this._needsSort){const h=this._zArray;h.sort((f,g)=>g.position.z-f.position.z||f.id-g.id),this._lastZIndex=h[h.length-AE].position.z,this._needsSort=!1}}}class _w{constructor(t){this.container=t,this.pixelRatio=lm,this.reduceFactor=cm}init(){const t=this.container,r=t.actualOptions;this.pixelRatio=!r.detectRetina||io()?lm:window.devicePixelRatio,this.reduceFactor=cm;const s=this.pixelRatio,l=t.canvas;if(l.element){const h=l.element;l.size.width=h.offsetWidth*s,l.size.height=h.offsetHeight*s}const u=r.particles,p=u.move;this.maxSpeed=ie(p.gravity.maxSpeed)*s,this.sizeAnimationSpeed=ie(u.size.animation.speed)*s}initParticle(t){const r=t.options,s=this.pixelRatio,l=r.move,u=l.distance,p=t.retina;p.moveDrift=ie(l.drift)*s,p.moveSpeed=ie(l.speed)*s,p.sizeAnimationSpeed=ie(r.size.animation.speed)*s;const h=p.maxDistance;h.horizontal=u.horizontal!==void 0?u.horizontal*s:void 0,h.vertical=u.vertical!==void 0?u.vertical*s:void 0,p.maxSpeed=ie(l.gravity.maxSpeed)*s}}function Me(i){return i&&!i.destroyed}function Tw(i,t=ku,r=!1){return{value:i,factor:r?ku/t:ku*i/dt}}function zo(i,t,...r){const s=new Sw(i,t);return nS(s,...r),s}class vw{constructor(t,r,s){this._intersectionManager=l=>{if(!(!Me(this)||!this.actualOptions.pauseOnOutsideViewport))for(const u of l)u.target===this.interactivity.element&&(u.isIntersecting?this.play():this.pause())},this._nextFrame=l=>{try{if(!this._smooth&&this._lastFrameTime!==void 0&&l<this._lastFrameTime+dt/this.fpsLimit){this.draw(!1);return}this._lastFrameTime??(this._lastFrameTime=l);const u=Tw(l-this._lastFrameTime,this.fpsLimit,this._smooth);if(this.addLifeTime(u.value),this._lastFrameTime=l,u.value>dt){this.draw(!1);return}if(this.particles.draw(u),!this.alive()){this.destroy();return}this.animationStatus&&this.draw(!1)}catch(u){no().error(`${Kt} in animation loop`,u)}},this._engine=t,this.id=Symbol(r),this.fpsLimit=120,this._smooth=!1,this._delay=0,this._duration=0,this._lifeTime=0,this._firstStart=!0,this.started=!1,this.destroyed=!1,this._paused=!0,this._lastFrameTime=0,this.zLayers=100,this.pageHidden=!1,this._clickHandlers=new Map,this._sourceOptions=s,this._initialSourceOptions=s,this.retina=new _w(this),this.canvas=new PC(this,this._engine),this.particles=new ww(this._engine,this),this.pathGenerators=new Map,this.interactivity={mouse:{clicking:!1,inside:!1}},this.plugins=new Map,this.effectDrawers=new Map,this.shapeDrawers=new Map,this._options=zo(this._engine,this),this.actualOptions=zo(this._engine,this),this._eventListeners=new XC(this),this._intersectionObserver=rC(l=>this._intersectionManager(l)),this._engine.dispatchEvent(mt.containerBuilt,{container:this})}get animationStatus(){return!this._paused&&!this.pageHidden&&Me(this)}get options(){return this._options}get sourceOptions(){return this._sourceOptions}addClickHandler(t){if(!Me(this))return;const r=this.interactivity.element;if(!r)return;const s=(R,y,w)=>{if(!Me(this))return;const b=this.retina.pixelRatio,X={x:y.x*b,y:y.y*b},N=this.particles.quadTree.queryCircle(X,w*b);t(R,N)},l=R=>{if(!Me(this))return;const y=R,w={x:y.offsetX||y.clientX,y:y.offsetY||y.clientY};s(R,w,Cb)},u=()=>{Me(this)&&(g=!0,E=!1)},p=()=>{Me(this)&&(E=!0)},h=R=>{if(Me(this)){if(g&&!E){const y=R;let w=y.touches[y.touches.length-hm];if(!w&&(w=y.changedTouches[y.changedTouches.length-hm],!w))return;const b=this.canvas.element,X=b?b.getBoundingClientRect():void 0,N={x:w.clientX-(X?X.left:cl),y:w.clientY-(X?X.top:cl)};s(R,N,Math.max(w.radiusX,w.radiusY))}g=!1,E=!1}},f=()=>{Me(this)&&(g=!1,E=!1)};let g=!1,E=!1;this._clickHandlers.set("click",l),this._clickHandlers.set("touchstart",u),this._clickHandlers.set("touchmove",p),this._clickHandlers.set("touchend",h),this._clickHandlers.set("touchcancel",f);for(const[R,y]of this._clickHandlers)r.addEventListener(R,y)}addLifeTime(t){this._lifeTime+=t}addPath(t,r,s=!1){return!Me(this)||!s&&this.pathGenerators.has(t)?!1:(this.pathGenerators.set(t,r),!0)}alive(){return!this._duration||this._lifeTime<=this._duration}clearClickHandlers(){var t;if(Me(this)){for(const[r,s]of this._clickHandlers)(t=this.interactivity.element)==null||t.removeEventListener(r,s);this._clickHandlers.clear()}}destroy(t=!0){var r,s;if(Me(this)){this.stop(),this.clearClickHandlers(),this.particles.destroy(),this.canvas.destroy();for(const l of this.effectDrawers.values())(r=l.destroy)==null||r.call(l,this);for(const l of this.shapeDrawers.values())(s=l.destroy)==null||s.call(l,this);for(const l of this.effectDrawers.keys())this.effectDrawers.delete(l);for(const l of this.shapeDrawers.keys())this.shapeDrawers.delete(l);if(this._engine.clearPlugins(this),this.destroyed=!0,t){const l=this._engine.items,u=l.findIndex(p=>p===this);u>=wb&&l.splice(u,DE)}this._engine.dispatchEvent(mt.containerDestroyed,{container:this})}}draw(t){if(!Me(this))return;let r=t;const s=l=>{r&&(this._lastFrameTime=void 0,r=!1),this._nextFrame(l)};this._drawAnimationFrame=Kb(l=>s(l))}async export(t,r={}){for(const s of this.plugins.values()){if(!s.export)continue;const l=await s.export(t,r);if(l.supported)return l.blob}no().error(`${Kt} - Export plugin with type ${t} not found`)}handleClickMode(t){var r;if(Me(this)){this.particles.handleClickMode(t);for(const s of this.plugins.values())(r=s.handleClickMode)==null||r.call(s,t)}}async init(){var g,E,R,y;if(!Me(this))return;const t=this._engine.getSupportedEffects();for(const w of t){const b=this._engine.getEffectDrawer(w);b&&this.effectDrawers.set(w,b)}const r=this._engine.getSupportedShapes();for(const w of r){const b=this._engine.getShapeDrawer(w);b&&this.shapeDrawers.set(w,b)}await this.particles.initPlugins(),this._options=zo(this._engine,this,this._initialSourceOptions,this.sourceOptions),this.actualOptions=zo(this._engine,this,this._options);const s=await this._engine.getAvailablePlugins(this);for(const[w,b]of s)this.plugins.set(w,b);this.retina.init(),await this.canvas.init(),this.updateActualOptions(),this.canvas.initBackground(),this.canvas.resize();const{zLayers:l,duration:u,delay:p,fpsLimit:h,smooth:f}=this.actualOptions;this.zLayers=l,this._duration=ie(u)*dt,this._delay=ie(p)*dt,this._lifeTime=0,this.fpsLimit=h>Tb?h:_b,this._smooth=f;for(const w of this.effectDrawers.values())await((g=w.init)==null?void 0:g.call(w,this));for(const w of this.shapeDrawers.values())await((E=w.init)==null?void 0:E.call(w,this));for(const w of this.plugins.values())await((R=w.init)==null?void 0:R.call(w));this._engine.dispatchEvent(mt.containerInit,{container:this}),await this.particles.init(),this.particles.setDensity();for(const w of this.plugins.values())(y=w.particlesSetup)==null||y.call(w);this._engine.dispatchEvent(mt.particlesSetup,{container:this})}async loadTheme(t){Me(this)&&(this._currentTheme=t,await this.refresh())}pause(){var t;if(Me(this)&&(this._drawAnimationFrame!==void 0&&(Zb(this._drawAnimationFrame),delete this._drawAnimationFrame),!this._paused)){for(const r of this.plugins.values())(t=r.pause)==null||t.call(r);this.pageHidden||(this._paused=!0),this._engine.dispatchEvent(mt.containerPaused,{container:this})}}play(t){if(!Me(this))return;const r=this._paused||t;if(this._firstStart&&!this.actualOptions.autoPlay){this._firstStart=!1;return}if(this._paused&&(this._paused=!1),r)for(const s of this.plugins.values())s.play&&s.play();this._engine.dispatchEvent(mt.containerPlay,{container:this}),this.draw(r??!1)}async refresh(){if(Me(this))return this.stop(),this.start()}async reset(t){if(Me(this))return this._initialSourceOptions=t,this._sourceOptions=t,this._options=zo(this._engine,this,this._initialSourceOptions,this.sourceOptions),this.actualOptions=zo(this._engine,this,this._options),this.refresh()}async start(){!Me(this)||this.started||(await this.init(),this.started=!0,await new Promise(t=>{const r=async()=>{var s;this._eventListeners.addListeners(),this.interactivity.element instanceof HTMLElement&&this._intersectionObserver&&this._intersectionObserver.observe(this.interactivity.element);for(const l of this.plugins.values())await((s=l.start)==null?void 0:s.call(l));this._engine.dispatchEvent(mt.containerStarted,{container:this}),this.play(),t()};this._delayTimeout=setTimeout(()=>void r(),this._delay)}))}stop(){var t;if(!(!Me(this)||!this.started)){this._delayTimeout&&(clearTimeout(this._delayTimeout),delete this._delayTimeout),this._firstStart=!0,this.started=!1,this._eventListeners.removeListeners(),this.pause(),this.particles.clear(),this.canvas.stop(),this.interactivity.element instanceof HTMLElement&&this._intersectionObserver&&this._intersectionObserver.unobserve(this.interactivity.element);for(const r of this.plugins.values())(t=r.stop)==null||t.call(r);for(const r of this.plugins.keys())this.plugins.delete(r);this._sourceOptions=this._options,this._engine.dispatchEvent(mt.containerStopped,{container:this})}}updateActualOptions(){this.actualOptions.responsive=[];const t=this.actualOptions.setResponsive(this.canvas.size.width,this.retina.pixelRatio,this._options);return this.actualOptions.setTheme(this._currentTheme),this._responsiveMaxWidth===t?!1:(this._responsiveMaxWidth=t,!0)}}class xw{constructor(){this._listeners=new Map}addEventListener(t,r){this.removeEventListener(t,r);let s=this._listeners.get(t);s||(s=[],this._listeners.set(t,s)),s.push(r)}dispatchEvent(t,r){const s=this._listeners.get(t);s==null||s.forEach(l=>l(r))}hasEventListener(t){return!!this._listeners.get(t)}removeAllEventListeners(t){t?this._listeners.delete(t):this._listeners=new Map}removeEventListener(t,r){const s=this._listeners.get(t);if(!s)return;const l=s.length,u=s.indexOf(r);u<Sd||(l===dl?this._listeners.delete(t):s.splice(u,dl))}}async function $u(i,t,r,s=!1){let l=t.get(i);return(!l||s)&&(l=await Promise.all([...r.values()].map(u=>u(i))),t.set(i,l)),l}async function Pw(i){const t=rn(i.url,i.index);if(!t)return i.fallback;const r=await fetch(t);return r.ok?await r.json():(no().error(`${Kt} ${r.status} while retrieving config file`),i.fallback)}const Xw=i=>{let t;if(i instanceof HTMLCanvasElement||i.tagName.toLowerCase()===zu)t=i,t.dataset[Yi]||(t.dataset[Yi]=rm);else{const s=i.getElementsByTagName(zu);s.length?(t=s[vb],t.dataset[Yi]=rm):(t=document.createElement(zu),t.dataset[Yi]=OE,i.appendChild(t))}const r="100%";return t.style.width||(t.style.width=r),t.style.height||(t.style.height=r),t},Nw=(i,t)=>{let r=t??document.getElementById(i);return r||(r=document.createElement("div"),r.id=i,r.dataset[Yi]=OE,document.body.append(r),r)};class Ow{constructor(){this._configs=new Map,this._domArray=[],this._eventDispatcher=new xw,this._initialized=!1,this.plugins=[],this.colorManagers=new Map,this.easingFunctions=new Map,this._initializers={interactors:new Map,movers:new Map,updaters:new Map},this.interactors=new Map,this.movers=new Map,this.updaters=new Map,this.presets=new Map,this.effectDrawers=new Map,this.shapeDrawers=new Map,this.pathGenerators=new Map}get configs(){const t={};for(const[r,s]of this._configs)t[r]=s;return t}get items(){return this._domArray}get version(){return"3.8.1"}async addColorManager(t,r=!0){this.colorManagers.set(t.key,t),await this.refresh(r)}addConfig(t){const r=t.key??t.name??"default";this._configs.set(r,t),this._eventDispatcher.dispatchEvent(mt.configAdded,{data:{name:r,config:t}})}async addEasing(t,r,s=!0){this.getEasing(t)||(this.easingFunctions.set(t,r),await this.refresh(s))}async addEffect(t,r,s=!0){Jt(t,l=>{this.getEffectDrawer(l)||this.effectDrawers.set(l,r)}),await this.refresh(s)}addEventListener(t,r){this._eventDispatcher.addEventListener(t,r)}async addInteractor(t,r,s=!0){this._initializers.interactors.set(t,r),await this.refresh(s)}async addMover(t,r,s=!0){this._initializers.movers.set(t,r),await this.refresh(s)}async addParticleUpdater(t,r,s=!0){this._initializers.updaters.set(t,r),await this.refresh(s)}async addPathGenerator(t,r,s=!0){this.getPathGenerator(t)||this.pathGenerators.set(t,r),await this.refresh(s)}async addPlugin(t,r=!0){this.getPlugin(t.id)||this.plugins.push(t),await this.refresh(r)}async addPreset(t,r,s=!1,l=!0){(s||!this.getPreset(t))&&this.presets.set(t,r),await this.refresh(l)}async addShape(t,r=!0){for(const s of t.validTypes)this.getShapeDrawer(s)||this.shapeDrawers.set(s,t);await this.refresh(r)}checkVersion(t){if(this.version!==t)throw new Error(`The tsParticles version is different from the loaded plugins version. Engine version: ${this.version}. Plugin version: ${t}`)}clearPlugins(t){this.updaters.delete(t),this.movers.delete(t),this.interactors.delete(t)}dispatchEvent(t,r){this._eventDispatcher.dispatchEvent(t,r)}dom(){return this.items}domItem(t){return this.item(t)}async getAvailablePlugins(t){const r=new Map;for(const s of this.plugins)s.needsPlugin(t.actualOptions)&&r.set(s.id,await s.getPlugin(t));return r}getEasing(t){return this.easingFunctions.get(t)??(r=>r)}getEffectDrawer(t){return this.effectDrawers.get(t)}async getInteractors(t,r=!1){return $u(t,this.interactors,this._initializers.interactors,r)}async getMovers(t,r=!1){return $u(t,this.movers,this._initializers.movers,r)}getPathGenerator(t){return this.pathGenerators.get(t)}getPlugin(t){return this.plugins.find(r=>r.id===t)}getPreset(t){return this.presets.get(t)}getShapeDrawer(t){return this.shapeDrawers.get(t)}getSupportedEffects(){return this.effectDrawers.keys()}getSupportedShapes(){return this.shapeDrawers.keys()}async getUpdaters(t,r=!1){return $u(t,this.updaters,this._initializers.updaters,r)}init(){this._initialized||(this._initialized=!0)}item(t){const{items:r}=this,s=r[t];if(!s||s.destroyed){r.splice(t,DE);return}return s}async load(t){var y;const r=t.id??((y=t.element)==null?void 0:y.id)??`tsparticles${Math.floor(Oe()*xb)}`,{index:s,url:l}=t,u=l?await Pw({fallback:t.options,url:l,index:s}):t.options,p=rn(u,s),{items:h}=this,f=h.findIndex(w=>w.id.description===r),g=new vw(this,r,p);if(f>=Pb){const w=this.item(f),b=w?Xb:ul;w&&!w.destroyed&&w.destroy(!1),h.splice(f,b,g)}else h.push(g);const E=Nw(r,t.element),R=Xw(E);return g.canvas.loadCanvas(R),await g.start(),g}loadOptions(t,r){this.plugins.forEach(s=>{var l;return(l=s.loadOptions)==null?void 0:l.call(s,t,r)})}loadParticlesOptions(t,r,...s){const l=this.updaters.get(t);l&&l.forEach(u=>{var p;return(p=u.loadOptions)==null?void 0:p.call(u,r,...s)})}async refresh(t=!0){t&&await Promise.all(this.items.map(r=>r.refresh()))}removeEventListener(t,r){this._eventDispatcher.removeEventListener(t,r)}setOnClickHandler(t){const{items:r}=this;if(!r.length)throw new Error(`${Kt} can only set click handlers after calling tsParticles.load()`);r.forEach(s=>s.addClickHandler(t))}}function Fw(){const i=new Ow;return i.init(),i}class yn{constructor(t){this.type=ar.external,this.container=t}}class Wd{constructor(t){this.type=ar.particles,this.container=t}}var on;(function(i){i.clockwise="clockwise",i.counterClockwise="counter-clockwise",i.random="random"})(on||(on={}));var Pm;(function(i){i.linear="linear",i.radial="radial",i.random="random"})(Pm||(Pm={}));var $i;(function(i){i.easeInBack="ease-in-back",i.easeInCirc="ease-in-circ",i.easeInCubic="ease-in-cubic",i.easeInLinear="ease-in-linear",i.easeInQuad="ease-in-quad",i.easeInQuart="ease-in-quart",i.easeInQuint="ease-in-quint",i.easeInExpo="ease-in-expo",i.easeInSine="ease-in-sine",i.easeOutBack="ease-out-back",i.easeOutCirc="ease-out-circ",i.easeOutCubic="ease-out-cubic",i.easeOutLinear="ease-out-linear",i.easeOutQuad="ease-out-quad",i.easeOutQuart="ease-out-quart",i.easeOutQuint="ease-out-quint",i.easeOutExpo="ease-out-expo",i.easeOutSine="ease-out-sine",i.easeInOutBack="ease-in-out-back",i.easeInOutCirc="ease-in-out-circ",i.easeInOutCubic="ease-in-out-cubic",i.easeInOutLinear="ease-in-out-linear",i.easeInOutQuad="ease-in-out-quad",i.easeInOutQuart="ease-in-out-quart",i.easeInOutQuint="ease-in-out-quint",i.easeInOutExpo="ease-in-out-expo",i.easeInOutSine="ease-in-out-sine"})($i||($i={}));const jd=Fw();io()||(window.tsParticles=jd);const Dw=i=>{const t=i.id??"tsparticles";return P.useEffect(()=>{let r;return jd.load({id:t,url:i.url,options:i.options}).then(s=>{var l;r=s,(l=i.particlesLoaded)==null||l.call(i,s)}),()=>{r==null||r.destroy()}},[t,i,i.url,i.options]),I.jsx("div",{id:t,className:i.className})};async function Hw(i){await i(jd)}const Cd=.5,Aw=2,Ei=0,lr=1,Xm=60,Nm=0,Lw=.01,Bw=Math.PI*Aw;function Mw(i){const t=i.initialPosition,{dx:r,dy:s}=Et(t,i.position),l=Math.abs(r),u=Math.abs(s),{maxDistance:p}=i.retina,h=p.horizontal,f=p.vertical;if(!h&&!f)return;const g=(h&&l>=h)??!1,E=(f&&u>=f)??!1;if((g||E)&&!i.misplaced)i.misplaced=!!h&&l>h||!!f&&u>f,h&&(i.velocity.x=i.velocity.y*Cd-i.velocity.x),f&&(i.velocity.y=i.velocity.x*Cd-i.velocity.y);else if((!h||l<h)&&(!f||u<f)&&i.misplaced)i.misplaced=!1;else if(i.misplaced){const R=i.position,y=i.velocity;h&&(R.x<t.x&&y.x<Ei||R.x>t.x&&y.x>Ei)&&(y.x*=-Oe()),f&&(R.y<t.y&&y.y<Ei||R.y>t.y&&y.y>Ei)&&(y.y*=-Oe())}}function qw(i,t,r,s,l,u){Ww(i,u);const p=i.gravity,h=p!=null&&p.enable&&p.inverse?-1:lr;l&&r&&(i.velocity.x+=l*u.factor/(Xm*r)),p!=null&&p.enable&&r&&(i.velocity.y+=h*(p.acceleration*u.factor)/(Xm*r));const f=i.moveDecay;i.velocity.multTo(f);const g=i.velocity.mult(r);p!=null&&p.enable&&s>Ei&&(!p.inverse&&g.y>=Ei&&g.y>=s||p.inverse&&g.y<=Ei&&g.y<=-s)&&(g.y=h*s,r&&(i.velocity.y=g.y/r));const E=i.options.zIndex,R=(lr-i.zIndexFactor)**E.velocityRate;g.multTo(R);const{position:y}=i;y.addTo(g),t.vibrate&&(y.x+=Math.sin(y.x*Math.cos(y.y)),y.y+=Math.cos(y.y*Math.sin(y.x)))}function Uw(i,t){const r=i.container;if(!i.spin)return;const s=i.spin.direction===on.clockwise,l={x:s?Math.cos:Math.sin,y:s?Math.sin:Math.cos};i.position.x=i.spin.center.x+i.spin.radius*l.x(i.spin.angle),i.position.y=i.spin.center.y+i.spin.radius*l.y(i.spin.angle),i.spin.radius+=i.spin.acceleration;const u=Math.max(r.canvas.size.width,r.canvas.size.height),p=u*Cd;i.spin.radius>p?(i.spin.radius=p,i.spin.acceleration*=-1):i.spin.radius<Nm&&(i.spin.radius=Nm,i.spin.acceleration*=-1),i.spin.angle+=t*Lw*(lr-i.spin.radius/u)}function Ww(i,t){var p;const r=i.options,s=r.move.path;if(!s.enable)return;if(i.lastPathTime<=i.pathDelay){i.lastPathTime+=t.value;return}const u=(p=i.pathGenerator)==null?void 0:p.generate(i,t);u&&i.velocity.addTo(u),s.clamp&&(i.velocity.x=Zt(i.velocity.x,-1,lr),i.velocity.y=Zt(i.velocity.y,-1,lr)),i.lastPathTime-=i.pathDelay}function jw(i){return i.slow.inRange?i.slow.factor:lr}function kw(i){const t=i.container,r=i.options,s=r.move.spin;if(!s.enable)return;const l=s.position??{x:50,y:50},u=.01,p={x:l.x*u*t.canvas.size.width,y:l.y*u*t.canvas.size.height},h=i.getPosition(),f=Mt(h,p),g=ie(s.acceleration);i.retina.spinAcceleration=g*t.retina.pixelRatio,i.spin={center:p,direction:i.velocity.x>=Ei?on.clockwise:on.counterClockwise,angle:Oe()*Bw,radius:f,acceleration:i.retina.spinAcceleration}}const zw=2,Gw=1,Qw=1;class Vw{init(t){const r=t.options,s=r.move.gravity;t.gravity={enable:s.enable,acceleration:ie(s.acceleration),inverse:s.inverse},kw(t)}isEnabled(t){return!t.destroyed&&t.options.move.enable}move(t,r){var X,N;const s=t.options,l=s.move;if(!l.enable)return;const u=t.container,p=u.retina.pixelRatio;(X=t.retina).moveSpeed??(X.moveSpeed=ie(l.speed)*p),(N=t.retina).moveDrift??(N.moveDrift=ie(t.options.move.drift)*p);const h=jw(t),f=t.retina.moveSpeed*u.retina.reduceFactor,g=t.retina.moveDrift,E=mn(s.size.value)*p,R=l.size?t.getRadius()/E:Gw,y=r.factor||Qw,w=f*R*h*y/zw,b=t.retina.maxSpeed??u.retina.maxSpeed;l.spin.enable?Uw(t,w):qw(t,l,w,b,g,r),Mw(t)}}async function Yw(i,t=!0){i.checkVersion("3.8.1"),await i.addMover("base",()=>Promise.resolve(new Vw),t)}const Kw=2,Zw=Math.PI*Kw,Jw=0,Om={x:0,y:0};function $w(i){const{context:t,particle:r,radius:s}=i;r.circleRange||(r.circleRange={min:Jw,max:Zw});const l=r.circleRange;t.arc(Om.x,Om.y,s,l.min,l.max,!1)}const e_=12,t_=360,Fm=0;class n_{constructor(){this.validTypes=["circle"]}draw(t){$w(t)}getSidesCount(){return e_}particleInit(t,r){const s=r.shapeData,l=(s==null?void 0:s.angle)??{max:t_,min:Fm};r.circleRange=Vo(l)?{min:Si(l.min),max:Si(l.max)}:{min:Fm,max:Si(l)}}}async function i_(i,t=!0){i.checkVersion("3.8.1"),await i.addShape(new n_,t)}class o_{constructor(t,r){this._container=t,this._engine=r}init(t){const r=Es(this._engine,t.options.color,t.id,t.options.reduceDuplicates);r&&(t.color=ZE(r,t.options.color.animation,this._container.retina.reduceFactor))}isEnabled(t){const{h:r,s,l}=t.options.color.animation,{color:u}=t;return!t.destroyed&&!t.spawning&&((u==null?void 0:u.h.value)!==void 0&&r.enable||(u==null?void 0:u.s.value)!==void 0&&s.enable||(u==null?void 0:u.l.value)!==void 0&&l.enable)}update(t,r){JE(t.color,r)}}async function r_(i,t=!0){i.checkVersion("3.8.1"),await i.addParticleUpdater("color",r=>Promise.resolve(new o_(r,i)),t)}var Qi;(function(i){i[i.r=1]="r",i[i.g=2]="g",i[i.b=3]="b",i[i.a=4]="a"})(Qi||(Qi={}));const s_=/^#?([a-f\d])([a-f\d])([a-f\d])([a-f\d])?$/i,a_=/^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})?$/i,ka=16,l_=1,c_=255;class u_{constructor(){this.key="hex",this.stringPrefix="#"}handleColor(t){return this._parseString(t.value)}handleRangeColor(t){return this._parseString(t.value)}parseString(t){return this._parseString(t)}_parseString(t){if(typeof t!="string"||!(t!=null&&t.startsWith(this.stringPrefix)))return;const r=t.replace(s_,(l,u,p,h,f)=>u+u+p+p+h+h+(f!==void 0?f+f:"")),s=a_.exec(r);return s?{a:s[Qi.a]!==void 0?parseInt(s[Qi.a],ka)/c_:l_,b:parseInt(s[Qi.b],ka),g:parseInt(s[Qi.g],ka),r:parseInt(s[Qi.r],ka)}:void 0}}async function d_(i,t=!0){i.checkVersion("3.8.1"),await i.addColorManager(new u_,t)}var Yo;(function(i){i[i.h=1]="h",i[i.s=2]="s",i[i.l=3]="l",i[i.a=5]="a"})(Yo||(Yo={}));class p_{constructor(){this.key="hsl",this.stringPrefix="hsl"}handleColor(t){const r=t.value,s=r.hsl??t.value;if(s.h!==void 0&&s.s!==void 0&&s.l!==void 0)return rr(s)}handleRangeColor(t){const r=t.value,s=r.hsl??t.value;if(s.h!==void 0&&s.l!==void 0)return rr({h:ie(s.h),l:ie(s.l),s:ie(s.s)})}parseString(t){if(!t.startsWith("hsl"))return;const r=/hsla?\(\s*(\d+)\s*[\s,]\s*(\d+)%\s*[\s,]\s*(\d+)%\s*([\s,]\s*(0|1|0?\.\d+|(\d{1,3})%)\s*)?\)/i,s=r.exec(t),l=4,u=1,p=10;return s?SC({a:s.length>l?qE(s[Yo.a]):u,h:parseInt(s[Yo.h],p),l:parseInt(s[Yo.l],p),s:parseInt(s[Yo.s],p)}):void 0}}async function h_(i,t=!0){i.checkVersion("3.8.1"),await i.addColorManager(new p_,t)}class f_{constructor(t){this.container=t}init(t){const r=t.options.opacity,s=1;t.opacity=zE(r,s);const l=r.animation;l.enable&&(t.opacity.velocity=ie(l.speed)/mi*this.container.retina.reduceFactor,l.sync||(t.opacity.velocity*=Oe()))}isEnabled(t){return!t.destroyed&&!t.spawning&&!!t.opacity&&t.opacity.enable&&((t.opacity.maxLoops??0)<=0||(t.opacity.maxLoops??0)>0&&(t.opacity.loops??0)<(t.opacity.maxLoops??0))}reset(t){t.opacity&&(t.opacity.time=0,t.opacity.loops=0)}update(t,r){!this.isEnabled(t)||!t.opacity||Bd(t,t.opacity,!0,t.options.opacity.animation.destroy,r)}}async function g_(i,t=!0){i.checkVersion("3.8.1"),await i.addParticleUpdater("opacity",r=>Promise.resolve(new f_(r)),t)}const hl=0,nr=0;function m_(i){if(i.outMode!==Qe.bounce&&i.outMode!==Qe.split||i.direction!==me.left&&i.direction!==me.right)return;i.bounds.right<nr&&i.direction===me.left?i.particle.position.x=i.size+i.offset.x:i.bounds.left>i.canvasSize.width&&i.direction===me.right&&(i.particle.position.x=i.canvasSize.width-i.size-i.offset.x);const t=i.particle.velocity.x;let r=!1;if(i.direction===me.right&&i.bounds.right>=i.canvasSize.width&&t>hl||i.direction===me.left&&i.bounds.left<=nr&&t<hl){const l=ie(i.particle.options.bounce.horizontal.value);i.particle.velocity.x*=-l,r=!0}if(!r)return;const s=i.offset.x+i.size;i.bounds.right>=i.canvasSize.width&&i.direction===me.right?i.particle.position.x=i.canvasSize.width-s:i.bounds.left<=nr&&i.direction===me.left&&(i.particle.position.x=s),i.outMode===Qe.split&&i.particle.destroy()}function E_(i){if(i.outMode!==Qe.bounce&&i.outMode!==Qe.split||i.direction!==me.bottom&&i.direction!==me.top)return;i.bounds.bottom<nr&&i.direction===me.top?i.particle.position.y=i.size+i.offset.y:i.bounds.top>i.canvasSize.height&&i.direction===me.bottom&&(i.particle.position.y=i.canvasSize.height-i.size-i.offset.y);const t=i.particle.velocity.y;let r=!1;if(i.direction===me.bottom&&i.bounds.bottom>=i.canvasSize.height&&t>hl||i.direction===me.top&&i.bounds.top<=nr&&t<hl){const l=ie(i.particle.options.bounce.vertical.value);i.particle.velocity.y*=-l,r=!0}if(!r)return;const s=i.offset.y+i.size;i.bounds.bottom>=i.canvasSize.height&&i.direction===me.bottom?i.particle.position.y=i.canvasSize.height-s:i.bounds.top<=nr&&i.direction===me.top&&(i.particle.position.y=s),i.outMode===Qe.split&&i.particle.destroy()}class S_{constructor(t){this.container=t,this.modes=[Qe.bounce,Qe.split]}update(t,r,s,l){if(!this.modes.includes(l))return;const u=this.container;let p=!1;for(const y of u.plugins.values())if(y.particleBounce!==void 0&&(p=y.particleBounce(t,s,r)),p)break;if(p)return;const h=t.getPosition(),f=t.offset,g=t.getRadius(),E=_s(h,g),R=u.canvas.size;m_({particle:t,outMode:l,direction:r,bounds:E,canvasSize:R,offset:f,size:g}),E_({particle:t,outMode:l,direction:r,bounds:E,canvasSize:R,offset:f,size:g})}}const za=0;class R_{constructor(t){this.container=t,this.modes=[Qe.destroy]}update(t,r,s,l){if(!this.modes.includes(l))return;const u=this.container;switch(t.outType){case an.normal:case an.outside:if(Hd(t.position,u.canvas.size,tt.origin,t.getRadius(),r))return;break;case an.inside:{const{dx:p,dy:h}=Et(t.position,t.moveCenter),{x:f,y:g}=t.velocity;if(f<za&&p>t.moveCenter.radius||g<za&&h>t.moveCenter.radius||f>=za&&p<-t.moveCenter.radius||g>=za&&h<-t.moveCenter.radius)return;break}}u.particles.remove(t,t.group,!0)}}const Ga=0;class I_{constructor(t){this.container=t,this.modes=[Qe.none]}update(t,r,s,l){if(!this.modes.includes(l)||((t.options.move.distance.horizontal&&(r===me.left||r===me.right))??(t.options.move.distance.vertical&&(r===me.top||r===me.bottom))))return;const u=t.options.move.gravity,p=this.container,h=p.canvas.size,f=t.getRadius();if(u.enable){const g=t.position;(!u.inverse&&g.y>h.height+f&&r===me.bottom||u.inverse&&g.y<-f&&r===me.top)&&p.particles.remove(t)}else{if(t.velocity.y>Ga&&t.position.y<=h.height+f||t.velocity.y<Ga&&t.position.y>=-f||t.velocity.x>Ga&&t.position.x<=h.width+f||t.velocity.x<Ga&&t.position.x>=-f)return;Hd(t.position,p.canvas.size,tt.origin,f,r)||p.particles.remove(t)}}}const Qa=0,Va=0;class y_{constructor(t){this.container=t,this.modes=[Qe.out]}update(t,r,s,l){if(!this.modes.includes(l))return;const u=this.container;switch(t.outType){case an.inside:{const{x:p,y:h}=t.velocity,f=tt.origin;f.length=t.moveCenter.radius,f.angle=t.velocity.angle+Math.PI,f.addTo(tt.create(t.moveCenter));const{dx:g,dy:E}=Et(t.position,f);if(p<=Qa&&g>=Va||h<=Qa&&E>=Va||p>=Qa&&g<=Va||h>=Qa&&E<=Va)return;t.position.x=Math.floor(Bt({min:0,max:u.canvas.size.width})),t.position.y=Math.floor(Bt({min:0,max:u.canvas.size.height}));const{dx:R,dy:y}=Et(t.position,t.moveCenter);t.direction=Math.atan2(-y,-R),t.velocity.angle=t.direction;break}default:{if(Hd(t.position,u.canvas.size,tt.origin,t.getRadius(),r))return;switch(t.outType){case an.outside:{t.position.x=Math.floor(Bt({min:-t.moveCenter.radius,max:t.moveCenter.radius}))+t.moveCenter.x,t.position.y=Math.floor(Bt({min:-t.moveCenter.radius,max:t.moveCenter.radius}))+t.moveCenter.y;const{dx:p,dy:h}=Et(t.position,t.moveCenter);t.moveCenter.radius&&(t.direction=Math.atan2(h,p),t.velocity.angle=t.direction);break}case an.normal:{const p=t.options.move.warp,h=u.canvas.size,f={bottom:h.height+t.getRadius()+t.offset.y,left:-t.getRadius()-t.offset.x,right:h.width+t.getRadius()+t.offset.x,top:-t.getRadius()-t.offset.y},g=t.getRadius(),E=_s(t.position,g);r===me.right&&E.left>h.width+t.offset.x?(t.position.x=f.left,t.initialPosition.x=t.position.x,p||(t.position.y=Oe()*h.height,t.initialPosition.y=t.position.y)):r===me.left&&E.right<-t.offset.x&&(t.position.x=f.right,t.initialPosition.x=t.position.x,p||(t.position.y=Oe()*h.height,t.initialPosition.y=t.position.y)),r===me.bottom&&E.top>h.height+t.offset.y?(p||(t.position.x=Oe()*h.width,t.initialPosition.x=t.position.x),t.position.y=f.top,t.initialPosition.y=t.position.y):r===me.top&&E.bottom<-t.offset.y&&(p||(t.position.x=Oe()*h.width,t.initialPosition.x=t.position.x),t.position.y=f.bottom,t.initialPosition.y=t.position.y);break}}break}}}}const b_=(i,t)=>i.default===t||i.bottom===t||i.left===t||i.right===t||i.top===t;class C_{constructor(t){this._addUpdaterIfMissing=(r,s,l)=>{const u=r.options.move.outModes;!this.updaters.has(s)&&b_(u,s)&&this.updaters.set(s,l(this.container))},this._updateOutMode=(r,s,l,u)=>{for(const p of this.updaters.values())p.update(r,u,s,l)},this.container=t,this.updaters=new Map}init(t){this._addUpdaterIfMissing(t,Qe.bounce,r=>new S_(r)),this._addUpdaterIfMissing(t,Qe.out,r=>new y_(r)),this._addUpdaterIfMissing(t,Qe.destroy,r=>new R_(r)),this._addUpdaterIfMissing(t,Qe.none,r=>new I_(r))}isEnabled(t){return!t.destroyed&&!t.spawning}update(t,r){const s=t.options.move.outModes;this._updateOutMode(t,r,s.bottom??s.default,me.bottom),this._updateOutMode(t,r,s.left??s.default,me.left),this._updateOutMode(t,r,s.right??s.default,me.right),this._updateOutMode(t,r,s.top??s.default,me.top)}}async function w_(i,t=!0){i.checkVersion("3.8.1"),await i.addParticleUpdater("outModes",r=>Promise.resolve(new C_(r)),t)}var Ko;(function(i){i[i.r=1]="r",i[i.g=2]="g",i[i.b=3]="b",i[i.a=5]="a"})(Ko||(Ko={}));class __{constructor(){this.key="rgb",this.stringPrefix="rgb"}handleColor(t){const r=t.value,s=r.rgb??t.value;if(s.r!==void 0)return s}handleRangeColor(t){const r=t.value,s=r.rgb??t.value;if(s.r!==void 0)return{r:ie(s.r),g:ie(s.g),b:ie(s.b)}}parseString(t){if(!t.startsWith(this.stringPrefix))return;const r=/rgba?\(\s*(\d{1,3})\s*[\s,]\s*(\d{1,3})\s*[\s,]\s*(\d{1,3})\s*([\s,]\s*(0|1|0?\.\d+|(\d{1,3})%)\s*)?\)/i,s=r.exec(t),l=10;return s?{a:s.length>4?qE(s[Ko.a]):1,b:parseInt(s[Ko.b],l),g:parseInt(s[Ko.g],l),r:parseInt(s[Ko.r],l)}:void 0}}async function T_(i,t=!0){i.checkVersion("3.8.1"),await i.addColorManager(new __,t)}const ji=0;class v_{init(t){const r=t.container,s=t.options.size,l=s.animation;l.enable&&(t.size.velocity=(t.retina.sizeAnimationSpeed??r.retina.sizeAnimationSpeed)/mi*r.retina.reduceFactor,l.sync||(t.size.velocity*=Oe()))}isEnabled(t){return!t.destroyed&&!t.spawning&&t.size.enable&&((t.size.maxLoops??ji)<=ji||(t.size.maxLoops??ji)>ji&&(t.size.loops??ji)<(t.size.maxLoops??ji))}reset(t){t.size.loops=ji}update(t,r){this.isEnabled(t)&&Bd(t,t.size,!0,t.options.size.animation.destroy,r)}}async function x_(i,t=!0){i.checkVersion("3.8.1"),await i.addParticleUpdater("size",()=>Promise.resolve(new v_),t)}async function P_(i,t=!0){i.checkVersion("3.8.1"),await d_(i,!1),await h_(i,!1),await T_(i,!1),await Yw(i,!1),await i_(i,!1),await r_(i,!1),await g_(i,!1),await w_(i,!1),await x_(i,!1),await i.refresh(t)}async function X_(i,t=!0){i.checkVersion("3.8.1"),await i.addEasing($i.easeInQuad,r=>r**2,!1),await i.addEasing($i.easeOutQuad,r=>1-(1-r)**2,!1),await i.addEasing($i.easeInOutQuad,r=>r<.5?2*r**2:1-(-2*r+2)**2/2,!1),await i.refresh(t)}function N_(i,t){const{context:r,opacity:s}=i,l=.5,u=r.globalAlpha;if(!t)return;const p=t.width,h=p*l;r.globalAlpha=s,r.drawImage(t,-h,-h,p,p),r.globalAlpha=u}const ed='"Twemoji Mozilla", Apple Color Emoji, "Segoe UI Emoji", "Noto Color Emoji", "EmojiOne Color"',Dm=0;class O_{constructor(){this.validTypes=["emoji"],this._emojiShapeDict=new Map}destroy(){for(const[t,r]of this._emojiShapeDict)r instanceof ImageBitmap&&(r==null||r.close()),this._emojiShapeDict.delete(t)}draw(t){const r=t.particle.emojiDataKey;if(!r)return;const s=this._emojiShapeDict.get(r);s&&N_(t,s)}async init(t){const r=t.actualOptions,{validTypes:s}=this;if(!s.find(p=>je(p,r.particles.shape.type)))return;const l=[ym(ed)],u=s.map(p=>r.particles.shape.options[p]).find(p=>!!p);u&&Jt(u,p=>{p.font&&l.push(ym(p.font))}),await Promise.all(l)}particleDestroy(t){t.emojiDataKey=void 0}particleInit(t,r){const l=r.shapeData;if(!(l!=null&&l.value))return;const u=rn(l.value,r.randomIndexData);if(!u)return;const p=typeof u=="string"?{font:l.font??ed,padding:l.padding??Dm,value:u}:{font:ed,padding:Dm,...l,...u},h=p.font,f=p.value,g=`${f}_${h}`;if(this._emojiShapeDict.has(g)){r.emojiDataKey=g;return}const E=p.padding*2,R=mn(r.size.value),y=R+E,w=y*2;let b;if(typeof OffscreenCanvas<"u"){const X=new OffscreenCanvas(w,w),N=X.getContext("2d");if(!N)return;N.font=`400 ${R*2}px ${h}`,N.textBaseline="middle",N.textAlign="center",N.fillText(f,y,y),b=X.transferToImageBitmap()}else{const X=document.createElement("canvas");X.width=w,X.height=w;const N=X.getContext("2d");if(!N)return;N.font=`400 ${R*2}px ${h}`,N.textBaseline="middle",N.textAlign="center",N.fillText(f,y,y),b=X}this._emojiShapeDict.set(g,b),r.emojiDataKey=g}}async function F_(i,t=!0){i.checkVersion("3.8.1"),await i.addShape(new O_,t)}const D_=1,H_=1,rS=0;function sS(i,t,r,s,l,u){const p=t.actualOptions.interactivity.modes.attract;if(!p)return;const h=t.particles.quadTree.query(l,u);for(const f of h){const{dx:g,dy:E,distance:R}=Et(f.position,r),y=p.speed*p.factor,w=Zt(i.getEasing(p.easing)(H_-R/s)*y,D_,p.maxSpeed),b=tt.create(R?g/R*w:y,R?E/R*w:y);f.position.subFrom(b)}}function A_(i,t,r){t.attract||(t.attract={particles:[]});const{attract:s}=t;if(s.finish||(s.count||(s.count=0),s.count++,s.count===t.particles.count&&(s.finish=!0)),s.clicking){const l=t.interactivity.mouse.clickPosition,u=t.retina.attractModeDistance;if(!u||u<rS||!l)return;sS(i,t,l,u,new ht(l.x,l.y,u),p=>r(p))}else s.clicking===!1&&(s.particles=[])}function L_(i,t,r){const s=t.interactivity.mouse.position,l=t.retina.attractModeDistance;!l||l<rS||!s||sS(i,t,s,l,new ht(s.x,s.y,l),u=>r(u))}class B_{constructor(){this.distance=200,this.duration=.4,this.easing=$i.easeOutQuad,this.factor=1,this.maxSpeed=50,this.speed=1}load(t){Q(t)||(t.distance!==void 0&&(this.distance=t.distance),t.duration!==void 0&&(this.duration=t.duration),t.easing!==void 0&&(this.easing=t.easing),t.factor!==void 0&&(this.factor=t.factor),t.maxSpeed!==void 0&&(this.maxSpeed=t.maxSpeed),t.speed!==void 0&&(this.speed=t.speed))}}const ps="attract";let M_=class extends yn{constructor(t,r){super(r),this._engine=t,r.attract||(r.attract={particles:[]}),this.handleClickMode=s=>{const l=this.container.actualOptions,u=l.interactivity.modes.attract;if(!(!u||s!==ps)){r.attract||(r.attract={particles:[]}),r.attract.clicking=!0,r.attract.count=0;for(const p of r.attract.particles)this.isEnabled(p)&&p.velocity.setTo(p.initialVelocity);r.attract.particles=[],r.attract.finish=!1,setTimeout(()=>{r.destroyed||(r.attract||(r.attract={particles:[]}),r.attract.clicking=!1)},u.duration*dt)}}}clear(){}init(){const t=this.container,r=t.actualOptions.interactivity.modes.attract;r&&(t.retina.attractModeDistance=r.distance*t.retina.pixelRatio)}interact(){const t=this.container,r=t.actualOptions,s=t.interactivity.status===eo,l=r.interactivity.events,{enable:u,mode:p}=l.onHover,{enable:h,mode:f}=l.onClick;s&&u&&je(ps,p)?L_(this._engine,this.container,g=>this.isEnabled(g)):h&&je(ps,f)&&A_(this._engine,this.container,g=>this.isEnabled(g))}isEnabled(t){const r=this.container,s=r.actualOptions,l=r.interactivity.mouse,u=((t==null?void 0:t.interactivity)??s.interactivity).events;if((!l.position||!u.onHover.enable)&&(!l.clickPosition||!u.onClick.enable))return!1;const p=u.onHover.mode,h=u.onClick.mode;return je(ps,p)||je(ps,h)}loadModeOptions(t,...r){t.attract||(t.attract=new B_);for(const s of r)t.attract.load(s==null?void 0:s.attract)}reset(){}};async function q_(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalAttract",r=>Promise.resolve(new M_(i,r)),t)}const U_=2,rl=.5,W_=Math.PI*rl,Hm=2,aS=10,j_=0;function lS(i,t,r,s,l){const u=i.particles.quadTree.query(s,l);for(const p of u)s instanceof ht?jE(Id(p),{position:t,mass:r**U_*W_,velocity:tt.origin,factor:tt.origin}):s instanceof ln&&uC(p,_s(t,r))}function k_(i,t,r,s){const l=document.querySelectorAll(t);l.length&&l.forEach(u=>{const p=u,h=i.retina.pixelRatio,f={x:(p.offsetLeft+p.offsetWidth*rl)*h,y:(p.offsetTop+p.offsetHeight*rl)*h},g=p.offsetWidth*rl*h,E=aS*h,R=r.type===sr.circle?new ht(f.x,f.y,g+E):new ln(p.offsetLeft*h-E,p.offsetTop*h-E,p.offsetWidth*h+E*Hm,p.offsetHeight*h+E*Hm);s(f,g,R)})}function z_(i,t,r,s){Ld(r,t,(l,u)=>k_(i,l,u,(p,h,f)=>lS(i,p,h,f,s)))}function G_(i,t){const r=i.retina.pixelRatio,s=aS*r,l=i.interactivity.mouse.position,u=i.retina.bounceModeDistance;!u||u<j_||!l||lS(i,l,u,new ht(l.x,l.y,u+s),t)}class Q_{constructor(){this.distance=200}load(t){Q(t)||t.distance!==void 0&&(this.distance=t.distance)}}const Ya="bounce";class V_ extends yn{constructor(t){super(t)}clear(){}init(){const t=this.container,r=t.actualOptions.interactivity.modes.bounce;r&&(t.retina.bounceModeDistance=r.distance*t.retina.pixelRatio)}interact(){const t=this.container,r=t.actualOptions,s=r.interactivity.events,l=t.interactivity.status===eo,u=s.onHover.enable,p=s.onHover.mode,h=s.onDiv;l&&u&&je(Ya,p)?G_(this.container,f=>this.isEnabled(f)):z_(this.container,h,Ya,f=>this.isEnabled(f))}isEnabled(t){const r=this.container,s=r.actualOptions,l=r.interactivity.mouse,u=((t==null?void 0:t.interactivity)??s.interactivity).events,p=u.onDiv;return!!l.position&&u.onHover.enable&&je(Ya,u.onHover.mode)||Ad(Ya,p)}loadModeOptions(t,...r){t.bounce||(t.bounce=new Q_);for(const s of r)t.bounce.load(s==null?void 0:s.bounce)}reset(){}}async function Y_(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalBounce",r=>Promise.resolve(new V_(r)),t)}class cS{constructor(){this.distance=200,this.duration=.4,this.mix=!1}load(t){if(!Q(t)){if(t.distance!==void 0&&(this.distance=t.distance),t.duration!==void 0&&(this.duration=t.duration),t.mix!==void 0&&(this.mix=t.mix),t.opacity!==void 0&&(this.opacity=t.opacity),t.color!==void 0){const r=In(this.color)?void 0:this.color;this.color=Jt(t.color,s=>ft.create(r,s))}t.size!==void 0&&(this.size=t.size)}}}class K_ extends cS{constructor(){super(),this.selectors=[]}load(t){super.load(t),!Q(t)&&t.selectors!==void 0&&(this.selectors=t.selectors)}}class Z_ extends cS{load(t){super.load(t),!Q(t)&&(this.divs=Jt(t.divs,r=>{const s=new K_;return s.load(r),s}))}}var gn;(function(i){i.color="color",i.opacity="opacity",i.size="size"})(gn||(gn={}));function Am(i,t,r,s){if(t>=r){const l=i+(t-r)*s;return Zt(l,i,t)}else if(t<r){const l=i-(r-t)*s;return Zt(l,t,i)}}const ki="bubble",td=0,J_=0,$_=2,Lm=1,Bm=1,eT=0,tT=0,nd=.5,id=1;class nT extends yn{constructor(t,r){super(t),this._clickBubble=()=>{var E;const s=this.container,l=s.actualOptions,u=s.interactivity.mouse.clickPosition,p=l.interactivity.modes.bubble;if(!p||!u)return;s.bubble||(s.bubble={});const h=s.retina.bubbleModeDistance;if(!h||h<td)return;const f=s.particles.quadTree.queryCircle(u,h,R=>this.isEnabled(R)),{bubble:g}=s;for(const R of f){if(!g.clicking)continue;R.bubble.inRange=!g.durationEnd;const y=R.getPosition(),w=Mt(y,u),b=(new Date().getTime()-(s.interactivity.mouse.clickTime??J_))/dt;b>p.duration&&(g.durationEnd=!0),b>p.duration*$_&&(g.clicking=!1,g.durationEnd=!1);const X={bubbleObj:{optValue:s.retina.bubbleModeSize,value:R.bubble.radius},particlesObj:{optValue:mn(R.options.size.value)*s.retina.pixelRatio,value:R.size.value},type:gn.size};this._process(R,w,b,X);const N={bubbleObj:{optValue:p.opacity,value:R.bubble.opacity},particlesObj:{optValue:mn(R.options.opacity.value),value:((E=R.opacity)==null?void 0:E.value)??Lm},type:gn.opacity};this._process(R,w,b,N),!g.durationEnd&&w<=h?this._hoverBubbleColor(R,w):delete R.bubble.color}},this._hoverBubble=()=>{const s=this.container,l=s.interactivity.mouse.position,u=s.retina.bubbleModeDistance;if(!u||u<td||!l)return;const p=s.particles.quadTree.queryCircle(l,u,h=>this.isEnabled(h));for(const h of p){h.bubble.inRange=!0;const f=h.getPosition(),g=Mt(f,l),E=Bm-g/u;g<=u?E>=tT&&s.interactivity.status===eo&&(this._hoverBubbleSize(h,E),this._hoverBubbleOpacity(h,E),this._hoverBubbleColor(h,E)):this.reset(h),s.interactivity.status===hd&&this.reset(h)}},this._hoverBubbleColor=(s,l,u)=>{const p=this.container.actualOptions,h=u??p.interactivity.modes.bubble;if(h){if(!s.bubble.finalColor){const f=h.color;if(!f)return;const g=rn(f);s.bubble.finalColor=Es(this._engine,g)}if(s.bubble.finalColor)if(h.mix){s.bubble.color=void 0;const f=s.getFillColor();s.bubble.color=f?VE(Md(f,s.bubble.finalColor,Bm-l,l)):s.bubble.finalColor}else s.bubble.color=s.bubble.finalColor}},this._hoverBubbleOpacity=(s,l,u)=>{var y,w;const p=this.container,h=p.actualOptions,f=(u==null?void 0:u.opacity)??((y=h.interactivity.modes.bubble)==null?void 0:y.opacity);if(!f)return;const g=s.options.opacity.value,E=((w=s.opacity)==null?void 0:w.value)??Lm,R=Am(E,f,mn(g),l);R!==void 0&&(s.bubble.opacity=R)},this._hoverBubbleSize=(s,l,u)=>{const p=this.container,h=u!=null&&u.size?u.size*p.retina.pixelRatio:p.retina.bubbleModeSize;if(h===void 0)return;const f=mn(s.options.size.value)*p.retina.pixelRatio,g=s.size.value,E=Am(g,h,f,l);E!==void 0&&(s.bubble.radius=E)},this._process=(s,l,u,p)=>{const h=this.container,f=p.bubbleObj.optValue,g=h.actualOptions,E=g.interactivity.modes.bubble;if(!E||f===void 0)return;const R=E.duration,y=h.retina.bubbleModeDistance,w=p.particlesObj.optValue,b=p.bubbleObj.value,X=p.particlesObj.value??eT,N=p.type;if(!(!y||y<td||f===w))if(h.bubble||(h.bubble={}),h.bubble.durationEnd)b&&(N===gn.size&&delete s.bubble.radius,N===gn.opacity&&delete s.bubble.opacity);else if(l<=y){if((b??X)!==f){const A=X-u*(X-f)/R;N===gn.size&&(s.bubble.radius=A),N===gn.opacity&&(s.bubble.opacity=A)}}else N===gn.size&&delete s.bubble.radius,N===gn.opacity&&delete s.bubble.opacity},this._singleSelectorHover=(s,l,u)=>{const p=this.container,h=document.querySelectorAll(l),f=p.actualOptions.interactivity.modes.bubble;!f||!h.length||h.forEach(g=>{const E=g,R=p.retina.pixelRatio,y={x:(E.offsetLeft+E.offsetWidth*nd)*R,y:(E.offsetTop+E.offsetHeight*nd)*R},w=E.offsetWidth*nd*R,b=u.type===sr.circle?new ht(y.x,y.y,w):new ln(E.offsetLeft*R,E.offsetTop*R,E.offsetWidth*R,E.offsetHeight*R),X=p.particles.quadTree.query(b,N=>this.isEnabled(N));for(const N of X){if(!b.contains(N.getPosition()))continue;N.bubble.inRange=!0;const v=f.divs,A=WE(v,E);(!N.bubble.div||N.bubble.div!==E)&&(this.clear(N,s,!0),N.bubble.div=E),this._hoverBubbleSize(N,id,A),this._hoverBubbleOpacity(N,id,A),this._hoverBubbleColor(N,id,A)}})},this._engine=r,t.bubble||(t.bubble={}),this.handleClickMode=s=>{s===ki&&(t.bubble||(t.bubble={}),t.bubble.clicking=!0)}}clear(t,r,s){t.bubble.inRange&&!s||(delete t.bubble.div,delete t.bubble.opacity,delete t.bubble.radius,delete t.bubble.color)}init(){const t=this.container,r=t.actualOptions.interactivity.modes.bubble;r&&(t.retina.bubbleModeDistance=r.distance*t.retina.pixelRatio,r.size!==void 0&&(t.retina.bubbleModeSize=r.size*t.retina.pixelRatio))}interact(t){const r=this.container.actualOptions,s=r.interactivity.events,l=s.onHover,u=s.onClick,p=l.enable,h=l.mode,f=u.enable,g=u.mode,E=s.onDiv;p&&je(ki,h)?this._hoverBubble():f&&je(ki,g)?this._clickBubble():Ld(ki,E,(R,y)=>this._singleSelectorHover(t,R,y))}isEnabled(t){const r=this.container,s=r.actualOptions,l=r.interactivity.mouse,u=((t==null?void 0:t.interactivity)??s.interactivity).events,{onClick:p,onDiv:h,onHover:f}=u,g=Ad(ki,h);return g||f.enable&&l.position||p.enable&&l.clickPosition?je(ki,f.mode)||je(ki,p.mode)||g:!1}loadModeOptions(t,...r){t.bubble||(t.bubble=new Z_);for(const s of r)t.bubble.load(s==null?void 0:s.bubble)}reset(t){t.bubble.inRange=!1}}async function iT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalBubble",r=>Promise.resolve(new nT(r,i)),t)}class oT{constructor(){this.opacity=.5}load(t){Q(t)||t.opacity!==void 0&&(this.opacity=t.opacity)}}class rT{constructor(){this.distance=80,this.links=new oT,this.radius=60}load(t){Q(t)||(t.distance!==void 0&&(this.distance=t.distance),this.links.load(t.links),t.radius!==void 0&&(this.radius=t.radius))}}const Mm=0,qm=1,sT=0;function aT(i,t,r,s){const l=Math.floor(r.getRadius()/t.getRadius()),u=t.getFillColor(),p=r.getFillColor();if(!u||!p)return;const h=t.getPosition(),f=r.getPosition(),g=Md(u,p,t.getRadius(),r.getRadius()),E=i.createLinearGradient(h.x,h.y,f.x,f.y);return E.addColorStop(Mm,Ss(u,s)),E.addColorStop(Zt(l,Mm,qm),Wn(g,s)),E.addColorStop(qm,Ss(p,s)),E}function lT(i,t,r,s,l){fs(i,s,l),i.lineWidth=t,i.strokeStyle=r,i.stroke()}function cT(i,t,r,s){const l=i.actualOptions,u=l.interactivity.modes.connect;if(u)return aT(t,r,s,u.links.opacity)}function uT(i,t,r){i.canvas.draw(s=>{const l=cT(i,s,t,r);if(!l)return;const u=t.getPosition(),p=r.getPosition();lT(s,t.retina.linksWidth??sT,l,u,p)})}const dT="connect",Um=0;class pT extends yn{constructor(t){super(t)}clear(){}init(){const t=this.container,r=t.actualOptions.interactivity.modes.connect;r&&(t.retina.connectModeDistance=r.distance*t.retina.pixelRatio,t.retina.connectModeRadius=r.radius*t.retina.pixelRatio)}interact(){const t=this.container;if(t.actualOptions.interactivity.events.onHover.enable&&t.interactivity.status==="pointermove"){const s=t.interactivity.mouse.position,{connectModeDistance:l,connectModeRadius:u}=t.retina;if(!l||l<Um||!u||u<Um||!s)return;const p=Math.abs(u),h=t.particles.quadTree.queryCircle(s,p,f=>this.isEnabled(f));h.forEach((f,g)=>{const E=f.getPosition(),R=1;for(const y of h.slice(g+R)){const w=y.getPosition(),b=Math.abs(l),X=Math.abs(E.x-w.x),N=Math.abs(E.y-w.y);X<b&&N<b&&uT(t,f,y)}})}}isEnabled(t){const r=this.container,s=r.interactivity.mouse,l=((t==null?void 0:t.interactivity)??r.actualOptions.interactivity).events;return l.onHover.enable&&s.position?je(dT,l.onHover.mode):!1}loadModeOptions(t,...r){t.connect||(t.connect=new rT);for(const s of r)t.connect.load(s==null?void 0:s.connect)}reset(){}}async function hT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalConnect",r=>Promise.resolve(new pT(r)),t)}class fT{constructor(){this.blink=!1,this.consent=!1,this.opacity=1}load(t){Q(t)||(t.blink!==void 0&&(this.blink=t.blink),t.color!==void 0&&(this.color=ft.create(this.color,t.color)),t.consent!==void 0&&(this.consent=t.consent),t.opacity!==void 0&&(this.opacity=t.opacity))}}class gT{constructor(){this.distance=100,this.links=new fT}load(t){Q(t)||(t.distance!==void 0&&(this.distance=t.distance),this.links.load(t.links))}}const mT=0;function ET(i,t,r,s,l,u){fs(i,r,s),i.strokeStyle=Wn(l,u),i.lineWidth=t,i.stroke()}function ST(i,t,r,s,l){i.canvas.draw(u=>{const p=t.getPosition();ET(u,t.retina.linksWidth??mT,p,l,r,s)})}const RT="grab",IT=0,yT=0;class bT extends yn{constructor(t,r){super(t),this._engine=r}clear(){}init(){const t=this.container,r=t.actualOptions.interactivity.modes.grab;r&&(t.retina.grabModeDistance=r.distance*t.retina.pixelRatio)}interact(){var h;const t=this.container,r=t.actualOptions,s=r.interactivity;if(!s.modes.grab||!s.events.onHover.enable||t.interactivity.status!==eo)return;const l=t.interactivity.mouse.position;if(!l)return;const u=t.retina.grabModeDistance;if(!u||u<IT)return;const p=t.particles.quadTree.queryCircle(l,u,f=>this.isEnabled(f));for(const f of p){const g=f.getPosition(),E=Mt(g,l);if(E>u)continue;const R=s.modes.grab.links,y=R.opacity,w=y-E*y/u;if(w<=yT)continue;const b=R.color??((h=f.options.links)==null?void 0:h.color);if(!t.particles.grabLineColor&&b){const N=s.modes.grab.links;t.particles.grabLineColor=KE(this._engine,b,N.blink,N.consent)}const X=bd(f,void 0,t.particles.grabLineColor);X&&ST(t,f,X,w,l)}}isEnabled(t){const r=this.container,s=r.interactivity.mouse,l=((t==null?void 0:t.interactivity)??r.actualOptions.interactivity).events;return l.onHover.enable&&!!s.position&&je(RT,l.onHover.mode)}loadModeOptions(t,...r){t.grab||(t.grab=new gT);for(const s of r)t.grab.load(s==null?void 0:s.grab)}reset(){}}async function CT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalGrab",r=>Promise.resolve(new bT(r,i)),t)}const wT="pause";class _T extends yn{constructor(t){super(t),this.handleClickMode=r=>{if(r!==wT)return;const s=this.container;s.animationStatus?s.pause():s.play()}}clear(){}init(){}interact(){}isEnabled(){return!0}reset(){}}async function TT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalPause",r=>Promise.resolve(new _T(r)),t)}class vT{constructor(){this.default=!0,this.groups=[],this.quantity=4}load(t){if(Q(t))return;t.default!==void 0&&(this.default=t.default),t.groups!==void 0&&(this.groups=t.groups.map(s=>s)),this.groups.length||(this.default=!0);const r=t.quantity;r!==void 0&&(this.quantity=ye(r))}}const xT="push",PT=0;class XT extends yn{constructor(t){super(t),this.handleClickMode=r=>{if(r!==xT)return;const s=this.container,l=s.actualOptions,u=l.interactivity.modes.push;if(!u)return;const p=ie(u.quantity);if(p<=PT)return;const h=Sl([void 0,...u.groups]),f=h!==void 0?s.actualOptions.particles.groups[h]:void 0;s.particles.push(p,s.interactivity.mouse,f,h)}}clear(){}init(){}interact(){}isEnabled(){return!0}loadModeOptions(t,...r){t.push||(t.push=new vT);for(const s of r)t.push.load(s==null?void 0:s.push)}reset(){}}async function NT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalPush",r=>Promise.resolve(new XT(r)),t)}class OT{constructor(){this.quantity=2}load(t){if(Q(t))return;const r=t.quantity;r!==void 0&&(this.quantity=ye(r))}}const FT="remove";class DT extends yn{constructor(t){super(t),this.handleClickMode=r=>{const s=this.container,l=s.actualOptions;if(!l.interactivity.modes.remove||r!==FT)return;const u=ie(l.interactivity.modes.remove.quantity);s.particles.removeQuantity(u)}}clear(){}init(){}interact(){}isEnabled(){return!0}loadModeOptions(t,...r){t.remove||(t.remove=new OT);for(const s of r)t.remove.load(s==null?void 0:s.remove)}reset(){}}async function HT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalRemove",r=>Promise.resolve(new DT(r)),t)}class uS{constructor(){this.distance=200,this.duration=.4,this.factor=100,this.speed=1,this.maxSpeed=50,this.easing=$i.easeOutQuad}load(t){Q(t)||(t.distance!==void 0&&(this.distance=t.distance),t.duration!==void 0&&(this.duration=t.duration),t.easing!==void 0&&(this.easing=t.easing),t.factor!==void 0&&(this.factor=t.factor),t.speed!==void 0&&(this.speed=t.speed),t.maxSpeed!==void 0&&(this.maxSpeed=t.maxSpeed))}}class AT extends uS{constructor(){super(),this.selectors=[]}load(t){super.load(t),!Q(t)&&t.selectors!==void 0&&(this.selectors=t.selectors)}}class LT extends uS{load(t){super.load(t),!Q(t)&&(this.divs=Jt(t.divs,r=>{const s=new AT;return s.load(r),s}))}}const zi="repulse",BT=0,MT=6,qT=3,UT=2,WT=0,jT=0,kT=1,od=.5;class zT extends yn{constructor(t,r){super(r),this._clickRepulse=()=>{const s=this.container,l=s.actualOptions.interactivity.modes.repulse;if(!l)return;const u=s.repulse??{particles:[]};if(u.finish||(u.count||(u.count=0),u.count++,u.count===s.particles.count&&(u.finish=!0)),u.clicking){const p=s.retina.repulseModeDistance;if(!p||p<BT)return;const h=Math.pow(p/MT,qT),f=s.interactivity.mouse.clickPosition;if(f===void 0)return;const g=new ht(f.x,f.y,h),E=s.particles.quadTree.query(g,R=>this.isEnabled(R));for(const R of E){const{dx:y,dy:w,distance:b}=Et(f,R.position),X=b**UT,N=l.speed,v=-h*N/X;if(X<=h){u.particles.push(R);const A=tt.create(y,w);A.length=v,R.velocity.setTo(A)}}}else if(u.clicking===!1){for(const p of u.particles)p.velocity.setTo(p.initialVelocity);u.particles=[]}},this._hoverRepulse=()=>{const s=this.container,l=s.interactivity.mouse.position,u=s.retina.repulseModeDistance;!u||u<WT||!l||this._processRepulse(l,u,new ht(l.x,l.y,u))},this._processRepulse=(s,l,u,p)=>{const h=this.container,f=h.particles.quadTree.query(u,N=>this.isEnabled(N)),g=h.actualOptions.interactivity.modes.repulse;if(!g)return;const{easing:E,speed:R,factor:y,maxSpeed:w}=g,b=this._engine.getEasing(E),X=((p==null?void 0:p.speed)??R)*y;for(const N of f){const{dx:v,dy:A,distance:k}=Et(N.position,s),U=Zt(b(kT-k/l)*X,jT,w),j=tt.create(k?v/k*U:X,k?A/k*U:X);N.position.addTo(j)}},this._singleSelectorRepulse=(s,l)=>{const u=this.container,p=u.actualOptions.interactivity.modes.repulse;if(!p)return;const h=document.querySelectorAll(s);h.length&&h.forEach(f=>{const g=f,E=u.retina.pixelRatio,R={x:(g.offsetLeft+g.offsetWidth*od)*E,y:(g.offsetTop+g.offsetHeight*od)*E},y=g.offsetWidth*od*E,w=l.type===sr.circle?new ht(R.x,R.y,y):new ln(g.offsetLeft*E,g.offsetTop*E,g.offsetWidth*E,g.offsetHeight*E),b=p.divs,X=WE(b,g);this._processRepulse(R,y,w,X)})},this._engine=t,r.repulse||(r.repulse={particles:[]}),this.handleClickMode=s=>{const l=this.container.actualOptions,u=l.interactivity.modes.repulse;if(!u||s!==zi)return;r.repulse||(r.repulse={particles:[]});const p=r.repulse;p.clicking=!0,p.count=0;for(const h of r.repulse.particles)this.isEnabled(h)&&h.velocity.setTo(h.initialVelocity);p.particles=[],p.finish=!1,setTimeout(()=>{r.destroyed||(p.clicking=!1)},u.duration*dt)}}clear(){}init(){const t=this.container,r=t.actualOptions.interactivity.modes.repulse;r&&(t.retina.repulseModeDistance=r.distance*t.retina.pixelRatio)}interact(){const t=this.container,r=t.actualOptions,s=t.interactivity.status===eo,l=r.interactivity.events,u=l.onHover,p=u.enable,h=u.mode,f=l.onClick,g=f.enable,E=f.mode,R=l.onDiv;s&&p&&je(zi,h)?this._hoverRepulse():g&&je(zi,E)?this._clickRepulse():Ld(zi,R,(y,w)=>this._singleSelectorRepulse(y,w))}isEnabled(t){const r=this.container,s=r.actualOptions,l=r.interactivity.mouse,u=((t==null?void 0:t.interactivity)??s.interactivity).events,p=u.onDiv,h=u.onHover,f=u.onClick,g=Ad(zi,p);if(!(g||h.enable&&l.position||f.enable&&l.clickPosition))return!1;const E=h.mode,R=f.mode;return je(zi,E)||je(zi,R)||g}loadModeOptions(t,...r){t.repulse||(t.repulse=new LT);for(const s of r)t.repulse.load(s==null?void 0:s.repulse)}reset(){}}async function GT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalRepulse",r=>Promise.resolve(new zT(i,r)),t)}class QT{constructor(){this.factor=3,this.radius=200}load(t){Q(t)||(t.factor!==void 0&&(this.factor=t.factor),t.radius!==void 0&&(this.radius=t.radius))}}const VT="slow",YT=0;class KT extends yn{constructor(t){super(t)}clear(t,r,s){t.slow.inRange&&!s||(t.slow.factor=1)}init(){const t=this.container,r=t.actualOptions.interactivity.modes.slow;r&&(t.retina.slowModeRadius=r.radius*t.retina.pixelRatio)}interact(){}isEnabled(t){const r=this.container,s=r.interactivity.mouse,l=((t==null?void 0:t.interactivity)??r.actualOptions.interactivity).events;return l.onHover.enable&&!!s.position&&je(VT,l.onHover.mode)}loadModeOptions(t,...r){t.slow||(t.slow=new QT);for(const s of r)t.slow.load(s==null?void 0:s.slow)}reset(t){t.slow.inRange=!1;const r=this.container,s=r.actualOptions,l=r.interactivity.mouse.position,u=r.retina.slowModeRadius,p=s.interactivity.modes.slow;if(!p||!u||u<YT||!l)return;const h=t.getPosition(),f=Mt(l,h),g=f/u,E=p.factor,{slow:R}=t;f>u||(R.inRange=!0,R.factor=g/E)}}async function ZT(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("externalSlow",r=>Promise.resolve(new KT(r)),t)}const JT=0,$T=1,ev=/(#(?:[0-9a-f]{2}){2,4}|(#[0-9a-f]{3})|(rgb|hsl)a?\((-?\d+%?[,\s]+){2,3}\s*[\d.]+%?\))|currentcolor/gi;function tv(i,t,r){const{svgData:s}=i;if(!s)return"";const l=Ss(t,r);if(s.includes("fill"))return s.replace(ev,()=>l);const u=s.indexOf(">");return`${s.substring(JT,u)} fill="${l}"${s.substring(u)}`}async function Rl(i){return new Promise(t=>{i.loading=!0;const r=new Image;i.element=r,r.addEventListener("load",()=>{i.loading=!1,t()}),r.addEventListener("error",()=>{i.element=void 0,i.error=!0,i.loading=!1,no().error(`${Kt} loading image: ${i.source}`),t()}),r.src=i.source})}async function nv(i){if(i.type!=="svg"){await Rl(i);return}i.loading=!0;const t=await fetch(i.source);t.ok?i.svgData=await t.text():(no().error(`${Kt} Image not found`),i.error=!0),i.loading=!1}function iv(i,t,r,s){var p;const l=tv(i,r,((p=s.opacity)==null?void 0:p.value)??$T),u={color:r,gif:t.gif,data:{...i,svgData:l},loaded:!1,ratio:t.width/t.height,replaceColor:t.replaceColor,source:t.src};return new Promise(h=>{const f=new Blob([l],{type:"image/svg+xml"}),g=URL||window.URL||window.webkitURL||window,E=g.createObjectURL(f),R=new Image;R.addEventListener("load",()=>{u.loaded=!0,u.element=R,h(u),g.revokeObjectURL(E)});const y=async()=>{g.revokeObjectURL(E);const w={...i,error:!1,loading:!0};await Rl(w),u.loaded=!0,u.element=w.element,h(u)};R.addEventListener("error",()=>void y()),R.src=E})}const rd=[0,4,2,1],Wm=[8,8,4,2];class ov{constructor(t){this.pos=0,this.data=new Uint8ClampedArray(t)}getString(t){const r=this.data.slice(this.pos,this.pos+t);return this.pos+=r.length,r.reduce((s,l)=>s+String.fromCharCode(l),"")}nextByte(){return this.data[this.pos++]}nextTwoBytes(){return this.pos+=2,this.data[this.pos-2]+(this.data[this.pos-1]<<8)}readSubBlocks(){let t="",r=0;const s=0,l=0;do{r=this.data[this.pos++];for(let u=r;--u>=s;t+=String.fromCharCode(this.data[this.pos++]));}while(r!==l);return t}readSubBlocksBin(){let t=this.data[this.pos],r=0;const s=0,l=1;for(let p=0;t!==s;p+=t+l,t=this.data[this.pos+p])r+=t;const u=new Uint8Array(r);t=this.data[this.pos++];for(let p=0;t!==s;t=this.data[this.pos++])for(let h=t;--h>=s;u[p++]=this.data[this.pos++]);return u}skipSubBlocks(){for(const t=1,r=0;this.data[this.pos]!==r;this.pos+=this.data[this.pos]+t);this.pos++}}var nn;(function(i){i[i.Replace=0]="Replace",i[i.Combine=1]="Combine",i[i.RestoreBackground=2]="RestoreBackground",i[i.RestorePrevious=3]="RestorePrevious",i[i.UndefinedA=4]="UndefinedA",i[i.UndefinedB=5]="UndefinedB",i[i.UndefinedC=6]="UndefinedC",i[i.UndefinedD=7]="UndefinedD"})(nn||(nn={}));var qn;(function(i){i[i.Extension=33]="Extension",i[i.ApplicationExtension=255]="ApplicationExtension",i[i.GraphicsControlExtension=249]="GraphicsControlExtension",i[i.PlainTextExtension=1]="PlainTextExtension",i[i.CommentExtension=254]="CommentExtension",i[i.Image=44]="Image",i[i.EndOfFile=59]="EndOfFile"})(qn||(qn={}));const Ct={x:0,y:0},rv=0,jm=.5,sv=0,km=0,wd=0;function dS(i,t){const r=[];for(let s=0;s<t;s++)r.push({r:i.data[i.pos],g:i.data[i.pos+1],b:i.data[i.pos+2]}),i.pos+=3;return r}function av(i,t,r,s){switch(i.nextByte()){case qn.GraphicsControlExtension:{const l=t.frames[r(!1)];i.pos++;const u=i.nextByte();l.GCreserved=(u&224)>>>5,l.disposalMethod=(u&28)>>>2,l.userInputDelayFlag=(u&2)===2;const p=(u&1)===1;l.delayTime=i.nextTwoBytes()*10;const h=i.nextByte();p&&s(h),i.pos++;break}case qn.ApplicationExtension:{i.pos++;const l={identifier:i.getString(8),authenticationCode:i.getString(3),data:i.readSubBlocksBin()};t.applicationExtensions.push(l);break}case qn.CommentExtension:{t.comments.push([r(!1),i.readSubBlocks()]);break}case qn.PlainTextExtension:{if(t.globalColorTable.length===0)throw new EvalError("plain text extension without global color table");i.pos++,t.frames[r(!1)].plainTextData={left:i.nextTwoBytes(),top:i.nextTwoBytes(),width:i.nextTwoBytes(),height:i.nextTwoBytes(),charSize:{width:i.nextTwoBytes(),height:i.nextTwoBytes()},foregroundColor:i.nextByte(),backgroundColor:i.nextByte(),text:i.readSubBlocks()};break}default:i.skipSubBlocks();break}}async function lv(i,t,r,s,l,u){const p=t.frames[s(!0)];p.left=i.nextTwoBytes(),p.top=i.nextTwoBytes(),p.width=i.nextTwoBytes(),p.height=i.nextTwoBytes();const h=i.nextByte(),f=(h&128)===128,g=(h&64)===64;p.sortFlag=(h&32)===32,p.reserved=(h&24)>>>3;const E=1<<(h&7)+1;f&&(p.localColorTable=dS(i,E));const R=v=>{const{r:A,g:k,b:U}=(f?p.localColorTable:t.globalColorTable)[v];return v!==l(null)?{r:A,g:k,b:U,a:255}:{r:A,g:k,b:U,a:r?~~((A+k+U)/3):0}},y=(()=>{try{return new ImageData(p.width,p.height,{colorSpace:"srgb"})}catch(v){if(v instanceof DOMException&&v.name==="IndexSizeError")return null;throw v}})();if(y==null)throw new EvalError("GIF frame size is to large");const w=i.nextByte(),b=i.readSubBlocksBin(),X=1<<w,N=(v,A)=>{const k=v>>>3,U=v&7;return(b[k]+(b[k+1]<<8)+(b[k+2]<<16)&(1<<A)-1<<U)>>>U};if(g){for(let v=0,A=w+1,k=0,U=[[0]],j=0;j<4;j++)if(rd[j]<p.height){let K=0,Z=0,ce=!1;for(;!ce;){const re=v;if(v=N(k,A),k+=A+1,v===X){A=w+1,U.length=X+2;for(let te=0;te<U.length;te++)U[te]=te<X?[te]:[]}else{v>=U.length?U.push(U[re].concat(U[re][0])):re!==X&&U.push(U[re].concat(U[v][0]));for(const te of U[v]){const{r:Xe,g:_e,b:Ee,a:qe}=R(te);y.data.set([Xe,_e,Ee,qe],rd[j]*p.width+Wm[j]*Z+K%(p.width*4)),K+=4}U.length===1<<A&&A<12&&A++}K===p.width*4*(Z+1)&&(Z++,rd[j]+Wm[j]*Z>=p.height&&(ce=!0))}}p.image=y,p.bitmap=await createImageBitmap(y)}else{let v=0,A=w+1,k=0,U=-4,j=!1;const K=[[0]];for(;!j;){const Z=v;if(v=N(k,A),k+=A,v===X){A=w+1,K.length=X+2;for(let ce=0;ce<K.length;ce++)K[ce]=ce<X?[ce]:[]}else{if(v===X+1){j=!0;break}v>=K.length?K.push(K[Z].concat(K[Z][0])):Z!==X&&K.push(K[Z].concat(K[v][0]));for(const ce of K[v]){const{r:re,g:te,b:Xe,a:_e}=R(ce);y.data.set([re,te,Xe,_e],U+=4)}K.length>=1<<A&&A<12&&A++}}p.image=y,p.bitmap=await createImageBitmap(y)}}async function cv(i,t,r,s,l,u){switch(i.nextByte()){case qn.EndOfFile:return!0;case qn.Image:await lv(i,t,r,s,l);break;case qn.Extension:av(i,t,s,l);break;default:throw new EvalError("undefined block found")}return!1}function uv(i){for(const t of i.applicationExtensions)if(t.identifier+t.authenticationCode==="NETSCAPE2.0")return t.data[1]+(t.data[2]<<8);return NaN}async function dv(i,t,r){r||(r=!1);const s=await fetch(i);if(!s.ok&&s.status===404)throw new EvalError("file not found");const l=await s.arrayBuffer(),u={width:0,height:0,totalTime:0,colorRes:0,pixelAspectRatio:0,frames:[],sortFlag:!1,globalColorTable:[],backgroundImage:new ImageData(1,1,{colorSpace:"srgb"}),comments:[],applicationExtensions:[]},p=new ov(new Uint8ClampedArray(l));if(p.getString(6)!=="GIF89a")throw new Error("not a supported GIF file");u.width=p.nextTwoBytes(),u.height=p.nextTwoBytes();const h=p.nextByte(),f=(h&128)===128;u.colorRes=(h&112)>>>4,u.sortFlag=(h&8)===8;const g=1<<(h&7)+1,E=p.nextByte();u.pixelAspectRatio=p.nextByte(),u.pixelAspectRatio!==0&&(u.pixelAspectRatio=(u.pixelAspectRatio+15)/64),f&&(u.globalColorTable=dS(p,g));const R=(()=>{try{return new ImageData(u.width,u.height,{colorSpace:"srgb"})}catch(U){if(U instanceof DOMException&&U.name==="IndexSizeError")return null;throw U}})();if(R==null)throw new Error("GIF frame size is to large");const{r:y,g:w,b}=u.globalColorTable[E];R.data.set(f?[y,w,b,255]:[0,0,0,0]);for(let U=4;U<R.data.length;U*=2)R.data.copyWithin(U,0,U);u.backgroundImage=R;let X=-1,N=!0,v=-1;const A=U=>(U&&(N=!0),X),k=U=>(U!=null&&(v=U),v);try{do N&&(u.frames.push({left:0,top:0,width:0,height:0,disposalMethod:nn.Replace,image:new ImageData(1,1,{colorSpace:"srgb"}),plainTextData:null,userInputDelayFlag:!1,delayTime:0,sortFlag:!1,localColorTable:[],reserved:0,GCreserved:0}),X++,v=-1,N=!1);while(!await cv(p,u,r,A,k,t));u.frames.length--;for(const U of u.frames){if(U.userInputDelayFlag&&U.delayTime===0){u.totalTime=1/0;break}u.totalTime+=U.delayTime}return u}catch(U){throw U instanceof EvalError?new Error(`error while parsing frame ${X} "${U.message}"`):U}}function pv(i){const{context:t,radius:r,particle:s,delta:l}=i,u=s.image;if(!(u!=null&&u.gifData)||!u.gif)return;const p=new OffscreenCanvas(u.gifData.width,u.gifData.height),h=p.getContext("2d");if(!h)throw new Error("could not create offscreen canvas context");h.imageSmoothingQuality="low",h.imageSmoothingEnabled=!1,h.clearRect(Ct.x,Ct.y,p.width,p.height),s.gifLoopCount===void 0&&(s.gifLoopCount=u.gifLoopCount??wd);let f=s.gifFrame??rv;const g={x:-u.gifData.width*jm,y:-u.gifData.height*jm},E=u.gifData.frames[f];if(s.gifTime===void 0&&(s.gifTime=sv),!!E.bitmap){switch(t.scale(r/u.gifData.width,r/u.gifData.height),E.disposalMethod){case nn.UndefinedA:case nn.UndefinedB:case nn.UndefinedC:case nn.UndefinedD:case nn.Replace:h.drawImage(E.bitmap,E.left,E.top),t.drawImage(p,g.x,g.y),h.clearRect(Ct.x,Ct.y,p.width,p.height);break;case nn.Combine:h.drawImage(E.bitmap,E.left,E.top),t.drawImage(p,g.x,g.y);break;case nn.RestoreBackground:h.drawImage(E.bitmap,E.left,E.top),t.drawImage(p,g.x,g.y),h.clearRect(Ct.x,Ct.y,p.width,p.height),u.gifData.globalColorTable.length?h.putImageData(u.gifData.backgroundImage,g.x,g.y):h.putImageData(u.gifData.frames[km].image,g.x+E.left,g.y+E.top);break;case nn.RestorePrevious:{const R=h.getImageData(Ct.x,Ct.y,p.width,p.height);h.drawImage(E.bitmap,E.left,E.top),t.drawImage(p,g.x,g.y),h.clearRect(Ct.x,Ct.y,p.width,p.height),h.putImageData(R,Ct.x,Ct.y)}break}if(s.gifTime+=l.value,s.gifTime>E.delayTime){if(s.gifTime-=E.delayTime,++f>=u.gifData.frames.length){if(--s.gifLoopCount<=wd)return;f=km,h.clearRect(Ct.x,Ct.y,p.width,p.height)}s.gifFrame=f}t.scale(u.gifData.width/r,u.gifData.height/r)}}async function hv(i){if(i.type!=="gif"){await Rl(i);return}i.loading=!0;try{i.gifData=await dv(i.source),i.gifLoopCount=uv(i.gifData)??wd,i.gifLoopCount||(i.gifLoopCount=1/0)}catch{i.error=!0}i.loading=!1}const fv=2,gv=1,mv=12,Ev=1;class Sv{constructor(t){this.validTypes=["image","images"],this.loadImageShape=async r=>{if(!this._engine.loadImage)throw new Error(`${Kt} image shape not initialized`);await this._engine.loadImage({gif:r.gif,name:r.name,replaceColor:r.replaceColor??!1,src:r.src})},this._engine=t}addImage(t){this._engine.images||(this._engine.images=[]),this._engine.images.push(t)}draw(t){const{context:r,radius:s,particle:l,opacity:u}=t,p=l.image,h=p==null?void 0:p.element;if(p){if(r.globalAlpha=u,p.gif&&p.gifData)pv(t);else if(h){const f=p.ratio,g={x:-s,y:-s},E=s*fv;r.drawImage(h,g.x,g.y,E,E/f)}r.globalAlpha=gv}}getSidesCount(){return mv}async init(t){const r=t.actualOptions;if(!(!r.preload||!this._engine.loadImage))for(const s of r.preload)await this._engine.loadImage(s)}loadShape(t){if(t.shape!=="image"&&t.shape!=="images")return;this._engine.images||(this._engine.images=[]);const r=t.shapeData;if(!r)return;this._engine.images.find(l=>l.name===r.name||l.source===r.src)||this.loadImageShape(r).then(()=>{this.loadShape(t)})}particleInit(t,r){if(r.shape!=="image"&&r.shape!=="images")return;this._engine.images||(this._engine.images=[]);const s=this._engine.images,l=r.shapeData;if(!l)return;const u=r.getFillColor(),p=s.find(f=>f.name===l.name||f.source===l.src);if(!p)return;const h=l.replaceColor??p.replaceColor;if(p.loading){setTimeout(()=>{this.particleInit(t,r)});return}(async()=>{let f;p.svgData&&u?f=await iv(p,l,u,r):f={color:u,data:p,element:p.element,gif:p.gif,gifData:p.gifData,gifLoopCount:p.gifLoopCount,loaded:!0,ratio:l.width&&l.height?l.width/l.height:p.ratio??Ev,replaceColor:h,source:l.src},f.ratio||(f.ratio=1);const g=l.fill??r.shapeFill,E=l.close??r.shapeClose,R={image:f,fill:g,close:E};r.image=R.image,r.shapeFill=R.fill,r.shapeClose=R.close})()}}class Rv{constructor(){this.src="",this.gif=!1}load(t){Q(t)||(t.gif!==void 0&&(this.gif=t.gif),t.height!==void 0&&(this.height=t.height),t.name!==void 0&&(this.name=t.name),t.replaceColor!==void 0&&(this.replaceColor=t.replaceColor),t.src!==void 0&&(this.src=t.src),t.width!==void 0&&(this.width=t.width))}}class Iv{constructor(t){this.id="imagePreloader",this._engine=t}async getPlugin(){return await Promise.resolve(),{}}loadOptions(t,r){if(!(r!=null&&r.preload))return;t.preload||(t.preload=[]);const s=t.preload;for(const l of r.preload){const u=s.find(p=>p.name===l.name||p.src===l.src);if(u)u.load(l);else{const p=new Rv;p.load(l),s.push(p)}}}needsPlugin(){return!0}}const yv=3;function bv(i){i.loadImage||(i.loadImage=async t=>{if(!t.name&&!t.src)throw new Error(`${Kt} no image source provided`);if(i.images||(i.images=[]),!i.images.find(r=>r.name===t.name||r.source===t.src))try{const r={gif:t.gif??!1,name:t.name??t.src,source:t.src,type:t.src.substring(t.src.length-yv),error:!1,loading:!0,replaceColor:t.replaceColor,ratio:t.width&&t.height?t.width/t.height:void 0};i.images.push(r);let s;t.gif?s=hv:s=t.replaceColor?nv:Rl,await s(r)}catch{throw new Error(`${Kt} ${t.name??t.src} not found`)}})}async function Cv(i,t=!0){i.checkVersion("3.8.1"),bv(i);const r=new Iv(i);await i.addPlugin(r,t),await i.addShape(new Sv(i),t)}class wv extends oo{constructor(){super(),this.sync=!1}load(t){Q(t)||(super.load(t),t.sync!==void 0&&(this.sync=t.sync))}}class _v extends oo{constructor(){super(),this.sync=!1}load(t){Q(t)||(super.load(t),t.sync!==void 0&&(this.sync=t.sync))}}class Tv{constructor(){this.count=0,this.delay=new wv,this.duration=new _v}load(t){Q(t)||(t.count!==void 0&&(this.count=t.count),this.delay.load(t.delay),this.duration.load(t.duration))}}const Go=0,vv=-1,zm=0,Gm=0;function xv(i,t,r){if(!i.life)return;const s=i.life;let l=!1;if(i.spawning)if(s.delayTime+=t.value,s.delayTime>=i.life.delay)l=!0,i.spawning=!1,s.delayTime=Go,s.time=Go;else return;if(s.duration===vv||i.spawning||(l?s.time=Go:s.time+=t.value,s.time<s.duration))return;if(s.time=Go,i.life.count>zm&&i.life.count--,i.life.count===zm){i.destroy();return}const u=ye(Gm,r.width),p=ye(Gm,r.width);i.position.x=Bt(u),i.position.y=Bt(p),i.spawning=!0,s.delayTime=Go,s.time=Go,i.reset();const h=i.options.life;h&&(s.delay=ie(h.delay.value)*dt,s.duration=ie(h.duration.value)*dt)}const Gi=0,Qm=1,Vm=-1;class Pv{constructor(t){this.container=t}init(t){const r=this.container,s=t.options,l=s.life;l&&(t.life={delay:r.retina.reduceFactor?ie(l.delay.value)*(l.delay.sync?Qm:Oe())/r.retina.reduceFactor*dt:Gi,delayTime:Gi,duration:r.retina.reduceFactor?ie(l.duration.value)*(l.duration.sync?Qm:Oe())/r.retina.reduceFactor*dt:Gi,time:Gi,count:l.count},t.life.duration<=Gi&&(t.life.duration=Vm),t.life.count<=Gi&&(t.life.count=Vm),t.life&&(t.spawning=t.life.delay>Gi))}isEnabled(t){return!t.destroyed}loadOptions(t,...r){t.life||(t.life=new Tv);for(const s of r)t.life.load(s==null?void 0:s.life)}update(t,r){!this.isEnabled(t)||!t.life||xv(t,r,this.container.canvas.size)}}async function Xv(i,t=!0){i.checkVersion("3.8.1"),await i.addParticleUpdater("life",async r=>Promise.resolve(new Pv(r)),t)}function Nv(i){const{context:t,particle:r,radius:s}=i,l=r.shapeData,u=0;t.moveTo(-s,u),t.lineTo(s,u),t.lineCap=(l==null?void 0:l.cap)??"butt"}const Ov=1;class Fv{constructor(){this.validTypes=["line"]}draw(t){Nv(t)}getSidesCount(){return Ov}}async function Dv(i,t=!0){i.checkVersion("3.8.1"),await i.addShape(new Fv,t)}const Ym=.5;class Hv{init(){}isEnabled(t){return!io()&&!t.destroyed&&t.container.actualOptions.interactivity.events.onHover.parallax.enable}move(t){const r=t.container,s=r.actualOptions,l=s.interactivity.events.onHover.parallax;if(io()||!l.enable)return;const u=l.force,p=r.interactivity.mouse.position;if(!p)return;const h=r.canvas.size,f={x:h.width*Ym,y:h.height*Ym},g=l.smooth,E=t.getRadius()/u,R={x:(p.x-f.x)*E,y:(p.y-f.y)*E},{offset:y}=t;y.x+=(R.x-y.x)/g,y.y+=(R.y-y.y)/g}}async function Av(i,t=!0){i.checkVersion("3.8.1"),await i.addMover("parallax",()=>Promise.resolve(new Hv),t)}const Km=1e3,Lv=1;class Bv extends Wd{constructor(t){super(t)}clear(){}init(){}interact(t){const r=this.container;t.attractDistance===void 0&&(t.attractDistance=ie(t.options.move.attract.distance)*r.retina.pixelRatio);const s=t.attractDistance,l=t.getPosition(),u=r.particles.quadTree.queryCircle(l,s);for(const p of u){if(t===p||!p.options.move.attract.enable||p.destroyed||p.spawning)continue;const h=p.getPosition(),{dx:f,dy:g}=Et(l,h),E=t.options.move.attract.rotate,R=f/(E.x*Km),y=g/(E.y*Km),w=p.size.value/t.size.value,b=Lv/w;t.velocity.x-=R*w,t.velocity.y-=y*w,p.velocity.x+=R*b,p.velocity.y+=y*b}}isEnabled(t){return t.options.move.attract.enable}reset(){}}async function Mv(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("particlesAttract",r=>Promise.resolve(new Bv(r)),t)}const qv=.5,Uv=10,Wv=0;function Zm(i,t,r,s,l,u){const p=Zt(i.options.collisions.absorb.speed*l.factor/Uv,Wv,s);i.size.value+=p*qv,r.size.value-=p,s<=u&&(r.size.value=0,r.destroy())}function jv(i,t,r,s){const l=i.getRadius(),u=t.getRadius();l===void 0&&u!==void 0?i.destroy():l!==void 0&&u===void 0?t.destroy():l!==void 0&&u!==void 0&&(l>=u?Zm(i,l,t,u,r,s):Zm(t,u,i,l,r,s))}const Jm=i=>{i.collisionMaxSpeed===void 0&&(i.collisionMaxSpeed=ie(i.options.collisions.maxSpeed)),i.velocity.length>i.collisionMaxSpeed&&(i.velocity.length=i.collisionMaxSpeed)};function pS(i,t){jE(Id(i),Id(t)),Jm(i),Jm(t)}function kv(i,t){!i.unbreakable&&!t.unbreakable&&pS(i,t),i.getRadius()===void 0&&t.getRadius()!==void 0?i.destroy():i.getRadius()!==void 0&&t.getRadius()===void 0?t.destroy():i.getRadius()!==void 0&&t.getRadius()!==void 0&&(i.getRadius()>=t.getRadius()?t:i).destroy()}function zv(i,t,r,s){switch(i.options.collisions.mode){case er.absorb:{jv(i,t,r,s);break}case er.bounce:{pS(i,t);break}case er.destroy:{kv(i,t);break}}}const Gv=2;class Qv extends Wd{constructor(t){super(t)}clear(){}init(){}interact(t,r){if(t.destroyed||t.spawning)return;const s=this.container,l=t.getPosition(),u=t.getRadius(),p=s.particles.quadTree.queryCircle(l,u*Gv);for(const h of p){if(t===h||!h.options.collisions.enable||t.options.collisions.mode!==h.options.collisions.mode||h.destroyed||h.spawning)continue;const f=h.getPosition(),g=h.getRadius();if(Math.abs(Math.round(l.z)-Math.round(f.z))>u+g)continue;const E=Mt(l,f),R=u+g;E>R||zv(t,h,r,s.retina.pixelRatio)}}isEnabled(t){return t.options.collisions.enable}reset(){}}async function Vv(i,t=!0){i.checkVersion("3.8.1"),await i.addInteractor("particlesCollisions",r=>Promise.resolve(new Qv(r)),t)}const sd=2;class Yv extends ht{constructor(t,r,s,l){super(t,r,s),this.canvasSize=l,this.canvasSize={...l}}contains(t){const{width:r,height:s}=this.canvasSize,{x:l,y:u}=t;return super.contains(t)||super.contains({x:l-r,y:u})||super.contains({x:l-r,y:u-s})||super.contains({x:l,y:u-s})}intersects(t){if(super.intersects(t))return!0;const r=t,s=t,l={x:t.position.x-this.canvasSize.width,y:t.position.y-this.canvasSize.height};if(s.radius!==void 0){const u=new ht(l.x,l.y,s.radius*sd);return super.intersects(u)}else if(r.size!==void 0){const u=new ln(l.x,l.y,r.size.width*sd,r.size.height*sd);return super.intersects(u)}return!1}}class Kv{constructor(){this.blur=5,this.color=new ft,this.color.value="#000",this.enable=!1}load(t){Q(t)||(t.blur!==void 0&&(this.blur=t.blur),this.color=ft.create(this.color,t.color),t.enable!==void 0&&(this.enable=t.enable))}}class Zv{constructor(){this.enable=!1,this.frequency=1}load(t){Q(t)||(t.color!==void 0&&(this.color=ft.create(this.color,t.color)),t.enable!==void 0&&(this.enable=t.enable),t.frequency!==void 0&&(this.frequency=t.frequency),t.opacity!==void 0&&(this.opacity=t.opacity))}}class Jv{constructor(){this.blink=!1,this.color=new ft,this.color.value="#fff",this.consent=!1,this.distance=100,this.enable=!1,this.frequency=1,this.opacity=1,this.shadow=new Kv,this.triangles=new Zv,this.width=1,this.warp=!1}load(t){Q(t)||(t.id!==void 0&&(this.id=t.id),t.blink!==void 0&&(this.blink=t.blink),this.color=ft.create(this.color,t.color),t.consent!==void 0&&(this.consent=t.consent),t.distance!==void 0&&(this.distance=t.distance),t.enable!==void 0&&(this.enable=t.enable),t.frequency!==void 0&&(this.frequency=t.frequency),t.opacity!==void 0&&(this.opacity=t.opacity),this.shadow.load(t.shadow),this.triangles.load(t.triangles),t.width!==void 0&&(this.width=t.width),t.warp!==void 0&&(this.warp=t.warp))}}const $m=2,$v=1,Ka={x:0,y:0},ex=0;function tx(i,t,r,s,l){const{dx:u,dy:p,distance:h}=Et(i,t);if(!l||h<=r)return h;const f={x:Math.abs(u),y:Math.abs(p)},g={x:Math.min(f.x,s.width-f.x),y:Math.min(f.y,s.height-f.y)};return Math.sqrt(g.x**$m+g.y**$m)}class nx extends Wd{constructor(t,r){super(t),this._setColor=s=>{if(!s.options.links)return;const l=this._linkContainer,u=s.options.links;let p=u.id===void 0?l.particles.linksColor:l.particles.linksColors.get(u.id);if(p)return;const h=u.color;p=KE(this._engine,h,u.blink,u.consent),u.id===void 0?l.particles.linksColor=p:l.particles.linksColors.set(u.id,p)},this._linkContainer=t,this._engine=r}clear(){}init(){this._linkContainer.particles.linksColor=void 0,this._linkContainer.particles.linksColors=new Map}interact(t){if(!t.options.links)return;t.links=[];const r=t.getPosition(),s=this.container,l=s.canvas.size;if(r.x<Ka.x||r.y<Ka.y||r.x>l.width||r.y>l.height)return;const u=t.options.links,p=u.opacity,h=t.retina.linksDistance??ex,f=u.warp;let g;f?g=new Yv(r.x,r.y,h,l):g=new ht(r.x,r.y,h);const E=s.particles.quadTree.query(g);for(const R of E){const y=R.options.links;if(t===R||!(y!=null&&y.enable)||u.id!==y.id||R.spawning||R.destroyed||!R.links||t.links.some(N=>N.destination===R)||R.links.some(N=>N.destination===t))continue;const w=R.getPosition();if(w.x<Ka.x||w.y<Ka.y||w.x>l.width||w.y>l.height)continue;const b=tx(r,w,h,l,f&&y.warp);if(b>h)continue;const X=($v-b/h)*p;this._setColor(t),t.links.push({destination:R,opacity:X})}}isEnabled(t){var r;return!!((r=t.options.links)!=null&&r.enable)}loadParticlesOptions(t,...r){t.links||(t.links=new Jv);for(const s of r)t.links.load(s==null?void 0:s.links)}reset(){}}async function ix(i,t=!0){await i.addInteractor("particlesLinks",async r=>Promise.resolve(new nx(r,i)),t)}function ox(i,t,r,s){i.beginPath(),i.moveTo(t.x,t.y),i.lineTo(r.x,r.y),i.lineTo(s.x,s.y),i.closePath()}function rx(i){let t=!1;const{begin:r,end:s,engine:l,maxDistance:u,context:p,canvasSize:h,width:f,backgroundMask:g,colorLine:E,opacity:R,links:y}=i;if(Mt(r,s)<=u)fs(p,r,s),t=!0;else if(y.warp){let b,X;const N={x:s.x-h.width,y:s.y},v=Et(r,N);if(v.distance<=u){const A=r.y-v.dy/v.dx*r.x;b={x:0,y:A},X={x:h.width,y:A}}else{const A={x:s.x,y:s.y-h.height},k=Et(r,A);if(k.distance<=u){const j=-(r.y-k.dy/k.dx*r.x)/(k.dy/k.dx);b={x:j,y:0},X={x:j,y:h.height}}else{const U={x:s.x-h.width,y:s.y-h.height},j=Et(r,U);if(j.distance<=u){const K=r.y-j.dy/j.dx*r.x;b={x:-K/(j.dy/j.dx),y:K},X={x:b.x+h.width,y:b.y+h.height}}}}b&&X&&(fs(p,r,b),fs(p,s,X),t=!0)}if(!t)return;p.lineWidth=f,g.enable&&(p.globalCompositeOperation=g.composite),p.strokeStyle=Wn(E,R);const{shadow:w}=y;if(w.enable){const b=sn(l,w.color);b&&(p.shadowBlur=w.blur,p.shadowColor=Wn(b))}p.stroke()}function sx(i){const{context:t,pos1:r,pos2:s,pos3:l,backgroundMask:u,colorTriangle:p,opacityTriangle:h}=i;ox(t,r,s,l),u.enable&&(t.globalCompositeOperation=u.composite),t.fillStyle=Wn(p,h),t.fill()}function ax(i){return i.sort((t,r)=>t-r),i.join("_")}function eE(i,t){const r=ax(i.map(l=>l.id));let s=t.get(r);return s===void 0&&(s=Oe(),t.set(r,s)),s}const tE=0,ad=0,nE=0,lx=.5,cx=1;class ux{constructor(t,r){this._drawLinkLine=(s,l)=>{const u=s.options.links;if(!(u!=null&&u.enable))return;const p=this._container,h=p.actualOptions,f=l.destination,g=s.getPosition(),E=f.getPosition();let R=l.opacity;p.canvas.draw(y=>{var A;let w;const b=(A=s.options.twinkle)==null?void 0:A.lines;if(b!=null&&b.enable){const k=b.frequency,U=sn(this._engine,b.color);Oe()<k&&U&&(w=U,R=ie(b.opacity))}if(!w){const k=u.id!==void 0?p.particles.linksColors.get(u.id):p.particles.linksColor;w=bd(s,f,k)}if(!w)return;const X=s.retina.linksWidth??ad,N=s.retina.linksDistance??nE,{backgroundMask:v}=h;rx({context:y,width:X,begin:g,end:E,engine:this._engine,maxDistance:N,canvasSize:p.canvas.size,links:u,backgroundMask:v,colorLine:w,opacity:R})})},this._drawLinkTriangle=(s,l,u)=>{const p=s.options.links;if(!(p!=null&&p.enable))return;const h=p.triangles;if(!h.enable)return;const f=this._container,g=f.actualOptions,E=l.destination,R=u.destination,y=h.opacity??(l.opacity+u.opacity)*lx;y<=tE||f.canvas.draw(w=>{const b=s.getPosition(),X=E.getPosition(),N=R.getPosition(),v=s.retina.linksDistance??nE;if(Mt(b,X)>v||Mt(N,X)>v||Mt(N,b)>v)return;let A=sn(this._engine,h.color);if(!A){const k=p.id!==void 0?f.particles.linksColors.get(p.id):f.particles.linksColor;A=bd(s,E,k)}A&&sx({context:w,pos1:b,pos2:X,pos3:N,backgroundMask:g.backgroundMask,colorTriangle:A,opacityTriangle:y})})},this._drawTriangles=(s,l,u,p)=>{var g,E,R;const h=u.destination;if(!((g=s.links)!=null&&g.triangles.enable&&((E=h.options.links)!=null&&E.triangles.enable)))return;const f=(R=h.links)==null?void 0:R.filter(y=>{const w=this._getLinkFrequency(h,y.destination);return h.options.links&&w<=h.options.links.frequency&&p.findIndex(X=>X.destination===y.destination)>=0});if(f!=null&&f.length)for(const y of f){const w=y.destination;this._getTriangleFrequency(l,h,w)>s.links.triangles.frequency||this._drawLinkTriangle(l,u,y)}},this._getLinkFrequency=(s,l)=>eE([s,l],this._freqs.links),this._getTriangleFrequency=(s,l,u)=>eE([s,l,u],this._freqs.triangles),this._container=t,this._engine=r,this._freqs={links:new Map,triangles:new Map}}drawParticle(t,r){const{links:s,options:l}=r;if(!(s!=null&&s.length))return;const u=s.filter(p=>l.links&&(l.links.frequency>=cx||this._getLinkFrequency(r,p.destination)<=l.links.frequency));for(const p of u)this._drawTriangles(l,r,p,u),p.opacity>tE&&(r.retina.linksWidth??ad)>ad&&this._drawLinkLine(r,p)}async init(){this._freqs.links=new Map,this._freqs.triangles=new Map,await Promise.resolve()}particleCreated(t){if(t.links=[],!t.options.links)return;const r=this._container.retina.pixelRatio,{retina:s}=t,{distance:l,width:u}=t.options.links;s.linksDistance=l*r,s.linksWidth=u*r}particleDestroyed(t){t.links=[]}}class dx{constructor(t){this.id="links",this._engine=t}getPlugin(t){return Promise.resolve(new ux(t,this._engine))}loadOptions(){}needsPlugin(){return!0}}async function px(i,t=!0){const r=new dx(i);await i.addPlugin(r,t)}async function hx(i,t=!0){i.checkVersion("3.8.1"),await ix(i,t),await px(i,t)}const fx=180,Za={x:0,y:0},gx=2;function mx(i,t,r){const{context:s}=i,l=r.count.numerator*r.count.denominator,u=r.count.numerator/r.count.denominator,p=fx*(u-gx)/u,h=Math.PI-Si(p);if(s){s.beginPath(),s.translate(t.x,t.y),s.moveTo(Za.x,Za.y);for(let f=0;f<l;f++)s.lineTo(r.length,Za.y),s.translate(r.length,Za.y),s.rotate(h)}}const Ex=5;class hS{draw(t){const{particle:r,radius:s}=t,l=this.getCenter(r,s),u=this.getSidesData(r,s);mx(t,l,u)}getSidesCount(t){const r=t.shapeData;return Math.round(ie((r==null?void 0:r.sides)??Ex))}}const iE=3.5,oE=2.66,Sx=3;class Rx extends hS{constructor(){super(...arguments),this.validTypes=["polygon"]}getCenter(t,r){return{x:-r/(t.sides/iE),y:-r/(oE/iE)}}getSidesData(t,r){const s=t.sides;return{count:{denominator:1,numerator:s},length:r*oE/(s/Sx)}}}const Ix=1.66,yx=3,bx=2;class Cx extends hS{constructor(){super(...arguments),this.validTypes=["triangle"]}getCenter(t,r){return{x:-r,y:r/Ix}}getSidesCount(){return yx}getSidesData(t,r){const s=r*bx;return{count:{denominator:2,numerator:3},length:s}}}async function wx(i,t=!0){i.checkVersion("3.8.1"),await i.addShape(new Rx,t)}async function _x(i,t=!0){i.checkVersion("3.8.1"),await i.addShape(new Cx,t)}async function Tx(i,t=!0){i.checkVersion("3.8.1"),await wx(i,t),await _x(i,t)}class vx{constructor(){this.enable=!1,this.speed=0,this.decay=0,this.sync=!1}load(t){Q(t)||(t.enable!==void 0&&(this.enable=t.enable),t.speed!==void 0&&(this.speed=ye(t.speed)),t.decay!==void 0&&(this.decay=ye(t.decay)),t.sync!==void 0&&(this.sync=t.sync))}}class xx extends oo{constructor(){super(),this.animation=new vx,this.direction=on.clockwise,this.path=!1,this.value=0}load(t){Q(t)||(super.load(t),t.direction!==void 0&&(this.direction=t.direction),this.animation.load(t.animation),t.path!==void 0&&(this.path=t.path))}}const fS=2,Px=Math.PI*fS,Xx=1,Nx=360;class Ox{constructor(t){this.container=t}init(t){const r=t.options.rotate;if(!r)return;t.rotate={enable:r.animation.enable,value:Si(ie(r.value)),min:0,max:Px},t.pathRotation=r.path;let s=r.direction;switch(s===on.random&&(s=Math.floor(Oe()*fS)>0?on.counterClockwise:on.clockwise),s){case on.counterClockwise:case"counterClockwise":t.rotate.status=Ge.decreasing;break;case on.clockwise:t.rotate.status=Ge.increasing;break}const l=r.animation;l.enable&&(t.rotate.decay=Xx-ie(l.decay),t.rotate.velocity=ie(l.speed)/Nx*this.container.retina.reduceFactor,l.sync||(t.rotate.velocity*=Oe())),t.rotation=t.rotate.value}isEnabled(t){const r=t.options.rotate;return r?!t.destroyed&&!t.spawning&&(!!r.value||r.animation.enable||r.path):!1}loadOptions(t,...r){t.rotate||(t.rotate=new xx);for(const s of r)t.rotate.load(s==null?void 0:s.rotate)}update(t,r){this.isEnabled(t)&&(t.isRotating=!!t.rotate,t.rotate&&(Bd(t,t.rotate,!1,to.none,r),t.rotation=t.rotate.value))}}async function Fx(i,t=!0){i.checkVersion("3.8.1"),await i.addParticleUpdater("rotate",r=>Promise.resolve(new Ox(r)),t)}const Dx=2,Hx=Math.sqrt(Dx),Ax=2;function Lx(i){const{context:t,radius:r}=i,s=r/Hx,l=s*Ax;t.rect(-s,-s,l,l)}const Bx=4;class Mx{constructor(){this.validTypes=["edge","square"]}draw(t){Lx(t)}getSidesCount(){return Bx}}async function qx(i,t=!0){i.checkVersion("3.8.1"),await i.addShape(new Mx,t)}const Ux=2,Qo={x:0,y:0};function Wx(i){const{context:t,particle:r,radius:s}=i,l=r.sides,u=r.starInset??Ux;t.moveTo(Qo.x,Qo.y-s);for(let p=0;p<l;p++)t.rotate(Math.PI/l),t.lineTo(Qo.x,Qo.y-s*u),t.rotate(Math.PI/l),t.lineTo(Qo.x,Qo.y-s)}const jx=2,kx=5;class zx{constructor(){this.validTypes=["star"]}draw(t){Wx(t)}getSidesCount(t){const r=t.shapeData;return Math.round(ie((r==null?void 0:r.sides)??kx))}particleInit(t,r){const s=r.shapeData;r.starInset=ie((s==null?void 0:s.inset)??jx)}}async function Gx(i,t=!0){i.checkVersion("3.8.1"),await i.addShape(new zx,t)}const Qx=1;class Vx{constructor(t,r){this._container=t,this._engine=r}init(t){var p;const r=this._container,s=t.options,l=rn(s.stroke,t.id,s.reduceDuplicates);t.strokeWidth=ie(l.width)*r.retina.pixelRatio,t.strokeOpacity=ie(l.opacity??Qx),t.strokeAnimation=(p=l.color)==null?void 0:p.animation;const u=Es(this._engine,l.color)??t.getFillColor();u&&(t.strokeColor=ZE(u,t.strokeAnimation,r.retina.reduceFactor))}isEnabled(t){const r=t.strokeAnimation,{strokeColor:s}=t;return!t.destroyed&&!t.spawning&&!!r&&((s==null?void 0:s.h.value)!==void 0&&s.h.enable||(s==null?void 0:s.s.value)!==void 0&&s.s.enable||(s==null?void 0:s.l.value)!==void 0&&s.l.enable)}update(t,r){this.isEnabled(t)&&JE(t.strokeColor,r)}}async function Yx(i,t=!0){i.checkVersion("3.8.1"),await i.addParticleUpdater("strokeColor",r=>Promise.resolve(new Vx(r,i)),t)}async function Kx(i,t=!0){i.checkVersion("3.8.1"),await Av(i,!1),await q_(i,!1),await Y_(i,!1),await iT(i,!1),await hT(i,!1),await CT(i,!1),await TT(i,!1),await NT(i,!1),await HT(i,!1),await GT(i,!1),await ZT(i,!1),await Mv(i,!1),await Vv(i,!1),await hx(i,!1),await X_(i,!1),await F_(i,!1),await Cv(i,!1),await Dv(i,!1),await Tx(i,!1),await qx(i,!1),await Gx(i,!1),await Xv(i,!1),await Fx(i,!1),await Yx(i,!1),await P_(i,t)}const Zx=i=>{const[t,r]=P.useState(!1);P.useEffect(()=>{Hw(async u=>{await Kx(u)}).then(()=>{r(!0)})},[]);const s=u=>{console.log(u)},l=P.useMemo(()=>({fpsLimit:120,interactivity:{events:{onClick:{enable:!0,mode:"repulse"},onHover:{enable:!0,mode:"grab"}},modes:{push:{distance:200,duration:15},grab:{distance:150}}},particles:{color:{value:"#FFFFFF"},links:{color:"#FFFFFF",distance:150,enable:!0,opacity:.3,width:1},move:{direction:"none",enable:!0,outModes:{default:"bounce"},random:!0,speed:1,straight:!1},number:{density:{enable:!0},value:150},opacity:{value:1},shape:{type:"circle"},size:{value:{min:1,max:3}}},detectRetina:!0}),[]);return I.jsx(Dw,{id:i.id,init:s,options:l})},Jx=()=>I.jsxs("div",{className:"flex flex-col items-center mt-24 my-12 lg:mt-32",children:[I.jsx("div",{className:"absolute inset-0 z-[-1]",children:I.jsx(Zx,{id:"particles"})}),I.jsxs("h1",{className:"text-4xl sm:text-6xl lg:text-7xl text-center tracking-wide",children:[I.jsxs("span",{className:"bg-gradient-to-r from-orange-500 to-red-700 text-transparent bg-clip-text",children:[" ","Learn SPARQL"]})," ","by example ",I.jsx("br",{}),"with SPARQL Fest"]}),I.jsxs("p",{className:"mt-10 text-lg text-center text-neutral-400 max-w-4xl",children:["Join a collaborative project around the Semantic Web.",I.jsx("br",{}),"SPARQL Fest centralises SPARQL queries shared by the community to make it easier to learn by example, make the most of everyone's contributions and develop your skills, one query at a time."]}),I.jsxs("div",{className:"flex-justify-center my-10",children:[I.jsx(ut,{to:"/sparql-queries",className:"bg-gradient-to-r from-orange-500 to-orange-800 py-3 px-4 mx-3 rounded-md",children:"See all queries"}),I.jsx("a",{href:"#about",className:"py-3 px-4 mx-3 rounded-md border",children:"More about us"})]})]});function gS(i){var t,r,s="";if(typeof i=="string"||typeof i=="number")s+=i;else if(typeof i=="object")if(Array.isArray(i)){var l=i.length;for(t=0;t<l;t++)i[t]&&(r=gS(i[t]))&&(s&&(s+=" "),s+=r)}else for(r in i)i[r]&&(s&&(s+=" "),s+=r);return s}function Ja(){for(var i,t,r=0,s="",l=arguments.length;r<l;r++)(i=arguments[r])&&(t=gS(i))&&(s&&(s+=" "),s+=t);return s}const bn=({title:i,description:t,variant:r="dark",id:s,bg:l=!0,children:u})=>{const p=r==="dark";return I.jsxs("section",{id:s,className:Ja("py-16 pb-16 px-[5vw] md:px-[7vw] lg:px-[10vw] font-sans clip-path-custom",l&&(p?"bg-stone-900 text-white":"bg-white text-black")),children:[I.jsxs("div",{className:"text-center mb-8",children:[I.jsx("h2",{className:Ja("text-3xl sm:text-4xl font-bold",p?"text-white":"text-stone-900"),children:i}),I.jsx("div",{className:Ja("w-40 h-1 mx-auto mt-2",p?"bg-orange-700":"bg-orange-400")}),I.jsx("p",{className:Ja("mt-4 text-lg font-semibold",p?"text-neutral-400":"text-neutral-700"),children:t})]}),u&&I.jsx("div",{className:"mt-10",children:u})]})},$x="/Sparql-fest/assets/network-bg-1-BA5TYA3w.jpg",eP="/Sparql-fest/assets/network-bg-2-DmVfYs01.avif",tP="/Sparql-fest/assets/network-bg-3-B7_7yXeW.jpg",nP=[{name:"Lyme Disease ORDO Mappings",slug:"ordo-lyme-mappings",date:"08/06/2025",description:"Obtain all external identifiers (via oboInOwl:hasDbXref) associated with Lyme disease in ORDO.",context:"This query seeks to retrieve all external databases linked to the ORDO entry for Lyme disease. This makes it possible to explore the connectivity of this entity to other repositories.",inidces:["Charger ORDO","Utilisez ordo:Orphanet_91546 comme point d'entrée.","Recherchez les prédicats oboInOwl:hasDbXref."],query:`PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
PREFIX ordo:     <http://www.orpha.net/ORDO/>

SELECT ?object
WHERE {
  ordo:Orphanet_91546 oboInOwl:hasDbXref ?object .
}

      `,ontologies:["ORDO","OBOInOWL"],sparqlConcepts:["PREFIX","SELECT"],category:"level 0",rdfResultExample:""},{name:"Getting a concept label from ORPHAcode.",slug:"label-from-orphacode",date:"08/06/2025",description:"Retrieve the label (official name) associated with a rare disease identifier.",context:"We want to display the official name of a specific rare disease based on its identifier. This is useful for making data more understandable to users, such as when generating reports or displaying disease information in user interfaces.",inidces:["Start by identifying the identifier of the concept you are interested in (e.g., Orphanet_558).","Ask for its associated label using a basic SELECT query.","Think of a triple where the subject is the disease identifier and the object is its name."],query:`PREFIX ordo:<http://www.orpha.net/ORDO/>
PREFIX w3: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?label
WHERE {
  ordo:Orphanet_558 w3:label ?label
}`,ontologies:["ORDO","RDFS"],sparqlConcepts:["PREFIX","SELECT"],category:"level 0",rdfResultExample:""},{name:" Getting the genetic material linked to more than 10 disorders.",slug:"genes-with-many-disorders",date:"08/06/2025",description:"Find the genetic elements that are each linked to more than 10 different disorders, and return their names along with the number of associated disorders.",context:"We want to identify elements of genetic material that are associated with a high number of distinct disorders. This kind of analysis can help prioritize genes for further research, diagnostics, or therapeutic development based on their widespread involvement in diseases.",inidces:["Start by counting how many disorders are linked to each genetic element.","Use a subquery to perform this count.","Filter results to keep only those with more than 10 links.","Return the label (name) of each gene and the total number of associated disorders.","Sort the result to show the most frequently involved genes first."],query:`PREFIX ordo: <http://www.orpha.net/ORDO/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>

SELECT ?gene ?geneLab ?nbD
WHERE {
  {
    SELECT ?g (COUNT(?d) AS ?nbD)
    WHERE {
      ?r owl:onProperty ?rel.
      ?g rdfs:label ?gLabel.
      ?g rdfs:subClassOf ?r.
      ?g rdfs:subClassOf ?class.
      ?class rdfs:subClassOf ?sc.
      FILTER (?sc = ordo:Orphanet_C010)
      ?r owl:someValuesFrom ?d.
    }
    GROUP BY ?g
  }

  FILTER (?nbD > 10)

  BIND (?g AS ?gene)
  ?gene rdfs:label ?geneLab.
}
ORDER BY DESC(?nbD)
`,ontologies:["ORDO","RDFS","OWL"],sparqlConcepts:["PREFIX","SELECT","COUNT()","FILTER","GROUP BY","BIND","ORDER BY"],category:"level 1",rdfResultExample:""},{name:"Getting the label and mapping information about a concept.",slug:"concept-label-and-mappings",date:"08/06/2025",description:"Retrieve all information related to a specific disease concept, including its label and any mapping-related metadata that might connect it to other vocabularies or systems.",context:"We want to explore all the available information about a specific disease concept, including its name and how it is connected to related concepts in other systems. This helps build a clearer understanding of how this disease is described and reused in various biomedical knowledge sources.",inidces:["Focus on one concept, using its identifier (e.g. Orphanet_558).","Search for all direct triples where this concept is the subject.","Optionally, look at the object of these triples and retrieve further labels if available (e.g. through mappings).","Use OPTIONAL to explore extra information without breaking the query if it doesn't exist.","Order the results for readability."],query:`PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX dc: <http://purl.org/dc/elements/1.1/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX Orphanet_: <http://www.orpha.net/ORDO/Orphanet_#>
PREFIX ORDO: <http://www.orpha.net/ORDO/>

SELECT ?s ?p ?o 
WHERE {
  ?s ?p ?o.
  OPTIONAL {
    ?o ?m ?v.
    ?v rdfs:label ?e
  }
  FILTER (?s = ORDO:Orphanet_558)
}
ORDER BY ?o
`,ontologies:["ORDO","RDFS","OWL","OBO","OBOInOwl","DC"],sparqlConcepts:["PREFIX","SELECT","Triple pattern","OPTIONAL","FILTER","ORDER BY"],category:"level 1",rdfResultExample:""},{name:"Counting the total number of Disorder-HPO associations in HOOM.",slug:"hoom-hpo-association-count",date:"08/06/2025",description:"Count how many formal associations exist between disorders and Human Phenotype Ontology (HPO) terms in the HOOM model.",context:"The goal is to quantify how many associations exist between diseases and phenotypic traits, which can be crucial for understanding how well a disease is phenotypically described. A higher number of associations can reflect a richer clinical characterization and support improved diagnosis or research.",inidces:["Look for association classes defined in the ontology (rdfs:subClassOf).","These associations are described using logical definitions with owl:equivalentClass and owl:intersectionOf.","Traverse RDF lists using rdf:rest*/rdf:first to access individual elements.","Focus only on counting, no need to return the actual associations."],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT (COUNT(?association) as ?Association_Amount)
WHERE {
    ?association rdfs:subClassOf ?type .
    ?association owl:equivalentClass ?node .
    ?node owl:intersectionOf ?list .
    ?list rdf:rest*/rdf:first ?item .
    ?item owl:onProperty ?Property .
    ?item owl:someValuesFrom ?Id .
}
`,ontologies:["HOOM","RDFS","RDF","OWL"],sparqlConcepts:["PREFIX","SELECT","rdf:rest*/rdf:first","COUNT()"],category:"level 2",rdfResultExample:""},{name:"Counting database cross-references for rare diseases",slug:"rare-diseases-dbxref-count",date:"08/06/2025",description:"List rare diseases with their labels and count how many database cross-references they have. Then, order this list from the diseases with the most references to those with fewer.",context:"We want to identify which rare diseases have the richest connections to external biomedical databases by counting how many database cross-references each disease has. This information helps prioritize diseases that are extensively linked and potentially better studied.",inidces:["Select diseases that have database cross-reference properties.","Retrieve their human-readable labels.","Use aggregation (COUNT) grouped by disease and label.","Sort the results by descending number of cross-references to highlight the most connected diseases."],query:`PREFIX ORDO: <http://www.orpha.net/ORDO/>
PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?disease ?label (COUNT(?dbxref) AS ?mappings) 
WHERE {
  ?disease oboInOwl:hasDbXref ?dbxref ;
           rdfs:label ?label .
}
GROUP BY ?disease ?label
ORDER BY DESC(?mappings)
`,ontologies:["HOOM","RDFS","RDF","OWL"],sparqlConcepts:["PREFIX","SELECT","GROUP BY","ORDER BY DESC","COUNT()"],category:"level 2",rdfResultExample:""},{name:"Most frequently referenced external database identifiers in Orphanet diseases",slug:"frequent-dbxref-orphanet",date:"08/06/2025",description:"Count how many distinct diseases reference each external database identifier and rank these identifiers by their frequency of use.",context:"The goal is to determine which external database identifiers are most commonly linked to rare diseases in Orphanet. This helps identify key reference databases that are widely used across disease descriptions.",inidces:["Focus on the property that links diseases to external database identifiers.","Use COUNT(DISTINCT ?disease) to avoid double counting diseases linked multiple times to the same identifier.","Group results by the external database identifier.","Sort results by descending count to find the most referenced identifiers."],query:`PREFIX ORDO: <http://www.orpha.net/ORDO/>
PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>

SELECT ?dbxref (COUNT(DISTINCT ?disease) AS ?linkedDiseases) WHERE {
  ?disease oboInOwl:hasDbXref ?dbxref .
}
GROUP BY ?dbxref
ORDER BY DESC(?linkedDiseases)
`,ontologies:["ORDO","OboInOwl"],sparqlConcepts:["PREFIX","SELECT","GROUP BY","ORDER BY DESC","COUNT(DISTINCT ...)"],category:"level 2",rdfResultExample:""},{name:"Getting ORDO mappings for Lyme disease",slug:"lyme-ordo-mappings",date:"08/06/2025",description:"Retrieve the external database mappings linked to Lyme disease.",context:"We want to find all external database identifiers that are mapped to Lyme disease within the ORDO dataset. This helps understand the external references and cross-links available for this specific disease.",inidces:["Identify the disease by its ORDO identifier (here Lyme disease = Orphanet_91546).","Select all values associated with the property for database cross-references (hasDbXref).","No grouping or sorting needed since it’s a direct property lookup."],query:`PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
PREFIX ordo: <http://www.orpha.net/ORDO/>

SELECT ?object WHERE { 
    ordo:Orphanet_91546 oboInOwl:hasDbXref ?object .
}
`,ontologies:["ORDO","OboInOwl"],sparqlConcepts:["PREFIX","SELECT","GROUP BY","ORDER BY DESC"],category:"level 0",rdfResultExample:""}],iP=nP.map(i=>({...i,source:"AMC"})),oP="/Sparql-fest/assets/SIB_logo-DZW0ir8s.jpg",rP=[{name:"Bgee - 001",slug:"Bgee_001",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the species present in Bgee?",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?species WHERE {
    ?species a up:Taxon .
    ?species up:rank up:Species .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss Bgee"},{name:"Bgee - 002",slug:"Bgee_002",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the species present in Bgee and their scientific and common names?",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?species ?sci_name ?common_name WHERE {
    ?species a up:Taxon ;
        up:scientificName ?sci_name ;
        up:rank up:Species .
    OPTIONAL { ?species up:commonName ?common_name . }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","IF"],category:"sib-swiss Bgee"},{name:"Bgee - 003",slug:"Bgee_003",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the APOC1 gene is expressed?",context:null,inidces:[],query:`PREFIX genex: <http://purl.org/genex#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?anat ?anatName WHERE {
    ?seq a orth:Gene ;
        genex:isExpressedIn ?anat ;
        rdfs:label "APOC1" .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss Bgee"},{name:"Bgee - 004",slug:"Bgee_004",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the APOC1 Homo sapiens gene is expressed?",context:null,inidces:[],query:`PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?geneName ?anat ?anatName WHERE {
    ?seq a orth:Gene ;
        genex:isExpressedIn ?anat ;
        rdfs:label ?geneName .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?seq orth:organism ?organism .
    ?organism obo:RO_0002162  ?species .
    ?species a up:Taxon ;
        up:scientificName "Homo sapiens" .
    FILTER (LCASE(?geneName) = LCASE('APOC1') )
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","IF","LCASE"],category:"sib-swiss Bgee"},{name:"Bgee - 005",slug:"Bgee_005",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the APOC1 gene is expressed independently of the developmental stage, sex, strain and cell type?",context:null,inidces:[],query:`PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?anat ?anatName {
    ?seq a orth:Gene ;
        genex:isExpressedIn ?condition ;
        rdfs:label "APOC1" .
    ?condition a genex:ExpressionCondition ;
        genex:hasAnatomicalEntity ?anat ;
        genex:hasAnatomicalEntity obo:GO_0005575 ;
        genex:hasDevelopmentalStage ?stage ;
        genex:hasSex "any" ;
        genex:hasStrain ?strain .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?stage a efo:EFO_0000399 ;
        rdfs:label "life cycle" .
    ?strain rdfs:label "wild-type" .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","DISTINCT","STR","IF"],category:"sib-swiss Bgee"},{name:"Bgee - 006",slug:"Bgee_006",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the human gene APOC1 is expressed in the post-juvenile stage?",context:null,inidces:[],query:`PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?anatName ?anat ?stageName ?stage WHERE {
    ?seq a orth:Gene ;
        rdfs:label "APOC1" ;
        genex:isExpressedIn ?condition ;
        orth:organism ?organism .
    ?condition a genex:ExpressionCondition ;
        genex:hasAnatomicalEntity ?anat ;
        genex:hasAnatomicalEntity obo:GO_0005575 ;
        genex:hasDevelopmentalStage ?stage .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?stage a efo:EFO_0000399 ;
        rdfs:label ?stageName .
    ?organism obo:RO_0002162  ?species .
    ?species a up:Taxon ;
        up:commonName "human" .
    FILTER (CONTAINS(LCASE(?stageName), 'post-juvenile' ))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","LCASE","CONTAINS"],category:"sib-swiss Bgee"},{name:"Bgee - 007",slug:"Bgee_007",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the mouse gene APOC1 is expressed in the adult stages?",context:null,inidces:[],query:`PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?geneName ?anatName ?anat ?stageName ?stage WHERE {
    ?seq a orth:Gene ;
        rdfs:label ?geneName ;
        genex:isExpressedIn ?condition ;
        orth:organism ?organism .
    ?condition a genex:ExpressionCondition ;
        genex:hasAnatomicalEntity ?anat ;
        genex:hasAnatomicalEntity obo:GO_0005575 ;
        genex:hasDevelopmentalStage ?stage .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?stage a efo:EFO_0000399 ; # developmental stage
        rdfs:label ?stageName .
    ?organism obo:RO_0002162 ?taxon . # in taxon
    ?taxon a up:Taxon ;
        up:commonName "mouse" .
    FILTER (CONTAINS(LCASE(?stageName), 'adult' ))
    FILTER (lcase(str(?geneName)) = "apoc1" ).
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","STR","LCASE","CONTAINS"],category:"sib-swiss Bgee"},{name:"Bgee - 008",slug:"Bgee_008",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the human gene APOC1 is expressed at the post-juvenile stage along with its expression score independently of the strain, sex, and cell type?",context:null,inidces:[],query:`PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?anat ?anatName ?score ?stage WHERE {
    ?seq a orth:Gene ;
        rdfs:label "APOC1" ;
        orth:organism ?organism .
    ?expression a genex:Expression ;
        genex:hasExpressionCondition ?condition ;
        genex:hasExpressionLevel ?score ;
        genex:hasSequenceUnit ?seq .
    ?condition a genex:ExpressionCondition ;
        genex:hasAnatomicalEntity ?anat ;
        genex:hasAnatomicalEntity obo:GO_0005575 ;
        genex:hasDevelopmentalStage ?stage ;
        genex:hasSex "any" ;
        genex:hasStrain ?strain .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?stage rdfs:label ?stageName .
    ?strain rdfs:label "wild-type" .
    ?organism obo:RO_0002162  ?species .
    ?species a up:Taxon ;
        up:commonName "human" .
    FILTER (?anat !=  obo:GO_0005575)
    FILTER (CONTAINS(LCASE(?stageName), 'post-juvenile' ))
} ORDER BY DESC(?score)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","STR","LCASE","CONTAINS"],category:"sib-swiss Bgee"},{name:"Bgee - 009",slug:"Bgee_009",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities including cell types, if any, where the human gene APOC1 is expressed at the post-juvenile stage along with its expression score independently of the strain and sex?",context:null,inidces:[],query:`PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?anat ?cellType ?anatName ?cellTypeName ?score ?stage WHERE {
    ?seq a orth:Gene ;
        rdfs:label "APOC1" ;
        orth:organism ?organism .
    ?expression a genex:Expression ;
        genex:hasExpressionCondition ?condition ;
        genex:hasExpressionLevel ?score ;
        genex:hasSequenceUnit ?seq .
    ?condition a genex:ExpressionCondition ;
        genex:hasAnatomicalEntity ?anat ;
        genex:hasAnatomicalEntity ?cellType ;
        genex:hasDevelopmentalStage ?stage ;
        genex:hasSex "any" ;
        genex:hasStrain ?strain .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?cellType rdfs:label ?cellTypeName .
    ?stage rdfs:label "post-juvenile" .
    ?strain a efo:EFO_0005135 ;
        rdfs:label "wild-type" .
    ?organism obo:RO_0002162  ?species .
    ?species a up:Taxon ;
        up:commonName "human" .
    FILTER (?anat != obo:GO_0005575)
    FILTER (?anat != ?cellType)
} ORDER BY DESC(?score)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","STR"],category:"sib-swiss Bgee"},{name:"Bgee - 010",slug:"Bgee_010",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities including cell types, if any, where the human gene APOC1 is expressed at the post-juvenile stage along with its expression score independently of the strain and sex?",context:null,inidces:[],query:`PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?anat ?cellType ?anatName ?cellTypeName ?score ?stage WHERE {
    ?seq a orth:Gene ;
        rdfs:label "APOC1" ;
        orth:organism ?organism .
    ?expression a genex:Expression ;
        genex:hasExpressionCondition ?condition ;
        genex:hasExpressionLevel ?score ;
        genex:hasSequenceUnit ?seq .
    ?condition a genex:ExpressionCondition ;
        genex:hasAnatomicalEntity ?anat ;
        genex:hasAnatomicalEntity ?cellType ;
        genex:hasDevelopmentalStage ?stage .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?cellType rdfs:label ?cellTypeName .
    ?stage a efo:EFO_0000399 ;
        rdfs:label "post-juvenile" .
    ?organism obo:RO_0002162  ?species .
    ?species a up:Taxon ;
        up:commonName "human" .
    FILTER (?anat !=  obo:GO_0005575)
    FILTER (?anat != ?cellType)
} ORDER BY DESC(?score)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT"],category:"sib-swiss Bgee"},{name:"Bgee - 011",slug:"Bgee_011",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the developmental stages present in Bgee?",context:null,inidces:[],query:`PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?stage ?stageName ?stageDescription WHERE {
    ?stage a efo:EFO_0000399 ;
        rdfs:label ?stageName ;
        dcterms:description ?stageDescription .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss Bgee"},{name:"Bgee - 012",slug:"Bgee_012",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the eel gene apoc1 is expressed along with its expression score independently of the strain, sex, and cell type?",context:null,inidces:[],query:`PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?anat ?anatName ?stageIRI ?score WHERE {
    ?seq a orth:Gene ;
        lscr:xrefNCBIGene <https://www.ncbi.nlm.nih.gov/gene/118230125> .
        # Or dcterms:identifier "118230125" .
    ?expression a genex:Expression ;
        genex:hasExpressionCondition ?condition ;
        genex:hasExpressionLevel ?score ;
        genex:hasSequenceUnit ?seq .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?condition a genex:ExpressionCondition ;
        genex:hasAnatomicalEntity ?anat ;
        genex:hasDevelopmentalStage ?stageIRI ;
        genex:hasSex "any" ;
        genex:hasStrain ?strain .
    ?stageIRI a efo:EFO_0000399 .
    ?strain a efo:EFO_0005135 ;
        rdfs:label "wild-type" .
    FILTER (?anat != obo:GO_0005575)
} ORDER BY DESC(?score)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","STR","IRI","IF"],category:"sib-swiss Bgee"},{name:"Bgee - 013",slug:"Bgee_013",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the P02654 gene is expressed? Note that P02654 is a UniProtKB identifier of the APOC1 human gene.",context:null,inidces:[],query:`PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>

SELECT DISTINCT ?anat ?anatName WHERE {
    ?seq a orth:Gene ;
        genex:isExpressedIn ?anat ;
        lscr:xrefUniprot uniprotkb:P02654 .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss Bgee"},{name:"Bgee - 014",slug:"Bgee_014",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What is all the metadata related to the ENSG00000130208 gene, where ENSG00000130208 is the identifier of the APOC1 human gene.",context:null,inidces:[],query:`PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?symbol ?description ?id
?links ?organism ?uniprot ?ensembl ?ncbi WHERE {
    ?seq a orth:Gene ;
        rdfs:label ?symbol ;
        rdfs:seeAlso ?links ;
        dcterms:description ?description ;
        dcterms:identifier ?id ;
        orth:organism ?organism .
    OPTIONAL{?seq lscr:xrefUniprot ?uniprot .}
    OPTIONAL{?seq lscr:xrefEnsemblGene ?ensembl .}
    OPTIONAL{?seq lscr:xrefNCBIGene ?ncbi .}
    FILTER (?id = "ENSG00000130208")
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","DISTINCT","IF"],category:"sib-swiss Bgee"},{name:"Bgee - 015",slug:"Bgee_015",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the anatomical entities where the APOC1 Homo sapiens gene is not expressed, that is where is APOC1 absent?",context:null,inidces:[],query:`PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?anat ?anatName WHERE {
    ?seq a orth:Gene ;
        rdfs:label "APOC1" ;
        genex:isAbsentIn ?anat ;
        orth:organism ?organism .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label ?anatName .
    ?organism obo:RO_0002162  ?species .
    ?species a up:Taxon ;
        up:scientificName "Homo sapiens" .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","IF","ABS"],category:"sib-swiss Bgee"},{name:"Bgee - 016",slug:"Bgee_016",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which are the genes in Human associated to a disease that are orthologous to a gene expressed in the rat's brain?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX sio: <http://semanticscience.org/resource/>

SELECT ?gene ?omaLink ?uniprot ?diseaseLabel ?annotationText WHERE {
    {
        SELECT ?gene {
            ?anat rdfs:label 'brain' .
            ?gene genex:isExpressedIn ?anat ;
                orth:organism/obo:RO_0002162 ?taxonRat .
            ?taxonRat up:commonName 'rat' .
        } LIMIT 20
    }
    SERVICE <https://sparql.uniprot.org/sparql> {
        ?taxonHuman up:commonName 'Human' .
    }
    SERVICE <https://sparql.omabrowser.org/sparql/> {
        ?cluster a orth:OrthologsCluster ;
            orth:hasHomologousMember ?node1 ;
            orth:hasHomologousMember ?node2 .
        ?node2 orth:hasHomologousMember* ?protein2 .
        ?node1 orth:hasHomologousMember* ?protein1 .
        ?protein1 a orth:Protein ;
            sio:SIO_010079 ?gene . # encoded by
        ?protein2 a orth:Protein ;
            rdfs:seeAlso ?omaLink ;
            orth:organism/obo:RO_0002162 ?taxonHuman ;
            lscr:xrefUniprot ?uniprot .
        FILTER(?node1 != ?node2)
    }
    SERVICE <https://sparql.uniprot.org/sparql> {
        ?uniprot up:annotation ?annotation .
        ?annotation a up:Disease_Annotation ;
            rdfs:comment ?annotationText ;
            up:disease ?disease .
        ?disease skos:prefLabel ?diseaseLabel .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","LIMIT"],category:"sib-swiss Bgee"},{name:"Bgee - 017",slug:"Bgee_017",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Anatomical entities where the ins zebrafish gene is expressed and its gene GO annotations.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?anatomicalEntity ?goClass ?goLabel {
    {
        SELECT ?ensemblGene ?anatomicalEntity {
            ?geneB a orth:Gene ;
                genex:isExpressedIn ?anat ;
                rdfs:label ?geneLabel ;
                lscr:xrefEnsemblGene ?ensemblGene ;
                orth:organism/obo:RO_0002162 ?taxon2 .
            ?taxon2 up:commonName 'zebrafish' .
            FILTER (UCASE(?geneLabel) = UCASE('ins'))
            ?anat rdfs:label ?anatomicalEntity .
        }
    }
    SERVICE <https://sparql.uniprot.org/sparql> {
        ?uniprot rdfs:seeAlso/up:transcribedFrom ?ensemblGene ;
            a up:Protein ;
            up:classifiedWith ?goClass .
        ?goClass rdfs:label ?goLabel .
    }
} LIMIT 20`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","FILTER","SERVICE","LIMIT","DISTINCT","FROM","WITH","IF","UCASE"],category:"sib-swiss Bgee"},{name:"Bgee - 018",slug:"Bgee_018",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which are the mouse's proteins encoded by genes which are expressed in the liver and are orthologous to human's INS gene?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX sio: <http://semanticscience.org/resource/>
PREFIX lscr: <http://purl.org/lscr#>

SELECT ?humanProtein ?mouseProtein ?mouseOmaLink {
    {
        ?humanTaxon up:commonName 'human' .
        ?mouseTaxon up:commonName 'mouse' .
    }
    SERVICE <https://sparql.omabrowser.org/sparql> {
        ?cluster a orth:OrthologsCluster ;
            orth:hasHomologousMember ?node1 ;
            orth:hasHomologousMember ?node2 .
        ?node2 orth:hasHomologousMember* ?mouseProtein .
        ?node1 orth:hasHomologousMember* ?humanProtein .
        ?humanProtein a orth:Protein ;
            rdfs:label 'INS' ;
            orth:organism/obo:RO_0002162 ?humanTaxon .
        ?mouseProtein a orth:Protein ;
            sio:SIO_010079 ?mouseGene ; # is encoded by
            orth:organism/obo:RO_0002162 ?mouseTaxon ;
            rdfs:seeAlso ?mouseOmaLink .
        ?mouseGene lscr:xrefEnsemblGene ?mouseGeneEnsembl .
        FILTER ( ?node1 != ?node2 )
    }
    {
        ?mouseGeneBgee a orth:Gene ;
            lscr:xrefEnsemblGene ?mouseGeneEnsembl ;
            genex:isExpressedIn ?cond ;
            orth:organism/obo:RO_0002162 ?mouseTaxon .
        ?cond genex:hasAnatomicalEntity/rdfs:label 'liver' .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","FILTER","SERVICE"],category:"sib-swiss Bgee"},{name:"Bgee - 019",slug:"Bgee_019",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which are the orthologs of a gene that is expressed in the fruit fly's brain?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX sio: <http://semanticscience.org/resource/>

SELECT DISTINCT ?geneEnsembl ?proteinOrtholog ?geneOrthologEnsembl ?taxonOrtholog ?omaLinkOrtholog WHERE {
    {
        SELECT DISTINCT * {
            ?gene a orth:Gene ;
                genex:isExpressedIn/rdfs:label 'brain' ;
                orth:organism/obo:RO_0002162 ?taxonFly ;
                dcterms:identifier ?geneEnsembl .
            ?taxonFly up:commonName 'fruit fly' .
        } LIMIT 20
    }

    SERVICE <https://sparql.omabrowser.org/sparql/> {
        ?cluster a orth:OrthologsCluster ;
            orth:hasHomologousMember ?node1 ;
            orth:hasHomologousMember ?node2 .
        ?node1 orth:hasHomologousMember* ?protein1 .
        ?node2 orth:hasHomologousMember* ?proteinOrtholog .
        ?protein1 sio:SIO_010079 ?gene . # is encoded by
        ?proteinOrtholog rdfs:seeAlso ?omaLinkOrtholog ;
            orth:organism/obo:RO_0002162 ?taxonOrthologUri ;
            sio:SIO_010079 ?geneOrtholog . # is encoded by
        FILTER ( ?node1 != ?node2 )
    }
    ?taxonOrthologUri up:commonName ?taxonOrtholog .
    ?geneOrtholog dcterms:identifier ?geneOrthologEnsembl .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","LIMIT","DISTINCT","URI","IF"],category:"sib-swiss Bgee"},{name:"Bgee - 020",slug:"Bgee_020",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which are the genes in Primates orthologous to a gene that is expressed in the fruit fly's eye?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX sio: <http://semanticscience.org/resource/>

SELECT DISTINCT ?flyEnsemblGene ?orthologTaxon ?orthologEnsemblGene ?orthologOmaLink WHERE {
	{
        SELECT DISTINCT ?gene ?flyEnsemblGene {
        ?gene a orth:Gene ;
            genex:isExpressedIn/rdfs:label 'eye' ;
            orth:organism/obo:RO_0002162 ?taxon ;
            dcterms:identifier ?flyEnsemblGene .
        ?taxon up:commonName 'fruit fly' .
        } LIMIT 100
    }
    SERVICE <https://sparql.omabrowser.org/sparql> {
        ?protein2 a orth:Protein .
        ?protein1 a orth:Protein .
        ?clusterPrimates a orth:OrthologsCluster .
        ?cluster a orth:OrthologsCluster ;
            orth:hasHomologousMember ?node1 ;
            orth:hasHomologousMember ?node2 .
        ?node1 orth:hasHomologousMember* ?protein1 .
        ?node2 orth:hasHomologousMember* ?clusterPrimates .
        ?clusterPrimates orth:hasHomologousMember* ?protein2 .
        ?protein1 sio:SIO_010079 ?gene . # is encoded by
        ?protein2 rdfs:seeAlso ?orthologOmaLink ;
            orth:organism/obo:RO_0002162 ?orthologTaxonUri ;
            sio:SIO_010079 ?orthologGene . # is encoded by
        ?clusterPrimates orth:hasTaxonomicRange ?taxRange .
        ?taxRange orth:taxRange 'Primates' .
        FILTER ( ?node1 != ?node2 )
    }
    ?orthologTaxonUri up:commonName ?orthologTaxon .
    ?orthologGene dcterms:identifier ?orthologEnsemblGene .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","LIMIT","DISTINCT","URI","IF"],category:"sib-swiss Bgee"},{name:"Bgee - 021",slug:"Bgee_021",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which species the Mt-co1 gene is present (without considering synonyms)?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX obo: <http://purl.obolibrary.org/obo/>

SELECT ?name
WHERE {
    ?gene a orth:Gene ;
        rdfs:label ?geneName ;
        orth:organism ?organism .
    ?organism obo:RO_0002162 ?taxon . # in taxon
    ?taxon up:scientificName ?name .
    FILTER ( UCASE(?geneName) = UCASE('Mt-co1') )
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","IF","UCASE"],category:"sib-swiss Bgee"},{name:"Bgee - 022",slug:"Bgee_022",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List the labels and identifiers of pig-tailed macaque genes in bgee",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX obo: <http://purl.obolibrary.org/obo/>

SELECT DISTINCT ?geneId ?geneName
WHERE {
    ?gene a orth:Gene ;
        dcterms:identifier ?geneId ;
        orth:organism/obo:RO_0002162/up:commonName ?commonName .
    OPTIONAL { ?gene rdfs:label ?geneName .}
    FILTER ( LCASE(?commonName) = "pig-tailed macaque" ) .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","DISTINCT","IF","LCASE"],category:"sib-swiss Bgee"},{name:"Bgee - 023",slug:"Bgee_023",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Human anatomical entities at young adult developmental stage",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>

SELECT DISTINCT ?anatomicalEntity ?stageName {
    ?condition genex:hasAnatomicalEntity ?anatEntity ;
        genex:hasDevelopmentalStage ?stage ;
        obo:RO_0002162/up:commonName ?commonName .
    ?anatEntity rdfs:label ?anatomicalEntity .
    ?stage rdfs:label ?stageName .
    FILTER ( lcase(?commonName) = "human" ).
    FILTER ( CONTAINS(lcase(?stageName), "young adult") )
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","FILTER","DISTINCT","LCASE","CONTAINS"],category:"sib-swiss Bgee"},{name:"Bgee - 024",slug:"Bgee_024",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Human anatomic entities at young adult developmental stage",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>

SELECT DISTINCT ?anatomicalEntity ?stageName {
    ?condition genex:hasAnatomicalEntity ?anatEntity ;
        genex:hasDevelopmentalStage ?stage ;
        obo:RO_0002162/up:commonName ?commonName .
    ?anatEntity rdfs:label ?anatomicalEntity .
    ?stage rdfs:label ?stageName .
    FILTER( LCASE(?commonName) = "human" )
    FILTER( CONTAINS(LCASE(?stageName), "young adult") )
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","FILTER","DISTINCT","LCASE","CONTAINS"],category:"sib-swiss Bgee"},{name:"Bgee - 025",slug:"Bgee_025",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What is the post-juvenile stage link and description?",context:null,inidces:[],query:`PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX efo: <http://www.ebi.ac.uk/efo/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?stage ?stageName ?stageDescription WHERE {
    ?stage a efo:EFO_0000399 ;
        rdfs:label ?stageName ;
        dcterms:description ?stageDescription .
    FILTER(CONTAINS(LCASE(?stageName), "post-juvenile"))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","LCASE","CONTAINS"],category:"sib-swiss Bgee"},{name:"Bgee - 026",slug:"Bgee_026",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the genes expressed in the human brain?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX genex: <http://purl.org/genex#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?gene ?geneName WHERE {
    ?gene a orth:Gene ;
        genex:isExpressedIn ?anat ;
        rdfs:label ?geneName ;
        orth:organism ?organism .
    ?anat a genex:AnatomicalEntity ;
        rdfs:label "brain" .
    ?organism obo:RO_0002162 ?species .
    ?species a up:Taxon ;
        up:scientificName "Homo sapiens" .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","IF"],category:"sib-swiss Bgee"},{name:"Bgee - 027-biosodafrontend",slug:"Bgee_027-biosodafrontend",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Genes expressed in the human pancreas and their annotations in UniProt.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
SELECT DISTINCT ?geneEns ?uniprot ?annotation_text {
	{
		SELECT ?geneEns {
			?geneB a orth:Gene .
			?geneB genex:isExpressedIn ?cond .
			?cond genex:hasAnatomicalEntity ?anat .
				?geneB lscr:xrefEnsemblGene ?geneEns .
			?anat rdfs:label 'pancreas' .
			?geneB orth:organism ?o .
			?o obo:RO_0002162 ?taxon2 .
			?taxon2 up:commonName 'human' .
		} LIMIT 100
	}
	SERVICE <https://sparql.uniprot.org/sparql> {
		?uniprot rdfs:seeAlso / up:transcribedFrom ?geneEns .
		?uniprot up:annotation ?annotation .
		?annotation rdfs:comment ?annotation_text .
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","SERVICE","LIMIT","DISTINCT","FROM"],category:"sib-swiss Bgee"},{name:"Bgee - 028-biosodafrontend",slug:"Bgee_028-biosodafrontend",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Genes expressed in the human's brain during the infant stage and their UniProt disease annotations.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
SELECT DISTINCT ?geneEns ?uniprot ?annotation {
	{
		SELECT ?geneEns {
			?geneB genex:isExpressedIn ?cond ;
				lscr:xrefEnsemblGene ?geneEns .
			?cond genex:hasDevelopmentalStage ?st .
			?cond genex:hasAnatomicalEntity ?anat .
			?st rdfs:label 'infant stage' .
			?anat rdfs:label 'brain' .
			?geneB orth:organism ?o .
			?o obo:RO_0002162 ?taxon2 .
			?taxon2 up:commonName 'human' .
		}
		LIMIT 10
	}
	SERVICE <https://sparql.uniprot.org/sparql> {
		?uniprot up:transcribedFrom ?geneEns .
		?uniprot up:annotation ?annotation .
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","SERVICE","LIMIT","DISTINCT","FROM"],category:"sib-swiss Bgee"}],sP=[{name:"Cellosaurus - 101 publi cnt by type",slug:"Cellosaurus_101_publi_cnt_by_type",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Count of referenced publications by publication type",context:null,inidces:[],query:`PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX xref: <https://purl.expasy.org/cellosaurus/rdf/xref/>
PREFIX widoco: <https://w3id.org/widoco/vocab#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX vann: <http://purl.org/vocab/vann/>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX sh: <http://www.w3.org/ns/shacl#>
PREFIX schema: <https://schema.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX pubmed: <https://www.ncbi.nlm.nih.gov/pubmed/>
PREFIX pub: <https://purl.expasy.org/cellosaurus/rdf/pub/>
PREFIX prism: <http://prismstandard.org/namespaces/basic/2.0/>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX orga: <https://purl.expasy.org/cellosaurus/rdf/orga/>
PREFIX oa: <http://www.w3.org/ns/oa#>
PREFIX help: <https://api.cellosaurus.org/>
PREFIX fabio: <http://purl.org/spar/fabio/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX db: <https://purl.expasy.org/cellosaurus/rdf/db/>
PREFIX cvcl: <https://purl.expasy.org/cellosaurus/rdf/cvcl/>
PREFIX cello: <https://purl.expasy.org/cellosaurus/rdf/ontology/>
PREFIX bibo: <http://purl.org/ontology/bibo/>
PREFIX ORDO: <http://www.orpha.net/ORDO/Orphanet_>
PREFIX OMIT: <http://purl.obolibrary.org/obo/OMIT_>
PREFIX OBI: <http://purl.obolibrary.org/obo/OBI_>
PREFIX NCIt: <http://purl.obolibrary.org/obo/NCIT_>
PREFIX IAO: <http://purl.org/ontology/IAO_>
PREFIX GENO: <http://purl.obolibrary.org/obo/GENO_>
PREFIX FBcv: <http://purl.obolibrary.org/obo/FBcv_>
PREFIX EDAM: <http://edamontology.org/>
PREFIX CLO: <http://purl.obolibrary.org/obo/CLO_>
PREFIX CL: <http://purl.obolibrary.org/obo/CL_>
PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX CARO: <http://purl.obolibrary.org/obo/CARO_>
PREFIX BTO: <http://purl.obolibrary.org/obo/BTO_>
PREFIX BFO: <http://purl.obolibrary.org/obo/BFO_>
PREFIX BAO: <http://www.bioassayontology.org/bao#BAO_>

select ?pub_type (count(?cl) as ?cnt) where { 
  ?pub_type rdfs:subClassOf* cello:Publication .
  optional {?cl rdf:type ?pub_type. }
}
group by ?pub_type
order by desc(count(*))`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","GROUP BY","ORDER BY","COUNT"],category:"sib-swiss Cellosaurus"},{name:"Cellosaurus - 102 cl cnt by category",slug:"Cellosaurus_102_cl_cnt_by_category",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Count of cell lines by cell line category",context:null,inidces:[],query:`PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX xref: <https://purl.expasy.org/cellosaurus/rdf/xref/>
PREFIX widoco: <https://w3id.org/widoco/vocab#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX vann: <http://purl.org/vocab/vann/>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX sh: <http://www.w3.org/ns/shacl#>
PREFIX schema: <https://schema.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX pubmed: <https://www.ncbi.nlm.nih.gov/pubmed/>
PREFIX pub: <https://purl.expasy.org/cellosaurus/rdf/pub/>
PREFIX prism: <http://prismstandard.org/namespaces/basic/2.0/>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX orga: <https://purl.expasy.org/cellosaurus/rdf/orga/>
PREFIX oa: <http://www.w3.org/ns/oa#>
PREFIX help: <https://api.cellosaurus.org/>
PREFIX fabio: <http://purl.org/spar/fabio/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX db: <https://purl.expasy.org/cellosaurus/rdf/db/>
PREFIX cvcl: <https://purl.expasy.org/cellosaurus/rdf/cvcl/>
PREFIX cello: <https://purl.expasy.org/cellosaurus/rdf/ontology/>
PREFIX bibo: <http://purl.org/ontology/bibo/>
PREFIX ORDO: <http://www.orpha.net/ORDO/Orphanet_>
PREFIX OMIT: <http://purl.obolibrary.org/obo/OMIT_>
PREFIX OBI: <http://purl.obolibrary.org/obo/OBI_>
PREFIX NCIt: <http://purl.obolibrary.org/obo/NCIT_>
PREFIX IAO: <http://purl.org/ontology/IAO_>
PREFIX GENO: <http://purl.obolibrary.org/obo/GENO_>
PREFIX FBcv: <http://purl.obolibrary.org/obo/FBcv_>
PREFIX EDAM: <http://edamontology.org/>
PREFIX CLO: <http://purl.obolibrary.org/obo/CLO_>
PREFIX CL: <http://purl.obolibrary.org/obo/CL_>
PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX CARO: <http://purl.obolibrary.org/obo/CARO_>
PREFIX BTO: <http://purl.obolibrary.org/obo/BTO_>
PREFIX BFO: <http://purl.obolibrary.org/obo/BFO_>
PREFIX BAO: <http://www.bioassayontology.org/bao#BAO_>

select ?class_name (count(?cl) as ?cnt) where { 
  ?class rdfs:subClassOf* cello:CellLine . # cell line generic class
  ?class rdfs:label ?class_name .
  optional {?cl rdf:type ?class. }
}
group by ?class_name
order by desc(count(*))`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","GROUP BY","ORDER BY","COUNT"],category:"sib-swiss Cellosaurus"}],aP=[{name:"dbgi - 001",slug:"dbgi_001",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Among the SIRIUS structural annotations from Tabernaemontana coffeoides (Apocynaceae) seeds extract, which ones are reported in the Tabernaemontana genus?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX emi: <https://purl.org/emi#>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX pr: <http://www.wikidata.org/prop/reference/>

SELECT DISTINCT ?ik2d ?genus
WHERE
    {
    ?material sosa:hasSample ?extract ;
        sosa:isSampleOf ?organe .
    ?organe emi:inTaxon ?wd_sp .
    ?wd_sp rdfs:label "tabernaemontana coffeoides" .
    ?extract sosa:isFeatureOfInterestOf ?lcms .
    ?lcms sosa:hasResult ?feature_list .
    ?feature_list emi:hasLCMSFeature ?feature .
    ?feature emi:hasAnnotation ?sirius_annotation .
    ?sirius_annotation a emi:StructuralAnnotation ;
        prov:wasGeneratedBy ?activity ;
        emi:hasChemicalStructure ?ik2d .
    ?activity prov:wasAssociatedWith <https://bio.informatik.uni-jena.de/software/sirius> .
    ?ik2d emi:hasSMILES ?smiles ;
        emi:isInChIKey2DOf ?ik .
    ?ik emi:isInChIKeyOf ?wd_id .
	{
        SELECT DISTINCT ?wd_id ?genus WHERE {
            ?material sosa:hasSample ?extract ;
                sosa:isSampleOf ?organe .
            ?organe emi:inTaxon ?wd_sp .
            ?wd_sp rdfs:label "tabernaemontana coffeoides" .
           # OPTIONAL {
                SERVICE <https://query.wikidata.org/sparql> {
                    ?wd_sp wdt:P225 ?species_name .
                    ?genus wdt:P31 wd:Q16521 ;
                        wdt:P105 wd:Q34740 ;
                        ^wdt:P171* ?wd_sp .
                    ?childtaxa wdt:P171* ?genus .
                    ?wd_id wdt:P703 ?childtaxa
                }
           # }
        } LIMIT 100
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","SERVICE","LIMIT","DISTINCT","WITH","STR","IRI","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 002",slug:"dbgi_002",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Among the structural annotations from Tabernaemontana coffeoides (Apocynaceae) seeds extract, which ones contain an aspidospermidine substructure?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX enpkg: <https://enpkg.commons-lab.org/kg/>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>

SELECT DISTINCT ?ik2d ?smiles
WHERE {
    ?extract a emi:ExtractSample ;
        sosa:isSampleOf* ?organe .
    ?organe emi:inTaxon ?taxon .
    ?taxon rdfs:label "tabernaemontana coffeoides" .
    ?extract sosa:isFeatureOfInterestOf ?lcms .
    ?lcms sosa:hasResult ?feature_list .
    ?feature_list emi:hasLCMSFeature ?feature .
    ?feature emi:hasAnnotation ?annotation .
    ?annotation emi:hasChemicalStructure ?ik2d .
    ?ik2d a emi:InChIKey2D ;
        emi:hasSMILES ?smiles ;
        emi:isInChIKey2DOf ?ik .
    ?ik emi:isInChIKeyOf ?wd_id .
    SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/wikidata> {
        SELECT * WHERE {
            ?wd_id sachem:substructureSearch [
                sachem:query "CCC12CCCN3C1C4(CC3)C(CC2)NC5=CC=CC=C45" # Aspidospermidine scaffold
            ] .
        }
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","STR","SUBSTR","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 003",slug:"dbgi_003",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Filter the positive ionization mode features of Melochia umbellate annotated as [M+H]+ by SIRIUS to keep the ones for which a feature in negative ionization mode is detected with the same retention time (± 3 seconds) and a mass corresponding to the [M-H]- adduct (± 5 ppm).",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>

SELECT DISTINCT ?feature ?rt ?pm ?feature_opp ?rt_opp ?pm_opp
WHERE {
    VALUES ?ppm {
        "5"^^xsd:decimal # m/z tolerance
    }
    VALUES ?rt_tol {
        "0.05"^^xsd:decimal # RT tolerance (minute)
    }
    ?sample a emi:ExtractSample ;
        sosa:isSampleOf* ?organe .
    ?organe emi:inTaxon ?taxon .
    ?taxon rdfs:label "melochia umbellata" .
    ?sample sosa:isFeatureOfInterestOf ?lcms .
    ?lcms a emi:LCMSAnalysisPos ;
        sosa:hasResult ?feature_list .
    ?feature_list emi:hasLCMSFeature ?feature .
    ?feature emi:hasParentMass ?pm ;
        emi:hasRetentionTime ?rt ;
        emi:hasAnnotation ?sirius .
	?sirius a emi:StructuralAnnotation ;
        prov:wasGeneratedBy ?activiy .
    ?activiy prov:wasAssociatedWith <https://bio.informatik.uni-jena.de/software/sirius> .
    ?sirius emi:hasAdduct ?adduct .
    FILTER(regex(str(?adduct), "[M+H]+"))
    ?sample sosa:isFeatureOfInterestOf ?lcms_opp .
    ?lcms_opp a emi:LCMSAnalysisNeg ;
        sosa:hasResult ?feature_list_opp .
    ?feature_list_opp emi:hasLCMSFeature ?feature_opp .
	?feature_opp emi:hasParentMass ?pm_opp ;
        emi:hasRetentionTime ?rt_opp .
    FILTER(((?rt - ?rt_tol) < ?rt_opp) && ((?rt + ?rt_tol) > ?rt_opp))
    FILTER((?pm_opp > ((?pm - 2.014) - ((?ppm * 0.000001) * (?pm - 2.014)))) && (?pm_opp < ((?pm - 2.014) + ((?ppm * 0.000001) * (?pm - 2.014)))))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","WITH","ADD","STR","REGEX","IRI","MIN","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 004",slug:"dbgi_004",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"For features from Melochia umbellata in positive ionization mode with SIRIUS annotations, get the ones for which a feature in negative ionization mode with the same retention time (± 3 sec) has the same SIRIUS annotation (2D InChIKey).",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>

SELECT DISTINCT ?feature ?feature_opp ?ik2d ?rt ?rt_opp
WHERE {
    VALUES ?rt_tol {
        "0.05"^^xsd:decimal # RT tolerance (minute)
    }
    ?sample a emi:ExtractSample ;
        sosa:isSampleOf* ?organe .
    ?organe emi:inTaxon ?taxon .
    ?taxon rdfs:label "melochia umbellata" .
    ?sample sosa:isFeatureOfInterestOf ?lcms .
    ?lcms a emi:LCMSAnalysisPos ;
        sosa:hasResult ?feature_list .
    ?feature_list emi:hasLCMSFeature ?feature .
    ?feature emi:hasRetentionTime ?rt ;
        emi:hasAnnotation ?sirius .
    ?sirius a emi:StructuralAnnotation ;
        prov:wasGeneratedBy ?activity .
    ?activity prov:wasAssociatedWith <https://bio.informatik.uni-jena.de/software/sirius> .
    ?sirius emi:hasChemicalStructure ?ik2d .
    ?sample sosa:isFeatureOfInterestOf ?lcms_opp .
    ?lcms_opp a emi:LCMSAnalysisNeg ;
        sosa:hasResult ?feature_list_opp .
    ?feature_list_opp emi:hasLCMSFeature ?feature_opp .
    ?feature_opp emi:hasRetentionTime ?rt_opp ;
        emi:hasAnnotation ?sirius_opp .
    ?sirius_opp a emi:StructuralAnnotation ;
    #    prov:wasGeneratedBy ?activity_opp ;
        emi:hasChemicalStructure ?ik2d .
    #?activity_opp prov:wasAssociatedWith <https://bio.informatik.uni-jena.de/software/sirius> .
    FILTER(((?rt - ?rt_tol) < ?rt_opp) && ((?rt + ?rt_tol) > ?rt_opp))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","WITH","STR","IRI","MIN","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 005",slug:"dbgi_005",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"How many features have the same SIRIUS and ISDB annotation?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX enpkg: <https://enpkg.commons-lab.org/kg/>
PREFIX emi: <https://purl.org/emi#>

SELECT (COUNT(?feature) AS ?count)
WHERE {
        ?lcms emi:hasMassiveDOI ?massive_doi ;
            emi:hasLCMSFeatureSet ?feature_list .
        # FILTER(regex(str(?massive_doi), "MSV000087728")) # MassIVE id filter, MSV000087728 is pf1600, MSV000093464 is 337 Korean medicinal plants, if you want to query both you can pipe "MSV000087728|MSV000093464". You can apply or not these filters by ucommenting/commenting this line.
        ?feature_list emi:hasLCMSFeature ?feature .
        ?feature emi:hasAnnotation ?sirius_annotation .
        ?sirius_annotation a emi:StructuralAnnotation ;
            prov:wasGeneratedBy ?activity ;
            emi:hasChemicalStructure ?sirius_ik2d .
        ?activity prov:wasAssociatedWith <https://bio.informatik.uni-jena.de/software/sirius> .
        ?feature emi:hasAnnotation ?isdb_annotation .
        ?activity_isdb prov:wasAssociatedWith <https://oolonek.github.io/ISDB> .
        ?isdb_annotation a emi:StructuralAnnotation ;
            prov:wasGeneratedBy ?activity_isdb ;
            emi:hasChemicalStructure ?isdb_ik2d .
    FILTER(?isdb_ik2d = ?sirius_ik2d) .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","WITH","STR","REGEX","IRI","IF","COUNT"],category:"sib-swiss dbgi"},{name:"dbgi - 006",slug:"dbgi_006",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Search spectrally related features with and within the VGF151_E05 sample of a datura stramonium through counts of common peaks and loss. Return their species, family and genus names",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>

SELECT ?family_name ?genus_name ?species_name ?wd_sp ?feature ?sub_rt ?sub_parent_mass ?count
WHERE {
    # Subquery to get the top N hits with their Wikidata IDs
    {
        SELECT ?wd_sp ?feature (SAMPLE(?rt) AS ?sub_rt) (SAMPLE(?parent_mass) AS ?sub_parent_mass) (COUNT(?peakloss) AS ?count)
        WHERE {
            ?material sosa:hasSample ?extract ;
                sosa:isSampleOf* ?organism.
            ?organism emi:inTaxon ?wd_sp .
            ?extract a emi:ExtractSample ;
                sosa:isFeatureOfInterestOf ?lcms .
            ?lcms a emi:LCMSAnalysisPos ;
                sosa:hasResult ?feature_list .
            ?feature_list emi:hasLCMSFeature ?feature .
            ?feature a emi:LCMSFeature ;
                emi:hasSpec2VecDoc ?doc ;
                emi:hasParentMass ?parent_mass ;
                emi:hasRetentionTime ?rt .
            ?doc emi:hasSpec2VecLoss|emi:hasSpec2VecPeak ?peakloss .

            {
                SELECT ?peakloss WHERE {
                    ?lcms a emi:LCMSAnalysisPos ;
                        sosa:hasResult ?feature_list ;
                        sosa:hasFeatureOfInterest ?sample .
                    ?sample dcterms:identifier "VGF151_E05" ;
                        sosa:isSampleOf* ?organism .
                    ?organism emi:inTaxon/rdfs:label "datura stramonium".
                    ?feature_list emi:hasLCMSFeature ?feature .
                    ?feature a emi:LCMSFeature ;
                        emi:hasSpec2VecDoc ?doc .
                    ?doc emi:hasSpec2VecLoss|emi:hasSpec2VecPeak ?peakloss .
                }
            }
        }
        GROUP BY ?wd_sp ?feature
        ORDER BY DESC(?count)
        LIMIT 5
    }

    # Outer query to fetch species names from Wikidata
    # Fetching taxonomic information from Wikidata
    SERVICE <https://query.wikidata.org/sparql> {
        ?wd_sp wdt:P225 ?species_name .
        OPTIONAL {
            ?family wdt:P31 wd:Q16521 ;
                wdt:P105 wd:Q35409 ;
                wdt:P225 ?family_name ;
                ^wdt:P171* ?wd_sp .
        }
        OPTIONAL {
            ?genus wdt:P31 wd:Q16521 ;
                wdt:P105 wd:Q34740 ;
                wdt:P225 ?genus_name ;
                ^wdt:P171* ?wd_sp .
        }
    }
} GROUP BY ?family_name ?genus_name ?species_name ?wd_sp ?feature ?sub_rt ?sub_parent_mass ?count
ORDER BY DESC(?count)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","SERVICE","GROUP BY","ORDER BY","LIMIT","FROM","WITH","STR","IF","COUNT","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 007",slug:"dbgi_007",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Show chemical structures in Melochia umbellata extracts with structural annotations where the cosmic score is greater than 0.5 and the zodiac score is less than 0.8. For ISDB annotations, consider taxon scores greater than or equal to 6",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>

SELECT DISTINCT ?ik ?wd_id ?taxon ?taxon_name
WHERE { 
    {SELECT  ?ik ?wd_id
        WHERE {
            ?sample a emi:ExtractSample ;
                sosa:isSampleOf* ?organe ;
                sosa:isFeatureOfInterestOf ?lcms .
            ?organe emi:inTaxon ?taxon .
            ?taxon rdfs:label "melochia umbellata" .
            ?lcms a emi:LCMSAnalysisPos ;
                sosa:hasResult ?feature_list .
            ?feature_list emi:hasLCMSFeature ?feature .
            {
                ?feature emi:hasAnnotation ?annotation .
                ?annotation a emi:StructuralAnnotation ;
                    prov:wasGeneratedBy ?activity .
                ?activity prov:wasAssociatedWith <https://bio.informatik.uni-jena.de/software/sirius> .
                ?annotation emi:hasCosmicScore ?cosmic ;
                    emi:hasZodiacScore ?zodiac .
                FILTER((?cosmic > 0.5) && (?zodiac > 0.8))
            } UNION {
                ?feature emi:hasAnnotation ?annotation .
                ?annotation emi:hasTaxoScore ?taxo .
                FILTER(?taxo >= 6)
            }
            ?annotation emi:hasChemicalStructure ?ik2d .
            ?ik2d emi:hasSMILES ?smiles ;
                emi:isInChIKey2DOf ?ik .
            ?ik emi:isInChIKeyOf ?wd_id .
        } LIMIT 50
    }
  SERVICE <https://query.wikidata.org/sparql> {
        ?wd_id wdt:P31 wd:Q113145171 ;
            wdt:P703 ?taxon .
        ?taxon wdt:P225 ?taxon_name .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","SERVICE","LIMIT","DISTINCT","WITH","STR","IRI","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 008",slug:"dbgi_008",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which features in the positive ionization mode have the most fragments and neutral losses in common with the feature identified as mzspec:MSV000087728:VGF143_H08_features_ms2_pos.mgf:scan:707",context:null,inidces:[],query:`PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>

SELECT ?feature ?massive_doi (SAMPLE(?rt) AS ?retention_time) (SAMPLE(?pm) AS ?parent_mass) (COUNT(?peakloss) AS ?count)
WHERE {
    ?lcms a emi:LCMSAnalysisPos ;
        emi:hasMassiveDOI ?massive_doi ;
        sosa:hasResult ?feature_list .
    #FILTER(regex(str(?massive_doi), "MSV000087728")) # MassIVE id filter
    ?feature_list emi:hasLCMSFeature ?feature .
    ?feature a emi:LCMSFeature ;
        emi:hasSpec2VecDoc ?doc ;
        emi:hasParentMass ?pm ;
        emi:hasRetentionTime ?rt .
    ?doc emi:hasSpec2VecLoss|emi:hasSpec2VecPeak ?peakloss .

    {
        SELECT ?peakloss WHERE {
            ?feature a emi:LCMSFeature ;
                emi:hasUSI "mzspec:MSV000087728:VGF143_H08_features_ms2_pos.mgf:scan:707" ;
                emi:hasSpec2VecDoc ?doc .
            ?doc emi:hasSpec2VecLoss|emi:hasSpec2VecPeak ?peakloss .
        }
    }
} GROUP BY ?feature ?massive_doi ORDER BY DESC(?count)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","ORDER BY","STR","REGEX","COUNT","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 009",slug:"dbgi_009",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which samples have features annotated as Aspidosperma_type alkaloids by CANOPUS with a probability score above 0.5, ordered by the decreasing count of features?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>

SELECT ?extract ?organe ?species_name ?genus_name ?family_name ?count_of_selected_class
WHERE {
    ?material sosa:hasSample ?extract ;
        sosa:isSampleOf ?organe .
    ?organe emi:inTaxon ?wd_sp .
    #OPTIONAL {
        SERVICE <https://query.wikidata.org/sparql> {
            ?wd_sp wdt:P225 ?species_name .
            ?family wdt:P31 wd:Q16521 ;
                wdt:P105 wd:Q35409 ;
                wdt:P225 ?family_name ;
                ^wdt:P171* ?wd_sp .
            ?genus wdt:P31 wd:Q16521 ;
                wdt:P105 wd:Q34740 ;
                wdt:P225 ?genus_name ;
                ^wdt:P171* ?wd_sp .
        }
    #}
    {
        SELECT ?extract (COUNT(DISTINCT ?feature) AS ?count_of_selected_class)
        WHERE {
            ?extract a emi:ExtractSample ;
                sosa:isFeatureOfInterestOf ?lcms .
            ?lcms a emi:LCMSAnalysis ;
                emi:hasLCMSFeatureSet ?feature_list .
            ?feature_list emi:hasLCMSFeature ?feature .
            ?feature emi:hasAnnotation ?canopus .
            ?canopus a emi:ChemicalTaxonAnnotation ;
                emi:hasClass ?np_class ;
                emi:hasClassProbability ?class_prob .
            ?np_class rdfs:label "Aspidosperma type" .
            FILTER((?class_prob > 0.5)) .
        } GROUP BY ?extract ORDER BY DESC(?count_of_selected_class)
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","SERVICE","GROUP BY","ORDER BY","DISTINCT","COUNT","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 011",slug:"dbgi_011",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List interactions of all species which have an IUCN status (wdt:P141) of near threatened (wd:Q719675).",context:null,inidces:[],query:`PREFIX emi: <https://purl.org/emi#>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
SELECT DISTINCT ?wdx_Source ?sourceName ?intxnLabel ?intxnType ?wdx_Target ?targetName WHERE {
    {
        SELECT DISTINCT ?wdx_Target ?targetName ?sourceName ?wdx_Source ?intxnType ?intxnLabel WHERE { #first select the source-target interaction-pairs
            #unidirectional interaction, e.g.: source-X hosts target target-Y
            ?intxn emi:hasSource ?source ;
                emi:hasTarget ?target ;
                emi:isClassifiedWith ?intxnType .
            ?intxnType rdfs:label ?intxnLabel .
            ?source emi:inTaxon ?wdx_Source ; # retrieve wikidata-id for source
                                rdfs:label ?sourceName . # scientific name of source as given in GloBI
                        ?target emi:inTaxon ?wdx_Target ; # retrieve wikidata-id for target
                                rdfs:label ?targetName . # scientific name of target as given in GloBI
                }
    }
    SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
        {
            ?wdx_Source wdt:P141 wd:Q719675 ; # filter source wikidata ids, which have IUCN status (wdt:P141) as near threatened (wd:Q719675) and which is a plant
                wdt:P171* wd:Q879246 ;
        } UNION 
        {
            ?wdx_Target wdt:P141 wd:Q719675 ; # filter target wikidata ids, which have IUCN status (wdt:P141) as near threatened (wd:Q719675) and which is a plant
                wdt:P171* wd:Q879246 ;
        }
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","SERVICE","DISTINCT","WITH","IF"],category:"sib-swiss dbgi"},{name:"dbgi - 012",slug:"dbgi_012",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List traits of all species which have an IUCN status (wdt:P141) of near threatened (wd:Q719675).",context:null,inidces:[],query:`PREFIX emi: <https://purl.org/emi#>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX qudt: <https://qudt.org/2.1/schema/qudt#>		
SELECT DISTINCT ?source_wd ?sourceName ?tryDataLab ?tryDataVal ?unit ?unitComment WHERE {
	?trySpObs sosa:isSampleOf ?source_trySpName ; #retrieve trait/non-trait data from trydb for trySpName (scientific name of plant species as listed in trydb)
		rdfs:label ?sourceName ;
		sosa:isFeatureOfInterestOf ?tryObId .
	?source_trySpName emi:inTaxon ?source_wd . #retrieve wikidata-id for trySpName		
	?tryObId sosa:hasResult ?tryData .
	?tryData rdfs:label ?tryDataLab ;			
		rdf:type emi:Trait ; #filter data which is labelled as 'Trait'
		rdf:value ?tryDataVal ;	#retrieve values for Trait data
		qudt:hasUnit ?unit . #retrieve units for Trait data
	OPTIONAL {
		?tryData rdfs:comment ?unitComment .
	} #retrieve comments (containing original unprocessed unit information - necessary for understanding some data) for Trait data
	SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
        	?source_wd wdt:P141 wd:Q719675 . #filter wikidata-ids for trySpName, which have IUCN status (wdt:P141) as near threatened (wd:Q719675)
        }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","VALUES","SERVICE","DISTINCT","FROM","IF","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 013",slug:"dbgi_013",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all metabolites produced by species with near threatened (wd:Q719675) IUCN status (wdt:P141) and with values available (or greater than a specific value) for trait 'Seed dry mass'.",context:null,inidces:[],query:`PREFIX emi: <https://purl.org/emi#>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
SELECT DISTINCT ?structure_inchikey ?wd_chem ?source_wdx ?sourceName WHERE {
	# trait data
	?trySpObs sosa:isSampleOf ?trySpName ; # trait/non-trait data for trySpName (scientific name of plant species as listed in trydb)
		rdfs:label ?sourceName ;
		sosa:isFeatureOfInterestOf ?tryObId .
	?tryObId sosa:hasResult ?tryData .
	?trySpName emi:inTaxon ?source_wdx . # wikidata-ids wdx for trySpName
	?tryData rdfs:label ?tryDataLab ;					
		rdf:type emi:Trait ; #retrieve data which is labelled as 'Trait' and its values
		rdf:value ?tryDataVal .
	FILTER (STR(?tryDataLab) = 'Seed dry mass' ) #filter on data label 'Seed dry mass'
	FILTER (?tryDataVal >= 100)						
	SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
        	?source_wdx wdt:P141 wd:Q719675 . # filter wikidata-ids for trySpName, which have IUCN status (wdt:P141) as near threatened (wd:Q719675)
        }	
	# metabolite data
	{ 
		SELECT ?source_wdx ?structure_inchikey ?wd_chem WHERE {	
			?material sosa:hasSample ?extract ;				
		        	sosa:isSampleOf ?organe .	
			?organe emi:inTaxon ?source_wdx . # filter metabolite data which is found in wikidata-ids wdx 
			?extract sosa:isFeatureOfInterestOf ?lcms .
		    	?lcms sosa:hasResult ?feature_list .
			?feature_list emi:hasLCMSFeature ?feature .
			?feature emi:hasAnnotation ?sirius_annotation .
			?sirius_annotation a emi:StructuralAnnotation ;
				emi:hasChemicalStructure ?ik2d .
			?ik2d emi:hasSMILES ?smiles ;
		    		emi:isInChIKey2DOf ?structure_inchikey .
			?structure_inchikey emi:isInChIKeyOf ?wd_chem .	# wikidata-ids for metabolites
		}
	}
	UNION # union with data from lotus (integrated in wikidata)
	{
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {		
			?wd_chem wdt:P235 ?structure_inchikey;				
   				wdt:P703 ?source_wdx .	
		}
	}
} LIMIT 1000`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","VALUES","SERVICE","LIMIT","DISTINCT","FROM","WITH","STR","IRI","IF","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 014",slug:"dbgi_014",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List traits (and their values) of plants producing Diterpenoids",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX qudt: <https://qudt.org/2.1/schema/qudt#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>		
SELECT DISTINCT ?source_wdx ?sourceName ?tryDataLab ?tryDataVal ?unit ?unitComment WHERE {
	# metabolite data
	{
		SELECT DISTINCT ?source_wdx WHERE {						
			?material sosa:hasSample ?extract ;
		        	sosa:isSampleOf ?organe .
		        ?organe emi:inTaxon ?source_wdx . # filter metabolite data which is found in wikidata-ids wdx
		        ?extract sosa:isFeatureOfInterestOf ?lcms .
		        ?lcms sosa:hasResult ?feature_list .
		        ?feature_list emi:hasLCMSFeature ?feature .
		        ?feature emi:hasAnnotation ?sirius_annotation .
		        ?sirius_annotation a emi:StructuralAnnotation ;
		        	emi:hasChemicalStructure ?ik2d .
		        ?ik2d emi:hasSMILES ?smiles ;
		        	emi:isInChIKey2DOf ?ik ;
		                emi:hasClass ?npcClass .
			?npcClass skos:broader ?npcSuperClass .
		        ?ik emi:isInChIKeyOf ?wd_chem .	# wikidata-ids for metabolites
			FILTER (REGEX(STR(?npcSuperClass), 'DITERPENOIDS'))
		}
	} UNION	#union with data from lotus (integrated in wikidata)
	{ 
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
			?wd_chem wdt:P235 ?structure_inchikey ;
				((wdt:P31|wdt:P279)/(wdt:P279*)) wd:Q47006367 ;	# check if the class/superclass of the chemical is Diterpenoids
   				wdt:P703 ?source_wdx .
		}
	}
	# trait data 	        
	?trySpName emi:inTaxon ?wdx .						
	?trySpObs sosa:isSampleOf ?trySpName ; # trait/non-trait data for trySpName (scientific name of plant species as listed in trydb)
		rdfs:label ?sourceName ;
		sosa:isFeatureOfInterestOf ?tryObId .
	?trySpName emi:inTaxon ?source_wdx . # wikidata-ids wdx for trySpName matching the one from metabolite queries 
	?tryObId sosa:hasResult ?tryData .
	?tryData rdfs:label ?tryDataLab ;
		rdf:type emi:Trait ; # retrieve data which is labelled as 'Trait' and its values
		rdf:value ?tryDataVal ;
		qudt:hasUnit ?unit ; 
		rdfs:comment ?unitComment . # original units for Trait data as listed in trydb
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","VALUES","SERVICE","DISTINCT","FROM","WITH","STR","REGEX","IRI","IF","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 015",slug:"dbgi_015",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List of possible natural locations of organisms, which produce Bahamaolide A,  (wd:Q75068439) which is an antifungal (https://doi.org/10.1021/np3001915).",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX wgs: <http://www.w3.org/2003/01/geo/wgs84_pos#>
PREFIX prov: <http://www.w3.org/ns/prov#>
SELECT DISTINCT ?organismName ?organismWD ?loc ?lats ?longs WHERE {
	# metabolite data
	{
		SELECT DISTINCT ?organismWD ?organismName WHERE {				
			?material sosa:hasSample ?extract ;
		        	sosa:isSampleOf ?organe .
			?organe emi:inTaxon ?organismWD ; # filter metabolite data which is found in wikidata-ids wdx	
				rdfs:label ?organismName .
			?extract sosa:isFeatureOfInterestOf ?lcms .
			?lcms sosa:hasResult ?feature_list .
			?feature_list emi:hasLCMSFeature ?feature .
			?feature emi:hasAnnotation ?sirius_annotation .
			?sirius_annotation a emi:StructuralAnnotation ;
				emi:hasChemicalStructure ?ik2d .
			?ik2d emi:hasSMILES ?smiles ;
				emi:isInChIKey2DOf ?ik .
			?ik emi:isInChIKeyOf ?wd_chem .	# retrieve wikidata-ids for metabolites
			VALUES ?wd_chem { wd:Q75068439 } # filter wikidata_id of metabolites matches wd:Q75068439
		}
	} UNION	#union with data from lotus (integrated in wikidata)
	{ 
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
			?wd_chem wdt:P235 ?ik ;
	    			wdt:P703 ?organismWD .
			 VALUES ?wd_chem { wd:Q75068439 }
			 FILTER(?organismWD = ?source_wdx || ?organismWD = ?target_wdx)	# ensure that organism-wd is either from source or target
			 OPTIONAL { ?organismWD wdt:P225 ?organismName . }
		}
	}
	# interaction data		
	?intxn emi:hasSource ?source ; #retrieve interaction-pairs
		emi:hasTarget ?target .
	?source emi:inTaxon ?source_wdx . # retrieve wikidata-id of source
	?target emi:inTaxon ?target_wdx . # retrieve wikidata-id of target
	?intxn prov:atLocation ?loc ; # retrieve the location of the interaction, and the latitude/longitude
		wgs:lat ?lats ;					
		wgs:long ?longs .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","UNION","VALUES","SERVICE","DISTINCT","FROM","WITH","STR","IRI","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 016",slug:"dbgi_016",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List of possible natural locations of plants, which produce  senkyunolide (wd:Q27251426)  which is a thalide for cerebral disorders (https://doi.org/10.3390/molecules28083636).",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX wgs: <http://www.w3.org/2003/01/geo/wgs84_pos#>
PREFIX prov: <http://www.w3.org/ns/prov#>
SELECT DISTINCT ?organismName ?organismWD ?loc ?lats ?longs WHERE {
	# metabolite data
	{
		SELECT DISTINCT ?organismWD ?organismName WHERE {
			?material sosa:hasSample ?extract ;
		        	sosa:isSampleOf ?organe .
			?organe emi:inTaxon ?organismWD ; # filter metabolite data which is found in wikidata-ids wdx	
				rdfs:label ?organismName .
			?extract sosa:isFeatureOfInterestOf ?lcms .
			?lcms sosa:hasResult ?feature_list .
			?feature_list emi:hasLCMSFeature ?feature .
			?feature emi:hasAnnotation ?sirius_annotation .
			?sirius_annotation a emi:StructuralAnnotation ;
				emi:hasChemicalStructure ?ik2d .
			?ik2d emi:hasSMILES ?smiles ;
				emi:isInChIKey2DOf ?ik .
			?ik emi:isInChIKeyOf ?wd_chem .	# retrieve wikidata-ids for metabolites
			VALUES ?wd_chem { wd:Q27251426 } # filter wikidata_id of metabolites matches wd:Q27251426
		}
	} UNION # union with data from lotus (integrated in wikidata)
	{
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
			?wd_chem wdt:P235 ?ik ;
	    			wdt:P703 ?organismWD .
			 VALUES ?wd_chem { wd:Q27251426 }
			 FILTER(?organismWD = ?source_wdx || ?organismWD = ?target_wdx)	# ensure that organism-wd is either from source or target
			 OPTIONAL { ?organismWD wdt:P225 ?organismName . }
		}
	}
	# interaction data			
	?intxn emi:hasSource ?source ; # retrieve interaction-pairs
		emi:hasTarget ?target .
	?source emi:inTaxon ?source_wdx . # retrieve wikidata-id of source
		?target emi:inTaxon ?target_wdx . # retrieve wikidata-id of target
		?intxn prov:atLocation ?loc ; # retrieve the location of the interaction, and the latitude/longitude
			wgs:lat ?lats ;					
			wgs:long ?longs .
} LIMIT 4000`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","UNION","VALUES","SERVICE","LIMIT","DISTINCT","FROM","WITH","STR","IRI","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 017",slug:"dbgi_017",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List metabolites of plants that interact with plant parasite moth Orgyia postica (wd:Q7102162) .",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
SELECT DISTINCT ?sourceWD ?sourceName ?intxnName ?intxnLabel ?targetWD ?targetName ?ik ?wd_chem WHERE { 
	# interaction data
	{
		SELECT DISTINCT ?sourceWD ?sourceName ?intxnName ?intxnLabel ?targetWD ?targetName WHERE {
                	?intxn emi:hasSource ?source ; # retrieve interaction-pairs
                		emi:hasTarget ?target ;
                                emi:isClassifiedWith ?intxnName .	
                        ?target emi:inTaxon ?targetWD ;	# retrieve wikidata-id for target plant 
				sosa:isSampleOf ?targetSample ;
				rdfs:label ?targetName .
			?source emi:inTaxon ?sourceWD ;	# retrieve wikidata-id for source parasite
				sosa:isSampleOf ?sourceSample ;
				rdfs:label ?sourceName .
			?intxnName rdfs:label ?intxnLabel
                        VALUES ?sourceWD { wd:Q7102162 } # retain results only if the source-WD matches the wikidata-id of Orgyia postica
                }
	}   
	# metabolite data        
	{
		SELECT DISTINCT ?targetWD ?ik ?wd_chem WHERE {
                	?material sosa:hasSample ?extract ;
                      		sosa:isSampleOf ?organe .
                      	?organe emi:inTaxon ?targetWD .	# filter metabolite data which is found in wikidata-ids targetWD
                      	?extract sosa:isFeatureOfInterestOf ?lcms .
                      	?lcms sosa:hasResult ?feature_list .
                      	?feature_list emi:hasLCMSFeature ?feature .
                      	?feature emi:hasAnnotation ?sirius_annotation .
                      	?sirius_annotation a emi:StructuralAnnotation ;
                      		emi:hasChemicalStructure ?ik2d .
                        ?ik2d emi:hasSMILES ?smiles ;
                        	emi:isInChIKey2DOf ?ik .
                      	?ik emi:isInChIKeyOf ?wd_chem .	# retrieve wikidata-ids for metabolites
                }
	} UNION	#union with data from lotus (integrated in wikidata)
	{
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
			?wd_chem wdt:P235 ?ik ;
				wdt:P703 ?targetWD .			
		}
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
               		?targetWD wdt:P141 wd:Q719675 ; # filter source wikidata ids, which have IUCN status (wdt:P141) as near threatened (wd:Q719675) and which is a plant
                        	wdt:P171* wd:Q879246 ;
                     	OPTIONAL { ?targetWD wdt:P225 ?targetName . }
               	}
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","UNION","VALUES","SERVICE","DISTINCT","FROM","WITH","STR","IRI","IF","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 018",slug:"dbgi_018",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List links between host-parasite and host-allelopathic interactions, where host is an agricultural crop, allelopath is usually a plant that inhibits growth of parasites which are usually insects. Inhibition happens by root/stem/leaf exudates of the allelopath.",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX dcterms: <http://purl.org/dc/terms/>
SELECT DISTINCT ?parasiteX_WD ?parasiteName ?intxn1Label ?allelopathX_WD ?allelopathName ?intxn2Label ?agriCrop_WD ?agriCropName ?intxn3Label ?study1_DOI ?study2_DOI ?study3_DOI WHERE {
	
	#part-1 interactions for parasite that negatively effects a plant or an agricultural crop
	?parasiteX1 emi:inTaxon ?parasiteX_WD . # parasite and its WD
	?agriCropX emi:inTaxon ?agriCrop_WD . # agricultural crop and its WD
	?intxn1 emi:hasSource ?parasiteX1 ; # retrieve interaction pairs part-1
    		emi:hasTarget ?agriCropX ;
    		emi:isClassifiedWith ?intxnName1 .
	?intxnName1 rdfs:label ?intxn1Label .
	# filter the interactions either by FILTER (interaction label) or VALUES (interaction ID)
	FILTER ((STR(?intxn1Label) IN ('hasHost','rootparasiteOf', 'hemiparasiteOf', 'hyperparasiteOf','rootparasiteOf','endoparasiteOf','ectoparasiteOf')))
	# VALUES ?intxnName1 { <http://purl.obolibrary.org/obo/RO_0002454> <http://purl.obolibrary.org/obo/RO_0002632> <http://purl.obolibrary.org/obo/RO_0002634> <http://purl.obolibrary.org/obo/RO_0002237> <http://purl.obolibrary.org/obo/RO_0002553> <http://purl.obolibrary.org/obo/RO_0008503> <http://purl.obolibrary.org/obo/RO_0002444> <http://purl.obolibrary.org/obo/RO_0002236> }
    	OPTIONAL { 
		?intxn1 dcterms:bibliographicCitation ?study1_DOI . 
	}

	# part-2 interactions for allelopath that impacts a plant or an agricultural crop
	{
		SELECT * WHERE { 
			?allelopathX1 emi:inTaxon ?allelopathX_WD . #allelopath and its WD
			?intxn2 emi:hasSource ?allelopathX1 ; #retrieve interaction pairs part-2
    				emi:hasTarget ?agriCrop ;
    				emi:isClassifiedWith ?intxnName2 .
			?intxnName2 rdfs:label ?intxn2Label .
			VALUES ?intxnName2 { <http://purl.obolibrary.org/obo/RO_0002555> } # filter interactions which have interaction ID of allelopathOf
			?agriCrop emi:inTaxon ?agriCrop_WD ; # retreieve agriCrop and its WD
	           		rdfs:label ?agriCropName .
    			OPTIONAL { 
				?intxn2 dcterms:bibliographicCitation ?study2_DOI . 
			}
			SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
	 			?agriCrop_WD wdt:P171* wd:Q879246 . # keep the interaction pairs part-2 only if agriCrop_WD has Kingdom Archaeplastida (wd:Q879246) in its lineage, this is to ensure the agriCrop is really a plant
			}
		}
	}

    	#part-3 interactions for parasite that impacts the allelopath
	{
		SELECT * WHERE {
	    		?intxn3 emi:hasSource ?parasiteX ; # retrieve interaction pairs part-3
	    			emi:hasTarget ?allelopathX ;
	    			emi:isClassifiedWith ?intxnName3 . 
			?intxnName3 rdfs:label ?intxn3Label .
			# filter interactions by interaction label (hasHost and allelopathOf) or interaction ID
	    		FILTER (STR(?intxn3Label) IN ('hasHost','hasAllelopath'))
			# VALUES ?intxnName3 { <http://purl.obolibrary.org/obo/RO_0020301> <http://purl.obolibrary.org/obo/RO_0002454> }
			?parasiteX emi:inTaxon ?parasiteX_WD ; # retreieve wikidata-ids for parasiteX
				rdfs:label ?parasiteName .
			?allelopathX emi:inTaxon ?allelopathX_WD ; # retreieve wikidata-ids for allelopathX
				rdfs:label ?allelopathName .
			OPTIONAL { 
				?intxn3 dcterms:bibliographicCitation ?study3_DOI . 
	    		}
		}
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","GRAPH","VALUES","SERVICE","DISTINCT","WITH","ALL","STR","IF"],category:"sib-swiss dbgi"},{name:"dbgi - 019",slug:"dbgi_019",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"A list of interactions depicting connections between parasatoids harmful for insects living as parasites on plants.",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX dcterms: <http://purl.org/dc/terms/>
SELECT DISTINCT ?parasitoidX_WD ?parasitoidName ?intxn2Label ?parasiteX_WD ?parasiteName ?intxn3Label ?hostPlant_WD ?hostPlantName ?study2_DOI ?study3_DOI WHERE {
	# part-1 interaction data parasite and parasitoid
	{ 
		SELECT DISTINCT ?parasitoidName ?parasiteName ?parasiteX_WD ?intxn2Label ?parasitoidX_WD ?study2_DOI WHERE {
			?intxn2 emi:hasSource ?parasitoidX ; # retrieve interaction pairs part-1
	    			emi:hasTarget ?parasiteX ;
	    			emi:isClassifiedWith ?intxnName2 .   
			VALUES ?intxnName2 { <http://purl.obolibrary.org/obo/RO_0002208> } # keep interactions only if the interaction-id is RO_0002208 (parasitoidOf), e.g: 'parasitoidX' is 'parasitoidOf' 'parasiteX'
			?intxnName2 rdfs:label ?intxn2Label .		
			?parasitoidX emi:inTaxon ?parasitoidX_WD ; # retreieve wikidata-ids for parasitoidX
			     	rdfs:label ?parasitoidName ;
			     	sosa:isSampleOf ?parasitoidSample . # retrieve scientific name of parasitoidX as listed in GloBI
			?parasiteX emi:inTaxon ?parasiteX_WD ; # retreieve wikidata-ids for parasiteX
				rdfs:label ?parasiteName ;
				sosa:isSampleOf ?parasiteSample . # retreieve scientific name for parasiteX as listed in GloBI
			OPTIONAL { 
				?intxn2 dcterms:bibliographicCitation ?study2_DOI . # optionally retrieve the doi of the study for parasitoid-parasite pairs
			}
		}
	}

	# part-2 interaction data parasite and host plant
	?parasiteX1 emi:inTaxon ?parasiteX_WD .	# check pasarsiteX1 is in wikidata-id parasiteX_WD (the ones obtained from interaction-pairs part-1)
	?intxn3 emi:hasSource ?parasiteX1 ; # retrieve interaction pairs part-2
    		emi:hasTarget ?hostPlant ;
    		emi:isClassifiedWith ?intxnName3 .
	?intxnName3 rdfs:label ?intxn3Label .
	?hostPlant emi:inTaxon ?hostPlant_WD ;	# retreieve wikidata-ids for hostPlant
		rdfs:label ?hostPlantName ;
		sosa:isSampleOf ?hostPlantSample . # retreieve scientific name for hostPlant as listed in GloBI
	OPTIONAL { 
		?intxn3 dcterms:bibliographicCitation ?study3_DOI . 
	}
    	FILTER (!(STR(?intxn3Label) IN ('visits', 'visitsFlowersOf', 'pollinates')))	# keep interactions only if the interaction-names are not 'visits', 'visitsFlowersOf' or 'pollinates'. e.g.: 'parasiteX1' is 'pathogenOf/parasiteOf/..' 'hostPlant'
	SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
	 	?hostPlant_WD wdt:P171* wd:Q879246 . # keep the interaction pairs part-2 only if hostPlant_WD has Kingdom Archaeplastida (wd:Q879246) in its lineage, this is to ensure the hostPlant is really a plant
	}
} LIMIT 1000`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","GRAPH","VALUES","SERVICE","LIMIT","DISTINCT","FROM","WITH","ALL","STR","IF","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 020",slug:"dbgi_020",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Natural producers (and their interactions that might be useful in agriculture) of Onpordopicrin (wd:Q27107580), which might exhibit antimicrobial and cytotoxic activities, especially against human-derived macrophages and against epidermoid carcinoma cells. There is limited scientific evidence to support these claims (https://www.sciencedirect.com/science/article/abs/pii/S138614251500685X).",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX wgs: <http://www.w3.org/2003/01/geo/wgs84_pos#>
PREFIX prov: <http://www.w3.org/ns/prov#>
SELECT DISTINCT ?sourceWD ?sourceName ?intxnName ?targetWD ?targetName ?loc WHERE {
	# trait data
	?intxn emi:hasSource ?xOrg ; # retrieve interaction-pairs
        	emi:hasTarget ?yOrg ;
		emi:isClassifiedWith ?intxnName ;
		prov:atLocation ?loc ; # retrieve location, latitude and longitude of interactions
		wgs:lat ?lats ;					
		wgs:long ?longs .
	?xOrg emi:inTaxon ?sourceWD ; # retrieve interaction-source and its scientific name as listed in GloBI
		rdfs:label ?sourceName ;
		sosa:isSampleOf ?sourceSample .
	?yOrg emi:inTaxon ?targetWD ; # retrieve interaction-target and its scientific name as listed in GloBI
		rdfs:label ?targetName ;
		sosa:isSampleOf ?targetSample .

	# metabolite data
        { 
		SELECT DISTINCT ?targetWD ?wd_chem WHERE {		
			?material sosa:hasSample ?extract ;
                        	sosa:isSampleOf ?organe .
                	?organe emi:inTaxon ?targetWD .	# filter by target wikidata-id
                	?extract sosa:isFeatureOfInterestOf ?lcms .
                	?lcms sosa:hasResult ?feature_list .
                	?feature_list emi:hasLCMSFeature ?feature .
                	?feature emi:hasAnnotation ?sirius_annotation .
                	?sirius_annotation a emi:StructuralAnnotation ;
	                	emi:hasChemicalStructure ?ik2d .
                	?ik2d emi:hasSMILES ?smiles ;
                      		emi:isInChIKey2DOf ?ik .
                	?ik emi:isInChIKeyOf ?wd_chem .
                	VALUES ?wd_chem { wd:Q27107580 } # filter on the wikidata-id of Onpordopicrin
		}
	} UNION	# union with the lotus metabolite data from wikidata
	{ 
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
			?wd_chem wdt:P235 ?ik ;
   				wdt:P703 ?targetWD .
		 	VALUES ?wd_chem { wd:Q27107580 }
		}
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","VALUES","SERVICE","DISTINCT","FROM","WITH","STR","IRI","IF","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 021",slug:"dbgi_021",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List possible interactions of plants that can produce Norhyocyamine (wd:Q27107545), a plant secondary metabolite.",context:null,inidces:[],query:`PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX emi: <https://purl.org/emi#>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
SELECT DISTINCT ?sourceWD ?sourceName ?intxnName ?intxnId ?targetWD ?targetName WHERE {
	# interaction data
	{
        	?intxn emi:hasSource ?xOrg ; # retrieve interaction-pairs
                	emi:hasTarget ?yOrg ;
                        emi:isClassifiedWith ?intxnId .
                ?intxnId rdfs:label ?intxnName .
                ?xOrg emi:inTaxon ?sourceWD ;
			rdfs:label ?sourceName .
		?yOrg emi:inTaxon ?targetWD ;
			rdfs:label ?targetName .
        }

	# metabolite data
	FILTER (?xWD = ?sourceWD || ?xWD = ?targetWD)	# ensure that xWD can be both source/target from the interaction-pairs retrieved before
        {
		SELECT DISTINCT ?xWD ?wd_chem WHERE {
                	?material sosa:hasSample ?extract ;
                        	sosa:isSampleOf ?organe .
                        ?organe emi:inTaxon ?xWD .
                        ?extract sosa:isFeatureOfInterestOf ?lcms .
                        ?lcms sosa:hasResult ?feature_list .
                        ?feature_list emi:hasLCMSFeature ?feature .
                        ?feature emi:hasAnnotation ?sirius_annotation .
                        ?sirius_annotation a emi:StructuralAnnotation ;
                        	emi:hasChemicalStructure ?ik2d .
                        ?ik2d emi:hasSMILES ?smiles ;
                        	emi:isInChIKey2DOf ?ik .
                        ?ik emi:isInChIKeyOf ?wd_chem .
                        VALUES ?wd_chem { wd:Q27107545 }
                }
	}  UNION # union with data from wikidata (lotus)
	{ 
		SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {
			?wd_chem wdt:P235 ?ik ;
   				wdt:P703 ?xWD .
			VALUES ?wd_chem { wd:Q27107545 }
		}
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","VALUES","SERVICE","DISTINCT","FROM","WITH","STR","IRI","IF","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 022",slug:"dbgi_022",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve data of 4 traits in leaf economics spectrum.",context:null,inidces:[],query:`PREFIX emi: <https://purl.org/emi#>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX qudt: <https://qudt.org/2.1/schema/qudt#>
SELECT DISTINCT ?source_wdx ?sourceName ?traitCategory ?traitLabel ?tryDataVal ?unit ?unitComment WHERE 
	{
		# trait data
		# trait/non-trait data for trySpName (scientific name of plant species as listed in trydb)
		?trySpObs sosa:isSampleOf ?trySpName ;
			rdfs:label ?sourceName ;
			sosa:isFeatureOfInterestOf ?tryObId .
		?tryObId sosa:hasResult ?tryData .
		?trySpName emi:inTaxon ?source_wdx . # wikidata-ids wdx for trySpName
		?tryData rdfs:label ?traitLabel ;
			rdf:type emi:Trait ; #retrieve data which is labelled as 'Trait' and its values
			rdf:value ?tryDataVal ;
			qudt:hasUnit ?unit . #retrieve units for Trait data
		OPTIONAL {
			?tryData rdfs:comment ?unitComment .
		} #retrieve comments (containing original unprocessed unit information - necessary for understanding some data) for Trait data
  		# retrieve trait labels
		BIND (IF(REGEX(STR(?traitLabel),"SLA","i"),"Specific leaf area",IF(REGEX(STR(?traitLabel),"leaf dry matter content","i"),"Leaf dry matter content",IF(REGEX(STR(?traitLabel),"leaf nitrogen content","i"),"Leaf nitrogen content",IF(REGEX(STR(?traitLabel),"leaf phosphorus content","i"),"Leaf phosphorus content","Other")))) AS ?traitCategory)
  		# filter to LES categories only
  		FILTER (?traitCategory != "Other")
	}
	# group results and retain only the ones from organisms with all four traits available 
	GROUP BY ?source_wdx ?sourceName ?traitCategory ?traitLabel ?tryDataVal ?unit ?unitComment
	HAVING (COUNT(?traitCategory) = 4)
	# order by organism and trait names
	ORDER BY ?source_wdx ?traitCategory`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","BIND","VALUES","GROUP BY","ORDER BY","HAVING","DISTINCT","FROM","WITH","ALL","STR","REGEX","IF","COUNT","SAMPLE"],category:"sib-swiss dbgi"},{name:"dbgi - 023",slug:"dbgi_023",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve metabolites of organisms with data available from 4 traits in leaf economics spectrum.",context:null,inidces:[],query:`PREFIX emi: <https://purl.org/emi#>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>
PREFIX sosa: <http://www.w3.org/ns/sosa/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
SELECT DISTINCT ?structure_inchikey ?wd_chem ?source_wdx ?sourceName  WHERE 
	{
		# trait data
		
		# trait/non-trait data for trySpName (scientific name of plant species as listed in trydb)
		?trySpObs sosa:isSampleOf ?trySpName ; 
			rdfs:label ?sourceName ;
			sosa:isFeatureOfInterestOf ?tryObId .
		?tryObId sosa:hasResult ?tryData .
		?trySpName emi:inTaxon ?source_wdx . # wikidata-ids wdx for trySpName
		?tryData rdfs:label ?traitLabel ; 
			rdf:type emi:Trait ; #retrieve data which is labelled as 'Trait' and its values
		
		# retrieve trait labels
		BIND (
        		IF (REGEX(STR(?traitLabel), "SLA", "i"), "Specific leaf area",
    			IF (REGEX(STR(?traitLabel), "leaf dry matter content", "i"), "Leaf dry matter content",
	    		IF (REGEX(STR(?traitLabel), "leaf nitrogen content", "i"), "Leaf nitrogen content",
    			IF (REGEX(STR(?traitLabel), "leaf phosphorus content", "i"), "Leaf phosphorus content",
    			"Other")))) AS ?traitCategory
  		)
  		
		# filter to LES categories only
	  	FILTER (?traitCategory != "Other")
		
		# metabolite data
		{ 
			SELECT ?source_wdx ?structure_inchikey ?wd_chem WHERE {	
				?material sosa:hasSample ?extract ;				
		        		sosa:isSampleOf ?organe .	
				?organe emi:inTaxon ?source_wdx . # filter metabolite data which is found in wikidata-ids wdx 
				?extract sosa:isFeatureOfInterestOf ?lcms .
			    	?lcms sosa:hasResult ?feature_list .
				?feature_list emi:hasLCMSFeature ?feature .
				?feature emi:hasAnnotation ?sirius_annotation .
				?sirius_annotation a emi:StructuralAnnotation ;
					emi:hasChemicalStructure ?ik2d .
				?ik2d emi:hasSMILES ?smiles ;
		    			emi:isInChIKey2DOf ?structure_inchikey .
				?structure_inchikey emi:isInChIKeyOf ?wd_chem .	# wikidata-ids for metabolites
			}
		}
		UNION # union with data from lotus (integrated in wikidata)
		{
			SERVICE <https://qlever.cs.uni-freiburg.de/api/wikidata> {		
				?wd_chem wdt:P235 ?structure_inchikey;				
	   				wdt:P703 ?source_wdx .	
			}
		}
	}

	# group results and retain only the ones from organisms with all four traits available 
	GROUP BY ?source_wdx ?sourceName ?traitCategory ?structure_inchikey ?wd_chem
	HAVING(COUNT(?traitCategory) = 4)

	# order by organism and trait names
	ORDER BY ?source_wdx ?traitCategory`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","VALUES","SERVICE","GROUP BY","ORDER BY","HAVING","DISTINCT","FROM","WITH","ALL","STR","REGEX","IRI","IF","COUNT","SAMPLE"],category:"sib-swiss dbgi"}],lP=[{name:"GlyConnect - 1",slug:"GlyConnect_1",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get all Glyconnect proteins their Uniprot reference isoform",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>

SELECT distinct ?glycoprotein ?isoform
where {
?glycosite faldo:reference ?isoform .
?specificglycosite faldo:location ?glycosite .
?glycoprotein glycan:glycosylated_at ?specificglycosite .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","IF"],category:"sib-swiss GlyConnect"},{name:"GlyConnect - 2",slug:"GlyConnect_2",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get all Glyconnect glycan with their graphical SNFG representation",context:null,inidces:[],query:`PREFIX foaf: <http://xmlns.com/foaf/0.1/>
PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>

SELECT distinct ?structure ?image 
where {
?structure glycan:glycosylates_at ?specificglycosite .
?structure foaf:depiction ?image  
  }`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","STR","IF"],category:"sib-swiss GlyConnect"},{name:"GlyConnect - 3",slug:"GlyConnect_3",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all glycosylations (glycan, protein, position) with their reference articles",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX foaf: <http://xmlns.com/foaf/0.1/>
PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>

SELECT distinct ?glycoprotein ?isoform ?position ?structure ?image  ?citation ?pmid
where {
?glycosite faldo:reference ?isoform .
?glycosite faldo:position ?position .
?specificglycosite faldo:location ?glycosite .
?glycoprotein glycan:glycosylated_at ?specificglycosite .
?structure glycan:glycosylates_at ?specificglycosite .
?structure foaf:depiction ?image .
?refconjugate glycan:has_protein_part ?glycoprotein .
?refconjugate glycan:published_in ?citation . 
?citation foaf:primaryTopicOf ?pmid .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","STR","IF"],category:"sib-swiss GlyConnect"},{name:"GlyConnect - 4",slug:"GlyConnect_4",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all glycosylations (glycan, protein, position) with their associated disease where the name start with 'cancer'",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sio: <http://semanticscience.org/resource/>

SELECT distinct ?glycoprotein ?isoform ?position ?structure ?disease ?diseasename
where {
?glycosite faldo:reference ?isoform .
?glycosite faldo:position ?position .
?specificglycosite faldo:location ?glycosite .
?glycoprotein glycan:glycosylated_at ?specificglycosite .
?structure glycan:glycosylates_at ?specificglycosite .
?refconjugate glycan:has_protein_part ?glycoprotein .
?refconjugate glycan:has_association ?refconjugatedisease .
?refconjugatedisease sio:SIO_000628 ?disease .
?disease rdfs:label ?diseasename .
FILTER regex(?diseasename, "^cancer", "i") .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","STR","REGEX","IF"],category:"sib-swiss GlyConnect"},{name:"GlyConnect - 5",slug:"GlyConnect_5",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get all Glyconnect reference papers",context:null,inidces:[],query:`PREFIX foaf: <http://xmlns.com/foaf/0.1/>
PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>

SELECT distinct ?citation ?pmid
where {
?refconjugate glycan:published_in ?citation. 
?citation foaf:primaryTopicOf ?pmid .  
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss GlyConnect"}],cP=[{name:"HAMAP - 1",slug:"HAMAP_1",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all hamap rules",context:null,inidces:[],query:`PREFIX sh: <http://www.w3.org/ns/shacl#>

SELECT ?rule
WHERE
{
    ?rule sh:construct ?query  .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","CONSTRUCT","WHERE","STR"],category:"sib-swiss HAMAP"},{name:"HAMAP - 3",slug:"HAMAP_3",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all hamap rules that deal with Catalytic Activity",context:null,inidces:[],query:`PREFIX sh: <http://www.w3.org/ns/shacl#>


SELECT ?rule ?query
WHERE
{
    ?rule sh:construct ?query  .
    FILTER (CONTAINS(?query, 'Catalytic_Activity_Annotation'))
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","CONSTRUCT","WHERE","FILTER","STR","CONTAINS"],category:"sib-swiss HAMAP"},{name:"HAMAP - 4",slug:"HAMAP_4",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all hamap rules that annotate a Rhea reaction with an EC enzyme class",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX sp: <http://spinrdf.org/sp#>
PREFIX up: <http://purl.uniprot.org/core/>


SELECT ?rule ?enzymeClass
WHERE
{
 ?rule a sp:Construct ;
     sp:templates/rdf:rest*/rdf:first ?annotationsToAdd .
 ?annotationsToAdd sp:predicate up:enzymeClass ;
                   sp:object ?enzymeClass .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","CONSTRUCT","WHERE","ADD","STR"],category:"sib-swiss HAMAP"}],uP=[{name:"neXtProt - NXQ 00001",slug:"neXtProt_NXQ_00001",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins phosphorylated and located in the cytoplasm",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?cytoloc {nextprot_cv:SL-0086 nextprot_cv:GO_0005737} # SL and GO values for cytoplasm
 ?entry :isoform ?iso.
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0597. # Phosphorylated
 ?iso :cellularComponent ?loc .
 ?loc :term /:childOf ?cytoloc.
 filter not exists {?loc :negativeEvidence ?negev} # No negative localization evidence
}

# Assign values to the variable ?cytoloc
# The values correspond to the two controlled vocabulary (cv) terms for cytoplam:
# SL-0086 is the UniProt subcellular location term
# GO_0005737 is the Gene Ontology cellular component term
#
# Use the same name of the variable (?iso and ?loc) in several statements.
# It is the name of the variable that enforces the constraints.
#
# Phosphorylated proteins are retrieved using a keyword:
# KW-0597 is the UniProt keyword for phosphorylated
#
# Use :childOf to include children of a term.
# Cytosol (SL-0091), the child term of cytoplasm, will thus be included.
#
# Exclude negative locatization evidences.
# This enforces that the protein is located in the cytoplasm.`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","USING","STR","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00002",slug:"neXtProt_NXQ_00002",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are located in both the nucleus and in the cytoplasm",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?cytoloc {nextprot_cv:GO_0005737 nextprot_cv:SL-0086} # GO and SL values for cytoplasm
 values ?nucloc {nextprot_cv:GO_0005634 nextprot_cv:SL-0191} # GO and SL values for nucleus
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc1, ?loc2 .
 ?loc1 :term /:childOf ?cytoloc .
 ?loc2 :term /:childOf ?nucloc .
 filter not exists {?loc1 :negativeEvidence ?negev} # No negative localization evidence
 filter not exists {?loc2 :negativeEvidence ?negev} # No negative localization evidence
}

# Note the dot "." at the end of statements.
# If missing, you will get a syntax error.
#
# Use a comma to retrieve two localizations in:
# ?iso :cellularComponent ?loc1, ?loc2 .`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00003",slug:"neXtProt_NXQ_00003",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with 7 transmembrane regions",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :topology ?statement.
 ?statement a :TransmembraneRegion.
} group by ?entry ?iso having(count( ?statement)=7)

# 'a' can also be used instead of 'rdf:type'
# 'a' is a synonym of 'rdf:type'
#
# Apply the grouping criterion (group by) for the results.
# This groups entries with isoforms having 7 transmembrane regions.
# Apply the aggregate function (count) to the variable ?statement.
# This counts the number of transmembrane region statements for each isoform.`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","WITH","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00004",slug:"neXtProt_NXQ_00004",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins expressed in brain with observed IHC expression \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 # get all expression
 ?iso :expression ?e1.
 # highly expressed in brain
 ?e1 :term/:childOf nextprot_cv:TS-0095;:evidence/:observedExpression :High.
 # not expressed in testis
 ?iso :undetectedExpression ?e2.
 ?e2 :term nextprot_cv:TS-1030.
# Use the semicolon ';' to refer to the previous subject (?e1)
# Only IHC data has observed expression "High" so ECO is not specified.
#
# Note that we also exclude expression detected in testis.
 filter not exists { ?iso :detectedExpression / :term / :childOf nextprot_cv:TS-1030 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","ALL","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00005",slug:"neXtProt_NXQ_00005",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins located in mitochondrion and that lack a transit peptide",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?mitoloc {nextprot_cv:SL-0173 nextprot_cv:GO_0005739 } # SL and GO values for mitochondrion
 ?entry :isoform ?iso.
 filter not exists { ?iso :uniprotKeyword /:term nextprot_cv:KW-0809 } # Transit peptide
 ?iso :cellularComponent ?loc .
 ?loc :term /:childOf ?mitoloc.
 filter not exists {?loc :negativeEvidence ?negev} # No negative localization evidence
}

# Variables start with ? and can be given any name.
# Variables such as ?entry and ?mitoloc (location in the mitochondrion) are easily understood by humans.`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","WITH","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00006",slug:"neXtProt_NXQ_00006",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose genes are on chromosome 13 and are associated with a disease",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene / :chromosome "13"^^xsd:string.
 ?entry :isoform ?iso.
 {
	 ?iso :medical / rdf:type :Disease.
 } union {
 ?iso :uniprotKeyword / :term ?kw .
	 ?kw :termType "Disease"^^xsd:string
	 filter (?kw != nextprot_cv:KW-0656)
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","STR","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00007",slug:"neXtProt_NXQ_00007",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins associated with diseases that are associated with cardiovascular aspects",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform /:medical / :term /:related / :childOf nextprot_cv:D002318.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00008",slug:"neXtProt_NXQ_00008",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose genes are less than 50000 bp away from the location of the gene coding for protein p53",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>

SELECT DISTINCT ?entry WHERE {
 nextprot:NX_P04637 :gene /:begin ?s;:gene/:chromosome ?chr.
 ?entry :gene/:begin ?gs;:gene/:chromosome ?chr.
 filter ( ?gs > (?s -50000) && ?gs <= (?s +50000) && ?entry != nextprot:NX_P04637 )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00009",slug:"neXtProt_NXQ_00009",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with 3 disulfide bonds and that are not annotated as hormones ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?isoform.
 ?isoform :ptm ?statement.
 ?statement a :DisulfideBond.
 filter not exists { ?entry :isoform / :function / :term /:childOf nextprot_cv:GO_0005179. } # GO Hormone activity
} group by ?entry ?isoform having (count(?statement) =3 )`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","HAVING","DISTINCT","EXISTS","NOT EXISTS","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00010",slug:"neXtProt_NXQ_00010",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are glycosylated and are not located in the membrane",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0325.
 filter not exists { ?iso :uniprotKeyword / :term nextprot_cv:KW-0812.} # KW for Transmembrane
 filter not exists { ?iso :topology / :term nextprot_cv:CVTO_0004 } # intramembrane segment
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00011",slug:"neXtProt_NXQ_00011",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are expressed in liver and involved in transport",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :detectedExpression/:term/:childOf nextprot_cv:TS-0564.
 nextprot_cv:KW-0813 :related ?tra.
 ?iso :generalAnnotation / :term ?tra.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00012",slug:"neXtProt_NXQ_00012",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that interact with protein RBM17 and that are involved in splicing",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 nextprot:NX_Q96I25 :isoform / :interaction / :interactant ?entry.
 ?entry :isoform / :uniprotKeyword / :term nextprot_cv:KW-0508
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00013",slug:"neXtProt_NXQ_00013",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that have a protein kinase domain but lack protein kinase activity",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :region /:term nextprot_cv:DO-00529.
 filter not exists { ?iso :enzymeClassification / :term /:childOf nextprot_cv:2_7_-_- }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00014",slug:"neXtProt_NXQ_00014",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with one SH2 and two SH3 domains",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 #with 1 SH3
 {select ?iso where{?iso :region ?stat1. ?stat1 :term nextprot_cv:DO-00614
 } group by ?iso having(count( ?stat1)=1)}
 #with 2 SH2
 {select ?iso where{?iso :region ?stat2. ?stat2 :term nextprot_cv:DO-00615
 } group by ?iso having(count( ?stat2)=2)}
} group by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","WITH","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00015",slug:"neXtProt_NXQ_00015",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a PDZ domain that interact with at least one protein which is expressed in brain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :region / :term nextprot_cv:DO-00477. #PDZ domain
 ?iso :binaryInteraction/:interactant/:isoform?/:detectedExpression/:term/:childOf nextprot_cv:TS-0095 #brain
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00016",slug:"neXtProt_NXQ_00016",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a mature chain of less than 1000 amino acids which are secreted and do not contain cysteines in the mature chain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?sloc {nextprot_cv:GO_0005576 nextprot_cv:SL-0243} # GO and SL values for secreted
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?anno .
 ?anno :term /:childOf ?sloc .
 filter not exists { ?anno :negativeEvidence ?_ }
 #
 # filter not exists {?iso :cellularComponent /:term nextprot_cv:GO_0070062 .
 # filter not exists {?iso :cellularComponent /:term /:childOf nextprot_cv:SL-0243 .}
 # }
 # you can uncomment this to filters out extracellular exosome only location (2240 entries),
 # most of them evidenced only by large-scale proteomic analysis
 #
 ?iso :sequence / :chain ?seq.
 ?iso :matureProtein [ :start ?mstart ; :end ?mend]
 filter ((?mend-?mstart > 0) && (?mend-?mstart < 1000))
 bind (?mend - ?mstart + 1 as ?mlen)
 bind (substr(?seq, ?mstart, ?mlen) as ?mseq)
 filter (!regex (?mseq,'C'))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","VALUES","DISTINCT","STR","REGEX","EXISTS","NOT EXISTS","SUBSTR","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00017",slug:"neXtProt_NXQ_00017",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins larger than 1000 amino acids that are located in the nucleus and expressed in the nervous system",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?nucloc {nextprot_cv:GO_0005634 nextprot_cv:SL-0191} # GO and SL values for nucleus

 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term/:childOf ?nucloc.
 filter not exists {?loc :negativeEvidence ?negev} # No negative localization evidence
 ?iso :detectedExpression/:term/:childOf nextprot_cv:TS-1313.
 ?iso :sequence/:length ?len.
 filter (?len>1000)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00018",slug:"neXtProt_NXQ_00018",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are acetylated and methylated and located in the nucleus",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?nucloc {nextprot_cv:GO_0005634 nextprot_cv:SL-0191} # GO and SL values for nucleus

 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term/:childOf ?nucloc.
 filter not exists {?loc :negativeEvidence ?negev} # No negative localization evidence
 ?iso :uniprotKeyword/:term nextprot_cv:KW-0007,nextprot_cv:KW-0488. # acetylated and methylated
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00019",slug:"neXtProt_NXQ_00019",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with more than 12 WD repeats (ie: with at least two beta-propellers)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 {select ?iso where{
 ?iso :repeat ?rep. ?rep :term nextprot_cv:DO-00722
 }
 group by ?iso having(count( ?rep) > 12)}
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00020",slug:"neXtProt_NXQ_00020",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least 2 HPA antibodies whose genes are located on chromosome 21 and that are highly expressed according to IHC in heart",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

select ?entry WHERE {
 SELECT DISTINCT ?entry ?id WHERE {
 ?entry :gene / :chromosome "21"^^xsd:string .
 ?entry :isoform / :expressionProfile ?s1.
 ?s1 :term / :childOf nextprot_cv:TS-0445.
 ?s1 :evidence ?evi.
 ?evi :observedExpression :High.
 ?evi :evidenceCode nextprot_cv:ECO_0001055.
 ?entry :isoform / :antibodyMapping ?map.
	 ?map :evidence / :reference / :accession ?id .
 }
} group by ?entry having (count(?id)>=2)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","STR","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00021",slug:"neXtProt_NXQ_00021",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one HPA antibody that are located in the peroxisome",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?peroxiloc {nextprot_cv:GO_0005777 nextprot_cv:SL-0204} # GO and SL values for peroxisome

 ?entry :isoform ?iso.
 ?iso :antibodyMapping ?map.
 ?iso :cellularComponent ?loc .
 ?loc :term /:childOf ?peroxiloc. # peroxisomal
 filter not exists { ?loc :negativeEvidence ?negev. } # No negative localization evidence
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00022",slug:"neXtProt_NXQ_00022",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with no function annotated",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 filter not exists { ?iso :functionInfo ?_. }
 filter not exists { ?iso :catalyticActivity ?_ .}
 filter not exists { ?iso :transportActivity ?_ .}
 filter not exists { ?iso :pathway ?_. }
 filter not exists {
 ?iso :function / :term ?fterm .
			 filter(?fterm != nextprot_cv:GO_0005524 && ?fterm != nextprot_cv:GO_0000287 && ?fterm != nextprot_cv:GO_0005515 && ?fterm != nextprot_cv:GO_0042802
			 && ?fterm != nextprot_cv:GO_0008270 && ?fterm != nextprot_cv:GO_0051260 && ?fterm != nextprot_cv:GO_0005509
					 && ?fterm != nextprot_cv:GO_0003676 && ?fterm != nextprot_cv:GO_0003824 && ?fterm != nextprot_cv:GO_0007165 && ?fterm != nextprot_cv:GO_0035556
					 && ?fterm != nextprot_cv:GO_0046914 && ?fterm != nextprot_cv:GO_0046872)
 }
 filter not exists { ?entry :existence :Uncertain } # Remove PE5 proteins
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","MOVE","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00023",slug:"neXtProt_NXQ_00023",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are involved in transport and located in a membrane and that are not glycosylated (experimentally or predicted)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 {
 ?iso :goBiologicalProcess ?gofunc .
	?gofunc :term / :childOf nextprot_cv:GO_0006810. # GO value for transport
	filter not exists {?gofunc :negativeEvidence ?negev} # No negative function evidence
 } union {
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0813. # KW for transport
 }
 {
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0812. # transmembrane
 } union {
 ?iso :topology / :term nextprot_cv:CVTO_0004. # intramembrane
 }
 filter not exists { ?iso :uniprotKeyword / :term nextprot_cv:KW-0325 } # not a glycoprotein
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00024",slug:"neXtProt_NXQ_00024",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with more than 10 reported interactions (of 'gold' quality)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform/:binaryInteraction ?interaction.
 ?interaction :interactant ?interactant; :quality :GOLD.
 ?interactant a :Entry.
} group by ?entry having (count(distinct ?interactant) > 10)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00025",slug:"neXtProt_NXQ_00025",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least 50 interactors that are not associated with a disease",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :binaryInteraction/:interactant ?interactant.

 # with no disease
 filter not exists {
 {
 ?iso :medical / rdf:type :Disease.
 } union {
 ?iso :uniprotKeyword / :term ?kw .
 ?kw :termType "Disease"^^xsd:string
 filter (?kw != nextprot_cv:KW-0656)
 }
 }

} group by ?entry ?iso having (count(?interactant) >= 50)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","GROUP BY","HAVING","DISTINCT","WITH","STR","EXISTS","NOT EXISTS","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00026",slug:"neXtProt_NXQ_00026",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins interacting with at least one protein which is located in the mitochondrion",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?mitoloc {nextprot_cv:SL-0173 nextprot_cv:GO_0005739} # SL and GO values for mitochondrion

 ?entry :isoform / :interaction ?it .
 ?it :quality :GOLD . # remove this filter for lower quality interactions
 ?it :interactant ?interactant.
 ?interactant :isoform? / :cellularComponent ?loc .
 # the question mark at :isoform allows to select also isoform-specific interactions
 ?loc :term / :childOf ?mitoloc.
 filter not exists { ?loc :negativeEvidence ?negev. } # No negative localization evidence
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","MOVE","ALL","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00027",slug:"neXtProt_NXQ_00027",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with one or more glycosylation sites reported in PubMed:20570859 or PubMed:14760718",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry ?publications WHERE {
 values ?pmid {"20570859"^^xsd:string "14760718"^^xsd:string}
 # get all assertions from the publications
 ?entry :isoform/:glycosylationSite ?statement.
 ?statement :evidence/:reference/:from ?xref .
 ?xref :accession ?pmid ; :provenance db:PubMed .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","DISTINCT","FROM","ALL","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00028",slug:"neXtProt_NXQ_00028",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins associated with a disease but without a disease-causing amino-acid substitution variant",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 {
	 ?iso :medical / rdf:type :Disease.
 } union {
 ?iso :uniprotKeyword / :term ?kw .
	 ?kw :termType "Disease"^^xsd:string
	 filter (?kw != nextprot_cv:KW-0656)
 }
 filter ( not exists { ?entry :isoform / :variant / :disease ?_ . } )
 filter ( not exists {
	?entry :isoform / :proteoform ?pf.
	?pf :difference / :evidence / :negative false. # positive evidence only
	?pf :diseaseRelatedVariant / :term nextprot_cv:ME_0000013 . # cause of disease
 } )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","STR","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00029",slug:"neXtProt_NXQ_00029",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins anchored to the membrane via a GPI-anchor",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :lipidationSite / rdfs:comment ?mod.
 filter(contains(?mod,"GPI-anchor")).
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00030",slug:"neXtProt_NXQ_00030",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose gene is located in chromosome 2 that belongs to families with at least 5 members in the human proteome",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

select ?entry WHERE {
 ?entry :familyName / :term/^:term/^:familyName ?member.
 ?entry :gene / :chromosome "2"^^xsd:string .
} group by ?entry having ( count(distinct ?member) >= 5 )`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","STR","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00031",slug:"neXtProt_NXQ_00031",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with more than 10 alternative isoforms",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
} group by ?entry having (count(?iso)>10)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00032",slug:"neXtProt_NXQ_00032",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a coiled coil region and that are involved in transcription but do not contain a bZIP domain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :function ?func .
 ?func :term / :childOf nextprot_cv:GO_0006351. # Transcription
 filter not exists {?func :negativeEvidence ?negev. } # no negative evidence
 ?iso :region/rdf:type :CoiledCoilRegion.
 filter not exists { ?iso :region/:term nextprot_cv:DO-00078 } # Bzip domain
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00033",slug:"neXtProt_NXQ_00033",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one phosphotyrosine but no phosphoserine or phosphothreonine",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 select ?entry ?iso (count(?ptm) as ?ptmCnt) WHERE {
 ?entry :isoform ?iso.
 ?iso :modifiedResidue ?ptm.
 ?ptm :term nextprot_cv:PTM-0255 #phosphotyrosine.
 filter (
 not exists { ?iso :modifiedResidue / :term nextprot_cv:PTM-0253. } #phosphoserine
 &&
 not exists { ?iso :modifiedResidue / :term nextprot_cv:PTM-0254. } #phosphothreonine
 )
 } group by ?entry ?iso having ( count(?ptm) >= 1 )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","HAVING","DISTINCT","IF","EXISTS","NOT EXISTS","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00034",slug:"neXtProt_NXQ_00034",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one homeobox domain and with at least one variant in the homeobox domain(s)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 # with >=1 homeobox domain
 ?iso :region ?st1.
 ?st1 :term nextprot_cv:DO-00312;:start ?start;:end ?end.
 # with >=1 variant
 ?iso :variant ?var.
 ?var :start ?varpos.
 # one variant in the homeobox domain
 filter (?varpos >=?start && ?varpos <=?end)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","WITH","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00035",slug:"neXtProt_NXQ_00035",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins located in the mitochondrion and which are enzymes",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?mitoloc {nextprot_cv:SL-0173 nextprot_cv:GO_0005739} # SL and GO values for mitochondrion
 ?entry :isoform / :enzymeClassification ?_ .
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term / :childOf ?mitoloc.
 filter not exists { ?loc :negativeEvidence ?negev. } # No negative localization evidence
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00036",slug:"neXtProt_NXQ_00036",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are oxidoreductases and that bind to NAD/NADP",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :uniprotKeyword /:term nextprot_cv:KW-0560. #oxidoreductase
 ?iso :uniprotKeyword /:term ?kw
 filter( ?kw in (nextprot_cv:KW-0520,nextprot_cv:KW-0521)) # NAD or NADP
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00037",slug:"neXtProt_NXQ_00037",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that bind RNA but do not contain a RRM domain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 ?iso :uniprotKeyword /:term nextprot_cv:KW-0694. #RNA-binding
 filter not exists {?iso :region/:term nextprot_cv:DO-00581} #RRM
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00038",slug:"neXtProt_NXQ_00038",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one selenocysteine in their sequence",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform/:selenocysteine ?statement.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00039",slug:"neXtProt_NXQ_00039",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a mutagenesis in a position that correspond to an annotated active site",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :mutagenesis /:start ?mutaPos.
 ?iso :activeSite /:start ?actsitePos.
 filter (?mutaPos=?actsitePos)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00040",slug:"neXtProt_NXQ_00040",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are enzymes and with at least one mutagenesis site that decrease or abolish activity",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?nextprot:isoform ?iso.
 ?iso :enzymeClassification ?_ .
 ?iso :mutagenesis/rdfs:comment ?comment
 filter regex(?comment, '(decrease|abolish).*activity','i')
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX","IF"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00041",slug:"neXtProt_NXQ_00041",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are annotated with GO \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :goMolecularFunction ?statement.
 ?statement :negativeEvidence ?ev.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00042",slug:"neXtProt_NXQ_00042",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that bind a metal and are secreted",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?sloc {nextprot_cv:GO_0005576 nextprot_cv:SL-0243} # GO and SL values for secreted
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term /:childOf ?sloc .
 filter not exists {?sloc :negativeEvidence ?negev} # No negative localization evidence
 #filter not exists {?iso :cellularComponent /:term nextprot_cv:GO_0070062 .
					 #filter not exists {?iso :cellularComponent /:term /:childOf nextprot_cv:SL-0243 .}
					#} # filters out extracellular exosome only location (2758 entries)
 ?iso :goMolecularFunction / :term /:childOf nextprot_cv:GO_0046872 . # metal-binding
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00043",slug:"neXtProt_NXQ_00043",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that bind zinc and are not oxidoreductase and not involved in transcription",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0479. # metal-binding
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0862. # zinc
 filter not exists {?iso :uniprotKeyword / :term nextprot_cv:KW-0804} # transcription
 filter not exists {?iso :uniprotKeyword / :term nextprot_cv:KW-0560} # oxidoreductase
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00044",slug:"neXtProt_NXQ_00044",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins involved in the Reactome pathway \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform / :pathway ?p.
 ?p :evidence /:reference ?xref.
 ?xref :provenance db:Reactome; :accession "R-HSA-611105"^^xsd:string.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00045",slug:"neXtProt_NXQ_00045",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with an active site that is a proton acceptor",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry ?comment WHERE {
 ?entry :isoform/ :activeSite /rdfs:comment ?comment.
 filter (contains (?comment,'Proton acceptor'))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00046",slug:"neXtProt_NXQ_00046",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a gene alternative name starting with IL27",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry ?recname ?altname WHERE {
 ?entry :gene ?gn.
 ?gn :recommendedName / rdfs:label ?recname.
 ?gn :alternativeName / rdfs:label ?altname.
 filter (regex(?altname, "^IL27"))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00047",slug:"neXtProt_NXQ_00047",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a gene name that starts with \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene / :name / rdfs:label ?name.
 filter (regex(?name, "^CLDN"@en))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00048",slug:"neXtProt_NXQ_00048",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one variant of the type \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 {
	?entry :isoform ?iso.
	?iso :variant ?var.
	?var :disease ?_.
	?var :original "C"^^xsd:string.
 } UNION {
 ?entry :isoform ?iso .
	?iso :proteoform ?pf.
	?pf :difference ?var.
	?var :evidence / :negative false. # positive evidence
	?pf :diseaseRelatedVariant / :term nextprot_cv:ME_0000013 . # cause of disease
	?var :original "C"^^xsd:string.

 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","DISTINCT","STR","IF","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00049",slug:"neXtProt_NXQ_00049",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one variant of the types \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform/:variant ?statement.
 {?statement :original "A"^^xsd:string;:variation "R"^^xsd:string}
 UNION
 {?statement :original "R"^^xsd:string;:variation "A"^^xsd:string}
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00051",slug:"neXtProt_NXQ_00051",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that have at least one 3D structure solved by NMR",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :pdbMapping ?map.
 ?map :method ?meth
 filter (contains(?meth,'NMR'))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00052",slug:"neXtProt_NXQ_00052",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a sequence that does not contain a lysine in the mature region",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 ?iso :sequence / :chain ?seq.
 ?iso :matureProtein [ :start ?mstart ; :end ?mend]
 bind(?mend - ?mstart + 1 as ?mlen)
 bind(substr(?seq, ?mstart, ?mlen) as ?mseq)
 filter (!regex (?mseq,'K'))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","STR","REGEX","SUBSTR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00053",slug:"neXtProt_NXQ_00053",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are involved in cell adhesion according to GO with an evidence which is not used in automatic assertion nor a sequence similarity evidence used in manual assertion",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform/:function ?statement.
 ?statement :term / :childOf nextprot_cv:GO_0007155.
 filter not exists { ?statement :negativeEvidence ?negev. } # No negative function evidence
 ?statement :evidence/:evidenceCode ?ecode.
 filter not exists { ?ecode :childOf nextprot_cv:ECO_0000501. } # Automatic assertion
 filter not exists { ?ecode :childOf nextprot_cv:ECO_0000250. } # Sequence similarity evidence used in manual assertion
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00054",slug:"neXtProt_NXQ_00054",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :function ?statement.
 ?statement :term / :childOf nextprot_cv:GO_0005102.
 filter not exists { ?statement :negativeEvidence ?negev. } # No negative function evidence
 ?iso :uniprotKeyword/:term nextprot_cv:KW-0391.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00055",slug:"neXtProt_NXQ_00055",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which have genes of length greater than 2 million bp",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene / :length ?l.
 filter (?l > 2000000)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00057",slug:"neXtProt_NXQ_00057",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are located in the mitochondrion with an experimental evidence originating not from HPA or DKFZ-GFP",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT ?entry WHERE {
 values ?mitoloc {nextprot_cv:SL-0173 nextprot_cv:GO_0005739} # SL and GO values for mitochondrion

 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc.
 ?loc :term /:childOf ?mitoloc.
 filter not exists { ?loc :negativeEvidence ?negev. } # No negative localization evidence
 ?loc :evidence /:assignedBy ?src.
 filter ( ?src not in (source:Human_protein_atlas, source:GFP-cDNAEMBL))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00058",slug:"neXtProt_NXQ_00058",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are located on the genome next to a protein which is involved in spermatogenesis",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT (?neighbour as ?entry) (str(?gen) as ?spergen) WHERE {
 {
 SELECT DISTINCT ?chr ?gen ?sperstart (MAX(?ldist) as ?lmin) (MIN(?rdist) as ?rmin) WHERE {
 ?entryx :isoform / :function ?func .
	 ?func :term / :childOf nextprot_cv:GO_0007283 . # Involvment in spermatogenesis
	 filter not exists {?func :negativeEvidence ?negev}
 ?entryx :gene ?gene.
 ?gene :begin ?sperstart; :end ?e; :chromosome ?chr; :recommendedName / rdfs:label ?gen.
 ?ngen :chromosome ?chr.
 ?ngen :begin ?n_begin; :end ?n_end .
 filter ( ?n_end > (?sperstart - 1000000) && ?n_begin <= (?sperstart + 1000000) && ?ngen != ?gene)
 bind(?sperstart - ?n_begin as ?dist)
 bind(if(?dist<0,?dist,-1000000) as ?ldist)
 bind(if(?dist>0,?dist,1000000) as ?rdist)
 } group by ?chr ?gen ?sperstart
 }
 bind(?sperstart - ?lmin as ?lchrpos)
 ?lgene :begin ?lchrpos; :chromosome ?chr.
 bind(?sperstart - ?rmin as ?rchrpos)
 ?rgene :begin ?rchrpos; :chromosome ?chr.
 { ?neighbour :gene ?lgene. } # left gene
	 union
 { ?neighbour :gene ?rgene. } # right gene
} order by ?gen`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","GROUP BY","ORDER BY","DISTINCT","STR","IF","EXISTS","NOT EXISTS","MIN","MAX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00059",slug:"neXtProt_NXQ_00059",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are glycosylated and phosphorylated on an extracellular topological domain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :topologicalDomain ?topodom.
 ?topodom :term nextprot_cv:CVTO_0002.
 ?topodom :start ?topostart ; :end ?topoend.
 ?iso :positionalAnnotation ?annot,?annot2.
 ?annot a :ModifiedResidue.
 ?annot :term ?ptmtype.
 filter (?ptmtype in (nextprot_cv:PTM-0253, nextprot_cv:PTM-0254, nextprot_cv:PTM-0255))
 ?annot2 a :GlycosylationSite.
 ?annot :start ?ptmpos.
 ?annot2 :start ?glypos.
 filter ((?ptmpos >= ?topostart) && (?ptmpos <= ?topoend))
 filter ((?glypos >= ?topostart) && (?glypos <= ?topoend))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","IF","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00060",slug:"neXtProt_NXQ_00060",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which have one or more negatively charged residue in a transmembrane domain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :transmembraneRegion ?tm.
 ?tm :start ?tmstart ; :end ?tmend.
 ?iso :sequence /:chain ?chain.
 bind (substr(?chain, ?tmstart, ?tmend - ?tmstart + 1) as ?tmseq)
 filter (regex(?tmseq, '([DE]+.*[^DE]){1,}')) # Change the 1 for the number of acidic residues required (5 seems to be the max)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","STR","REGEX","SUBSTR","MAX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00061",slug:"neXtProt_NXQ_00061",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which have at least one 3D structure that spans the complete sequence of the mature protein",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :pdbMapping ?pdbmap.
 ?pdbmap :start ?pdbstart ; :end ?pdbend.
 ?iso :matureProtein [ :start ?mstart ; :end ?mend]
 filter (?mend-?mstart > 0)
 filter ((?pdbstart <= ?mstart) && (?pdbend >= ?mend))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00062",slug:"neXtProt_NXQ_00062",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which have at least one zinc finger of any subtype",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :zincFingerRegion ?_.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00063",slug:"neXtProt_NXQ_00063",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that have at least one RRM RNA-binding domain and either no GO \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 {
 ?entry :isoform ?iso.
 # >=1 RRM RNA-binding domain
 ?iso :region/:term nextprot_cv:DO-00581.
 # No GO "RNA binding"
 filter not exists {
 ?iso :function/:term /:childOf nextprot_cv:GO_0003723
 }
 }
 union{
 ?entry :isoform ?iso.
 # >=1 RRM RNA-binding domain
 ?iso :region/:term nextprot_cv:DO-00581.
 # GO "RNA binding" with evidence IEA or ISS
 ?iso :function ?s1.
 ?s1 :term /:childOf nextprot_cv:GO_0003723.
 filter not exists { ?s1 :negativeEvidence ?negev. } # No negative function evidence
 ?s1 :evidence /:evidenceCode /:childOf ?pcode.
 values ?pcode { nextprot_cv:ECO_0000501 nextprot_cv:ECO_0000250 }
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","VALUES","DISTINCT","WITH","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00064",slug:"neXtProt_NXQ_00064",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are enzymes and that have an incomplete EC number",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform/ :enzymeClassification ?ec.
 ?ec :term / rdfs:label ?eclabel.
 filter regex(?eclabel, '-'^^xsd:string)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","STR","REGEX","IF"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00065",slug:"neXtProt_NXQ_00065",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that have more than one catalytic activity",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform / :enzymeClassification / :term ?ec .
}
group by ?entry having (count(distinct ?ec)>1)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","IF","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00066",slug:"neXtProt_NXQ_00066",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are cytoplasmic with alternate O-glycosylation or phosphorylation at the same positions",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?cytoloc {nextprot_cv:SL-0086 nextprot_cv:GO_0005737} # SL and GO values for cytoplasm

 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term /:childOf ?cytoloc .
 filter not exists {?loc :negativeEvidence ?negev} # No negative localization evidence
 ?iso :modifiedResidue /:term ?ptmtype.
 ?iso :modifiedResidue /:start ?ptmpos.
 filter (?ptmtype in (nextprot_cv:PTM-0253, nextprot_cv:PTM-0254, nextprot_cv:PTM-0255))
 ?iso :glycosylationSite /:start ?ptmpos.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00067",slug:"neXtProt_NXQ_00067",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with alternative acetylation or Ubl conjugation (SUMO or Ubiquitin) at the same positions",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :crossLink /:start ?ptmpos.
 ?iso :modifiedResidue ?ptm.
 ?ptm :start ?ptmpos.
 ?ptm rdfs:comment ?comment.
 # We use this filter to select acetylations
 filter regex(?comment, "acetyl","i")
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX","IF"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00068",slug:"neXtProt_NXQ_00068",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a protein existence \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :existence :Evidence_at_transcript_level
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00069",slug:"neXtProt_NXQ_00069",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are the substrate of protein kinase SYK",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 {
 ?iso :modifiedResidue ?ptm.
 ?ptm :term ?ptmtype.
 ?ptm rdfs:comment ?comment.
 filter (?ptmtype in (nextprot_cv:PTM-0253, nextprot_cv:PTM-0254, nextprot_cv:PTM-0255))
 filter regex(?comment, "SYK","i")
 # filter regex(?comment, "auto","i")
 }
 union
 {
 ?iso :ptmInfo / rdfs:comment ?ptmtext .
 filter regex(?ptmtext, "SYK","i")
 #filter regex(?ptmtext, "autophos","i")
 }
 }`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","REGEX","IF","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00070",slug:"neXtProt_NXQ_00070",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins secreted but without a signal sequence",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?sloc {nextprot_cv:GO_0005576 nextprot_cv:SL-0243} # GO and SL values for secreted
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term/:childOf ?sloc .
 filter not exists {?sloc :negativeEvidence ?negev} # No negative localization evidence
 filter not exists {?iso :cellularComponent /:term nextprot_cv:GO_0070062 .
					 filter not exists {?iso :cellularComponent /:term /:childOf nextprot_cv:SL-0243 .}
					} # filters out extracellular exosome only location (2758 entries)
 filter not exists {?iso :signalPeptide ?_}
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00072",slug:"neXtProt_NXQ_00072",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a cross-reference to CCDS",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

SELECT DISTINCT ?entry WHERE {
 ?entry :reference / :provenance db:CCDS.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00073",slug:"neXtProt_NXQ_00073",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with no known (annotated) domain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>

SELECT DISTINCT ?entry WHERE {
 ?entry a :Entry.
 filter not exists { ?entry :isoform / :region/rdf:type :Domain}
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00074",slug:"neXtProt_NXQ_00074",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins belonging to a family with at least two members in the human proteome",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :familyName / :term ?ac.
 ?member :familyName / :term ?ac.
} group by ?entry having (count(distinct ?member) >= 2 )`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00075",slug:"neXtProt_NXQ_00075",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which have been detected in the HUPO liver proteome set but not the HUPO plasma proteome set",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT ?entry WHERE {
values ?pepsources {
 source:PeptideAtlas_human_Liver
 source:MassIVE_human_Liver
 }
 ?entry a :Entry.
 ?entry :isoform / :peptideMapping / :evidence / :assignedBy ?pepsources .
 filter not exists { ?entry :isoform / :peptideMapping / :evidence / :assignedBy source:PeptideAtlas_human_Blood_Plasma . }
 filter not exists { ?entry :isoform / :peptideMapping / :evidence / :assignedBy source:MassIVE_human_Blood_Plasma . }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00076",slug:"neXtProt_NXQ_00076",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are located in mitochondrion and have at least one HPA antibody and exist in at least one proteome identification sets",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?mitoloc {nextprot_cv:SL-0173 nextprot_cv:GO_0005739} # SL and GO values for mitochondrion

 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term /:childOf ?mitoloc .
 filter not exists { ?loc :negativeEvidence ?negev. } # No negative localization evidence
 ?iso :antibodyMapping ?abmap.
 ?iso :peptideMapping ?pepmap.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00077",slug:"neXtProt_NXQ_00077",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are expressed in liver according to IHC data but not found in HUPO liver proteome set",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform /:expression ?s1.
 ?s1 :evidence ?evi;:term/:childOf nextprot_cv:TS-0564. #Liver
 ?evi :evidenceCode nextprot_cv:ECO_0001055; :observedExpression ?level .
 filter (?level not in (:Negative))
 filter not exists { ?entry :isoform / :peptideMapping / :evidence / :assignedBy source:PeptideAtlas_human_Liver . }
 filter not exists { ?entry :isoform / :peptideMapping / :evidence / :assignedBy source:MassIVE_human_Liver . }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00078",slug:"neXtProt_NXQ_00078",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which have been identified in at least one proteomics set and that are secreted",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?sloc {nextprot_cv:GO_0005576 nextprot_cv:SL-0243} # GO and SL values for secreted
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?loc .
 ?loc :term/:childOf ?sloc .
 filter not exists {?sloc :negativeEvidence ?negev} # No negative localization evidence
 #filter not exists {?iso :cellularComponent /:term nextprot_cv:GO_0070062 .
 #					 filter not exists {?iso :cellularComponent /:term /:childOf nextprot_cv:SL-0243 .}
 #					} # you can uncomment this to filters out extracellular exosome only location (2240 entries), most of them evidenced only by large-scale proteomic analysis
 ?iso :peptideMapping ?map.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00079",slug:"neXtProt_NXQ_00079",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one 3D structure and that are phosphorylated",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0597. # Phosphoprotein
 ?iso :pdbMapping ?map.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00080",slug:"neXtProt_NXQ_00080",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one 3D structure of resolution less than 3 Angstroms",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :pdbMapping ?map.
 ?map :resolution ?res_and_unit.
 bind (xsd:float(substr(?res_and_unit,0,strlen(?res_and_unit)-2)) as ?res)
 filter (?res <= 3.0)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","STR","STRLEN","SUBSTR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00081",slug:"neXtProt_NXQ_00081",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one 3D structure and that are located in the mitochondrion and are linked with a disease",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 values ?mitoloc {nextprot_cv:SL-0173 nextprot_cv:GO_0005739} # SL and GO values for mitochondrion
 ?entry :isoform ?iso.
 ?iso :pdbMapping ?map.
 ?iso :cellularComponent ?loc .
 ?loc :term /:childOf ?mitoloc . # mitochondrial
 filter not exists { ?loc :negativeEvidence ?negev. } # No negative localization evidence
 {
 ?iso :medical / rdf:type :Disease.
 } union {
 ?iso :uniprotKeyword / :term ?kw .
 ?kw :termType "Disease"^^xsd:string
 filter (?kw != nextprot_cv:KW-0656)
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","VALUES","DISTINCT","STR","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00082",slug:"neXtProt_NXQ_00082",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose genes are on chromosome 21 that have \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene / :chromosome "21"^^xsd:string; :isoform / :variant ?variant.
 ?variant :evidence / :quality :GOLD.
 filter not exists { ?variant :disease ?disease. }
 filter ( not exists {
	?entry :isoform / :proteoform ?pf.
	?pf :difference ?variant.
	?variant :evidence / :negative false. # variant with positive evidence
	?pf :diseaseRelatedVariant / :term nextprot_cv:ME_0000013 . # variant is causing disease
 } )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","WITH","USING","STR","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00083",slug:"neXtProt_NXQ_00083",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are expressed only in liver",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

Select distinct ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :detectedExpression /:term ?tiss, ?tiss2.
 ?tiss2 :childOf nextprot_cv:TS-0564. # Liver
} group by ?entry ?iso having(count(distinct ?tiss) = count(distinct ?tiss2))`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00084",slug:"neXtProt_NXQ_00084",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose genes are on chromosome 18 and that are experimentally (cv:ECO_0000269) known to be glycosylated or phosphorylated or acetylated",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry {
 ?entry :gene / :chromosome "18"^^xsd:string.
 ?entry :isoform ?iso.
 {
 ?iso :glycosylationSite ?ptm1.
 ?ptm1 :evidence /:evidenceCode / :childOf nextprot_cv:ECO_0000269
 }
 union {
 ?iso :modifiedResidue ?ptm2.
 ?ptm2 rdfs:comment ?com.
 ?ptm2 :evidence / :evidenceCode / :childOf nextprot_cv:ECO_0000269
 filter (regex(?com, '^phospho|acetyl','i'))
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","FILTER","UNION","DISTINCT","STR","REGEX","IF","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00085",slug:"neXtProt_NXQ_00085",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with Poly-Proline stretches",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 ?iso :compositionallyBiasedRegion / rdfs:comment ?desc.
 filter (contains(?desc,'Pro residues')).
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","ALL","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00086",slug:"neXtProt_NXQ_00086",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose genes are located in chromosome 2 region from 2p12 to 2p24.2",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene ?gene .
 ?gene :chromosome "2"^^xsd:string .
 ?gene :band ?band.
 filter (?band >= "p12"@en && ?band <= "p24.2"@en )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00087",slug:"neXtProt_NXQ_00087",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose genes are on chromosome X and which do not have a ortholog in mouse",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene / :chromosome "X"^^xsd:string .
 filter not exists { ?entry :reference / :provenance db:MGI . }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","STR","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00089",slug:"neXtProt_NXQ_00089",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Protein that have a signal sequence which is not cleaved",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform / :signalPeptide / rdfs:comment ?comment .
 filter(contains(?comment,"Not cleaved"))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00090",slug:"neXtProt_NXQ_00090",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are linked to a disease and that do not have orthologs/paralogs in mouse",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 {
 ?iso :medical / rdf:type :Disease.
 } union {
 ?iso :uniprotKeyword / :term ?kw .
 ?kw :termType "Disease"^^xsd:string
 filter (?kw != nextprot_cv:KW-0656)
 }
 FILTER NOT EXISTS { ?entry :reference / :provenance db:MGI . }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","STR","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00092",slug:"neXtProt_NXQ_00092",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a sequence containing \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>

SELECT DISTINCT ?entry WHERE {
 ?entry rdf:type :Entry.
 ?entry :isoform / :sequence / :chain ?chain.
 filter ( regex(?chain, "FF.QYE") )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00093",slug:"neXtProt_NXQ_00093",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a sequence containing \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform / :sequence / :chain ?chain.
 filter ( regex(?chain, "FF.+QYE") )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00097",slug:"neXtProt_NXQ_00097",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins located on chromosome 2 and having at least one variant in a phosphorylated tyrosine",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene / :chromosome "2"^^xsd:string .
 ?entry :isoform ?iso.
 ?iso :variant /:start ?varpos.
 ?iso :modifiedResidue ?modres.
 ?modres :term nextprot_cv:PTM-0255.
 ?modres :start ?varpos.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","STR","IF","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00099",slug:"neXtProt_NXQ_00099",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Secreted proteins that have at least one PTM in a position of a variant",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 values ?sloc {nextprot_cv:GO_0005576 nextprot_cv:SL-0243} # GO and SL values for secreted
 ?entry :isoform ?iso .
 ?iso :swissprotDisplayed true . # restricted to main isoform due to computing time (> 98% of sites)
 ?iso :cellularComponent ?anno .
 ?anno :quality :GOLD; :term /:childOf ?sloc. # secreted
 filter not exists { ?anno :negativeEvidence ?_ }
 # excludes "extracellular exosome"-only location, most of them evidenced only by large-scale proteomic analysis
 filter not exists {
	 ?iso :cellularComponent /:term nextprot_cv:GO_0070062 . # extracellular exosome
 	 filter not exists { ?iso :cellularComponent /:term /:childOf nextprot_cv:SL-0243 .}
 	}
 ?iso :variant /:start ?varpos.
 ?iso :ptm /:start ?varpos.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","STR","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00103",slug:"neXtProt_NXQ_00103",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that do not have a cross-reference to an Ensembl ENSG",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

SELECT DISTINCT ?entry WHERE {
 ?entry a :Entry.
 filter not exists { ?entry :reference / :provenance db:Ensembl . }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00104",slug:"neXtProt_NXQ_00104",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose gene is on chromosome 21 with at least one disease annotation from Orphanet",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX source: <http://nextprot.org/rdf/source/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry (str(?disname) as ?disease) WHERE {
 ?entry :isoform ?iso; :gene / :chromosome "21"^^xsd:string .
 ?iso :disease ?medannot .
 ?medannot rdfs:comment ?disname; :evidence / :assignedBy source:Orphanet.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00105",slug:"neXtProt_NXQ_00105",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one cross-reference to SMR (Swiss Model Repository) but no cross-references to PDB",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

SELECT DISTINCT ?entry WHERE {
 ?entry a :Entry.
 ?entry :reference / :provenance db:SMR.
 filter not exists { ?entry :reference / :provenance db:PDB }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00106",slug:"neXtProt_NXQ_00106",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins annotated as glycosylated but with no recorded glycosylation site",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0325.
 filter not exists { ?entry :isoform /:glycosylationSite ?_. }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00107",slug:"neXtProt_NXQ_00107",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a protein existence not \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

SELECT DISTINCT ?entry WHERE {
 ?entry :existence ?level.
 filter (?level != :Evidence_at_protein_level).
 ?entry :reference ?r.
 ?r :provenance db:HGNC ; :accession ?ac.
 filter (regex(?ac,'orf')) .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00108",slug:"neXtProt_NXQ_00108",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that have a 3D structure in PDB that overlap by at least 50 amino acids with a SH3 domain",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?isoform.
 ?isoform :pdbMapping ?struc.
 ?struc :start ?s1 ; :end ?s2.
 ?isoform :region ?dom.
 ?dom :term nextprot_cv:DO-00615 ; :start ?d1 ; :end ?d2.
 bind ( if(?d2<?s2, ?d2, ?s2) - if(?d1>?s1, ?d1, ?s1) as ?overlap) .
 filter (?overlap>50)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","STR","IF","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00109",slug:"neXtProt_NXQ_00109",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that have a MS-identified proteotypic peptide that maps partly or fully into a signal sequence",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry ?overlap WHERE {
 ?entry :isoform ?iso.
 #get Signal position (start is always 1)
 ?iso :signalPeptide [:end ?s2] .
 ?iso :peptideMapping ?pm .
 ?pm :proteotypic true .
 #get peptide position
 ?pm :start ?p1 ; :end ?p2 .
 #match positions
 filter ( ?p1 < ?s2 )
 bind ( if(?p2<?s2, ?p2, ?s2) - ?p1 as ?overlap) .
 filter(?overlap > 2)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","IF"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00111",slug:"neXtProt_NXQ_00111",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with one transmembrane domain and no annotated topology",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :swissprotDisplayed true .
 ?iso :topology ?tm.
 ?tm a :TransmembraneRegion.
 filter (not exists { ?iso :topology ?topodom.
 ?topodom a :TopologicalDomain.
 })
} group by ?entry having (count( ?tm)=1)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","HAVING","DISTINCT","EXISTS","NOT EXISTS","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00112",slug:"neXtProt_NXQ_00112",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins located in nucleus and nowhere else",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

select ?entry WHERE {
 {
 select ?entry (count(?loc) as ?anyLoc) (sum(?inNucleus) as ?nucleusLoc)
 WHERE {
 ?entry :isoform / :cellularComponent ?loc .
 ?loc :evidence / :negative false .
 ?loc :term ?locterm.
 bind( exists {
 ?locterm :childOf / rdfs:label ?label.
 filter (regex(?label, '^[Nn]ucleus$'))
 } as ?inNucleus)
 }
 group by ?entry
 }
 filter ( ?anyLoc = ?nucleusLoc)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","GROUP BY","REGEX","EXISTS","COUNT","SUM"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00113",slug:"neXtProt_NXQ_00113",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with protein existence \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform / :peptideMapping ?pm .
 ?entry :existence :Evidence_at_protein_level .
 ?pm :proteotypic true .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00117",slug:"neXtProt_NXQ_00117",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one sequence variant",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :variant ?var.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00118",slug:"neXtProt_NXQ_00118",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one somatic variant",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :variant ?var.
 {
 ?var rdfs:comment ?comment.
 filter (contains (?comment,'somatic'))
 }
 union
 {
 ?var :evidence ?ev.
 ?ev :assignedBy source:Cosmic.
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00119",slug:"neXtProt_NXQ_00119",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one annotated mutagenesis site",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :mutagenesis ?mut.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00124",slug:"neXtProt_NXQ_00124",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the 25 most frequent families with member count",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

select (str(?famlab) as ?familylabel) (count(distinct(?member)) as ?membercnt) where{
 ?member :familyName /:term /:childOf / rdfs:label ?famlab .
} group by ?famlab
order by desc(?membercnt)
limit 25`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","LIMIT","DISTINCT","STR","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00125",slug:"neXtProt_NXQ_00125",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Domains that are entirely covered by 3D structures",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?domlabel WHERE {
 ?entry :isoform ?isoform.
 ?isoform :pdbMapping ?struc.
 ?struc :start ?s1 ; :end ?s2.
 ?isoform :domain ?dom.
 ?dom :start ?d1 ; :end ?d2.
 ?dom :term/rdfs:label ?domlabel
 filter ((?s1 <= ?d1) && (?s2 >= ?d2))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00126",slug:"neXtProt_NXQ_00126",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Peptides that are potential neo N-termini from undescribed isoforms",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry ?pep WHERE {
 ?entry :isoform ?iso .
 ?iso :peptideMapping ?pm.
 ?pm :start ?p1 ; :end ?p2 .
 ?pm :proteotypic true .
 filter(?p1 > 2). # must not be already N-terminal
 ?iso :sequence / :chain ?chain.
 bind (substr(?chain, ?p1, ?p2 - ?p1 + 1) as ?pep) .
 bind (substr(?chain, ?p1-1, 1) as ?prevAA) .
 bind (substr(?chain, ?p1, 1) as ?firstAA) .
 bind (substr(?chain, ?p2, 1) as ?lastAA) .
 filter(!regex (?prevAA,'[KR]')) # must be semi-tryptic in N-ter
 filter(regex (?prevAA,'M') || regex (?firstAA,'M')) # must be N-terminal
 filter(!regex (?firstAA,'[DEFIKLRY]')) #plausible 2nd AA
 filter(regex (?lastAA,'[KR]')) # must be tryptic in C-ter
 filter not exists { # The candidate peptide must not already exist as N-ter in a described isoform
 ?entry :isoform ?iso2.
 ?iso2 :sequence / :chain ?chain2.
 ?iso2 :matureProtein [ :start ?mstart ; :end ?mend]
 bind (substr(?chain2, ?mstart, ?mend - ?mstart + 1) as ?mat2) .
 filter(strlen(?mat2) > 30).
 bind (substr(?mat2, 2, strlen(?mat2) - 1) as ?mat22) .
 filter(regex(?mat2,concat("^", ?pep)) || regex(?mat22,concat("^", ?pep))).
 }
}
order by ?pep

# overestimated, need additional filter(s)
# but for instance AELEEVTLDGKPLQALR, AELEEVTLDGKPLQALRVTDLKAALEQR and AELEEVTLDGKPLQALRVTDLKAALEQR in Q9UKV3
# are N-acetylated and good markers for an additional iso starting at M-59`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","DESCRIBE","WHERE","FILTER","BIND","ORDER BY","DISTINCT","ADD","STR","REGEX","EXISTS","NOT EXISTS","CONCAT","STRLEN","SUBSTR","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00127",slug:"neXtProt_NXQ_00127",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one known SUMOylation site",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :crossLink /rdfs:comment ?comment.
 filter(contains (?comment,'SUMO'))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","CONTAINS","SUM"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00128",slug:"neXtProt_NXQ_00128",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose gene is annotated to be induced by interferons",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :induction /rdfs:comment ?comment.
 filter(regex (?comment,'IFN|interferon',"i"))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX","IF"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00130",slug:"neXtProt_NXQ_00130",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a DNA-binding region but not located in the nucleus",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?entry :gene / :name / rdfs:label ?gen .
 ?iso :dnaBindingRegion ?dnab.
 filter not exists { ?entry :isoform / :cellularComponent /:term /:childOf nextprot_cv:SL-0191} # SL for nucleus
 filter not exists { ?entry :isoform / :cellularComponent /:term /:childOf nextprot_cv:GO_0005634} # GO for nucleus
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00131",slug:"neXtProt_NXQ_00131",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Terms of controlled vocabularies containing some word(s)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT * WHERE {
 ?term rdfs:label ?label ; a ?type .
 # ---------------------------
 # exact match
 # ---------------------------
 filter(?label = "peroxisome"^^xsd:string)

 # ---------------------------
 # starting with peroxisome
 # ---------------------------
 # filter(regex(?label,"^peroxisome"))

 # ---------------------------
 # containing peroxisome
 # ---------------------------
 # filter(regex(?label,"peroxisome"))

 # ------------------------------------
 # containing peroxisome and receptor
 # ------------------------------------
 # filter(regex(?label,"peroxisome.*receptor"))
} limit 20`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","LIMIT","DISTINCT","WITH","STR","REGEX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00132",slug:"neXtProt_NXQ_00132",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a 3D structure in complex with another human protein which is not reported as binary interactant",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

SELECT DISTINCT ?entry WHERE {
 ?entry a :Entry.
 ?entry :reference ?ref.
 ?ref :provenance db:PDB; :accession ?ac.
 ?entry2 a :Entry.
 ?ac ^:accession/^:reference ?entry2.
 filter(?entry != ?entry2).
 filter not exists{?entry :isoform /:binaryInteraction / :interactant ?entry2.}
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00135",slug:"neXtProt_NXQ_00135",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins involved in both Wnt and Hippo signaling pathways",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 # nextprot_cv:GO:0016055 wnt pathway
 ?iso :goBiologicalProcess ?func1 .
 ?func1 :term / :childOf nextprot_cv:GO_0016055.
 filter not exists { ?func1 :negativeEvidence ?negev. } # No negative function evidence
 # nextprot_cv:GO:0035329 Hippo pathway
 ?iso :goBiologicalProcess ?func2 .
 ?func2 :term / :childOf nextprot_cv:GO_0035329.
 filter not exists { ?func2 :negativeEvidence ?negev. } # No negative function evidence
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00136",slug:"neXtProt_NXQ_00136",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins whose sequence was updated in 2014",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry a :Entry.
 ?entry :history / :lastSequenceUpdate ?sequpd.
 filter(contains(str(?sequpd),"2014"))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","STR","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00137",slug:"neXtProt_NXQ_00137",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that potentially interact with Class I PDZ domains (whose C-terminal sequence has a PDZ-binding consensus)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :matureProtein / :end ?mend.
 ?iso :sequence / :chain ?seq.
 filter (strlen(?seq) > 2).
 bind (substr(?seq, ?mend-2, 3) as ?cterseq).
 filter(regex(?cterseq,'[ST].[ILV]')). # short motif consensus for C-terminal PDZ-binding
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","STR","REGEX","IF","STRLEN","SUBSTR","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00138",slug:"neXtProt_NXQ_00138",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with 10 or more gold interactions with SH3 domain-containing proteins",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :interaction ?it.
 ?it :quality :GOLD.
 ?it :interactant ?interactant.
 ?interactant :isoform? / :domain / :term nextprot_cv:DO-00615. # SH3
}
group by ?entry having (count (distinct ?interactant) >= 10)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00139",slug:"neXtProt_NXQ_00139",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Protein kinases which are high-confidence drug targets according to CHEMBL",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX cco: <http://rdf.ebi.ac.uk/terms/chembl#>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>

SELECT DISTINCT ?entry (str(?gen) as ?gene) WHERE {
    SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/idsm> {
        SELECT DISTINCT ?protein WHERE {
            ?target cco:hasTargetComponent ?tarComp .
            ?target cco:taxonomy <http://identifiers.org/taxonomy/9606> . # human protein target
            ?tarComp cco:targetCmptXref ?protein .
            #?protein a cco:UniprotRef .
            FILTER(contains(str(?protein),"uniprot"))
            ?activity a cco:Activity ; cco:hasMolecule ?drug ; cco:hasAssay ?assay .
            ?drug cco:hasMechanism / cco:hasTarget ?target . # known drug action mechanism
            ?assay cco:hasTarget ?target; cco:targetConfScore ?score .
            FILTER(?score > 8) # high-confidence assay
        }
    }
    ?entry skos:exactMatch ?protein .
    ?entry :isoform / :uniprotKeyword / :term nextprot_cv:KW-0418. #kinase
    ?entry :gene / :recommendedName / rdfs:label ?gen.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","DISTINCT","STR","IF","NOW","CONTAINS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00140",slug:"neXtProt_NXQ_00140",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that interact with viral proteins",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 optional {?iso :interactionInfo ?itinfo .}
 {
 ?itinfo rdfs:comment ?txt.
 filter(contains(?txt,"viral") || contains(?txt,"virus")).
 } # refers to a virus in the free-text interactionInfo topic
UNION
 { ?iso :uniprotKeyword / :term nextprot_cv:KW-0945.} # host-virus interaction keyword
UNION
 {
 SERVICE <https://sparql.uniprot.org/sparql> {
 SELECT distinct ?viralinteractor WHERE # get viral proteins with an IntAct xref
 {
	?viralinteractor a up:Protein .
	?viralinteractor rdfs:seeAlso ?db .
	?db up:database <http://purl.uniprot.org/database/IntAct> .
 	?viralinteractor up:organism ?tax .
	?tax up:scientificName ?orgname.
	filter(contains(?orgname,"virus"))
 }
 }
 ?entry :isoform / :binaryInteraction ?it .
 ?it :interactant ?interactant; :quality :GOLD. # NeXtprot entries with a GOLD IntAct binary interaction
 ?interactant skos:exactMatch ?viralinteractor . # interactant must be in the uniprot service result set to select the entry
 }
UNION
 {
 SERVICE <https://sparql.uniprot.org/sparql> {
 SELECT distinct ?humprotein WHERE # get human proteins that share a PDB xref with a viral protein (same PDB id)
 {
	?humprotein a up:Protein .
 ?humprotein up:organism taxon:9606 .
	?humprotein rdfs:seeAlso ?db .
	?db up:database <http://purl.uniprot.org/database/PDB> .
	?viralprotein a up:Protein .
	?viralprotein rdfs:seeAlso ?db .
 ?viralprotein up:organism ?tax .
	?tax up:scientificName ?orgname.
	filter(contains(?orgname,"virus"))
 }
 }
 ?entry skos:exactMatch ?humprotein .
 }
}
order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","UNION","SERVICE","ORDER BY","DISTINCT","WITH","IF","CONTAINS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00143",slug:"neXtProt_NXQ_00143",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins for which none of the reported proteotypic peptides is from PeptideAtlas nor MassIVE",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX source: <http://nextprot.org/rdf/source/>

select ?entry WHERE {
 ?entry :isoform / :peptideMapping ?pm.
 ?pm :proteotypic true .
 ?pm :peptideSource ?src .
 bind (?src = source:MassIVE as ?massive)
 bind (?src = source:PeptideAtlas as ?pa)
 bind (?src != source:MassIVE && ?src != source:PeptideAtlas as ?other)
}
group by ?entry
having (sum(?massive)=0 && sum(?pa)=0 && sum(?other)>0)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","GROUP BY","HAVING","SUM"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00144",slug:"neXtProt_NXQ_00144",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that contains a sequence \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
?entry :isoform ?isoform.
?isoform :ptm ?ptm.
?ptm :term nextprot_cv:PTM-0257 . # Proline amide
?ptm :start ?ptmpos .
?isoform :sequence / :chain ?seq.
bind(substr(?seq, ?ptmpos-2, 3) as ?modseq)
filter ( regex(?modseq, "QHP") )
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT","STR","REGEX","SUBSTR","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00145",slug:"neXtProt_NXQ_00145",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which include a mature chain of less or equal to 50 amino acid residues whose C-terminus is amidated",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry WHERE {
?entry :isoform ?isoform.
?isoform :ptm ?ptm.
?ptm rdfs:comment ?comment.
filter regex(?comment, "amid","i")
?ptm :start ?ptmpos .
?isoform :matureProtein [ :start ?mstart ; :end ?mend]
filter ((?mend = ?ptmpos) && (?mend-?mstart < 50))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00146",slug:"neXtProt_NXQ_00146",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins having at least 2 proteotypic peptides of 7 or 8aa but no proteotypic peptide >= 9 aa",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
?entry :isoform / :peptideMapping ?pm.
?pm :proteotypic true .
?pm :peptideName ?pepid .
#?pm :start ?pos1 ; :end ?pos2 .
#filter((?pos2 - ?pos1 + 1) >= 7) # This is not absolutely required for current query since 7 is the minimum length for accepted proteotypic peptides
filter not exists {
 ?entry :isoform / :peptideMapping ?pm2.
 ?pm2 :start ?p1 ; :end ?p2 .
 ?pm2 :proteotypic true .
 filter((?p2 - ?p1 + 1) >= 9)
 }
}
group by ?entry having(count (distinct ?pepid) >= 2)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","HAVING","DISTINCT","EXISTS","NOT EXISTS","ABS","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00147",slug:"neXtProt_NXQ_00147",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a variant having an impact on the nucleus localization at level GOLD",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry
where {
 values ?ioTerm {nextprot_cv:GO_0005634} . # nucleus
 ?entry :isoform / :proteoform / :phenotypicVariation ?phvar .
 ?phvar :term / :childOf nextprot_cv:ME_0000002 . # children of impact
 ?phvar :evidence / :quality :GOLD .
 ?phvar :impactedObject / :term ?ioTerm .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00148",slug:"neXtProt_NXQ_00148",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Variants with normal \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT distinct ?pfname ?impact_on_ubi_trsfrt_activity ?impact_on_binding_UBE2D1 WHERE {
 values (?ubi_trsfrt_activity ?no_impact ?UBE2D1 ?impact)
 { (nextprot_cv:GO_0004842 nextprot_cv:ME_0000003 nextprot:NX_P51668 nextprot_cv:ME_0000002) }
 ?pf a :Proteoform .
 ?pf rdfs:label ?pfname .
 ?pf :phenotypicVariation ?pv1 .
 ?pv1 :impactedObject / :term ?ubi_trsfrt_activity .
 ?pv1 :term ?no_impact .
 ?no_impact rdfs:label ?impact_on_ubi_trsfrt_activity .
 ?pf :phenotypicVariation ?pv2 .
 ?pv2 :impactedObject / :interactant ?UBE2D1 .
 ?pv2 :term ?sub_impact .
 ?sub_impact :childOf ?impact .
 ?sub_impact rdfs:label ?impact_on_binding_UBE2D1 .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","VALUES","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00149",slug:"neXtProt_NXQ_00149",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"BRCA1 variants with at least 5 different Severe phenotypes",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT
?pfname
(count(?pvlabel) as ?severe_count)
(group_concat(?pvlabel;separator=", and ") as ?severe_list)
WHERE {
 nextprot:NX_P38398 :isoform / :proteoform ?pf .
 ?pf rdfs:label ?pfname .
 ?pf :phenotypicVariation ?pv1 .
 ?pv1 rdfs:comment ?pvlabel .
 ?pv1 :evidence / :severity :Severe .
 }
group by ?pfname having (count(?pvlabel)>=5)
order by desc(count(?pvlabel)) ?pfname`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","HAVING","CONCAT","COUNT","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00202",slug:"neXtProt_NXQ_00202",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with from 2 to 4 transmembrane regions",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :topology ?statement.
 ?statement a :TransmembraneRegion.
} group by ?entry ?iso having(count( ?statement)>=2 && count( ?statement)<=4)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00203",slug:"neXtProt_NXQ_00203",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a mature chain of from 300 to 400 residues",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :matureProtein [ :start ?mstart ; :end ?mend].
 filter (?mend-?mstart > 0). # chain fully defined
 bind(?mend - ?mstart as ?mlen)
 filter(?mlen >= 300 && ?mlen <= 400)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00204",slug:"neXtProt_NXQ_00204",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with protein existence \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 values ?level {
#	:Evidence_at_protein_level # PE=1
 :Evidence_at_transcript_level # PE=2
 :Inferred_from_homology # PE=3
 :Predicted # PE=4
#	:Uncertain # PE=5
 }
 ?entry a :Entry .
 ?entry :existence ?level .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","DISTINCT","FROM"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00208",slug:"neXtProt_NXQ_00208",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are ion channels and are associated with a disease",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :uniprotKeyword /:term nextprot_cv:KW-0407. #ion channel
 {
 ?iso :medical / rdf:type :Disease.
 } union {
 ?iso :uniprotKeyword / :term ?kw .
 ?kw :termType "Disease"^^xsd:string
 filter (?kw != nextprot_cv:KW-0656)
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","STR","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00209",slug:"neXtProt_NXQ_00209",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Glycosylation sites and cross links positions on SwissProt canonical isoforms",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry ?iso ?ptmtype ?pos ?modres WHERE {
?entry :isoform ?iso.
?iso :swissprotDisplayed true .
?iso :sequence / :chain ?seq .
?iso :ptm ?ptm.
?ptm :term ?modterm.
?ptm :start ?pos.
bind (substr(?seq,?pos,1) as ?modres)	.
{
?ptm a :GlycosylationSite .
bind("glyco" as ?ptmtype)
}
union
{
?ptm a :CrossLink .
bind("cross-link" as ?ptmtype)
}
}
order by ?iso
#limit 100`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","BIND","ORDER BY","LIMIT","DISTINCT","STR","SUBSTR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00216",slug:"neXtProt_NXQ_00216",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Phosphorylation sites from PeptideAtlas Phosphoproteome",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT (sample(?iso) as ?isospl) (sample(?pos) as ?posspl) WHERE {
 ?entry :isoform ?iso .
 ?iso :ptm ?ptm.
 ?ptm :entryAnnotationId ?ptmid; :start ?pos .
 ?ptm :evidence / :assignedBy source:PeptideAtlas_human_phosphoproteome .
}
group by ?ptmid
order by ?isospl ?posspl`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","SAMPLE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00217",slug:"neXtProt_NXQ_00217",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins associated with an incomplete EC number, no function annotated and high expression in liver",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?isox.
 ?isox :enzymeClassification / :term / rdfs:label ?eclabel.
 filter regex(?eclabel, '-'^^xsd:string) # incomplete EC number
 ?isox :highExpression /:term /:childOf nextprot_cv:TS-0564. # highly expressed in liver
 {
 ?isox :functionInfo / rdfs:comment ?functext .
 filter(strstarts(?functext,"Probable") || strstarts(?functext,"Putative"))
 }
 UNION
 {
 filter not exists {?isox :functionInfo ?_ } # entries with no function at all
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","WITH","ALL","STR","REGEX","IF","EXISTS","NOT EXISTS","STRSTARTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00218",slug:"neXtProt_NXQ_00218",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins for which different splice isoforms have a different subcellular location or function",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

#A0A1B0GVQ0,A0AVT1 ,A0PJK1...



SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso1, ?iso2.
 ?iso1 :swissprotDisplayed true .
 ?iso2 :swissprotDisplayed false .
 {
 ?iso2 :function / :term ?functerm .
 filter (?functerm != nextprot_cv:GO_0005515) #protein-binding
 filter not exists {?iso1 :function / :term ?functerm .}
 }
 UNION
 {
 ?iso2 :functionInfo / rdfs:comment ?functext .
 filter not exists {?iso1 :functionInfo / rdfs:comment ?functext .}
 }
 UNION
 {
 ?iso2 :cellularComponent / :term ?locterm .
 filter not exists {?iso1 :cellularComponent / :term ?locterm}
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00219",slug:"neXtProt_NXQ_00219",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins entries from a list of gene names",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 values ?gene
 {"UBA6" "PXDNL" "CNOT1" "TCAF2" "ARHGAP32" "HACD1" "AKR1B15" "SGK1" "MYO1C" "LGALS9"
 "PIK3CD" "PDE2A" "DNM1L"} # space or nl-separated

 bind (STRDT (?gene,xsd:string) as ?genename ) . # converts raw string to ^^xsd:string
 ?entry :gene / :recommendedName / rdfs:label ?genename .
 }`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","VALUES","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00220",slug:"neXtProt_NXQ_00220",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins located on chromosome MT (mitochondrial) coded by a gene located on the plus strand",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry WHERE {
 ?entry :gene ?gene .
 ?gene :chromosome "MT"^^xsd:string; :strand "1"^^xsd:string .
 }`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","STR","RAND"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00221",slug:"neXtProt_NXQ_00221",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with RNA-seq observed expression \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 ?iso :expression ?exp, ?exp2.
 ?exp :term ?tiss .
 ?exp :negativeEvidence / :evidenceCode nextprot_cv:ECO_0000295. # RNA-seq
 ?exp2 :quality :GOLD .
 ?exp2 :term ?tiss2 .
 ?exp2 :evidence ?evi2.
 ?evi2 :evidenceCode nextprot_cv:ECO_0001055. # IHC
 ?evi2 :observedExpression :High.
 ?tiss2 :childOf ?tiss . # same tissue or children
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00222",slug:"neXtProt_NXQ_00222",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with both RNA-seq expression and observed IHC expression \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso .
 ?iso :expression ?exprna, ?expihc.
 ?exprna :term ?rnatiss .
 ?exprna :evidence ?evirna.
 ?evirna :observedExpression :Positive.
 ?evirna :evidenceCode nextprot_cv:ECO_0000295. # RNA-seq
 ?rnatiss :childOf nextprot_cv:TS-0095. #brain
 ?expihc :quality :GOLD .
 ?expihc :term ?ihctiss .
 ?expihc :evidence ?eviihc.
 ?eviihc :evidenceCode nextprot_cv:ECO_0001055. #IHC
 ?eviihc :observedExpression :High.
 ?ihctiss :childOf ?rnatiss .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00223",slug:"neXtProt_NXQ_00223",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"100 most-cited publications in neXtProt and the corresponding number of associated entries",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

select (str(?publiid) as ?PMID) (str(?tt) as ?title) (count(distinct ?entry) as ?ecnt)
where {
 ?entry a :Entry .
 ?entry :reference ?ref .
 ?ref a :Publication.
 ?ref :title ?tt .
 ?ref :from ?xref .
 ?xref :accession ?publiid .
 ?xref :provenance db:PubMed .
}
group by ?publiid ?tt
order by desc(?ecnt)
limit 100`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","LIMIT","DISTINCT","FROM","STR","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00224",slug:"neXtProt_NXQ_00224",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Transmembrane proteins with at least 100 consecutive aa located in the EXTRACELLULAR OR LUMENAL compartment.",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :topologicalDomain ?topdom .
 ?topdom :term ?topterm .
 filter (?topterm in (nextprot_cv:CVTO_0002, nextprot_cv:CVTO_0003, nextprot_cv:CVTO_0007)) # extracellular, exoplasmic loop, lumenal
 ?topdom :start ?domstart; :end ?domend .
 filter ((?domend - ?domstart) >= 100)
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00225",slug:"neXtProt_NXQ_00225",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with high proline content",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry ?AAcnt ?seqlen ?AAdensity WHERE {
 ?entry :isoform ?iso.
 ?iso :swissprotDisplayed true .
 ?iso :sequence / :chain ?seq .
 ?iso :sequence / :length ?seqlen .
 bind( strlen( replace( str(?seq), "[^P]", "")) as ?AAcnt) # replace all non-Proline by empty string
 bind(xsd:float(?AAcnt)/xsd:float(?seqlen) as ?AAdensity)
 filter(?AAcnt > 10)
}
order by desc(?AAdensity)
limit 100 # will bring-up the 100 most relevant cases`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","ORDER BY","LIMIT","DISTINCT","ALL","STR","STRLEN","REPLACE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00226",slug:"neXtProt_NXQ_00226",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least 2 validating peptides >=9aa found in blood plasma, urine or cerebrospinal fluid (criteria for biomarker)].",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT ?entry WHERE {
values ?pepsources {
 source:PeptideAtlas_human_Cerebrospinal_Fluid
 source:PeptideAtlas_human_Blood_Plasma
 source:PeptideAtlas_human_Urine
 source:MassIVE_human_Cerebrospinal_Fluid
 source:MassIVE_human_Blood_Plasma
 source:MassIVE_human_Urine
 }
 ?entry :isoform ?iso.
 ?iso :peptideMapping ?pm .
 ?pm :peptideName ?pepid .
 ?pm :evidence / :assignedBy ?pepsources .
 ?pm :proteotypic true .
 ?pm :start ?p1 ; :end ?p2 .
 filter(?p2-?p1 >= 8) # peptide length >= 9
}
group by ?entry having(count (distinct ?pepid) > 1) # at least two such peptides`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","GROUP BY","HAVING","DISTINCT","URI","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00230",slug:"neXtProt_NXQ_00230",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with experimentally determined lengthy alpha-helices (> 75 aa)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (str(?gen) as ?gene) ?hlen WHERE {
 ?entry :isoform ?iso.
 ?entry :gene / :recommendedName / rdfs:label ?gen.
 ?iso :helix ?hel .
 ?hel :start ?s; :end ?e .
 bind((?e - ?s + 1) as ?hlen )
 filter(?hlen > 75)
}
order by desc(?hlen)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","ORDER BY","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00231",slug:"neXtProt_NXQ_00231",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with sequences 100% identical to other proteins",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform / :sequence / :chain ?chain1.
 ?entry2 :isoform / :sequence / :chain ?chain2.
 filter ( (?chain1 = ?chain2) && (?entry != ?entry2))
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00232",slug:"neXtProt_NXQ_00232",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with sequences 100% identical to other proteins, grouped",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry (group_concat(distinct substr(str(?entry2),34); separator = ",") as ?sameseq) WHERE {
 ?entry :isoform / :sequence / :chain ?chain1.
 ?entry2 :isoform / :sequence / :chain ?chain2.
 filter ( (?chain1 = ?chain2) && (?entry != ?entry2))
}
group by ?entry
order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","ORDER BY","DISTINCT","STR","CONCAT","SUBSTR","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00233",slug:"neXtProt_NXQ_00233",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are PE>1 with at least one proteotypic peptide of at least 9 amino-acids identified in a human sample ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 values ?level {
 	:Evidence_at_transcript_level # PE=2
 	:Inferred_from_homology 	# PE=3
 	:Predicted 	# PE=4
	:Uncertain 	# PE=5
	}
 ?entry a :Entry .
 ?entry :existence ?level .
 ?entry :isoform / :peptideMapping ?pm.
 ?pm :proteotypic true .
 ?pm :start ?pos1 ; :end ?pos2 .
 filter((?pos2 - ?pos1 + 1) >= 9)
 }`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","FROM"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00234",slug:"neXtProt_NXQ_00234",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least two proteotypic synthetic peptides from SRMAtlas of at least 9 amino acids in length",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :srmPeptideMapping ?srm .
 ?srm :start ?ps; :end ?pe .
 filter(?pe - ?ps + 1 >= 9)
 ?srm :proteotypic true .
 ?srm :peptideName ?pepid .
}
group by ?entry having(count(distinct ?pepid) > 1)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","HAVING","DISTINCT","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00235",slug:"neXtProt_NXQ_00235",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least two antibodies available from Human Protein Atlas that have associated tissue expression annotations from immunohistochemistry studies ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :antibodyMapping ?abmap.
 ?abmap :evidence / :reference / :accession ?aid .
 ?iso :expressionProfile / :evidence ?ev .
 ?ev :evidenceCode nextprot_cv:ECO_0001055.
# ?ev :quality :GOLD .
}
group by ?entry having(count(distinct ?aid) > 1)
order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","HAVING","DISTINCT","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00236",slug:"neXtProt_NXQ_00236",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with subcellular location gold and list of all these locations",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (group_concat(distinct str(?loclab); SEPARATOR = ",") as ?locs) WHERE {
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?locannot .
 ?locannot :term ?locterm .
 ?locterm rdfs:label ?loclab .
 ?locannot :evidence ?locev .
 ?locev :quality :GOLD .
 filter not exists {?locannot :negativeEvidence ?locev .}
}
GROUP BY ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","DISTINCT","STR","EXISTS","NOT EXISTS","CONCAT","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00237",slug:"neXtProt_NXQ_00237",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Transmembrane proteins with the number of transmembrane domains in the canonical isoform by decreasing order",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

select ?entry (count(distinct(?statement)) as ?tmcnt) WHERE {
 ?entry :isoform ?iso.
 ?iso :swissprotDisplayed true .
 ?iso :topology ?statement.
 ?statement a :TransmembraneRegion.
}
group by ?entry
order by desc(?tmcnt)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00238",slug:"neXtProt_NXQ_00238",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are lipoproteins",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :uniprotKeyword / :term nextprot_cv:KW-0449.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00239",slug:"neXtProt_NXQ_00239",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with cross-references to InterPro",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

SELECT DISTINCT ?entry ?iprac WHERE {
 ?entry :reference ?ref.
 ?ref :provenance db:InterPro.
 ?ref :accession ?iprac}
order by ?entry ?iprac`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00240",slug:"neXtProt_NXQ_00240",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Gold binary interactions with other human proteins ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry ?interactant WHERE {
 ?entry :isoform ?iso.
 ?entry :isoform/:binaryInteraction ?interaction.
 ?interaction :interactant ?interactant; :quality :GOLD.
 filter not exists { ?interactant a :Xref . }
 filter(?interactant != ?entry) # remove self-interacting proteins
}
order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","MOVE","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00241",slug:"neXtProt_NXQ_00241",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Interaction annotations from SwissProt ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (str(?txt) as ?itinfo) WHERE {
 SELECT DISTINCT ?entry ?txt WHERE {
 ?entry :isoform ?iso.
 ?iso :interactionInfo / rdfs:comment ?txt.
 } order by ?entry
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00242",slug:"neXtProt_NXQ_00242",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a mitochondrial transit peptide",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :mitochondrialTransitPeptide ?mit.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00243",slug:"neXtProt_NXQ_00243",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a signal sequence ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :signalPeptide ?_.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00244",slug:"neXtProt_NXQ_00244",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a variant having an impact on a binary interaction ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry ?interactant WHERE {
 ?entry :isoform / :proteoform / :phenotypicVariation ?phvar .
 ?phvar :term / :childOf nextprot_cv:ME_0000002 . # children of impact
 ?phvar :evidence / :quality :GOLD .
 ?phvar :impactedObject / :interactant ?interactant.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00245",slug:"neXtProt_NXQ_00245",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a propeptide ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform / :propeptide ?propep .
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00246",slug:"neXtProt_NXQ_00246",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins which are enzymes catalyzing a reaction involving lipids",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>

SELECT DISTINCT ?entry WHERE {
#SELECT DISTINCT ?entry str(?xlab) WHERE {
 SERVICE <https://sparql.rhea-db.org/sparql> {
 SELECT distinct ?chebi WHERE {
 ?reaction rdfs:subClassOf rh:Reaction .
 ?reaction rh:status rh:Approved .
 ?reaction rh:side ?reactionSide .
 ?reactionSide rh:contains ?participant .
 ?participant rh:compound ?compound .
 ?compound rh:chebi ?chebi .
 ?chebi rdfs:subClassOf+ CHEBI:18059 .
 }
 }
 ?entry :isoform / :smallMoleculeInteraction / :interactant ?x .
 ?x rdfs:label ?xlab .
 ?x skos:exactMatch ?chebi .
} order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","ORDER BY","DISTINCT","ALL","STR","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00247",slug:"neXtProt_NXQ_00247",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with a molecular weight less than 25 kDa",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry ?mw WHERE {
 ?entry :isoform / :sequence / :molecularWeight ?mw.
 filter ( ?mw < 25000)
} order by ?mw`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00248",slug:"neXtProt_NXQ_00248",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Variants with phenotype annotation that map to a 3D structure",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (str(?pfname) as ?phenovar) (concat(str(sample(?pdbac)),"...") as ?pdbsample) WHERE {
 ?entry :gene / :name / rdfs:label ?gen .
 ?entry :isoform ?iso.
 ?iso :proteoform ?pf .
 ?pf rdfs:label ?pfname .
 ?pf :phenotypicVariation ?pv1 .
 ?pf :difference ?pfmod .
 ?pfmod a :Variant .
 ?pfmod :start ?vstart; :end ?vend .
 ?iso :pdbMapping ?pdbmap.
 ?pdbmap rdfs:comment ?pdbac.
 ?pdbmap :start ?pdbstart ; :end ?pdbend.
 filter(?pdbstart <= ?vstart && ?pdbend >= ?vend )
} group by ?entry ?pfname`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","DISTINCT","STR","IF","CONCAT","SAMPLE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00250",slug:"neXtProt_NXQ_00250",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that act as transporters and their TCDB numbers",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (str(?trcom) as ?trannot) (str(?acc) as ?trac) {
 ?entry :isoform/ :transportActivity ?tr.
 ?tr rdfs:comment ?trcom; :evidence / :reference ?ref.
 ?ref a :Xref; :accession ?acc .
} order by ?acc`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","ORDER BY","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00251",slug:"neXtProt_NXQ_00251",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with at least one proteotypic peptide 9aa+ not mapping on canonical isoform",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

select ?entry (sample(?isopos) as ?samplematch) WHERE {
 ?entry :isoform ?iso1, ?isononcano.
 ?iso1 :swissprotDisplayed true .
 ?isononcano :swissprotDisplayed false .
 ?isononcano :peptideMapping ?pm.
 ?pm :peptideName ?pepname.
 ?pm :proteotypic true.
 ?pm :start ?pos ; :end ?pos2 .
 filter(?pos2 - ?pos >= 8) # peptide length >= 9
 bind(strafter(str(?isononcano),"-") as ?isostr)
 bind(concat(concat(?isostr,"-"),?pos) as ?isopos)
 filter not exists {?iso1 :peptideMapping/ :peptideName ?pepname.}
 }
group by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","GROUP BY","STR","EXISTS","NOT EXISTS","CONCAT","SAMPLE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00253",slug:"neXtProt_NXQ_00253",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Human pathways in which at least one protein is mitochondrial GOLD",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wp: <http://vocabularies.wikipathways.org/wp#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>


select ?pathwayname (group_concat(distinct ?gen ; SEPARATOR = ',') as ?gene) WHERE {
 SERVICE <https://sparql.wikipathways.org/sparql> {

 { ?geneProduct a wp:GeneProduct . }
 union
 { ?geneProduct a wp:Protein . }
 ?geneProduct rdfs:label ?genraw .
 bind (concat( ""^^xsd:string, ?genraw) as ?gen).
 filter(!regex(?gen,"[ a-z-]")). # ensures official gene names for subsequent neXtprot matching
 ?geneProduct dcterms:isPartOf ?pathway .
 ?pathway a wp:Pathway .
 ?pathway wp:organism ?organism .
 filter(contains(str(?organism),"9606"))
 ?pathway dcterms:title ?pathwayname .
 }

 ?entry a :Entry .
 ?entry :gene / :recommendedName / rdfs:label ?gen .
 ?entry :isoform / :cellularComponent ?loc .
 values ?mitoloc {nextprot_cv:SL-0173 nextprot_cv:GO_0005739 } # SL and GO values for mitochondrion
 ?loc :term / :childOf ?mitoloc. # mitochondrion
 ?loc :evidence / :quality :GOLD .
 filter not exists {?loc :negativeEvidence ?negev} # No negative localization evidence
}
group by ?pathwayname
order by ?pathwayname`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","VALUES","SERVICE","GROUP BY","ORDER BY","DISTINCT","STR","REGEX","EXISTS","NOT EXISTS","CONCAT","CONTAINS","MIN","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00254",slug:"neXtProt_NXQ_00254",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with associated pathways in WikiPathways",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wp: <http://vocabularies.wikipathways.org/wp#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry ?gen ?pathwayname WHERE {
 SERVICE <https://sparql.wikipathways.org/sparql> {

 {?geneProduct a wp:Protein}
 union
 {?geneProduct a wp:GeneProduct}

 ?geneProduct rdfs:label ?genraw .
	 bind (concat( ""^^xsd:string, ?genraw) as ?gen).
 filter(!regex(?gen,"[ a-z-]")). # ensures official gene names for subsequent neXtprot matching

 ?geneProduct dcterms:isPartOf ?pathway .
 ?pathway a wp:Pathway .
 ?pathway wp:organism ?organism .
 filter(contains(str(?organism),"9606")) # Human proteins
 ?pathway dcterms:title ?pathwayname .
 }
 ?entry a :Entry .
 ?entry :gene / :recommendedName / rdfs:label ?gen .
}
order by ?pathwayname`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","SERVICE","ORDER BY","DISTINCT","STR","REGEX","CONCAT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00255",slug:"neXtProt_NXQ_00255",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Variants identified in exome datasets with a frequency of >0.1",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry ?varpos ?freq where
{
 ?entry :isoform ?iso .
 ?iso :swissprotDisplayed true .
 ?iso :variant ?var .
 ?var :start ?varpos .
 ?var :evidence ?ev .
 ?ev :alleleNumber ?anb .
 ?ev :alleleFrequency ?freq .
 filter(?freq > 0.1 && ?anb > 100000)
 } order by desc(?freq)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","ALL"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00256",slug:"neXtProt_NXQ_00256",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Variants identified in exome datasets in a frequent homozygote state",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT ?entry ?varpos where
{
 ?entry :isoform ?iso .
 ?iso :swissprotDisplayed true .
 ?iso :variant ?var .
 ?var :start ?varpos .
 ?var :evidence ?ev .
 ?var :evidence / :homozygoteCount ?hcnt .
 filter(?hcnt > 100000)
 }
 order by desc(?hcnt)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00258",slug:"neXtProt_NXQ_00258",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins involved in diseases due to intronic variants with one selected publication",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX ncit: <http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sio: <http://semanticscience.org/resource/>
PREFIX so: <http://purl.obolibrary.org/obo/SO_>

SELECT DISTINCT ?entry (str(?gen) as ?gene) ?diseaseTitle ?sampleArticle WHERE {
 SERVICE <https://rdf.disgenet.org/sparql> {
 SELECT DISTINCT ?protein ?diseaseTitle (sample(?article) as ?sampleArticle) WHERE {
 ?vda sio:SIO_000628 ?variant,?disease .
	?vda sio:SIO_000772 ?article .
 ?disease a sio:SIO_010299 ; dcterms:title ?diseaseTitle . # true disease, use ncit:C7057 for traits
 ?variant a so:0001627 ; dcterms:title ?variantTitle . # intron variant
 ?variant so:associated_with ?gene .
 ?gene a ncit:C16612; sio:SIO_010078 ?protein .
 } group by ?protein ?diseaseTitle
 }
 BIND(IRI(replace(str(?protein),"purl","www")) AS ?unipage) .
 ?entry :swissprotPage ?unipage .
 ?entry :gene / :recommendedName / rdfs:label ?gen .
} order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","SERVICE","GROUP BY","ORDER BY","DISTINCT","WITH","STR","IRI","REPLACE","SAMPLE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00259",slug:"neXtProt_NXQ_00259",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins involved in diseases with clinical manifestations that include long organs ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX ncit: <http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sio: <http://semanticscience.org/resource/>

SELECT DISTINCT ?entry (str(?gen) as ?genx) ?umlsTerm WHERE {
 SERVICE <https://rdf.disgenet.org/sparql> {
 SELECT DISTINCT ?protein ?umlsTerm WHERE {
 ?gda sio:SIO_000628 ?gene,?disease .
 ?disease a sio:SIO_010056 . # traits or phenotypes
 ?disease dcterms:title ?umlsTerm .
 filter(contains(str(?umlsTerm),"Long "))
 ?gene a ncit:C16612; sio:SIO_010078 ?protein .
 }
 }
 BIND(IRI(replace(str(?protein),"purl","www")) AS ?unipage) .
 ?entry :swissprotPage ?unipage .
 ?entry :gene / :recommendedName / rdfs:label ?gen .
} order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","SERVICE","ORDER BY","DISTINCT","STR","IRI","CONTAINS","REPLACE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00260",slug:"neXtProt_NXQ_00260",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with high-frequency missense variants involved in bacterial infection, with dbSNP identifiers and position on the canonical isoform",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX ncit: <http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sio: <http://semanticscience.org/resource/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX so: <http://purl.obolibrary.org/obo/SO_>

SELECT DISTINCT ?entry ?umlsTerm (str(?snpac) as ?varid) ?pos (str(?orgaa) as ?orgAA) (str(?varaa) as ?varAA) WHERE {
 SERVICE <https://rdf.disgenet.org/sparql> {
 SELECT DISTINCT ?protein ?variantTitle ?umlsTerm
 WHERE {
 ?vda sio:SIO_000628 ?umls, ?variant .
 ?umls dcterms:title ?umlsTerm ; skos:exactMatch ?doid .
 ?doid rdfs:subClassOf+ <http://purl.obolibrary.org/obo/DOID_104> . # DO id for bacterial infection
 ?variant a so:0001583 ; dcterms:title ?variantTitle . # Missense variant
 ?variant so:associated_with ?gene .
 ?variant sio:SIO_000223 ?altAl .
 ?altAl rdf:type <http://purl.obolibrary.org/obo/GENO_0000476>; sio:SIO_000900 ?altAlFreq .
 ?altAlFreq a sio:SIO_001367; sio:SIO_000300 ?altAlFreqVal .
 filter(?altAlFreqVal > 0.01) # freq > 1%
 ?gene a ncit:C16612; sio:SIO_010078 ?protein .
 }
 }

 BIND(IRI(replace(str(?protein),"purl","www")) AS ?unipage) .
 ?entry :swissprotPage ?unipage .
 ?entry :isoform ?iso .
 ?iso :swissprotDisplayed true .
 ?iso :variant ?var .
 ?var :evidence /:reference ?xref .
 ?xref :provenance db:dbSNP; :accession ?snpac .
 ?var :start ?pos ; :original ?orgaa; :variation ?varaa .
 filter(contains(?snpac,str(?variantTitle))) # matches the exact same variant Disgenet returned
} order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","SERVICE","ORDER BY","DISTINCT","WITH","STR","IRI","CONTAINS","REPLACE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00261",slug:"neXtProt_NXQ_00261",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Glycosylation sites from GlyConnect",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT ?entry ?iso ?pos ?comment WHERE {
?entry :isoform ?iso.
?iso :swissprotDisplayed true .
?iso :glycosylationSite ?glyco.
?glyco :evidence / :assignedBy source:GlyConnect.
?glyco rdfs:comment ?comment.
?glyco :start ?pos.

} order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00265",slug:"neXtProt_NXQ_00265",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all positional annotations at a given position on a neXtProt isoform",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX isoform: <http://nextprot.org/rdf/isoform/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?iso ?spos ?epos ?annot_type (str(?txt) as ?note) where
{
 values ?iso { isoform:NX_Q99728-1 }
 values ?poi {107} # position of interest

{
 ?iso :positionalAnnotation ?statement . optional {?statement rdfs:comment ?txt .}

 ?statement a ?annot_type .
 ?statement :start ?spos; :end ?epos .
}
 union

{
 ?iso :proteoform ?pf .
 ?pf :difference ?varmut; :phenotypicVariation ?phvar .
 ?varmut :start ?spos; :end ?epos.
 ?phvar :term ?phtype; :impactedObject /:term /rdfs:label ?ioTermlab .
 ?phvar a ?annot_type .
 ?phtype :childOf nextprot_cv:ME_0000002; rdfs:label ?effect . # children of impact
 bind (concat(CONCAT(?effect," "),?ioTermlab) as ?txt) }

 filter((?spos <= ?poi) && (?epos >= ?poi)) # select annotations encompassing the position of interest
} order by ?spos`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","UNION","BIND","VALUES","ORDER BY","DISTINCT","STR","IF","CONCAT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00266",slug:"neXtProt_NXQ_00266",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins binding estradiol and/or similar molecules (substructure search with SMILES) and their associated GO_MF terms",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX cco: <http://rdf.ebi.ac.uk/terms/chembl#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>

SELECT distinct ?entry (group_concat(distinct str(?gomflab); SEPARATOR = ",") as ?gomfx) WHERE {
	SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/idsm> {
		SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/cco> {
		 ?compound sachem:substructureSearch [ sachem:query "CC12CCC3C(C1CCC2O)CCC4=C3C=CC(=C4)O" ] . # smiles chain for estradiol
		}
		?ACTIVITY rdf:type cco:Activity;
		cco:hasMolecule ?compound;
		cco:hasAssay ?ASSAY.
		?ASSAY cco:hasTarget ?TARGET.
		?TARGET cco:hasTargetComponent ?COMPONENT.
		?TARGET cco:taxonomy <http://identifiers.org/taxonomy/9606> . # human protein target
		?COMPONENT cco:targetCmptXref ?UNIPROT.
		#?UNIPROT rdf:type cco:UniprotRef.
		filter(contains(str(?UNIPROT),"uniprot"))
	}

	?entry skos:exactMatch ?UNIPROT.
	?entry :isoform ?iso.
	?iso :goMolecularFunction / :term ?gomf .
	?gomf rdfs:label ?gomflab .
}

GROUP BY ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","GROUP BY","DISTINCT","STR","IF","CONCAT","SUBSTR","CONTAINS","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00267",slug:"neXtProt_NXQ_00267",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins binding estradiol and/or similar molecules (similarity search with SMILES), and their associated GO_MF terms",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX cco: <http://rdf.ebi.ac.uk/terms/chembl#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>


SELECT distinct ?entry (group_concat(distinct str(?gomflab); SEPARATOR = ",") as ?gomfx) WHERE {
 		SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/idsm> {
 		SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/cco> {
 		?compound sachem:similarCompoundSearch [ sachem:query "CC12CCC3C(C1CCC2O)CCC4=C3C=CC(=C4)O" ] . # smiles chain for estradiol
		 }
		 		?ACTIVITY rdf:type cco:Activity;
		cco:hasMolecule ?compound;
 cco:hasAssay ?ASSAY.
		 ?ASSAY cco:hasTarget ?TARGET.
		 ?TARGET cco:taxonomy <http://identifiers.org/taxonomy/9606> . # human protein target
		 ?TARGET cco:hasTargetComponent ?COMPONENT.
		 ?COMPONENT cco:targetCmptXref ?UNIPROT.
		 filter(contains(str(?UNIPROT),"uniprot"))
		 }
	?entry skos:exactMatch ?UNIPROT.
	?entry :isoform ?iso.
	?iso :goMolecularFunction / :term ?gomf .
	?gomf rdfs:label ?gomflab .
}
group by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","GROUP BY","DISTINCT","STR","IF","CONCAT","CONTAINS","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00269",slug:"neXtProt_NXQ_00269",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with associated cancer pathways in WikiPathways (via Disease Ontology classification)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wp: <http://vocabularies.wikipathways.org/wp#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry (str(?gen) as ?gene) ?pathwayname WHERE {
 SERVICE <https://sparql.wikipathways.org/sparql> {
 {?geneProduct a wp:Protein}
 union
 {?geneProduct a wp:GeneProduct}
 ?geneProduct rdfs:label ?genraw .
 bind (concat( ""^^xsd:string, ?genraw) as ?gen).
 filter(!regex(?gen,"[ a-z-]")). # ensures official gene names for subsequent neXtprot matching
 ?geneProduct dcterms:isPartOf ?pathway .
 ?pathway a wp:Pathway .
 ?pathway wp:organism ?organism .
 filter(contains(str(?organism),"9606")) # Human proteins
 ?pathway dcterms:title ?pathwayname .
 ?pathway wp:ontologyTag <http://purl.obolibrary.org/obo/DOID_162> . # Parent id for all cancers
 }
 ?entry a :Entry .
 ?entry :gene / :recommendedName / rdfs:label ?gen .
}
order by ?pathwayname`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","SERVICE","ORDER BY","DISTINCT","ALL","STR","REGEX","CONCAT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00270",slug:"neXtProt_NXQ_00270",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins belonging to Rett syndrome pathways, and their subcellular locations (GOLD)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wp: <http://vocabularies.wikipathways.org/wp#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

select ?entry (str(?gen) AS ?gene) ?pathwayname (group_concat(distinct str(?loclab); SEPARATOR = ",") as ?locs) WHERE {
 SERVICE <https://sparql.wikipathways.org/sparql> {

 {?geneProduct a wp:Protein}
 union
 {?geneProduct a wp:GeneProduct}

 ?geneProduct rdfs:label ?genraw .
 bind (concat( ""^^xsd:string, ?genraw) as ?gen).

 filter(!regex(?gen,"[ a-z-]")). # ensures official gene names for subsequent neXtprot matching
 ?geneProduct dcterms:isPartOf ?pathway .
 ?pathway a wp:Pathway .
 ?pathway wp:organism ?organism .
 filter(contains(str(?organism),"9606"))
 ?pathway dcterms:title ?pathwayname .
 filter(regex(?pathwayname,"rett","i")).
 }
 ?entry a :Entry .
 ?entry :gene / :recommendedName / rdfs:label ?gen .
 ?entry :isoform ?iso.
 ?iso :cellularComponent ?locannot .
 ?locannot :term ?locterm .
 ?locterm rdfs:label ?loclab .
 ?locannot :evidence ?locev .
 ?locev :quality :GOLD .
 filter not exists {?locannot :negativeEvidence ?locev .}
}
group by ?entry ?gen ?pathwayname
order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","SERVICE","GROUP BY","ORDER BY","DISTINCT","STR","REGEX","EXISTS","NOT EXISTS","CONCAT","CONTAINS","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00271",slug:"neXtProt_NXQ_00271",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Polymorphisms located on ACE2 and TMPRSS2 and affecting proteins’ activity, structure, PTM...",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT (str(?gn) as ?gene) ?pos (str(?snpac) as ?varid) (str(?orgaa) as ?orgAA) ?annot_type (str(?txt) as ?note) (str(?varaa) as ?varAA) ?freq where
{
 values ?entry {nextprot:NX_Q9BYF1 nextprot:NX_O15393} # proteins of interest (ACE2, TMPRSS2)
 ?entry :gene /:recommendedName / rdfs:label ?gn.
 ?entry :isoform ?iso .
 ?iso :swissprotDisplayed true; :variant ?var .
 ?var :start ?pos ; :original ?orgaa; :variation ?varaa .
 optional {?var :evidence / :alleleFrequency ?freq .}
 ?var :evidence / :reference ?xref .
 ?iso :positionalAnnotation ?annot .
 optional {?annot rdfs:comment ?txt .}
 ?annot a ?annot_type .
 {
 ?annot :start ?pos; :end ?pos.
 filter not exists {?annot a :Variant. }
 filter not exists {?annot a :SequenceConflict. }
 }
 union
 {
 ?annot a :DisulfideBond.
 {?annot :start ?pos. }
 union
 {?annot :end ?pos. }
 }
 } order by ?entry ?pos`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","UNION","VALUES","ORDER BY","DISTINCT","ALL","STR","EXISTS","NOT EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00272",slug:"neXtProt_NXQ_00272",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins involved in coronaviruses/SARS-CoV-2 pathways with associated medical information",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX wp: <http://vocabularies.wikipathways.org/wp#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry (str(?gen) AS ?gene) ?pathwayname (str(?discom) AS ?discom1) ?medsource WHERE {
 SERVICE <http://sparql.wikipathways.org/sparql> {
 {?geneProduct a wp:Protein}
 union
 {?geneProduct a wp:GeneProduct}

 ?geneProduct rdfs:label ?genraw .
 bind (concat( ""^^xsd:string, ?genraw) as ?gen).
 filter(!regex(?gen,"[ a-z-]")). # ensures official gene names for subsequent neXtprot matching

 ?geneProduct dcterms:isPartOf ?pathway .
 ?pathway a wp:Pathway .
 ?pathway wp:organism ?organism .
 filter(contains(str(?organism),"9606"))

 ?pathway dcterms:title ?pathwayname .
 filter(regex(?pathwayname,"sars-cov-2","i")|| regex(?pathwayname,"corona","i") ).
 }

 ?entry a :Entry .
 ?entry :gene / :recommendedName / rdfs:label ?gen .
 ?entry :isoform ?iso.
 ?iso :medical ?med.
 ?med rdfs:comment ?discom.
 ?med :evidence/:assignedBy ?medsource.
}
order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","SERVICE","ORDER BY","DISTINCT","STR","REGEX","CONCAT","CONTAINS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00276",slug:"neXtProt_NXQ_00276",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Diseases/phenotypes associated with coding variants and associated publications for a given gene",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX dcterms: <http://purl.org/dc/terms/>
PREFIX ncit: <http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#>
PREFIX sio: <http://semanticscience.org/resource/>
PREFIX so: <http://purl.obolibrary.org/obo/SO_>

SELECT DISTINCT ?entry (str(?umlsTerm) as ?umlsTermSTR) (str(?snpac) as ?varid) ?pos (str(?orgaa) as ?orgAA) (str(?varaa) as ?varAA) ?article WHERE {
 SERVICE <https://rdf.disgenet.org/sparql> {
 ?vda sio:SIO_000628 ?umls, ?variant .
 ?vda sio:SIO_000772 ?article .
 ?umls dcterms:title ?umlsTerm.
 {?umls a sio:SIO_010299 .} # disease
 union
 {?umls a sio:SIO_010056 .} # or phenotype
 ?variant a so:0001583 ; dcterms:title ?variantTitle . # Missense variant
 ?variant so:associated_with ?gene .
 ?gene a ncit:C16612; sio:SIO_010078 ?protein .
 ?gene sio:SIO_000205 ?gname.
 filter(contains(str(?gname),"HBB")) # Hemoglobin gene (NX_P68871)
 }

 BIND(IRI(replace(str(?protein),"purl","www")) AS ?unipage) .
 ?entry :swissprotPage ?unipage .
 ?entry :isoform ?iso .
 ?iso :swissprotDisplayed true .
 ?iso :variant ?var .
 ?var :evidence /:reference ?xref .
 ?xref :provenance db:dbSNP; :accession ?snpac .
 ?var :start ?pos ; :original ?orgaa; :variation ?varaa .
 filter(contains(?snpac,str(?variantTitle))) # matches the exact same variant Disgenet returned
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","SERVICE","DISTINCT","WITH","STR","IRI","CONTAINS","REPLACE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00278",slug:"neXtProt_NXQ_00278",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Authors who have reported more than 25000 human protein-protein interactions ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>

SELECT DISTINCT (str(?auth) AS ?auth1) (count (distinct ?itid) as ?itcnt) WHERE {
 ?entry :isoform / :binaryInteraction ?interaction.
 ?interaction :evidence / :reference ?publi .
 ?publi :author / :name ?auth .
 ?interaction :interactant ?interactant; :quality :GOLD; :entryAnnotationId ?itid.
} group by ?auth having (count (distinct ?itid) > 25000)
order by desc(?itcnt)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","HAVING","DISTINCT","STR","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00279",slug:"neXtProt_NXQ_00279",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins for which an interaction mapping region is described for both interactants ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (str(?gen) AS ?gene) ?entry2 (str(?gen2) AS ?gene2) WHERE {
 ?entry :isoform ?iso; :gene / :recommendedName / rdfs:label ?gen .
 ?iso :interactionMapping / :interactant ?entry2.
 ?entry2 :gene / :recommendedName / rdfs:label ?gen2 .
 filter exists { ?entry2 :isoform / :interactionMapping / :interactant ?entry.}
} order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","STR","EXISTS"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00280",slug:"neXtProt_NXQ_00280",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins interacting with at least 10 members of a protein family ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry (str(?gen) as ?gene) (str(?xlab) as ?itfamily) ?itcnt ?membercnt ((xsd:float(?itcnt) / xsd:float(?membercnt)) as ?ratio) WHERE {
 {
 select ?xterm ?xlab (count(distinct ?member) as ?membercnt) WHERE {
 ?member :isoform? / :familyName /:term /:childOf ?xterm.
 ?xterm rdfs:label ?xlab .
 } group by ?xterm ?xlab
 }
 {
 select ?xterm ?entry ?gen (count (distinct ?interactant) as ?itcnt) WHERE {
 ?entry :isoform ?iso; :gene / :name / rdfs:label ?gen .
 ?iso :interaction ?it.
 ?it :quality :GOLD; :interactant ?interactant.
 ?interactant :familyName /:term /:childOf ?xterm .
 } group by ?xterm ?entry ?gen having (count (distinct ?interactant) >= 10)
 }
}
order by desc(?ratio)
limit 60`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","LIMIT","HAVING","DISTINCT","STR","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00281",slug:"neXtProt_NXQ_00281",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Protein pairs with at least 50 common interactors (excluding keratins) ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry ?gene ?entry2 ?gene2 ?commonitcnt ?sampleit WHERE {
 {
 SELECT DISTINCT ?entry (str(?gen) as ?gene) ?entry2 (str(?gen2) as ?gene2) (count (distinct ?commonpartner) as ?commonitcnt) (str(sample(?genit)) as ?sampleit) WHERE {
 ?entry :isoform / :interaction ?it1; :gene / :name / rdfs:label ?gen.
 ?it1 :quality :GOLD; :interactant ?commonpartner .
 ?entry2 :isoform / :interaction ?it2; :gene / :name / rdfs:label ?gen2.
 ?it2 :quality :GOLD;:interactant ?commonpartner .
 ?commonpartner :gene / :name / rdfs:label ?genit .
 filter(?entry != ?entry2)
 filter(!contains(?gen,"KRT") && !contains(?gen2,"KRT") && !contains(?genit,"KRT")) # remove sticky keratins
} group by ?entry ?gen ?entry2 ?gen2 having (count (distinct ?commonpartner) >= 50) 				}
filter(substr(str(?entry),34) < substr(str(?entry2),34)) # ensures only one row by it pair
}
order by desc(?commonitcnt)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","ORDER BY","HAVING","DISTINCT","MOVE","STR","SUBSTR","CONTAINS","COUNT","SAMPLE"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00282",slug:"neXtProt_NXQ_00282",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with an interaction mapping encompassing a coiled coil or bZip region ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry (str(?gen) as ?gene) ?itlen ?coilen (str(?gen2) as ?partnergene) WHERE {
 ?entry :isoform ?iso; :gene / :recommendedName / rdfs:label ?gen .
 ?iso :swissprotDisplayed true; :interactionMapping ?itmap .
 ?itmap :interactant ?entry2; :start ?its; :end ?ite .
 bind(?ite - ?its + 1 as ?itlen)
 ?entry2 :isoform? ?iso2; :gene / :recommendedName / rdfs:label ?gen2.
 ?iso2 :swissprotDisplayed true.
 {?iso :coiledCoilRegion ?coil . }
 union
 {?iso :region ?coil .
 ?coil :term nextprot_cv:DO-00078 # bzip
 }
 ?coil :start ?cs; :end ?ce .
 bind(?ce - ?cs + 1 as ?coilen)
 filter (?cs >= ?its -15 && ?ce <= ?ite + 15) # overlap
 filter(xsd:float(?itlen) / xsd:float(?coilen) < 2.0) # itlen no more than 2 coilen
} order by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","ORDER BY","DISTINCT","STR","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00283",slug:"neXtProt_NXQ_00283",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with no reported mitochondrial localization but interacting with 20 or more mitochondrial proteins ('gold' quality) ",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (str(?gen) as ?gene) (count(distinct ?mitopartner) as ?mitcount) (count(distinct ?partner) as ?itcount) WHERE {
values ?mitloc {nextprot_cv:GO_0005739 nextprot_cv:SL-0173} # GO and SL values for mitochondrion
 ?entry :isoform ?iso.
 ?entry :gene / :recommendedName / rdfs:label ?gen.
 filter not exists { ?iso :cellularComponent /:term / :childOf ?mitloc }
 ?iso :binaryInteraction ?it, ?it2.
 ?it :interactant ?mitopartner; :quality :GOLD .
 ?mitopartner :isoform / :cellularComponent ?loc .
 ?loc :quality :GOLD; :term / :childOf ?mitloc .
 ?it2 :interactant ?partner; :quality :GOLD .
} group by ?entry ?gen ?mitcnt having (count(distinct ?mitopartner) >= 20)
order by desc(?mitcnt)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","GROUP BY","ORDER BY","HAVING","DISTINCT","STR","EXISTS","NOT EXISTS","COUNT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00286",slug:"neXtProt_NXQ_00286",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Protein domains or regions that frequently occur in interaction mappings",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT (str(?rlab) as ?rlab1) (count(distinct ?itid) as ?itcnt) WHERE {
 ?entry :isoform ?iso; :gene / :name / rdfs:label ?gen .
 ?iso :swissprotDisplayed true; :interactionMapping ?itmap .
 ?itmap :entryAnnotationId ?itid; :start ?its; :end ?ite .
 bind(?ite - ?its + 1 as ?itlen)
 ?iso :region ?reg .
 ?reg :start ?rs; :end ?re; :term ?rterm .
 ?rterm rdfs:label ?rlab .
 bind(?re - ?rs + 1 as ?rlen)
 filter (?rs >= ?its - 15 && ?re <= ?ite + 15)
 filter(xsd:float(?itlen) / xsd:float(?rlen) < 2.0)
} group by ?rlab having(count(distinct ?itid) >= 10)
order by desc(?itcnt)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","GROUP BY","ORDER BY","HAVING","DISTINCT","STR","COUNT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00287",slug:"neXtProt_NXQ_00287",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with interactions obtained from x-ray crystallography",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :isoform ?iso.
 ?iso :binaryInteraction /:evidence ?ev.
 ?ev :interactionDetectionMethod nextprot_cv:MI_0114. # x-ray crystallography
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","DISTINCT","ALL","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00288",slug:"neXtProt_NXQ_00288",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Biological Process and Molecular Function GO terms related to UniPathway metabolic pathways",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT (str(?pathwayname) as ?unipathway) ?related (str(?pathlabel) as ?GO_BP) (str(?pathlabel2) as ?GO_MF) WHERE {
 ?entry :isoform ?iso.
 ?iso :pathway ?pathannot .
 ?pathannot :term ?pathterm; rdfs:comment ?pathwayname.
 ?pathterm :related ?related .
 {?related a :GoBiologicalProcessCv ; rdfs:label ?pathlabel .}
 UNION
 {?related a :GoMolecularFunctionCv ; rdfs:label ?pathlabel2 .}
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00289",slug:"neXtProt_NXQ_00289",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"KEGG and Reactome pathway names",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX source: <http://nextprot.org/rdf/source/>

SELECT DISTINCT ?src (str(?pathwayname) as ?pw) WHERE {
 ?entry :isoform / :pathway ?pathannot .
 ?pathannot rdfs:comment ?pathwayname; :evidence / :assignedBy ?src.
 filter(?src = source:Reactome || ?src = source:KEGG_PTW)
} order by ?pathwayname`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","DISTINCT","STR"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00290",slug:"neXtProt_NXQ_00290",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins that are enzymes requiring the cofactor Ca(2+) and with a calcium binding site",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?entry (str(?name) as ?prot_name) (str(?ec) as ?ecs) where

{ ?entry :recommendedName ?name_entity .
 ?name_entity a :ProteinName; rdfs:label ?name.
 ?entry :isoform ?iso.
 ?iso :enzymeClassification /rdfs:comment ?ec.
 ?iso :cofactor /:interactant /skos:exactMatch CHEBI:29108. # Ca(2+) cofactor
 ?iso :bindingSite /:interactant ?inter.
 ?inter :accession "CHEBI:29108"^^xsd:string # binding site interactant Ca(2+)
}
order by asc(?ec)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","ORDER BY","DISTINCT","STR","IF"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00294",slug:"neXtProt_NXQ_00294",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Variants in MECP2 causing Rett Syndrome",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?start ?stop ?original ?modified WHERE {
 nextprot:NX_P51608 :isoform ?iso. #MECP2 entry
 ?iso :swissprotDisplayed true. #Swissprot canonical isoform
 {
 ?iso :variant ?v.
 ?v :disease nextprot_cv:DI-00999. #UniProtKB term for Rett Syndrome
 ?v :start ?start.
 ?v :end ?stop.
 filter((?stop - ?start) = 0) #single amino acid variants (SAAVs)
 ?v :original ?original.
 ?v :variation ?modified.
 }
 union
 {
 ?iso :proteoform ?pf.
 ?pf :diseaseRelatedVariant ?ann.
 filter not exists {?ann :negativeEvidence ?negev} # No negative disease evidence
 ?ann :impactedObject /:term nextprot_cv:C75488. #NCI Thesaurus term for Rett Syndrome
 ?pf :difference ?v.
 ?v :start ?start.
 ?v :end ?stop.
 filter((?stop - ?start) = 0)
 ?v :original ?original.
 ?v :variation ?modified.
 }
}
order by asc(?start)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","ORDER BY","DISTINCT","IF","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00295",slug:"neXtProt_NXQ_00295",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"References for SCN1A variants causing Dravet syndrome",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT (str(?publiid) as ?PMID) ?title WHERE {
 nextprot:NX_P35498 :isoform ?iso. #SCN1A entry
 ?iso :swissprotDisplayed true. #Swissprot canonical isoform
 {
 ?iso :variant ?v.
 ?v :disease nextprot_cv:DI-01023. #UniProtKB term for Dravet syndrome
 ?v :evidence /:reference ?pub.
 ?pub :title ?title.
 ?pub :from ?xref .
 ?xref :accession ?publiid ; :provenance db:PubMed .
 }
 union
 {
 ?iso :proteoform ?pf.
 ?pf :diseaseRelatedVariant ?ann.
 filter not exists {?ann :negativeEvidence ?negev} # No negative disease evidence
 ?ann :impactedObject /:term nextprot_cv:C116573. #NCI Thesaurus term for Dravet syndrome
 ?ann :evidence /:reference ?pub.
 ?pub :title ?title.
 ?pub :from ?xref .
 ?xref :accession ?publiid ; :provenance db:PubMed .
 }
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","FROM","STR","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00296",slug:"neXtProt_NXQ_00296",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Pathways in which SCN1A GOLD interactants are involved",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?pathway WHERE {
 nextprot:NX_P35498 :isoform /:binaryInteraction ?interaction.
 ?interaction :interactant ?entry; :quality :GOLD.
 ?entry a :Entry.
 ?entry :isoform /:pathway /rdfs:comment ?pathway.
}
order by asc(?pathway)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00297",slug:"neXtProt_NXQ_00297",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"PDB structures spanning the complete sequence of the mature protein",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>

select ?entry (group_concat(distinct str(?pdbac); SEPARATOR = ",") as ?pdbacs) WHERE {
 ?entry :isoform ?iso.
 ?iso :pdbMapping ?pdbmap.
 ?pdbmap :evidence /:reference ?ref.
 ?ref :provenance db:PDB.
 ?ref :accession ?pdbac.
 ?pdbmap :start ?pdbstart ; :end ?pdbend.
 ?iso :matureProtein [ :start ?mstart ; :end ?mend]
 filter (?mend-?mstart > 0)
 filter ((?pdbstart <= ?mstart) && (?pdbend >= ?mend))
} group by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","DISTINCT","STR","CONCAT","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00298",slug:"neXtProt_NXQ_00298",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Recommended isoform names for MSH6",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot: <http://nextprot.org/rdf/entry/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?iso ?name WHERE {
 nextprot:NX_P52701 :isoform ?iso. # MSH6 entry
 ?iso :recommendedName ?name_entity .
 ?name_entity a :IsoformName; rdfs:label ?name.
}
order by asc(?iso)`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00299",slug:"neXtProt_NXQ_00299",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins interacting with small molecules according to DrugBank",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX db: <http://nextprot.org/rdf/db/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT DISTINCT ?entry (group_concat(distinct str(?lbl); SEPARATOR = ";") as ?mol) WHERE {
 ?entry :isoform / :smallMoleculeInteraction / :interactant ?ref .
 ?ref :provenance db:DrugBank.
 ?ref rdfs:label ?lbl.
}
group by ?entry`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","DISTINCT","ALL","STR","CONCAT","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00300",slug:"neXtProt_NXQ_00300",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with protein existence \\",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>
PREFIX orthodb: <http://purl.orthodb.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

select ?entry (str(?name) as ?human_name) (group_concat(distinct str(?fly_name); SEPARATOR = ",") as ?fly_names) where
{
?entry :isoform ?iso.
?entry :gene / :recommendedName / rdfs:label ?name.
?entry :existence :Evidence_at_protein_level .
filter not exists { ?iso :functionInfo ?_. }
filter not exists { ?iso :catalyticActivity ?_ .}
filter not exists { ?iso :transportActivity ?_ .}
filter not exists { ?iso :pathway ?_. }
filter not exists { ?iso :function / :term ?fterm . filter(?fterm != nextprot_cv:GO_0005524 && ?fterm != nextprot_cv:GO_0000287 && ?fterm != nextprot_cv:GO_0005515 && ?fterm != nextprot_cv:GO_0042802 && ?fterm != nextprot_cv:GO_0008270 && ?fterm != nextprot_cv:GO_0051260 && ?fterm != nextprot_cv:GO_0005509 && ?fterm != nextprot_cv:GO_0003676 && ?fterm != nextprot_cv:GO_0003824 && ?fterm != nextprot_cv:GO_0007165 && ?fterm != nextprot_cv:GO_0035556 && ?fterm != nextprot_cv:GO_0046914 && ?fterm != nextprot_cv:GO_0046872)}
?iso :expression ?e1.
?e1 :term/:childOf nextprot_cv:TS-0095;:evidence/:observedExpression :High.
{
SERVICE <https://sparql.orthodb.org/sparql>
{?gene rdfs:seeAlso ?entry; orthodb:memberOf ?og.
?og orthodb:ogBuiltAt [up:scientificName ?clade]
; orthodb:hasMember ?ortholog.
?ortholog orthodb:name ?fly_name; up:organism/a/up:scientificName 'Drosophila melanogaster'.
filter (?clade='Metazoa') }
}
} group by ?entry ?name`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","GROUP BY","DISTINCT","STR","IF","EXISTS","NOT EXISTS","CONCAT","MIN","GROUP_CONCAT"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00301",slug:"neXtProt_NXQ_00301",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"PE1 entries that comply with HPP guidelines (at least 2 non overlapping peptides of at least 9aa from a single data source)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :existence :Evidence_at_protein_level .
 ?entry :isoform / :uniprotKeyword / :term nextprot_cv:KW-1267.
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","MIN"],category:"sib-swiss neXtProt"},{name:"neXtProt - NXQ 00302",slug:"neXtProt_NXQ_00302",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"PE1 entries that do not comply with HPP guidelines (at least 2 non overlapping peptides of at least 9aa from a single data source)",context:null,inidces:[],query:`PREFIX : <http://nextprot.org/rdf/>
PREFIX nextprot_cv: <http://nextprot.org/rdf/terminology/>

SELECT DISTINCT ?entry WHERE {
 ?entry :existence :Evidence_at_protein_level .
 ?entry :isoform ?iso.
 filter not exists { ?iso :uniprotKeyword / :term nextprot_cv:KW-1267.}
}`,ontologies:["RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","EXISTS","NOT EXISTS","MIN"],category:"sib-swiss neXtProt"}],dP=[{name:"OMA - 01-rat-proteins",slug:"OMA_01-rat-proteins",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all Rattus norvegicus proteins present in OMA RDF database.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX obo: <http://purl.obolibrary.org/obo/>

SELECT ?protein ?OMA_link
WHERE {
    ?protein a orth:Protein ;
        orth:organism ?organism .
    ?organism obo:RO_0002162 ?taxon . # in taxon
    ?taxon up:scientificName 'Rattus norvegicus' .
    ?protein rdfs:seeAlso ?OMA_link .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","IF"],category:"sib-swiss OMA"},{name:"OMA - 02-all-species",slug:"OMA_02-all-species",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which species are available on OMA database and their scientific names?",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?species ?sciname
WHERE {
    ?species a up:Taxon ;
        up:scientificName ?sciname ;
        up:rank up:Species .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","IF"],category:"sib-swiss OMA"},{name:"OMA - 03-ins-encoded-proteins",slug:"OMA_03-ins-encoded-proteins",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all proteins in OMA that is encoded by the INS gene and their mnemonics and evidence types from Uniprot database (federated query).",context:null,inidces:[],query:`PREFIX lscr: <http://purl.org/lscr#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?proteinOMA ?species ?mnemonic ?evidenceType ?UniProt_URI
WHERE {
    ?proteinOMA a orth:Protein ;
        orth:organism/obo:RO_0002162/up:scientificName ?species ;
        rdfs:label 'INS' .
    ?proteinOMA lscr:xrefUniprot ?UniProt_URI .
    # Search the INS gene mnemonics and evidence types from Uniprot database
    SERVICE <https://sparql.uniprot.org/sparql> {
        ?UniProt_URI up:mnemonic ?mnemonic ;
            up:existence/rdfs:label ?evidenceType.
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","FROM","URI","IF"],category:"sib-swiss OMA"},{name:"OMA - 04-orthologs-of-ensembl-gene",slug:"OMA_04-orthologs-of-ensembl-gene",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all genes that are orthologous to ENSLACG00000002497 Ensembl gene (identifier)",context:null,inidces:[],query:`PREFIX ensembl: <http://rdf.ebi.ac.uk/resource/ensembl/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sio: <http://semanticscience.org/resource/>

SELECT ?protein2 ?OMA_LINK
WHERE {
    # The three that contains Orthologs. The leafs are proteins.
    # This graph pattern defines the relationship protein1 is Orthologs to protein2
    ?cluster a orth:OrthologsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?protein2 .
    ?node1 orth:hasHomologousMember* ?protein1 .

    # Specify the protein to look for its orthologs
    ?protein1 sio:SIO_010079/lscr:xrefEnsemblGene ensembl:ENSLACG00000002497 .

    # The OMA link to the second protein
    ?protein2 rdfs:seeAlso ?OMA_LINK .

    filter(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GRAPH","IF","CONTAINS"],category:"sib-swiss OMA"},{name:"OMA - 05-paralogs-of-ensembl-gene",slug:"OMA_05-paralogs-of-ensembl-gene",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all genes that are paralogous to ENSG00000244734 Ensembl gene (identifier).",context:null,inidces:[],query:`PREFIX ensembl: <http://rdf.ebi.ac.uk/resource/ensembl/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sio: <http://semanticscience.org/resource/>

SELECT ?protein2 ?OMA_LINK
WHERE {
    # The three that contains paralogs. The leafs are proteins.
    # This graph pattern defines the relationship protein1 is paralogous to protein2
    ?cluster a orth:ParalogsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?protein2 .
    ?node1 orth:hasHomologousMember* ?protein1 .

    # Specify the protein to look for its paralogs
    ?protein1 sio:SIO_010079/lscr:xrefEnsemblGene ensembl:ENSG00000244734 .

    # The OMA link to the second protein
    ?protein2 rdfs:seeAlso ?OMA_LINK .

    FILTER(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GRAPH","IF","CONTAINS"],category:"sib-swiss OMA"},{name:"OMA - 06-paralogs-with-uniprot-xrefs",slug:"OMA_06-paralogs-with-uniprot-xrefs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all genes that are paralogous to HUMAN00529 OMA protein (identifier) and their cross-reference links to OMA and Uniprot.",context:null,inidces:[],query:`PREFIX dc: <http://purl.org/dc/terms/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>

SELECT ?protein2 ?Uniprot_link
WHERE {
    ?cluster a orth:ParalogsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?protein2 .
    ?node1 orth:hasHomologousMember* ?protein1 .
    ?protein1 a orth:Protein ;
        dc:identifier 'HUMAN00529'.
    ?protein2 a orth:Protein ;
        lscr:xrefUniprot ?Uniprot_link .
    filter(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","IF"],category:"sib-swiss OMA"},{name:"OMA - 07-orthologs-with-uniprot-xrefs",slug:"OMA_07-orthologs-with-uniprot-xrefs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all genes that are orthologous to HUMAN22169 OMA protein (identifier) and their cross-reference links to OMA and Uniprot.",context:null,inidces:[],query:`PREFIX dc: <http://purl.org/dc/terms/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>

SELECT ?protein2 ?Uniprot_link
WHERE {
    ?cluster a orth:OrthologsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?protein2 .
    ?node1 orth:hasHomologousMember* ?protein1 .
    ?protein1 a orth:Protein ;
        dc:identifier 'HUMAN22169' .
    ?protein2 a orth:Protein ;
        lscr:xrefUniprot ?Uniprot_link .
    FILTER(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","IF"],category:"sib-swiss OMA"},{name:"OMA - 08-rabbit-apoci-orthologs",slug:"OMA_08-rabbit-apoci-orthologs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all genes per species that are orthologous to Rabbit's APOCI or APOC1 gene and their cross-reference links to OMA and Uniprot including the corresponding Ensembl gene identifier.",context:null,inidces:[],query:`PREFIX dc: <http://purl.org/dc/terms/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sio: <http://semanticscience.org/resource/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein1 ?protein2 ?geneName2 ?species2 ?Prot2_uniprot ?prot2_ensemblGeneId
WHERE {
    ?cluster a orth:OrthologsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?protein2 .
    ?node1 orth:hasHomologousMember* ?protein1 .
    ?protein1 a orth:Protein ;
        orth:organism/obo:RO_0002162/up:scientificName 'Oryctolagus cuniculus';
        rdfs:label 'APOCI' .
    ?protein2 a orth:Protein ;
        lscr:xrefUniprot ?Prot2_uniprot ;
        sio:SIO_010079/lscr:xrefEnsemblGene/dc:identifier ?prot2_ensemblGeneId ;
        rdfs:label ?geneName2 ;
        orth:organism/obo:RO_0002162/up:scientificName ?species2.
    FILTER(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","IF"],category:"sib-swiss OMA"},{name:"OMA - 09-rabbit-orthologs-of-mouse-homoglobinY",slug:"OMA_09-rabbit-orthologs-of-mouse-homoglobinY",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all Rabbit's proteins encoded by genes that are orthologous to Mouses's hemoglobin Y gene and their cross-reference links to Uniprot.",context:null,inidces:[],query:`PREFIX lscr: <http://purl.org/lscr#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?MOUSE_PROTEIN ?RABIT_PROTEIN ?MOUSE_UNIPROT_XREF ?RABIT_UNIPROT_XREF
WHERE {
    ?cluster a orth:OrthologsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?RABIT_PROTEIN .
    ?node1 orth:hasHomologousMember* ?MOUSE_PROTEIN .
    ?MOUSE_PROTEIN a orth:Protein .
    ?MOUSE_PROTEIN  orth:organism/obo:RO_0002162/up:scientificName 'Mus musculus' ;
        rdfs:label 'HBB-Y';
        lscr:xrefUniprot ?MOUSE_UNIPROT_XREF .
    ?RABIT_PROTEIN a orth:Protein .
    ?RABIT_PROTEIN orth:organism/obo:RO_0002162/up:scientificName 'Oryctolagus cuniculus' .
    ?RABIT_PROTEIN lscr:xrefUniprot ?RABIT_UNIPROT_XREF.
    FILTER(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","IF"],category:"sib-swiss OMA"},{name:"OMA - 10-paralogs-in-human-of-hbb",slug:"OMA_10-paralogs-in-human-of-hbb",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all human proteins that are paralogous to the HBB gene and their UniProt cross-references.",context:null,inidces:[],query:`PREFIX lscr: <http://purl.org/lscr#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?PROTEIN_HBB ?IS_PARALOGOUS_TO_PROTEIN ?PARALOG_GENE_LABEL ?HBB_UNIPROT_XREF ?PARALOG_UNIPROT_XREF
WHERE {
    ?cluster a orth:OrthologsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?PROTEIN_HBB .
    ?node1 orth:hasHomologousMember* ?IS_PARALOGOUS_TO_PROTEIN .
    ?PROTEIN_HBB a orth:Protein ;
        orth:organism/obo:RO_0002162/up:scientificName 'Homo sapiens' ;
        rdfs:label 'HBB';
        lscr:xrefUniprot ?HBB_UNIPROT_XREF .
    ?IS_PARALOGOUS_TO_PROTEIN a orth:Protein ;
        orth:organism/obo:RO_0002162/up:scientificName 'Homo sapiens' ;
        lscr:xrefUniprot ?PARALOG_UNIPROT_XREF ;
        rdfs:label ?PARALOG_GENE_LABEL .
    FILTER(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","IF"],category:"sib-swiss OMA"},{name:"OMA - 11-percentage-of-proteins-with-paralogs",slug:"OMA_11-percentage-of-proteins-with-paralogs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"The percentage of proteins in Drosophila melanogaster that has at least one paralogous sequence (protein).",context:null,inidces:[],query:`PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT (((xsd:float(?num_paralogy)*100)/xsd:float(?total)) as ?result)
WHERE {
    {
        SELECT (count(distinct ?PROTEIN) as ?num_paralogy )
        WHERE {
            ?cluster a orth:ParalogsCluster.
            ?cluster orth:hasHomologousMember ?node1.
            ?cluster orth:hasHomologousMember ?node2.
            ?node2 orth:hasHomologousMember* ?PROTEIN.
            ?node1 orth:hasHomologousMember* ?IS_PARALOGOUS_TO_PROTEIN.
            ?PROTEIN a orth:Protein.
            ?PROTEIN orth:organism/obo:RO_0002162/up:scientificName ?species.
            ?IS_PARALOGOUS_TO_PROTEIN a orth:Protein.
            ?IS_PARALOGOUS_TO_PROTEIN orth:organism/obo:RO_0002162/up:scientificName ?species .
            values(?species ){( 'Drosophila melanogaster' )}
            filter(?node1 != ?node2)
        }
    }
    {
        SELECT (count(distinct ?protein_total) as ?total)
            WHERE {
               ?protein_total a orth:Protein .
               ?protein_total orth:organism/obo:RO_0002162/up:scientificName ?species .
               values(?species ){( 'Drosophila melanogaster' )}
            }
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","DISTINCT","IF","COUNT"],category:"sib-swiss OMA"},{name:"OMA - 12-orthologs-between-two-species",slug:"OMA_12-orthologs-between-two-species",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all orthologs between mouse and rabbit, together with their HOG id",context:null,inidces:[],query:`PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?MOUSE_PROTEIN ?RABIT_PROTEIN ?HOG
WHERE {
    ?HOG a orth:OrthologsCluster ;
        orth:hasHomologousMember ?node1 ;
        orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?RABIT_PROTEIN .
    ?node1 orth:hasHomologousMember* ?MOUSE_PROTEIN .
    ?MOUSE_PROTEIN a orth:Protein ;
        orth:organism/obo:RO_0002162/up:scientificName 'Mus musculus' .
    ?RABIT_PROTEIN a orth:Protein ;
        orth:organism/obo:RO_0002162/up:scientificName 'Oryctolagus cuniculus' .
    FILTER(?node1 != ?node2)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","IF"],category:"sib-swiss OMA"},{name:"OMA - 13-hog-members-at-level-from-query-protein",slug:"OMA_13-hog-members-at-level-from-query-protein",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve all proteins belongong to the Hierarchical Orthologous Group (HOG) at the level 'Vertebrata' to which humans' CDIN1 gene belong, together with their gene name symbol if available.",context:null,inidces:[],query:`PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?HOG ?MEMBER ?GENE_LABEL
WHERE {
    ?HOG a orth:OrthologsCluster ;
      orth:hasHomologousMember ?node1 ;
      orth:hasTaxonomicRange ?taxRange .
    ?taxRange orth:taxRange 'Vertebrata' .
    ?node1 orth:hasHomologousMember* ?query ;
      orth:hasHomologousMember* ?MEMBER .
    ?MEMBER a orth:Protein .
    OPTIONAL {
        ?MEMBER rdfs:label ?GENE_LABEL .
    }
    ?query a orth:Protein ;
      orth:organism/obo:RO_0002162/up:scientificName 'Homo sapiens';
      rdfs:label 'CDIN1'.
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","DISTINCT","IF"],category:"sib-swiss OMA"},{name:"OMA - 14-compare-mbgd-oma",slug:"OMA_14-compare-mbgd-oma",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve pairwise orthologous genes of the Cyanobacterium-aponinum psb27- gene that are found in the MBGD database but are not present in OMA",context:null,inidces:[],query:`PREFIX oma: <http://omabrowser.org/ontology/oma#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX sio: <http://semanticscience.org/resource/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX mbgd: <http://purl.jp/bio/11/mbgd#>

SELECT ?protein2 ?species WHERE {
  SERVICE <http://sparql.nibb.ac.jp/sparql> {
    SELECT ?protein2 ?species WHERE {
      ?cluster_mbgd a orth:OrthologsCluster ;
          orth:hasHomologous ?node1_mbgd ;
        orth:hasHomologous ?node2_mbgd .
      ?node1_mbgd orth:hasHomologous* ?gene1 .
      ?node2_mbgd orth:hasHomologous* ?gene2 .
      ?gene1 mbgd:uniprot <http://purl.uniprot.org/uniprot/K9Z723> .
      ?gene2 mbgd:uniprot ?protein2 ;
        mbgd:organism ?taxon .
      OPTIONAL {
        ?taxon mbgd:species ?species .
      }
      FILTER (?node1_mbgd != ?node2_mbgd)
    }
  }
  FILTER NOT EXISTS { # keep only those that do not exist in OMA
    ?cluster a orth:OrthologsCluster ;
      orth:hasHomologousMember ?node1 ;
      orth:hasHomologousMember ?node2 .
    ?node1 orth:hasHomologousMember* ?protein_OMA_1.
    ?node2 orth:hasHomologousMember* ?protein_OMA_2.
    ?protein_OMA_1 lscr:xrefUniprot <http://purl.uniprot.org/uniprot/K9Z723>.
    ?protein_OMA_2 lscr:xrefUniprot ?protein2.
    FILTER (?node1 != ?node2)
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","SERVICE","EXISTS","NOT EXISTS"],category:"sib-swiss OMA"},{name:"OMA - 15-biosodafrontend-rat-TP53",slug:"OMA_15-biosodafrontend-rat-TP53",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Rattus norvegicus' proteins encoded by genes that are paralogous to its TP53 gene and their Uniprot function annotations.",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>
SELECT DISTINCT ?PROTEIN ?IS_PARALOGOUS_TO_PROTEIN ?UNIPROT_XREF ?PARALOG_UNIPROT_XREF ?annotation_text WHERE {
	{
		?cluster a orth:ParalogsCluster .
		?cluster orth:hasHomologousMember ?node1 .
		?cluster orth:hasHomologousMember ?node2 .
		?node2 orth:hasHomologousMember* ?PROTEIN .
		?node1 orth:hasHomologousMember* ?IS_PARALOGOUS_TO_PROTEIN .
		?PROTEIN a orth:Protein .
		?PROTEIN orth:organism/obo:RO_0002162/up:scientificName 'Rattus norvegicus' ;
			rdfs:label 'TP53' ;
			lscr:xrefUniprot ?UNIPROT_XREF .
		?IS_PARALOGOUS_TO_PROTEIN a orth:Protein .
		?IS_PARALOGOUS_TO_PROTEIN orth:organism/obo:RO_0002162/up:scientificName 'Rattus norvegicus' .
		?IS_PARALOGOUS_TO_PROTEIN lscr:xrefUniprot ?PARALOG_UNIPROT_XREF .
	}
	SERVICE <https://sparql.uniprot.org/sparql> {
		?PARALOG_UNIPROT_XREF up:annotation ?annotation .
		?annotation a up:Function_Annotation .
		?annotation rdfs:comment ?annotation_text .
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","IF"],category:"sib-swiss OMA"}],pP=[{name:"OrthoDB - 1",slug:"OrthoDB_1",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all OrthoDB species with attached OrthoDB organisms containing \\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?tx a :Species; up:scientificName ?tax_name .
    ?org a ?tx; up:scientificName ?org_name.
    FILTER(strstarts (lcase(?tax_name ), "drosophila" ))
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","STR","IF","LCASE","STRSTARTS"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 10",slug:"OrthoDB_10",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find most conservative single copy Danio rerio genes at the Vertebrata level",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX : <http://purl.orthodb.org/>

SELECT ?og ?gene ?description ?evolrate
WHERE {
    ?gene a :Gene; :description ?description; up:organism/a [up:scientificName "Danio rerio"].
    ?gene :memberOf ?og.
    ?og :ogBuiltAt [up:scientificName "Vertebrata"].
    ?og a :OrthoGroup; :ogPercentSingleCopy 100; :ogEvolRate ?evolrate.
    FILTER(?evolrate < 1)
}
ORDER BY ?evolrate`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","COPY","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 11",slug:"OrthoDB_11",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find shortest rapidly evolving Caudovirales genes",context:null,inidces:[],query:`PREFIX : <http://purl.orthodb.org/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT *
WHERE {
?gene a :Gene; :description ?description.
?gene :geneTranslatedLength ?aa_length.
?gene :memberOf ?og.
?og :ogBuiltAt [up:scientificName "Caudovirales"].
?og a :OrthoGroup; :ogEvolRate ?evolrate.
FILTER(?evolrate > 1)
bind (?evolrate/?aa_length as ?x)
}
ORDER BY desc(?x) limit 99`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","ORDER BY","LIMIT","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 12",slug:"OrthoDB_12",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find orthologous human and mouse genes in the group(s) annotated with GO MF GO:0005164 (tumor necrosis factor receptor binding) and GO BP GO:0045739 (positive regulation of DNA repair)",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX GO: <http://purl.obolibrary.org/obo/GO_>
PREFIX : <http://purl.orthodb.org/>

SELECT ?og ?og_description ?gene_m ?gene_m_name ?gene_h ?gene_h_name
WHERE {
    ?gene_m a :Gene.
    ?gene_h a :Gene.
    ?gene_m up:organism/a [up:scientificName "Mus musculus"].
    ?gene_h up:organism/a taxon:9606.
    ?gene_m :name ?gene_m_name.
    ?gene_h :name ?gene_h_name.
    ?gene_m :memberOf ?og.
    ?gene_h :memberOf ?og.
    ?og a :OrthoGroup; :name ?og_description.
    ?og :xref [a :Xref; :xrefResource GO:0005164],[a :Xref; :xrefResource GO:0045739].
} ORDER BY ?og`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 13",slug:"OrthoDB_13",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all xrefs attached to the groups annotated with at least three Interpro domains (IPR011990, IPR013083 and IPR001841)",context:null,inidces:[],query:`PREFIX : <http://purl.orthodb.org/>
PREFIX interpro: <http://www.ebi.ac.uk/interpro/entry/>

SELECT ?og ?og_description (group_concat(distinct ?xref; SEPARATOR="; ") as ?xrefs)
WHERE {
    ?og a :OrthoGroup; :name ?og_description.
    ?og :xref [a :Xref; :xrefResource ?xref].
    ?og :xref [a :Xref; :xrefResource interpro:IPR011990].
    ?og :xref [a :Xref; :xrefResource interpro:IPR013083].
    ?og :xref [a :Xref; :xrefResource interpro:IPR001841].
} GROUP BY ?og ?og_description
ORDER BY ?og_description`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","CONCAT","GROUP_CONCAT"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 14",slug:"OrthoDB_14",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find genes with their properties by a list of their xrefs",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX ensembl: <http://rdf.ebi.ac.uk/resource/ensembl/>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX entrezgene: <http://www.ncbi.nlm.nih.gov/gene/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?gene a :Gene.
    ?gene :name ?gene_name; :description ?description; up:organism [up:scientificName ?org_name].
    ?gene :xref [a :Xref; :xrefResource ?xref]
    FILTER(?xref in (
    ensembl:ENSPTRG00000022217
    ,ensembl:ENSPVAG00000015405
    ,uniprotkb:Q9D4H7
    ,uniprotkb:H2PWL0
    ,entrezgene:105598395
    ,entrezgene:105062977
    ))
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 15",slug:"OrthoDB_15",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find orthologous group/s at any level containing protein with Uniprot id P12345",context:null,inidces:[],query:`PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?og a :OrthoGroup ;
        :ogBuiltAt ?level;
        :hasMember/rdfs:seeAlso uniprotkb:P12345 .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 16",slug:"OrthoDB_16",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find statistics on translated sequence length for genes matching both IPR002117 and GO:0008219",context:null,inidces:[],query:`PREFIX GO: <http://purl.obolibrary.org/obo/GO_>
PREFIX interpro: <http://www.ebi.ac.uk/interpro/entry/>
PREFIX : <http://purl.orthodb.org/>

SELECT
    (count(?gene) as ?count_genes)
    (min(?aa_seq_length) as ?min_aa_length)
    (avg(?aa_seq_length) as ?avg_aa_length)
    (max(?aa_seq_length) as ?max_aa_length)
WHERE {
    ?gene a :Gene; :geneTranslatedLength ?aa_seq_length.
    ?gene :xref [a :Xref; :xrefResource interpro:IPR002117].
    ?gene :xref [a :Xref; :xrefResource GO:0008219].
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","COUNT","MIN","MAX","AVG"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 17",slug:"OrthoDB_17",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find genes having Uniprot xrefs in the group 6400at314295, along with their names fetched from Uniprot SPARQL endpoint",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX odbgroup: <http://purl.orthodb.org/odbgroup/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?gene a :Gene;
        :memberOf odbgroup:6400at314295 .
    ?gene :xref [a :Xref; :xrefResource ?xref] .
    ?xref a :Uniprot .
    SERVICE <https://sparql.uniprot.org/sparql> {
        ?xref a up:Protein ;
            up:recommendedName [up:fullName ?name] .
    }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 18",slug:"OrthoDB_18",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find zebrafish orthologs of disease-implicated (according to Nextprot SPARQL endpoint) human genes via Vertebrata-level orthogroup/s annotated with both IPR000719 (protein kinase domain) and GO:0048013 (ephrin receptor signaling pathway)",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX GO: <http://purl.obolibrary.org/obo/GO_>
PREFIX : <http://purl.orthodb.org/>
PREFIX np: <http://nextprot.org/rdf#>
PREFIX interpro: <http://www.ebi.ac.uk/interpro/entry/>

SELECT ?gene_zf ?gene_zf_name ?gene ?gene_name ?description ?go ?disease
WHERE {
	SERVICE <https://sparql.nextprot.org/> {
        select distinct ?entry ?disease WHERE {
            ?entry np:isoform / np:disease / rdfs:comment ?disease
        }
    }
    ?gene rdfs:seeAlso ?entry;
        :name ?gene_name ;
        :description ?description.
    ?gene :memberOf ?og .
    ?og :xref/:xrefResource interpro:IPR000719 , GO:0048013.
    ?og :ogBuiltAt/up:scientificName "Vertebrata";
        :hasMember ?gene_zf.
    ?gene_zf :name ?gene_zf_name;
        up:organism/a [up:scientificName "Danio rerio"].
} ORDER BY ?disease`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","ORDER BY","DISTINCT","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 19",slug:"OrthoDB_19",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find distribution of oxidation reactions catalyzed by genes arranged into several Eukaryota-level orthologous groups annotated with IPR002328 across species in Alveolata, Stramenopiles, Liliopsida and Coleoptera via cooperation with both Uniprot SPARQL endpoint and RHEA SPARQL endpoint",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX : <http://purl.orthodb.org/>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX interpro: <http://www.ebi.ac.uk/interpro/entry/>

SELECT ?og ?equation
    (group_concat (distinct ?clade_name; SEPARATOR="; ") as ?clades)
    (count(distinct ?taxon) as ?cnt_taxons)
    (count(distinct ?gene) as ?cnt_genes)
    (group_concat (distinct ?org_name; SEPARATOR="; ") as ?taxons)
WHERE {
    ?taxon a :Species; up:scientificName ?org_name ;
        rdfs:subClassOf+/up:scientificName ?clade_name.
    ?gene a :Gene ;
        up:organism/a ?taxon.
    ?gene rdfs:seeAlso ?xref.
    ?xref a :Uniprot.
    ?gene :memberOf ?og.
    ?og :xref/:xrefResource interpro:IPR002328 ;
        :ogBuiltAt/up:scientificName "Eukaryota".

    # join via uniprot ?xref
    SERVICE <https://sparql.uniprot.org/sparql> {
        ?xref a up:Protein ;
            up:annotation/up:catalyticActivity/up:catalyzedReaction ?rh .
    }

    # join via reaction ?rh
    SERVICE <https://sparql.rhea-db.org/sparql> {
        ?rh rdfs:subClassOf rh:Reaction ;
            rh:equation ?equation.
    }

    # restrict taxons to several clades
    VALUES ?clade_name {
        "Stramenopiles"
        "Alveolata"
        "Liliopsida"
        "Coleoptera"
    }
} GROUP BY ?equation ?og ORDER BY ?og ?equation`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","SERVICE","GROUP BY","ORDER BY","DISTINCT","STR","IF","CONCAT","COUNT","GROUP_CONCAT"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 2",slug:"OrthoDB_2",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all clades, i.e. taxonomic levels where orthologous groups were built on, with their Latin names and ranks underneath Bacteria, sorted alphabetically by rank, then name",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX : <http://purl.orthodb.org/>

SELECT * WHERE {
    ?taxon a :Clade ;
        rdfs:subClassOf+ taxon:2 ;
        up:scientificName ?name ;
        up:rank ?rank .
} ORDER BY ?rank ?name`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 20",slug:"OrthoDB_20",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find single-copy phyletic profile at all orthologous levels of STRING-anotated well known interacting genes pertinent to mouse gene/s from the OrthoDB orthologous group annotated with GO MF GO:0005164 (tumor necrosis factor receptor binding) and GO BP GO:0045739 (positive regulation of DNA repair)",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX GO: <http://purl.obolibrary.org/obo/GO_>
PREFIX : <http://purl.orthodb.org/>
PREFIX ip_highest: <http://string-db.org/rdf/interaction/physical-highest-confidence-cutoff>

SELECT ?gene_m ?gene_m_name ?partner_m ?string_partner_genename ?level ?percent_single_copy
WHERE {
    ?gene_m a :Gene .
    ?gene_m up:organism/a [up:scientificName "Mus musculus"] .
    ?gene_m :name ?gene_m_name .
    ?gene_m :memberOf ?og .
    ?og a :OrthoGroup ;
        :name ?og_description ;
        :ogBuiltAt [up:scientificName "Mammalia"] .
    ?og :xref [a :Xref; :xrefResource GO:0005164], [a :Xref; :xrefResource GO:0045739] .
    ?gene_m :xref [a :Xref; :xrefResource ?xref_m] .
    ?xref_m a :Entrezgene.

    SERVICE <https://sparql.string-db.org/sparql> {
        ?string_gene_m rdfs:seeAlso ?xref_m ;
            ip_highest: ?partner_m .
        ?partner_m rdfs:label ?string_partner_genename ;
            rdfs:comment ?string_partner_description .
    }

    ?gene_m2 a :Gene ;
        up:organism/a [up:scientificName "Mus musculus"] ;
        :name ?string_partner_genename ;
        :memberOf ?og2 .
    ?og2 a :OrthoGroup ;
        :ogBuiltAt/up:scientificName ?level ;
        :ogPercentSingleCopy ?percent_single_copy .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","COPY","STR","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 3",slug:"OrthoDB_3",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find LCA in the OrthoDB tree for fruit fly and honey bee taxons",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX : <http://purl.orthodb.org/>

SELECT DISTINCT *
WHERE {
    ?lca a :Clade ; up:scientificName ?lcaname .
    taxon:7227  rdfs:subClassOf* ?lca .
    taxon:7460 rdfs:subClassOf* ?lca .
    FILTER(not exists {
        ?xca a :Clade ; rdfs:subClassOf ?lca .
        taxon:7227    rdfs:subClassOf* ?xca .
        taxon:7460   rdfs:subClassOf* ?xca .
    })
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","IF","EXISTS","NOT EXISTS"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 4",slug:"OrthoDB_4",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find URI and some properties of the orthologous group 6400at314295 by its short text label",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?og a :OrthoGroup;
        rdfs:label "6400at314295";
        :name ?description;
        :ogBuiltAt [up:scientificName ?clade] .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 5",slug:"OrthoDB_5",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all properties of the orthologous group 6400at314295 by its URI",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX odbgroup: <http://purl.orthodb.org/odbgroup/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?og a :OrthoGroup;
        rdfs:label ?label;
        :name ?description;
        :ogBuiltAt [up:scientificName ?clade];
        :ogEvolRate ?evolRate;
        :ogPercentSingleCopy ?percentSingleCopy;
        :ogPercentInSpecies ?percentInSpecies;
        :ogTotalGenesCount ?totalGenesCount;
        :ogMultiCopyGenesCount ?multiCopyGenesCount;
        :ogSingleCopyGenesCount ?singleCopyGenesCount;
        :ogInSpeciesCount ?inSpeciesCount;
        :cladeTotalSpeciesCount ?cladeTotalSpeciesCount .
    OPTIONAL { ?og :ogMedianProteinLength ?medianProteinLength}
    OPTIONAL { ?og :ogStddevProteinLength ?stddevProteinLength}
    OPTIONAL { ?og :ogMedianExonsCount ?medianExonsCount}
    OPTIONAL { ?og :ogStddevExonsCount ?stddevExonsCount}
    FILTER(?og = odbgroup:6400at314295)
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","COPY","IF","COUNT"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 6",slug:"OrthoDB_6",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all genes with their name and description in the orthologous group 6400at314295",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX odbgroup: <http://purl.orthodb.org/odbgroup/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?gene a :Gene.
    ?gene :name ?gene_name.
    ?gene :description ?description.
    ?gene up:organism/a ?taxon.
    ?taxon up:scientificName ?org_name.
    ?gene :memberOf odbgroup:6400at314295.
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","IF"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 7",slug:"OrthoDB_7",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find A.thaliana genes with their properties by a list of their gene symbols/names",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?gene a :Gene;
        up:organism [up:scientificName ?sciname] .
    FILTER(strstarts (?sciname, "Arabidopsis thaliana" ))
    ?gene :name ?gene_name; :description ?description; :geneTranslatedLength ?aa_seq_length.
    OPTIONAL {?gene :geneNbExons ?geneNbExons}.
    ?gene :aaSequence ?sequence
    FILTER(?gene_name in(
        "CHX10"
        ,"GPX1"
        ,"MAG"
        ,"MSRB6"
        ,"MYH"
        ,"POLH"
        ,"RPA3A"
        ,"UVR3"
        ,"XRCC1"
    ))
} ORDER BY ?aa_seq_length`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","ORDER BY","STR","IF","STRSTARTS"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 8",slug:"OrthoDB_8",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find in any organsim under Gammaproteobacteria all genes containing word \\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?tx rdfs:subClassOf+/up:scientificName "Gammaproteobacteria".
    ?gene a :Gene; up:organism/a ?tx; :description ?description.
    FILTER(contains (lcase(?description), "protease" ))
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","IF","LCASE","CONTAINS"],category:"sib-swiss OrthoDB"},{name:"OrthoDB - 9",slug:"OrthoDB_9",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"At the LCA level, find all human orthologs of mouse genes with name containing \\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX : <http://purl.orthodb.org/>

SELECT *
WHERE {
    ?og a :OrthoGroup.
    ?og :ogBuiltAt taxon:314146.
    ?gene_m a :Gene.
    ?gene_h a :Gene.
    ?gene_m up:organism/a taxon:10090.
    ?gene_h up:organism/a taxon:9606.
    ?gene_m :memberOf ?og.
    ?gene_h :memberOf ?og.
    ?gene_m :name ?gene_m_name.
    ?gene_h :name ?gene_h_name.
    FILTER(contains (UCASE(?gene_m_name), "MAPK" ))
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UCASE","CONTAINS"],category:"sib-swiss OrthoDB"}],hP=[{name:"Rhea - 1",slug:"Rhea_1",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 1
# Select all Rhea reactions (unspecified direction) 
# and return identifier (id), accession, boolean attributes (isChemicallyBalanced, isTransport) and chemical equation.
#
SELECT ?rhea ?id ?accession ?isChemicallyBalanced ?isTransport  ?equation 
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:id ?id .
  ?rhea rh:accession ?accession .
  ?rhea rh:equation ?equation .
  ?rhea rh:isTransport ?isTransport .
  ?rhea rh:isChemicallyBalanced ?isChemicallyBalanced .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ALL","STR","IF"],category:"sib-swiss Rhea"},{name:"Rhea - 10",slug:"Rhea_10",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions linked to an enzyme classification (sub)-class",context:null,inidces:[],query:`PREFIX ec: <http://purl.uniprot.org/enzyme/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 10
# Select all Rhea reactions mapped to 
# 
# This query corresponds to the Rhea website query:
# https://www.rhea-db.org/rhea?query=ec:1.*
# https://www.rhea-db.org/rhea?query=ec:1.1.*
# https://www.rhea-db.org/rhea?query=ec:1.1.1.*
#
SELECT ?ec ?ecNumber ?rhea ?accession ?equation
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:accession ?accession .
  ?rhea rh:ec ?ec .
  BIND(strafter(str(?ec),str(ec:)) as ?ecNumber)
  # class (e.g EC 1.-.-.-)
  FILTER (regex(?ecNumber,'^1\\\\\\\\.')) 
  # sub-class  (e.g EC 1.1.-.-)
  #FILTER (regex(?ecNumber,'^1\\\\\\\\.1\\\\\\\\.')) 
  # sub-sub-class (e.g EC 1.1.1.-)
  #FILTER (regex(?ecNumber,'^1\\\\\\\\.1\\\\\\\\.1\\\\\\\\.')) 
  ?rhea rh:equation ?equation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","ALL","STR","REGEX"],category:"sib-swiss Rhea"},{name:"Rhea - 100 Get the number of ChEBI participant with at least one xref to CAS number as provided by ChEBI",slug:"Rhea_100_Get_the_number_of_ChEBI_participant_with_at_least_one_xref_to_CAS_number_as_provided_by_ChEBI",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get the number of ChEBI participant with at least one xref to CAS number as provided by ChEBI",context:null,inidces:[],query:`PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX allie: <http://allie.dbcls.jp/>
PREFIX GO: <http://purl.obolibrary.org/obo/GO_>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX chebihash:<http://purl.obolibrary.org/obo/chebi#>
PREFIX oboInOwl:<http://www.geneontology.org/formats/oboInOwl#>

SELECT
  (count(distinct ?chebi) as ?chebiCount)
  (count(distinct ?dbXref) as ?xrefCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  {
    ?compound rh:chebi ?chebi .
    ?chebi oboInOwl:hasDbXref ?dbXref .
    FILTER (regex(?dbXref,'^CAS:'))
  }
  UNION
  {
    ?compound rh:chebi ?chebi .
    ?chebi2 rdfs:subClassOf ?chebiRestriction .
    ?chebiRestriction a owl:Restriction .
    ?chebiRestriction owl:onProperty chebihash:has_major_microspecies_at_pH_7_3 .
    ?chebiRestriction owl:someValuesFrom ?chebi .
    ?chebi2 oboInOwl:hasDbXref ?dbXref .
    FILTER (regex(?dbXref,'^CAS:'))
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","VALUES","DISTINCT","FROM","ALL","STR","REGEX","NOW","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 101 Pairs of reaction participants that belong to the same reaction but on distinct sides and number of reactions in which they occur",slug:"Rhea_101_Pairs_of_reaction_participants_that_belong_to_the_same_reaction_but_on_distinct_sides_and_number_of_reactions_in_which_they_occur",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Pairs of reaction participants that belong to the same reaction but on distinct sides and number of reactions in which they occur",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?chebi1
  ?compound1Name

  ?chebi2
  ?compound2Name

  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .

  ?reaction rh:side ?reactionSide1 .
  ?reactionSide1 rh:contains ?participant1 .
  ?participant1 rh:compound ?compound1 .
  ?compound1 rh:name ?compound1Name .
  ?compound1 rh:chebi ?chebi1 .

  ?reaction rh:side ?reactionSide2 .
  ?reactionSide2 rh:contains ?participant2 .
  ?participant2 rh:compound ?compound2 .
  ?compound2 rh:name ?compound2Name .
  ?compound2 rh:chebi ?chebi2 .


  ?reactionSide1 rh:transformableTo ?reactionSide2

  FILTER(?chebi1<?chebi2)
}
GROUP BY ?chebi1 ?chebi2 ?compound1Name ?compound2Name
ORDER BY DESC (?reactionCount)
LIMIT 100`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","ORDER BY","LIMIT","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 103 Select all approved reactions linked to a given EC number",slug:"Rhea_103_Select_all_approved_reactions_linked_to_a_given_EC_number",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions linked to a given EC number (EC 1.1.1.353)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>

SELECT
  ?reaction
  ?reactionEquation
WHERE {
  BIND(ec:1.1.1.353 AS ?ecNumber)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:ec ?ecNumber
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 104 Select the number of reactions linked to EC numbers",slug:"Rhea_104_Select_the_number_of_reactions_linked_to_EC_numbers",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of reactions linked to EC numbers",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:ec ?ecNumber
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 105 Select all reactions linked to EC numbers",slug:"Rhea_105_Select_all_reactions_linked_to_EC_numbers",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions linked to EC numbers",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?reaction
  ?ecNumber
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:ec ?ecNumber
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 106 Select all reactions linked to more than one EC number",slug:"Rhea_106_Select_all_reactions_linked_to_more_than_one_EC_number",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions linked to more than one EC number",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?reaction
  (count(?ecNumber) as ?ecCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:ec ?ecNumber
}
GROUP BY ?reaction
HAVING (count(?ecNumber) > 1)
ORDER BY DESC(?ecCount)`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","HAVING","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 107 Select the number of reactions with no links to EC numbers",slug:"Rhea_107_Select_the_number_of_reactions_with_no_links_to_EC_numbers",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of reactions with no links to EC numbers",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  FILTER (NOT EXISTS {?reaction rh:ec ?ecNumber .})
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","EXISTS","NOT EXISTS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 108 Select all reactions with no links to EC numbers",slug:"Rhea_108_Select_all_reactions_with_no_links_to_EC_numbers",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions with no links to EC numbers",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT ?reaction WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  FILTER (NOT EXISTS {?reaction rh:ec ?ecNumber .})
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","EXISTS","NOT EXISTS"],category:"sib-swiss Rhea"},{name:"Rhea - 109 Distribution of reactions according to the first level of enzyme classification",slug:"Rhea_109_Distribution_of_reactions_according_to_the_first_level_of_enzyme_classification",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos:<http://www.w3.org/2004/02/skos/core#>

SELECT
  ?ecClass
  (str(?ecName) as ?ecClassName)
  (count(?reaction) as ?reactionCount)
WHERE
{
  SERVICE <http://sparql.uniprot.org/sparql> {
    VALUES (?ecClass) { (ec:1.-.-.-)(ec:2.-.-.-)(ec:3.-.-.-)(ec:4.-.-.-)(ec:5.-.-.-) (ec:6.-.-.-) (ec:7.-.-.-) }
    ?ecNumber rdfs:subClassOf ?ecClass .
    ?ecClass skos:prefLabel ?ecName .
  }
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:ec ?ecNumber .
}
GROUP BY ?ecClass ?ecName`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","VALUES","SERVICE","GROUP BY","STR","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 11",slug:"Rhea_11",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Distribution of reactions according to the first class of the enzyme classification (federated query)",context:null,inidces:[],query:`PREFIX ec: <http://purl.uniprot.org/enzyme/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>

# Query 11
# Retrieve the count of reactions mapped to each level (main class) of the enzyme classification
#
# This query mimics the Filter section of the Rhea website (Browse all reactions)
# https://www.rhea-db.org/rhea?query=
SELECT ?ecClass (STR(?ecName) AS ?ecClassName) (COUNT(?rhea) AS ?rheaCount)
WHERE {
  SERVICE <https://sparql.uniprot.org/sparql> {
    VALUES (?ecClass) { (ec:1.-.-.-)(ec:2.-.-.-)(ec:3.-.-.-)(ec:4.-.-.-)(ec:5.-.-.-) (ec:6.-.-.-) (ec:7.-.-.-)}
    ?ecNumber rdfs:subClassOf ?ecClass .
    ?ecClass skos:prefLabel ?ecName .
  }
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:ec ?ecNumber .
} GROUP BY ?ecClass ?ecName`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","SERVICE","GROUP BY","ALL","STR","IF","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 110 Select reaction ec protein for a given reaction",slug:"Rhea_110_Select_reaction_ec_protein_for_a_given_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select <reaction> <ec> <protein> for a given reaction",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX up:<http://purl.uniprot.org/core/>

SELECT
  ?reaction
  ?enzyme
  ?protein
WHERE {
  BIND(rh:11672 AS ?reaction)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:ec ?enzyme
  SERVICE <http://sparql.uniprot.org/sparql> {
    ?protein up:reviewed true .
    OPTIONAL {?protein up:enzyme ?enzyme. } .
    OPTIONAL {?protein up:domain/up:enzyme ?enzyme . } .
    OPTIONAL {?protein up:component/up:enzyme ?enzyme .} .
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","BIND","SERVICE"],category:"sib-swiss Rhea"},{name:"Rhea - 111 Select reaction ec protein for a given EC number",slug:"Rhea_111_Select_reaction_ec_protein_for_a_given_EC_number",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select <reaction> <ec> <protein> for a given EC number",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX up:<http://purl.uniprot.org/core/>

SELECT
  ?reaction
  ?enzyme
  ?protein
WHERE {
  BIND(ec:2.6.1.62 AS ?enzyme)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:ec ?enzyme .
  SERVICE <http://sparql.uniprot.org/sparql> {
    ?protein up:reviewed true .
    OPTIONAL {?protein up:enzyme ?enzyme. } .
    OPTIONAL {?protein up:domain/up:enzyme ?enzyme . } .
    OPTIONAL {?protein up:component/up:enzyme ?enzyme .} .
  }

}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","BIND","SERVICE"],category:"sib-swiss Rhea"},{name:"Rhea - 112 Total number of reaction ec protein links",slug:"Rhea_112_Total_number_of_reaction_ec_protein_links",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Total number of <reaction> <ec> <protein> links",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX up:<http://purl.uniprot.org/core/>

SELECT
  (count(?reaction) as ?reactionEcProteinLinkCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:ec ?enzyme
  SERVICE <http://sparql.uniprot.org/sparql> {
    ?protein up:reviewed true .
    ?protein up:enzyme ?enzyme.
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","SERVICE","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 113 Number of distinct rhea EC number UniProtKB Swiss-Prot entries involved in links",slug:"Rhea_113_Number_of_distinct_rhea_EC_number_UniProtKB_Swiss-Prot_entries_involved_in_links",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Number of distinct rhea, EC number, UniProtKB/Swiss-Prot entries involved in (<reaction> <ec> <protein>) links",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX up:<http://purl.uniprot.org/core/>

SELECT
  (count(distinct ?reaction) as ?reactionCount)
  (count(distinct ?enzyme) as ?enzymeCount)
  (count(distinct ?protein) as ?proteinCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:ec ?enzyme
  SERVICE <http://sparql.uniprot.org/sparql> {
    ?protein up:reviewed true .
    ?protein up:enzyme ?enzyme.
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 114 Retrieve the set of reactions catalyzed by enzymes of the ECOLI reference proteome reaction EC protein",slug:"Rhea_114_Retrieve_the_set_of_reactions_catalyzed_by_enzymes_of_the_ECOLI_reference_proteome_reaction_EC_protein",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve the set of reactions catalyzed by enzymes of the ECOLI reference proteome (<reaction> - <EC> - <protein>)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX up:<http://purl.uniprot.org/core/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>
PREFIX keywords:<http://purl.uniprot.org/keywords/>

SELECT ?protein ?ecNumber ?reaction where {

  SERVICE <https://sparql.uniprot.org/sparql> {
    ?protein a up:Protein .
    ?protein up:reviewed true .
    ?protein up:organism taxon:83333 .
    ?protein up:classifiedWith keywords:1185 .
      {?protein up:enzyme ?ecNumber}
    UNION
      {?protein up:domain/up:enzyme ?ecNumber}
    UNION
    {?protein up:component/up:enzyme ?ecNumber} .
  }
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:ec ?ecNumber .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","SERVICE","WITH","IF"],category:"sib-swiss Rhea"},{name:"Rhea - 115 Retrieve the set of metabolites involved in reactions catalyzed by human genes HUMAN reference proteome",slug:"Rhea_115_Retrieve_the_set_of_metabolites_involved_in_reactions_catalyzed_by_human_genes_HUMAN_reference_proteome",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>
PREFIX up:<http://purl.uniprot.org/core/>
PREFIX keywords:<http://purl.uniprot.org/keywords/>

SELECT
  ?chebi
  ?reaction
  ?ecNumber
  ?protein
  ?ensemblTranscript
  ?ensemblGene
WHERE {
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?protein a up:Protein .
    ?protein up:reviewed true .
    ?protein up:organism taxon:9606 .
    ?protein up:classifiedWith keywords:1185 .

    ?protein rdfs:seeAlso ?ensemblTranscript .
    ?ensemblTranscript a up:Transcript_Resource .
    ?ensemblTranscript up:database <http://purl.uniprot.org/database/Ensembl> .
    ?ensemblTranscript up:transcribedFrom ?ensemblGene .

      {?protein up:enzyme ?ecNumber .}
    UNION
      {?protein up:domain/up:enzyme ?ecNumber .}
    UNION
      {?protein up:component/up:enzyme ?ecNumber .}
  }

  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:ec ?ecNumber .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi . # only considering small molecules participants
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","UNION","SERVICE","FROM","WITH","ALL","IF","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 116 Retrieve the set of reactions catalyzed by human enzymes reactions EC proteins from HUMAN reference proteome",slug:"Rhea_116_Retrieve_the_set_of_reactions_catalyzed_by_human_enzymes_reactions_EC_proteins_from_HUMAN_reference_proteome",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve the set of reactions catalyzed by human enzymes (reactions - EC - proteins from HUMAN reference proteome)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX up:<http://purl.uniprot.org/core/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>
PREFIX keywords:<http://purl.uniprot.org/keywords/>

SELECT
  ?protein
  ?ecNumber
  ?reaction
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:ec ?ecNumber .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  # ?compound rh:chebi CHEBI:57970 .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:18059 .


  SERVICE <https://sparql.uniprot.org/sparql> {
    ?protein a up:Protein ;
    up:reviewed true ;
    up:organism taxon:83333 ;
    up:classifiedWith keywords:1185 .
      {?protein up:enzyme ?ecNumber}
    UNION
      {?protein up:domain/up:enzyme ?ecNumber}
    UNION
      {?protein up:component/up:enzyme ?ecNumber} .
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","SERVICE","WITH","IF","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 117 Retrieve parent reactions and their catalyzing enzymes for a metabolite that is not directly involved in a reaction",slug:"Rhea_117_Retrieve_parent_reactions_and_their_catalyzing_enzymes_for_a_metabolite_that_is_not_directly_involved_in_a_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve parent reactions (and their catalyzing enzymes) for a metabolite that is not directly involved in a reaction (ex: CHEBI:83137).",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX up:<http://purl.uniprot.org/core/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>
PREFIX keywords:<http://purl.uniprot.org/keywords/>

SELECT
  ?chebi
  ?ecNumber
  (count(?protein) as ?proteinCount)
  ?reaction
  ?equation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?equation .
  ?reaction rh:ec ?ecNumber .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  CHEBI:83137 rdfs:subClassOf+ ?chebi.

  SERVICE <https://sparql.uniprot.org/sparql> {
  ?protein a up:Protein .
  ?protein up:reviewed true .
    {?protein up:enzyme ?ecNumber}
  UNION
    {?protein up:domain/up:enzyme ?ecNumber}
  UNION
    {?protein up:component/up:enzyme ?ecNumber} .
  }
} 
GROUP BY ?chebi ?ecNumber ?reaction ?equation`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","SERVICE","GROUP BY","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 118 Number of approved reactions missing citations",slug:"Rhea_118_Number_of_approved_reactions_missing_citations",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Number of approved reactions missing citations",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
SELECT
  (count(?reaction) as ?eactionMissingCitationCount)
WHERE
{
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  OPTIONAL {?reaction rh:citation ?citation .}
  FILTER (NOT EXISTS {?reaction rh:citation ?citation .})
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","EXISTS","NOT EXISTS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 119 Give me the set of approved reactions missing citations",slug:"Rhea_119_Give_me_the_set_of_approved_reactions_missing_citations",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Give me the set of approved reactions missing citations",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  ?reaction
WHERE
{
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  OPTIONAL {?reaction rh:citation ?citation .}
  FILTER (NOT EXISTS {?reaction rh:citation ?citation .} )
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","ORDER BY","EXISTS","NOT EXISTS"],category:"sib-swiss Rhea"},{name:"Rhea - 12",slug:"Rhea_12",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions linked to protein sequences (enzymes and transporters) in UniProtKB",context:null,inidces:[],query:`PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

# Query 12
# Select all Rhea reactions used to annotate enzyme sequences in UniProtKB
# return the number of UniProtKB entries
# Federated query using a service to UniProt SPARQL endpoint
#
# This query corresponds to the Rhea website query:
# https://www.rhea-db.org/rhea?query=uniprot:*
#
SELECT ?uniprotCount ?rhea ?accession ?equation 
WHERE {
  SERVICE <https://sparql.uniprot.org/sparql> { 
  	SELECT ?rhea (count(?uniprot) as ?uniprotCount) {
      ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea . 
  	}
  	GROUP BY ?rhea
  }
  ?rhea rh:accession ?accession .
  ?rhea rh:equation ?equation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","GROUP BY","USING","ALL","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 126 list protein components ec",slug:"Rhea_126_list_protein_components_ec",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX up:<http://purl.uniprot.org/core/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>
SELECT
  distinct
    ?enzyme
WHERE
{
  ?protein up:reviewed true .
  ?protein up:mnemonic ?mnemonic .
  ?protein up:domain ?domain .
  ?domain up:enzyme ?enzyme .
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","DISTINCT"],category:"sib-swiss Rhea"},{name:"Rhea - 128 list EC numbers not linked to UniProt entries",slug:"Rhea_128_list_EC_numbers_not_linked_to_UniProt_entries",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX up:<http://purl.uniprot.org/core/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>

SELECT
  (count(distinct ?ecNumber) as ?ecCount)
WHERE
{
  ?ecNumber rdfs:subClassOf up:Enzyme .
  FILTER (NOT EXISTS {?ecNumber up:obsolete true .} )
  FILTER (!regex(?ecNumber,'-')) .

  FILTER (NOT EXISTS {
    ?protein up:enzyme ?ecNumber .
    ?protein up:reviewed true .
  })
  FILTER (NOT EXISTS {
    ?protein up:domain/up:enzyme ?ecNumber .
    ?protein up:reviewed true .
  })
  FILTER (NOT EXISTS {
    ?protein up:reviewed true .
    ?protein up:component/up:enzyme ?ecNumber .
  })
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","FILTER","DISTINCT","REGEX","EXISTS","NOT EXISTS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 13",slug:"Rhea_13",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions used in UniProtKB/Swiss-Prot for a given organism (NCBI taxonomy ID).",context:null,inidces:[],query:`PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

# Query 13
# Select all Rhea reactions used to annotate Escherichia coli (taxid=83333) in UniProtKB/Swiss-Prot
# return the number of UniProtKB entries
# 
# Federated query using a service to UniProt SPARQL endpoint
#
# This query cannot be performed using the Rhea search website
SELECT ?uniprot ?mnemo ?rhea ?accession ?equation 
WHERE {
  SERVICE <https://sparql.uniprot.org/sparql> { 
    VALUES (?taxid) { (taxon:83333) }
    GRAPH <http://sparql.uniprot.org/uniprot> {
      ?uniprot up:reviewed true . 
      ?uniprot up:mnemonic ?mnemo . 
      ?uniprot up:organism ?taxid .
      ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea . 
    }
  }
  ?rhea rh:accession ?accession .
  ?rhea rh:equation ?equation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","VALUES","SERVICE","USING","ALL"],category:"sib-swiss Rhea"},{name:"Rhea - 130 Where are the human genes encoding enzymes metabolizing cholesterol expressed",slug:"Rhea_130_Where_are_the_human_genes_encoding_enzymes_metabolizing_cholesterol_expressed",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Where are the human genes encoding enzymes metabolizing cholesterol expressed?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX : <http://purl.orthodb.org/>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX odbgroup:<http://purl.orthodb.org/odbgroup/>
SELECT *
WHERE {
  SERVICE <https://sparql.orthodb.org/sparql/> {
    ?gene a :Gene; :memberOf odbgroup:2906at28890.
    ?gene :xref [a :Xref; :xrefResource ?xref].
    ?xref a :Uniprot.
  }
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?xref a up:Protein; up:recommendedName [up:fullName ?name] ;
    up:annotation/up:catalyticActivity/up:catalyzedReaction ?reaction .
  }
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:directionalReaction ?directionalReaction .
  OPTIONAL { ?directionalReaction rdfs:seeAlso ?xref2 . }
  ?reaction rh:bidirectionalReaction ?bidirectionalReaction .
  OPTIONAL { ?bidirectionalReaction rdfs:seeAlso ?xref2 . }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","SERVICE"],category:"sib-swiss Rhea"},{name:"Rhea - 131 Where are the human genes encoding enzymes metabolizing cholesterol expressed",slug:"Rhea_131_Where_are_the_human_genes_encoding_enzymes_metabolizing_cholesterol_expressed",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Where are the human genes encoding enzymes metabolizing cholesterol expressed?",context:null,inidces:[],query:`PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX up:<http://purl.uniprot.org/core/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>

SELECT
  distinct
    ?protein
    ?ensemblGene
    ?reaction
    ?anatomicEntityLabel
    ?anatomicEntity
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
      ?reaction rh:status rh:Approved .
      ?reaction rh:equation ?reactionEquation .
      ?reaction rh:side ?reactionSide .
      ?reactionSide rh:contains ?participant .
      ?participant rh:compound ?compound .
      # compound constraint (CHEBI:16113 == cholesterol)
      ?compound rh:chebi CHEBI:16113 .
  SERVICE <https://sparql.uniprot.org/sparql> {
    # taxonomy constraint (taxon:9606 == Homo sapiens)
    ?protein up:organism taxon:9606 .
    ?protein up:annotation ?a .
    ?a a up:Catalytic_Activity_Annotation .
    ?a up:catalyticActivity ?ca .
    ?ca up:catalyzedReaction ?reaction .
    ?protein rdfs:seeAlso / up:transcribedFrom ?ensemblGene .
  }
  # federated query to Bgee (expression data)
  SERVICE <https://www.bgee.org/sparql/> {
    ?gene genex:isExpressedIn ?anatomicEntity .
    ?gene lscr:xrefEnsemblGene ?ensemblGene .
    ?anatomicEntity rdfs:label ?anatomicEntityLabel .
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","FROM","STR","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 14",slug:"Rhea_14",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions that have a given ChEBI ID as reaction participant",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 14
# Select all Rhea reactions that have CHEBI:29985 (L-glutamate) as reaction participant
# 
# This query corresponds to the Rhea website query:
# https://www.rhea-db.org/rhea?query=chebi:29985
SELECT distinct ?chebi ?rhea ?equation
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:equation ?equation .
  ?rhea rh:side/rh:contains/rh:compound ?compound .
  #
  # the ChEBI can be used either as a small molecule, the reactive part of a macromolecule or as a polymer.
  #
  ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  VALUES (?chebi) { (CHEBI:29985) }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","DISTINCT","ALL","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 15",slug:"Rhea_15",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all ChEBI compounds used in Rhea as reaction participant",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

# Query 15
# Select all ChEBI compounds used in Rhea as reaction participant
# 
# This query can not be expressed in the Rhea website
SELECT ?chebi ?name (count(?rhea) as ?countRhea)
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:side/rh:contains/rh:compound ?compound .
  #
  # the ChEBI can be used either as a small molecule, the reactive part of a macromolecule or as a polymer.
  #
  ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  ?chebi up:name ?name .
}
GROUP BY ?chebi ?name
ORDER BY DESC(?countRhea)`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","ALL","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 16",slug:"Rhea_16",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions that have a pair of ChEBI IDs as reaction participant and in opposite side",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

# Query 16
# Select all Rhea reactions that have a pair of ChEBI IDs as reaction participant and in opposite side
# Return Rhea reactions that have CHEBI:29985 (L-glutamate) as reaction participant in one side
# and CHEBI:58359 (L-glutamine) in the other side
#
# This query cannot be expressed in the Rhea website
SELECT ?chebi1 ?name1 ?chebi2 ?name2 ?rhea ?equation
WHERE {
  VALUES (?chebi1) { (CHEBI:29985) }
  ?chebi1 up:name ?name1 .
  ?rhea rh:side ?reactionSide1 .
  ?reactionSide1  rh:contains / rh:compound / rh:chebi ?chebi1 .

  VALUES (?chebi2) { (CHEBI:58359) }
  ?chebi2 up:name ?name2 .

  ?rhea rh:side ?reactionSide2 .
  ?reactionSide2  rh:contains / rh:compound / rh:chebi ?chebi2 .
  
  ?reactionSide1 rh:transformableTo ?reactionSide2 .
  
  ?rhea rh:equation ?equation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ALL","CONTAINS","MIN"],category:"sib-swiss Rhea"},{name:"Rhea - 17",slug:"Rhea_17",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions that involve a lipid, i.e. children of CHEBI:18059 in the ChEBI hierarchy.",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX chebihash: <http://purl.obolibrary.org/obo/chebi#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

# Query 17
# Select all Rhea reactions that involve a lipid, i.e. children of CHEBI:18059 in the ChEBI hierarchy.
# 
# This query corresponds to the Rhea website query:
# https://www.rhea-db.org/rhea?query=chebi:18059
#
SELECT distinct ?chebi ?name ?rhea ?equation
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:equation ?equation .
  ?rhea rh:side/rh:contains/rh:compound ?compound .
  #
  # the ChEBI can be used either as a small molecule, the reactive part of a macromolecule or as a polymer.
  #
  { 
    ?chebi rdfs:subClassOf* CHEBI:18059 . # lipid
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  }
  UNION 
  { # add non-pH 7.3 species
    ?not7_3 rdfs:subClassOf* CHEBI:18059 . # lipid
    ?not7_3 rdfs:subClassOf ?chebiRestriction .
    ?chebiRestriction a owl:Restriction .
    ?chebiRestriction owl:onProperty chebihash:has_major_microspecies_at_pH_7_3 .
    ?chebiRestriction owl:someValuesFrom ?chebi .
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  }
  ?chebi up:name ?name .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","VALUES","DISTINCT","FROM","ADD","ALL","STR","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 18",slug:"Rhea_18",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Use <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>


SELECT 
  ?rhea 
  ?chebi
WHERE {
  SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/chebi> {
    ?chebi sachem:substructureSearch [
        sachem:query "[C@]12(CCC3CCCC[C@]3(C)[C@@]1([H])CC[C@]1(C)[C@@]([H])([C@@](C)([H])CCCC(C)C)CC[C@@]21[H])[H]" ].
  }
  ?rhea rh:side/rh:contains/rh:compound/rdfs:subClassOf ?chebi .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","STR","SUBSTR","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 19 draft human metabolome",slug:"Rhea_19_draft_human_metabolome",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Generate a draft human metabolome",context:null,inidces:[],query:`PREFIX chebislash: <http://purl.obolibrary.org/obo/chebi/>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?uniprot ?mnemonic ?rhea ?chebi ?smiles ?inchiKey
WHERE
{
  ?rhea rh:side/rh:contains/rh:compound ?compound .
  ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  ?chebi chebislash:smiles ?smiles ;
          chebislash:inchikey ?inchiKey .
  SERVICE <https://sparql.uniprot.org/sparql> {
     ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea ;
                                             up:organism taxon:9606 ;
                                             up:mnemonic ?mnemonic .

   }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 2",slug:"Rhea_2",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions annotated with a given Pubmed ID",context:null,inidces:[],query:`PREFIX pubmed: <http://rdf.ncbi.nlm.nih.gov/pubmed/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 2 
# Select all Rhea reactions annotated with a given Pubmed identifier (PMID = 29867142)
#
SELECT ?pubmed ?rhea ?accession ?isTransport  ?equation 
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:accession ?accession .
  ?rhea rh:citation ?pubmed .
  VALUES (?pubmed) { (pubmed:29867142) }
  ?rhea rh:isTransport ?isTransport .
  ?rhea rh:equation ?equation .
} ORDER BY ?rhea`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY","WITH","ALL","STR","IF"],category:"sib-swiss Rhea"},{name:"Rhea - 20 search chemical names in japanese",slug:"Rhea_20_search_chemical_names_in_japanese",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Search for Chemical names starting in Japanese, translated to english using allie",context:null,inidces:[],query:`PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT
*
WHERE {
SERVICE <https://data.allie.dbcls.jp/sparql>{
        ?x rdfs:label "1,2,4-トリクロロベンゼン"@ja ;
            rdfs:label ?englishLabel .
        FILTER(lang(?englishLabel) = "en")
    }
  BIND(STR(?englishLabel) AS ?englishLabelStr)
  ?chebi rdfs:label|oboInOwl:hasSynonym ?englishLabelStr .
}`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","SERVICE","ALL","STR","LANG","NOW"],category:"sib-swiss Rhea"},{name:"Rhea - 21 count rhea examples",slug:"Rhea_21_count_rhea_examples",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"SELECT the number of reactions in Rhea",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT (count(?reaction) as ?reactionCount) WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
}`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 22 Select all reactions and their equation ordered by reaction identifier",slug:"Rhea_22_Select_all_reactions_and_their_equation_ordered_by_reaction_identifier",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions and their equation, ordered by reaction identifier",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?reaction ?reactionId ?equation WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:equation ?equation .
  ?reaction rh:id ?reactionId . 
}
ORDER BY ?reaction`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 23 Select all reaction IDs status and equations ordered by reaction identifier",slug:"Rhea_23_Select_all_reaction_IDs_status_and_equations_ordered_by_reaction_identifier",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reaction IDs, status and equations ordered by reaction identifier",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?reactionId ?status ?equation WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:equation ?equation .
  ?reaction rh:status ?status .
  ?reaction rh:id ?reactionId . 
}
ORDER BY ?reaction`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 24 Display only a subset of reactions",slug:"Rhea_24_Display_only_a_subset_of_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the 10 reactions, with the lowest identifier by alphabetic sort",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?reaction ?reactionId WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:id ?reactionId . 
}
ORDER BY ?reaction
LIMIT 10`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","ORDER BY","LIMIT"],category:"sib-swiss Rhea"},{name:"Rhea - 25 Select the first 10 reactions starting from the fifth reaction",slug:"Rhea_25_Select_the_first_10_reactions_starting_from_the_fifth_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the fifth to fiftheenth (inclusive) reactions, with the lowest identifier by alphabetic sort",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
SELECT ?reaction ?reactionId WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:id ?reactionId . 
}
ORDER BY ?reaction
LIMIT 10
OFFSET 5`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","ORDER BY","LIMIT","OFFSET"],category:"sib-swiss Rhea"},{name:"Rhea - 26 Select the number of reactions with status Approved",slug:"Rhea_26_Select_the_number_of_reactions_with_status_Approved",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of reactions with status Approved",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT (count(?reaction) as ?reactionCount) WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
}`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 27 Select the number of reactions that have Xrefs",slug:"Rhea_27_Select_the_number_of_reactions_that_have_Xrefs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of reactions that have cross-references",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT (count(distinct ?reaction) as ?distinctReactionCount) WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:directionalReaction ?directionalReaction .
  ?reaction rh:bidirectionalReaction ?bidirectionalReaction .
  OPTIONAL { ?directionalReaction rdfs:seeAlso ?xref . }
  OPTIONAL { ?bidirectionalReaction rdfs:seeAlso ?xref . }
  FILTER (BOUND(?xref))
}`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","DISTINCT","BOUND","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 28 Select all reactions with Xrefs",slug:"Rhea_28_Select_all_reactions_with_Xrefs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions with cross-references",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT 
  DISTINCT ?reaction 
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:directionalReaction ?directionalReaction .
  ?reaction rh:bidirectionalReaction ?bidirectionalReaction .
  OPTIONAL { ?directionalReaction rdfs:seeAlso ?xref . }
  OPTIONAL { ?bidirectionalReaction rdfs:seeAlso ?xref . }
  FILTER (BOUND(?xref))
}
ORDER BY ?reaction`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","ORDER BY","DISTINCT","BOUND"],category:"sib-swiss Rhea"},{name:"Rhea - 29 Select all approved reactions linked to PMID X",slug:"Rhea_29_Select_all_approved_reactions_linked_to_PMID_X",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions linked to PMID:2460092",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX pubmed:<http://rdf.ncbi.nlm.nih.gov/pubmed/>

SELECT ?reaction ?reactionEquation  WHERE {
  BIND(pubmed:2460092 AS ?cit)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:citation ?cit .
}
ORDER BY ?reaction`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","BIND","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 3",slug:"Rhea_3",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the specific form of <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 3
# Select the specific form of RHEA:11628.
# This query mimics the Related reactions sections of
# https://www.rhea-db.org/rhea/11628
#
SELECT ?rhea  ?equation ?childrenRhea ?childrenEquation 
WHERE {
  VALUES (?rhea) {(rh:11628)}
  ?rhea rh:equation ?equation .
  ?childrenRhea rdfs:subClassOf+ ?rhea .
  ?childrenRhea rh:equation ?childrenEquation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","IF"],category:"sib-swiss Rhea"},{name:"Rhea - 30 Select all approved reactions annotated with a given Pubmed ID",slug:"Rhea_30_Select_all_approved_reactions_annotated_with_a_given_Pubmed_ID",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions annotated with a given Pubmed ID (2460092, and show the pubmed id in result as text, using strafter and BIND)",context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX pubmed:<http://rdf.ncbi.nlm.nih.gov/pubmed/>

SELECT ?reaction ?pubMedID ?reactionEquation  WHERE {
  BIND(pubmed:2460092 AS ?cit)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:citation ?cit .
  BIND(strafter(str(?cit), str(pubmed:)) AS ?pubMedID  )
}`,ontologies:["EX","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","BIND","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 31 Select the average number of citation of reactions that have at least one citation",slug:"Rhea_31_Select_the_average_number_of_citation_of_reactions_that_have_at_least_one_citation",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rh: <http://rdf.rhea-db.org/>

SELECT (AVG(?linksToPubmedPerReaction) AS ?avgLinksToPubmedPerReaction)
WHERE
{
    SELECT ?reaction (COUNT(DISTINCT ?citation) AS ?linksToPubmedPerReaction)
    WHERE
    {
        ?reaction rh:citation ?citation .
    }
GROUP BY ?reaction ORDER BY DESC(?linksToPubmedPerReaction)
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","COUNT","AVG"],category:"sib-swiss Rhea"},{name:"Rhea - 32 Select the distribution of reactions according to their status",slug:"Rhea_32_Select_the_distribution_of_reactions_according_to_their_status",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the distribution of reactions according to their status",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?status
  (count(?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status ?status .
}
GROUP BY ?status`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 33 Select all approved transport reactions",slug:"Rhea_33_Select_all_approved_transport_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved transport reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?reaction
  ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:isTransport true
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 34 Select all cross-references for a given reaction",slug:"Rhea_34_Select_all_cross-references_for_a_given_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all cross-references for a given reaction (RHEA:11680)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT 
  ?reaction 
  ?xref 
WHERE {
  BIND (rh:11680 AS ?reaction)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:directionalReaction ?directionalReaction .
  OPTIONAL { ?directionalReaction rdfs:seeAlso ?xref . }
  ?reaction rh:bidirectionalReaction ?bidirectionalReaction .
  OPTIONAL { ?bidirectionalReaction rdfs:seeAlso ?xref . }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 35 Select all cross-references Kegg MetaCyc Macie for a given reaction",slug:"Rhea_35_Select_all_cross-references_Kegg_MetaCyc_Macie_for_a_given_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all cross-references (to KEGG, MetaCyc, Macie, ...) for a given reaction (RHEA:11932)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?reaction
  ?xref
WHERE {
  BIND (rh:11932 AS ?reaction)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:directionalReaction ?directionalReaction .
  OPTIONAL { ?directionalReaction rdfs:seeAlso ?xref . }
  ?reaction rh:bidirectionalReaction ?bidirectionalReaction .
  OPTIONAL { ?bidirectionalReaction rdfs:seeAlso ?xref . }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 36 Select the number of reactions with cross-references to KEGG resource",slug:"Rhea_36_Select_the_number_of_reactions_with_cross-references_to_KEGG_resource",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of reactions with cross-references to KEGG resource (via bidirectional rhea reactions)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX kegg: <http://identifiers.org/kegg.reaction/>

SELECT (count(?reaction) as ?reactionCount) 
WHERE {
    ?reaction rdfs:subClassOf rh:Reaction .
    ?reaction rh:bidirectionalReaction ?bidirectionalReaction .
    ?bidirectionalReaction rdfs:seeAlso ?xref .
    FILTER (regex(?xref, str(kegg:)))
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","STR","REGEX","IF","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 37 Select the number of reactions with cross-references to MetaCyc",slug:"Rhea_37_Select_the_number_of_reactions_with_cross-references_to_MetaCyc",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of reactions with cross-references to MetaCyc",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX up:<http://purl.uniprot.org/core/>
PREFIX keywords:<http://purl.uniprot.org/keywords/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>

SELECT
  (count(distinct ?rhea) as ?rheaMetacycCount)
  (count(distinct ?xref) as ?metacycRheaCount)
  (count(?rhea) as ?rheaMetacycXrefCount)
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:status rh:Approved .
  ?rhea rh:directionalReaction ?directionalReaction .
  ?rhea rh:bidirectionalReaction ?bidirectionalReaction .
  {
    ?directionalReaction rdfs:seeAlso ?xref .
    FILTER regex(str(?xref), \\"METACYC\\") .
  }
  UNION
  {
    ?bidirectionalReaction rdfs:seeAlso ?xref .
    FILTER regex(str(?xref), \\"METACYC\\") .
  }
  UNION
  {
    ?rhea rdfs:seeAlso ?xref .
    FILTER regex(str(?xref), \\"METACYC\\") .
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT","STR","REGEX","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 39 Select all reactions annotated with a given Pubmed ID",slug:"Rhea_39_Select_all_reactions_annotated_with_a_given_Pubmed_ID",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions annotated with a given Pubmed ID",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX pubmed:<http://rdf.ncbi.nlm.nih.gov/pubmed/>

SELECT
  ?reaction
  ?pubMedID
  ?reactionEquation
WHERE {
  BIND(pubmed:2460092 AS ?pubMedID)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:citation ?pubMedID .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 4",slug:"Rhea_4",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all cross-references for a given reaction",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 4
# Select all cross-references mapped to RHEA:21016
# 
# This query mimics the Cross-references section of 
# https://www.rhea-db.org/rhea/21016
#
SELECT distinct ?rhea ?rheaDir ?xref 
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  VALUES (?rhea) {(rh:21016)}
  {
    ?rhea rdfs:seeAlso ?xref .
    BIND(?rhea as ?rheaDir)
  }
  UNION
  {
    ?rhea rh:directionalReaction ?directionalReaction .
    ?directionalReaction rdfs:seeAlso ?xref . 
    BIND(?directionalReaction as ?rheaDir  )
  }
  UNION
  {
    ?rhea rh:bidirectionalReaction ?bidirectionalReaction .
    ?bidirectionalReaction rdfs:seeAlso ?xref . 
    BIND(?bidirectionalReaction as ?rheaDir  )
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","BIND","VALUES","DISTINCT","ALL"],category:"sib-swiss Rhea"},{name:"Rhea - 40 Select all citations of a given reaction",slug:"Rhea_40_Select_all_citations_of_a_given_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all citations of a given reaction",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX pubmed:<http://rdf.ncbi.nlm.nih.gov/pubmed/>

SELECT
  ?reaction
  ?citation
WHERE {
  BIND(rh:11680 AS ?reaction)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:citation ?citation .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 41 Select all reactions with citations display the number of citations and order by reaction ID",slug:"Rhea_41_Select_all_reactions_with_citations_display_the_number_of_citations_and_order_by_reaction_ID",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions with citations, display the number of citations and ORDER BY reaction ID",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?reaction
  (COUNT(DISTINCT ?citation) AS ?countPubmedPerReaction)
WHERE
{
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:citation ?citation .
}
GROUP BY ?reaction
ORDER BY DESC (COUNT(DISTINCT ?citation))`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 42 Select the average number of citation of reactions that have at least one citation",slug:"Rhea_42_Select_the_average_number_of_citation_of_reactions_that_have_at_least_one_citation",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rh: <http://rdf.rhea-db.org/>
SELECT
  (AVG(?linksToPubmedPerReaction) AS ?avgLinksToPubmedPerReaction)
WHERE
{
  SELECT
    ?reaction
    (COUNT(DISTINCT ?citation) AS ?linksToPubmedPerReaction)
  WHERE
  {
    ?reaction rh:citation ?citation .
  }
GROUP BY ?reaction ORDER BY DESC(?linksToPubmedPerReaction)
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","COUNT","AVG"],category:"sib-swiss Rhea"},{name:"Rhea - 43 Select the child reactions of a given reaction in the Rhea hierarchy",slug:"Rhea_43_Select_the_child_reactions_of_a_given_reaction_in_the_Rhea_hierarchy",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the child reaction(s) of a given reaction (RHEA:11628) in the Rhea hierarchy, (using rdfs:subClassOf)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT ?reaction ?equation ?childReaction ?childEquation 
WHERE {
  BIND (rh:11628 AS ?reaction)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?equation .

  ?childReaction rdfs:subClassOf rh:Reaction .
  ?childReaction rh:status rh:Approved .
  ?childReaction rdfs:subClassOf ?reaction .
  ?childReaction rh:equation ?childEquation .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 44 Select the descendant reaction of a given reaction in the Rhea hierarchy",slug:"Rhea_44_Select_the_descendant_reaction_of_a_given_reaction_in_the_Rhea_hierarchy",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the descendant reaction(s) of a given reaction (RHEA:11628) in the Rhea hierarchy, (using rdfs:subClassOf+)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT 
  ?reaction 
  ?equation 
  ?descendantReaction 
  ?descendantEquation 
WHERE {
  BIND(rh:11628 AS ?reaction)
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?equation .

  ?descendantReaction rdfs:subClassOf rh:Reaction .
  ?descendantReaction rh:status rh:Approved .
  ?descendantReaction rdfs:subClassOf+ ?reaction .
  ?descendantReaction rh:equation ?descendantEquation .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 45 Select the parent reactions of a given reaction",slug:"Rhea_45_Select_the_parent_reactions_of_a_given_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the parent reaction(s) of a given reaction",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT 
  ?reactionChild 
  ?childEquation 
  ?reactionParent 
  ?parentEquation 
WHERE {
  BIND (rh:39155 AS ?reactionChild)
  ?reactionChild rdfs:subClassOf rh:Reaction .
  ?reactionChild rh:status rh:Approved .
  ?reactionChild rh:equation ?childEquation .
  ?reactionParent rdfs:subClassOf rh:Reaction .
  ?reactionParent rh:status rh:Approved .
  ?reactionParent rh:equation ?parentEquation .
  ?reactionChild rdfs:subClassOf ?reactionParent .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 46 Number of IsA relationships",slug:"Rhea_46_Number_of_IsA_relationships",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Number of IsA relationships, distinct child reactions and distinct parent reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT 
    (COUNT(?reactionChild) AS ?isARelationCount) 
    (COUNT(distinct ?reactionChild) AS ?uniqueReactionChildCount)
    (COUNT(distinct ?reactionParent) AS ?uniqueReactionParentCount)
WHERE {
    ?reactionChild rdfs:subClassOf rh:Reaction .
    ?reactionChild rh:status rh:Approved .
    ?reactionParent rdfs:subClassOf rh:Reaction .
    ?reactionParent rh:status rh:Approved .
    ?reactionChild rdfs:subClassOf ?reactionParent .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 47 Select all children reactions and give the number of parent reactions",slug:"Rhea_47_Select_all_children_reactions_and_give_the_number_of_parent_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all children reactions and give the number of parent reactions.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT 
  (count(?reactionChild) as ?isARelationCount) 
  (count(distinct ?reactionChild) as ?uniqueReactionChildCount) 
  (count(distinct ?reactionParent) as ?uniqueReactionParentCount) 
WHERE {
  ?reactionChild rdfs:subClassOf rh:Reaction .
  ?reactionChild rh:status rh:Approved .
  ?reactionParent rdfs:subClassOf rh:Reaction .
  ?reactionParent rh:status rh:Approved .
  ?reactionChild rdfs:subClassOf ?reactionParent .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 48 Select all reactions with at least one parent reaction and give the number ancestor reactions",slug:"Rhea_48_Select_all_reactions_with_at_least_one_parent_reaction_and_give_the_number_ancestor_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions with at least one parent reaction and give the number ancestor reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  ?reaction
  (count(?parentReaction) as ?parentReactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .

  ?reaction rdfs:subClassOf ?parentReaction .

  ?parentReaction rdfs:subClassOf rh:Reaction .
  ?parentReaction rh:status rh:Approved .
}
GROUP BY ?reaction
ORDER BY DESC (?parentReactionCount) ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 49 Select parent reactions give the number of child and descendant reactions",slug:"Rhea_49_Select_parent_reactions_give_the_number_of_child_and_descendant_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select parent reactions, give the number of child and descendant reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT 
  ?reaction
  (count(?ancestorReaction) as ?ancestorReactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?ancestorReaction rdfs:subClassOf rh:Reaction .
  ?ancestorReaction rh:status rh:Approved .
  ?reaction rdfs:subClassOf+ ?ancestorReaction .
}
GROUP BY ?reaction
ORDER BY (count(?ancestorReaction))`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 5",slug:"Rhea_5",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions mapped to KEGG reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 5 
# Select all Rhea reactions mapped to KEGG reactions
# KEGG reactions are mapped to Rhea bidirectional reactions
# Rhea web query: https://www.rhea-db.org/rhea?query=kegg:*
#
SELECT ?rhea ?kegg ?rheaDir 
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:bidirectionalReaction ?rheaDir .
  ?rheaDir rdfs:seeAlso ?kegg .
  FILTER (regex(str(?kegg),'kegg'))
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ALL","STR","REGEX"],category:"sib-swiss Rhea"},{name:"Rhea - 50 Number of reactions that have parent and child reactions",slug:"Rhea_50_Number_of_reactions_that_have_parent_and_child_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Number of reactions that have parent and child reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT ?reaction
  (count(distinct ?reactionChild) as ?reactionChildCount)
  (count(distinct ?reactionDescendant) as ?reactionDescendantCount)
  ?equation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?equation .

  ?reactionChild rdfs:subClassOf rh:Reaction .
  ?reactionChild rh:status rh:Approved .

  ?reactionDescendant rdfs:subClassOf rh:Reaction .
  ?reactionDescendant rh:status rh:Approved .

  ?reaction ^rdfs:subClassOf ?reactionChild .
  ?reaction ^rdfs:subClassOf+ ?reactionDescendant .
}
GROUP BY ?reaction ?equation
ORDER BY DESC (count(?reactionChild))`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 51 Select all reactions that have parents and children reactions",slug:"Rhea_51_Select_all_reactions_that_have_parents_and_children_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all reactions that have parents and children reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .

  ?childReaction rdfs:subClassOf rh:Reaction .
  ?childReaction rh:status rh:Approved .

  ?parentReaction rdfs:subClassOf rh:Reaction .
  ?parentReaction rh:status rh:Approved .

  ?reaction rdfs:subClassOf ?parentReaction .
  ?reaction ^rdfs:subClassOf ?childReaction .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 52 List all small molecules",slug:"Rhea_52_List_all_small_molecules",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all small molecules",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  ?compound
  ?compoundName
WHERE
{
  ?compound rdfs:subClassOf rh:SmallMolecule .
  ?compound rh:name ?compoundName
}
ORDER BY (?compound)`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","ALL"],category:"sib-swiss Rhea"},{name:"Rhea - 53 List all polymers",slug:"Rhea_53_List_all_polymers",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all polymers",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  ?compound
  ?compoundName
WHERE
{
  ?compound rdfs:subClassOf rh:Polymer .
  ?compound rh:name ?compoundName
}
ORDER BY (?compound)`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 54 List all generic compounds",slug:"Rhea_54_List_all_generic_compounds",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all generic compounds",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  ?compound
  ?compoundName
WHERE
{
  ?compound rdfs:subClassOf rh:Polymer .
  ?compound rh:name ?compoundName .
}
ORDER BY (?compound)`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 55 Number of compounds according to their category",slug:"Rhea_55_Number_of_compounds_according_to_their_category",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Number of compounds according to their category (Small molecule, Generic compound, Polymer)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT 
  ?compoundCategory 
  ?compoundCategoryLabel?genericCategory
WHERE
{
  VALUES (?compoundCategory) {(rh:GenericCompound) (rh:Polymer) (rh:SmallMolecule)}
  ?compoundCategory rdfs:subClassOf rh:Compound .
  ?compoundCategory rdfs:label ?compoundCategoryLabel .
}
ORDER BY ?genericCategory DESC(?compound)`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY","ALL"],category:"sib-swiss Rhea"},{name:"Rhea - 56 Distribution of GenericParticipant",slug:"Rhea_56_Distribution_of_GenericParticipant",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>

SELECT
  ?genericSubClass
  ?labelGenericSubClass
  (count(?genericParticipant) as ?countGenericParticipant)
WHERE
{
  ?genericSubClass rdfs:subClassOf rh:GenericParticipant .
  ?genericSubClass rdfs:label ?labelGenericSubClass .
  ?genericParticipant rdfs:subClassOf* ?genericSubClass .
}
GROUP BY ?genericSubClass ?labelGenericSubClass`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","GROUP BY","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 57 Give me the list of Rhea generics",slug:"Rhea_57_Give_me_the_list_of_Rhea_generics",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Give me the list of Rhea generics, type polypeptide (rh:GenericPolypeptideParticipant)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  ?genericSubClass
  ?labelGenericSubClass
  (count(?genericParticipant) as ?countGenericParticipant)
WHERE
{
  ?genericSubClass rdfs:subClassOf rh:GenericParticipant .
  ?genericSubClass rdfs:label ?labelGenericSubClass .
  ?genericParticipant rdfs:subClassOf* ?genericSubClass .
}
GROUP BY ?genericSubClass ?labelGenericSubClass`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 58 Give me the reactions involving a given Rhea generic",slug:"Rhea_58_Give_me_the_reactions_involving_a_given_Rhea_generic",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Give me the reactions involving a given Rhea generic (GENERIC:11964, \\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

SELECT
  ?genericSubClass
  ?labelGenericSubClass
  (count(?genericParticipant) as ?countGenericParticipant)
WHERE
{
  ?genericSubClass rdfs:subClassOf rh:GenericParticipant .
  ?genericSubClass rdfs:label ?labelGenericSubClass .
  ?genericParticipant rdfs:subClassOf* ?genericSubClass .
} GROUP BY ?genericSubClass ?labelGenericSubClass`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 59 Select reaction participants for a given reaction ID. Display their coefficient and their name",slug:"Rhea_59_Select_reaction_participants_for_a_given_reaction_ID._Display_their_coefficient_and_their_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select reaction participants for a given reaction ID. Display their coefficient and their name.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
SELECT
  ?genericCompound
  ?genericCompoundName
  ?reaction
  ?equation
WHERE
{
  ?genericCategory rdfs:subClassOf rh:GenericCompound .
  ?genericCompound rdfs:subClassOf ?genericCategory .
  ?genericCompound rh:name ?genericCompoundName .
  ?genericCompound rh:accession ?ac .
  FILTER (?ac='GENERIC:11964') .

  ?participant rh:compound ?genericCompound .
  ?reactionSide rh:contains ?participant .
  ?reaction rh:side ?reactionSide .
  ?reaction rh:equation ?equation .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 6",slug:"Rhea_6",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions mapped to MetaCyc reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 6 
# Select all Rhea reactions mapped to MetaCyc reactions
# MetaCyc reactions are mapped to either undirected, left-to-right, right-to-left or bidirectional reactions
# Rhea web query: https://www.rhea-db.org/rhea?query=metacyc:*
#
SELECT distinct ?rhea ?rheaDir ?metacyc
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  {
    ?rhea rdfs:seeAlso ?metacyc .
    FILTER CONTAINS(str(?metacyc), "METACYC") 
    BIND(?rhea as ?rheaDir)
  }
  UNION
  {
    ?rhea rh:directionalReaction ?directionalReaction .
    ?directionalReaction rdfs:seeAlso ?metacyc . 
    FILTER CONTAINS(str(?metacyc), "METACYC") 
    BIND(?directionalReaction as ?rheaDir  )
  }
  UNION
  {
    ?rhea rh:bidirectionalReaction ?bidirectionalReaction .
    ?bidirectionalReaction rdfs:seeAlso ?metacyc . 
    FILTER CONTAINS(str(?metacyc), "METACYC") 
    BIND(?bidirectionalReaction as ?rheaDir  )
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","DISTINCT","ALL","STR","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 60 Select the number of approved reactions using CHEBI 29985 as small molecule participant",slug:"Rhea_60_Select_the_number_of_approved_reactions_using_CHEBI_29985_as_small_molecule_participant",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of approved reactions using CHEBI:29985 (L-glutamate) as small molecule participant",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?reaction
  ?reactionSide
  ?coefficient
  ?participant
  ?name
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:side ?reactionSide .
  ?contains rdfs:subPropertyOf rh:contains .
  ?contains rh:coefficient ?coefficient .
  ?reactionSide ?contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:name ?name .

  FILTER (?reaction=rh:11680)
}
ORDER BY ?reactionSide`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 61 Select all approved reactions using CHEBI 29985 as small molecule participant",slug:"Rhea_61_Select_all_approved_reactions_using_CHEBI_29985_as_small_molecule_participant",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions using CHEBI:29985 (L-glutamate) as small molecule participant",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?smallMolecule .
  ?smallMolecule rdfs:subClassOf+ CHEBI:29985
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ALL","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 62 Select the number of approved reactions using L-glutamine AND L-glutamate in different reaction sides",slug:"Rhea_62_Select_the_number_of_approved_reactions_using_L-glutamine_AND_L-glutamate_in_different_reaction_sides",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of approved reactions using L-glutamine (CHEBI:29985) AND L-glutamate (CHEBI:58359) in different reaction sides",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?reaction
  ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi CHEBI:29985
}

# There are several way to indicate that the reaction sides must be different:
# - ?reactionSide1 rh:transformableTo ?reactionSide2
# FILTER (?reactionSide1 != ?reactionSide2)
#It is better to use rh:transformableTo predicate.`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","IF","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 63 Select all approved reactions using L-glutamine AND L-glutamate in different reaction sides",slug:"Rhea_63_Select_all_approved_reactions_using_L-glutamine_AND_L-glutamate_in_different_reaction_sides",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions using L-glutamine (CHEBI:29985) AND L-glutamate (CHEBI:58359) in different reaction sides",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .

  ?reaction rh:side ?reactionSide1 .
  ?reactionSide1 rh:contains ?participant1 .
  ?participant1 rh:compound ?compound1 .
  ?compound1 rh:chebi CHEBI:29985 .

  ?reaction rh:side ?reactionSide2 .
  ?reactionSide2 rh:contains ?participant2 .
  ?participant2 rh:compound ?compound2 .
  ?compound2 rh:chebi CHEBI:58359 .

  ?reactionSide1 rh:transformableTo ?reactionSide2
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 64 Number of reaction participants per reaction side",slug:"Rhea_64_Number_of_reaction_participants_per_reaction_side",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Distribution: number of reaction participants per reaction Side",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?reaction
  ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .

  ?reaction rh:side ?reactionSide1 .
  ?reactionSide1 rh:contains ?participant1 .
  ?participant1 rh:compound ?compound1 .
  ?compound1 rh:chebi CHEBI:29985 .

  ?reaction rh:side ?reactionSide2 .
  ?reactionSide2 rh:contains ?participant2 .
  ?participant2 rh:compound ?compound2 .
  ?compound2 rh:chebi CHEBI:58359 .

  ?reactionSide1 rh:transformableTo ?reactionSide2 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 65 Number of reaction participants per reaction side",slug:"Rhea_65_Number_of_reaction_participants_per_reaction_side",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Distribution: number of reaction participants per reaction Side",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?reaction
  ?reactionSide1
  (count(distinct ?participant1) as ?side1ParticipantCount)
  ?reactionSide2
  (count(distinct ?participant2) as ?side2ParticipantCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .

  ?reaction rh:side ?reactionSide1 .
  ?reactionSide1 rh:contains ?participant1 .

  ?reaction rh:side ?reactionSide2 .
  ?reactionSide2 rh:contains ?participant2 .

  ?reactionSide1 rh:transformableTo ?reactionSide2
}
GROUP BY ?reaction ?reactionSide1 ?reactionSide2
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 66 Select all compounds and count their occurrence in Rhea reactions",slug:"Rhea_66_Select_all_compounds_and_count_their_occurrence_in_Rhea_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all compounds and count their occurrence in Rhea reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?reaction
  ?reactionSide1
  (count(distinct ?participant1) as ?side1ParticipantCount)
  ?reactionSide2
  (count(distinct ?participant2) as ?side2ParticipantCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .

  ?reaction rh:side ?reactionSide1 .
  ?reactionSide1 rh:contains ?participant1 .
  ?reactionSide1 rh:curatedOrder ?curatedOrder1 .

  ?reaction rh:side ?reactionSide2 .
  ?reactionSide2 rh:contains ?participant2 .
  ?reactionSide2 rh:curatedOrder ?curatedOrder2 .

  ?reactionSide1 rh:transformableTo ?reactionSide2 .
  FILTER (?curatedOrder1 < ?curatedOrder2) .
}
GROUP BY ?reaction ?reactionSide1 ?reactionSide2
ORDER BY DESC(?side1ParticipantCount) DESC(?side2ParticipantCount) ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","GROUP BY","ORDER BY","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 67 Select all compounds and count their occurrence in Rhea reactions",slug:"Rhea_67_Select_all_compounds_and_count_their_occurrence_in_Rhea_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all compounds and count their occurrence in Rhea reactions",context:null,inidces:[],query:`PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT
  ?reactionParticipantAc
  ?chebi
  (count(distinct ?reaction) as ?reactionCount)
  ?reactionParticipantName
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  OPTIONAL {
    ?compound rh:chebi ?chebi .
  }
  ?compound rh:name ?reactionParticipantName .
  ?compound rh:accession ?reactionParticipantAc .
}
GROUP BY ?reactionParticipantAc ?chebi ?reactionParticipantName
ORDER BY DESC(?reactionCount)`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","GROUP BY","ORDER BY","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 68 Select reaction participants that appear in only one reaction",slug:"Rhea_68_Select_reaction_participants_that_appear_in_only_one_reaction",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select reaction participants that appear in only one reaction",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?compound
  ?compoundAc
  ?compoundName
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:name ?compoundName .
  ?compound rh:accession ?compoundAc .
}
GROUP BY ?compound ?compoundName ?compoundAc
HAVING (count(distinct ?reaction) = 1)
ORDER BY ?compoundAc`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","HAVING","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 69 Select all the direct children of carbohydrate in the ChEBI ontology used in Rhea reaction or not",slug:"Rhea_69_Select_all_the_direct_children_of_carbohydrate_in_the_ChEBI_ontology_used_in_Rhea_reaction_or_not",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all the (direct) children of *carbohydrate* (CHEBI:16646) in the ChEBI ontology , used in Rhea reaction or not",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?chebiChildCarbohydrate
  ?chebiLabel
WHERE {
  ?chebiChildCarbohydrate rdfs:subClassOf CHEBI:16646 .
  ?chebiChildCarbohydrate rdfs:label ?chebiLabel .
}
ORDER BY ?chebiChildCarbohydrate`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY"],category:"sib-swiss Rhea"},{name:"Rhea - 7",slug:"Rhea_7",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions mapped to Reactome reactions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 7 
# Select all Rhea reactions mapped to Reactome reactions
# MetaCyc reactions are mapped to either undirected, left-to-right or right-to-left reactions
# Rhea web query: https://www.rhea-db.org/rhea?query=reactome:*
#    
SELECT distinct ?rhea ?rheaDir ?reactome  
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  {
    ?rhea rdfs:seeAlso ?reactome .
    FILTER CONTAINS(str(?reactome), "reactome") 
    BIND(?rhea as ?rheaDir)
  }
  UNION
  {
    ?rhea rh:directionalReaction ?directionalReaction .
    ?directionalReaction rdfs:seeAlso ?reactome . 
    FILTER CONTAINS(str(?reactome), "reactome") 
    BIND(?directionalReaction as ?rheaDir  )
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","DISTINCT","ALL","STR","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 70 Select all the descendants of  carbohydrate  in the ChEBI ontology used in Rhea reaction or not",slug:"Rhea_70_Select_all_the_descendants_of__carbohydrate__in_the_ChEBI_ontology_used_in_Rhea_reaction_or_not",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all the descendants of *carbohydrate* (CHEBI:16646) in the ChEBI ontology, used in Rhea reaction or not",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?chebiDescendantCarbohydrate
  (str(?label) as ?chebiLabel)
WHERE {
  ?chebiDescendantCarbohydrate rdfs:subClassOf+ CHEBI:16646 .
  ?chebiDescendantCarbohydrate rdfs:label ?label .
}
ORDER BY ?chebiDescendantCarbohydrate`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 71 Select count of descendants of  carbohydrate  in the ChEBI ontology used in Rhea reaction or not",slug:"Rhea_71_Select_count_of_descendants_of__carbohydrate__in_the_ChEBI_ontology_used_in_Rhea_reaction_or_not",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select count of descendants of *carbohydrate* (CHEBI:16646) in the ChEBI ontology, used in Rhea reaction or not",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(?chebiDescendantCarbohydrate) AS ?chebiDescendantCarbohydrateCount)
WHERE {
  ?chebiDescendantCarbohydrate rdfs:subClassOf+ CHEBI:16646 .
  ?chebiDescendantCarbohydrate rdfs:label ?label .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 72 Select children of ChEBI 35179 in the ChEBI hierarchy used in Rhea reactions or not",slug:"Rhea_72_Select_children_of_ChEBI_35179_in_the_ChEBI_hierarchy_used_in_Rhea_reactions_or_not",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select children of CHEBI:35179 (a 2-oxo carboxylate) in the ChEBI hierarchy (using rdfs:subClassOf), used in Rhea reactions or not",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?chebiChild
  (str(?label) as ?chebiChildName)
WHERE {
  ?chebiChild rdfs:subClassOf CHEBI:35179 .
  ?chebiChild rdfs:label ?label .
}
ORDER BY ?chebiChild`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 73 Select all the descendants of CHEBI 35179 in the ChEBI hierarchy used in Rhea reaction or not",slug:"Rhea_73_Select_all_the_descendants_of_CHEBI_35179_in_the_ChEBI_hierarchy_used_in_Rhea_reaction_or_not",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all the descendants of CHEBI:35179 (a 2-oxo carboxylate) in the ChEBI hierarchy (using rdfs:subClassOf+), used in Rhea reaction or not",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?chebiDescendant
  (str(?label) as ?chebiDescendantName)
WHERE {
  ?chebiDescendant rdfs:subClassOf+ CHEBI:35179 .
  ?chebiDescendant rdfs:label ?label .
}
ORDER BY ?chebiDescendant`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 74 Select children of CHEBI 35179 in the ChEBI hierarchy used in Rhea reactions and show the reactions",slug:"Rhea_74_Select_children_of_CHEBI_35179_in_the_ChEBI_hierarchy_used_in_Rhea_reactions_and_show_the_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select children of CHEBI:35179 (a 2-oxo carboxylate) in the ChEBI hierarchy (using rdfs:subClassOf) used in Rhea reaction(s), and show the reaction(s)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  ?chebi
  ?compoundName
  ?reaction
  ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:name ?compoundName .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf CHEBI:35179 .
}
ORDER BY ?chebi`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 75 Select all the descendants of CHEBI 35179 in the ChEBI hierarchy used in Rhea reactions and show the reactions",slug:"Rhea_75_Select_all_the_descendants_of_CHEBI_35179_in_the_ChEBI_hierarchy_used_in_Rhea_reactions_and_show_the_reactions",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all the descendants of CHEBI:35179 (a 2-oxo carboxylate) in the ChEBI hierarchy (using rdfs:subClassOf+) used in Rhea reaction(s), and show the reaction(s)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  ?chebi
  ?compoundName
  ?reaction
  ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:name ?compoundName .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:35179 .
}
ORDER BY ?chebi`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 76 Select the number of approved reactions involving a monosaccharide derivative  based on the ChEBI ontology",slug:"Rhea_76_Select_the_number_of_approved_reactions_involving_a_monosaccharide_derivative__based_on_the_ChEBI_ontology",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of approved reactions involving a *_monosaccharide derivative_* (subclass of *CHEBI:63367*) based on the ChEBI ontology (use of *subClassOf+*)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:63367 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 77 Select all approved reactions involving a monosaccharide derivative",slug:"Rhea_77_Select_all_approved_reactions_involving_a_monosaccharide_derivative",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions involving a *_monosaccharide derivative_* (subclass of *CHEBI:63367*) based on the ChEBI ontology (use of *subClassOf+*)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  distinct
    ?reaction
    ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:63367 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 78 Select the number of approved reactions involving a monosaccharide",slug:"Rhea_78_Select_the_number_of_approved_reactions_involving_a_monosaccharide",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of approved reactions involving a *_monosaccharide_* (subclass of *CHEBI:35381*) based on the ChEBI ontology (use of *subClassOf+*)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:35381 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 79 Select all approved reactions involving a monosaccharide",slug:"Rhea_79_Select_all_approved_reactions_involving_a_monosaccharide",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions involving a *_monosaccharide_* (subclass of *CHEBI:35381*) based on the ChEBI ontology (use of *subClassOf+*)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>

SELECT
  distinct
    ?reaction
    ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:35381 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 8",slug:"Rhea_8",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions mapped to GO molecular functions",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 8
# Select all Rhea reactions mapped to GO molecular function
# 
# This query corresponds to the Rhea website query:
# https://www.rhea-db.org/rhea?query=go:*
#
SELECT ?go ?rhea ?equation 
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rdfs:seeAlso ?go .
  FILTER CONTAINS(str(?go), "GO_")   
  ?rhea rh:equation ?equation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ALL","STR","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 80 Select the number of approved reactions involving a carbohydrate derivative",slug:"Rhea_80_Select_the_number_of_approved_reactions_involving_a_carbohydrate_derivative",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of approved reactions involving a *_carbohydrate derivative_* (subclass of *CHEBI:63299*) based on the ChEBI ontology (use of *subClassOf+*)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:63299 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 81 Select all approved reactions involving a carbohydrate derivative",slug:"Rhea_81_Select_all_approved_reactions_involving_a_carbohydrate_derivative",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions involving a *_carbohydrate derivative_* (subclass of *CHEBI:63299*) based on the ChEBI ontology",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>

SELECT
  distinct
    ?reaction
    ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:63299 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 82 Select the number of approved reactions involving a carbohydrate",slug:"Rhea_82_Select_the_number_of_approved_reactions_involving_a_carbohydrate",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of approved reactions involving a *_carbohydrate_* (subclass of *CHEBI:16646*) based on the ChEBI ontology (use of *subClassOf+*)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:16646 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 83 Select all approved reactions involving a carbohydrate",slug:"Rhea_83_Select_all_approved_reactions_involving_a_carbohydrate",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions involving a *_carbohydrate_* (subclass of *CHEBI:16646*) based on the ChEBI ontology (use of *subClassOf+*)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  distinct
    ?reaction
    ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:16646 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 84 Select the number of approved reactions involving  lipids",slug:"Rhea_84_Select_the_number_of_approved_reactions_involving__lipids",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of approved reactions involving *lipids* (subclass of *CHEBI:18059*) based on the ChEBI ontology",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:18059 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 85 Select all approved reactions involving lipids based on the ChEBI ontology",slug:"Rhea_85_Select_all_approved_reactions_involving_lipids_based_on_the_ChEBI_ontology",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions involving *lipids* (subclass of *CHEBI:18059*) based on the ChEBI ontology",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>

SELECT
  distinct
    ?reaction
    ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:18059 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 86 Select all approved reactions involving lipids",slug:"Rhea_86_Select_all_approved_reactions_involving_lipids",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions involving *lipids* (subclass of *CHEBI:18059*) that have at least one parent reaction",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>

SELECT
  distinct
    ?reaction
    ?reactionParent
WHERE {
  ?reactionParent rdfs:subClassOf rh:Reaction .
  ?reactionParent rh:status rh:Approved .

  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rdfs:subClassOf ?reactionParent .
  ?reaction rh:side ?reactionSide .

  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf+ CHEBI:18059 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 87 Select all approved reactions involving lipids that are parent reactions and give their number of descendants",slug:"Rhea_87_Select_all_approved_reactions_involving_lipids_that_are_parent_reactions_and_give_their_number_of_descendants",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions involving *lipids* (subclass of *CHEBI:18059*) that are parent reactions and give their number of descendants",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT distinct ?reaction (count(distinct ?reactionDescendant) as ?reactionDescendantCount) ?reactionEquation WHERE {
?reaction rdfs:subClassOf rh:Reaction .
?reaction rh:status rh:Approved .
?reaction rh:equation ?reactionEquation .
?reaction rh:side ?reactionSide .

?reactionSide rh:contains ?participant .
?participant rh:compound ?compound .
?compound rh:chebi ?chebi .
?chebi rdfs:subClassOf+ CHEBI:18059 .

?reactionDescendant rdfs:subClassOf rh:Reaction .
?reactionDescendant rh:status rh:Approved .

?reactionDescendant rdfs:subClassOf+ ?reaction .
}
group by ?reaction ?reactionEquation
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 88 How many descendants of CHEBI 17815",slug:"Rhea_88_How_many_descendants_of_CHEBI_17815",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"How many descendants of CHEBI:17815 (a 1,2-diacyl-sn-glycerol) in the ChEBI ontology?",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT (count(?chebiDescendant) as ?countChild) WHERE {
  ?chebiDescendant rdfs:subClassOf+ CHEBI:17815 .
  ?chebiDescendant rdfs:label ?label .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 89 Give me all CHEBI identifier and label for the descendants of CHEBI 17815",slug:"Rhea_89_Give_me_all_CHEBI_identifier_and_label_for_the_descendants_of_CHEBI_17815",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Give me all CHEBI identifier and label for the descendants of CHEBI:17815 (a 1,2-diacyl-sn-glycerol)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
SELECT
  ?chebiDescendant
  (str(?label) as ?chebiLabel)
WHERE {
  ?chebiDescendant rdfs:subClassOf+ CHEBI:17815 .
  ?chebiDescendant rdfs:label ?label .
}
ORDER BY ?chebiDescendantCarbohydrate`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 9",slug:"Rhea_9",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all Rhea reactions mapped to enzyme classification (EC numbers)",context:null,inidces:[],query:`PREFIX ec: <http://purl.uniprot.org/enzyme/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Query 9
# Select all Rhea reactions mapped to EC numbers (enzyme classification)
#
# This query corresponds to the Rhea website query:
# https://www.rhea-db.org/rhea?query=ec:*
#
SELECT ?ec ?ecNumber ?rhea ?accession ?equation
WHERE {
  ?rhea rdfs:subClassOf rh:Reaction .
  ?rhea rh:accession ?accession .
  ?rhea rh:ec ?ec .
  BIND(strafter(str(?ec),str(ec:)) as ?ecNumber)
  ?rhea rh:isTransport ?isTransport .
  ?rhea rh:equation ?equation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","ALL","STR","IF"],category:"sib-swiss Rhea"},{name:"Rhea - 90 How many reactions involve CHEBI 17815",slug:"Rhea_90_How_many_reactions_involve_CHEBI_17815",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"How many reactions involve CHEBI:17815 (a 1,2-diacyl-sn-glycerol)?",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

PREFIX rh:<http://rdf.rhea-db.org/>
SELECT
  (count(?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .

  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi CHEBI:17815 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 91 How many reactions involve CHEBI 17815",slug:"Rhea_91_How_many_reactions_involve_CHEBI_17815",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"How many reactions involve CHEBI:17815 (a 1,2-diacyl-sn-glycerol) or one of its descendant?",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  (count(distinct ?reaction) as ?countReaction)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction ;
  rh:status rh:Approved ;
    rh:side ?reactionSide .

  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf* CHEBI:17815 .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 92 Select all approved reactions with participants being CHEBI 17815  or one of its descendant",slug:"Rhea_92_Select_all_approved_reactions_with_participants_being_CHEBI_17815__or_one_of_its_descendant",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions with participant(s) being CHEBI:17815 (a 1,2-diacyl-sn-glycerol) or one of its descendant",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>

SELECT
  DISTINCT
    ?reaction
    ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf* CHEBI:17815 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 93 Select all approved reactions with CHEBI 17815 or one of its descendant optional EC",slug:"Rhea_93_Select_all_approved_reactions_with_CHEBI_17815_or_one_of_its_descendant_optional_EC",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions with CHEBI:17815 (a 1,2-diacyl-sn-glycerol) or one of its descendant. Display the EC numbers if the rhea-ec link exists.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>

SELECT
  distinct
    ?reaction
    ?ec
    ?reactionEquation
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .

  OPTIONAL {?reaction rh:ec ?ec .} .

  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf* CHEBI:17815 .
}
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","ORDER BY","DISTINCT","CONTAINS"],category:"sib-swiss Rhea"},{name:"Rhea - 94 Select all approved reactions with CHEBI or one of its descendant optional UniProtKB reviewed EC",slug:"Rhea_94_Select_all_approved_reactions_with_CHEBI_or_one_of_its_descendant_optional_UniProtKB_reviewed_EC",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all approved reactions with CHEBI:17815 (a 1,2-diacyl-sn-glycerol) or one of its descendant. Display links to UniProtKB/Swiss-Prot entries via EC numbers if the link rhea-ec exists.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX up:<http://purl.uniprot.org/core/>

SELECT
  distinct
    ?reaction
    ?enzyme
    (count(distinct ?protein) as ?proteinCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .

  OPTIONAL {?reaction rh:ec ?enzyme .
    SERVICE <https://sparql.uniprot.org/sparql> {
      ?protein up:reviewed true .
      ?protein up:enzyme ?enzyme.
    }
  } .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?chebi rdfs:subClassOf* CHEBI:17815 .
}
GROUP BY ?reaction ?enzyme
ORDER BY ?reaction`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH","SPEX"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","SERVICE","GROUP BY","ORDER BY","DISTINCT","CONTAINS","COUNT"],category:"sib-swiss Rhea"},{name:"Rhea - 95 Get uniprot name of a given chebi compound",slug:"Rhea_95_Get_uniprot_name_of_a_given_chebi_compound",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get uniprot name of a given chebi compound",context:null,inidces:[],query:`PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX oboInOwl:<http://www.geneontology.org/formats/oboInOwl#>
PREFIX owl:<http://www.w3.org/2002/07/owl#>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>

SELECT
  ?chebi
  ?chebiName
  ?chebiUniprotName
WHERE {
  BIND(CHEBI:57416 AS ?chebi)
  ?chebi rdfs:label ?chebiName .
  ?chebi oboInOwl:hasRelatedSynonym ?chebiUniprotName .
  ?axiom a owl:Axiom .
  ?axiom owl:annotatedSource ?chebi .
  ?axiom owl:annotatedProperty oboInOwl:hasRelatedSynonym .
  ?axiom owl:annotatedTarget ?chebiUniprotName .
  ?axiom oboInOwl:hasDbXref ?dbXref .
  FILTER (?dbXref="UniProt")
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","NOW"],category:"sib-swiss Rhea"},{name:"Rhea - 96 Get tautomer of a given chebi compound",slug:"Rhea_96_Get_tautomer_of_a_given_chebi_compound",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get tautomer of a given chebi compound",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX obo:<http://purl.obolibrary.org/obo/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX chebihash:<http://purl.obolibrary.org/obo/chebi#>
PREFIX owl:<http://www.w3.org/2002/07/owl#>
select
  ?chebi
  ?chebiName
  ?chebiTautomer
  ?chebiTautomerName
where {
  BIND(CHEBI:57416 AS ?chebi)
  ?chebi rdfs:subClassOf ?chebiRestriction .
  ?chebiRestriction a owl:Restriction .
  ?chebiRestriction owl:onProperty chebihash:is_tautomer_of .
  ?chebiRestriction owl:someValuesFrom ?chebiTautomer .
  ?chebi rdfs:label ?chebiName .
  ?chebiTautomer rdfs:label ?chebiTautomerName .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","VALUES","FROM","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 97 Get structural data of a given ChEBI compound",slug:"Rhea_97_Get_structural_data_of_a_given_ChEBI_compound",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`#
# get structural data of a given ChEBI compound
#
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX chebihash:<http://purl.obolibrary.org/obo/chebi/>
SELECT
  ?chebi
  ?formula
  ?charge
  ?mass
  ?monoisotopicmass
  ?inchikey
  ?smiles
  ?inchi
WHERE {
  BIND (CHEBI:29985 AS ?chebi)
  ?chebi chebihash:inchi ?inchi ;
    chebihash:smiles ?smiles ;
    chebihash:formula ?formula ;
    chebihash:mass ?mass ;
    chebihash:charge ?charge ;
    chebihash:inchikey ?inchikey ;
    chebihash:monoisotopicmass ?monoisotopicmass .
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","BIND","STR"],category:"sib-swiss Rhea"},{name:"Rhea - 98 Get chebi compound of a given InChI",slug:"Rhea_98_Get_chebi_compound_of_a_given_InChI",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX obo:<http://purl.obolibrary.org/obo/>
PREFIX chebihash:<http://purl.obolibrary.org/obo/chebi/>
SELECT
  ?chebi
  ?inchi
WHERE {
  BIND("InChI=1S/C5H9NO4/c6-3(5(9)10)1-2-4(7)8/h3H,1-2,6H2,(H,7,8)(H,9,10)/t3-/m0/s1" AS ?inchi)
  ?chebi chebihash:inchi ?inchi .
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","BIND"],category:"sib-swiss Rhea"},{name:"Rhea - 99 Get the list of reactions involving a ChEBI participant with  has role antifungal agent",slug:"Rhea_99_Get_the_list_of_reactions_involving_a_ChEBI_participant_with__has_role_antifungal_agent",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get the list of reactions involving a ChEBI participant with <em>has role</em> <strong>antifungal agent</strong> (CHEBI:35718)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX chebihash:<http://purl.obolibrary.org/obo/chebi#>
SELECT
  (count(distinct ?reaction) as ?reactionCount)
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:equation ?reactionEquation .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebiInRhea .
  {
    ?chebiInRhea rdfs:subClassOf ?chebiRestrictionRole .
    ?chebiRestrictionRole a owl:Restriction .
    ?chebiRestrictionRole owl:onProperty obo:RO_0000087 .
    ?chebiRestrictionRole owl:someValuesFrom CHEBI:35718 .
  }
  UNION
  {
    ?chebi rdfs:subClassOf ?chebiRestrictionRole .
    ?chebiRestrictionRole a owl:Restriction .
    ?chebiRestrictionRole owl:onProperty obo:RO_0000087 .
    ?chebiRestrictionRole owl:someValuesFrom CHEBI:35718 .

    ?chebi rdfs:subClassOf ?chebiRestrictionPH .
    ?chebiRestrictionPH a owl:Restriction .
    ?chebiRestrictionPH owl:onProperty chebihash:has_major_microspecies_at_pH_7_3 .
    ?chebiRestrictionPH owl:someValuesFrom ?chebiInRhea .
  }
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","VALUES","DISTINCT","FROM","STR","CONTAINS","COUNT"],category:"sib-swiss Rhea"}],fP=[{name:"SwissLipids - 1",slug:"SwissLipids_1",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the SwissLipids categories and their labels.",context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 1
SELECT ?category ?label
WHERE
{
    ?category SWISSLIPID:rank SWISSLIPID:Category .
    ?category rdfs:label ?label .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 10",slug:"SwissLipids_10",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 10
SELECT ?slm ?slmName ?hmdb
WHERE {
  ?slm rdfs:label ?slmName . 
  # Corresponding HMDB entries
  ?slm rdfs:seeAlso ?hmdb .
  FILTER regex(str(?hmdb), "hmdb") .
}
ORDER BY ?slm`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","STR","REGEX"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 11",slug:"SwissLipids_11",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 11
prefix HMDB:<http://identifiers.org/hmdb/>
SELECT ?hmdb ?slm ?slmName 
WHERE {
  VALUES ?hmdb {HMDB:HMDB00269 HMDB:HMDB00032 HMDB:HMDB01383}
  ?slm rdfs:label ?slmName . 
  # Corresponding SwissLipids entries
  ?slm rdfs:seeAlso ?hmdb .
}
ORDER BY ?hmdb`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY","IF"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 12",slug:"SwissLipids_12",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 12
SELECT ?slm ?slmName ?chebi
WHERE {
  ?slm rdfs:label ?slmName . 
  # Corresponding ChEBI entries
  ?slm owl:equivalentClass ?chebi .
}
ORDER BY ?slm`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 13",slug:"SwissLipids_13",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 13
SELECT ?chebi ?slm ?slmName 
WHERE {
  VALUES ?chebi {CHEBI:70846 CHEBI:70771 CHEBI:70829}
  ?slm rdfs:label ?slmName . 
  # Corresponding SwissLipids entries
  ?slm owl:equivalentClass ?chebi .
}
ORDER BY ?chebi`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 14",slug:"SwissLipids_14",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX lipidmaps: <https://www.lipidmaps.org/rdf/>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 14
SELECT ?lipidmaps ?slm ?slmName ?chebi
WHERE {
  #VALUES ?lipidmaps {lipidmaps:LMSP01020001 lipidmaps:LMST01010069 lipidmaps:LMST04030222}
  ?slm rdfs:label ?slmName . 
  # Corresponding SwissLipids entries
  ?slm rdfs:seeAlso ?lipidmaps .
  FILTER regex(str(?lipidmaps), "lipidmaps") .

  # Hierarchical search
  #?children rdfs:subClassOf* ?slm .
  #?children rdfs:label ?name .
  # Corresponding ChEBI entries
  ?slm owl:equivalentClass ?chebi .
}
ORDER BY ?lipidmaps ?slm ?chebi`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","ORDER BY","STR","REGEX"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 15",slug:"SwissLipids_15",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT
    ?swisslipid
    ?lipidx
    ?isomericsubSwisslipid
    ?pubmed
WHERE {
    VALUES ?hmdb {'HMDB00269' 'MDB00032'}
    BIND (IRI(CONCAT('http://identifiers.org/lipidx/', ?lipidx)) AS ?lipidxIRI)
    ?swisslipid rdfs:seeAlso ?lipidxIRI .
    ?swisslipid rdfs:subClassOf ?isomericsubSwisslipid .
    ?isomericsubSwisslipid SWISSLIPID:rank SWISSLIPID:Isomeric_Subspecies .
    ?isomericsubSwisslipid SWISSLIPID:citation ?pubmed .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","VALUES","IRI","IF","CONCAT"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 16",slug:"SwissLipids_16",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

# Example 16
SELECT ?startId ?startName ?uniprot
WHERE
{
  # SwissLipids query (list of) identifier(s)
  VALUES ?startId { SWISSLIPID:000399814 }
  # name
  ?startId rdfs:label ?startName .
  # ChEBI x-ref
  ?startId owl:equivalentClass ?chebi .
  # federated query to Rhea
  SERVICE <https://sparql.rhea-db.org/sparql> {
  	?rhea rh:side/rh:contains/rh:compound/rh:chebi ?chebi .
  }    
  # federated query to UniProt
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?uniprot up:reviewed true .
    ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea .
  }
} 
ORDER BY ?startId`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","SERVICE","ORDER BY","IF","CONTAINS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 2",slug:"SwissLipids_2",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 2
SELECT ?startId ?startName ?rank ?id ?name
WHERE
{
  # query: list of SwissLipids Species identifiers
  VALUES ?startId { SWISSLIPID:000056420 SWISSLIPID:000308470 }
  ?startId SWISSLIPID:rank SWISSLIPID:Species .
  # SwissLipids Species name
  ?startId rdfs:label ?startName .  
  # Hierarchical search
  ?id rdfs:subClassOf+ ?startId .
  ?id SWISSLIPID:rank ?rank .
  VALUES ?rank { SWISSLIPID:Isomeric_Subspecies SWISSLIPID:Molecular_Subspecies SWISSLIPID:Structural_Subspecies}
  ?id rdfs:label ?name .
} 
ORDER BY DESC(?rank) ?startId`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY","STR","IF"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 26",slug:"SwissLipids_26",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Lipids produced in different organisms",context:null,inidces:[],query:`PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?swisslipid  ?organism {
  ?swisslipid owl:equivalentClass ?chebi .
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?rhea rh:side/rh:contains/rh:compound ?compound .
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?metabolite .
  }
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?catalyticActivityAnnotation up:catalyticActivity/up:catalyzedReaction ?rhea .
    ?protein up:annotation ?catalyticActivityAnnotation ;
             up:organism ?organism .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","SERVICE","CONTAINS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 27",slug:"SwissLipids_27",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find the list of SwissLipids for all organisms and their Isomeric subspecies",context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?swisslipid  ?organism {
  ?swisslipid owl:equivalentClass ?chebi .
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?rhea rh:side/rh:contains/rh:compound ?compound .
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?metabolite .
  }
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?catalyticActivityAnnotation up:catalyticActivity/up:catalyzedReaction ?rhea .
    ?protein up:annotation ?catalyticActivityAnnotation ;
             up:organism ?organism .
  }
  ?swisslipidIsomericSubpecies rdfs:subClassOf ?swisslipid ;
        SWISSLIPID:rank SWISSLIPID:Isomeric_Subspecies .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","SERVICE","CONTAINS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 3",slug:"SwissLipids_3",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 3
SELECT ?startId ?startLabel ?rank ?id ?name ?citation
WHERE
{
  # Queried lipid species
  VALUES ?startId { SWISSLIPID:000056493 }
  ?startId rdfs:label ?startLabel .
 
  ?id rdfs:subClassOf+ ?startId .  
  VALUES ?rank { SWISSLIPID:Isomeric_Subspecies SWISSLIPID:Molecular_Subspecies SWISSLIPID:Structural_Subspecies}
  ?id SWISSLIPID:rank ?rank ;
	  rdfs:label ?name ;
      SWISSLIPID:citation ?citation .
      
} 
ORDER BY ?startId ?id DESC(?rank)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY","STR"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 34",slug:"SwissLipids_34",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Lipids affected by human enzymes",context:null,inidces:[],query:`PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?swisslipid  {
  ?swisslipid owl:equivalentClass ?chebi .
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?rhea rh:side/rh:contains/rh:compound ?compound .
	?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?metabolite . 
  }
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?catalyticActivityAnnotation up:catalyticActivity/up:catalyzedReaction ?rhea .
    ?protein up:annotation ?catalyticActivityAnnotation ;
             up:organism taxon:9606 .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","SERVICE","CONTAINS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 35",slug:"SwissLipids_35",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Lipids by fatty acid component (hexadecanoate) at position sn1",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>

SELECT 
?swisslipid 
WHERE {
  VALUES (?chebi) {
    (CHEBI:7896) #hexadecanoate
  }
  ?swisslipidHexadeconate owl:equivalentClass ?chebi .
  ?swisslipid SWISSLIPID:hasPart [
    SWISSLIPID:derived_from ?swisslipidHexadecanoate ;
    SWISSLIPID:position SWISSLIPID:sn1 
  ]
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","FROM"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 36",slug:"SwissLipids_36",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>


SELECT ?startId ?startName ?tail1 ?tailName1 ?tail2 ?tailName2 ?id ?name ?pubmed
WHERE {
  VALUES ?startId { SWISSLIPID:000121946 }
  ?startId rdfs:label ?startName .
  # tail components
  ?startId SWISSLIPID:haspart ?tail1 .
  ?startId SWISSLIPID:haspart ?tail2 .
  FILTER(!sameTerm(?tail1, ?tail2))
  ?tail1 rdfs:label ?tailName1 .
  ?tail2 rdfs:label ?tailName2 .

  # Retrieve lipids with similar tails
  ?id SWISSLIPID:haspart ?tail1 , ?tail2 .
  
  # Restrict to isomeric subspecies with PubMed citation(s)
  ?id SWISSLIPID:rank SWISSLIPID:Isomeric_Subspecies .
  ?id SWISSLIPID:citation ?pubmed .

  # Retrieve name
  ?id rdfs:label ?name .
}
ORDER BY ?id ?tail1 ?tail2`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","ORDER BY","WITH","STR","SAMETERM"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 37",slug:"SwissLipids_37",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 37
SELECT ?startId ?startName ?tail1 ?tailName1 ?tail2 ?tailName2 ?id ?name
WHERE {
  VALUES ?startId { SWISSLIPID:000121946 }
  ?startId rdfs:label ?startName .
  # tail components
  ?startId SWISSLIPID:haspart ?tail1 .
  ?startId SWISSLIPID:haspart ?tail2 .
  FILTER(!sameTerm(?tail1, ?tail2))
  ?tail1 rdfs:label ?tailName1 .
  ?tail2 rdfs:label ?tailName2 .

  # Retrieve lipids with similar tails
  ?id SWISSLIPID:haspart ?tail1 , ?tail2 .
  
  # Restrict to isomeric subspecies with PubMed citation(s)
  ?id SWISSLIPID:rank SWISSLIPID:Isomeric_Subspecies .

  # Retrieve name
  ?id rdfs:label ?name .
}
ORDER BY ?id ?tail1 ?tail2`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","ORDER BY","WITH","STR","SAMETERM"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 38",slug:"SwissLipids_38",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 38
SELECT ?startId ?startName ?tail1 ?tailName1 ?tail2 ?tailName2 ?id ?name ?pubmed
WHERE {
  VALUES ?startId { SWISSLIPID:000121946 }
  ?startId rdfs:label ?startName .
  # tail components
  ?startId SWISSLIPID:haspart ?tail1 .
  ?startId SWISSLIPID:haspart ?tail2 .
  FILTER(!sameTerm(?tail1, ?tail2))
  ?tail1 rdfs:label ?tailName1 .
  ?tail2 rdfs:label ?tailName2 .

  # Retrieve lipids with similar tails
  ?id SWISSLIPID:haspart ?tail1 , 
                  ?tail2 .
  
  # Restrict to isomeric subspecies with PubMed citation(s)
  ?id SWISSLIPID:rank SWISSLIPID:Isomeric_Subspecies .
  ?id SWISSLIPID:citation ?pubmed .

  # Retrieve name
  ?id rdfs:label ?name .
}
ORDER BY ?id ?tail1 ?tail2`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","VALUES","ORDER BY","WITH","STR","SAMETERM"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 4",slug:"SwissLipids_4",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 4
SELECT ?startId ?startName ?id ?name
WHERE
{
  VALUES ?startId { SWISSLIPID:000399814 SWISSLIPID:000000567}
  ?startId SWISSLIPID:rank SWISSLIPID:Class .
  ?startId rdfs:label ?startName .
  ?id rdfs:subClassOf+ ?startId .
  ?id SWISSLIPID:rank SWISSLIPID:Isomeric_Subspecies ;
	  rdfs:label ?name .    
} 
ORDER BY ?startId`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 5",slug:"SwissLipids_5",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>

# Example 5
SELECT ?startId ?startName ?id ?name ?chebi ?rhea ?rheaEquation 
WHERE
{
  # SwissLipids query (list of) identifier(s)
  VALUES ?startId { SWISSLIPID:000399814 }
  # name
  ?startId rdfs:label ?startName .
  # Retrieve children lipids 
  ?id rdfs:subClassOf* ?startId .
  ?id rdfs:label ?name .
  # ChEBI x-ref
  ?id owl:equivalentClass ?chebi .
  # federated query to Rhea
  SERVICE <https://sparql.rhea-db.org/sparql> {
  	?rhea rh:equation ?rheaEquation .
  	?rhea rh:side/rh:contains/rh:compound/rh:chebi ?chebi .
  }    
} 
ORDER BY  ?startId ?id`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","SERVICE","ORDER BY","IF","CONTAINS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 6",slug:"SwissLipids_6",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

# Example 6
SELECT ?startId ?startName ?chebi ?rhea ?rheaEquation ?uniprot ?uniprotName
WHERE
{
  # SwissLipids query (list of) identifier(s)
  VALUES ?startId { SWISSLIPID:000399814 }
  # name
  ?startId rdfs:label ?startName .
  # ChEBI x-ref
  ?startId owl:equivalentClass ?chebi .
  # federated query to Rhea
  SERVICE <https://sparql.rhea-db.org/sparql> {
  	?rhea rh:equation ?rheaEquation .
  	?rhea rh:side/rh:contains/rh:compound/rh:chebi ?chebi .
  }    
  # federated query to UniProt
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?uniprot up:reviewed true .
    ?uniprot up:recommendedName/up:fullName ?uniprotName .
    ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea .
  }
} 
ORDER BY ?startId`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","SERVICE","ORDER BY","IF","CONTAINS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 7",slug:"SwissLipids_7",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX SWISSLIPID: <https://swisslipids.org/rdf/SLM_>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

# Example 7
SELECT ?startId ?startLabel ?id ?name 
WHERE {
  # Queried lipid species
  VALUES ?startId { SWISSLIPID:000056871 }
  ?startId rdfs:label ?startLabel . 
  # Corresponding Isomeric subspecies
  ?id rdfs:subClassOf+ ?startId .  
  ?id SWISSLIPID:rank SWISSLIPID:Isomeric_Subspecies ;
	  rdfs:label ?name .
  # Mapped ChEBI
  ?id owl:equivalentClass ?chebi .  
  # federated query to Rhea
  SERVICE <https://sparql.rhea-db.org/sparql> {
  	?rhea rh:equation ?rheaEquation .
  	?rhea rh:side/rh:contains/rh:compound/rh:chebi ?chebi .
  }    
  # federated query to UniProt
  SERVICE <https://sparql.uniprot.org/sparql> {
    ?uniprot up:reviewed true .
    ?uniprot up:recommendedName/up:fullName ?uniprotName .
    ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea .
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","SERVICE","CONTAINS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 8",slug:"SwissLipids_8",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX lipidmaps: <https://www.lipidmaps.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 8
SELECT ?slm ?slmName ?lipidmaps
WHERE {
  ?slm rdfs:label ?slmName . 
  # Corresponding LIPID MAPS
  ?slm rdfs:seeAlso ?lipidmaps .
  FILTER (strstarts(str(?lipidmaps), str(lipidmaps:))) .
}
ORDER BY ?slm`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","ORDER BY","STR","STRSTARTS"],category:"sib-swiss SwissLipids"},{name:"SwissLipids - 9",slug:"SwissLipids_9",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX lipidmaps: <https://www.lipidmaps.org/rdf/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

# Example 9
SELECT ?lipidmaps ?slm ?slmName 
WHERE {
  VALUES ?lipidmaps {lipidmaps:LMSP01020001 lipidmaps:LMST01010069 lipidmaps:LMST04030222}
  ?slm rdfs:label ?slmName . 
  # Corresponding SwissLipds entries
  ?slm rdfs:seeAlso ?lipidmaps .
}
ORDER BY ?lipidmaps`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","ORDER BY"],category:"sib-swiss SwissLipids"}],gP=[{name:"UniProt - 100 uniprot organelles or plasmids",slug:"UniProt_100_uniprot_organelles_or_plasmids",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List the proteins encoded by a gene that is located in an organelle other than the nucleus, or on a plasmid rather than a chromosome. In these cases the gene location is stored with encodedIn properties. Note that if a plasmid has several names, they are listed as multiple <em>rdfs:label</em> properties.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?protein 
    ?plasmidOrOrganelle
    ?label
WHERE {
    ?protein a up:Protein ;
      up:encodedIn ?plasmidOrOrganelle .
    OPTIONAL {
        ?plasmidOrOrganelle rdfs:label ?label .
    }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL"],category:"sib-swiss UniProt"},{name:"UniProt - 101 uniprot potential isoforms",slug:"UniProt_101_uniprot_potential_isoforms",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all human UniProtKB entries and their computationaly mapped potential isoforms.",context:null,inidces:[],query:`PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?entry ?sequence ?isCanonical
WHERE {
  # We don't want to look into the UniParc graph which will 
  # confuse matters
  GRAPH <http://sparql.uniprot.org/uniprot> {
      # we need the UniProt entries that are human
      ?entry a up:Protein ;
        up:organism taxon:9606 ;
      # and we select the computationally mapped sequences
        up:potentialSequence ?sequence .
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","ALL"],category:"sib-swiss UniProt"},{name:"UniProt - 102 uniprot primary accession",slug:"UniProt_102_uniprot_primary_accession",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Extracting an UniProtKB primary accession from our IRIs. Is done with a bit of string manipulation. While UniProt primary accession are unique within UniProtKB they may be reused by accident or itentionally by other data sources. If we provided them as strings (not IRI) and if you used them in a query that way, you might accidentaly retrieve completely wrong records.",context:null,inidces:[],query:`PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?primaryAccession
  ?protein
WHERE {
  ?protein a up:Protein .
  BIND(substr(str(?protein), strlen(str(uniprotkb:))+1) AS ?primaryAccession)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","STR","STRLEN","SUBSTR"],category:"sib-swiss UniProt"},{name:"UniProt - 103 uniprot proteome location of gene",slug:"UniProt_103_uniprot_proteome_location_of_gene",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List UniProtKB proteins with genetic replicon that they are encoded on using the Proteome data.",context:null,inidces:[],query:`PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  DISTINCT
    ?proteomeData
    ?replicon
    ?proteome  
WHERE {
  # reviewed entries (UniProtKB/Swiss-Prot)
  ?protein up:reviewed true . 
  # restricted to Human taxid
  ?uniprot up:organism taxon:9606 . 
  ?uniprot up:proteome ?proteomeData .
  BIND( strbefore( str(?proteomeData), "#" ) as ?proteome )
  BIND( strafter( str(?proteomeData), "#" ) as ?replicon )
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","DISTINCT","STR"],category:"sib-swiss UniProt"},{name:"UniProt - 104 uniprot recomended protein full name",slug:"UniProt_104_uniprot_recomended_protein_full_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"The recommended protein full names for UniProtKB entries",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein
  ?fullName
WHERE {
  ?protein a up:Protein ;
           up:recommendedName ?recommendedName .
  ?recommendedName up:fullName ?fullName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 105 uniprot recomended protein short name",slug:"UniProt_105_uniprot_recomended_protein_short_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"The recommended protein short names for UniProtKB entries",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein 
  ?fullName
WHERE {
  ?protein a up:Protein ;
           up:recommendedName ?recommendedName .
  ?recommendedName up:shortName ?fullName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 106 uniprot reviewed or not",slug:"UniProt_106_uniprot_reviewed_or_not",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all UniProtKB proteins and if they are reviewed (Swiss-Prot) or unreviewed (TrEMBL)",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
  ?reviewed
WHERE {
  ?protein a up:Protein . 
  ?protein up:reviewed ?reviewed . 
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 107 uniprot sequences and mark which is cannonical for human",slug:"UniProt_107_uniprot_sequences_and_mark_which_is_cannonical_for_human",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all human UniProtKB entries and their sequences, marking if the sequence listed is the cannonical sequence of the matching entry.",context:null,inidces:[],query:`PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?entry ?sequence ?isCanonical
WHERE {
  # We don't want to look into the UniParc graph which will 
  # confuse matters
  GRAPH <http://sparql.uniprot.org/uniprot> {
      # we need the UniProt entries that are human
      ?entry a up:Protein ;
	up:organism taxon:9606 ;
      	up:sequence ?sequence .
      # If the sequence is a "Simple_Sequence" it is likely to be the 
      # cannonical sequence
      OPTIONAL {
       	?sequence a up:Simple_Sequence .
        BIND(true AS ?likelyIsCanonical)
      }
      # unless we are dealing with an external isoform
      # see https://www.uniprot.org/help/canonical_and_isoforms
      OPTIONAL {
       	FILTER(?likelyIsCanonical)
        ?sequence a up:External_Sequence .
        BIND(true AS ?isComplicated)
      }
      # If it is an external isoform it's id would not match the 
      # entry primary accession
      BIND(IF(?isComplicated, STRENDS(STR(?entry), STRBEFORE(SUBSTR(STR(?sequence), 34),'-')),?likelyIsCanonical) AS ?isCanonical)
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","GRAPH","BIND","WITH","STR","IF","SUBSTR","STRENDS"],category:"sib-swiss UniProt"},{name:"UniProt - 108 uniprot signature match start end",slug:"UniProt_108_uniprot_signature_match_start_end",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all InterPro member database signature match start and end for a specific UniProtKB protein.",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?protein ?interproMemberDatabaseXref ?matchStart ?matchEnd
WHERE{
  GRAPH <http://sparql.uniprot.org/uniprot> {
    VALUES ?protein {<http://purl.uniprot.org/uniprot/P05067>} .
    ?protein rdfs:seeAlso ?sa .
  }
  GRAPH <http://sparql.uniprot.org/uniparc> {
    ?uniparc up:sequenceFor ?protein ;
      rdfs:seeAlso ?interproMemberDatabaseXref .
    ?interproDatabaseXref up:signatureSequenceMatch ?sam .
    ?sam faldo:begin ?sab ;
      faldo:end ?sae .
    ?sab faldo:position ?matchStart ;
      faldo:reference ?uniparc .
    ?sae  faldo:position ?matchEnd ;
      faldo:reference ?uniparc .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","VALUES"],category:"sib-swiss UniProt"},{name:"UniProt - 109 uniprot transporter in liver",slug:"UniProt_109_uniprot_transporter_in_liver",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find human transporter proteins in reviewed UniProtKB/Swiss-Prot, that are expressed in the liver (Uses Bgee and UBERON).",context:null,inidces:[],query:`PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX uberon: <http://purl.obolibrary.org/obo/uo#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?rhea
  ?protein
  ?anat
WHERE
{
  GRAPH <https://sparql.rhea-db.org/rhea> {
    ?rhea rh:isTransport true .
  }
  ?protein up:annotation ?ann .
  ?protein up:organism taxon:9606 .
  ?ann up:catalyticActivity ?ca .
  ?ca up:catalyzedReaction ?rhea .
  BIND(uberon:0002107 AS ?anat)
  SERVICE <https://www.bgee.org/sparql/> {
    ?seq genex:isExpressedIn ?anat .
    ?seq lscr:xrefUniprot ?protein .
    ?seq orth:organism ?organism .
    ?organism obo:RO_0002162 taxon:9606 .
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","BIND","SERVICE","STR"],category:"sib-swiss UniProt"},{name:"UniProt - 10 human variant leading to transposition of tyrosine to phenylalanine",slug:"UniProt_10_human_variant_leading_to_transposition_of_tyrosine_to_phenylalanine",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all human UniProtKB entries with a sequence variant that leads to a tyrosine to phenylalanine substitution",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?annotation ?begin ?text
WHERE
{
        ?protein a up:Protein ;
            up:organism taxon:9606 ; 
            up:annotation ?annotation .
        ?annotation a up:Natural_Variant_Annotation ;
            rdfs:comment ?text ;
            up:substitution ?substitution ;
            up:range/faldo:begin
                [ faldo:position ?begin ;
                  faldo:reference ?sequence ] .
        ?sequence rdf:value ?value .
        BIND (substr(?value, ?begin, 1) as ?original) .
        FILTER(?original = 'Y' && ?substitution = 'F') .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","STR","SUBSTR"],category:"sib-swiss UniProt"},{name:"UniProt - 110 uniprot unamed plasmids",slug:"UniProt_110_uniprot_unamed_plasmids",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Sometimes it is known that a gene encoding a protein UniProtKB is located on a plasmid or an organelle, but the name of the plasmid is unknown.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?protein 
    ?plasmidOrOrganelle
    ?label
WHERE {
    ?protein a up:Protein ;
      up:encodedIn ?plasmidOrOrganelle .
    OPTIONAL {
        ?plasmidOrOrganelle rdfs:label ?label .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL"],category:"sib-swiss UniProt"},{name:"UniProt - 111 uniprot make chain sequence fasta",slug:"UniProt_111_uniprot_make_chain_sequence_fasta",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Given an UniProt chain identifier, generate the matching sequence as a FASTA",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX faldo: <http://biohackathon.org/resource/faldo#>

SELECT (CONCAT('>', ?chainSeqId, '\\\\n', (SUBSTR(?iupacAA, ?begin, (?end-?begin+1))))  AS ?chainFasta)
WHERE {
  BIND("PRO_0000268053" AS ?chainSeqId) 
  BIND(IRI(CONCAT("http://purl.uniprot.org/annotation/", ?chainSeqId)) AS ?annId)
  ?annId up:range ?range .
  ?range faldo:begin [ faldo:reference ?reference ; faldo:position ?begin ] ;
     faldo:end [ faldo:position ?end ] .
  ?reference rdf:value ?iupacAA .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","STR","IRI","CONCAT","SUBSTR"],category:"sib-swiss UniProt"},{name:"UniProt - 112 count human transporters",slug:"UniProt_112_count_human_transporters",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Count the number of Human Transporter Proteins",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX faldo: <http://biohackathon.org/resource/faldo#>

SELECT  
  (COUNT(DISTINCT ?protein) AS ?humanTransportEnzymes)
WHERE {
  GRAPH <http://sparql.uniprot.org/uniprot> {
    ?protein up:organism taxon:9606 ;
             up:annotation ?a .
    ?a a up:Catalytic_Activity_Annotation ;
      up:catalyticActivity ?ca .
    ?ca up:catalyzedReaction ?rhea .
  }
  GRAPH <https://sparql.rhea-db.org/rhea>{
  	?rhea rh:isTransport true .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","DISTINCT","STR","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 113 UniProtKB Swiss-Prot entries annotated with CC-CA Rhea involving lipids",slug:"UniProt_113_UniProtKB_Swiss-Prot_entries_annotated_with_CC-CA_Rhea_involving_lipids",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"UniProtKB/Swiss-Prot entries annotated with CC-CA/Rhea involving lipids",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
# select all pairs (sprot,rhea) where rhea involves a lipid
PREFIX rh:<http://rdf.rhea-db.org/>
PREFIX CHEBI:<http://purl.obolibrary.org/obo/CHEBI_>
PREFIX up:<http://purl.uniprot.org/core/>
PREFIX uniprotkb:<http://purl.uniprot.org/uniprot/>

SELECT 
  (COUNT(*) as ?cnt) 
WHERE 
{
  {
    SELECT ?protein ?rhea 
    WHERE {
      SERVICE <https://sparql.rhea-db.org/sparql> {
        SELECT 
          distinct 
           ?rhea 
        WHERE {
          ?rhea rdfs:subClassOf rh:Reaction .
          ?rhea rh:status rh:Approved .
          ?rhea rh:side ?reactionSide .
          ?reactionSide rh:contains ?participant .
          ?participant rh:compound ?compound .
          ?compound rh:chebi ?chebi .
          ?chebi rdfs:subClassOf+ CHEBI:18059 .
        }
      }
    }
  }
  ?protein a up:Protein .
  ?protein up:reviewed true .
  ?protein up:annotation ?a .
  ?a a up:Catalytic_Activity_Annotation .
  ?a up:catalyticActivity ?ca .
  ?ca up:catalyzedReaction ?rhea .
}`,ontologies:["DCTERMS","EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","ALL","CONTAINS","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 114 Number of EC numbers described at protein domain and component levels",slug:"UniProt_114_Number_of_EC_numbers_described_at_protein_domain_and_component_levels",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX up:<http://purl.uniprot.org/core/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>

SELECT 
  (count(distinct ?protein) as ?proteinCount)
WHERE
{
  ?protein up:reviewed true .
  OPTIONAL{?protein up:enzyme ?ecNumber}
  OPTIONAL{?protein up:domain/up:enzyme ?ecNumber}
  OPTIONAL{?protein up:component/up:enzyme ?ecNumber}
  FILTER (BOUND(?ecNumber))
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","DISTINCT","BOUND","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 115 Number of complete EC numbers described at protein level at domain levels at component level",slug:"UniProt_115_Number_of_complete_EC_numbers_described_at_protein_level_at_domain_levels_at_component_level",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX up:<http://purl.uniprot.org/core/>
PREFIX ec:<http://purl.uniprot.org/enzyme/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>

SELECT 
  (count(distinct ?ecProt) as ?ecNumberInProteinCount)
  (count(distinct ?ecDomain) as ?ecNumberInDomainCount)
  (count(distinct ?ecComponent) as ?ecNumberInComponentCount)
WHERE
{
  ?protein up:reviewed true .
  OPTIONAL {?protein up:enzyme ?ecProt . } .
  OPTIONAL {?protein up:domain/up:enzyme ?ecDomain . } .
  OPTIONAL {?protein up:component/up:enzyme ?ecComponent . } .
}`,ontologies:["EX","RDFS","SCHEMA"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","DISTINCT","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 116 biosodafrontend rabit mouse orthologs",slug:"UniProt_116_biosodafrontend_rabit_mouse_orthologs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Rabbit's proteins encoded by genes that are orthologous to Mouse's HBB-Y gene and their cross reference links to Uniprot",context:null,inidces:[],query:`PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
SELECT DISTINCT ?PROTEIN_1 ?PROTEIN_2 ?UNIPROT_XREF_1 ?UNIPROT_XREF_2 WHERE {
	?taxon_1 up:commonName 'Mouse' .
	?taxon_2 up:commonName 'Rabbit' .
	SERVICE <https://sparql.omabrowser.org/sparql/> {
		?cluster a orth:OrthologsCluster .
		?cluster orth:hasHomologousMember ?node1 .
		?cluster orth:hasHomologousMember ?node2 .
		?node2 orth:hasHomologousMember* ?PROTEIN_2 .
		?node1 orth:hasHomologousMember* ?PROTEIN_1 .
		?PROTEIN_1 a orth:Protein .
		?PROTEIN_1 orth:organism/obo:RO_0002162 ?taxon_1 ;
			rdfs:label 'HBB-Y' ;
			lscr:xrefUniprot ?UNIPROT_XREF_1 .
		?PROTEIN_2 a orth:Protein .
		?PROTEIN_2 orth:organism/obo:RO_0002162 ?taxon_2 .
		?PROTEIN_2 lscr:xrefUniprot ?UNIPROT_XREF_2 .
		FILTER ( ?node1 != ?node2 )
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","DISTINCT"],category:"sib-swiss UniProt"},{name:"UniProt - 117 biosodafrontend glioblastoma orthologs rat",slug:"UniProt_117_biosodafrontend_glioblastoma_orthologs_rat",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Which are the proteins associated with glioblastoma and the orthologs expressed in the rat brain?",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX sio: <http://semanticscience.org/resource/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX genex: <http://purl.org/genex#>
SELECT DISTINCT ?protein ?orthologous_protein ?gene ?annotation_text WHERE {
  {
  	SELECT ?protein ?annotation_text WHERE {
      ?protein a up:Protein ;
          up:organism taxon:9606 ;
          up:annotation ?annotation .
      ?annotation rdfs:comment ?annotation_text .
      ?annotation a up:Disease_Annotation .
      FILTER CONTAINS (?annotation_text, "glioblastoma")
    }
  }
  SERVICE <https://sparql.omabrowser.org/sparql/> {
    SELECT ?orthologous_protein ?protein ?gene WHERE {
    ?protein_OMA a orth:Protein .
    ?orthologous_protein a orth:Protein .
    ?cluster a orth:OrthologsCluster .
    ?cluster orth:hasHomologousMember ?node1 .
    ?cluster
    orth:hasHomologousMember ?node2 .
    ?node2 orth:hasHomologousMember* ?protein_OMA .
    ?node1 orth:hasHomologousMember* ?orthologous_protein .
    ?orthologous_protein orth:organism/obo:RO_0002162 taxon:10116 . # rattus norvegicus
    ?orthologous_protein sio:SIO_010079 ?gene .
    ?protein_OMA lscr:xrefUniprot ?protein .
    FILTER(?node1 != ?node2)
		}
	}
  SERVICE <https://www.bgee.org/sparql/> {
    ?gene genex:isExpressedIn ?a .
    ?a rdfs:label "brain" .
    ?gene orth:organism ?s . 
    ?s obo:RO_0002162 taxon:10116.
	}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","DISTINCT","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 118 biosodafrontend rat brain human cancer",slug:"UniProt_118_biosodafrontend_rat_brain_human_cancer",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"What are the Homo sapiens genes associated with cancer and their orthologs expressed in the Rattus norvegicus brain?",context:null,inidces:[],query:`PREFIX up:<http://purl.uniprot.org/core/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>
PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#>
PREFIX orth:<http://purl.org/net/orth#>
PREFIX dcterms:<http://purl.org/dc/terms/>
PREFIX obo:<http://purl.obolibrary.org/obo/>
PREFIX lscr:<http://purl.org/lscr#>
PREFIX genex:<http://purl.org/genex#>
PREFIX sio: <http://semanticscience.org/resource/>
SELECT ?gene ?orthologous_protein2 WHERE {
  {
    SELECT ?protein1 WHERE {
      ?protein1 a up:Protein;
        up:organism/up:scientificName 'Homo sapiens' ;
        up:annotation ?annotation .
      ?annotation rdfs:comment ?annotation_text.
      ?annotation a up:Disease_Annotation .
      FILTER CONTAINS (?annotation_text, "cancer")
    }
  }
  SERVICE <https://sparql.omabrowser.org/sparql/> {
    SELECT ?orthologous_protein2 ?protein1 ?gene WHERE {
      ?protein_OMA a orth:Protein .
      ?orthologous_protein2 a orth:Protein .
      ?cluster a orth:OrthologsCluster .
      ?cluster orth:hasHomologousMember ?node1 .
      ?cluster orth:hasHomologousMember ?node2 .
      ?node2 orth:hasHomologousMember* ?protein_OMA .
      ?node1 orth:hasHomologousMember* ?orthologous_protein2 
      .?orthologous_protein2 orth:organism/obo:RO_0002162/up:scientificName 'Rattus norvegicus' .
      ?orthologous_protein2 sio:SIO_010079 ?gene .
      ?protein_OMA lscr:xrefUniprot ?protein1 .
      FILTER(?node1 != ?node2)
    }
  }
  SERVICE <https://www.bgee.org/sparql/> {
    ?gene genex:isExpressedIn ?anatEntity .
    ?anatEntity rdfs:label 'brain' .
    ?gene orth:organism ?org . 
    ?org obo:RO_0002162 taxon:10116 .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","SERVICE","IF","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 119 uniref distinct ec in seed",slug:"UniProt_119_uniref_distinct_ec_in_seed",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Distinct Enzymes in UniRef50 seed entries",context:null,inidces:[],query:`PREFIX up:<http://purl.uniprot.org/core/>
SELECT 
  (COUNT(DISTINCT ?enzyme) AS ?distinctEnzymesInUniRef50Seed)
WHERE {
  GRAPH <http://sparql.uniprot.org/uniprot>{
  	?protein ( up:enzyme | up:domain/up:enzyme | up:component/up:enzyme ) ?enzyme .
  }
  GRAPH <http://sparql.uniprot.org/uniref>{
    ?protein up:seedFor ?cluster .
    ?cluster up:identity 0.5 .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","DISTINCT","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 11 entries with transmem on cannonical sequence",slug:"UniProt_11_entries_with_transmem_on_cannonical_sequence",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all UniProtKB entries with annotated transmembrane regions and the regions' begin and end coordinates on the canonical sequence",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?begin ?end
WHERE 
{
	?protein a up:Protein .
	?protein up:annotation ?annotation .
	?annotation a up:Transmembrane_Annotation .
	?annotation up:range ?range .
	?range faldo:begin/faldo:position ?begin .
	?range faldo:end/faldo:position ?end
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 120 for taxon find reference proteomes",slug:"UniProt_120_for_taxon_find_reference_proteomes",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"For a ncbi taxon identifier as a digit find it's reference proteome(s) if they exist",context:null,inidces:[],query:`PREFIX up:<http://purl.uniprot.org/core/>
PREFIX taxon:<http://purl.uniprot.org/taxonomy/>

SELECT ?taxon ?reference_proteome
WHERE
{
  VALUES (?taxid) {
    (623) # Shigella flexneri
    (633) # Yersinia pseudotuberculosis
  } 
  # Convert the digit to a correct IRI
  BIND(IRI(CONCAT(STR(taxon:), ?taxid)) AS ?taxon)
  ?taxon up:scientificName ?taxonName .
  OPTIONAL {
    ?reference_proteome a up:Reference_Proteome .
    ?reference_proteome up:organism ?taxon .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","BIND","VALUES","STR","IRI","IF","CONCAT"],category:"sib-swiss UniProt"},{name:"UniProt - 121 proteins and diseases linked",slug:"UniProt_121_proteins_and_diseases_linked",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all UniProtKB proteins and the diseases are annotated to be related.",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT
	?protein
	?disease
WHERE {
	?protein a up:Protein ;
    	up:annotation ?annotation .
	?annotation a up:Disease_Annotation ;
    	up:disease ?disease .
	?disease a up:Disease .
}`,ontologies:["EX","EX120","RDFS","SCHEMA","SH","UP"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 122 enzymes transporting lipids in reviewed human",slug:"UniProt_122_enzymes_transporting_lipids_in_reviewed_human",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all UniProtKB SwissProt entries, facilitation the transport of lipids.",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>

SELECT
  ?protein
  ?chebi
WHERE {
  GRAPH <https://sparql.rhea-db.org/rhea> {
    ?rhea rh:isTransport true .
    ?rhea rh:side/rh:contains/rh:compound ?compound .
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  }
  # CHEBI:18059 is the class for all Lipids
  ?chebi rdfs:subClassOf* CHEBI:18059 .
 
  
  # Select human reviewed entries from Swiss-Prot
  ?protein up:reviewed true ; 
    up:organism taxon:9606 .
   # Link protein to catalytic activity, then to Rhea reaction
  ?protein up:annotation?annotation .
  ?annotation up:catalyticActivity?catalytic_activity_obj .
  ?catalytic_activity up:catalyzedReaction?rhea .
}`,ontologies:["EX","EX120","RDFS","SCHEMA","SH","UP"],sparqlConcepts:["SELECT","WHERE","GRAPH","FROM","ALL","STR","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 12 entries integrated on date",slug:"UniProt_12_entries_integrated_on_date",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all UniProtKB entries that were integrated on the 30th of November 2010",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT ?protein
WHERE
{
	?protein a up:Protein . 
	?protein up:created '2010-11-30'^^xsd:date
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","CREATE"],category:"sib-swiss UniProt"},{name:"UniProt - 17 average count to PDB",slug:"UniProt_17_average_count_to_PDB",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the average number of cross-references to the <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT (AVG(?linksToPdbPerEntry) AS ?avgLinksToPdbPerEntry)
WHERE
{
	SELECT ?protein (COUNT(DISTINCT ?db) AS ?linksToPdbPerEntry)
	WHERE
	{
		?protein a up:Protein .
		?protein rdfs:seeAlso ?db .
		?db up:database <http://purl.uniprot.org/database/PDB> .
	}
	GROUP BY ?protein ORDER BY DESC(?linksToPdbPerEntry)
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","DISTINCT","COUNT","AVG"],category:"sib-swiss UniProt"},{name:"UniProt - 18 top level ec classification group by count",slug:"UniProt_18_top_level_ec_classification_group_by_count",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the number of UniProtKB entries for each of the EC (Enzyme Commission) top level categories",context:null,inidces:[],query:`PREFIX ec: <http://purl.uniprot.org/enzyme/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?ecClass (COUNT(?protein) as ?size)
WHERE
{
    VALUES (?ecClass) {(ec:1.-.-.-) (ec:2.-.-.-) (ec:3.-.-.-) (ec:4.-.-.-) (ec:5.-.-.-) (ec:6.-.-.-) (ec:7.-.-.-)} .
    ?protein ( up:enzyme | up:domain/up:enzyme | up:component/up:enzyme ) ?enzyme .
    # Enzyme subclasses are materialized, do not use rdfs:subClassOf+
    ?enzyme rdfs:subClassOf ?ecClass .
}
GROUP BY ?ecClass ORDER BY ?ecClass`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","GROUP BY","ORDER BY","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 19 natural variants associated with pubmed id",slug:"UniProt_19_natural_variants_associated_with_pubmed_id",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all natural variant annotations if associated via an evidence tag to an article with a PubMed identifier",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?accession
    ?annotation_acc 
    ?pubmed
WHERE
{
        ?protein a up:Protein ;
            up:annotation ?annotation .
        ?annotation a up:Natural_Variant_Annotation .
        ?linkToEvidence rdf:object ?annotation ;
                        up:attribution ?attribution .
        ?attribution up:source ?source .
        ?source a up:Journal_Citation .
  BIND(SUBSTR(STR(?protein),33) AS ?accession)
  BIND(IF(CONTAINS(STR(?annotation), "#SIP"), SUBSTR(STR(?annotation),33), SUBSTR(STR(?annotation),36))AS?annotation_acc)
  BIND(SUBSTR(STR(?source),35) AS ?pubmed)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","STR","IF","SUBSTR","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 1 select all taxa used in uniprot",slug:"UniProt_1_select_all_taxa_used_in_uniprot",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all taxa from the <a href=\\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?taxon
FROM <http://sparql.uniprot.org/taxonomy>
WHERE
{
    ?taxon a up:Taxon .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FROM"],category:"sib-swiss UniProt"},{name:"UniProt - 20 how often citation used in evidence tag",slug:"UniProt_20_how_often_citation_used_in_evidence_tag",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find how often an article in PubMed was used in an evidence tag in a human protein (ordered by most used to least)",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?source 
    (COUNT(?attribution) AS ?attribitions)
WHERE
{
        ?protein a up:Protein ;
            up:organism taxon:9606 ;
            up:annotation ?annotation .
        ?linkToEvidence rdf:object ?annotation ;
                        up:attribution ?attribution .
        ?attribution up:source ?source .
        ?source a up:Journal_Citation .
} GROUP BY ?source ORDER BY DESC(COUNT(?attribution))`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","ORDER BY","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 21 where are genetic disease related proteins in a cell",slug:"UniProt_21_where_are_genetic_disease_related_proteins_in_a_cell",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find where disease related proteins are known to be located in the cell",context:null,inidces:[],query:`PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>


SELECT 
    ?protein 
    ?disease 
    ?location_inside_cell 
    ?cellcmpt
WHERE
{
    ?protein up:annotation ?diseaseAnnotation , ?subcellAnnotation .
    ?diseaseAnnotation up:disease/skos:prefLabel ?disease .
    ?subcellAnnotation up:locatedIn/up:cellularComponent ?cellcmpt .
    ?cellcmpt skos:prefLabel ?location_inside_cell .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 22 go term labels per go category for multiple proteins",slug:"UniProt_22_go_term_labels_per_go_category_for_multiple_proteins",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"For two accession numbers (ACs) find the GO term labels and group them into GO process,function and component",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX GO:<http://purl.obolibrary.org/obo/GO_>

SELECT 
    (CONCAT(SUBSTR(STR(?protein), 33)) AS ?uniprot)
    (GROUP_CONCAT(?celtype; separator=";") AS ?celtypes)
    (GROUP_CONCAT(?biotype; separator=";") AS ?biotypes)
    (GROUP_CONCAT(?moltype; separator=";") AS ?moltypes)
WHERE
{
    VALUES (?ac) {("Q6GZX4") ("Q96375")}
    BIND (IRI(CONCAT("http://purl.uniprot.org/uniprot/",?ac)) AS ?protein)
    ?protein a up:Protein .
    ?protein up:classifiedWith ?goTerm .
    #Determine if the type is biological_process
    OPTIONAL {
        ?goTerm rdfs:subClassOf GO:0008150 .
        ?goTerm rdfs:label ?biotype .
    }
    #Determine if the type is cellular_component
    OPTIONAL {
        ?goTerm rdfs:subClassOf GO:0005575 .
        ?goTerm rdfs:label ?celtype .
    }
    #Determine if the type is molecular_function
    OPTIONAL {
        ?goTerm rdfs:subClassOf GO:0003674 .
        ?goTerm rdfs:label ?moltype .
    }
    #Filter out the uniprot keywords
    FILTER(bound(?biotype) || bound(?celtype) || bound(?moltype))
} GROUP BY ?protein`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","OPTIONAL","BIND","VALUES","GROUP BY","WITH","STR","BOUND","IRI","IF","CONCAT","SUBSTR","MIN","GROUP_CONCAT"],category:"sib-swiss UniProt"},{name:"UniProt - 23 human proteins related to kinase activity",slug:"UniProt_23_human_proteins_related_to_kinase_activity",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Number of reviewed entries (UniProtKB/Swiss-Prot) that are related to <a href=\\",context:null,inidces:[],query:`PREFIX GO: <http://purl.obolibrary.org/obo/GO_>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
    (COUNT(DISTINCT(?protein)) AS ?pc)
WHERE
{   
    ?protein rdf:type up:Protein ;
        up:reviewed true  ;
        up:organism taxon:9606 ;
        up:classifiedWith|(up:classifiedWith/rdfs:subClassOf) GO:0016301 .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","WITH","IF","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 24 uniprot release in endpoint",slug:"UniProt_24_uniprot_release_in_endpoint",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find the release number of the UniProt data that is currently being queried",context:null,inidces:[],query:`SELECT ?version
FROM <https://sparql.uniprot.org/.well-known/void>
WHERE
{
    [] <http://purl.org/pav/version> ?version
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FROM","NOW"],category:"sib-swiss UniProt"},{name:"UniProt - 25 protein name of any kind with value",slug:"UniProt_25_protein_name_of_any_kind_with_value",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find UniProtKB entry which has a protein name 'HLA class I histocompatibility antigen, B-73 alpha chain'",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?anyKindOfName
WHERE
{
		?protein a up:Protein .
		?protein (up:recommendedName|up:alternativeName) ?structuredName .
		?structuredName ?anyKindOfName  "HLA class I histocompatibility antigen, B alpha chain" .
		?anyKindOfName rdfs:subPropertyOf up:structuredNameType .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","STR"],category:"sib-swiss UniProt"},{name:"UniProt - 26 component HLA class I histocompatibility domain",slug:"UniProt_26_component_HLA_class_I_histocompatibility_domain",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find UniProtKB entry, or an UniProtKB entries domain or component which has a name 'HLA class I histocompatibility antigen, B-73 alpha chain'",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?anyKindOfName
WHERE
{
		?protein a up:Protein .
		?protein (up:recommendedName|up:alternativeName)|((up:domain|up:component)/(up:recommendedName|up:alternativeName)) ?structuredName .
		?structuredName ?anyKindOfName  "HLA class I histocompatibility antigen, B-73 alpha chain" .
		?anyKindOfName rdfs:subPropertyOf up:structuredNameType .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","STR"],category:"sib-swiss UniProt"},{name:"UniProt - 27 all names of protein",slug:"UniProt_27_all_names_of_protein",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all names associated with UniProtKB entry P05067, and if the name is associated with the entry it's domains or its components",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
  ?anyKindOfName 
  ?names 
  ?partType
WHERE
{
  BIND(<http://purl.uniprot.org/uniprot/P05067> AS ?protein)
  ?protein a up:Protein .
  {
    ?protein (up:recommendedName|up:alternativeName) ?structuredName .
  }
    UNION
  {
    VALUES(?partType){(up:domain) (up:component)}
    ?protein ?partType ?part .
    ?part (up:recommendedName|up:alternativeName) ?structuredName .
  }
  ?structuredName ?anyKindOfName  ?names .
  ?anyKindOfName rdfs:subPropertyOf up:structuredNameType .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","BIND","VALUES","STR"],category:"sib-swiss UniProt"},{name:"UniProt - 28 all entries encoded on chromosome of proteome",slug:"UniProt_28_all_entries_encoded_on_chromosome_of_proteome",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Get the list of UniProtKB entries for the chromosome of proteome <a href=\\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?protein
  ?proteome 
WHERE
{
  ?protein a up:Protein ;
           up:reviewed true ;
           up:proteome ?proteome .
  VALUES (?proteome) {(<http://purl.uniprot.org/proteomes/UP000000625#Chromosome>)}
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES"],category:"sib-swiss UniProt"},{name:"UniProt - 29 search uniprot in japanese with allie",slug:"UniProt_29_search_uniprot_in_japanese_with_allie",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Use <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?englishLabelStr
WHERE {
    SERVICE <https://data.allie.dbcls.jp/sparql>{
        ?x rdfs:label "アミロイド前駆体タンパク質"@ja ;
            rdfs:label ?englishLabel .
        FILTER(lang(?englishLabel) = "en")
    }
    BIND (STR(?englishLabel) AS ?englishLabelStr)
    ?protein a up:Protein .
    {
        ?protein (up:recommendedName|up:alternativeName) ?structuredName .
    }
    UNION
    {
        VALUES(?partType){(up:domain) (up:component)}
            ?protein ?partType ?part .
        ?part (up:recommendedName|up:alternativeName) ?structuredName .
    }
    ?structuredName ?anyKindOfName  ?englishLabelStr .
    ?anyKindOfName rdfs:subPropertyOf up:structuredNameType .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","VALUES","SERVICE","ALL","STR","LANG"],category:"sib-swiss UniProt"},{name:"UniProt - 2 bacteria taxa and their scientific name",slug:"UniProt_2_bacteria_taxa_and_their_scientific_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all bacterial taxa and their scientific name from the <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?taxon ?name
WHERE
{
    ?taxon a up:Taxon .
    ?taxon up:scientificName ?name .
    # Taxon subclasses are materialized, do not use rdfs:subClassOf+
    ?taxon rdfs:subClassOf taxon:2 .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 30 merged loci",slug:"UniProt_30_merged_loci",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find UniProtKB entries with merged loci in <a href=\\",context:null,inidces:[],query:`PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein 
  (GROUP_CONCAT(?locusName; separator=',') AS ?locusNames)
WHERE 
{ 
  ?protein a up:Protein ;
    up:organism taxon:360910 ;
    up:encodedBy ?gene .
  ?gene up:locusName ?locusName .
} 
GROUP BY ?protein 
HAVING (COUNT(?locusName) > 1)`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","CONCAT","COUNT","GROUP_CONCAT"],category:"sib-swiss UniProt"},{name:"UniProt - 31 uniparc record with most subentries",slug:"UniProt_31_uniparc_record_with_most_subentries",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find UniParc records whose sequence point to the most database entries",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?sequence ?entries
WHERE
{
    SELECT 
        ?sequence 
        (COUNT(?entry) AS ?entries)
    WHERE
    {
        GRAPH <http://sparql.uniprot.org/uniparc> {
            ?sequence up:sequenceFor ?entry .
        }
    } GROUP BY ?sequence
} ORDER BY DESC(?entries)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","GROUP BY","ORDER BY","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 32 uniprot with more than x topodom",slug:"UniProt_32_uniprot_with_more_than_x_topodom",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find UniProtKB entries with more than 1 <a href=\\",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?protein 
    (GROUP_CONCAT(?comment; separator=", ") AS ?comments)
WHERE
{
    ?protein a up:Protein ;
            up:annotation ?annotation . 
    ?annotation rdf:type up:Topological_Domain_Annotation ;
            rdfs:comment ?comment .
} 
GROUP BY ?protein 
HAVING (COUNT(?annotation) > 1)`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","CONCAT","COUNT","GROUP_CONCAT"],category:"sib-swiss UniProt"},{name:"UniProt - 33 longest variant comment",slug:"UniProt_33_longest_variant_comment",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find longest comment text associated with a UniProtKB natural variant annotation",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?annotation ?comment
WHERE {
    ?annotation a up:Natural_Variant_Annotation ;
        rdfs:comment ?comment . 
} 
ORDER BY DESC(STRLEN(?comment))`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","STR","STRLEN"],category:"sib-swiss UniProt"},{name:"UniProt - 34 cooccurence count of topodom",slug:"UniProt_34_cooccurence_count_of_topodom",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find the co-occurence count of topological domain comment text in UniProtKB entries",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?comment1 
    ?comment2 
    (COUNT(?comment1) AS ?count1)
WHERE
{
    ?protein a up:Protein ;
               up:annotation ?annotation1 , 
                             ?annotation2 . 
    ?annotation1 rdf:type up:Topological_Domain_Annotation ;
        rdfs:comment ?rawComment1 .
    ?annotation2 rdf:type up:Topological_Domain_Annotation ;
        rdfs:comment ?rawComment2 . 
    BIND(IF(contains(?rawComment1, ';'), 
            STRBEFORE(?rawComment1,';'), 
            ?rawComment1) AS ?comment1)
    BIND(IF(contains(?rawComment2, ';'), 
            STRBEFORE(?rawComment2,';'), 
            ?rawComment2) AS ?comment2)
    FILTER(?annotation1 != ?annotation2)
} 
GROUP BY ?comment1 ?comment2 
ORDER BY DESC(COUNT(?comment1))`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","GROUP BY","ORDER BY","STR","IF","CONTAINS","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 35 similar proteins via uniref clusters",slug:"UniProt_35_similar_proteins_via_uniref_clusters",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find the similar proteins for UniProtKB entry P05067 sorted by UniRef cluster identity",context:null,inidces:[],query:`PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?similar ?identity
FROM <http://sparql.uniprot.org/uniref>
FROM <http://sparql.uniprot.org/uniprot>
WHERE
{
    BIND (uniprotkb:P05607 AS ?protein)
    ?cluster up:member ?member ;
             up:member/up:sequenceFor ?protein;
             up:identity ?identity .
    ?member up:sequenceFor ?similar .
    FILTER(!sameTerm(?similar, ?protein))
} 
ORDER BY DESC(?identity)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","ORDER BY","FROM","SAMETERM"],category:"sib-swiss UniProt"},{name:"UniProt - 36 orthologous proteins via orthodb",slug:"UniProt_36_orthologous_proteins_via_orthodb",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find the orthologous proteins for UniProtKB entry P05067 using the <a href=\\",context:null,inidces:[],query:`PREFIX orthodb: <http://purl.orthodb.org/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
  ?orthoGroup
  ?scientificName
  ?functionComment
  ?prefferedGeneName
  ((STRLEN(?value) - ?medianLength) as ?deviationFromMedianLength)
WHERE
{
  uniprotkb:P05067 a up:Protein ;
        up:organism/up:scientificName ?scientificName ;
        rdfs:seeAlso ?orthoGroup ;
        up:encodedBy/skos:prefLabel ?prefferedGeneName ;
          up:sequence/rdf:value ?value .
  OPTIONAL {
    ?protein up:annotation ?functionAnnotation .
    ?functionAnnotation a up:Function_Annotation ;
      rdfs:comment ?functionComment .
  }
  SERVICE <https://sparql.orthodb.org/sparql>{
    ?orthoGroup orthodb:ogMedianProteinLength ?medianLength .
    ?orthoGroup orthodb:hasMember ?xref .
    ?xref orthodb:xref/orthodb:xrefResource uniprotkb:P05067 .
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","SERVICE","FROM","STR","IF","STRLEN"],category:"sib-swiss UniProt"},{name:"UniProt - 37 Epitope VSTQ where T is a phosporylated threonine",slug:"UniProt_37_Epitope_VSTQ_where_T_is_a_phosporylated_threonine",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find the human protein which contains an epitope VSTQ, where T is a phosphorylated threonine",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?protein 
  ?comment
  ?begin
  ?end 
WHERE
{
  ?protein a up:Protein ;
    up:organism taxon:9606 ; 
    up:sequence ?sequence ;
    up:annotation ?annotation .
  ?annotation a up:Modified_Residue_Annotation ;
    rdfs:comment ?comment ;
    up:range ?range .
  ?range 
    faldo:begin [ faldo:position ?begin ; faldo:reference ?sequence ] ;
    faldo:end [ faldo:position ?end ; faldo:reference ?sequence ] .
  ?sequence rdf:value ?aaSequence .
  FILTER (SUBSTR(?aaSequence, ?begin -2 , 4) = "VSTQ")     
  FILTER (CONTAINS(?comment, "Phosphothreonine"))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","STR","IF","SUBSTR","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 38 find gene region in wikidata for known entry",slug:"UniProt_38_find_gene_region_in_wikidata_for_known_entry",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"For the human entry P05067 (Amyloid-beta precursor protein) find the gene start ends in WikiData",context:null,inidces:[],query:`PREFIX p: <http://www.wikidata.org/prop/>
PREFIX pq: <http://www.wikidata.org/prop/qualifier/>
PREFIX ps: <http://www.wikidata.org/prop/statement/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>


SELECT 
	?protein 
	?begin
	?end
	?chromosome
	?assembly
WHERE {
    {
        BIND(uniprotkb:P05067 AS ?proteinIRI)
        BIND (SUBSTR(STR(?proteinIRI), STRLEN(STR(uniprotkb:))+1) AS ?protein)
    }
    SERVICE <https://query.wikidata.org/sparql> {
        ?wp wdt:P352 ?protein ;
            wdt:P702 ?wg . 
        ?wg p:P644   ?wgss .
        ?wgss ps:P644        ?begin ;
          pq:P1057/wdt:P1813 ?chromosome ;
          pq:P659/rdfs:label ?assembly .
        ?wg p:P645 ?wgse .
        ?wgse ps:P645        ?end ;
          pq:P1057/wdt:P1813 ?chromosome ;
          pq:P659/rdfs:label ?assembly .
        FILTER(lang(?assembly) = "en")
  } 
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","SERVICE","STR","LANG","IRI","IF","STRLEN","SUBSTR"],category:"sib-swiss UniProt"},{name:"UniProt - 39 experimental catalytic activities in swissprot",slug:"UniProt_39_experimental_catalytic_activities_in_swissprot",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve entries and catalytic activities in the reviewed (UniProtKB/Swiss-Prot) section that have experimental evidences,",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT  
  ?protein
  ?rhea 
WHERE {
  # ECO 269 is experimental evidence
  BIND (<http://purl.obolibrary.org/obo/ECO_0000269> as ?evidence)
  GRAPH <http://sparql.uniprot.org/uniprot> {
    ?protein up:reviewed true ;
      up:annotation ?a ;
      up:attribution ?attribution  .

    ?a a up:Catalytic_Activity_Annotation ;
      up:catalyticActivity ?ca .
    ?ca up:catalyzedReaction ?rhea .
  
    [] rdf:subject ?a ;
      rdf:predicate up:catalyticActivity ;
      rdf:object ?ca ;
      up:attribution ?attribution .

    ?attribution up:evidence ?evidence .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","BIND"],category:"sib-swiss UniProt"},{name:"UniProt - 3 entry sequences organism",slug:"UniProt_3_entry_sequences_organism",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all UniProtKB entries, and their organism and amino acid sequences (including isoforms), for <a href=\\",context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?organism ?isoform ?sequence
WHERE
{
    ?protein a up:Protein .
    ?protein up:organism ?organism .
    # Taxon subclasses are materialized, do not use rdfs:subClassOf+
    ?organism rdfs:subClassOf taxon:83333 .
    ?protein up:sequence ?isoform .
    ?isoform rdf:value ?sequence .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 40 human enzymes that metabolize sphingolipids",slug:"UniProt_40_human_enzymes_that_metabolize_sphingolipids",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve human enzymes that metabolize sphingolipids and are annotated in ChEMBL",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?protein ?chemblEntry
WHERE {
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?rhea rdfs:subClassOf rh:Reaction ;
      rh:side/rh:contains/rh:compound/rh:chebi/rdfs:subClassOf+ CHEBI:26739 .
  }
  ?ca up:catalyzedReaction ?rhea .
  ?protein up:annotation/up:catalyticActivity ?ca ;
    up:organism taxon:9606 ;
    rdfs:seeAlso ?chemblEntry .
  ?chemblEntry up:database <http://purl.uniprot.org/database/ChEMBL> .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 41 fragmented sequences",slug:"UniProt_41_fragmented_sequences",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve UniProtKB entries with sequences that are composed of fragments",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>


SELECT DISTINCT 
  ?protein
WHERE {
  ?protein a up:Protein ;
    up:sequence ?sequence .
  MINUS { ?sequence up:fragment [] }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","MINUS","DISTINCT","MIN"],category:"sib-swiss UniProt"},{name:"UniProt - 42 connect patents to epo",slug:"UniProt_42_connect_patents_to_epo",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Connect patents cited in UniProtKB with those in the patent database at EPO via publication number.",context:null,inidces:[],query:`PREFIX patent: <http://data.epo.org/linked-data/def/patent/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?citation ?patent ?application ?applicationNo
WHERE
{
  ?citation a up:Patent_Citation ;
    skos:exactMatch ?patent .
  FILTER(CONTAINS(STR(?patent), 'EP'))
  BIND(SUBSTR(STR(?patent), 35) AS ?applicationNo)
  SERVICE <https://data.epo.org/linked-data/query>{
    ?application patent:publicationNumber ?applicationNo
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","SERVICE","STR","SUBSTR","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 43 patents used in uniprot granted twenty years in the past",slug:"UniProt_43_patents_used_in_uniprot_granted_twenty_years_in_the_past",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Connect patents cited in UniProtKB with those in the patent database at EPO via publication number, whose grant date is more than twenty years in the past.",context:null,inidces:[],query:`PREFIX patent: <http://data.epo.org/linked-data/def/patent/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?grantDate ?patent ?application ?applicationNo
WHERE
{
    ?citation a up:Patent_Citation ;
  skos:exactMatch ?patent .
  BIND(SUBSTR(STR(?patent), 35) AS ?applicationNo)
  BIND(SUBSTR(STR(?patent), 33, 2) AS ?countryCode)
  SERVICE <https://data.epo.org/linked-data/query>{
    ?publication patent:publicationNumber ?applicationNo ;
      patent:application ?application .
    ?application patent:grantDate ?grantDate .
  }
  BIND((year(now()) - 20) AS ?thisYearMinusTwenty)
  BIND(year(?grantDate) AS ?grantYear)
  FILTER(?grantYear < ?thisYearMinusTwenty)
} ORDER BY ?grantYear`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","MINUS","SERVICE","ORDER BY","STR","NOW","SUBSTR","YEAR","COUNT","MIN"],category:"sib-swiss UniProt"},{name:"UniProt - 44 rhea interpro union in uniprot",slug:"UniProt_44_rhea_interpro_union_in_uniprot",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find the Rhea and InterPro combinations in UniProtKB entries.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?interpro
	?rhea
FROM <http://sparql.uniprot.org/uniprot>
WHERE 
{
  ?protein up:reviewed true .
  ?protein up:annotation ?annotation .
  ?annotation up:catalyticActivity ?rhea .
  ?protein rdfs:seeAlso ?interpro .
  ?interpro up:database <http://purl.uniprot.org/database/InterPro> .
} ORDER BY ?rhea`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","ORDER BY","FROM"],category:"sib-swiss UniProt"},{name:"UniProt - 45 drugs targeting human sterol metabolism enzymes",slug:"UniProt_45_drugs_targeting_human_sterol_metabolism_enzymes",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve drugs that target human enzymes involved in sterol metabolism (federated query with WikiData and Rhea).",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX chebihash: <http://purl.obolibrary.org/obo/chebi#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX wdt: <http://www.wikidata.org/prop/direct/>

SELECT DISTINCT ?protein ?proteinFullName ?wikiChemical ?wikiChemicalLabel ?medicalConditionTreatedLabel
WHERE {
  # ChEBI: retrieve members of the ChEBI class ChEBI:15889 (sterol)
  # Rhea: retrieve the reactions involving these ChEBI as participants
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?reaction rdfs:subClassOf rh:Reaction ;
      rh:status rh:Approved ;
      rh:side ?reactionSide .
    ?reactionSide
      rh:contains ?participant .
    ?participant rh:compound ?compound
    {
      ?compound rh:chebi ?chebi .
      ?chebi (rdfs:subClassOf)+ CHEBI:15889
    } UNION {
      ?compound rh:chebi ?chebi .
      ?chebi2 rdfs:subClassOf ?chebiRestriction .
      ?chebiRestriction
		a owl:Restriction ;
		owl:onProperty chebihash:has_major_microspecies_at_pH_7_3 ;
		owl:someValuesFrom ?chebi .
      ?chebi2 (rdfs:subClassOf)+ CHEBI:15889
    }
  }
  # UniProt: retrieve the human (taxid:9606) enzymes catalyzing these Rhea reactions
  ?ca up:catalyzedReaction  ?reaction .
  ?a up:catalyticActivity  ?ca .
  ?proteinIRI  up:annotation ?a ;
    up:organism taxon:9606 ;
    up:recommendedName ?proteinRecName .
  ?proteinRecName up:fullName ?proteinFullName .
  # Find drugs in wikidata that interact with the UniProt Proteins
  BIND (SUBSTR(STR(?proteinIRI), STRLEN(STR(uniprotkb:))+1) AS ?protein)
  SERVICE <https://query.wikidata.org/sparql> {
    ?wp wdt:P352  ?protein .
    ?wikiChemical wdt:P129 ?wp . # Physically interacts with
    ?wikiChemical rdfs:label ?wikiChemicalLabel .
    ?wikiChemical wdt:P2175 ?wmc . # Medical conndition treated
    ?wmc rdfs:label ?medicalConditionTreatedLabel .
    FILTER(lang(?medicalConditionTreatedLabel) = 'en')
    FILTER(lang(?wikiChemicalLabel) = 'en')
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","BIND","VALUES","SERVICE","DISTINCT","FROM","WITH","ALL","STR","LANG","IRI","STRLEN","SUBSTR","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 46 duck pictures via eepa and taxonomy",slug:"UniProt_46_duck_pictures_via_eepa_and_taxonomy",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve images of 'Anas' (Ducks) from the European Environmental Agency databases (federated query).",context:null,inidces:[],query:`PREFIX eunisSpecies: <http://eunis.eea.europa.eu/rdf/species-schema.rdf#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?taxon
    ?ncbiTaxid
    ?eunisTaxon
    ?eunisName 
    ?image
WHERE
{
    GRAPH <http://sparql.uniprot.org/taxonomy>
    {
        ?taxon a up:Taxon .
        # Taxon subclasses are materialized, do not use rdfs:subClassOf+
        ?taxon rdfs:subClassOf taxon:8835 .
        BIND(strafter(str(?taxon), 'onomy/') AS ?ncbiTaxid)
    }
    SERVICE <https://semantic.eea.europa.eu/sparql>
    {
        ?eunisTaxon a eunisSpecies:SpeciesSynonym ;
           eunisSpecies:binomialName ?eunisName ;
           eunisSpecies:sameSpeciesNCBI ?ncbiTaxid ;
           <http://xmlns.com/foaf/0.1/depiction> ?image .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","BIND","SERVICE","STR"],category:"sib-swiss UniProt"},{name:"UniProt - 47 protein with transmembrane region with close by alanine",slug:"UniProt_47_protein_with_transmembrane_region_with_close_by_alanine",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find UniProtKB entries with a transmembrane region, with an alanine in the 15 amino acid region preceding the transmembrane",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX up: <http://purl.uniprot.org/core/>


SELECT ?protein ?from ?interestingRegion
WHERE
{
  ?protein up:annotation ?annotation .
  ?annotation a up:Transmembrane_Annotation .
  # Get the coordinates of the Transmembrane
  ?annotation up:range ?range .
  ?range faldo:begin ?beginI .
  ?beginI faldo:position ?begin .
  ?beginI faldo:reference ?sequence .
  # The aas will have the specific IUPAC aminoacids
  ?sequence rdf:value ?aas .
  # We calculate the start by substracting 10
  BIND(?begin - 10 AS ?tenBeforeBegin)
  # Can't start before the sequence starts or we might miss some results
  BIND(IF(?tenBeforeBegin < 1, 0, ?tenBeforeBegin) AS ?from)
  # Substring the IUPAC aminoacids
  BIND(SUBSTR(?aas, ?from, 15) AS ?interestingRegion)
  # The interestingRegion needds to contain an Alanine
  FILTER(CONTAINS(?interestingRegion, 'A'))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","FROM","STR","IF","SUBSTR","CONTAINS","MIN"],category:"sib-swiss UniProt"},{name:"UniProt - 48 glycosylation sites and glycans",slug:"UniProt_48_glycosylation_sites_and_glycans",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve glycosylation sites and glycans on human enzymes (federated with Glyconnect)",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX foaf: <http://xmlns.com/foaf/0.1/>
PREFIX glycan: <http://purl.jp/bio/12/glyco/glycan#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
        DISTINCT
            ?protein ?glycosite ?glycostructure ?glycoimage
WHERE{
  ?protein up:annotation ?annotation .
  ?protein up:organism taxon:9606 .
  ?annotation a up:Catalytic_Activity_Annotation .
  ?protein up:sequence ?isoform .
   
  SERVICE <https://glyconnect.expasy.org/sparql> {
    ?glycosite faldo:reference ?isoform .
    ?glycosite faldo:position ?position .
    ?specificglycosite faldo:location ?glycosite .
    ?glycoprotein glycan:glycosylated_at ?specificglycosite .
    ?glycostructure glycan:glycosylates_at ?specificglycosite .
    ?glycostructure foaf:depiction ?glycoimage .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","STR","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 49 tissues where genes metabolizing cholestrol are expressed",slug:"UniProt_49_tissues_where_genes_metabolizing_cholestrol_are_expressed",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve the UniProtKB proteins, their catalyzed Rhea reactions, their encoding genes (Ensembl) and the anatomic entities where the genes are expressed (UBERON anatomic entites from Bgee expression data resource).",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?protein ?ensemblGene ?reaction ?anatomicEntityLabel ?anatomicEntity
WHERE {
  # federated query to Rhea enadpoint
  {
    SELECT DISTINCT ?reaction WHERE {
      SERVICE <https://sparql.rhea-db.org/sparql> {
        ?reaction rdfs:subClassOf rh:Reaction ;
          rh:equation ?reactionEquation ;
          rh:side ?reactionSide .
        ?reactionSide rh:contains ?participant .
        ?participant rh:compound ?compound .
        # compound constraint (CHEBI:16113 == cholesterol)
        ?compound rh:chebi CHEBI:16113 .
      }
    }
  }
  # taxonomy constraint (taxon:9606 == Homo sapiens)
  ?protein up:organism taxon:9606 ;
    up:annotation ?a ;
    rdfs:seeAlso / up:transcribedFrom ?ensemblGene .
  ?a a up:Catalytic_Activity_Annotation ;
    up:catalyticActivity ?ca .
  ?ca up:catalyzedReaction ?reaction .
  # federated query to Bgee (expression data)
  BIND(IRI(REPLACE(STR(?ensemblGene), "\\\\\\\\.[0-9]+$", "")) AS ?ensemblGeneNoVersion)
  SERVICE <https://www.bgee.org/sparql/> {
    ?gene lscr:xrefEnsemblGene ?ensemblGeneNoVersion ;
      genex:isExpressedIn ?anatomicEntity .
    ?anatomicEntity rdfs:label ?anatomicEntityLabel .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","SERVICE","DISTINCT","FROM","STR","IRI","CONTAINS","REPLACE"],category:"sib-swiss UniProt"},{name:"UniProt - 4 uniprot mnemonic id",slug:"UniProt_4_uniprot_mnemonic_id",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the UniProtKB entry with the <a href=\\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein
WHERE
{
    ?protein a up:Protein .
    ?protein up:mnemonic 'A4_HUMAN'
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 50 tissues where genes sphingosines are expressed",slug:"UniProt_50_tissues_where_genes_sphingosines_are_expressed",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX genex: <http://purl.org/genex#>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  DISTINCT
    ?chebi
    ?reaction
    ?protein ?ensemblGene
    ?anatomicEntityLabel
    ?anatomicEntity
WHERE {
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?reaction rdfs:subClassOf rh:Reaction .
    ?reaction rh:equation ?reactionEquation .
    ?reaction rh:side ?reactionSide .
    ?reactionSide rh:contains ?participant .
    ?participant rh:compound ?compound .
    ?compound rh:chebi ?chebi .
    ?chebi rdfs:subClassOf* CHEBI:52639
}
  ?protein up:organism taxon:9606 .
  ?protein up:annotation ?a .
  ?a a up:Catalytic_Activity_Annotation .
  ?a up:catalyticActivity ?ca .
  ?ca up:catalyzedReaction ?reaction .
  ?protein rdfs:seeAlso / up:transcribedFrom ?ensemblGene .

  SERVICE  <https://www.bgee.org/sparql/> {
    ?gene genex:isExpressedIn ?anatomicEntity .
    ?gene lscr:xrefEnsemblGene ?ensemblGene .
    ?anatomicEntity rdfs:label ?anatomicEntityLabel .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","FROM","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 51 all proteins linked to arachidonate",slug:"UniProt_51_all_proteins_linked_to_arachidonate",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all proteins linked to arachidonate (CHEBI:32395)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>


SELECT 
    DISTINCT
        ?uniprot
        ?uniprotID
        ?recname
        ?gene
        ?chebi
        ?uniprotName
WHERE {
  SERVICE <https://sparql.rhea-db.org/sparql> {
     VALUES (?chebi) { (CHEBI:32395) }
     ?rhea rh:side/rh:contains/rh:compound ?compound .
     ?compound rh:chebi ?chebi .
     ?chebi up:name ?uniprotName .
  }
  ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea .
  ?uniprot up:mnemonic ?uniprotID .
  ?uniprot up:recommendedName/up:fullName ?recname .
  OPTIONAL {?uniprot up:encodedBy/skos:prefLabel ?gene .}
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","VALUES","SERVICE","DISTINCT","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 52 drugs targeting sterol metabolism",slug:"UniProt_52_drugs_targeting_sterol_metabolism",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve drugs that target human enzymes involved in sterol metabolism (federated query with Rhea and ChEMBL via IDSM/Elixir czech republic).",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX cco: <http://rdf.ebi.ac.uk/terms/chembl#>
PREFIX chebihash: <http://purl.obolibrary.org/obo/chebi#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
        DISTINCT
            ?protein
            ?proteinFullName
            ?activityType
            ?standardActivityValue
            ?standardActivityUnit
            ?chemblMolecule
            ?chemlbMoleculePrefLabel
WHERE
  {
  # ChEBI: retrieve members of the ChEBI class ChEBI:15889 (sterol)
  # Rhea: retrieve the reactions involving these ChEBI as participants
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?reaction rdfs:subClassOf  rh:Reaction ;
	      rh:status        rh:Approved ;
	      rh:side          ?reactionSide .
    ?reactionSide
	      rh:contains      ?participant .
    ?participant rh:compound  ?compound
    {
      ?compound  rh:chebi  ?chebi .
      ?chebi (rdfs:subClassOf)+ CHEBI:15889
    } UNION {
      ?compound  rh:chebi           ?chebi .
      ?chebi2   rdfs:subClassOf     ?chebiRestriction .
      ?chebiRestriction
		a           owl:Restriction ;
		owl:onProperty      chebihash:has_major_microspecies_at_pH_7_3 ;
		owl:someValuesFrom  ?chebi .
      ?chebi2 (rdfs:subClassOf)+ CHEBI:15889
    }
  }
  # UniProt: retrieve the human (taxid:9606) enzymes catalyzing these Rhea reactions
  ?ca       up:catalyzedReaction  ?reaction .
  ?a        up:catalyticActivity  ?ca .
  ?protein  up:annotation         ?a ;
	    up:organism           taxon:9606 ;
	    up:recommendedName    ?proteinRecName .
  ?proteinRecName
	    up:fullName           ?proteinFullName .
  # Find drugs in wikidata that interact with the UniProt Proteins
  # ChEMBL: retrieve the corresponding targets and with drugs in clinical phase 4
  # Via https://idsm.elixir-czech.cz/sparql/
  SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/idsm> {
    ?activity a cco:Activity ;
      cco:hasMolecule ?chemblMolecule ;
      cco:hasAssay ?assay ;
      cco:standardType ?activityType ;
      cco:standardValue ?standardActivityValue ;
      cco:standardUnits ?standardActivityUnit .
    ?chemblMolecule cco:highestDevelopmentPhase ?highestDevelopmentPhase ;
      rdfs:label ?chemblMoleculeLabel ;
      skos:prefLabel ?chemlbMoleculePrefLabel .
    FILTER (?highestDevelopmentPhase > 3)
    ?assay cco:hasTarget ?target .
    ?target cco:hasTargetComponent/cco:targetCmptXref ?protein .
    ?protein a cco:UniprotRef .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","VALUES","SERVICE","DISTINCT","FROM","WITH","STR","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 53 mouse homologs of sterol enzymes via omabrowser",slug:"UniProt_53_mouse_homologs_of_sterol_enzymes_via_omabrowser",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find mouse homologs in OMABrowser of human enzymes that catalyze reactions involving sterols (CHEBI:15889). Federating with Rhea and OMABrowser.",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX lscr: <http://purl.org/lscr#>
PREFIX orth: <http://purl.org/net/orth#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  DISTINCT
    ?chebi
    ?reaction
    ?humanProtein
    ?mouseProtein
    ?cluster 
WHERE {
  SERVICE <https://sparql.rhea-db.org/sparql> {
    ?reaction rdfs:subClassOf rh:Reaction .
    ?reaction rh:side/rh:contains/rh:compound ?compound .
    ?compound rh:chebi ?chebi .
    ?chebi rdfs:subClassOf* CHEBI:15889
  }

  ?humanProtein up:organism taxon:9606 .
  ?humanProtein up:annotation ?a .
  ?a a up:Catalytic_Activity_Annotation .
  ?a up:catalyticActivity ?ca .
  ?ca up:catalyzedReaction ?reaction .

  SERVICE  <https://sparql.omabrowser.org/sparql> {
    ?cluster a orth:ParalogsCluster .
    ?cluster orth:hasHomologousMember ?node1 , ?node2 .
    ?node1 orth:hasHomologousMember* ?orthoProtein1 .
    ?node2 orth:hasHomologousMember* ?orthoProtein2 .
    ?orthoProtein1 lscr:xrefUniprot ?mouseProtein .
    ?orthoProtein2 lscr:xrefUniprot ?humanProtein .
    # inTaxon mouse
    ?orthoProtein1 orth:organism/<http://purl.obolibrary.org/obo/RO_0002162> taxon:10090 . 
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","DISTINCT","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 54 binding sites for ligands similar to heme sachem",slug:"UniProt_54_binding_sites_for_ligands_similar_to_heme_sachem",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Proteins with binding sites for ligands similar to heme",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT 
  ?protein
  ?mnemonic
  ?proteinName
  ?ligandSimilarityScore
  ?ligand
WHERE {
  SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/chebi> {
    ?ssc sachem:compound ?ligand; 
      sachem:score ?ligandSimilarityScore ;
      sachem:similaritySearch ?sss .
        # Smiles of Heme
    ?sss    sachem:query "CC1=C(CCC([O-])=O)C2=[N+]3C1=Cc1c(C)c(C=C)c4C=C5C(C)=C(C=C)C6=[N+]5[Fe-]3(n14)n1c(=C6)c(C)c(CCC([O-])=O)c1=C2";
      sachem:cutoff "8e-1"^^xsd:double ;
      sachem:aromaticityMode sachem:aromaticityDetect ;
      sachem:similarityRadius 1 ;
      sachem:tautomerMode sachem:ignoreTautomers .
  }
  ?protein up:mnemonic ?mnemonic ;
    up:recommendedName/up:fullName ?proteinName ;
    up:annotation ?annotation .
  ?annotation a up:Binding_Site_Annotation ;
      up:ligand/rdfs:subClassOf ?ligand .
}
ORDER BY DESC(?ligandSimilarityScore)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","SERVICE","ORDER BY"],category:"sib-swiss UniProt"},{name:"UniProt - 55 metal or sulphur cluster binding sites experimental",slug:"UniProt_55_metal_or_sulphur_cluster_binding_sites_experimental",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Number of proteins with binding sites for metals or metal sulfur clusters (and experimental evidence for the binding)",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?ligand
  ?ligandName 
  (COUNT(DISTINCT ?protein) as ?entries)
WHERE {
   ?protein up:annotation ?annotation .
   
   VALUES ?evs { obo:ECO_0000269 obo:ECO_0007744 } .
   VALUES ?chebids { CHEBI:25213 CHEBI:25214 } .
   ?st rdf:subject ?protein ; 
       rdf:predicate up:annotation ; 
       rdf:object ?annotation ;
       up:attribution/up:evidence ?evs .

   ?annotation up:ligand/rdfs:subClassOf ?ligand .
   ?ligand rdfs:subClassOf+ ?chebids ;
     rdfs:label ?ligandName .
}
GROUP BY ?ligand ?ligandName
ORDER BY DESC(?entries)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","GROUP BY","ORDER BY","DISTINCT","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 56 enzymes that have a known allosteric effect",slug:"UniProt_56_enzymes_that_have_a_known_allosteric_effect",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select enzymes that have ligands known to have an allosteric effect",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
  ?ligandName 
  ?ligandNote
  ?chebi
WHERE {
   ?protein up:annotation ?annotation .
   ?annotation a up:Binding_Site_Annotation . 
   ?annotation up:ligand ?ligand .
   ?ligand rdfs:comment ?ligandNote ;
     rdfs:subClassOf ?chebi ;
     rdfs:label ?ligandName .
   FILTER(REGEX(?ligandNote, "allosteric", "i"))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","ALL","REGEX"],category:"sib-swiss UniProt"},{name:"UniProt - 57 map pdb identifiers plus chains to uniprot",slug:"UniProt_57_map_pdb_identifiers_plus_chains_to_uniprot",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Map a selection of PDB identifiers plus chains to UniProtKB",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?pdbId ?chain ?pdbChain ?uniprot
WHERE
{
  # A space separated list of pairs of PDB identifiers and the chain code.
  VALUES(?pdbId ?pdbChain) { ('6VXC' 'A') ('1BG3' 'B') }

  # Make an IRI out of the pdbId
  BIND(iri(concat('http://rdf.wwpdb.org/pdb/', ?pdbId)) AS ?pdb)

  # Map to UniProt entries
  ?uniprot rdfs:seeAlso ?pdb .
  ?pdb up:database <http://purl.uniprot.org/database/PDB> ;
       up:chainSequenceMapping ?chainSm .
  ?chainSm up:chain ?chainsPlusRange .

  # Extract the list of chains from the text representation.
  BIND(STRBEFORE(?chainsPlusRange, '=') AS ?chain)

  # Filter those that match.
  FILTER(CONTAINS(?chain, ?pdbChain))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","BIND","VALUES","FROM","STR","IRI","IF","CONCAT","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 58 uniprot to HGNC and symbols",slug:"UniProt_58_uniprot_to_HGNC_and_symbols",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Map a selection of UniProtKB accession numbers (ACs) to HGNC identifiers and symbols",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?uniprot
  ?hgnc
  ?hgncSymbol
WHERE
{
  # A space separated list of UniProt primary accessions.
  VALUES (?acc) {('P05067') ('P00750')}
  BIND(iri(concat(str(uniprotkb:), ?acc)) AS ?uniprot)
  ?uniprot rdfs:seeAlso ?hgnc .
  ?hgnc up:database <http://purl.uniprot.org/database/HGNC> ;
       rdfs:comment ?hgncSymbol .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","VALUES","STR","IRI","CONCAT"],category:"sib-swiss UniProt"},{name:"UniProt - 59 all isoforms for a given proteome",slug:"UniProt_59_all_isoforms_for_a_given_proteome",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Count all isoforms for a given proteome",context:null,inidces:[],query:`PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

PREFIX proteome:<http://purl.uniprot.org/proteomes/>
SELECT
  (COUNT(DISTINCT ?sequence) AS ?allIsoforms)
WHERE
{
  ?protein up:reviewed true .
  ?protein up:organism taxon:9606 .
  ?protein up:sequence ?sequence .
  ?protein up:proteome/^skos:narrower proteome:UP000005640 .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","ALL","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 5 mapping to PDB",slug:"UniProt_5_mapping_to_PDB",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select a mapping of UniProtKB to PDB entries using the UniProtKB cross-references to the <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?db
WHERE
{
    ?protein a up:Protein .
    ?protein rdfs:seeAlso ?db .
    ?db up:database <http://purl.uniprot.org/database/PDB>
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 60 enzymes working on substrate with Cholestene backbone",slug:"UniProt_60_enzymes_working_on_substrate_with_Cholestene_backbone",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find human proteins that catalyze reactions where substrates or product have a <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  DISTINCT
    ?protein
    ?disease
    ?rhea
    ?chebi
    ?omim
WHERE {
    # Find complete ChEBIs with a Cholestane skeleton, via the Czech Elixir node IDSM Sachem chemical substructure search.
    SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/chebi> {
      ?chebi sachem:substructureSearch [
        sachem:query
"[C@]12(CCC3CCCC[C@]3(C)[C@@]1([H])CC[C@]1(C)[C@@]([H])([C@@](C)([H])CCCC(C)C)CC[C@@]21[H])[H]"
].
   }
   # Use the fact that UniProt catalytic activities are annotated using Rhea 
   # Mapping the found ChEBIs to Rhea reactions
   SERVICE <https://sparql.rhea-db.org/sparql>{
     ?rhea rh:side/rh:contains/rh:compound/rdfs:subClassOf ?chebi .
   }
   # Match the found Rhea reactions with human UniProtKB proteins
   ?protein up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea .
   ?protein up:organism taxon:9606 .
   # Find only those human entries that have an annotated related disease, and optionaly map these to OMIM
   ?protein up:annotation/up:disease ?disease .
   OPTIONAL {
     ?disease rdfs:seeAlso ?omim .
     ?omim up:database <http://purl.uniprot.org/database/MIM>
   }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","SERVICE","DISTINCT","WITH","USING","STR","SUBSTR","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 61 Gene Protein Reaction sets",slug:"UniProt_61_Gene_Protein_Reaction_sets",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the Gene Protein Reaction sets for Human (Ensembl Gene, Human UniProtKB, Catalyzed Rhea reactions)",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  DISTINCT # Disinct because there might be more than one transcript for a gene leading to duplicates
    ?ensemblGene
    ?protein 
    ?rhea
WHERE {
  ?protein up:reviewed true ;
           up:organism taxon:9606 .
  ?protein up:annotation ?caa ;
           rdfs:seeAlso ?ensemblTranscript .
  ?ensemblTranscript up:database <http://purl.uniprot.org/database/Ensembl> .
	?caa up:catalyticActivity ?ca .
  ?ca up:catalyzedReaction ?rhea .
  ?ensemblTranscript up:transcribedFrom ?ensemblGene 
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","FROM"],category:"sib-swiss UniProt"},{name:"UniProt - 62 diseases involving enzymes",slug:"UniProt_62_diseases_involving_enzymes",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find diseases that are thought to be related to enzymes",context:null,inidces:[],query:`PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
?disease ?diseaseLabel
WHERE {
 ?protein up:enzyme|up:annotation/up:catalyticActivity/up:enzymeClass ?enzyme ;
                   up:annotation ?diseaseAnnotation .
 ?diseaseAnnotation a up:Disease_Annotation ;
                    up:disease ?disease .
 ?disease skos:prefLabel ?diseaseLabel .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 63 diseases involving enzymes located in mitochondrion",slug:"UniProt_63_diseases_involving_enzymes_located_in_mitochondrion",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find (Human genetic) diseases that are thought to be related to Enzymes, known to be located in the Mitochondrion",context:null,inidces:[],query:`PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  DISTINCT
    ?disease
WHERE {
  ?protein a up:Protein ;
  up:organism taxon:9606 ;
  up:annotation ?disease_annotation ,
                ?subcellularLocation .
  {
    ?protein up:enzyme [] .
  } UNION {
    ?protein up:annotation/a up:Catalytic_Activity_Annotation .
  }
  ?disease_annotation a up:Disease_Annotation ;
    up:disease ?disease .
  ?subcellularLocation a up:Subcellular_Location_Annotation ;
    up:locatedIn ?location .
  ?location up:cellularComponent ?component .
  ?component up:partOf* <http://purl.uniprot.org/locations/173> .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","DISTINCT"],category:"sib-swiss UniProt"},{name:"UniProt - 64 diseases related to mutation in active site",slug:"UniProt_64_diseases_related_to_mutation_in_active_site",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find (Human Genetic) diseases, that are related to a natural variant on the active site of an enzyme.",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  DISTINCT
   ?disease
WHERE {
  ?protein a up:Protein ;
    up:organism taxon:9606 ;
    up:annotation ?disease_annotation, ?active_site_annotation, ?natural_variant_annotation .
  {
    ?protein up:enzyme [] .
  } UNION {
  ?protein up:annotation/a up:Catalytic_Activity_Annotation .
  }
  ?disease_annotation a up:Disease_Annotation ;
  up:disease ?disease .
  ?active_site_annotation a up:Active_Site_Annotation ;
  up:range ?active_site_range .
  ?active_site_range faldo:begin ?active_site_begin .
  ?active_site_begin faldo:position ?active_site_position ;
                     faldo:reference ?sequence .
  ?natural_variant_annotation a up:Natural_Variant_Annotation ;
  up:range ?natural_variant_range ;
  skos:related ?disease .
  ?natural_variant_range faldo:begin ?natural_variant_begin ;
                         faldo:end ?natural_variant_end .
  ?natural_variant_begin faldo:position ?natural_variant_begin_position .
  ?natural_variant_end faldo:position ?natural_variant_end_position ;
                       faldo:reference ?sequence .

  FILTER(?natural_variant_begin_position >= ?active_site_position && ?active_site_position <= ?natural_variant_end_position)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","UNION","DISTINCT"],category:"sib-swiss UniProt"},{name:"UniProt - 65 distinct extinct organisms in uniprotkb",slug:"UniProt_65_distinct_extinct_organisms_in_uniprotkb",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"How many distinct extinct organisms are represented in UniProtKB",context:null,inidces:[],query:`PREFIX keywords: <http://purl.uniprot.org/keywords/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?taxon
  (SAMPLE(?name) AS ?anName)
  (COUNT(DISTINCT ?protein) AS ?entriesPerExtinctTaxon)
WHERE
{
  GRAPH<http://sparql.uniprot.org/taxonomy>{
    ?taxon a up:Taxon ;
           up:scientificName ?name .
  }
  ?protein up:organism ?taxon ;
           up:classifiedWith keywords:952 .
} GROUP BY ?taxon ORDER BY ?taxon`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","GROUP BY","ORDER BY","DISTINCT","WITH","IF","COUNT","SAMPLE"],category:"sib-swiss UniProt"},{name:"UniProt - 66 distinct rhea transport in reviewed uniprot",slug:"UniProt_66_distinct_rhea_transport_in_reviewed_uniprot",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Count number of unique Rhea transport reactions annotated in reviewed UniProtKB entries.",context:null,inidces:[],query:`PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
(COUNT(DISTINCT ?rhea) AS ?distinctRheaTransportInUniProt)
WHERE
{
  GRAPH <https://sparql.rhea-db.org/rhea> {
    ?rhea rh:isTransport true .
  }
  ?protein up:annotation ?ann .
  ?ann up:catalyticActivity ?ca .
  ?ca up:catalyzedReaction ?rhea .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","DISTINCT","STR","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 67 draft human metabolome",slug:"UniProt_67_draft_human_metabolome",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Generate a draft human metabolome",context:null,inidces:[],query:`PREFIX chebislash: <http://purl.obolibrary.org/obo/chebi/>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
?uniprot ?mnemonic ?rhea ?chebi ?smiles ?inchiKey
WHERE
{
  ?uniprot up:annotation/up:catalyticActivity/up:catalyzedReaction ?rhea ;
        up:organism taxon:9606 ;
        up:mnemonic ?mnemonic .
  SERVICE <https://sparql.rhea-db.org/sparql> {
	?rhea rh:side/rh:contains/rh:compound ?compound .
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
   }
   ?chebi chebislash:smiles ?smiles ;
          chebislash:inchikey ?inchiKey .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","SERVICE","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 68 drosophila enzymes derived from at least two mRNAs",slug:"UniProt_68_drosophila_enzymes_derived_from_at_least_two_mRNAs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find Drosophila enzymes, that depend on more than one transcript (requires cross-references to Ensembl family of resources).",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
WHERE {
  ?protein up:organism taxon:7227 ;
    a up:Protein ;
    up:reviewed true ;
    rdfs:seeAlso ?transcript .
  {
    ?protein up:annotation/a up:Catalytic_Activity_Annotation ;
  } UNION {
    ?protein up:enzyme ?enzyme .
  }
  ?transcript  a up:Transcript_Resource .
} GROUP BY ?protein HAVING(COUNT(?transcript) >= 2)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","GROUP BY","HAVING","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 69 drosophila proteins derived from at least two mRNAs",slug:"UniProt_69_drosophila_proteins_derived_from_at_least_two_mRNAs",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find Drosophila proteins, that depend on more than one transcript (requires cross-references to Ensembl family of resources).",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
WHERE {
  ?protein up:organism taxon:7227 ;
    a up:Protein ;
    up:reviewed true ;
    rdfs:seeAlso ?transcript .
  ?transcript  a up:Transcript_Resource .
} GROUP BY ?protein HAVING(COUNT(?transcript) >= 2)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 6 cross ref in category 3D",slug:"UniProt_6_cross_ref_in_category_3D",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all cross-references to external databases of the category <a href=\\",context:null,inidces:[],query:`PREFIX keywords: <http://purl.uniprot.org/keywords/>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT DISTINCT ?link
WHERE
{
    ?protein a up:Protein .
    ?protein up:classifiedWith keywords:5 .
    ?protein rdfs:seeAlso ?link .
    ?link up:database ?db .
    ?db up:category '3D structure databases'
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","DISTINCT","WITH","STR","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 70 enzymes interacting with molecules similar to dopamine",slug:"UniProt_70_enzymes_interacting_with_molecules_similar_to_dopamine",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Look for enzymes catalyzing reactions with molecules similar to dopamine.",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
  ?rhea
  ?chebi
WHERE {
  # Use the smiles of dopamine CHEBI:18243
  SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/chebi>{
    ?chebi sachem:similarCompoundSearch [
      sachem:query "NCCc1ccc(O)c(O)c1" ] .
  }
  GRAPH<https://sparql.rhea-db.org/rhea>{
    ?rhea rh:side/rh:contains/rh:compound ?compound .
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  }
  ?protein up:reviewed true ;
    up:annotation ?caa .
  ?caa up:catalyticActivity/up:catalyzedReaction ?rhea .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","SERVICE","CONTAINS","MIN"],category:"sib-swiss UniProt"},{name:"UniProt - 71 enzymes interacting with molecules similar to dopamine with variants related to disease",slug:"UniProt_71_enzymes_interacting_with_molecules_similar_to_dopamine_with_variants_related_to_disease",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Look for enzymes catalyzing reactions with molecules similar to dopamine, with natural variants related to a disease.",context:null,inidces:[],query:`PREFIX CHEBI: <http://purl.obolibrary.org/obo/CHEBI_>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
  ?rhea
  ?chebi
  ?disease
WHERE {
  # Use the smiles of dopamine CHEBI:18243
  SERVICE <https://idsm.elixir-czech.cz/sparql/endpoint/chebi>{
    ?chebi sachem:similarCompoundSearch [
      sachem:query "NCCc1ccc(O)c(O)c1" ] .
  }
  GRAPH<https://sparql.rhea-db.org/rhea>{
    ?rhea rh:side/rh:contains/rh:compound ?compound .
    ?compound (rh:chebi|(rh:reactivePart/rh:chebi)|rh:underlyingChebi) ?chebi .
  }
  ?protein up:reviewed true ;
    up:annotation ?caa, ?natural_variant_annotation, ?disease_annotation .
  ?caa up:catalyticActivity/up:catalyzedReaction ?rhea .
  ?natural_variant_annotation a up:Natural_Variant_Annotation ;
    skos:related ?disease .
  ?disease_annotation a up:Disease_Annotation ;
     up:disease ?disease .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","SERVICE","CONTAINS","MIN"],category:"sib-swiss UniProt"},{name:"UniProt - 72 enzymes mapping to PDB",slug:"UniProt_72_enzymes_mapping_to_PDB",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
  ?db
WHERE
{
  ?protein a up:Protein .
  ?protein rdfs:seeAlso ?db .
  ?db up:database <http://purl.uniprot.org/database/PDB> .
  {
    ?protein up:enzyme ?enzyme .
  } UNION {
    ?protein up:annotation/rdf:type up:Catalytic_Activity_Annotation .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH","RDF"],sparqlConcepts:["SELECT","WHERE","UNION"],category:"sib-swiss UniProt"},{name:"UniProt - 73 enzymes related to protein",slug:"UniProt_73_enzymes_related_to_protein",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:null,context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
SELECT
  DISTINCT ?enzyme
WHERE {
  ?protein a up:Protein .
  {
    ?protein up:enzyme ?enzyme .
  } UNION {
    ?protein up:annotation ?caa .
    ?caa a up:Catalytic_Activity_Annotation .
    ?caa up:catalyticActivity ?ca .
    ?ca up:enzymeClass ?enzyme
  } UNION {
    ?protein up:component/up:enzyme ?enzyme .
  } UNION {
    ?protein up:domain/up:enzyme ?enzyme .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH","RDF"],sparqlConcepts:["SELECT","WHERE","UNION","DISTINCT"],category:"sib-swiss UniProt"},{name:"UniProt - 74 enzymes with at least two transmembrane domains",slug:"UniProt_74_enzymes_with_at_least_two_transmembrane_domains",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find Hydrolases (enzymes that catalyse the hydrolysis of various bonds) that have at least two transmembrane domains",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
WHERE {
 ?protein up:enzyme|up:annotation/up:catalyticActivity/up:enzymeClass ?enzymeClass ;
                   up:annotation ?transMembraneAnnotation .
 ?enzymeClass rdfs:subClassOf <http://purl.uniprot.org/enzyme/3.-.-.-> .
 ?transMembraneAnnotation a up:Transmembrane_Annotation .
} GROUP BY ?protein HAVING (COUNT(DISTINCT ?transMembraneAnnotation) >= 2)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GROUP BY","HAVING","DISTINCT","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 75 enzymes with at least two transmembrane domains PDB xray",slug:"UniProt_75_enzymes_with_at_least_two_transmembrane_domains_PDB_xray",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find enzymes that have at least two transmembrane domains, whose 3D structure is elucidated through X-Ray analysis",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
WHERE {
  ?protein a up:Protein ;
    up:annotation ?transmembrane_annotation ;
    rdfs:seeAlso ?wwPDB .
  {
    ?protein up:enzyme ?enzyme .
  } UNION {
    ?protein up:annotation/a up:Catalytic_Activity_Annotation .
  }
  ?wwPDB up:database <http://purl.uniprot.org/database/PDB> ;
    up:method up:X-Ray_Crystallography .
  ?transmembrane_annotation a up:Transmembrane_Annotation .
} GROUP BY ?protein HAVING(COUNT(DISTINCT ?transmembrane_annotation ) >= 2)`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","UNION","GRAPH","GROUP BY","HAVING","DISTINCT","ALL","COUNT"],category:"sib-swiss UniProt"},{name:"UniProt - 76 enzymes with mutagenesis affecting active site",slug:"UniProt_76_enzymes_with_mutagenesis_affecting_active_site",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find enzymes, where the active site is a region affected by mutagenesis and show the comment regarding mutagenesis effect.",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein ?mutagenesisBeginPosition ?activeSiteBeginPosition ?mutagenesisRangeEndPosition ?mutagenesisComment
WHERE {
  ?protein up:annotation ?activeSiteAnnotation ,
      ?mutagenesisAnnotation .
  ?mutagenesisAnnotation a up:Mutagenesis_Annotation ;
     up:range ?mutagenesisRange ;
	 rdfs:comment ?mutagenesisComment .
  ?activeSiteAnnotation a up:Active_Site_Annotation ;
    up:range ?activeSiteRange .
  ?activeSiteRange faldo:begin ?activeSiteBegin .
  ?activeSiteBegin faldo:position ?activeSiteBeginPosition ;
    faldo:reference ?sequence .
  ?mutagenesisRange faldo:begin ?mutagenesisRangeBegin ;
    faldo:end ?mutagenesisRangeEnd .
  ?mutagenesisRangeBegin faldo:position ?mutagenesisBeginPosition ;
    faldo:reference ?sequence .
  ?mutagenesisRangeEnd faldo:position ?mutagenesisRangeEndPosition .
  FILTER(?mutagenesisBeginPosition <= ?activeSiteBeginPosition && ?activeSiteBeginPosition <= ?mutagenesisRangeEndPosition)
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER"],category:"sib-swiss UniProt"},{name:"UniProt - 77 enzymes with tyrosine as active site",slug:"UniProt_77_enzymes_with_tyrosine_as_active_site",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find enzymes with a Tyrosine (Y) as an active site",context:null,inidces:[],query:`PREFIX faldo: <http://biohackathon.org/resource/faldo#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein
WHERE {
  ?protein up:annotation ?activeSiteAnnotation .
  ?activeSiteAnnotation a up:Active_Site_Annotation ;
    up:range ?range .
  ?range faldo:begin ?begin .
  ?begin faldo:position ?beginPosition ;
    faldo:reference ?sequence .
  ?sequence rdf:value ?sequenceVal .
  FILTER(SUBSTR(?sequenceVal, ?beginPosition, 1) = 'Y')
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","STR","SUBSTR"],category:"sib-swiss UniProt"},{name:"UniProt - 78 genetic disease related proteins",slug:"UniProt_78_genetic_disease_related_proteins",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all UniProtKB proteins annotated to be related to a genetic disease.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?uniprot ?disease ?diseaseComment ?mim
WHERE
{
  GRAPH <http://sparql.uniprot.org/uniprot> {
    ?uniprot a up:Protein ;
       up:annotation ?diseaseAnnotation .
    ?diseaseAnnotation up:disease ?disease .
  }
  GRAPH <http://sparql.uniprot.org/diseases> {
    ?disease a up:Disease ;
             rdfs:comment ?diseaseComment .
    OPTIONAL {
      ?disease rdfs:seeAlso ?mim .
       ?mim up:database <http://purl.uniprot.org/database/MIM> .
    }
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","GRAPH"],category:"sib-swiss UniProt"},{name:"UniProt - 79 mnemonic also known as id",slug:"UniProt_79_mnemonic_also_known_as_id",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all UniProtKB protein ID (mnemonic) for current UniProtKB entries.",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?uniprot ?mnemonic 
WHERE
{
  GRAPH <http://sparql.uniprot.org/uniprot> {
        ?uniprot a up:Protein ;
            up:mnemonic ?mnemonic .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH"],category:"sib-swiss UniProt"},{name:"UniProt - 7 swissprot with rec protein name if pref gene name contains word DNA",slug:"UniProt_7_swissprot_with_rec_protein_name_if_pref_gene_name_contains_word_DNA",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select reviewed UniProtKB entries (Swiss-Prot), and their recommended protein name, that have a preferred gene name that contains the text 'DNA'",context:null,inidces:[],query:`PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?name
WHERE
{
        ?protein a up:Protein .
        ?protein up:reviewed true .
        ?protein up:recommendedName ?recommended .
        ?recommended up:fullName ?name .
        ?protein up:encodedBy ?gene .
        ?gene skos:prefLabel ?text .
        FILTER CONTAINS(?text, 'DNA')
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","CONTAINS"],category:"sib-swiss UniProt"},{name:"UniProt - 80 obsolete mnemonic also known as id",slug:"UniProt_80_obsolete_mnemonic_also_known_as_id",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List all UniProtKB protein ID (mnemonic) that where used in the past for current UniProtKB entries.",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?uniprot ?obsoleteMnemonic 
WHERE
{
  GRAPH <http://sparql.uniprot.org/uniprot> {
        ?uniprot a up:Protein ;
            up:oldMnemonic ?obsoleteMnemonic .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH"],category:"sib-swiss UniProt"},{name:"UniProt - 81 rhea reactions annotated as experimental and only small molecules",slug:"UniProt_81_rhea_reactions_annotated_as_experimental_and_only_small_molecules",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all Rhea (only small molecule) that are used in UniProtKB where the annotation has a paper and is tagged having experimental evidence.",context:null,inidces:[],query:`PREFIX ECO: <http://purl.obolibrary.org/obo/ECO_>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX rh: <http://rdf.rhea-db.org/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?rhea 
  ?catalyzedReaction 
  ?source
WHERE {
  {  
    SELECT 
      DISTINCT 
        ?rhea
      WHERE {
        GRAPh<https://sparql.rhea-db.org/rhea> {
          ?rhea rdfs:subClassOf rh:Reaction .
          ?rhea rh:side/rh:contains/rh:compound ?compound2 . 
          ?uc rdfs:subClassOf rh:Compound .
        }
        ?compound2 rdfs:subClassOf ?uc . 
    	BIND(IF(?uc = rh:SmallMolecule, 0, 1) AS ?c)  
     } GROUP BY ?rhea  HAVING (SUM(?c) = 0)
  }
  ?catalyzedReaction up:catalyzedReaction ?rhea .
  ?reif rdf:object ?catalyzedReaction ;
        up:attribution ?attr .
  ?attr up:evidence ECO:0000269 ;
        up:source ?source .
  ?source a up:Citation .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","BIND","GROUP BY","HAVING","DISTINCT","ALL","IF","CONTAINS","SUM"],category:"sib-swiss UniProt"},{name:"UniProt - 82 rhea reactions associated with ec in uniprotkb",slug:"UniProt_82_rhea_reactions_associated_with_ec_in_uniprotkb",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List <a href=\\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?rhea 
  ?EC 
WHERE {
  ?CatalyticActivity  up:catalyzedReaction   ?rhea ;
    up:enzymeClass         ?EC .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 83 rhea reactions not associated with ec in uniprotkb",slug:"UniProt_83_rhea_reactions_not_associated_with_ec_in_uniprotkb",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List <a href=\\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?rhea 
  ?EC 
WHERE {
  ?CatalyticActivity up:catalyzedReaction ?rhea .
  MINUS {
    ?CatalyticActivity up:enzymeClass ?EC .
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","MINUS","MIN"],category:"sib-swiss UniProt"},{name:"UniProt - 84 taxonomy hierarchy",slug:"UniProt_84_taxonomy_hierarchy",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find all taxonomic records that describe species of the genus <a href=\\",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?species 
  ?genus 
WHERE {
  BIND(taxon:9605 AS ?genus)
  ?species a up:Taxon ;
           up:rank up:Species ;
           rdfs:subClassOf ?genus .
  ?genus a up:Taxon ;
         up:rank up:Genus .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND"],category:"sib-swiss UniProt"},{name:"UniProt - 85 taxonomy host",slug:"UniProt_85_taxonomy_host",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find taxon records that are known to have part of their life cycle in other organisms (e.g. parasite, symbiont, infection)",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?virus 
  ?host 
WHERE {
    ?virus up:host ?host .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 86 taxonomy rank and scientific name",slug:"UniProt_86_taxonomy_rank_and_scientific_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Retrieve the rank and the scientific name of an taxonomic record. Not all taxonomic records have a <a href=\\",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?taxon 
  ?scientificName
  ?rank
WHERE {
  ?taxon a up:Taxon ;
         up:scientificName ?scientificName .
  OPTIONAL {
    ?taxon up:rank ?rank
  }
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","OPTIONAL","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 87 taxonomy with at least one swissprot",slug:"UniProt_87_taxonomy_with_at_least_one_swissprot",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find taxon records for which one reviewed UniProtKB/Swiss-Prot entry exists. We might expect species, strains, subspecies and isolates in the taxon list.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>


SELECT
    DISTINCT
         ?taxid
         ?scientificName
         ?domain
         ?domainName
WHERE {
  ?uniprot a up:Protein .
  # reviewed entries
  ?uniprot up:reviewed true .
  ?uniprot up:organism ?taxid . 
  ?taxid up:scientificName ?scientificName .
    
  VALUES ?domain { taxon:2 # bacteria
                   taxon:2157 # archaea
                   taxon:2759 # eukaryota
                   taxon:10239 #viruses
                 } .
  ?taxid rdfs:subClassOf ?domain .
}`,ontologies:["EX","RDF","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","DISTINCT","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 88 uniparc linked to active uniprot",slug:"UniProt_88_uniparc_linked_to_active_uniprot",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Show for a given UniParc accessions which active UniProtKB entries have the same amino acid sequence",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?uniparc
  ?uniprot
WHERE {
  GRAPH <http://sparql.uniprot.org/uniparc>{
    BIND(<http://purl.uniprot.org/uniparc/UPI000002DB1C> AS ?uniparc) 
    ?uniparc up:sequenceFor ?uniprot .
  }
  GRAPH <http://sparql.uniprot.org/uniprot> {
    ?uniprot a up:Protein .
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","BIND"],category:"sib-swiss UniProt"},{name:"UniProt - 89 uniparc triples directly associated",slug:"UniProt_89_uniparc_triples_directly_associated",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Predicates and objects, for a given UniParc accession as a subject",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
  ?predicate 
  ?object 
WHERE { 
  <http://purl.uniprot.org/uniparc/UPI000012A0AD> ?predicate ?object
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 8 prefered gene name of human disease related proteins",slug:"UniProt_8_prefered_gene_name_of_human_disease_related_proteins",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select the preferred gene name and disease annotation of all human UniProtKB entries that are known to be involved in a disease",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?name ?text
WHERE
{
        ?protein a up:Protein . 
        ?protein up:organism taxon:9606 .
        ?protein up:encodedBy ?gene . 
        ?gene skos:prefLabel ?name .
        ?protein up:annotation ?annotation .
        ?annotation a up:Disease_Annotation .
        ?annotation rdfs:comment ?text
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 90 uniprot affected by metabolic diseases using MeSH",slug:"UniProt_90_uniprot_affected_by_metabolic_diseases_using_MeSH",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find UniProtKB proteins involved in metabolic diseases using the MeSH concept for 'Metabolic Diseases'.",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT
 ?disease ?protein
WHERE {
  SERVICE<https://id.nlm.nih.gov/mesh/sparql> {
    GRAPH <http://id.nlm.nih.gov/mesh> {
      # Mesh M0013493 represents the concept 'Metabolic Diseases'
	  ?mesh <http://id.nlm.nih.gov/mesh/vocab#broaderDescriptor>* ?broader .
      ?broader <http://id.nlm.nih.gov/mesh/vocab#preferredConcept> <http://id.nlm.nih.gov/mesh/M0013493> .
    }
  }
  GRAPH <http://sparql.uniprot.org/diseases>{
    ?disease a up:Disease ;
    	rdfs:seeAlso ?mesh .
    ?mesh up:database <http://purl.uniprot.org/database/MeSH> .
  }
  GRAPH <http://sparql.uniprot.org/uniprot> {
     ?protein up:annotation/up:disease ?disease . 
  }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","GRAPH","SERVICE"],category:"sib-swiss UniProt"},{name:"UniProt - 91 uniprot alternative protein full name",slug:"UniProt_91_uniprot_alternative_protein_full_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Alternative protein full names for UniProtKB entries",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein 
  ?fullName
WHERE {
  ?protein a up:Protein ;
           up:alternativeName ?recommendedName .
  ?recommendedName up:fullName ?fullName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 92 uniprot bioregistry iri translation",slug:"UniProt_92_uniprot_bioregistry_iri_translation",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Translate the global unique identifier for a UniProtKB record into other options using the bioregistry translating endpoint.",context:null,inidces:[],query:`PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?protein 
    ?otherIdentifier
WHERE {
    BIND(uniprotkb:P00750 AS ?protein)
    ?protein a up:Protein .
    SERVICE <https://bioregistry.io/sparql> {
        ?protein owl:sameAs ?otherIdentifier .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","BIND","SERVICE","STR","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 93 uniprot created modified updated",slug:"UniProt_93_uniprot_created_modified_updated",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List the created, last modified, and last sequence update dates for UniProtKB proteins.",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?protein 
  ?created
  ?modified
  ?version
WHERE {
  ?protein a up:Protein ;
           up:created ?created ;
           up:modified ?modified ;
           up:version ?version .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","CREATE","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 94 uniprot encoding gene",slug:"UniProt_94_uniprot_encoding_gene",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List UniProtKB proteins with their associated named gene",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein
  ?gene 
WHERE {
  ?protein a up:Protein ;
           up:encodedBy ?gene .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 95 uniprot encoding gene alternative names",slug:"UniProt_95_uniprot_encoding_gene_alternative_names",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List UniProtKB proteins with their associated gene and the gene's names that are used in the field, but not recommended for use by UniProtKB",context:null,inidces:[],query:`PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein
  ?gene ?altGeneName
WHERE {
  ?protein a up:Protein ;
           up:encodedBy ?gene .
  ?gene skos:altLabel ?altGeneName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 96 uniprot encoding gene name",slug:"UniProt_96_uniprot_encoding_gene_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List UniProtKB proteins with their associated gene and the gene's preffered name",context:null,inidces:[],query:`PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein
  ?gene 
WHERE {
  ?protein a up:Protein ;
           up:encodedBy ?gene .
  ?gene skos:prefLabel ?recommendedGeneName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 97 uniprot encoding gene org name",slug:"UniProt_97_uniprot_encoding_gene_org_name",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"List UniProtKB proteins with their associated gene and the gene's ORF label",context:null,inidces:[],query:`PREFIX up: <http://purl.uniprot.org/core/>

SELECT
  ?gene 
  ?orfName
WHERE {
  ?protein a up:Protein ;
           up:encodedBy ?gene .
  ?gene up:orfName ?orfName .
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE"],category:"sib-swiss UniProt"},{name:"UniProt - 98 uniprot entries with more than two geneid crossrefences",slug:"UniProt_98_uniprot_entries_with_more_than_two_geneid_crossrefences",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Find GeneID's crosslinked to more than one human or mouse UniProtKB entry",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX up: <http://purl.uniprot.org/core/>
PREFIX taxon:<http://purl.uniprot.org/taxon/>
SELECT 
  ?geneid 
  ?organism 
  (GROUP_CONCAT(?protein; separator=', ') AS ?entries)
WHERE
{
  VALUES ?organism {taxon:9606 taxon:10090}
  ?geneid up:database <http://purl.uniprot.org/database/GeneID> .
  ?protein rdfs:seeAlso ?geneid ; 
           up:organism ?organism
} GROUP BY ?geneid ?organism HAVING (COUNT(?protein) > 1) ORDER BY ?organism ?geneid`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","GROUP BY","ORDER BY","HAVING","CONCAT","COUNT","GROUP_CONCAT"],category:"sib-swiss UniProt"},{name:"UniProt - 99 uniprot identifiers org translation",slug:"UniProt_99_uniprot_identifiers_org_translation",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Translate a selection of UniProtKB accession numbers into other options using the identifiers.org translating endpoint.",context:null,inidces:[],query:`PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX uniprotkb: <http://purl.uniprot.org/uniprot/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT 
    ?protein 
    ?otherIdentifier
WHERE {
    VALUES (?protein) {(uniprotkb:P00750) (uniprotkb:P05067)}
    ?protein a up:Protein .
    SERVICE <https://sparql.api.identifiers.org/sparql> {
        ?protein owl:sameAs ?otherIdentifier .
    }
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","VALUES","SERVICE","IF"],category:"sib-swiss UniProt"},{name:"UniProt - 9 human variant loss of function",slug:"UniProt_9_human_variant_loss_of_function",date:"18-06-2025",image:"src/assets/images/SIB_logo.jpg",source:"SIB",description:"Select all human UniProtKB entries with a sequence variant that leads to a 'loss of function'",context:null,inidces:[],query:`PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX taxon: <http://purl.uniprot.org/taxonomy/>
PREFIX up: <http://purl.uniprot.org/core/>

SELECT ?protein ?text
WHERE
{ 
        ?protein a up:Protein .
        ?protein up:organism taxon:9606 . 
        ?protein up:annotation ?annotation .
        ?annotation a up:Natural_Variant_Annotation . 
        ?annotation rdfs:comment ?text .
        FILTER (CONTAINS(?text, 'loss of function'))
}`,ontologies:["EX","RDFS","SCHEMA","SH"],sparqlConcepts:["SELECT","WHERE","FILTER","CONTAINS"],category:"sib-swiss UniProt"}];function mP(){return[...rP,...sP,...aP,...lP,...cP,...uP,...dP,...pP,...hP,...fP,...gP].map((t,r)=>({...t,image:oP,source:"SIB"}))}const EP=mP(),SP=[$x,eP,tP];function RP(i,t){return[...i,...t].map((s,l)=>{var p,h;const u=((h=(p=s.level)==null?void 0:p.match(/\d+/))==null?void 0:h[0])||"0";return{...s,id:l+1,image:s.image||SP[u]}})}const Vi=RP(iP,EP);function mS({data:i,withPrefix:t=!1}){const{name:r,image:s,category:l,date:u,ontologies:p,slug:h}=i,f=t?`/sparql-queries/${h}`:h;return I.jsxs("div",{className:"rounded-xl overflow-hidden shadow-md shadow-stone-950 ring-1 ring-white/10 flex flex-col hover:shadow-lg hover:shadow-orange-900/50 transition-shadow bg-neutral-800 ",children:[I.jsxs("div",{className:"relative",children:[I.jsxs(ut,{to:f,children:[I.jsx("img",{className:"w-full h-42 object-cover",src:s,alt:r}),I.jsx("div",{className:"hover:bg-transparent transition duration-300 absolute inset-0 bg-gray-900 opacity-25"})]}),I.jsx("div",{className:"font-bold shadow-xl text-s absolute top-0 right-0  bg-orange-700 px-4 py-2 text-white mt-3 mr-3 hover:bg-white hover:text-orange-600 transition duration-500 ease-in-out rounded-xl",children:l})]}),I.jsxs("div",{className:"px-6 py-4 mb-auto",children:[I.jsx("p",{className:"text-gray-500 text-sm mb-2",children:u}),I.jsx("h3",{className:"font-medium text-lg inline-block hover:text-orange-600 transition duration-500 ease-in-out",children:r})]}),I.jsx("div",{className:"px-6 pb-6",children:p.map((g,E)=>I.jsx("span",{className:"inline-block bg-stone-500 rounded-full px-3 py-1 text-sm font-semibold text-neutral-900 mr-2 mb-2",children:g},E))})]})}const IP=()=>{const i=[...Vi].sort((t,r)=>new Date(r.date)-new Date(t.date)).slice(0,3);return I.jsxs(I.Fragment,{children:[I.jsx("div",{className:"h-32 bg-gradient-to-b from-transparent to-stone-900 w-full"}),I.jsxs(bn,{id:"queries",title:"LAST QUERIES",description:"Discover our most recently updated SPARQL queries.",variant:"dark",children:[I.jsx("div",{className:"grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-10",children:i.map((t,r)=>I.jsx(mS,{data:t,withPrefix:!0},r))}),I.jsx("div",{className:"flex justify-end mt-10",children:I.jsx(ut,{to:"/sparql-queries",className:"bg-gradient-to-r from-orange-500 to-orange-800 py-3 px-4 mx-3 rounded-md",children:"View all queries"})})]})]})},yP=()=>I.jsx(bn,{id:"why-sparql",title:"WHY LEARN SPARQL",description:"Unlock the power of querying linked data efficiently.",variant:"dark",children:I.jsx("div",{className:"flex flex-wrap mt-10 lg:mt-20",children:II.map((i,t)=>I.jsx("div",{className:"w-full sm:w-1/2 lg:w-1/3",children:I.jsxs("div",{className:"flex",children:[I.jsx("div",{className:"flex mx-6 h-12 w-12 bg-neutral-800 text-orange-700 justify-center items-center rounded-full aspect-square",children:wt.cloneElement(i.icon,{className:"w-5 h-5"})}),I.jsxs("div",{children:[I.jsx("h5",{className:"mt-1 mb-6 text-xl",children:i.text}),I.jsx("p",{className:"text-md mb-20 text-neutral-500 w-full text-justify",children:i.description})]})]})},t))})}),bP="/Sparql-fest/assets/SPARQL-ex2-CUHw8cA-.png",CP=()=>I.jsx(I.Fragment,{children:I.jsxs(bn,{id:"about",title:"ABOUT SPARQL FEST",description:"A community gathering to explore, learn, and create with SPARQL.",variant:"light",children:[I.jsxs("p",{className:"text-neutral-900 mb-4",children:[I.jsx("span",{className:"bg-gradient-to-r from-orange-500 to-red-700 text-transparent bg-clip-text",children:"SPARQL Fest"})," is a collaborative platform that brings together the community to learn, share, and explore SPARQL. It curates real SPARQL queries from people, organizations, and events to make the language easier to understand and apply."]}),I.jsxs("div",{className:"flex flex-wrap justify-center items-stretch",children:[I.jsx("div",{className:"relative p-2 w-full lg:w-1/2 flex items-stretch justify-center mb-12",children:I.jsx("img",{src:bP,alt:"Coding",className:"object-contain h-full w-auto max-h-full"})}),I.jsx("div",{className:"pt-12 w-full lg:w-1/2 flex flex-col justify-center",children:yI.map((i,t)=>I.jsxs("div",{className:"flex mb-12",children:[I.jsx("div",{className:"text-green-400 mx-6 bg-neutral-100 h-10 w-10 p-2 justify-center items-center rounded-full",children:I.jsx(xI,{})}),I.jsxs("div",{children:[I.jsx("h5",{className:"mt-1 mb-2 text-xl",children:i.title}),I.jsx("p",{className:"text-md text-neutral-500",children:i.description})]})]},t))})]})]})}),wP="/Sparql-fest/assets/PrefixeSparql-C_33p_FH.png",_P=()=>I.jsx(I.Fragment,{children:I.jsxs("div",{className:"flex flex-col md:flex-row items-center bg-white shadow-lg px-12 gap-6 py-20",children:[I.jsx("div",{className:"w-full md:w-1/2 px-10",children:I.jsx("img",{src:wP,alt:"SPARQL prefixes",className:"w-full h-auto rounded-xl object-cover"})}),I.jsxs("div",{className:"w-full md:w-1/2 mx-16 text-left space-y-4",children:[I.jsx("p",{className:"text-sm text-orange-600 uppercase tracking-wide",children:"What are prefixes?"}),I.jsx("h2",{className:"text-2xl font-bold text-gray-800",children:"Understand SPARQL Prefixes"}),I.jsx("p",{className:"text-gray-600 text-base",children:"SPARQL prefixes are shortcuts for long URIs, helping you write clearer, more readable queries. They reference vocabularies like RDF, OWL, schema.org, and biomedical ontologies like OBO or ORDO."}),I.jsx(ut,{to:"/sparql-prefixes",className:"inline-block mt-2 px-5 py-2 rounded-xl bg-gradient-to-r from-orange-500 to-orange-800  text-white font-medium hover:shadow-xl",children:"Explore all prefixes that we use →"})]})]})}),TP=()=>I.jsx(I.Fragment,{children:I.jsx(bn,{id:"contact",title:"CONTACT US",description:"Get in touch with the SPARQL Fest team.",variant:"dark",bg:!0})}),vP=()=>{const i=Ci();return P.useEffect(()=>{setTimeout(()=>{if(i.hash){const t=i.hash.slice(1),r=document.getElementById(t);r&&r.scrollIntoView({behavior:"smooth"})}},0)},[i]),I.jsxs("div",{children:[I.jsx(Jx,{}),I.jsx(IP,{}),I.jsx(CP,{}),I.jsx(yP,{}),I.jsx(_P,{}),I.jsx(TP,{})]})},xP=()=>{const[i,t]=P.useState(20),[r,s]=P.useState(""),[l,u]=P.useState(""),[p,h]=P.useState(""),[f,g]=P.useState([]),[E,R]=P.useState([]),y=[...new Set(Vi.map(v=>v.category))],w=[...new Set(Vi.map(v=>v.source).filter(Boolean))],b=[...new Set(Vi.flatMap(v=>v.sparqlConcepts))],X=[...new Set(Vi.flatMap(v=>v.ontologies))];P.useEffect(()=>{t(20)},[r,l,p,f,E]);const N=P.useMemo(()=>Vi.filter(v=>{const A=typeof v.name=="string"&&typeof r=="string"&&v.name.toLowerCase().includes(r.toLowerCase()),k=l?v.category===l:!0,U=p?v.source===p:!0,j=f.length>0?f.every(Z=>v.sparqlConcepts.includes(Z)):!0,K=E.length>0?E.every(Z=>v.ontologies.includes(Z)):!0;return A&&k&&U&&j&&K}),[r,l,p,f,E]);return I.jsx("div",{children:I.jsxs(bn,{id:"all-queries",title:"ALL SPARQL QUERIES",description:"Discover all our SPARQL queries.",variant:"dark",bg:!0,children:[I.jsxs("div",{className:"mb-8 grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4",children:[I.jsx("input",{type:"text",placeholder:"Search by name...",className:"p-2 rounded",value:r,onChange:v=>s(v.target.value)}),I.jsxs("select",{className:"p-2 rounded",value:l,onChange:v=>u(v.target.value),children:[I.jsx("option",{value:"",children:"All Categoires"}),y.map(v=>I.jsx("option",{value:v,children:v},v))]}),I.jsxs("select",{className:"p-2 rounded",value:p,onChange:v=>h(v.target.value),children:[I.jsx("option",{value:"",children:"All Sources"}),w.map(v=>I.jsx("option",{value:v,children:v},v))]}),I.jsx("select",{multiple:!0,className:"p-2 rounded",onChange:v=>g([...v.target.selectedOptions].map(A=>A.value)),children:b.map(v=>I.jsx("option",{value:v,children:v},v))}),I.jsx("select",{multiple:!0,className:"p-2 rounded",onChange:v=>R([...v.target.selectedOptions].map(A=>A.value)),children:X.map(v=>I.jsx("option",{value:v,children:v},v))})]}),I.jsx("div",{className:"grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-10",children:N.slice(0,i).map((v,A)=>I.jsx(mS,{data:v},A))}),i<N.length&&I.jsx("div",{className:"mt-6 text-center",children:I.jsx("button",{onClick:()=>t(v=>v+20),className:"bg-gradient-to-r from-orange-500 to-orange-800 py-2 px-3 rounded-md",children:"Load More"})})]})})};function PP(i,t,r){return t in i?Object.defineProperty(i,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):i[t]=r,i}function rE(i,t){var r=Object.keys(i);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(i);t&&(s=s.filter(function(l){return Object.getOwnPropertyDescriptor(i,l).enumerable})),r.push.apply(r,s)}return r}function sE(i){for(var t=1;t<arguments.length;t++){var r=arguments[t]!=null?arguments[t]:{};t%2?rE(Object(r),!0).forEach(function(s){PP(i,s,r[s])}):Object.getOwnPropertyDescriptors?Object.defineProperties(i,Object.getOwnPropertyDescriptors(r)):rE(Object(r)).forEach(function(s){Object.defineProperty(i,s,Object.getOwnPropertyDescriptor(r,s))})}return i}function XP(i,t){if(i==null)return{};var r={},s=Object.keys(i),l,u;for(u=0;u<s.length;u++)l=s[u],!(t.indexOf(l)>=0)&&(r[l]=i[l]);return r}function NP(i,t){if(i==null)return{};var r=XP(i,t),s,l;if(Object.getOwnPropertySymbols){var u=Object.getOwnPropertySymbols(i);for(l=0;l<u.length;l++)s=u[l],!(t.indexOf(s)>=0)&&Object.prototype.propertyIsEnumerable.call(i,s)&&(r[s]=i[s])}return r}function OP(i,t){return FP(i)||DP(i,t)||HP(i,t)||AP()}function FP(i){if(Array.isArray(i))return i}function DP(i,t){if(!(typeof Symbol>"u"||!(Symbol.iterator in Object(i)))){var r=[],s=!0,l=!1,u=void 0;try{for(var p=i[Symbol.iterator](),h;!(s=(h=p.next()).done)&&(r.push(h.value),!(t&&r.length===t));s=!0);}catch(f){l=!0,u=f}finally{try{!s&&p.return!=null&&p.return()}finally{if(l)throw u}}return r}}function HP(i,t){if(i){if(typeof i=="string")return aE(i,t);var r=Object.prototype.toString.call(i).slice(8,-1);if(r==="Object"&&i.constructor&&(r=i.constructor.name),r==="Map"||r==="Set")return Array.from(i);if(r==="Arguments"||/^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r))return aE(i,t)}}function aE(i,t){(t==null||t>i.length)&&(t=i.length);for(var r=0,s=new Array(t);r<t;r++)s[r]=i[r];return s}function AP(){throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`)}function LP(i,t,r){return t in i?Object.defineProperty(i,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):i[t]=r,i}function lE(i,t){var r=Object.keys(i);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(i);t&&(s=s.filter(function(l){return Object.getOwnPropertyDescriptor(i,l).enumerable})),r.push.apply(r,s)}return r}function cE(i){for(var t=1;t<arguments.length;t++){var r=arguments[t]!=null?arguments[t]:{};t%2?lE(Object(r),!0).forEach(function(s){LP(i,s,r[s])}):Object.getOwnPropertyDescriptors?Object.defineProperties(i,Object.getOwnPropertyDescriptors(r)):lE(Object(r)).forEach(function(s){Object.defineProperty(i,s,Object.getOwnPropertyDescriptor(r,s))})}return i}function BP(){for(var i=arguments.length,t=new Array(i),r=0;r<i;r++)t[r]=arguments[r];return function(s){return t.reduceRight(function(l,u){return u(l)},s)}}function hs(i){return function t(){for(var r=this,s=arguments.length,l=new Array(s),u=0;u<s;u++)l[u]=arguments[u];return l.length>=i.length?i.apply(this,l):function(){for(var p=arguments.length,h=new Array(p),f=0;f<p;f++)h[f]=arguments[f];return t.apply(r,[].concat(l,h))}}}function fl(i){return{}.toString.call(i).includes("Object")}function MP(i){return!Object.keys(i).length}function ys(i){return typeof i=="function"}function qP(i,t){return Object.prototype.hasOwnProperty.call(i,t)}function UP(i,t){return fl(t)||Ri("changeType"),Object.keys(t).some(function(r){return!qP(i,r)})&&Ri("changeField"),t}function WP(i){ys(i)||Ri("selectorType")}function jP(i){ys(i)||fl(i)||Ri("handlerType"),fl(i)&&Object.values(i).some(function(t){return!ys(t)})&&Ri("handlersType")}function kP(i){i||Ri("initialIsRequired"),fl(i)||Ri("initialType"),MP(i)&&Ri("initialContent")}function zP(i,t){throw new Error(i[t]||i.default)}var GP={initialIsRequired:"initial state is required",initialType:"initial state should be an object",initialContent:"initial state shouldn't be an empty object",handlerType:"handler should be an object or a function",handlersType:"all handlers should be a functions",selectorType:"selector should be a function",changeType:"provided value of changes should be an object",changeField:'it seams you want to change a field in the state which is not specified in the "initial" state',default:"an unknown error accured in `state-local` package"},Ri=hs(zP)(GP),$a={changes:UP,selector:WP,handler:jP,initial:kP};function QP(i){var t=arguments.length>1&&arguments[1]!==void 0?arguments[1]:{};$a.initial(i),$a.handler(t);var r={current:i},s=hs(KP)(r,t),l=hs(YP)(r),u=hs($a.changes)(i),p=hs(VP)(r);function h(){var g=arguments.length>0&&arguments[0]!==void 0?arguments[0]:function(E){return E};return $a.selector(g),g(r.current)}function f(g){BP(s,l,u,p)(g)}return[h,f]}function VP(i,t){return ys(t)?t(i.current):t}function YP(i,t){return i.current=cE(cE({},i.current),t),t}function KP(i,t,r){return ys(t)?t(i.current):Object.keys(r).forEach(function(s){var l;return(l=t[s])===null||l===void 0?void 0:l.call(t,i.current[s])}),r}var ZP={create:QP},JP={paths:{vs:"https://cdn.jsdelivr.net/npm/monaco-editor@0.52.2/min/vs"}};function $P(i){return function t(){for(var r=this,s=arguments.length,l=new Array(s),u=0;u<s;u++)l[u]=arguments[u];return l.length>=i.length?i.apply(this,l):function(){for(var p=arguments.length,h=new Array(p),f=0;f<p;f++)h[f]=arguments[f];return t.apply(r,[].concat(l,h))}}}function e2(i){return{}.toString.call(i).includes("Object")}function t2(i){return i||uE("configIsRequired"),e2(i)||uE("configType"),i.urls?(n2(),{paths:{vs:i.urls.monacoBase}}):i}function n2(){console.warn(ES.deprecation)}function i2(i,t){throw new Error(i[t]||i.default)}var ES={configIsRequired:"the configuration object is required",configType:"the configuration object should be an object",default:"an unknown error accured in `@monaco-editor/loader` package",deprecation:`Deprecation warning!
    You are using deprecated way of configuration.

    Instead of using
      monaco.config({ urls: { monacoBase: '...' } })
    use
      monaco.config({ paths: { vs: '...' } })

    For more please check the link https://github.com/suren-atoyan/monaco-loader#config
  `},uE=$P(i2)(ES),o2={config:t2},r2=function(){for(var t=arguments.length,r=new Array(t),s=0;s<t;s++)r[s]=arguments[s];return function(l){return r.reduceRight(function(u,p){return p(u)},l)}};function SS(i,t){return Object.keys(t).forEach(function(r){t[r]instanceof Object&&i[r]&&Object.assign(t[r],SS(i[r],t[r]))}),sE(sE({},i),t)}var s2={type:"cancelation",msg:"operation is manually canceled"};function ld(i){var t=!1,r=new Promise(function(s,l){i.then(function(u){return t?l(s2):s(u)}),i.catch(l)});return r.cancel=function(){return t=!0},r}var a2=ZP.create({config:JP,isInitialized:!1,resolve:null,reject:null,monaco:null}),RS=OP(a2,2),Ts=RS[0],Il=RS[1];function l2(i){var t=o2.config(i),r=t.monaco,s=NP(t,["monaco"]);Il(function(l){return{config:SS(l.config,s),monaco:r}})}function c2(){var i=Ts(function(t){var r=t.monaco,s=t.isInitialized,l=t.resolve;return{monaco:r,isInitialized:s,resolve:l}});if(!i.isInitialized){if(Il({isInitialized:!0}),i.monaco)return i.resolve(i.monaco),ld(cd);if(window.monaco&&window.monaco.editor)return IS(window.monaco),i.resolve(window.monaco),ld(cd);r2(u2,p2)(h2)}return ld(cd)}function u2(i){return document.body.appendChild(i)}function d2(i){var t=document.createElement("script");return i&&(t.src=i),t}function p2(i){var t=Ts(function(s){var l=s.config,u=s.reject;return{config:l,reject:u}}),r=d2("".concat(t.config.paths.vs,"/loader.js"));return r.onload=function(){return i()},r.onerror=t.reject,r}function h2(){var i=Ts(function(r){var s=r.config,l=r.resolve,u=r.reject;return{config:s,resolve:l,reject:u}}),t=window.require;t.config(i.config),t(["vs/editor/editor.main"],function(r){IS(r),i.resolve(r)},function(r){i.reject(r)})}function IS(i){Ts().monaco||Il({monaco:i})}function f2(){return Ts(function(i){var t=i.monaco;return t})}var cd=new Promise(function(i,t){return Il({resolve:i,reject:t})}),yS={config:l2,init:c2,__getMonacoInstance:f2},g2={wrapper:{display:"flex",position:"relative",textAlign:"initial"},fullWidth:{width:"100%"},hide:{display:"none"}},ud=g2,m2={container:{display:"flex",height:"100%",width:"100%",justifyContent:"center",alignItems:"center"}},E2=m2;function S2({children:i}){return wt.createElement("div",{style:E2.container},i)}var R2=S2,I2=R2;function y2({width:i,height:t,isEditorReady:r,loading:s,_ref:l,className:u,wrapperProps:p}){return wt.createElement("section",{style:{...ud.wrapper,width:i,height:t},...p},!r&&wt.createElement(I2,null,s),wt.createElement("div",{ref:l,style:{...ud.fullWidth,...!r&&ud.hide},className:u}))}var b2=y2,bS=P.memo(b2);function C2(i){P.useEffect(i,[])}var CS=C2;function w2(i,t,r=!0){let s=P.useRef(!0);P.useEffect(s.current||!r?()=>{s.current=!1}:i,t)}var Lt=w2;function gs(){}function Zo(i,t,r,s){return _2(i,s)||T2(i,t,r,s)}function _2(i,t){return i.editor.getModel(wS(i,t))}function T2(i,t,r,s){return i.editor.createModel(t,r,s?wS(i,s):void 0)}function wS(i,t){return i.Uri.parse(t)}function v2({original:i,modified:t,language:r,originalLanguage:s,modifiedLanguage:l,originalModelPath:u,modifiedModelPath:p,keepCurrentOriginalModel:h=!1,keepCurrentModifiedModel:f=!1,theme:g="light",loading:E="Loading...",options:R={},height:y="100%",width:w="100%",className:b,wrapperProps:X={},beforeMount:N=gs,onMount:v=gs}){let[A,k]=P.useState(!1),[U,j]=P.useState(!0),K=P.useRef(null),Z=P.useRef(null),ce=P.useRef(null),re=P.useRef(v),te=P.useRef(N),Xe=P.useRef(!1);CS(()=>{let J=yS.init();return J.then(L=>(Z.current=L)&&j(!1)).catch(L=>(L==null?void 0:L.type)!=="cancelation"&&console.error("Monaco initialization: error:",L)),()=>K.current?qe():J.cancel()}),Lt(()=>{if(K.current&&Z.current){let J=K.current.getOriginalEditor(),L=Zo(Z.current,i||"",s||r||"text",u||"");L!==J.getModel()&&J.setModel(L)}},[u],A),Lt(()=>{if(K.current&&Z.current){let J=K.current.getModifiedEditor(),L=Zo(Z.current,t||"",l||r||"text",p||"");L!==J.getModel()&&J.setModel(L)}},[p],A),Lt(()=>{let J=K.current.getModifiedEditor();J.getOption(Z.current.editor.EditorOption.readOnly)?J.setValue(t||""):t!==J.getValue()&&(J.executeEdits("",[{range:J.getModel().getFullModelRange(),text:t||"",forceMoveMarkers:!0}]),J.pushUndoStop())},[t],A),Lt(()=>{var J,L;(L=(J=K.current)==null?void 0:J.getModel())==null||L.original.setValue(i||"")},[i],A),Lt(()=>{let{original:J,modified:L}=K.current.getModel();Z.current.editor.setModelLanguage(J,s||r||"text"),Z.current.editor.setModelLanguage(L,l||r||"text")},[r,s,l],A),Lt(()=>{var J;(J=Z.current)==null||J.editor.setTheme(g)},[g],A),Lt(()=>{var J;(J=K.current)==null||J.updateOptions(R)},[R],A);let _e=P.useCallback(()=>{var V;if(!Z.current)return;te.current(Z.current);let J=Zo(Z.current,i||"",s||r||"text",u||""),L=Zo(Z.current,t||"",l||r||"text",p||"");(V=K.current)==null||V.setModel({original:J,modified:L})},[r,t,l,i,s,u,p]),Ee=P.useCallback(()=>{var J;!Xe.current&&ce.current&&(K.current=Z.current.editor.createDiffEditor(ce.current,{automaticLayout:!0,...R}),_e(),(J=Z.current)==null||J.editor.setTheme(g),k(!0),Xe.current=!0)},[R,g,_e]);P.useEffect(()=>{A&&re.current(K.current,Z.current)},[A]),P.useEffect(()=>{!U&&!A&&Ee()},[U,A,Ee]);function qe(){var L,V,ne,Re;let J=(L=K.current)==null?void 0:L.getModel();h||((V=J==null?void 0:J.original)==null||V.dispose()),f||((ne=J==null?void 0:J.modified)==null||ne.dispose()),(Re=K.current)==null||Re.dispose()}return wt.createElement(bS,{width:w,height:y,isEditorReady:A,loading:E,_ref:ce,className:b,wrapperProps:X})}var x2=v2;P.memo(x2);function P2(i){let t=P.useRef();return P.useEffect(()=>{t.current=i},[i]),t.current}var X2=P2,el=new Map;function N2({defaultValue:i,defaultLanguage:t,defaultPath:r,value:s,language:l,path:u,theme:p="light",line:h,loading:f="Loading...",options:g={},overrideServices:E={},saveViewState:R=!0,keepCurrentModel:y=!1,width:w="100%",height:b="100%",className:X,wrapperProps:N={},beforeMount:v=gs,onMount:A=gs,onChange:k,onValidate:U=gs}){let[j,K]=P.useState(!1),[Z,ce]=P.useState(!0),re=P.useRef(null),te=P.useRef(null),Xe=P.useRef(null),_e=P.useRef(A),Ee=P.useRef(v),qe=P.useRef(),J=P.useRef(s),L=X2(u),V=P.useRef(!1),ne=P.useRef(!1);CS(()=>{let B=yS.init();return B.then(z=>(re.current=z)&&ce(!1)).catch(z=>(z==null?void 0:z.type)!=="cancelation"&&console.error("Monaco initialization: error:",z)),()=>te.current?_():B.cancel()}),Lt(()=>{var z,G,$,de;let B=Zo(re.current,i||s||"",t||l||"",u||r||"");B!==((z=te.current)==null?void 0:z.getModel())&&(R&&el.set(L,(G=te.current)==null?void 0:G.saveViewState()),($=te.current)==null||$.setModel(B),R&&((de=te.current)==null||de.restoreViewState(el.get(u))))},[u],j),Lt(()=>{var B;(B=te.current)==null||B.updateOptions(g)},[g],j),Lt(()=>{!te.current||s===void 0||(te.current.getOption(re.current.editor.EditorOption.readOnly)?te.current.setValue(s):s!==te.current.getValue()&&(ne.current=!0,te.current.executeEdits("",[{range:te.current.getModel().getFullModelRange(),text:s,forceMoveMarkers:!0}]),te.current.pushUndoStop(),ne.current=!1))},[s],j),Lt(()=>{var z,G;let B=(z=te.current)==null?void 0:z.getModel();B&&l&&((G=re.current)==null||G.editor.setModelLanguage(B,l))},[l],j),Lt(()=>{var B;h!==void 0&&((B=te.current)==null||B.revealLine(h))},[h],j),Lt(()=>{var B;(B=re.current)==null||B.editor.setTheme(p)},[p],j);let Re=P.useCallback(()=>{var B;if(!(!Xe.current||!re.current)&&!V.current){Ee.current(re.current);let z=u||r,G=Zo(re.current,s||i||"",t||l||"",z||"");te.current=(B=re.current)==null?void 0:B.editor.create(Xe.current,{model:G,automaticLayout:!0,...g},E),R&&te.current.restoreViewState(el.get(z)),re.current.editor.setTheme(p),h!==void 0&&te.current.revealLine(h),K(!0),V.current=!0}},[i,t,r,s,l,u,g,E,R,p,h]);P.useEffect(()=>{j&&_e.current(te.current,re.current)},[j]),P.useEffect(()=>{!Z&&!j&&Re()},[Z,j,Re]),J.current=s,P.useEffect(()=>{var B,z;j&&k&&((B=qe.current)==null||B.dispose(),qe.current=(z=te.current)==null?void 0:z.onDidChangeModelContent(G=>{ne.current||k(te.current.getValue(),G)}))},[j,k]),P.useEffect(()=>{if(j){let B=re.current.editor.onDidChangeMarkers(z=>{var $;let G=($=te.current.getModel())==null?void 0:$.uri;if(G&&z.find(de=>de.path===G.path)){let de=re.current.editor.getModelMarkers({resource:G});U==null||U(de)}});return()=>{B==null||B.dispose()}}return()=>{}},[j,U]);function _(){var B,z;(B=qe.current)==null||B.dispose(),y?R&&el.set(u,te.current.saveViewState()):(z=te.current.getModel())==null||z.dispose(),te.current.dispose()}return wt.createElement(bS,{width:w,height:b,isEditorReady:j,loading:f,_ref:Xe,className:X,wrapperProps:N})}var O2=N2,F2=P.memo(O2);const D2=({dataQuery:i})=>{const[t,r]=P.useState(!1),[s,l]=P.useState({x:0,y:0}),u=P.useRef(null),[p,h]=P.useState(100),f=b=>{u.current=b,g(),b.onDidChangeModelContent(g)},g=()=>{var v,A;if(!u.current)return;const b=((A=(v=u.current.getModel())==null?void 0:v.getLineCount)==null?void 0:A.call(v))??1,X=u.current.getOption(28),N=b*(X*20)+20;h(N)},[E,R]=P.useState(""),y=()=>{navigator.clipboard.writeText(E),r(!0),setTimeout(()=>r(!1),1200)},w=b=>{l({x:b.clientX,y:b.clientY})};return I.jsxs("div",{className:"relative bg-gray-800 rounded-xl overflow-hidden mb-6 shadow-lg",onMouseMove:w,children:[I.jsx("div",{className:"bg-gray-700 text-white text-xs uppercase px-4 py-4 font-mono tracking-wide",children:"sparql"}),I.jsx("button",{onClick:y,className:"absolute top-2 right-2 text-xs bg-gray-600 text-white px-2 py-1 rounded hover:bg-gray-500",children:"Copier"}),t&&I.jsx("div",{className:"fixed z-50  text-white text-xs px-2 py-1 rounded shadow transition-opacity duration-300 pointer-events-none",style:{top:s.y+10,left:s.x+10},children:"✅ Copié !"}),I.jsx(F2,{height:p,defaultLanguage:"sparql",defaultValue:i,onMount:f,onChange:b=>R(b??""),options:{readOnly:!1,minimap:{enabled:!1},scrollBeyondLastLine:!1,fontSize:14,theme:"vs-dark",lineNumbers:"on"}})]})},H2=({data:i})=>{const{category:t,date:r,ontologies:s,sparqlConcepts:l}=i;return I.jsx("div",{children:I.jsxs("ul",{class:"mt-8 mx-auto max-w-xs text-left font-medium text-lg leading-none",children:[I.jsxs("li",{children:[I.jsxs("p",{class:"py-3.5 w-full flex items-center text-orange-100 border-b border-neutral-500 pb-2",children:[I.jsx("span",{class:"ml-5 mr-2.5 w-1 h-7 bg-orange-700 rounded-r-md"}),"SPARQL Prefixes used"]}),I.jsx("div",{className:"py-4 px-2 pb-6 text-right",children:(s||[]).map((u,p)=>I.jsx(ut,{to:`/sparql-prefixes/${u}`,className:"inline-block bg-stone-300 rounded-full px-3 py-1 text-sm font-semibold text-orange-950 mr-2 mb-2",children:u},p))})]}),I.jsxs("li",{children:[I.jsxs("p",{class:"py-3.5 w-full flex items-center text-orange-100 border-b border-neutral-500 pb-2",children:[I.jsx("span",{class:"ml-5 mr-2.5 w-1 h-7 bg-orange-700 rounded-r-md"}),"Category"]}),I.jsx("div",{className:"py-4 px-2 pb-6 text-right",children:I.jsx("span",{className:"inline-block bg-orange-800 rounded-full px-3 py-1 text-sm font-semibold text-orange-100 mr-2 mb-2",children:t})})]}),I.jsxs("li",{children:[I.jsxs("p",{class:"py-3.5 w-full flex items-center text-orange-100 border-b border-neutral-500 pb-2",children:[I.jsx("span",{class:"ml-5 mr-2.5 w-1 h-7 bg-orange-700 rounded-r-md"}),"SPARQL Concepts"]}),I.jsx("div",{className:"py-4 px-2 pb-6 text-right",children:(l||[]).map((u,p)=>I.jsx("span",{className:"inline-block bg-orange-300 rounded-full px-3 py-1 text-sm font-semibold text-orange-950 mr-2 mb-2",children:u},p))})]})]})})};function A2(i){return bi({attr:{viewBox:"0 0 24 24",fill:"none",stroke:"currentColor",strokeWidth:"2",strokeLinecap:"round",strokeLinejoin:"round"},child:[{tag:"path",attr:{d:"M3 12h1m8 -9v1m8 8h1m-15.4 -6.4l.7 .7m12.1 -.7l-.7 .7"},child:[]},{tag:"path",attr:{d:"M9 16a5 5 0 1 1 6 0a3.5 3.5 0 0 0 -1 3a2 2 0 0 1 -4 0a3.5 3.5 0 0 0 -1 -3"},child:[]},{tag:"path",attr:{d:"M9.7 17l4.6 0"},child:[]}]})(i)}const L2="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAMCAgMCAgMDAwMEAwMEBQgFBQQEBQoHBwYIDAoMDAsKCwsNDhIQDQ4RDgsLEBYQERMUFRUVDA8XGBYUGBIUFRT/2wBDAQMEBAUEBQkFBQkUDQsNFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBT/wgARCABkAGQDASIAAhEBAxEB/8QAGwABAAMBAQEBAAAAAAAAAAAAAAQGBwUDAgj/xAAbAQEAAgMBAQAAAAAAAAAAAAAABAYBAgMFB//aAAwDAQACEAMQAAAB/VIAAAAAAAAKl6Zb8eBauVvWK/c3z9r6tOuPeAEuIAABglGv1PpH1qNbKZf5W+jXGp2z3vmAT4IAAFHzrfeNX7BlkPTbLGlPYtFWDOAIudaPyNufN5lo5OdYPp0/piFHmyyZULZHxtXrX4+52Br1AAAAAAAAAAAAAA//xAAlEAABBAEEAQQDAAAAAAAAAAAEAgMFBgEAFDA1FQcSFjYgQFD/2gAIAQEAAQUC/ecsozBdlnMV+HpF9NNmJGVYjExsgmSH4JjtJWc8XcjY8f09CwrLlbqvV8Ex2l9+zXMhb+m/rNV6vgnW8ty17HU4Xak5Wo5rYg1hvKIngskOoxI0gsZpUy2lQAD8wWy0lhrhLhxDcorACFNMoYR+JT+BRRLfjC2bJhZOLkh8f5qBuSLOrAg1lQ4n5a37VHZ8oqfOfT504bQW/S5ohhJI5cIOalyHYdVirD6TXmEZJr7JOXKwMtaYAZIxkUgrTMCMw0RCDE6Cjto5/I//xAAuEQABAgQBCAsAAAAAAAAAAAABAgMABAUREhAgISJRkcHRExQwMTJAQXGhsvD/2gAIAQMBAT8B7FdG6wkCV8WAq3W5xTKQ44wh+YOqpVhuJ4RNtpafWhPcDm0lQEwwdqCPqeEO4RLSrY2qPwrnE8bzTnvmydSS230Mwm6YcqrLSSJZOnafxgm+k5VBXpBxxrxrwQq8AG/kP//EACoRAAAFAgQDCQAAAAAAAAAAAAABAgMEESEFMUHREBIgExQiIzBAYXGx/9oACAECAQE/AfRhT1uLcS9kR77DEZnYNqNrMtyERxTrCFqzMumLZx9OtdxiB+S5X4/SEAqRW/rpl4atbveIyuVQThkmQsjmLsWhAipYuJU1Fh4RYWFvYf/EADwQAAIBAgQCBAsECwAAAAAAAAECAwARBBIhMRMiBUFRYRQjMDIzQlJxcpGygYKh0RAgJEBQYsHC0vDx/9oACAEBAAY/Av36SCVXTKbZtxU2M4ZlZdFUdp7e6vA8eTiFxLcjKvoz+VKZbktsFG9cZVKC9rHyOK+M1iYZ/GdH4hUSaJtrZRrWJkwpz47GOVhc7xR/7/Suh2YlmIlJJ+KvvnyOK+M1iPhT6RXQ+Y3/AGCN/tP/ACuhfhk+qvvnyOJB62vWG6QUXhxES838w3FdCKoLMejoQAPtro7An0kEPOOxjqaQn1mJ8iMRCLyoLFe0U8DxpPh286GUXFRPBgYYpooxEkh5iijsrrNzd5D1UsaCyqLDyWaWEZvaGhq+Rn7magkahFHUv600xFxGhe3uocfEYPFK0Jly4I3aMi3Kddd99K4MmCxMFpBC7vksrkXA0bst864mGwU03jI1tmT1mt7Wh7jani5uUuubMmrLe4te/UdbVK8XR+KMiw8Yjk8Wp80nm7thrURaCbhFlibEgDhiQ9W99za9qzjAYsxcLwgPyWMXt+d+G9YfDAAxywSS5uvQoP7qxBgWBfBllkfiAnOFkdQBrp5m9LHOkEs86I0PDuoBZrWbfa+9MuL4EiWuskIK69liT87/AKJIX82RSpt31CHzDhLlGU+7/EVKzZ/GTLOdfWAAH0imLT4iR7KqyMwzJlbMLG2uoG96ltNPwpM94cwyjNudu+vTTwhoxDIImtxFHUfx2o2mxEcBlWbwdH8XmBBvt3UsAL5Fwvge/qfnULLNLh5YQVSWIjNY7jUEdQ+VSIue0kPBYltSLsSffdjQz5tIhELNa1jcH33FPI2InxMjDLmmbYe4WH8J/8QAKRABAQACAQMDAwMFAAAAAAAAAREAITFBUWEwcfCRocEQIPFAUIGx4f/aAAgBAQABPyH+uQgPM/6ZomQhnSrp8dcpIB3HLIfx+uKTZ3zHPjriOs+t16WtM9hWxa+PuYdMP95MPJTfVezFjjHVetz5/t6em3XM+vb/AKfon8/29HsIH2S45QcjgKJ54++HxHKq9JjxsSXgmakJftZ+PR0OYXPi8mMG3qb+HOSKF8oXxy4g2XI/8vGRUMXg9Jug/mHP+csHnGfbOLjxQ/c6ZqHKC/jGhUV2NZe2p1DjOTP0Di7LVsCaYWswneS1Fu77ZYVJKIADIFDPgEBTnZaFnsUD3K07IOOMfG1PQHgy4F1csWJBIPPUnDpy4wrqD3Dp4j+hiA8fpl8KRrtt4xhHfB+3miDEsdGX/N8ny+QaP0sp3rQRyUcsw0r3rBvKn54e8T7mJt2aaEQUIttYJkPkEWXLqZYLozVAE0XTS99od85ZJCYjEbchlnWXeW8SNbYHbs5xEjgQSpQHqHRib2rhFJ3LXzg4LoKABE2AEfGCQsjMOgAe8r3/ALT/AP/aAAwDAQACAAMAAAAQ888888888BbU8888CiX8888K1888Ag1BO1888888888888888//EACQRAQABAwIFBQAAAAAAAAAAAAERACExUZEQIEFhgTBAscHx/9oACAEDAQE/EPRdZDgN5cJos06SBbNXJwLqw5nSYat8RcE4QHKuWwHdQDZPinZtInYBdzvRINXzyimFhMnXthwiJSU5kZI3UxpYmkakrxYy6GI66eKiLLUYzSbG3590G5t7D//EACQRAQABAwIFBQAAAAAAAAAAAAERACExQVEQIIGRsTBAYXGh/9oACAECAQE/EPRjIRhFmFy7KiORDfFwx9NI3IF5bz0WPgS/qVC6zDrJ4GkA7fHLeQc7PnOpCNAFesdgCd7tAQIDihQddTSaiClRb2H/xAAlEAEBAAICAQMEAwEAAAAAAAABEQAhMUFRMGHwcYGhsSBAUMH/2gAIAQEAAT8Q/vJH2lWMqDH2HNhW+m0ZcSirOg2KaGickdL1E3e6ysT7B4zOSgkcpzhQodBUG9fX0fm/OKWPN3vpqh37h4QjqSAIkeYD2BJGPYYQtajtV3XOf5a9H835wz38nf6ore7+/wCDBLZOf5a9GFKNbyIfvA5qVVzXQD6pHDj7i1LIANqupihhoLYfUj+zCoQLfg5+i3KYem091HXY+QHSrZ7eQPg59uKZdpTvCGGnJtbu537bwC1V7XX6Ap7F7TCH39LlYUqz5VH5MNGBp/wC5wByEvsd+/8AJ0rDxSwfLpiGH7jgEW+A1EDRqlGO1aCWGJqKxp6T9FYAKC1YAyp7DNq2aDxCRC4WEJ6Ixm7cQJQoS+Te8f3mCEWLRTflveGkeI0VAnXNj4sQM7SltLeDdE9oPTVZqZAk8LXI9kUcxNAAaXY3DaUi50RBtsezE2Ls4Wydxi4eKDorDkrfyedLQHck+gpvC2qEukN/spyC2MrRUUTi5CRQXSqJJhHWNGXZXlrCBWmzSASKgjWARFw3bDRVJxdkCvgPCc6wBjxxCax9F2hEx5GgKsUVdTS8DKsu4j9wMlIjWOn3V6U3sKvSVQ/yf//Z",B2="/Sparql-fest/assets/SIB2-B9o4XN4z.jpg",kd=[{id:"AMC",name:"Amsterdam University Medical Center",prefix:"AMC",logo:L2,namespace:"https://www.amsterdamumc.org/nl.htm"},{id:"SIB",name:"SIB Swiss",logo:B2,prefix:"SIB",namespace:"https://sib-swiss.github.io/sparql-examples/",other:[{name:"A large collection of bioinformatics question-query pairs over federated knowledge graphs: methodology and applications",type:"article",link:"https://arxiv.org/abs/2410.06010",metadata:"author : Jerven Bolleman and Vincent Emonet and Adrian Altenhoff and Amos Bairoch and Marie-Claude Blatter and Alan Bridge and Severine Duvaud and Elisabeth Gasteiger and Dmitry Kuznetsov and Sebastien Moretti and Pierre-Andre Michel and Anne Morgat and Marco Pagni and Nicole Redaschi and Monique Zahn-Zabal and Tarcisio Mendes de Farias and Ana Claudia Sima"}]}],M2=()=>{const[i,t]=P.useState(!1),{slug:r}=Pd(),[s,l]=P.useState([]);if(P.useEffect(()=>{const f=Vi.filter(g=>g.slug===r);f.length>0?l(f[0]):window.location.href="/"},[r]),!s)return I.jsx("div",{className:"text-center py-10",children:"Loading or not found..."});const u=s.image||"https://images.unsplash.com/photo-1506744038136-46273834b3fb?auto=format&fit=crop&w=1400&q=80",p=s.source==="SIB",h=kd.find(f=>f.id===s.source);return console.log(h),I.jsxs(I.Fragment,{children:[I.jsx("div",{className:`relative w-full h-64 ${p?"bg-contain bg-no-repeat":"bg-cover"} bg-center bg-white overflow-hidden flex pl-6`,style:{backgroundImage:`url(${u})`},children:I.jsxs("p",{class:"px-4 py-6 text-xl font-bold text-neutral-300/30 text-opacity-80",children:["  ",I.jsx("pre",{children:s.query})]})}),I.jsx("div",{class:"container mx-auto mt-8",children:I.jsxs("div",{class:"flex flex-wrap justify-between",children:[I.jsx("div",{class:"w-full md:w-8/12 px-4",children:I.jsxs(bn,{id:s.slug,title:s.name,description:s.date,variant:"dark",bg:!1,children:[(s.source||s.context)&&I.jsxs(I.Fragment,{children:[I.jsx("h2",{className:"text-xl font-bold mt-4 mb-2",children:"Context"}),s.context?I.jsx("p",{className:"text-neutral-300 mb-6",children:s.context}):h?I.jsxs("div",{className:"text-neutral-300 mb-6",children:["This query was created by ",I.jsx("strong",{children:h.name}),". You can find more information at:"," ",I.jsxs("ul",{className:"list-disc list-inside mt-2",children:[I.jsx("li",{children:I.jsx("a",{href:h.namespace,target:"_blank",rel:"noopener noreferrer",className:"underline text-blue-400",children:h.namespace})}),I.jsx("li",{children:I.jsxs(ut,{to:`/other-resources/${h.id}`,className:"underline text-blue-400",children:["Our internal resources about ",h.name]})})]})]}):I.jsx("div",{className:"text-neutral-300 mb-6",children:"No context available."})]}),I.jsx("h2",{class:"text-xl font-bold mt-4 mb-2",children:"Description"}),I.jsx("p",{class:"text-neutral-300 mb-6",children:s.description}),I.jsxs("h2",{className:"flex items-center cursor-pointer select-none text-xl font-semibold mb-3",onClick:()=>t(!i),children:[I.jsx(A2,{className:"text-yellow-500 mr-2",size:24}),"Hints",I.jsx("button",{"aria-label":i?"Hide hints":"Show hints",className:"ml-auto text-sm text-stone-200 hover:text-orange-è00",onClick:f=>{f.stopPropagation(),t(!i)},children:i?"Hide":"Show"})]}),i&&I.jsx("ol",{className:"ml-6 list-disc list-inside space-y-2 text-neutral-300",children:(s.inidces||[]).map((f,g)=>I.jsx("li",{children:f},g))}),I.jsx("h2",{class:"text-xl font-bold mt-10 mb-4",children:"Query"}),I.jsx(D2,{dataQuery:s.query}),I.jsx("h2",{class:"text-xl font-bold mt-10 mb-2",children:"Results"}),(s==null?void 0:s.rdfResultExample)==""&&I.jsx("p",{class:"text-neutral-300 mb-6 flex justify-center",children:" No result available for this query "})]})}),I.jsx("div",{class:"w-full md:w-4/12 px-4 mb-8",children:I.jsx("div",{class:"px-4 py-6 rounded",children:I.jsx(H2,{data:s})})})]})})]})},q2=()=>I.jsxs("div",{className:"mt-10 flex flex-col items-center justify-center text-center",children:[I.jsx("h1",{className:"text-2xl mb-6 ",children:"This section is still in Progress ..."}),I.jsx(ut,{to:"/",className:"bg-gradient-to-r from-orange-500 to-orange-800 py-3 px-4 mx-3 rounded-md",children:"Back to home"})]});function U2(i){return bi({attr:{viewBox:"0 0 512 512"},child:[{tag:"path",attr:{d:"M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"},child:[]}]})(i)}const _S=({data:i})=>I.jsxs("div",{className:"w-full p-10",children:[I.jsxs("div",{className:"flex items-center justify-between mb-4",children:[I.jsx("h5",{className:"text-xl font-bold leading-none text-gray-100 ",children:"View all"}),I.jsx("a",{href:"#",className:"text-sm font-medium text-orange-600 hover:underline",children:"Later filter"})]}),I.jsx("div",{className:"flow-root",children:I.jsx("ul",{role:"list",className:"divide-y divide-gray-200",children:i.map((t,r)=>I.jsx("li",{className:"py-3 sm:py-4",children:I.jsxs("div",{className:"flex items-center",children:[I.jsx("div",{className:"shrink-0",children:I.jsx("img",{className:"w-20 h-20 rounded-2xl p-1 border border-orange-500",src:t.logo,alt:`${t.name} image`})}),I.jsxs("div",{className:"flex-1 min-w-0 ms-4",children:[I.jsx("p",{className:"text-md font-medium text-gray-100  ",children:t.name}),I.jsx("p",{className:"text-sm text-stone-400 ",children:t.description}),I.jsx("p",{className:"text-xs text-orange-800 pt-2  ",children:t.namespace})]}),I.jsxs("div",{className:"relative w-64 h-22",children:[I.jsx("div",{className:"absolute top-0 right-0 text-sm font-semibold  bg-orange-800 rounded-full px-3 py-1 text-orange-100 mr-2 mb-2",children:t.prefix}),I.jsxs(ut,{to:t.prefix,className:"absolute bottom-0 right-0 flex items-center gap-1 text-xs text-stone-400 hover:text-orange-500",children:[I.jsx("span",{children:"View details"}),I.jsx(U2,{})]})]})]})},r))})})]}),dE="/Sparql-fest/assets/rdf-8oo4Zwka.png",W2="/Sparql-fest/assets/owl-BgRu7oc_.png",j2="/Sparql-fest/assets/ordo-C2qQvMGb.png",TS="/Sparql-fest/assets/unknown-CnBua-wC.avif",k2="/Sparql-fest/assets/hoom-B3r_aD00.jpg",z2="/Sparql-fest/assets/dc-Cj7eLQMX.jpeg",G2=[{prefix:"EX",name:"ex",namespace:"https://www.bgee.org/sparql/.well-known/sparql-examples/"},{prefix:"RDFS",name:"rdfs",namespace:"http://www.w3.org/2000/01/rdf-schema#"},{prefix:"SCHEMA",name:"schema",namespace:"https://schema.org/"},{prefix:"SH",name:"sh",namespace:"http://www.w3.org/ns/shacl#"},{prefix:"RDF",name:"rdf",namespace:"http://www.w3.org/1999/02/22-rdf-syntax-ns#"},{prefix:"SPEX",name:"spex",namespace:"https://purl.expasy.org/sparql-examples/ontology#"},{prefix:"DCTERMS",name:"dcterms",namespace:"http://purl.org/dc/terms/"},{prefix:"EX120",name:"ex120",namespace:"https://sparql.uniprot.org/.well-known/sparql-examples/120_proteins_and_diseases_linked#"},{prefix:"UP",name:"up",namespace:"http://purl.uniprot.org/core/"}],Q2=[{prefix:"RDF",name:"RDF ",namespace:"http://www.w3.org/1999/02/22-rdf-syntax-ns#",description:"RDF provides a data model for representing information about resources in the web.",logo:dE,documentation:"https://www.w3.org/TR/rdf-schema/",download:"https://www.w3.org/TR/rdf-schema/rdf-schema.rdf",version:"1.1",category:"W3C Vocabulary",usedIn:["Linked Data","OWL","SPARQL Basics"]},{prefix:"OWL",name:"OWL Ontology",namespace:"http://www.w3.org/2002/07/owl#",description:"The Web Ontology Language (OWL) is used to define complex vocabularies and relationships.",logo:W2,documentation:"https://www.w3.org/TR/owl2-overview/",download:"https://www.w3.org/2002/07/owl.rdf",version:"2",category:"Ontology",usedIn:["BioPortal","Ontology Mapping"]},{prefix:"ORDO",name:"Orphanet Rare Disease Ontology",namespace:"http://www.orpha.net/ORDO/",description:"ORDO is a structured vocabulary for rare diseases derived from Orphanet data.",logo:j2,documentation:"https://www.orpha.net/consor/cgi-bin/Education_AboutOrdo.php",download:"http://www.orphadata.org/data/ORDO/ORDO.owl",version:"2024-05-01",category:"Biomedical Ontology",usedIn:["Rare Disease Research","SPARQL for Health"]},{prefix:"RDFS",name:"RDF Schema",namespace:"http://www.w3.org/2000/01/rdf-schema#",description:"RDFS is a semantic extension of RDF providing mechanisms to describe groups of related resources and the relationships between them.",logo:dE,documentation:"https://www.w3.org/TR/rdf-schema/",download:"https://www.w3.org/TR/rdf-schema/rdf-schema.rdf",version:"1.1",category:"W3C Vocabulary",usedIn:["Class/Property Hierarchies","Linked Data"]},{prefix:"OBOINOWL",name:"OBO in OWL",namespace:"http://www.geneontology.org/formats/oboInOwl#",description:"OBOinOWL provides properties and conventions used for representing OBO ontologies in OWL.",logo:TS,documentation:"https://github.com/information-artifact-ontology/oboInOwl",download:"http://purl.obolibrary.org/obo/iao/oboInOwl.owl",version:"2024-01",category:"Ontology Metadata",usedIn:["OBO Ontologies","Ontology Metadata"]},{prefix:"HOOM",name:"Health Outcome Ontology Model",namespace:"http://w3id.org/hoom#",description:"HOOM is an ontology for modeling health outcomes, especially in the context of rare diseases.",logo:k2,documentation:"https://github.com/rare-disease-semantic-models/HOOM",download:"https://raw.githubusercontent.com/rare-disease-semantic-models/HOOM/main/hoom.owl",version:"1.0",category:"Biomedical Ontology",usedIn:["Health Outcome Modeling","Rare Disease Projects"]},{prefix:"DC",name:"Dublin Core",namespace:"http://purl.org/dc/elements/1.1/",description:"The Dublin Core Metadata Element Set is a standard for cross-domain resource description.",logo:z2,documentation:"https://www.dublincore.org/specifications/dublin-core/dces/",download:"http://dublincore.org/2012/06/14/dcelements.rdf",version:"1.1",category:"Metadata Standard",usedIn:["Metadata Description","Digital Libraries","SPARQL Metadata"]}],_d=new Map(Q2.map(i=>[i.prefix.toUpperCase(),i]));for(const i of G2){const t=i.prefix.toUpperCase();_d.has(t)||_d.set(t,{...i,description:"No description available.",logo:TS,documentation:null,download:null,version:null,category:"Unknown",usedIn:[]})}const vS=Array.from(_d.values()),V2=()=>I.jsx(I.Fragment,{children:I.jsx(bn,{id:"sparql-prefixes",title:"SPARQL PREFIXES WE USE",description:"List of all prefixes that are used in our queries.",variant:"dark",children:I.jsx("div",{className:"flex-center",children:I.jsx(_S,{data:vS})})})}),Y2=()=>I.jsx(q2,{}),K2=()=>I.jsx(bn,{id:"resources",title:"Other resources",description:"List of all the resources we use or which may be interesting for learning sparql.",variant:"dark",children:I.jsx("div",{className:"flex-center",children:I.jsx(_S,{data:kd})})}),Z2=()=>{const{slug:i}=Pd(),[t,r]=P.useState([]);return P.useEffect(()=>{const s=vS.filter(l=>l.prefix===i);s.length>0?r(s[0]):window.location.href="/"},[i]),t?I.jsxs("div",{class:"w-full px-4",children:[I.jsx("div",{className:"flex flex-col items-center mt-20",children:I.jsx("img",{src:t.logo,className:"w-25 h-25",alt:t.prefix})}),I.jsxs(bn,{id:t.prefix,title:t.name,description:t.prefix,variant:"dark",bg:!1,children:[I.jsx("h2",{class:"text-xl font-bold mt-4 mb-2",children:"Description"}),I.jsx("p",{class:"text-neutral-300 mb-6",children:t.description}),I.jsx("p",{class:"text-neutral-400 mb-6",children:t.version}),I.jsx("p",{class:"text-neutral-400 mb-6",children:t.category}),I.jsx("h2",{class:"text-xl font-bold mt-4 mb-2",children:"documentation"}),I.jsx("p",{class:"text-neutral-300 mb-6",children:t.documentation}),I.jsx("h2",{class:"text-md font-bold mt-4 mb-2",children:"download"}),I.jsx("p",{class:"text-neutral-300 mb-6",children:t.download}),I.jsx("h2",{class:"text-md font-bold mt-4 mb-2",children:"version"}),I.jsx("p",{class:"text-neutral-300 mb-6",children:t.version})]})]}):I.jsx("div",{className:"text-center py-10",children:"Loading or not found..."})},J2=()=>{const{slug:i}=Pd(),[t,r]=P.useState([]);return P.useEffect(()=>{const s=kd.filter(l=>l.id===i);s.length>0?r(s[0]):window.location.href="/"},[i]),t?I.jsxs("div",{class:"w-full px-4",children:[I.jsx("div",{className:"flex flex-col items-center mt-20",children:I.jsx("img",{src:t.logo,className:"w-25 h-25",alt:t.id})}),I.jsxs(bn,{id:t.id,title:t.name,description:t.prefix,variant:"dark",bg:!1,children:[I.jsx("h2",{class:"text-xl font-bold mt-4 mb-2",children:"Description"}),I.jsxs("p",{class:"text-neutral-300 mb-6",children:["Find more information at : ",I.jsx("a",{href:t.namespace,target:"_blank",className:"text-blue-400 hover:underline",children:t.namespace})]}),t!=null&&t.other?I.jsxs(I.Fragment,{children:[I.jsx("h2",{class:"text-xl font-bold mt-4 mb-2",children:"Other Info"}),I.jsx("ul",{className:"space-y-4",children:t.other.map((s,l)=>I.jsxs("li",{className:"p-4 border rounded-lg shadow-md text-neutral-300",children:[I.jsxs("div",{className:"font-bold mb-2",children:["Type: ",I.jsx("span",{className:"bg-gradient-to-r from-orange-500 to-orange-800 py-1 px-2 rounded-md",children:s.type})]}),I.jsxs("div",{className:"font-bold",children:["Name: ",I.jsx("span",{className:"font-normal",children:s.name})]}),I.jsxs("div",{className:"font-bold",children:["Metadata: ",I.jsx("br",{})," ",I.jsx("span",{className:"font-normal",children:s.metadata})]}),I.jsxs("div",{className:"font-bold",children:["Link: ",I.jsx("a",{href:s.link,target:"_blank",rel:"noopener noreferrer",className:"text-blue-400 hover:underline font-normal",children:s.link})]})]},l))})]}):void 0]})]}):I.jsx("div",{className:"text-center py-10",children:"Loading or not found..."})},$2=()=>I.jsxs(Vy,{basename:"/Sparql-fest",children:[I.jsx(ob,{}),I.jsxs(wy,{children:[I.jsx(Bn,{path:"/",element:I.jsx(vP,{})}),I.jsx(Bn,{path:"/sparql-queries",element:I.jsx(xP,{})}),I.jsx(Bn,{path:"sparql-queries/:slug",element:I.jsx(M2,{})}),I.jsx(Bn,{path:"/sparql-prefixes",element:I.jsx(V2,{})}),I.jsx(Bn,{path:"sparql-prefixes/:slug",element:I.jsx(Z2,{})}),I.jsx(Bn,{path:"/tutorials",element:I.jsx(Y2,{})}),I.jsx(Bn,{path:"/other-resources",element:I.jsx(K2,{})}),I.jsx(Bn,{path:"/other-resources/:slug",element:I.jsx(J2,{})})]}),I.jsx(rb,{})]});sI.createRoot(document.getElementById("root")).render(I.jsx(P.StrictMode,{children:I.jsx($2,{})}));
